<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>8.2 Adding Pseudo-scRNA profiles for each scATAC cell | ArchR</title>
  <meta name="description" content="A guide to ArchR" />
  <meta name="generator" content="bookdown 0.17 and GitBook 2.6.7" />

  <meta property="og:title" content="8.2 Adding Pseudo-scRNA profiles for each scATAC cell | ArchR" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="A guide to ArchR" />
  <meta name="github-repo" content="GreenleafLab/ArchR" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="8.2 Adding Pseudo-scRNA profiles for each scATAC cell | ArchR" />
  
  <meta name="twitter:description" content="A guide to ArchR" />
  

<meta name="author" content="Jeffrey Granja and Ryan Corces" />


<meta name="date" content="2020-02-18" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="labeling-scatac-cells-with-scrna-clusters.html"/>
<link rel="next" href="labeling-scatac-clusters-with-scrna.html"/>
<script src="libs/header-attrs-2.1/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="https://greenleaflab.github.io/ArchR/">Back to ArchR Home</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="chapter" data-level="1" data-path="getting-started-with-archr.html"><a href="getting-started-with-archr.html"><i class="fa fa-check"></i><b>1</b> Getting Started with ArchR</a>
<ul>
<li class="chapter" data-level="1.1" data-path="what-is-an-arrowfile-archrproject.html"><a href="what-is-an-arrowfile-archrproject.html"><i class="fa fa-check"></i><b>1.1</b> What is an <code>ArrowFile</code> / <code>ArchRProject</code>?</a></li>
<li class="chapter" data-level="1.2" data-path="getting-set-up.html"><a href="getting-set-up.html"><i class="fa fa-check"></i><b>1.2</b> Getting Set Up</a></li>
<li class="chapter" data-level="1.3" data-path="setting-a-genome-and-geneannotation.html"><a href="setting-a-genome-and-geneannotation.html"><i class="fa fa-check"></i><b>1.3</b> Setting a Genome and GeneAnnotation</a></li>
<li class="chapter" data-level="1.4" data-path="creating-arrow-files.html"><a href="creating-arrow-files.html"><i class="fa fa-check"></i><b>1.4</b> Creating Arrow Files</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="doublet-inference-with-archr.html"><a href="doublet-inference-with-archr.html"><i class="fa fa-check"></i><b>2</b> Doublet Inference with ArchR</a>
<ul>
<li class="chapter" data-level="2.1" data-path="inferring-scatac-seq-doublets-with-archr.html"><a href="inferring-scatac-seq-doublets-with-archr.html"><i class="fa fa-check"></i><b>2.1</b> Inferring scATAC-seq Doublets with ArchR</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="creating-an-archrproject.html"><a href="creating-an-archrproject.html"><i class="fa fa-check"></i><b>3</b> Creating an ArchRProject</a>
<ul>
<li class="chapter" data-level="3.1" data-path="creating-an-archrproject-1.html"><a href="creating-an-archrproject-1.html"><i class="fa fa-check"></i><b>3.1</b> Creating An ArchRProject</a></li>
<li class="chapter" data-level="3.2" data-path="manipulating-an-archrproject.html"><a href="manipulating-an-archrproject.html"><i class="fa fa-check"></i><b>3.2</b> Manipulating An ArchRProject</a></li>
<li class="chapter" data-level="3.3" data-path="plotting-sample-statistics-from-an-archrproject.html"><a href="plotting-sample-statistics-from-an-archrproject.html"><i class="fa fa-check"></i><b>3.3</b> Plotting Sample Statistics from an ArchRProject</a></li>
<li class="chapter" data-level="3.4" data-path="plotting-sample-fragment-size-distribution-and-tss-enrichment-profiles-.html"><a href="plotting-sample-fragment-size-distribution-and-tss-enrichment-profiles-.html"><i class="fa fa-check"></i><b>3.4</b> Plotting Sample Fragment Size Distribution and TSS Enrichment Profiles.</a></li>
<li class="chapter" data-level="3.5" data-path="filtering-doublets-from-an-archrproject.html"><a href="filtering-doublets-from-an-archrproject.html"><i class="fa fa-check"></i><b>3.5</b> Filtering Doublets from an ArchRProject</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="dimensionality-reduction-with-archr.html"><a href="dimensionality-reduction-with-archr.html"><i class="fa fa-check"></i><b>4</b> Dimensionality Reduction with ArchR</a>
<ul>
<li class="chapter" data-level="4.1" data-path="iterative-latent-semantic-indexing-lsi.html"><a href="iterative-latent-semantic-indexing-lsi.html"><i class="fa fa-check"></i><b>4.1</b> Iterative Latent Semantic Indexing (LSI)</a></li>
<li class="chapter" data-level="4.2" data-path="batch-effect-correction-wtih-harmony.html"><a href="batch-effect-correction-wtih-harmony.html"><i class="fa fa-check"></i><b>4.2</b> Batch Effect Correction wtih Harmony</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="clustering-with-archr.html"><a href="clustering-with-archr.html"><i class="fa fa-check"></i><b>5</b> Clustering with ArchR</a></li>
<li class="chapter" data-level="6" data-path="single-cell-embeddings.html"><a href="single-cell-embeddings.html"><i class="fa fa-check"></i><b>6</b> Single Cell Embeddings</a>
<ul>
<li class="chapter" data-level="6.1" data-path="umap.html"><a href="umap.html"><i class="fa fa-check"></i><b>6.1</b> UMAP</a></li>
<li class="chapter" data-level="6.2" data-path="tsne.html"><a href="tsne.html"><i class="fa fa-check"></i><b>6.2</b> tSNE</a></li>
<li class="chapter" data-level="6.3" data-path="umap-of-previous-reduced-dimensions.html"><a href="umap-of-previous-reduced-dimensions.html"><i class="fa fa-check"></i><b>6.3</b> UMAP of previous Reduced Dimensions</a></li>
<li class="chapter" data-level="6.4" data-path="tsne-of-previous-reduced-dimensions.html"><a href="tsne-of-previous-reduced-dimensions.html"><i class="fa fa-check"></i><b>6.4</b> tSNE of previous Reduced Dimensions</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="marker-genes-with-scatac.html"><a href="marker-genes-with-scatac.html"><i class="fa fa-check"></i><b>7</b> Marker Genes with scATAC</a>
<ul>
<li class="chapter" data-level="7.1" data-path="identification-of-marker-genes.html"><a href="identification-of-marker-genes.html"><i class="fa fa-check"></i><b>7.1</b> Identification of Marker Genes</a></li>
<li class="chapter" data-level="7.2" data-path="marker-genes.html"><a href="marker-genes.html"><i class="fa fa-check"></i><b>7.2</b> Marker Genes</a></li>
<li class="chapter" data-level="7.3" data-path="marker-genes-imputation-with-magic.html"><a href="marker-genes-imputation-with-magic.html"><i class="fa fa-check"></i><b>7.3</b> Marker Genes Imputation with Magic</a></li>
<li class="chapter" data-level="7.4" data-path="track-plotting-with-archrbrowser.html"><a href="track-plotting-with-archrbrowser.html"><i class="fa fa-check"></i><b>7.4</b> Track Plotting with ArchRBrowser</a></li>
<li class="chapter" data-level="7.5" data-path="launching-the-archrbrowser.html"><a href="launching-the-archrbrowser.html"><i class="fa fa-check"></i><b>7.5</b> Launching the ArchRBrowser</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="labeling-clusters-with-scrna.html"><a href="labeling-clusters-with-scrna.html"><i class="fa fa-check"></i><b>8</b> Labeling Clusters with scRNA</a>
<ul>
<li class="chapter" data-level="8.1" data-path="labeling-scatac-cells-with-scrna-clusters.html"><a href="labeling-scatac-cells-with-scrna-clusters.html"><i class="fa fa-check"></i><b>8.1</b> Labeling scATAC cells with scRNA clusters</a></li>
<li class="chapter" data-level="8.2" data-path="adding-pseudo-scrna-profiles-for-each-scatac-cell.html"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html"><i class="fa fa-check"></i><b>8.2</b> Adding Pseudo-scRNA profiles for each scATAC cell</a></li>
<li class="chapter" data-level="8.3" data-path="labeling-scatac-clusters-with-scrna.html"><a href="labeling-scatac-clusters-with-scrna.html"><i class="fa fa-check"></i><b>8.3</b> Labeling scATAC clusters with scRNA</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="calling-peaks-with-archr.html"><a href="calling-peaks-with-archr.html"><i class="fa fa-check"></i><b>9</b> Calling Peaks with ArchR</a>
<ul>
<li class="chapter" data-level="9.1" data-path="calling-peaks-w-macs2.html"><a href="calling-peaks-w-macs2.html"><i class="fa fa-check"></i><b>9.1</b> Calling Peaks w/ Macs2</a></li>
<li class="chapter" data-level="9.2" data-path="calling-peaks-w-tilematrix.html"><a href="calling-peaks-w-tilematrix.html"><i class="fa fa-check"></i><b>9.2</b> Calling Peaks w/ TileMatrix</a></li>
<li class="chapter" data-level="9.3" data-path="add-peak-matrix.html"><a href="add-peak-matrix.html"><i class="fa fa-check"></i><b>9.3</b> Add Peak Matrix</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="identifying-marker-peaks-with-archr.html"><a href="identifying-marker-peaks-with-archr.html"><i class="fa fa-check"></i><b>10</b> Identifying Marker Peaks with ArchR</a>
<ul>
<li class="chapter" data-level="10.1" data-path="identifying-marker-peaks-with-archr-1.html"><a href="identifying-marker-peaks-with-archr-1.html"><i class="fa fa-check"></i><b>10.1</b> Identifying Marker Peaks with ArchR</a></li>
<li class="chapter" data-level="10.2" data-path="pairwise-testing-groups.html"><a href="pairwise-testing-groups.html"><i class="fa fa-check"></i><b>10.2</b> Pairwise Testing Groups</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="peak-annotation-enrichment-with-archr.html"><a href="peak-annotation-enrichment-with-archr.html"><i class="fa fa-check"></i><b>11</b> Peak Annotation Enrichment with ArchR</a>
<ul>
<li class="chapter" data-level="11.1" data-path="motif-enrichment.html"><a href="motif-enrichment.html"><i class="fa fa-check"></i><b>11.1</b> Motif Enrichment</a></li>
<li class="chapter" data-level="11.2" data-path="archr-enrichment.html"><a href="archr-enrichment.html"><i class="fa fa-check"></i><b>11.2</b> ArchR Enrichment</a>
<ul>
<li class="chapter" data-level="11.2.1" data-path="archr-enrichment.html"><a href="archr-enrichment.html#encode-tfbs"><i class="fa fa-check"></i><b>11.2.1</b> Encode TFBS</a></li>
<li class="chapter" data-level="11.2.2" data-path="archr-enrichment.html"><a href="archr-enrichment.html#bulk-atac-seq"><i class="fa fa-check"></i><b>11.2.2</b> Bulk ATAC-seq</a></li>
<li class="chapter" data-level="11.2.3" data-path="archr-enrichment.html"><a href="archr-enrichment.html#codex-tfbs"><i class="fa fa-check"></i><b>11.2.3</b> Codex TFBS</a></li>
</ul></li>
<li class="chapter" data-level="11.3" data-path="custom-enrichment.html"><a href="custom-enrichment.html"><i class="fa fa-check"></i><b>11.3</b> Custom Enrichment</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="chromvar-deviatons-enrichment-with-archr.html"><a href="chromvar-deviatons-enrichment-with-archr.html"><i class="fa fa-check"></i><b>12</b> ChromVAR Deviatons Enrichment with ArchR</a>
<ul>
<li class="chapter" data-level="12.1" data-path="motif-deviations.html"><a href="motif-deviations.html"><i class="fa fa-check"></i><b>12.1</b> Motif Deviations</a></li>
<li class="chapter" data-level="12.2" data-path="archr-deviations.html"><a href="archr-deviations.html"><i class="fa fa-check"></i><b>12.2</b> ArchR Deviations</a>
<ul>
<li class="chapter" data-level="12.2.1" data-path="archr-deviations.html"><a href="archr-deviations.html#encode-tfbs-1"><i class="fa fa-check"></i><b>12.2.1</b> Encode TFBS</a></li>
<li class="chapter" data-level="12.2.2" data-path="archr-deviations.html"><a href="archr-deviations.html#bulk-atac-seq-1"><i class="fa fa-check"></i><b>12.2.2</b> Bulk ATAC-seq</a></li>
</ul></li>
<li class="chapter" data-level="12.3" data-path="custom-deviations.html"><a href="custom-deviations.html"><i class="fa fa-check"></i><b>12.3</b> Custom Deviations</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="footprinting-with-archr.html"><a href="footprinting-with-archr.html"><i class="fa fa-check"></i><b>13</b> Footprinting with ArchR</a>
<ul>
<li class="chapter" data-level="13.1" data-path="motif-footprinting.html"><a href="motif-footprinting.html"><i class="fa fa-check"></i><b>13.1</b> Motif Footprinting</a>
<ul>
<li class="chapter" data-level="13.1.1" data-path="motif-footprinting.html"><a href="motif-footprinting.html#no-normalization"><i class="fa fa-check"></i><b>13.1.1</b> No Normalization</a></li>
<li class="chapter" data-level="13.1.2" data-path="motif-footprinting.html"><a href="motif-footprinting.html#divide-tn5-bias"><i class="fa fa-check"></i><b>13.1.2</b> Divide Tn5 Bias</a></li>
<li class="chapter" data-level="13.1.3" data-path="motif-footprinting.html"><a href="motif-footprinting.html#subtract-tn5-bias"><i class="fa fa-check"></i><b>13.1.3</b> Subtract Tn5 Bias</a></li>
</ul></li>
<li class="chapter" data-level="13.2" data-path="feature-footprinting.html"><a href="feature-footprinting.html"><i class="fa fa-check"></i><b>13.2</b> Feature Footprinting</a>
<ul>
<li class="chapter" data-level="13.2.1" data-path="feature-footprinting.html"><a href="feature-footprinting.html#no-normalization-1"><i class="fa fa-check"></i><b>13.2.1</b> No Normalization</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="14" data-path="trajectory-analysis-with-archr.html"><a href="trajectory-analysis-with-archr.html"><i class="fa fa-check"></i><b>14</b> Trajectory Analysis with ArchR</a>
<ul>
<li class="chapter" data-level="14.1" data-path="myeloid-trajectory.html"><a href="myeloid-trajectory.html"><i class="fa fa-check"></i><b>14.1</b> Myeloid Trajectory</a></li>
<li class="chapter" data-level="14.2" data-path="lymphoid-trajectory.html"><a href="lymphoid-trajectory.html"><i class="fa fa-check"></i><b>14.2</b> Lymphoid Trajectory</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="peak-linkage-with-archr.html"><a href="peak-linkage-with-archr.html"><i class="fa fa-check"></i><b>15</b> Peak Linkage with ArchR</a>
<ul>
<li class="chapter" data-level="15.1" data-path="coaccessibility-with-archr.html"><a href="coaccessibility-with-archr.html"><i class="fa fa-check"></i><b>15.1</b> CoAccessibility with ArchR</a></li>
<li class="chapter" data-level="15.2" data-path="peak2genelinkage-with-archr.html"><a href="peak2genelinkage-with-archr.html"><i class="fa fa-check"></i><b>15.2</b> Peak2GeneLinkage with ArchR</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://bookdown.org" target="_blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">ArchR</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="adding-pseudo-scrna-profiles-for-each-scatac-cell" class="section level2" number="8.2">
<h2><span class="header-section-number">8.2</span> Adding Pseudo-scRNA profiles for each scATAC cell</h2>
<p>First we need to download scRNA data for hematopoiesis from Granja* et al (2019).</p>
<div class="sourceCode" id="cb84"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb84-1"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb84-1"></a><span class="co">#259 MB Download</span></span>
<span id="cb84-2"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb84-2"></a><span class="cf">if</span>(<span class="op">!</span><span class="kw">file.exists</span>(<span class="st">&quot;scRNA-Healthy-Hematopoiesis-191120.rds&quot;</span>)){</span>
<span id="cb84-3"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb84-3"></a>    <span class="kw">download.file</span>(</span>
<span id="cb84-4"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb84-4"></a>        <span class="dt">url =</span> <span class="st">&quot;https://jeffgranja.s3.amazonaws.com/MPAL-10x/Supplementary_Data/Healthy-Data/scRNA-Healthy-Hematopoiesis-191120.rds&quot;</span>,</span>
<span id="cb84-5"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb84-5"></a>        <span class="dt">destfile =</span> <span class="st">&quot;scRNA-Healthy-Hematopoiesis-191120.rds&quot;</span></span>
<span id="cb84-6"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb84-6"></a>    )</span>
<span id="cb84-7"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb84-7"></a>}</span>
<span id="cb84-8"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb84-8"></a></span>
<span id="cb84-9"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb84-9"></a>seRNA &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="st">&quot;scRNA-Healthy-Hematopoiesis-191120.rds&quot;</span>)</span>
<span id="cb84-10"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb84-10"></a>seRNA</span>
<span id="cb84-11"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb84-11"></a><span class="co"># class: RangedSummarizedExperiment </span></span>
<span id="cb84-12"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb84-12"></a><span class="co"># dim: 20287 35582 </span></span>
<span id="cb84-13"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb84-13"></a><span class="co"># metadata(6): variableGenes optimizeLSI ... UMAP_Params colorMap</span></span>
<span id="cb84-14"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb84-14"></a><span class="co"># assays(1): counts</span></span>
<span id="cb84-15"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb84-15"></a><span class="co"># rownames(20287): FAM138A OR4F5 ... S100B PRMT2</span></span>
<span id="cb84-16"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb84-16"></a><span class="co"># rowData names(3): gene_name gene_id exonLength</span></span>
<span id="cb84-17"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb84-17"></a><span class="co"># colnames(35582): CD34_32_R5:AAACCTGAGTATCGAA-1</span></span>
<span id="cb84-18"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb84-18"></a><span class="co">#   CD34_32_R5:AAACCTGAGTCGTTTG-1 ...</span></span>
<span id="cb84-19"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb84-19"></a><span class="co">#   BMMC_10x_GREENLEAF_REP2:TTTGTTGCATGTGTCA-1</span></span>
<span id="cb84-20"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb84-20"></a><span class="co">#   BMMC_10x_GREENLEAF_REP2:TTTGTTGCATTGAAAG-1</span></span>
<span id="cb84-21"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb84-21"></a><span class="co"># colData names(10): Group nUMI_pre ... BioClassification Barcode</span></span>
<span id="cb84-22"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb84-22"></a></span>
<span id="cb84-23"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb84-23"></a><span class="kw">colnames</span>(<span class="kw">colData</span>(seRNA))</span>
<span id="cb84-24"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb84-24"></a><span class="co"># [1] &quot;Group&quot;             &quot;nUMI_pre&quot;          &quot;nUMI&quot;             </span></span>
<span id="cb84-25"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb84-25"></a><span class="co"># [4] &quot;nGene&quot;             &quot;initialClusters&quot;   &quot;UMAP1&quot;            </span></span>
<span id="cb84-26"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb84-26"></a><span class="co"># [7] &quot;UMAP2&quot;             &quot;Clusters&quot;          &quot;BioClassification&quot;</span></span>
<span id="cb84-27"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb84-27"></a><span class="co"># [10] &quot;Barcode&quot;          </span></span>
<span id="cb84-28"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb84-28"></a></span>
<span id="cb84-29"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb84-29"></a><span class="kw">table</span>(<span class="kw">colData</span>(seRNA)<span class="op">$</span>BioClassification)</span>
<span id="cb84-30"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb84-30"></a><span class="co">#         01_HSC 02_Early.Eryth  03_Late.Eryth  04_Early.Baso    05_CMP.LMPP </span></span>
<span id="cb84-31"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb84-31"></a><span class="co">#           1425           1653            446            111           2260 </span></span>
<span id="cb84-32"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb84-32"></a><span class="co">#       06_CLP.1         07_GMP    08_GMP.Neut         09_pDC         10_cDC </span></span>
<span id="cb84-33"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb84-33"></a><span class="co">#            903           2097           1050            544            325 </span></span>
<span id="cb84-34"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb84-34"></a><span class="co"># 11_CD14.Mono.1 12_CD14.Mono.2   13_CD16.Mono         14_Unk       15_CLP.2 </span></span>
<span id="cb84-35"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb84-35"></a><span class="co">#           1800           4222            292            520            377 </span></span>
<span id="cb84-36"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb84-36"></a><span class="co">#       16_Pre.B           17_B      18_Plasma       19_CD8.N      20_CD4.N1 </span></span>
<span id="cb84-37"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb84-37"></a><span class="co">#            710           1711             62           1521           2470 </span></span>
<span id="cb84-38"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb84-38"></a><span class="co">#      21_CD4.N2       22_CD4.M      23_CD8.EM      24_CD8.CM          25_NK </span></span>
<span id="cb84-39"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb84-39"></a><span class="co">#           2364           3539            796           2080           2143 </span></span>
<span id="cb84-40"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb84-40"></a><span class="co">#         26_Unk </span></span>
<span id="cb84-41"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb84-41"></a><span class="co">#            161 </span></span></code></pre></div>
<p>We can now save our original projHeme2 using <code>saveArchRProject</code> from ArchR.</p>
<div class="sourceCode" id="cb85"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb85-1"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb85-1"></a><span class="kw">saveArchRProject</span>(projHeme2)</span></code></pre></div>
<p>Constrained Integration (recommended)</p>
<div class="sourceCode" id="cb86"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb86-1"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb86-1"></a>sampleList &lt;-<span class="st"> </span><span class="kw">SimpleList</span>(</span>
<span id="cb86-2"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb86-2"></a>    <span class="dt">scATAC_BMMC_R1 =</span> <span class="kw">grep</span>(<span class="st">&quot;BMMC&quot;</span>, <span class="kw">colnames</span>(seRNA), <span class="dt">ignore.case =</span> <span class="ot">TRUE</span>, <span class="dt">value =</span> <span class="ot">TRUE</span>),</span>
<span id="cb86-3"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb86-3"></a>    <span class="dt">scATAC_CD34_BMMC_R1 =</span> <span class="kw">grep</span>(<span class="st">&quot;CD34&quot;</span>, <span class="kw">colnames</span>(seRNA), <span class="dt">ignore.case =</span> <span class="ot">TRUE</span>, <span class="dt">value =</span> <span class="ot">TRUE</span>),</span>
<span id="cb86-4"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb86-4"></a>    <span class="dt">scATAC_PBMC_R1 =</span> <span class="kw">grep</span>(<span class="st">&quot;PBMC&quot;</span>, <span class="kw">colnames</span>(seRNA), <span class="dt">ignore.case =</span> <span class="ot">TRUE</span>, <span class="dt">value =</span> <span class="ot">TRUE</span>)</span>
<span id="cb86-5"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb86-5"></a>)</span>
<span id="cb86-6"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb86-6"></a></span>
<span id="cb86-7"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb86-7"></a><span class="co">#To see size of each list element</span></span>
<span id="cb86-8"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb86-8"></a><span class="kw">lapply</span>(sampleList, length)</span>
<span id="cb86-9"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb86-9"></a><span class="co"># $scATAC_BMMC_R1</span></span>
<span id="cb86-10"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb86-10"></a><span class="co"># [1] 12602</span></span>
<span id="cb86-11"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb86-11"></a></span>
<span id="cb86-12"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb86-12"></a><span class="co"># $scATAC_CD34_BMMC_R1</span></span>
<span id="cb86-13"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb86-13"></a><span class="co"># [1] 8176</span></span>
<span id="cb86-14"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb86-14"></a></span>
<span id="cb86-15"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb86-15"></a><span class="co"># $scATAC_PBMC_R1</span></span>
<span id="cb86-16"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb86-16"></a><span class="co"># [1] 14804</span></span>
<span id="cb86-17"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb86-17"></a></span>
<span id="cb86-18"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb86-18"></a><span class="co">#~5 minutes</span></span>
<span id="cb86-19"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb86-19"></a>projHeme3 &lt;-<span class="st"> </span><span class="kw">addGeneIntegrationMatrix</span>(</span>
<span id="cb86-20"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb86-20"></a>    <span class="dt">ArchRProj =</span> projHeme2, </span>
<span id="cb86-21"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb86-21"></a>    <span class="dt">seRNA =</span> seRNA, </span>
<span id="cb86-22"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb86-22"></a>    <span class="dt">sampleList =</span> sampleList,</span>
<span id="cb86-23"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb86-23"></a>    <span class="dt">addToArrow =</span> <span class="ot">TRUE</span>, <span class="co"># We are going to add scRNA to each Arrow</span></span>
<span id="cb86-24"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb86-24"></a>    <span class="dt">groupBy =</span> <span class="st">&quot;BioClassification&quot;</span>,</span>
<span id="cb86-25"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb86-25"></a>    <span class="dt">nameGroup =</span> <span class="st">&quot;predictedGroup&quot;</span>,</span>
<span id="cb86-26"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb86-26"></a>    <span class="dt">nameScore =</span> <span class="st">&quot;predictedScore&quot;</span></span>
<span id="cb86-27"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb86-27"></a>)</span></code></pre></div>
<p>What matrices are available?</p>
<div class="sourceCode" id="cb87"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb87-1"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb87-1"></a><span class="kw">getAvailableMatrices</span>(projHeme3)</span>
<span id="cb87-2"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb87-2"></a><span class="co">#[1] &quot;GeneIntegrationMatrix&quot; &quot;GeneScoreMatrix&quot;       &quot;TileMatrix&quot;</span></span></code></pre></div>
<p>Now lets see how this effects our marker gene scores overlayed on our 2-d embedding.</p>
<div class="sourceCode" id="cb88"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb88-1"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb88-1"></a>markerGenes  &lt;-<span class="st"> </span><span class="kw">c</span>(</span>
<span id="cb88-2"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb88-2"></a>    <span class="st">&quot;CD34&quot;</span>, <span class="co">#Early Progenitor</span></span>
<span id="cb88-3"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb88-3"></a>    <span class="st">&quot;GATA1&quot;</span>, <span class="co">#Erythroid</span></span>
<span id="cb88-4"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb88-4"></a>    <span class="st">&quot;PAX5&quot;</span>, <span class="st">&quot;MS4A1&quot;</span>, <span class="co">#B-Cell Trajectory</span></span>
<span id="cb88-5"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb88-5"></a>    <span class="st">&quot;CD14&quot;</span>, <span class="co">#Monocytes</span></span>
<span id="cb88-6"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb88-6"></a>    <span class="st">&quot;CD3D&quot;</span>, <span class="st">&quot;CD8A&quot;</span>, <span class="st">&quot;TBX21&quot;</span>, <span class="st">&quot;IL7R&quot;</span> <span class="co">#TCells</span></span>
<span id="cb88-7"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb88-7"></a>  )</span>
<span id="cb88-8"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb88-8"></a></span>
<span id="cb88-9"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb88-9"></a>p1 &lt;-<span class="st"> </span><span class="kw">plotEmbedding</span>(</span>
<span id="cb88-10"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb88-10"></a>    <span class="dt">ArchRProj =</span> projHeme3, </span>
<span id="cb88-11"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb88-11"></a>    <span class="dt">colorBy =</span> <span class="st">&quot;GeneIntegrationMatrix&quot;</span>, </span>
<span id="cb88-12"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb88-12"></a>    <span class="dt">name =</span> markerGenes, </span>
<span id="cb88-13"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb88-13"></a>    <span class="dt">continuousSet =</span> <span class="st">&quot;horizonExtra&quot;</span>,</span>
<span id="cb88-14"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb88-14"></a>    <span class="dt">embedding =</span> <span class="st">&quot;UMAP&quot;</span>,</span>
<span id="cb88-15"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb88-15"></a>    <span class="dt">imputeWeights =</span> <span class="kw">getImputeWeights</span>(projHeme3)</span>
<span id="cb88-16"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb88-16"></a>)</span>
<span id="cb88-17"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb88-17"></a></span>
<span id="cb88-18"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb88-18"></a>p2 &lt;-<span class="st"> </span><span class="kw">plotEmbedding</span>(</span>
<span id="cb88-19"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb88-19"></a>    <span class="dt">ArchRProj =</span> projHeme3, </span>
<span id="cb88-20"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb88-20"></a>    <span class="dt">colorBy =</span> <span class="st">&quot;GeneScoreMatrix&quot;</span>, </span>
<span id="cb88-21"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb88-21"></a>    <span class="dt">continuousSet =</span> <span class="st">&quot;horizonExtra&quot;</span>,</span>
<span id="cb88-22"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb88-22"></a>    <span class="dt">name =</span> markerGenes, </span>
<span id="cb88-23"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb88-23"></a>    <span class="dt">embedding =</span> <span class="st">&quot;UMAP&quot;</span>,</span>
<span id="cb88-24"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb88-24"></a>    <span class="dt">imputeWeights =</span> <span class="kw">getImputeWeights</span>(projHeme3)</span>
<span id="cb88-25"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb88-25"></a>)</span></code></pre></div>
<p>To plot all marker genes we can use cowplot</p>
<div class="sourceCode" id="cb89"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb89-1"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb89-1"></a><span class="co">#Rearrange for grid plotting</span></span>
<span id="cb89-2"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb89-2"></a>p1c &lt;-<span class="st"> </span><span class="kw">lapply</span>(p1, <span class="cf">function</span>(x){</span>
<span id="cb89-3"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb89-3"></a>    x <span class="op">+</span><span class="st"> </span><span class="kw">guides</span>(<span class="dt">color =</span> <span class="ot">FALSE</span>, <span class="dt">fill =</span> <span class="ot">FALSE</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb89-4"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb89-4"></a><span class="st">    </span><span class="kw">theme_ArchR</span>(<span class="dt">baseSize =</span> <span class="fl">6.5</span>) <span class="op">+</span></span>
<span id="cb89-5"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb89-5"></a><span class="st">    </span><span class="kw">theme</span>(<span class="dt">plot.margin =</span> <span class="kw">unit</span>(<span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>), <span class="st">&quot;cm&quot;</span>)) <span class="op">+</span></span>
<span id="cb89-6"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb89-6"></a><span class="st">    </span><span class="kw">theme</span>(</span>
<span id="cb89-7"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb89-7"></a>        <span class="dt">axis.text.x=</span><span class="kw">element_blank</span>(), </span>
<span id="cb89-8"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb89-8"></a>        <span class="dt">axis.ticks.x=</span><span class="kw">element_blank</span>(), </span>
<span id="cb89-9"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb89-9"></a>        <span class="dt">axis.text.y=</span><span class="kw">element_blank</span>(), </span>
<span id="cb89-10"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb89-10"></a>        <span class="dt">axis.ticks.y=</span><span class="kw">element_blank</span>()</span>
<span id="cb89-11"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb89-11"></a>    )</span>
<span id="cb89-12"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb89-12"></a>})</span>
<span id="cb89-13"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb89-13"></a></span>
<span id="cb89-14"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb89-14"></a><span class="co">#Rearrange for grid plotting</span></span>
<span id="cb89-15"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb89-15"></a>p2c &lt;-<span class="st"> </span><span class="kw">lapply</span>(p2, <span class="cf">function</span>(x){</span>
<span id="cb89-16"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb89-16"></a>    x <span class="op">+</span><span class="st"> </span><span class="kw">guides</span>(<span class="dt">color =</span> <span class="ot">FALSE</span>, <span class="dt">fill =</span> <span class="ot">FALSE</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb89-17"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb89-17"></a><span class="st">    </span><span class="kw">theme_ArchR</span>(<span class="dt">baseSize =</span> <span class="fl">6.5</span>) <span class="op">+</span></span>
<span id="cb89-18"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb89-18"></a><span class="st">    </span><span class="kw">theme</span>(<span class="dt">plot.margin =</span> <span class="kw">unit</span>(<span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>), <span class="st">&quot;cm&quot;</span>)) <span class="op">+</span></span>
<span id="cb89-19"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb89-19"></a><span class="st">    </span><span class="kw">theme</span>(</span>
<span id="cb89-20"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb89-20"></a>        <span class="dt">axis.text.x=</span><span class="kw">element_blank</span>(), </span>
<span id="cb89-21"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb89-21"></a>        <span class="dt">axis.ticks.x=</span><span class="kw">element_blank</span>(), </span>
<span id="cb89-22"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb89-22"></a>        <span class="dt">axis.text.y=</span><span class="kw">element_blank</span>(), </span>
<span id="cb89-23"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb89-23"></a>        <span class="dt">axis.ticks.y=</span><span class="kw">element_blank</span>()</span>
<span id="cb89-24"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb89-24"></a>    )</span>
<span id="cb89-25"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb89-25"></a>})</span>
<span id="cb89-26"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb89-26"></a></span>
<span id="cb89-27"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb89-27"></a><span class="kw">do.call</span>(cowplot<span class="op">::</span>plot_grid, <span class="kw">c</span>(<span class="kw">list</span>(<span class="dt">ncol =</span> <span class="dv">3</span>), p1c))</span></code></pre></div>
<p><img src="images/HemeWalkthrough/PNG/Plot-UMAP-Markers-RNA-W-Imputation_1.png" width="800" height="800" /></p>
<div class="sourceCode" id="cb90"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb90-1"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb90-1"></a><span class="kw">do.call</span>(cowplot<span class="op">::</span>plot_grid, <span class="kw">c</span>(<span class="kw">list</span>(<span class="dt">ncol =</span> <span class="dv">3</span>), p2c))</span></code></pre></div>
<p><img src="images/HemeWalkthrough/PNG/Plot-UMAP-Markers-RNA-W-Imputation_2.png" width="800" height="800" /></p>
<p>To save a nice looking pdf we use plotPDF which removes white pages and tries to make the plots
nice looking.</p>
<div class="sourceCode" id="cb91"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb91-1"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb91-1"></a><span class="kw">plotPDF</span>(<span class="dt">plotList =</span> p1, </span>
<span id="cb91-2"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb91-2"></a>    <span class="dt">name =</span> <span class="st">&quot;Plot-UMAP-Marker-Genes-RNA-W-Imputation.pdf&quot;</span>, </span>
<span id="cb91-3"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb91-3"></a>    <span class="dt">ArchRProj =</span> projHeme3, </span>
<span id="cb91-4"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html#cb91-4"></a>    <span class="dt">addDOC =</span> <span class="ot">FALSE</span>, <span class="dt">width =</span> <span class="dv">5</span>, <span class="dt">height =</span> <span class="dv">5</span>)</span></code></pre></div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="labeling-scatac-cells-with-scrna-clusters.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="labeling-scatac-clusters-with-scrna.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/GreenleafLab/ArchR_Website/edit/dev/08_scRNA_Labeling.Rmd",
"text": "Edit this chapter"
},
"history": {
"link": "https://github.com/GreenleafLab/ArchR_Website/commits/dev/08_scRNA_Labeling.Rmd",
"text": "Chapter edit history"
},
"view": {
"link": "https://github.com/GreenleafLab/ArchR_Website/view/dev/08_scRNA_Labeling.Rmd",
"text": "View source on GitHub"
},
"download": ["ArchR.pdf"],
"toc": {
"collapse": "none"
}
});
});
</script>

</body>

</html>
