<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>10.1 Identifying Marker Peaks with ArchR | ArchR</title>
  <meta name="description" content="A guide to ArchR" />
  <meta name="generator" content="bookdown 0.17 and GitBook 2.6.7" />

  <meta property="og:title" content="10.1 Identifying Marker Peaks with ArchR | ArchR" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="A guide to ArchR" />
  <meta name="github-repo" content="GreenleafLab/ArchR" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="10.1 Identifying Marker Peaks with ArchR | ArchR" />
  
  <meta name="twitter:description" content="A guide to ArchR" />
  

<meta name="author" content="Jeffrey Granja and Ryan Corces" />


<meta name="date" content="2020-02-18" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="identifying-marker-peaks-with-archr.html"/>
<link rel="next" href="pairwise-testing-groups.html"/>
<script src="libs/header-attrs-2.1/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="https://greenleaflab.github.io/ArchR/">Back to ArchR Home</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="chapter" data-level="1" data-path="getting-started-with-archr.html"><a href="getting-started-with-archr.html"><i class="fa fa-check"></i><b>1</b> Getting Started with ArchR</a>
<ul>
<li class="chapter" data-level="1.1" data-path="what-is-an-arrowfile-archrproject.html"><a href="what-is-an-arrowfile-archrproject.html"><i class="fa fa-check"></i><b>1.1</b> What is an <code>ArrowFile</code> / <code>ArchRProject</code>?</a></li>
<li class="chapter" data-level="1.2" data-path="getting-set-up.html"><a href="getting-set-up.html"><i class="fa fa-check"></i><b>1.2</b> Getting Set Up</a></li>
<li class="chapter" data-level="1.3" data-path="setting-a-genome-and-geneannotation.html"><a href="setting-a-genome-and-geneannotation.html"><i class="fa fa-check"></i><b>1.3</b> Setting a Genome and GeneAnnotation</a></li>
<li class="chapter" data-level="1.4" data-path="creating-arrow-files.html"><a href="creating-arrow-files.html"><i class="fa fa-check"></i><b>1.4</b> Creating Arrow Files</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="doublet-inference-with-archr.html"><a href="doublet-inference-with-archr.html"><i class="fa fa-check"></i><b>2</b> Doublet Inference with ArchR</a>
<ul>
<li class="chapter" data-level="2.1" data-path="inferring-scatac-seq-doublets-with-archr.html"><a href="inferring-scatac-seq-doublets-with-archr.html"><i class="fa fa-check"></i><b>2.1</b> Inferring scATAC-seq Doublets with ArchR</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="creating-an-archrproject.html"><a href="creating-an-archrproject.html"><i class="fa fa-check"></i><b>3</b> Creating an ArchRProject</a>
<ul>
<li class="chapter" data-level="3.1" data-path="creating-an-archrproject-1.html"><a href="creating-an-archrproject-1.html"><i class="fa fa-check"></i><b>3.1</b> Creating An ArchRProject</a></li>
<li class="chapter" data-level="3.2" data-path="manipulating-an-archrproject.html"><a href="manipulating-an-archrproject.html"><i class="fa fa-check"></i><b>3.2</b> Manipulating An ArchRProject</a></li>
<li class="chapter" data-level="3.3" data-path="plotting-sample-statistics-from-an-archrproject.html"><a href="plotting-sample-statistics-from-an-archrproject.html"><i class="fa fa-check"></i><b>3.3</b> Plotting Sample Statistics from an ArchRProject</a></li>
<li class="chapter" data-level="3.4" data-path="plotting-sample-fragment-size-distribution-and-tss-enrichment-profiles-.html"><a href="plotting-sample-fragment-size-distribution-and-tss-enrichment-profiles-.html"><i class="fa fa-check"></i><b>3.4</b> Plotting Sample Fragment Size Distribution and TSS Enrichment Profiles.</a></li>
<li class="chapter" data-level="3.5" data-path="filtering-doublets-from-an-archrproject.html"><a href="filtering-doublets-from-an-archrproject.html"><i class="fa fa-check"></i><b>3.5</b> Filtering Doublets from an ArchRProject</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="dimensionality-reduction-with-archr.html"><a href="dimensionality-reduction-with-archr.html"><i class="fa fa-check"></i><b>4</b> Dimensionality Reduction with ArchR</a>
<ul>
<li class="chapter" data-level="4.1" data-path="iterative-latent-semantic-indexing-lsi.html"><a href="iterative-latent-semantic-indexing-lsi.html"><i class="fa fa-check"></i><b>4.1</b> Iterative Latent Semantic Indexing (LSI)</a></li>
<li class="chapter" data-level="4.2" data-path="batch-effect-correction-wtih-harmony.html"><a href="batch-effect-correction-wtih-harmony.html"><i class="fa fa-check"></i><b>4.2</b> Batch Effect Correction wtih Harmony</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="clustering-with-archr.html"><a href="clustering-with-archr.html"><i class="fa fa-check"></i><b>5</b> Clustering with ArchR</a></li>
<li class="chapter" data-level="6" data-path="single-cell-embeddings.html"><a href="single-cell-embeddings.html"><i class="fa fa-check"></i><b>6</b> Single Cell Embeddings</a>
<ul>
<li class="chapter" data-level="6.1" data-path="umap.html"><a href="umap.html"><i class="fa fa-check"></i><b>6.1</b> UMAP</a></li>
<li class="chapter" data-level="6.2" data-path="tsne.html"><a href="tsne.html"><i class="fa fa-check"></i><b>6.2</b> tSNE</a></li>
<li class="chapter" data-level="6.3" data-path="umap-of-previous-reduced-dimensions.html"><a href="umap-of-previous-reduced-dimensions.html"><i class="fa fa-check"></i><b>6.3</b> UMAP of previous Reduced Dimensions</a></li>
<li class="chapter" data-level="6.4" data-path="tsne-of-previous-reduced-dimensions.html"><a href="tsne-of-previous-reduced-dimensions.html"><i class="fa fa-check"></i><b>6.4</b> tSNE of previous Reduced Dimensions</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="marker-genes-with-scatac.html"><a href="marker-genes-with-scatac.html"><i class="fa fa-check"></i><b>7</b> Marker Genes with scATAC</a>
<ul>
<li class="chapter" data-level="7.1" data-path="identification-of-marker-genes.html"><a href="identification-of-marker-genes.html"><i class="fa fa-check"></i><b>7.1</b> Identification of Marker Genes</a></li>
<li class="chapter" data-level="7.2" data-path="marker-genes.html"><a href="marker-genes.html"><i class="fa fa-check"></i><b>7.2</b> Marker Genes</a></li>
<li class="chapter" data-level="7.3" data-path="marker-genes-imputation-with-magic.html"><a href="marker-genes-imputation-with-magic.html"><i class="fa fa-check"></i><b>7.3</b> Marker Genes Imputation with Magic</a></li>
<li class="chapter" data-level="7.4" data-path="track-plotting-with-archrbrowser.html"><a href="track-plotting-with-archrbrowser.html"><i class="fa fa-check"></i><b>7.4</b> Track Plotting with ArchRBrowser</a></li>
<li class="chapter" data-level="7.5" data-path="launching-the-archrbrowser.html"><a href="launching-the-archrbrowser.html"><i class="fa fa-check"></i><b>7.5</b> Launching the ArchRBrowser</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="labeling-clusters-with-scrna.html"><a href="labeling-clusters-with-scrna.html"><i class="fa fa-check"></i><b>8</b> Labeling Clusters with scRNA</a>
<ul>
<li class="chapter" data-level="8.1" data-path="labeling-scatac-cells-with-scrna-clusters.html"><a href="labeling-scatac-cells-with-scrna-clusters.html"><i class="fa fa-check"></i><b>8.1</b> Labeling scATAC cells with scRNA clusters</a></li>
<li class="chapter" data-level="8.2" data-path="adding-pseudo-scrna-profiles-for-each-scatac-cell.html"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html"><i class="fa fa-check"></i><b>8.2</b> Adding Pseudo-scRNA profiles for each scATAC cell</a></li>
<li class="chapter" data-level="8.3" data-path="labeling-scatac-clusters-with-scrna.html"><a href="labeling-scatac-clusters-with-scrna.html"><i class="fa fa-check"></i><b>8.3</b> Labeling scATAC clusters with scRNA</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="calling-peaks-with-archr.html"><a href="calling-peaks-with-archr.html"><i class="fa fa-check"></i><b>9</b> Calling Peaks with ArchR</a>
<ul>
<li class="chapter" data-level="9.1" data-path="calling-peaks-w-macs2.html"><a href="calling-peaks-w-macs2.html"><i class="fa fa-check"></i><b>9.1</b> Calling Peaks w/ Macs2</a></li>
<li class="chapter" data-level="9.2" data-path="calling-peaks-w-tilematrix.html"><a href="calling-peaks-w-tilematrix.html"><i class="fa fa-check"></i><b>9.2</b> Calling Peaks w/ TileMatrix</a></li>
<li class="chapter" data-level="9.3" data-path="add-peak-matrix.html"><a href="add-peak-matrix.html"><i class="fa fa-check"></i><b>9.3</b> Add Peak Matrix</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="identifying-marker-peaks-with-archr.html"><a href="identifying-marker-peaks-with-archr.html"><i class="fa fa-check"></i><b>10</b> Identifying Marker Peaks with ArchR</a>
<ul>
<li class="chapter" data-level="10.1" data-path="identifying-marker-peaks-with-archr-1.html"><a href="identifying-marker-peaks-with-archr-1.html"><i class="fa fa-check"></i><b>10.1</b> Identifying Marker Peaks with ArchR</a></li>
<li class="chapter" data-level="10.2" data-path="pairwise-testing-groups.html"><a href="pairwise-testing-groups.html"><i class="fa fa-check"></i><b>10.2</b> Pairwise Testing Groups</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="peak-annotation-enrichment-with-archr.html"><a href="peak-annotation-enrichment-with-archr.html"><i class="fa fa-check"></i><b>11</b> Peak Annotation Enrichment with ArchR</a>
<ul>
<li class="chapter" data-level="11.1" data-path="motif-enrichment.html"><a href="motif-enrichment.html"><i class="fa fa-check"></i><b>11.1</b> Motif Enrichment</a></li>
<li class="chapter" data-level="11.2" data-path="archr-enrichment.html"><a href="archr-enrichment.html"><i class="fa fa-check"></i><b>11.2</b> ArchR Enrichment</a>
<ul>
<li class="chapter" data-level="11.2.1" data-path="archr-enrichment.html"><a href="archr-enrichment.html#encode-tfbs"><i class="fa fa-check"></i><b>11.2.1</b> Encode TFBS</a></li>
<li class="chapter" data-level="11.2.2" data-path="archr-enrichment.html"><a href="archr-enrichment.html#bulk-atac-seq"><i class="fa fa-check"></i><b>11.2.2</b> Bulk ATAC-seq</a></li>
<li class="chapter" data-level="11.2.3" data-path="archr-enrichment.html"><a href="archr-enrichment.html#codex-tfbs"><i class="fa fa-check"></i><b>11.2.3</b> Codex TFBS</a></li>
</ul></li>
<li class="chapter" data-level="11.3" data-path="custom-enrichment.html"><a href="custom-enrichment.html"><i class="fa fa-check"></i><b>11.3</b> Custom Enrichment</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="chromvar-deviatons-enrichment-with-archr.html"><a href="chromvar-deviatons-enrichment-with-archr.html"><i class="fa fa-check"></i><b>12</b> ChromVAR Deviatons Enrichment with ArchR</a>
<ul>
<li class="chapter" data-level="12.1" data-path="motif-deviations.html"><a href="motif-deviations.html"><i class="fa fa-check"></i><b>12.1</b> Motif Deviations</a></li>
<li class="chapter" data-level="12.2" data-path="archr-deviations.html"><a href="archr-deviations.html"><i class="fa fa-check"></i><b>12.2</b> ArchR Deviations</a>
<ul>
<li class="chapter" data-level="12.2.1" data-path="archr-deviations.html"><a href="archr-deviations.html#encode-tfbs-1"><i class="fa fa-check"></i><b>12.2.1</b> Encode TFBS</a></li>
<li class="chapter" data-level="12.2.2" data-path="archr-deviations.html"><a href="archr-deviations.html#bulk-atac-seq-1"><i class="fa fa-check"></i><b>12.2.2</b> Bulk ATAC-seq</a></li>
</ul></li>
<li class="chapter" data-level="12.3" data-path="custom-deviations.html"><a href="custom-deviations.html"><i class="fa fa-check"></i><b>12.3</b> Custom Deviations</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="footprinting-with-archr.html"><a href="footprinting-with-archr.html"><i class="fa fa-check"></i><b>13</b> Footprinting with ArchR</a>
<ul>
<li class="chapter" data-level="13.1" data-path="motif-footprinting.html"><a href="motif-footprinting.html"><i class="fa fa-check"></i><b>13.1</b> Motif Footprinting</a>
<ul>
<li class="chapter" data-level="13.1.1" data-path="motif-footprinting.html"><a href="motif-footprinting.html#no-normalization"><i class="fa fa-check"></i><b>13.1.1</b> No Normalization</a></li>
<li class="chapter" data-level="13.1.2" data-path="motif-footprinting.html"><a href="motif-footprinting.html#divide-tn5-bias"><i class="fa fa-check"></i><b>13.1.2</b> Divide Tn5 Bias</a></li>
<li class="chapter" data-level="13.1.3" data-path="motif-footprinting.html"><a href="motif-footprinting.html#subtract-tn5-bias"><i class="fa fa-check"></i><b>13.1.3</b> Subtract Tn5 Bias</a></li>
</ul></li>
<li class="chapter" data-level="13.2" data-path="feature-footprinting.html"><a href="feature-footprinting.html"><i class="fa fa-check"></i><b>13.2</b> Feature Footprinting</a>
<ul>
<li class="chapter" data-level="13.2.1" data-path="feature-footprinting.html"><a href="feature-footprinting.html#no-normalization-1"><i class="fa fa-check"></i><b>13.2.1</b> No Normalization</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="14" data-path="trajectory-analysis-with-archr.html"><a href="trajectory-analysis-with-archr.html"><i class="fa fa-check"></i><b>14</b> Trajectory Analysis with ArchR</a>
<ul>
<li class="chapter" data-level="14.1" data-path="myeloid-trajectory.html"><a href="myeloid-trajectory.html"><i class="fa fa-check"></i><b>14.1</b> Myeloid Trajectory</a></li>
<li class="chapter" data-level="14.2" data-path="lymphoid-trajectory.html"><a href="lymphoid-trajectory.html"><i class="fa fa-check"></i><b>14.2</b> Lymphoid Trajectory</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="peak-linkage-with-archr.html"><a href="peak-linkage-with-archr.html"><i class="fa fa-check"></i><b>15</b> Peak Linkage with ArchR</a>
<ul>
<li class="chapter" data-level="15.1" data-path="coaccessibility-with-archr.html"><a href="coaccessibility-with-archr.html"><i class="fa fa-check"></i><b>15.1</b> CoAccessibility with ArchR</a></li>
<li class="chapter" data-level="15.2" data-path="peak2genelinkage-with-archr.html"><a href="peak2genelinkage-with-archr.html"><i class="fa fa-check"></i><b>15.2</b> Peak2GeneLinkage with ArchR</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://bookdown.org" target="_blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">ArchR</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="identifying-marker-peaks-with-archr-1" class="section level2" number="10.1">
<h2><span class="header-section-number">10.1</span> Identifying Marker Peaks with ArchR</h2>
<p>Often times, we are interested to know which peaks are unique to an individual cluster or a small group of clusters. We can do this in an unsupervised fashion in ArchR:</p>
<div class="sourceCode" id="cb99"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb99-1"><a href="identifying-marker-peaks-with-archr-1.html#cb99-1"></a><span class="kw">addArchRThreads</span>(<span class="dv">8</span>)</span>
<span id="cb99-2"><a href="identifying-marker-peaks-with-archr-1.html#cb99-2"></a></span>
<span id="cb99-3"><a href="identifying-marker-peaks-with-archr-1.html#cb99-3"></a><span class="co">#Our scRNA labels</span></span>
<span id="cb99-4"><a href="identifying-marker-peaks-with-archr-1.html#cb99-4"></a><span class="kw">table</span>(projHeme5<span class="op">$</span>Clusters2)</span>
<span id="cb99-5"><a href="identifying-marker-peaks-with-archr-1.html#cb99-5"></a><span class="co"># C1_Early.Eryth C10_CD14.Mono.2           C11_B       C12_Pre.B       C13_CLP.2 </span></span>
<span id="cb99-6"><a href="identifying-marker-peaks-with-archr-1.html#cb99-6"></a><span class="co">#             28            1070             425             351             408 </span></span>
<span id="cb99-7"><a href="identifying-marker-peaks-with-archr-1.html#cb99-7"></a><span class="co">#        C14_pDC       C2_CD4.N1       C3_CD4.N2           C4_NK   C5_Late.Eryth </span></span>
<span id="cb99-8"><a href="identifying-marker-peaks-with-archr-1.html#cb99-8"></a><span class="co">#            345            1330             562             901             860 </span></span>
<span id="cb99-9"><a href="identifying-marker-peaks-with-archr-1.html#cb99-9"></a><span class="co">#         C6_HSC     C7_CMP.LMPP     C8_GMP.Neut  C9_CD14.Mono.1 </span></span>
<span id="cb99-10"><a href="identifying-marker-peaks-with-archr-1.html#cb99-10"></a><span class="co">#           1218             974             274            1505 </span></span>
<span id="cb99-11"><a href="identifying-marker-peaks-with-archr-1.html#cb99-11"></a></span>
<span id="cb99-12"><a href="identifying-marker-peaks-with-archr-1.html#cb99-12"></a><span class="co">#Identify Marker Peaks while controling for TSS and Depth Biases</span></span>
<span id="cb99-13"><a href="identifying-marker-peaks-with-archr-1.html#cb99-13"></a>markersPeaks &lt;-<span class="st"> </span><span class="kw">markerFeatures</span>(</span>
<span id="cb99-14"><a href="identifying-marker-peaks-with-archr-1.html#cb99-14"></a>    <span class="dt">ArchRProj =</span> projHeme5, </span>
<span id="cb99-15"><a href="identifying-marker-peaks-with-archr-1.html#cb99-15"></a>    <span class="dt">useMatrix =</span> <span class="st">&quot;PeakMatrix&quot;</span>, </span>
<span id="cb99-16"><a href="identifying-marker-peaks-with-archr-1.html#cb99-16"></a>    <span class="dt">groupBy =</span> <span class="st">&quot;Clusters2&quot;</span>,</span>
<span id="cb99-17"><a href="identifying-marker-peaks-with-archr-1.html#cb99-17"></a>    <span class="dt">bias =</span> <span class="kw">c</span>(<span class="st">&quot;TSSEnrichment&quot;</span>, <span class="st">&quot;log10(nFrags)&quot;</span>),</span>
<span id="cb99-18"><a href="identifying-marker-peaks-with-archr-1.html#cb99-18"></a>    <span class="dt">testMethod =</span> <span class="st">&quot;wilcoxon&quot;</span></span>
<span id="cb99-19"><a href="identifying-marker-peaks-with-archr-1.html#cb99-19"></a>)</span>
<span id="cb99-20"><a href="identifying-marker-peaks-with-archr-1.html#cb99-20"></a></span>
<span id="cb99-21"><a href="identifying-marker-peaks-with-archr-1.html#cb99-21"></a>markersPeaks</span>
<span id="cb99-22"><a href="identifying-marker-peaks-with-archr-1.html#cb99-22"></a><span class="co"># class: SummarizedExperiment </span></span>
<span id="cb99-23"><a href="identifying-marker-peaks-with-archr-1.html#cb99-23"></a><span class="co"># dim: 158378 14 </span></span>
<span id="cb99-24"><a href="identifying-marker-peaks-with-archr-1.html#cb99-24"></a><span class="co"># metadata(2): MatchInfo Params</span></span>
<span id="cb99-25"><a href="identifying-marker-peaks-with-archr-1.html#cb99-25"></a><span class="co"># assays(5): Log2FC Mean FDR AUC MeanBGD</span></span>
<span id="cb99-26"><a href="identifying-marker-peaks-with-archr-1.html#cb99-26"></a><span class="co"># rownames(158378): 1 2 ... 158377 158378</span></span>
<span id="cb99-27"><a href="identifying-marker-peaks-with-archr-1.html#cb99-27"></a><span class="co"># rowData names(4): seqnames idx start end</span></span>
<span id="cb99-28"><a href="identifying-marker-peaks-with-archr-1.html#cb99-28"></a><span class="co"># colnames(14): C1_Early.Eryth C2_CD4.N1 ... C13_CLP.2 C14_pDC</span></span>
<span id="cb99-29"><a href="identifying-marker-peaks-with-archr-1.html#cb99-29"></a><span class="co"># colData names(0):</span></span></code></pre></div>
<p>We can get the peak regions for each marker as <code>DataFrame</code> or <code>GRanges</code>.</p>
<div class="sourceCode" id="cb100"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb100-1"><a href="identifying-marker-peaks-with-archr-1.html#cb100-1"></a><span class="co">#DataFrame</span></span>
<span id="cb100-2"><a href="identifying-marker-peaks-with-archr-1.html#cb100-2"></a>markerList &lt;-<span class="st"> </span><span class="kw">getMarkers</span>(markersPeaks, <span class="dt">cutOff =</span> <span class="st">&quot;FDR &lt;= 0.01 &amp; Log2FC &gt;= 1&quot;</span>)</span>
<span id="cb100-3"><a href="identifying-marker-peaks-with-archr-1.html#cb100-3"></a>markerList</span>
<span id="cb100-4"><a href="identifying-marker-peaks-with-archr-1.html#cb100-4"></a><span class="co"># List of length 14</span></span>
<span id="cb100-5"><a href="identifying-marker-peaks-with-archr-1.html#cb100-5"></a><span class="co"># names(14): C1_Early.Eryth C2_CD4.N1 C3_CD4.N2 ... C12_Pre.B C13_CLP.2 C14_pDC</span></span>
<span id="cb100-6"><a href="identifying-marker-peaks-with-archr-1.html#cb100-6"></a></span>
<span id="cb100-7"><a href="identifying-marker-peaks-with-archr-1.html#cb100-7"></a>markerList<span class="op">$</span>C5_Late.Eryth</span>
<span id="cb100-8"><a href="identifying-marker-peaks-with-archr-1.html#cb100-8"></a><span class="co"># DataFrame with 2262 rows and 6 columns</span></span>
<span id="cb100-9"><a href="identifying-marker-peaks-with-archr-1.html#cb100-9"></a><span class="co">#        seqnames     idx     start       end           Log2FC</span></span>
<span id="cb100-10"><a href="identifying-marker-peaks-with-archr-1.html#cb100-10"></a><span class="co">#           &lt;Rle&gt; &lt;array&gt;   &lt;array&gt;   &lt;array&gt;        &lt;numeric&gt;</span></span>
<span id="cb100-11"><a href="identifying-marker-peaks-with-archr-1.html#cb100-11"></a><span class="co"># 97705     chr22    1313  30129829  30130329 3.51592607805769</span></span>
<span id="cb100-12"><a href="identifying-marker-peaks-with-archr-1.html#cb100-12"></a><span class="co"># 10135      chr1   10135 164681301 164681801 4.44617535438744</span></span>
<span id="cb100-13"><a href="identifying-marker-peaks-with-archr-1.html#cb100-13"></a><span class="co"># 48828     chr15    1349  50552672  50553172 5.18141784055201</span></span>
<span id="cb100-14"><a href="identifying-marker-peaks-with-archr-1.html#cb100-14"></a><span class="co"># 54061     chr16    1457  15615745  15616245 8.33999798117364</span></span>
<span id="cb100-15"><a href="identifying-marker-peaks-with-archr-1.html#cb100-15"></a><span class="co"># 2928       chr1    2928  27869062  27869562 5.21444029812509</span></span>
<span id="cb100-16"><a href="identifying-marker-peaks-with-archr-1.html#cb100-16"></a><span class="co"># ...         ...     ...       ...       ...              ...</span></span>
<span id="cb100-17"><a href="identifying-marker-peaks-with-archr-1.html#cb100-17"></a><span class="co"># 41787     chr13    3145 112037183 112037683 3.09392983218513</span></span>
<span id="cb100-18"><a href="identifying-marker-peaks-with-archr-1.html#cb100-18"></a><span class="co"># 97647     chr22    1255  29702383  29702883 2.97001510006876</span></span>
<span id="cb100-19"><a href="identifying-marker-peaks-with-archr-1.html#cb100-19"></a><span class="co"># 143218     chr8    1908  37949093  37949593  2.9624832757185</span></span>
<span id="cb100-20"><a href="identifying-marker-peaks-with-archr-1.html#cb100-20"></a><span class="co"># 39746     chr13    1104  41255455  41255955 1.27203148042162</span></span>
<span id="cb100-21"><a href="identifying-marker-peaks-with-archr-1.html#cb100-21"></a><span class="co"># 128023     chr6    3784  41169564  41170064 2.67070386367678</span></span>
<span id="cb100-22"><a href="identifying-marker-peaks-with-archr-1.html#cb100-22"></a></span>
<span id="cb100-23"><a href="identifying-marker-peaks-with-archr-1.html#cb100-23"></a><span class="co">#GRanges</span></span>
<span id="cb100-24"><a href="identifying-marker-peaks-with-archr-1.html#cb100-24"></a>markerList &lt;-<span class="st"> </span><span class="kw">getMarkers</span>(markersPeaks, <span class="dt">cutOff =</span> <span class="st">&quot;FDR &lt;= 0.01 &amp; Log2FC &gt;= 1&quot;</span>, <span class="dt">returnGR =</span> <span class="ot">TRUE</span>)</span>
<span id="cb100-25"><a href="identifying-marker-peaks-with-archr-1.html#cb100-25"></a>markerList</span>
<span id="cb100-26"><a href="identifying-marker-peaks-with-archr-1.html#cb100-26"></a><span class="co"># GenomicRangesList of length 14</span></span>
<span id="cb100-27"><a href="identifying-marker-peaks-with-archr-1.html#cb100-27"></a><span class="co"># names(14): C1_Early.Eryth C2_CD4.N1 C3_CD4.N2 ... C12_Pre.B C13_CLP.2 C14_pDC</span></span>
<span id="cb100-28"><a href="identifying-marker-peaks-with-archr-1.html#cb100-28"></a></span>
<span id="cb100-29"><a href="identifying-marker-peaks-with-archr-1.html#cb100-29"></a>markerList<span class="op">$</span>C5_Late.Eryth</span>
<span id="cb100-30"><a href="identifying-marker-peaks-with-archr-1.html#cb100-30"></a><span class="co"># GRanges object with 2262 ranges and 2 metadata columns:</span></span>
<span id="cb100-31"><a href="identifying-marker-peaks-with-archr-1.html#cb100-31"></a><span class="co">#          seqnames              ranges strand |           Log2FC</span></span>
<span id="cb100-32"><a href="identifying-marker-peaks-with-archr-1.html#cb100-32"></a><span class="co">#             &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; |        &lt;numeric&gt;</span></span>
<span id="cb100-33"><a href="identifying-marker-peaks-with-archr-1.html#cb100-33"></a><span class="co">#      [1]    chr22   30129829-30130329      * | 3.51592607805769</span></span>
<span id="cb100-34"><a href="identifying-marker-peaks-with-archr-1.html#cb100-34"></a><span class="co">#      [2]     chr1 164681301-164681801      * | 4.44617535438744</span></span>
<span id="cb100-35"><a href="identifying-marker-peaks-with-archr-1.html#cb100-35"></a><span class="co">#      [3]    chr15   50552672-50553172      * | 5.18141784055201</span></span>
<span id="cb100-36"><a href="identifying-marker-peaks-with-archr-1.html#cb100-36"></a><span class="co">#      [4]    chr16   15615745-15616245      * | 8.33999798117364</span></span>
<span id="cb100-37"><a href="identifying-marker-peaks-with-archr-1.html#cb100-37"></a><span class="co">#      [5]     chr1   27869062-27869562      * | 5.21444029812509</span></span>
<span id="cb100-38"><a href="identifying-marker-peaks-with-archr-1.html#cb100-38"></a><span class="co">#      ...      ...                 ...    ... .              ...</span></span>
<span id="cb100-39"><a href="identifying-marker-peaks-with-archr-1.html#cb100-39"></a><span class="co">#   [2258]    chr13 112037183-112037683      * | 3.09392983218513</span></span>
<span id="cb100-40"><a href="identifying-marker-peaks-with-archr-1.html#cb100-40"></a><span class="co">#   [2259]    chr22   29702383-29702883      * | 2.97001510006876</span></span>
<span id="cb100-41"><a href="identifying-marker-peaks-with-archr-1.html#cb100-41"></a><span class="co">#   [2260]     chr8   37949093-37949593      * |  2.9624832757185</span></span>
<span id="cb100-42"><a href="identifying-marker-peaks-with-archr-1.html#cb100-42"></a><span class="co">#   [2261]    chr13   41255455-41255955      * | 1.27203148042162</span></span>
<span id="cb100-43"><a href="identifying-marker-peaks-with-archr-1.html#cb100-43"></a><span class="co">#   [2262]     chr6   41169564-41170064      * | 2.67070386367678</span></span></code></pre></div>
<p>We can visualize this as a heatmap</p>
<div class="sourceCode" id="cb101"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb101-1"><a href="identifying-marker-peaks-with-archr-1.html#cb101-1"></a><span class="co">#Visualize Markers as a heatmap</span></span>
<span id="cb101-2"><a href="identifying-marker-peaks-with-archr-1.html#cb101-2"></a>heatmapPeaks &lt;-<span class="st"> </span><span class="kw">markerHeatmap</span>(</span>
<span id="cb101-3"><a href="identifying-marker-peaks-with-archr-1.html#cb101-3"></a>  <span class="dt">seMarker =</span> markersPeaks, </span>
<span id="cb101-4"><a href="identifying-marker-peaks-with-archr-1.html#cb101-4"></a>  <span class="dt">cutOff =</span> <span class="st">&quot;FDR &lt;= 0.1 &amp; Log2FC &gt;= 0.5&quot;</span>,</span>
<span id="cb101-5"><a href="identifying-marker-peaks-with-archr-1.html#cb101-5"></a>  <span class="dt">transpose =</span> <span class="ot">TRUE</span></span>
<span id="cb101-6"><a href="identifying-marker-peaks-with-archr-1.html#cb101-6"></a>)</span>
<span id="cb101-7"><a href="identifying-marker-peaks-with-archr-1.html#cb101-7"></a></span>
<span id="cb101-8"><a href="identifying-marker-peaks-with-archr-1.html#cb101-8"></a><span class="kw">draw</span>(heatmapPeaks, <span class="dt">heatmap_legend_side =</span> <span class="st">&quot;bot&quot;</span>, <span class="dt">annotation_legend_side =</span> <span class="st">&quot;bot&quot;</span>)</span></code></pre></div>
<p><img src="images/HemeWalkthrough/PNG/Peak-Marker-Heatmap_1.png" width="800" height="600" /></p>
<p>We can then plot this</p>
<div class="sourceCode" id="cb102"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb102-1"><a href="identifying-marker-peaks-with-archr-1.html#cb102-1"></a><span class="kw">plotPDF</span>(heatmapPeaks, <span class="dt">name =</span> <span class="st">&quot;Peak-Marker-Heatmap&quot;</span>, <span class="dt">width =</span> <span class="dv">8</span>, <span class="dt">height =</span> <span class="dv">6</span>, <span class="dt">ArchRProj =</span> projHeme5, <span class="dt">addDOC =</span> <span class="ot">FALSE</span>)</span></code></pre></div>
<p>We can also plot an MA/Volcano Plot for any individual cluster</p>
<div class="sourceCode" id="cb103"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb103-1"><a href="identifying-marker-peaks-with-archr-1.html#cb103-1"></a><span class="co">#Marker Peaks for Erythroid</span></span>
<span id="cb103-2"><a href="identifying-marker-peaks-with-archr-1.html#cb103-2"></a>pma &lt;-<span class="st"> </span><span class="kw">markerPlot</span>(<span class="dt">seMarker =</span> markersPeaks, <span class="dt">name =</span> <span class="st">&quot;C5_Late.Eryth&quot;</span>, <span class="dt">cutOff =</span> <span class="st">&quot;FDR &lt;= 0.1 &amp; Log2FC &gt;= 1&quot;</span>, <span class="dt">plotAs =</span> <span class="st">&quot;MA&quot;</span>)</span>
<span id="cb103-3"><a href="identifying-marker-peaks-with-archr-1.html#cb103-3"></a>pma</span></code></pre></div>
<p><img src="images/HemeWalkthrough/PNG/Erythroid-Markers-MA-Volcano_1.png" width="600" height="600" /></p>
<div class="sourceCode" id="cb104"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb104-1"><a href="identifying-marker-peaks-with-archr-1.html#cb104-1"></a>pv &lt;-<span class="st"> </span><span class="kw">markerPlot</span>(<span class="dt">seMarker =</span> markersPeaks, <span class="dt">name =</span> <span class="st">&quot;C5_Late.Eryth&quot;</span>, <span class="dt">cutOff =</span> <span class="st">&quot;FDR &lt;= 0.1 &amp; Log2FC &gt;= 1&quot;</span>, <span class="dt">plotAs =</span> <span class="st">&quot;Volcano&quot;</span>)</span>
<span id="cb104-2"><a href="identifying-marker-peaks-with-archr-1.html#cb104-2"></a>pv</span></code></pre></div>
<p><img src="images/HemeWalkthrough/PNG/Erythroid-Markers-MA-Volcano_2.png" width="600" height="600" /></p>
<p>We can then plot this as a nice pdf</p>
<div class="sourceCode" id="cb105"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb105-1"><a href="identifying-marker-peaks-with-archr-1.html#cb105-1"></a><span class="kw">plotPDF</span>(pma, pv, <span class="dt">name =</span> <span class="st">&quot;Erythroid-Markers-MA-Volcano&quot;</span>, <span class="dt">width =</span> <span class="dv">5</span>, <span class="dt">height =</span> <span class="dv">5</span>, <span class="dt">ArchRProj =</span> projHeme5, <span class="dt">addDOC =</span> <span class="ot">FALSE</span>)</span></code></pre></div>
<p>Additionally we can see these peak regions overlayed on our browser tracks</p>
<div class="sourceCode" id="cb106"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb106-1"><a href="identifying-marker-peaks-with-archr-1.html#cb106-1"></a>p &lt;-<span class="st"> </span><span class="kw">ArchRBrowserTrack</span>(</span>
<span id="cb106-2"><a href="identifying-marker-peaks-with-archr-1.html#cb106-2"></a>    <span class="dt">ArchRProj =</span> projHeme5, </span>
<span id="cb106-3"><a href="identifying-marker-peaks-with-archr-1.html#cb106-3"></a>    <span class="dt">groupBy =</span> <span class="st">&quot;Clusters&quot;</span>, </span>
<span id="cb106-4"><a href="identifying-marker-peaks-with-archr-1.html#cb106-4"></a>    <span class="dt">geneSymbol =</span> <span class="kw">c</span>(<span class="st">&quot;GATA1&quot;</span>),</span>
<span id="cb106-5"><a href="identifying-marker-peaks-with-archr-1.html#cb106-5"></a>    <span class="dt">features =</span>  <span class="kw">getMarkers</span>(markersPeaks, <span class="dt">cutOff =</span> <span class="st">&quot;FDR &lt;= 0.1 &amp; Log2FC &gt;= 1&quot;</span>, <span class="dt">returnGR =</span> <span class="ot">TRUE</span>)[<span class="st">&quot;C5_Late.Eryth&quot;</span>],</span>
<span id="cb106-6"><a href="identifying-marker-peaks-with-archr-1.html#cb106-6"></a>    <span class="dt">upstream =</span> <span class="dv">50000</span>,</span>
<span id="cb106-7"><a href="identifying-marker-peaks-with-archr-1.html#cb106-7"></a>    <span class="dt">downstream =</span> <span class="dv">50000</span></span>
<span id="cb106-8"><a href="identifying-marker-peaks-with-archr-1.html#cb106-8"></a>)</span>
<span id="cb106-9"><a href="identifying-marker-peaks-with-archr-1.html#cb106-9"></a>grid<span class="op">::</span><span class="kw">grid.newpage</span>()</span>
<span id="cb106-10"><a href="identifying-marker-peaks-with-archr-1.html#cb106-10"></a>grid<span class="op">::</span><span class="kw">grid.draw</span>(p<span class="op">$</span>GATA1)</span></code></pre></div>
<p><img src="images/HemeWalkthrough/PNG/Plot-Tracks-With-Features_1.png" width="600" height="600" /></p>
<p>We can then plot this as a nice pdf</p>
<div class="sourceCode" id="cb107"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb107-1"><a href="identifying-marker-peaks-with-archr-1.html#cb107-1"></a><span class="kw">plotPDF</span>(p, <span class="dt">name =</span> <span class="st">&quot;Plot-Tracks-With-Features&quot;</span>, <span class="dt">width =</span> <span class="dv">5</span>, <span class="dt">height =</span> <span class="dv">5</span>, <span class="dt">ArchRProj =</span> projHeme5, <span class="dt">addDOC =</span> <span class="ot">FALSE</span>)</span></code></pre></div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="identifying-marker-peaks-with-archr.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="pairwise-testing-groups.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/GreenleafLab/ArchR_Website/edit/dev/10_MarkerPeaks.Rmd",
"text": "Edit this chapter"
},
"history": {
"link": "https://github.com/GreenleafLab/ArchR_Website/commits/dev/10_MarkerPeaks.Rmd",
"text": "Chapter edit history"
},
"view": {
"link": "https://github.com/GreenleafLab/ArchR_Website/view/dev/10_MarkerPeaks.Rmd",
"text": "View source on GitHub"
},
"download": ["ArchR.pdf"],
"toc": {
"collapse": "none"
}
});
});
</script>

</body>

</html>
