<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Function reference â€¢ ArchR</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/yeti/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><!-- docsearch --><script src="../docsearch.js"></script><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.3/docsearch.min.css" integrity="sha256-QOSRU/ra9ActyXkIBbiIB144aDBdtvXBcNc3OTNuX/Q=" crossorigin="anonymous"><link href="../docsearch.css" rel="stylesheet"><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/jquery.mark.min.js" integrity="sha256-4HLtjeVgH0eIB3aZ9mLYF6E8oU5chNdjU6p6rrXpl9U=" crossorigin="anonymous"></script><link href="../extra.css" rel="stylesheet"><meta property="og:title" content="Function reference"><meta property="og:image" content="https://greenleaflab.github.io/ArchR_Website/logo.svg"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-158769145-1"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-158769145-1');
</script></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-index">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">ArchR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.0.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../index.html">
    <span class="fa fa-home"></span>
     
  </a>
</li>
<li>
  <a href="../articles/Articles/tutorial.html">
    <span class="fa fa-graduation-cap"></span>
     
    Quick Tutorial
  </a>
</li>
<li>
  <a href="../bookdown/index.html">
    <span class="fa fa-file-text-o"></span>
     
    Full Manual
  </a>
</li>
<li>
  <a href="../reference/index.html">
    <span class="fa fa-code"></span>
     
    Functions
  </a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/GreenleafLab/ArchR" class="external-link">
    <span class="fa fa-github"></span>
     
    ArchR on GitHub
  </a>
</li>
      </ul><form class="navbar-form navbar-right hidden-xs hidden-sm" role="search">
        <div class="form-group">
          <input type="search" class="form-control" name="search-input" id="search-input" placeholder="Search..." aria-label="Search for..." autocomplete="off"></div>
      </form>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">
    <div class="page-header">
      <h1>Reference</h1>
    </div>

    <table class="ref-index"><colgroup><col class="alias"><col class="title"></colgroup><tbody><tr><th colspan="2">
          <h2 id="arrow-file-creation">Arrow file creation <a href="#arrow-file-creation" class="anchor" aria-hidden="true"></a></h2>
          <p class="section-desc"></p><p>Functions that help create / set up an ArrowFile</p>
        </th>
      </tr></tbody><tbody><tr><td>
          <p><code><a href="createArrowFiles.html">createArrowFiles()</a></code> </p>
        </td>
        <td><p>Create Arrow Files</p></td>
      </tr><tr><td>
          <p><code><a href="createGenomeAnnotation.html">createGenomeAnnotation()</a></code> </p>
        </td>
        <td><p>Create a genome annotation object for ArchR</p></td>
      </tr><tr><td>
          <p><code><a href="filterDoublets.html">filterDoublets()</a></code> </p>
        </td>
        <td><p>Filter Doublets From an ArchRProject</p></td>
      </tr><tr><td>
          <p><code><a href="nCells.html">nCells()</a></code> </p>
        </td>
        <td><p>Get the number of cells from an ArchRProject/ArrowFile</p></td>
      </tr><tr><td>
          <p><code><a href="reformatFragmentFiles.html">reformatFragmentFiles()</a></code> </p>
        </td>
        <td><p>Reformat Fragment Files to be Tabix and Chr Sorted</p></td>
      </tr></tbody><tbody><tr><th colspan="2">
          <h2 id="add-ers">Add-ers <a href="#add-ers" class="anchor" aria-hidden="true"></a></h2>
          <p class="section-desc"></p><p>Functions that add information to ArrowFiles or an ArchRProject</p>
        </th>
      </tr></tbody><tbody><tr><td>
          <p><code><a href="addArchRAnnotations.html">addArchRAnnotations()</a></code> </p>
        </td>
        <td><p>Add ArchR annotations to an ArchRProject</p></td>
      </tr><tr><td>
          <p><code><a href="addArchRChrPrefix.html">addArchRChrPrefix()</a></code> </p>
        </td>
        <td><p>Add a globally-applied requirement for chromosome prefix</p></td>
      </tr><tr><td>
          <p><code><a href="addArchRDebugging.html">addArchRDebugging()</a></code> </p>
        </td>
        <td><p>Set ArchR Debugging</p></td>
      </tr><tr><td>
          <p><code><a href="addArchRGenome.html">addArchRGenome()</a></code> </p>
        </td>
        <td><p>Add a globally defined genome to all ArchR functions.</p></td>
      </tr><tr><td>
          <p><code><a href="addArchRLogging.html">addArchRLogging()</a></code> </p>
        </td>
        <td><p>Set ArchR Logging</p></td>
      </tr><tr><td>
          <p><code><a href="addArchRThreads.html">addArchRThreads()</a></code> </p>
        </td>
        <td><p>Add a globally-applied number of threads to use for parallel computing.</p></td>
      </tr><tr><td>
          <p><code><a href="addArchRVerbose.html">addArchRVerbose()</a></code> </p>
        </td>
        <td><p>Set ArchR Verbosity for Log Messaging</p></td>
      </tr><tr><td>
          <p><code><a href="addBgdPeaks.html">addBgdPeaks()</a></code> </p>
        </td>
        <td><p>Add Background Peaks to an ArchRProject</p></td>
      </tr><tr><td>
          <p><code><a href="addCellColData.html">addCellColData()</a></code> </p>
        </td>
        <td><p>Add information to cellColData in an ArchRProject</p></td>
      </tr><tr><td>
          <p><code><a href="addClusters.html">addClusters()</a></code> </p>
        </td>
        <td><p>Add cluster information to an ArchRProject</p></td>
      </tr><tr><td>
          <p><code><a href="addCoAccessibility.html">addCoAccessibility()</a></code> </p>
        </td>
        <td><p>Add Peak Co-Accessibility to an ArchRProject</p></td>
      </tr><tr><td>
          <p><code><a href="addCombinedDims.html">addCombinedDims()</a></code> </p>
        </td>
        <td><p>Combine two or more modalities dimensionality reductions.</p></td>
      </tr><tr><td>
          <p><code><a href="addDemuxletResults.html">addDemuxletResults()</a></code> </p>
        </td>
        <td><p>Add Demuxlet Results to ArchR Project</p></td>
      </tr><tr><td>
          <p><code><a href="addDeviationsMatrix.html">addDeviationsMatrix()</a></code> </p>
        </td>
        <td><p>Add a matrix of deviations for a given peakAnnotation to Arrow Files in ArchRProject</p></td>
      </tr><tr><td>
          <p><code><a href="addDoubletScores.html">addDoubletScores()</a></code> </p>
        </td>
        <td><p>Add Doublet Scores to a collection of ArrowFiles or an ArchRProject</p></td>
      </tr><tr><td>
          <p><code><a href="addFeatureCounts.html">addFeatureCounts()</a></code> </p>
        </td>
        <td><p>This function will add total counts of scATAC cells in provided features into ArchRProject.</p></td>
      </tr><tr><td>
          <p><code><a href="addFeatureMatrix.html">addFeatureMatrix()</a></code> </p>
        </td>
        <td><p>Add a feature matrix to an ArchRProject or a set of ArrowFiles</p></td>
      </tr><tr><td>
          <p><code><a href="addGeneExpressionMatrix.html">addGeneExpressionMatrix()</a></code> </p>
        </td>
        <td><p>Add Gene Expression Matrix to ArrowFiles or an ArchRProject</p></td>
      </tr><tr><td>
          <p><code><a href="addGeneIntegrationMatrix.html">addGeneIntegrationMatrix()</a></code> </p>
        </td>
        <td><p>Add a GeneIntegrationMatrix to ArrowFiles or an ArchRProject</p></td>
      </tr><tr><td>
          <p><code><a href="addGeneScoreMatrix.html">addGeneScoreMatrix()</a></code> </p>
        </td>
        <td><p>Add GeneScoreMatrix to ArrowFiles or an ArchRProject</p></td>
      </tr><tr><td>
          <p><code><a href="addGroupCoverages.html">addGroupCoverages()</a></code> </p>
        </td>
        <td><p>Add Group Coverages to an ArchRProject object</p></td>
      </tr><tr><td>
          <p><code><a href="addHarmony.html">addHarmony()</a></code> </p>
        </td>
        <td><p>Add Harmony Batch Corrected Reduced Dims to an ArchRProject</p></td>
      </tr><tr><td>
          <p><code><a href="addImputeWeights.html">addImputeWeights()</a></code> </p>
        </td>
        <td><p>Add Imputation Weights to an ArchRProject</p></td>
      </tr><tr><td>
          <p><code><a href="addIterativeLSI.html">addIterativeLSI()</a></code> </p>
        </td>
        <td><p>Add an Iterative LSI-based dimensionality reduction to an ArchRProject</p></td>
      </tr><tr><td>
          <p><code><a href="addModuleScore.html">addModuleScore()</a></code> </p>
        </td>
        <td><p>Add Module Scores to an ArchRProject</p></td>
      </tr><tr><td>
          <p><code><a href="addMonocleTrajectory.html">addMonocleTrajectory()</a></code> </p>
        </td>
        <td><p>Add a Monocle Trajectory to an ArchR Project #NEW</p></td>
      </tr><tr><td>
          <p><code><a href="addMotifAnnotations.html">addMotifAnnotations()</a></code> </p>
        </td>
        <td><p>Add motif annotations to an ArchRProject</p></td>
      </tr><tr><td>
          <p><code><a href="addPeak2GeneLinks.html">addPeak2GeneLinks()</a></code> </p>
        </td>
        <td><p>Add Peak2GeneLinks to an ArchRProject</p></td>
      </tr><tr><td>
          <p><code><a href="addPeakAnnotations.html">addPeakAnnotations()</a></code> </p>
        </td>
        <td><p>Add peak annotations to an ArchRProject</p></td>
      </tr><tr><td>
          <p><code><a href="addPeakMatrix.html">addPeakMatrix()</a></code> </p>
        </td>
        <td><p>Add a Peak Matrix to the ArrowFiles of an ArchRProject</p></td>
      </tr><tr><td>
          <p><code><a href="addPeakSet.html">addPeakSet()</a></code> </p>
        </td>
        <td><p>Add a peak set to an ArchRProject</p></td>
      </tr><tr><td>
          <p><code><a href="addProjectSummary.html">addProjectSummary()</a></code> </p>
        </td>
        <td><p>Add projectSummary to an ArchRProject</p></td>
      </tr><tr><td>
          <p><code><a href="addReproduciblePeakSet.html">addReproduciblePeakSet()</a></code> </p>
        </td>
        <td><p>Add a Reproducible Peak Set to an ArchRProject</p></td>
      </tr><tr><td>
          <p><code><a href="addSampleColData.html">addSampleColData()</a></code> </p>
        </td>
        <td><p>Add information to sampleColData in an ArchRProject</p></td>
      </tr><tr><td>
          <p><code><a href="addSlingShotTrajectories.html">addSlingShotTrajectories()</a></code> </p>
        </td>
        <td><p>Add a Slingshot Trajectories to an ArchR Project #NEW</p></td>
      </tr><tr><td>
          <p><code><a href="addTSNE.html">addTSNE()</a></code> </p>
        </td>
        <td><p>Add a TSNE embedding of a reduced dimensions object to an ArchRProject</p></td>
      </tr><tr><td>
          <p><code><a href="addTileMatrix.html">addTileMatrix()</a></code> </p>
        </td>
        <td><p>Add TileMatrix to ArrowFiles or an ArchRProject</p></td>
      </tr><tr><td>
          <p><code><a href="addTrajectory.html">addTrajectory()</a></code> </p>
        </td>
        <td><p>Add a Supervised Trajectory to an ArchR Project</p></td>
      </tr><tr><td>
          <p><code><a href="addUMAP.html">addUMAP()</a></code> </p>
        </td>
        <td><p>Add a UMAP embedding of a reduced dimensions object to an ArchRProject</p></td>
      </tr></tbody><tbody><tr><th colspan="2">
          <h2 id="get-ers">Get-ers <a href="#get-ers" class="anchor" aria-hidden="true"></a></h2>
          <p class="section-desc"></p><p>Functions that get information from ArrowFiles or an ArchRProject</p>
        </th>
      </tr></tbody><tbody><tr><td>
          <p><code><a href="getArchRChrPrefix.html">getArchRChrPrefix()</a></code> </p>
        </td>
        <td><p>Get a globally-applied requirement for chromosome prefix</p></td>
      </tr><tr><td>
          <p><code><a href="getArchRDebugging.html">getArchRDebugging()</a></code> </p>
        </td>
        <td><p>Get ArchR Debugging</p></td>
      </tr><tr><td>
          <p><code><a href="getArchRGenome.html">getArchRGenome()</a></code> </p>
        </td>
        <td><p>Get the globally defined genome, the geneAnnotation, or genomeAnnotation objects associated with the globally defined genome.</p></td>
      </tr><tr><td>
          <p><code><a href="getArchRLogging.html">getArchRLogging()</a></code> </p>
        </td>
        <td><p>Get ArchR Logging</p></td>
      </tr><tr><td>
          <p><code><a href="getArchRThreads.html">getArchRThreads()</a></code> </p>
        </td>
        <td><p>Get globally-applied number of threads to use for parallel computing.</p></td>
      </tr><tr><td>
          <p><code><a href="getArchRVerbose.html">getArchRVerbose()</a></code> </p>
        </td>
        <td><p>Set ArchR Verbosity for Log Messaging</p></td>
      </tr><tr><td>
          <p><code><a href="getArrowFiles.html">getArrowFiles()</a></code> </p>
        </td>
        <td><p>Get ArrowFiles from an ArchRProject</p></td>
      </tr><tr><td>
          <p><code><a href="getAvailableMatrices.html">getAvailableMatrices()</a></code> </p>
        </td>
        <td><p>Get a list available matrices in the ArrowFiles storted in an ArchRProject</p></td>
      </tr><tr><td>
          <p><code><a href="getBgdPeaks.html">getBgdPeaks()</a></code> </p>
        </td>
        <td><p>Get Background Peaks from an ArchRProject</p></td>
      </tr><tr><td>
          <p><code><a href="getBlacklist.html">getBlacklist()</a></code> </p>
        </td>
        <td><p>Get the blacklist from an ArchRProject</p></td>
      </tr><tr><td>
          <p><code><a href="getCellColData.html">getCellColData()</a></code> </p>
        </td>
        <td><p>Get cellColData from an ArchRProject</p></td>
      </tr><tr><td>
          <p><code><a href="getCellNames.html">getCellNames()</a></code> </p>
        </td>
        <td><p>Get cellNames from an ArchRProject</p></td>
      </tr><tr><td>
          <p><code><a href="getChromLengths.html">getChromLengths()</a></code> </p>
        </td>
        <td><p>Get chromLengths from ArchRProject</p></td>
      </tr><tr><td>
          <p><code><a href="getChromSizes.html">getChromSizes()</a></code> </p>
        </td>
        <td><p>Get chromSizes from ArchRProject</p></td>
      </tr><tr><td>
          <p><code><a href="getCoAccessibility.html">getCoAccessibility()</a></code> </p>
        </td>
        <td><p>Get the peak co-accessibility from an ArchRProject</p></td>
      </tr><tr><td>
          <p><code><a href="getEmbedding.html">getEmbedding()</a></code> </p>
        </td>
        <td><p>Get embedding information stored in an ArchRProject</p></td>
      </tr><tr><td>
          <p><code><a href="getExons.html">getExons()</a></code> </p>
        </td>
        <td><p>Get the exons from an ArchRProject</p></td>
      </tr><tr><td>
          <p><code><a href="getFeatures.html">getFeatures()</a></code> </p>
        </td>
        <td><p>Get the features that could be selected from a given data matrix within an ArchRProject</p></td>
      </tr><tr><td>
          <p><code><a href="getFootprints.html">getFootprints()</a></code> </p>
        </td>
        <td><p>Calculate footprints from an ArchRProject</p></td>
      </tr><tr><td>
          <p><code><a href="getFragmentsFromArrow.html">getFragmentsFromArrow()</a></code> </p>
        </td>
        <td><p>Get the fragments from an ArrowFile</p></td>
      </tr><tr><td>
          <p><code><a href="getFragmentsFromProject.html">getFragmentsFromProject()</a></code> </p>
        </td>
        <td><p>Get the fragments from an ArchRProject</p></td>
      </tr><tr><td>
          <p><code><a href="getGeneAnnotation.html">getGeneAnnotation()</a></code> </p>
        </td>
        <td><p>Get geneAnnotation from an ArchRProject</p></td>
      </tr><tr><td>
          <p><code><a href="getGenes.html">getGenes()</a></code> </p>
        </td>
        <td><p>Get the genes from an ArchRProject</p></td>
      </tr><tr><td>
          <p><code><a href="getGenome.html">getGenome()</a></code> </p>
        </td>
        <td><p>Get the genome used by an ArchRProject</p></td>
      </tr><tr><td>
          <p><code><a href="getGenomeAnnotation.html">getGenomeAnnotation()</a></code> </p>
        </td>
        <td><p>Get the genomeAnnotation from an ArchRProject</p></td>
      </tr><tr><td>
          <p><code><a href="getGroupBW.html">getGroupBW()</a></code> </p>
        </td>
        <td><p>Export Group BigWigs</p></td>
      </tr><tr><td>
          <p><code><a href="getGroupSE.html">getGroupSE()</a></code> </p>
        </td>
        <td><p>Export Group Summarized Experiment</p></td>
      </tr><tr><td>
          <p><code><a href="getGroupSummary.html">getGroupSummary()</a></code> </p>
        </td>
        <td><p>Get summary for Groups in ArchRProject</p></td>
      </tr><tr><td>
          <p><code><a href="getImputeWeights.html">getImputeWeights()</a></code> </p>
        </td>
        <td><p>Get Imputation Weights from ArchRProject</p></td>
      </tr><tr><td>
          <p><code><a href="getInputFiles.html">getInputFiles()</a></code> </p>
        </td>
        <td><p>Get Input Files from paths to create arrows</p></td>
      </tr><tr><td>
          <p><code><a href="getMarkerFeatures.html">getMarkerFeatures()</a></code> </p>
        </td>
        <td><p>Identify Marker Features for each cell grouping</p></td>
      </tr><tr><td>
          <p><code><a href="getMarkers.html">getMarkers()</a></code> </p>
        </td>
        <td><p>Get Marker Features from a marker summarized experiment</p></td>
      </tr><tr><td>
          <p><code><a href="getMatches.html">getMatches()</a></code> </p>
        </td>
        <td><p>Get peak annotation matches from an ArchRProject</p></td>
      </tr><tr><td>
          <p><code><a href="getMatrixFromArrow.html">getMatrixFromArrow()</a></code> </p>
        </td>
        <td><p>Get a data matrix stored in an ArrowFile</p></td>
      </tr><tr><td>
          <p><code><a href="getMatrixFromProject.html">getMatrixFromProject()</a></code> </p>
        </td>
        <td><p>Get a data matrix stored in an ArchRProject</p></td>
      </tr><tr><td>
          <p><code><a href="getMonocleTrajectories.html">getMonocleTrajectories()</a></code> </p>
        </td>
        <td><p>Get a Monocle CDS of Trajectories that can be added to an ArchRProject #NEW</p></td>
      </tr><tr><td>
          <p><code><a href="getOutputDirectory.html">getOutputDirectory()</a></code> </p>
        </td>
        <td><p>Get outputDirectory from an ArchRProject</p></td>
      </tr><tr><td>
          <p><code><a href="getPeak2GeneLinks.html">getPeak2GeneLinks()</a></code> </p>
        </td>
        <td><p>Get the peak-to-gene links from an ArchRProject</p></td>
      </tr><tr><td>
          <p><code><a href="getPeakAnnotation.html">getPeakAnnotation()</a></code> </p>
        </td>
        <td><p>Get peakAnnotation from an ArchRProject</p></td>
      </tr><tr><td>
          <p><code><a href="getPeakSet.html">getPeakSet()</a></code> </p>
        </td>
        <td><p>Get the peak set from an ArchRProject</p></td>
      </tr><tr><td>
          <p><code><a href="getPositions.html">getPositions()</a></code> </p>
        </td>
        <td><p>Get peak annotation positions from an ArchRProject</p></td>
      </tr><tr><td>
          <p><code><a href="getProjectSummary.html">getProjectSummary()</a></code> </p>
        </td>
        <td><p>Get projectSummary from an ArchRProject</p></td>
      </tr><tr><td>
          <p><code><a href="getReducedDims.html">getReducedDims()</a></code> </p>
        </td>
        <td><p>Get dimensionality reduction information stored in an ArchRProject</p></td>
      </tr><tr><td>
          <p><code><a href="getSampleColData.html">getSampleColData()</a></code> </p>
        </td>
        <td><p>Get sampleColData from an ArchRProject</p></td>
      </tr><tr><td>
          <p><code><a href="getSampleNames.html">getSampleNames()</a></code> </p>
        </td>
        <td><p>Get the sample names from an ArchRProject</p></td>
      </tr><tr><td>
          <p><code><a href="getSeqnames.html">getSeqnames()</a></code> </p>
        </td>
        <td><p>Get the seqnames that could be selected from a given data matrix within an ArchRProject</p></td>
      </tr><tr><td>
          <p><code><a href="getTSS.html">getTSS()</a></code> </p>
        </td>
        <td><p>Get the transcription start sites of all genes in an ArchRProject</p></td>
      </tr><tr><td>
          <p><code><a href="getTestFragments.html">getTestFragments()</a></code> </p>
        </td>
        <td><p>Get PBMC Small Test Fragments</p></td>
      </tr><tr><td>
          <p><code><a href="getTestProject.html">getTestProject()</a></code> </p>
        </td>
        <td><p>Get PBMC Small Test Project</p></td>
      </tr><tr><td>
          <p><code><a href="getTrajectory.html">getTrajectory()</a></code> </p>
        </td>
        <td><p>Get Supervised Trajectory from an ArchR Project</p></td>
      </tr><tr><td>
          <p><code><a href="getTutorialData.html">getTutorialData()</a></code> </p>
        </td>
        <td><p>Get Relevant Data For ArchR Tutorials</p></td>
      </tr><tr><td>
          <p><code><a href="getValidBarcodes.html">getValidBarcodes()</a></code> </p>
        </td>
        <td><p>Get Valid Barcodes from 10x Cell Ranger output to pre-filter barcodes</p></td>
      </tr><tr><td>
          <p><code><a href="getVarDeviations.html">getVarDeviations()</a></code> </p>
        </td>
        <td><p>Get Variable Deviations across cells in ArchRProject.</p></td>
      </tr></tbody><tbody><tr><th colspan="2">
          <h2 id="manipulating-an-archrproject">Manipulating an ArchRProject <a href="#manipulating-an-archrproject" class="anchor" aria-hidden="true"></a></h2>
          <p class="section-desc"></p><p>Functions for manipulating an ArchRProject</p>
        </th>
      </tr></tbody><tbody><tr><td>
          <p><code><a href="ArchRProject.html">ArchRProject()</a></code> </p>
        </td>
        <td><p>Create ArchRProject from ArrowFiles</p></td>
      </tr><tr><td>
          <p><code><a href="loadArchRProject.html">loadArchRProject()</a></code> </p>
        </td>
        <td><p>Load Previous ArchRProject into R</p></td>
      </tr><tr><td>
          <p><code><a href="recoverArchRProject.html">recoverArchRProject()</a></code> </p>
        </td>
        <td><p>Recover ArchRProject if broken sampleColData/cellColData</p></td>
      </tr><tr><td>
          <p><code><a href="saveArchRProject.html">saveArchRProject()</a></code> </p>
        </td>
        <td><p>Save ArchRProject for Later Usage</p></td>
      </tr><tr><td>
          <p><code><a href="subsetArchRProject.html">subsetArchRProject()</a></code> </p>
        </td>
        <td><p>Subset an ArchRProject for downstream analysis</p></td>
      </tr><tr><td>
          <p><code><a href="subsetCells.html">subsetCells()</a></code> </p>
        </td>
        <td><p>Subset cells in an ArchRProject.</p></td>
      </tr></tbody><tbody><tr><th colspan="2">
          <h2 id="plotting-aesthetic-functions">Plotting / aesthetic functions <a href="#plotting-aesthetic-functions" class="anchor" aria-hidden="true"></a></h2>
          <p class="section-desc"></p><p>Functions for plotting in ArchR</p>
        </th>
      </tr></tbody><tbody><tr><td>
          <p><code><a href="ArchRBrowser.html">ArchRBrowser()</a></code> </p>
        </td>
        <td><p>Launch ArchR Genome Browser</p></td>
      </tr><tr><td>
          <p><code><a href="ArchRPalettes.html">ArchRPalettes</a></code> </p>
        </td>
        <td><p>List of color palettes that can be used in plots</p></td>
      </tr><tr><td>
          <p><code><a href="ggAlignPlots.html">ggAlignPlots()</a></code> </p>
        </td>
        <td><p>Align ggplot plots vertically or horizontally</p></td>
      </tr><tr><td>
          <p><code><a href="ggGroup.html">ggGroup()</a></code> </p>
        </td>
        <td><p>A ggplot-based ridge/violin plot wrapper function</p></td>
      </tr><tr><td>
          <p><code><a href="ggHex.html">ggHex()</a></code> </p>
        </td>
        <td><p>A ggplot-based Hexplot wrapper function summary of points in a standardized manner</p></td>
      </tr><tr><td>
          <p><code><a href="ggOneToOne.html">ggOneToOne()</a></code> </p>
        </td>
        <td><p>A ggplot-based one-to-one dot plot wrapper function</p></td>
      </tr><tr><td>
          <p><code><a href="ggPoint.html">ggPoint()</a></code> </p>
        </td>
        <td><p>A ggplot-based dot plot wrapper function</p></td>
      </tr><tr><td>
          <p><code><a href="paletteContinuous.html">paletteContinuous()</a></code> </p>
        </td>
        <td><p>Continuous Color Palette</p></td>
      </tr><tr><td>
          <p><code><a href="paletteDiscrete.html">paletteDiscrete()</a></code> </p>
        </td>
        <td><p>Optimized discrete color palette generation</p></td>
      </tr><tr><td>
          <p><code><a href="plotBrowserTrack.html">plotBrowserTrack()</a></code> </p>
        </td>
        <td><p>Plot an ArchR Region Track</p></td>
      </tr><tr><td>
          <p><code><a href="plotEmbedding.html">plotEmbedding()</a></code> </p>
        </td>
        <td><p>Visualize an Embedding from ArchR Project</p></td>
      </tr><tr><td>
          <p><code><a href="plotEnrichHeatmap.html">plotEnrichHeatmap()</a></code> </p>
        </td>
        <td><p>Plot a Heatmap of Peak Annotation Hypergeometric Enrichment in Marker Peaks.</p></td>
      </tr><tr><td>
          <p><code><a href="plotFootprints.html">plotFootprints()</a></code> </p>
        </td>
        <td><p>Plot Footprints</p></td>
      </tr><tr><td>
          <p><code><a href="plotFragmentSizes.html">plotFragmentSizes()</a></code> </p>
        </td>
        <td><p>Plot the fragment size distribution for each sample</p></td>
      </tr><tr><td>
          <p><code><a href="plotGroups.html">plotGroups()</a></code> </p>
        </td>
        <td><p>Visualize Groups from ArchR Project</p></td>
      </tr><tr><td>
          <p><code><a href="plotMarkerHeatmap.html">plotMarkerHeatmap()</a></code> </p>
        </td>
        <td><p>Plot a Heatmap of Identified Marker Features</p></td>
      </tr><tr><td>
          <p><code><a href="plotMarkers.html">plotMarkers()</a></code> </p>
        </td>
        <td><p>Plot Differential Markers</p></td>
      </tr><tr><td>
          <p><code><a href="plotPDF.html">plotPDF()</a></code> </p>
        </td>
        <td><p>Plot PDF in outputDirectory of an ArchRProject</p></td>
      </tr><tr><td>
          <p><code><a href="plotPeak2GeneHeatmap.html">plotPeak2GeneHeatmap()</a></code> </p>
        </td>
        <td><p>Plot Peak2Gene Heatmap from an ArchRProject</p></td>
      </tr><tr><td>
          <p><code><a href="plotTSSEnrichment.html">plotTSSEnrichment()</a></code> </p>
        </td>
        <td><p>Plot a TSS Enrichment Plot for Each Sample</p></td>
      </tr><tr><td>
          <p><code><a href="plotTrajectory.html">plotTrajectory()</a></code> </p>
        </td>
        <td><p>Visualize a Trajectory from ArchR Project</p></td>
      </tr><tr><td>
          <p><code><a href="plotTrajectoryHeatmap.html">plotTrajectoryHeatmap()</a></code> </p>
        </td>
        <td><p>Plot a Heatmap of Features across a Trajectory</p></td>
      </tr><tr><td>
          <p><code><a href="theme_ArchR.html">theme_ArchR()</a></code> </p>
        </td>
        <td><p>ggplot2 default theme for ArchR</p></td>
      </tr></tbody><tbody><tr><th colspan="2">
          <h2 id="helper-functions">Helper functions <a href="#helper-functions" class="anchor" aria-hidden="true"></a></h2>
          <p class="section-desc"></p><p>Functions that make things easier</p>
        </th>
      </tr></tbody><tbody><tr><td>
          <p><code><a href="grapes-ni-grapes.html">`%ni%`</a></code> </p>
        </td>
        <td><p>Negated Value Matching</p></td>
      </tr><tr><td>
          <p><code><a href="grapes-bcin-grapes.html">`%bcin%`</a></code> </p>
        </td>
        <td><p>Generic matching function for S4Vector objects</p></td>
      </tr><tr><td>
          <p><code><a href="grapes-bcni-grapes.html">`%bcni%`</a></code> </p>
        </td>
        <td><p>Negated matching function for S4Vector objects</p></td>
      </tr><tr><td>
          <p><code><a href="createLogFile.html">createLogFile()</a></code> </p>
        </td>
        <td><p>Create a Log File for ArchR</p></td>
      </tr><tr><td>
          <p><code><a href="findMacs2.html">findMacs2()</a></code> </p>
        </td>
        <td><p>Find the installed location of the MACS2 executable</p></td>
      </tr><tr><td>
          <p><code><a href="installExtraPackages.html">installExtraPackages()</a></code> </p>
        </td>
        <td><p>Install extra packages used in ArchR that are not installed by default</p></td>
      </tr><tr><td>
          <p><code><a href="validBSgenome.html">validBSgenome()</a></code> </p>
        </td>
        <td><p>Get/Validate BSgenome</p></td>
      </tr></tbody><tbody><tr><th colspan="2">
          <h2 id="granges-manipulation">GRanges manipulation <a href="#granges-manipulation" class="anchor" aria-hidden="true"></a></h2>
          <p class="section-desc"></p><p>Functions to manipulate GRanges objects</p>
        </th>
      </tr></tbody><tbody><tr><td>
          <p><code><a href="extendGR.html">extendGR()</a></code> </p>
        </td>
        <td><p>Extend regions from a Genomic Ranges object</p></td>
      </tr><tr><td>
          <p><code><a href="filterChrGR.html">filterChrGR()</a></code> </p>
        </td>
        <td><p>Filters unwanted seqlevels from a Genomic Ranges object or similar object</p></td>
      </tr><tr><td>
          <p><code><a href="nonOverlappingGR.html">nonOverlappingGR()</a></code> </p>
        </td>
        <td><p>Retreive a non-overlapping set of regions from a Genomic Ranges object</p></td>
      </tr></tbody><tbody><tr><th colspan="2">
          <h2 id="other-archr-functions">Other ArchR Functions <a href="#other-archr-functions" class="anchor" aria-hidden="true"></a></h2>
          <p class="section-desc"></p><p>Functions that didnt fit into one of the above categories</p>
        </th>
      </tr></tbody><tbody><tr><td>
          <p><code><a href="confusionMatrix.html">confusionMatrix()</a></code> </p>
        </td>
        <td><p>Create a Confusion Matrix based on two value vectors</p></td>
      </tr><tr><td>
          <p><code><a href="correlateMatrices.html">correlateMatrices()</a></code> </p>
        </td>
        <td><p>Correlate Matrices within an ArchRProject</p></td>
      </tr><tr><td>
          <p><code><a href="correlateTrajectories.html">correlateTrajectories()</a></code> </p>
        </td>
        <td><p>Correlate Trajectories</p></td>
      </tr><tr><td>
          <p><code><a href="imputeMatrix.html">imputeMatrix()</a></code> </p>
        </td>
        <td><p>Impute a matrix with impute weights</p></td>
      </tr><tr><td>
          <p><code><a href="mapLabels.html">mapLabels()</a></code> </p>
        </td>
        <td><p>Re-map a character vector of labels from an old set of labels to a new set of labels</p></td>
      </tr><tr><td>
          <p><code><a href="peakAnnoEnrichment.html">peakAnnoEnrichment()</a></code> </p>
        </td>
        <td><p>Peak Annotation Hypergeometric Enrichment in Marker Peaks.</p></td>
      </tr><tr><td>
          <p><code><a href="projectBulkATAC.html">projectBulkATAC()</a></code> </p>
        </td>
        <td><p>Project Bulk ATAC-seq data into single cell subspace</p></td>
      </tr></tbody></table></div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by <a href="https://twitter.com/JeffreyGranja" class="external-link">Jeffrey Granja</a>, <a href="https://twitter.com/doctorcorces" class="external-link">Ryan Corces</a>.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.1.</p>
</div>

      </footer></div>

  
<script src="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.1/docsearch.min.js" integrity="sha256-GKvGqXDznoRYHCwKXGnuchvKSwmx9SRMrZOTh2g4Sb0=" crossorigin="anonymous"></script><script>
  docsearch({
    
    
    apiKey: '252c2c36fedc32b3ed803d0db4877b96',
    indexName: 'archr',
    inputSelector: 'input#search-input.form-control',
    transformData: function(hits) {
      return hits.map(function (hit) {
        hit.url = updateHitURL(hit);
        return hit;
      });
    }
  });
</script></body></html>

