<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta name="viewport" content="width=device-width, initial-scale=1" />
<meta property="og:title" content="Chapter 19 Renv solo locally | ArchR: Robust and scaleable analysis of single-cell chromatin accessibility data." />
<meta property="og:type" content="book" />


<meta property="og:description" content="A guide to ArchR" />
<meta name="github-repo" content="GreenleafLab/ArchR" />

<meta name="author" content="Jeffrey Granja and Ryan Corces" />

<meta name="date" content="2022-01-17" />

<script type="text/x-mathjax-config">
MathJax.Hub.Config({
  TeX: { equationNumbers: { autoNumber: "AMS" } }
});
</script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<meta name="description" content="A guide to ArchR">

<title>Chapter 19 Renv solo locally | ArchR: Robust and scaleable analysis of single-cell chromatin accessibility data.</title>

<script src="libs/header-attrs-2.10/header-attrs.js"></script>
<script src="libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="libs/navigation-1.1/tabsets.js"></script>
<link href="libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>



<link rel="stylesheet" href="css/style.css" type="text/css" />
<link rel="stylesheet" href="css/toc.css" type="text/css" />

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
/* show arrow before summary tag as in bootstrap
TODO: remove if boostrap in updated in html_document (rmarkdown#1485) */
details > summary {
  display: list-item;
  cursor: pointer;
}
</style>
</head>

<body>

<div class="container-fluid main-container">


<div class="row">
<div class="col-sm-12">
<div id="TOC">
<ul>
<li><a href="index.html#section"><img src="images/ArchR_Logo_Integrated.png" alt="" width="200"></a></li>
<li><a href="1-getting-started-with-archr.html#getting-started-with-archr"><span class="toc-section-number">1</span> Getting Started with ArchR</a>
<ul>
<li><a href="1.1-a-brief-primer-on-atac-seq-terminology.html#a-brief-primer-on-atac-seq-terminology"><span class="toc-section-number">1.1</span> A Brief Primer on ATAC-seq Terminology</a></li>
<li><a href="1.2-why-use-archr.html#why-use-archr"><span class="toc-section-number">1.2</span> Why use ArchR?</a></li>
<li><a href="1.3-what-is-an-arrow-file-archrproject.html#what-is-an-arrow-file-archrproject"><span class="toc-section-number">1.3</span> What is an Arrow file / <code>ArchRProject</code>?</a></li>
<li><a href="1.4-input-file-types-in-archr.html#input-file-types-in-archr"><span class="toc-section-number">1.4</span> Input File Types in ArchR</a></li>
<li><a href="1.5-debugging-in-archr.html#debugging-in-archr"><span class="toc-section-number">1.5</span> Debugging in ArchR</a></li>
<li><a href="1.6-getting-set-up.html#getting-set-up"><span class="toc-section-number">1.6</span> Getting Set Up</a>
<ul>
<li><a href="1.6-getting-set-up.html#creating-a-custom-archrgenome"><span class="toc-section-number">1.6.1</span> Creating a Custom <code>ArchRGenome</code></a></li>
<li><a href="1.6-getting-set-up.html#using-non-standard-genomes-in-archr"><span class="toc-section-number">1.6.2</span> Using Non-standard Genomes in ArchR</a></li>
</ul></li>
<li><a href="1.7-creating-arrow-files.html#creating-arrow-files"><span class="toc-section-number">1.7</span> Creating Arrow Files</a>
<ul>
<li><a href="1.7-creating-arrow-files.html#input-file-types-for-archr"><span class="toc-section-number">1.7.1</span> Input file types for ArchR</a></li>
<li><a href="1.7-creating-arrow-files.html#arrow-file-creation-from-fragment-files"><span class="toc-section-number">1.7.2</span> Arrow file creation from fragment files</a></li>
</ul></li>
<li><a href="1.8-per-cell-quality-control.html#per-cell-quality-control"><span class="toc-section-number">1.8</span> Per-cell Quality Control</a></li>
</ul></li>
<li><a href="2-doublet-inference-with-archr.html#doublet-inference-with-archr"><span class="toc-section-number">2</span> Doublet Inference with ArchR</a>
<ul>
<li><a href="2.1-how-does-doublet-identification-work-in-archr.html#how-does-doublet-identification-work-in-archr"><span class="toc-section-number">2.1</span> How does doublet identification work in ArchR?</a></li>
<li><a href="2.2-inferring-scatac-seq-doublets-with-archr.html#inferring-scatac-seq-doublets-with-archr"><span class="toc-section-number">2.2</span> Inferring scATAC-seq Doublets with ArchR</a></li>
</ul></li>
<li><a href="3-creating-an-archrproject.html#creating-an-archrproject"><span class="toc-section-number">3</span> Creating an ArchRProject</a>
<ul>
<li><a href="3.1-creating-an-archrproject-1.html#creating-an-archrproject-1"><span class="toc-section-number">3.1</span> Creating An ArchRProject</a></li>
<li><a href="3.2-manipulating-an-archrproject.html#manipulating-an-archrproject"><span class="toc-section-number">3.2</span> Manipulating An ArchRProject</a></li>
<li><a href="3.3-plotting-sample-statistics-from-an-archrproject.html#plotting-sample-statistics-from-an-archrproject"><span class="toc-section-number">3.3</span> Plotting Sample Statistics from an ArchRProject</a></li>
<li><a href="3.4-plotting-sample-fragment-size-distribution-and-tss-enrichment-profiles..html#plotting-sample-fragment-size-distribution-and-tss-enrichment-profiles."><span class="toc-section-number">3.4</span> Plotting Sample Fragment Size Distribution and TSS Enrichment Profiles.</a></li>
<li><a href="3.5-saving-and-loading-an-archrproject.html#saving-and-loading-an-archrproject"><span class="toc-section-number">3.5</span> Saving and Loading an <code>ArchRProject</code></a></li>
<li><a href="3.6-filtering-doublets-from-an-archrproject.html#filtering-doublets-from-an-archrproject"><span class="toc-section-number">3.6</span> Filtering Doublets from an ArchRProject</a></li>
</ul></li>
<li><a href="4-dimensionality-reduction-with-archr.html#dimensionality-reduction-with-archr"><span class="toc-section-number">4</span> Dimensionality Reduction with ArchR</a>
<ul>
<li><a href="4.1-archrs-lsi-implementation.html#archrs-lsi-implementation"><span class="toc-section-number">4.1</span> ArchR’s LSI Implementation</a></li>
<li><a href="4.2-iterative-latent-semantic-indexing-lsi.html#iterative-latent-semantic-indexing-lsi"><span class="toc-section-number">4.2</span> Iterative Latent Semantic Indexing (LSI)</a></li>
<li><a href="4.3-estimated-lsi.html#estimated-lsi"><span class="toc-section-number">4.3</span> Estimated LSI</a></li>
<li><a href="4.4-batch-effect-correction-wtih-harmony.html#batch-effect-correction-wtih-harmony"><span class="toc-section-number">4.4</span> Batch Effect Correction wtih Harmony</a></li>
</ul></li>
<li><a href="5-clustering-with-archr.html#clustering-with-archr"><span class="toc-section-number">5</span> Clustering with ArchR</a>
<ul>
<li><a href="5.1-clustering-using-seurats-findclusters-function.html#clustering-using-seurats-findclusters-function"><span class="toc-section-number">5.1</span> Clustering using Seurat’s <code>FindClusters()</code> function</a>
<ul>
<li><a href="5.1-clustering-using-seurats-findclusters-function.html#clustering-using-scran"><span class="toc-section-number">5.1.1</span> Clustering using <code>scran</code></a></li>
</ul></li>
</ul></li>
<li><a href="6-single-cell-embeddings.html#single-cell-embeddings"><span class="toc-section-number">6</span> Single-cell Embeddings</a>
<ul>
<li><a href="6.1-uniform-manifold-approximation-and-projection-umap.html#uniform-manifold-approximation-and-projection-umap"><span class="toc-section-number">6.1</span> Uniform Manifold Approximation and Projection (UMAP)</a></li>
<li><a href="6.2-t-stocastic-neighbor-embedding-t-sne.html#t-stocastic-neighbor-embedding-t-sne"><span class="toc-section-number">6.2</span> t-Stocastic Neighbor Embedding (t-SNE)</a></li>
<li><a href="6.3-dimensionality-reduction-after-harmony.html#dimensionality-reduction-after-harmony"><span class="toc-section-number">6.3</span> Dimensionality Reduction After Harmony</a></li>
</ul></li>
<li><a href="7-gene-scores-and-marker-genes-with-archr.html#gene-scores-and-marker-genes-with-archr"><span class="toc-section-number">7</span> Gene Scores and Marker Genes with ArchR</a>
<ul>
<li><a href="7.1-calculating-gene-scores-in-archr.html#calculating-gene-scores-in-archr"><span class="toc-section-number">7.1</span> Calculating Gene Scores in ArchR</a></li>
<li><a href="7.2-identification-of-marker-features.html#identification-of-marker-features"><span class="toc-section-number">7.2</span> Identification of Marker Features</a>
<ul>
<li><a href="7.2-identification-of-marker-features.html#how-does-marker-feature-identification-happen"><span class="toc-section-number">7.2.1</span> How does marker feature identification happen?</a></li>
</ul></li>
<li><a href="7.3-identifying-marker-genes.html#identifying-marker-genes"><span class="toc-section-number">7.3</span> Identifying Marker Genes</a></li>
<li><a href="7.4-visualizing-marker-genes-on-an-embedding.html#visualizing-marker-genes-on-an-embedding"><span class="toc-section-number">7.4</span> Visualizing Marker Genes on an Embedding</a></li>
<li><a href="7.5-marker-genes-imputation-with-magic.html#marker-genes-imputation-with-magic"><span class="toc-section-number">7.5</span> Marker Genes Imputation with MAGIC</a></li>
<li><a href="7.6-track-plotting-with-archrbrowser.html#track-plotting-with-archrbrowser"><span class="toc-section-number">7.6</span> Track Plotting with ArchRBrowser</a></li>
<li><a href="7.7-launching-the-archrbrowser.html#launching-the-archrbrowser"><span class="toc-section-number">7.7</span> Launching the ArchRBrowser</a></li>
</ul></li>
<li><a href="8-defining-cluster-identity-with-scrna-seq.html#defining-cluster-identity-with-scrna-seq"><span class="toc-section-number">8</span> Defining Cluster Identity with scRNA-seq</a>
<ul>
<li><a href="8.1-cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells"><span class="toc-section-number">8.1</span> Cross-platform linkage of scATAC-seq cells with scRNA-seq cells</a>
<ul>
<li><a href="8.1-cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#unconstrained-integration"><span class="toc-section-number">8.1.1</span> Unconstrained Integration</a></li>
<li><a href="8.1-cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#constrained-integration"><span class="toc-section-number">8.1.2</span> Constrained Integration</a></li>
<li><a href="8.1-cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#comparing-unconstrained-and-constrained-integrations"><span class="toc-section-number">8.1.3</span> Comparing Unconstrained and Constrained Integrations</a></li>
</ul></li>
<li><a href="8.2-adding-pseudo-scrna-seq-profiles-for-each-scatac-seq-cell.html#adding-pseudo-scrna-seq-profiles-for-each-scatac-seq-cell"><span class="toc-section-number">8.2</span> Adding Pseudo-scRNA-seq profiles for each scATAC-seq cell</a></li>
<li><a href="8.3-labeling-scatac-seq-clusters-with-scrna-seq-information.html#labeling-scatac-seq-clusters-with-scrna-seq-information"><span class="toc-section-number">8.3</span> Labeling scATAC-seq clusters with scRNA-seq information</a></li>
</ul></li>
<li><a href="9-pseudo-bulk-replicates-in-archr.html#pseudo-bulk-replicates-in-archr"><span class="toc-section-number">9</span> Pseudo-bulk Replicates in ArchR</a>
<ul>
<li><a href="9.1-how-does-archr-make-pseudo-bulk-replicates.html#how-does-archr-make-pseudo-bulk-replicates"><span class="toc-section-number">9.1</span> How Does ArchR Make Pseudo-bulk Replicates?</a>
<ul>
<li><a href="9.1-how-does-archr-make-pseudo-bulk-replicates.html#cluster1"><span class="toc-section-number">9.1.1</span> Cluster1</a></li>
<li><a href="9.1-how-does-archr-make-pseudo-bulk-replicates.html#cluster2"><span class="toc-section-number">9.1.2</span> Cluster2</a></li>
<li><a href="9.1-how-does-archr-make-pseudo-bulk-replicates.html#cluster3"><span class="toc-section-number">9.1.3</span> Cluster3</a></li>
<li><a href="9.1-how-does-archr-make-pseudo-bulk-replicates.html#cluster4"><span class="toc-section-number">9.1.4</span> Cluster4</a></li>
<li><a href="9.1-how-does-archr-make-pseudo-bulk-replicates.html#cluster5"><span class="toc-section-number">9.1.5</span> Cluster5</a></li>
</ul></li>
<li><a href="9.2-making-pseudo-bulk-replicates.html#making-pseudo-bulk-replicates"><span class="toc-section-number">9.2</span> Making Pseudo-bulk Replicates</a></li>
</ul></li>
<li><a href="10-calling-peaks-with-archr.html#calling-peaks-with-archr"><span class="toc-section-number">10</span> Calling Peaks with ArchR</a>
<ul>
<li><a href="10.1-the-iterative-overlap-peak-merging-procedure.html#the-iterative-overlap-peak-merging-procedure"><span class="toc-section-number">10.1</span> The Iterative Overlap Peak Merging Procedure</a>
<ul>
<li><a href="10.1-the-iterative-overlap-peak-merging-procedure.html#fixed-width-vs-variable-width-peaks"><span class="toc-section-number">10.1.1</span> Fixed-width vs Variable-width Peaks</a></li>
<li><a href="10.1-the-iterative-overlap-peak-merging-procedure.html#raw-peak-overlap-using-bedtools-merge"><span class="toc-section-number">10.1.2</span> Raw Peak Overlap Using <em>bedtools merge</em></a></li>
<li><a href="10.1-the-iterative-overlap-peak-merging-procedure.html#clustered-overlap-using-bedtools-cluster"><span class="toc-section-number">10.1.3</span> Clustered Overlap Using <em>bedtools cluster</em></a></li>
<li><a href="10.1-the-iterative-overlap-peak-merging-procedure.html#iterative-overlap-in-archr"><span class="toc-section-number">10.1.4</span> Iterative Overlap In ArchR</a></li>
<li><a href="10.1-the-iterative-overlap-peak-merging-procedure.html#comparison-of-peak-calling-methods"><span class="toc-section-number">10.1.5</span> Comparison of Peak Calling Methods</a></li>
<li><a href="10.1-the-iterative-overlap-peak-merging-procedure.html#so-how-does-this-all-work-in-archr"><span class="toc-section-number">10.1.6</span> So how does this all work in ArchR?</a></li>
<li><a href="10.1-the-iterative-overlap-peak-merging-procedure.html#what-if-i-dont-like-this-iterative-overlap-peak-merging-process"><span class="toc-section-number">10.1.7</span> What if I don’t like this iterative overlap peak merging process?</a></li>
</ul></li>
<li><a href="10.2-calling-peaks-w-macs2.html#calling-peaks-w-macs2"><span class="toc-section-number">10.2</span> Calling Peaks w/ Macs2</a></li>
<li><a href="10.3-calling-peaks-w-tilematrix.html#calling-peaks-w-tilematrix"><span class="toc-section-number">10.3</span> Calling Peaks w/ TileMatrix</a>
<ul>
<li><a href="10.3-calling-peaks-w-tilematrix.html#comparing-the-two-peak-calling-methods"><span class="toc-section-number">10.3.1</span> Comparing the two peak calling methods</a></li>
</ul></li>
<li><a href="10.4-add-peak-matrix.html#add-peak-matrix"><span class="toc-section-number">10.4</span> Add Peak Matrix</a></li>
</ul></li>
<li><a href="11-identifying-marker-peaks-with-archr.html#identifying-marker-peaks-with-archr"><span class="toc-section-number">11</span> Identifying Marker Peaks with ArchR</a>
<ul>
<li><a href="11.1-identifying-marker-peaks-with-archr-1.html#identifying-marker-peaks-with-archr-1"><span class="toc-section-number">11.1</span> Identifying Marker Peaks with ArchR</a></li>
<li><a href="11.2-plotting-marker-peaks-in-archr.html#plotting-marker-peaks-in-archr"><span class="toc-section-number">11.2</span> Plotting Marker Peaks in ArchR</a>
<ul>
<li><a href="11.2-plotting-marker-peaks-in-archr.html#marker-peak-heatmaps"><span class="toc-section-number">11.2.1</span> Marker Peak Heatmaps</a></li>
<li><a href="11.2-plotting-marker-peaks-in-archr.html#marker-peak-ma-and-volcano-plots"><span class="toc-section-number">11.2.2</span> Marker Peak MA and Volcano Plots</a></li>
<li><a href="11.2-plotting-marker-peaks-in-archr.html#marker-peaks-in-browser-tracks"><span class="toc-section-number">11.2.3</span> Marker Peaks in Browser Tracks</a></li>
</ul></li>
<li><a href="11.3-pairwise-testing-between-groups.html#pairwise-testing-between-groups"><span class="toc-section-number">11.3</span> Pairwise Testing Between Groups</a></li>
</ul></li>
<li><a href="12-motif-and-feature-enrichment-with-archr.html#motif-and-feature-enrichment-with-archr"><span class="toc-section-number">12</span> Motif and Feature Enrichment with ArchR</a>
<ul>
<li><a href="12.1-motif-enrichment-in-differential-peaks.html#motif-enrichment-in-differential-peaks"><span class="toc-section-number">12.1</span> Motif Enrichment in Differential Peaks</a></li>
<li><a href="12.2-motif-enrichment-in-marker-peaks.html#motif-enrichment-in-marker-peaks"><span class="toc-section-number">12.2</span> Motif Enrichment in Marker Peaks</a></li>
<li><a href="12.3-archr-enrichment.html#archr-enrichment"><span class="toc-section-number">12.3</span> ArchR Enrichment</a>
<ul>
<li><a href="12.3-archr-enrichment.html#encode-tf-binding-sites"><span class="toc-section-number">12.3.1</span> Encode TF Binding Sites</a></li>
<li><a href="12.3-archr-enrichment.html#bulk-atac-seq"><span class="toc-section-number">12.3.2</span> Bulk ATAC-seq</a></li>
<li><a href="12.3-archr-enrichment.html#codex-tfbs"><span class="toc-section-number">12.3.3</span> Codex TFBS</a></li>
</ul></li>
<li><a href="12.4-custom-enrichment.html#custom-enrichment"><span class="toc-section-number">12.4</span> Custom Enrichment</a></li>
</ul></li>
<li><a href="13-chromvar-deviatons-enrichment-with-archr.html#chromvar-deviatons-enrichment-with-archr"><span class="toc-section-number">13</span> ChromVAR Deviatons Enrichment with ArchR</a>
<ul>
<li><a href="13.1-motif-deviations.html#motif-deviations"><span class="toc-section-number">13.1</span> Motif Deviations</a></li>
<li><a href="13.2-archr-and-custom-deviations.html#archr-and-custom-deviations"><span class="toc-section-number">13.2</span> ArchR and Custom Deviations</a>
<ul>
<li><a href="13.2-archr-and-custom-deviations.html#encode-tfbs"><span class="toc-section-number">13.2.1</span> Encode TFBS</a></li>
<li><a href="13.2-archr-and-custom-deviations.html#bulk-atac-seq-1"><span class="toc-section-number">13.2.2</span> Bulk ATAC-seq</a></li>
<li><a href="13.2-archr-and-custom-deviations.html#custom-deviations"><span class="toc-section-number">13.2.3</span> Custom Deviations</a></li>
</ul></li>
</ul></li>
<li><a href="14-footprinting-with-archr.html#footprinting-with-archr"><span class="toc-section-number">14</span> Footprinting with ArchR</a>
<ul>
<li><a href="14.1-motif-footprinting.html#motif-footprinting"><span class="toc-section-number">14.1</span> Motif Footprinting</a></li>
<li><a href="14.2-normalization-of-footprints-for-tn5-bias.html#normalization-of-footprints-for-tn5-bias"><span class="toc-section-number">14.2</span> Normalization of Footprints for Tn5 Bias</a>
<ul>
<li><a href="14.2-normalization-of-footprints-for-tn5-bias.html#subtracting-the-tn5-bias"><span class="toc-section-number">14.2.1</span> Subtracting the Tn5 Bias</a></li>
<li><a href="14.2-normalization-of-footprints-for-tn5-bias.html#dividing-by-the-tn5-bias"><span class="toc-section-number">14.2.2</span> Dividing by the Tn5 Bias</a></li>
<li><a href="14.2-normalization-of-footprints-for-tn5-bias.html#footprinting-without-normalization-for-tn5-bias"><span class="toc-section-number">14.2.3</span> Footprinting Without Normalization for Tn5 Bias</a></li>
</ul></li>
<li><a href="14.3-feature-footprinting.html#feature-footprinting"><span class="toc-section-number">14.3</span> Feature Footprinting</a></li>
</ul></li>
<li><a href="15-integrative-analysis-with-archr.html#integrative-analysis-with-archr"><span class="toc-section-number">15</span> Integrative Analysis with ArchR</a>
<ul>
<li><a href="15.1-creating-low-overlapping-aggregates-of-cells.html#creating-low-overlapping-aggregates-of-cells"><span class="toc-section-number">15.1</span> Creating Low-Overlapping Aggregates of Cells</a></li>
<li><a href="15.2-co-accessibility-with-archr.html#co-accessibility-with-archr"><span class="toc-section-number">15.2</span> Co-accessibility with ArchR</a>
<ul>
<li><a href="15.2-co-accessibility-with-archr.html#plotting-browser-tracks-of-co-accessibility"><span class="toc-section-number">15.2.1</span> Plotting browser tracks of Co-accessibility</a></li>
</ul></li>
<li><a href="15.3-peak2genelinkage-with-archr.html#peak2genelinkage-with-archr"><span class="toc-section-number">15.3</span> Peak2GeneLinkage with ArchR</a>
<ul>
<li><a href="15.3-peak2genelinkage-with-archr.html#plotting-browser-tracks-with-peak-to-gene-links"><span class="toc-section-number">15.3.1</span> Plotting browser tracks with peak-to-gene links</a></li>
<li><a href="15.3-peak2genelinkage-with-archr.html#plotting-a-heatmap-of-peak-to-gene-links"><span class="toc-section-number">15.3.2</span> Plotting a heatmap of peak-to-gene links</a></li>
</ul></li>
<li><a href="15.4-identification-of-positive-tf-regulators.html#identification-of-positive-tf-regulators"><span class="toc-section-number">15.4</span> Identification of Positive TF-Regulators</a>
<ul>
<li><a href="15.4-identification-of-positive-tf-regulators.html#step-1.-identify-deviant-tf-motifs"><span class="toc-section-number">15.4.1</span> Step 1. Identify Deviant TF Motifs</a></li>
<li><a href="15.4-identification-of-positive-tf-regulators.html#step-2.-identify-correlated-tf-motifs-and-tf-gene-scoreexpression"><span class="toc-section-number">15.4.2</span> Step 2. Identify Correlated TF Motifs and TF Gene Score/Expression</a></li>
</ul></li>
</ul></li>
<li><a href="16-trajectory-analysis-with-archr.html#trajectory-analysis-with-archr"><span class="toc-section-number">16</span> Trajectory Analysis with ArchR</a>
<ul>
<li><a href="16.1-myeloid-trajectory---monocyte-differentiation.html#myeloid-trajectory---monocyte-differentiation"><span class="toc-section-number">16.1</span> Myeloid Trajectory - Monocyte Differentiation</a>
<ul>
<li><a href="16.1-myeloid-trajectory---monocyte-differentiation.html#pseudo-time-umaps-and-individual-feature-plots"><span class="toc-section-number">16.1.1</span> Pseudo-time UMAPs and individual feature plots</a></li>
<li><a href="16.1-myeloid-trajectory---monocyte-differentiation.html#pseudo-time-heatmaps"><span class="toc-section-number">16.1.2</span> Pseudo-time heatmaps</a></li>
<li><a href="16.1-myeloid-trajectory---monocyte-differentiation.html#integrative-pseudo-time-analyses"><span class="toc-section-number">16.1.3</span> Integrative pseudo-time analyses</a></li>
</ul></li>
<li><a href="16.2-lymphoid-trajectory---b-cell-cifferentiation.html#lymphoid-trajectory---b-cell-cifferentiation"><span class="toc-section-number">16.2</span> Lymphoid Trajectory - B Cell Cifferentiation</a>
<ul>
<li><a href="16.2-lymphoid-trajectory---b-cell-cifferentiation.html#pseudo-time-umaps-and-individual-feature-plots-1"><span class="toc-section-number">16.2.1</span> Pseudo-time UMAPs and individual feature plots</a></li>
<li><a href="16.2-lymphoid-trajectory---b-cell-cifferentiation.html#pseudo-time-heatmaps-1"><span class="toc-section-number">16.2.2</span> Pseudo-time heatmaps</a></li>
<li><a href="16.2-lymphoid-trajectory---b-cell-cifferentiation.html#integrative-pseudo-time-analyses-1"><span class="toc-section-number">16.2.3</span> Integrative pseudo-time analyses</a></li>
</ul></li>
</ul></li>
<li><a href="17-plot-aesthetics-in-archr.html#plot-aesthetics-in-archr"><span class="toc-section-number">17</span> Plot aesthetics in ArchR</a></li>
<li><a href="18-manage-archrs-dependencies.html#manage-archrs-dependencies"><span class="toc-section-number">18</span> Manage ArchR’s Dependencies</a>
<ul>
<li><a href="18.1-start-container-locally.html#start-container-locally"><span class="toc-section-number">18.1</span> Start Container Locally</a></li>
</ul></li>
<li><a href="19-renv-solo-locally.html#renv-solo-locally"><span class="toc-section-number">19</span> Renv solo locally</a>
<ul>
<li><a href="19.1-general-notes-on-renv.html#general-notes-on-renv"><span class="toc-section-number">19.1</span> General Notes on renv</a></li>
<li><a href="19.2-general-notes-on-docker.html#general-notes-on-docker"><span class="toc-section-number">19.2</span> General Notes on Docker</a></li>
</ul></li>
<li><a href="20-session-information.html#session-information"><span class="toc-section-number">20</span> Session Information</a></li>
</ul>
</div>
</div>
</div>
<div class="row">
<div class="col-sm-12">
<div id="renv-solo-locally" class="section level1" number="19">
<h1><span class="header-section-number">Chapter 19</span> Renv solo locally</h1>
<ol style="list-style-type: decimal">
<li><p><code>cd</code> into the directory where you have your data</p></li>
<li><p>Run
<code>wget -O renv.lock "https://raw.githubusercontent.com/rcorces/ArchR_docker/main/renv.lock?token=AOHUDISVOOHAHDXW7AJBJFLB3XHEM"</code></p></li>
<li><p>Install <code>renv</code> in R</p></li>
</ol>
<p><code>install.packages('renv')</code></p>
<ol style="list-style-type: decimal">
<li>Check your working directory in R.</li>
</ol>
<p>Use <code>getwd()</code></p>
<ol style="list-style-type: decimal">
<li><p>Run <code>renv::init()</code></p></li>
<li><p>Wait for <code>renv</code> to install your project library. Output should be:</p></li>
</ol>
<pre><code>Restoring project ...
The following package(s) will be updated:</code></pre>
<p>Followed by a list of ArchR’s dependencies.
You may also see:</p>
<pre><code>This project already has a lockfile. What would you like to do?
1: Restore the project from the lockfile.
2: Discard the lockfile and re-initialize the project.
3: Activate the project without snapshotting or installing any packages.
4: Abort project initialization.</code></pre>
<p>In this case, write 1.</p>
<ol style="list-style-type: decimal">
<li>Restart your R session. No need to save your workspace. If you are in RStudio: command/ctrl + shift + F10. You can also use <code>.rs.restartR().</code></li>
</ol>
</div>
<p style="text-align: center;">
<a href="18.1-start-container-locally.html"><button class="btn btn-default">Previous</button></a>
<a href="19.1-general-notes-on-renv.html"><button class="btn btn-default">Next</button></a>
</p>
</div>
</div>


</div>

<script>

// add bootstrap table styles to pandoc tables
$(document).ready(function () {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
});

</script>

</body>
</html>
