<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>12.2 ArchR Deviations | ArchR: Robust and scaleable analysis of single-cell chromatin accessibility data.</title>
  <meta name="description" content="A guide to ArchR" />
  <meta name="generator" content="bookdown 0.17 and GitBook 2.6.7" />

  <meta property="og:title" content="12.2 ArchR Deviations | ArchR: Robust and scaleable analysis of single-cell chromatin accessibility data." />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="A guide to ArchR" />
  <meta name="github-repo" content="GreenleafLab/ArchR" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="12.2 ArchR Deviations | ArchR: Robust and scaleable analysis of single-cell chromatin accessibility data." />
  
  <meta name="twitter:description" content="A guide to ArchR" />
  

<meta name="author" content="Jeffrey Granja and Ryan Corces" />


<meta name="date" content="2020-04-18" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="images/favicon.ico" type="image/x-icon" />
<link rel="prev" href="motif-deviations.html"/>
<link rel="next" href="custom-deviations.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<link href="libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="../index.html">Back to ArchR Home</a></li>

<li class="divider"></li>
<li><a href="index.html#section"><img src="images/ArchR_Logo_Integrated.png" alt="" width="200"></a></li>
<li class="chapter" data-level="1" data-path="getting-started-with-archr.html"><a href="getting-started-with-archr.html"><i class="fa fa-check"></i><b>1</b> Getting Started with ArchR</a>
<ul>
<li class="chapter" data-level="1.1" data-path="a-brief-primer-on-atac-seq-terminology.html"><a href="a-brief-primer-on-atac-seq-terminology.html"><i class="fa fa-check"></i><b>1.1</b> A Brief Primer on ATAC-seq Terminology</a></li>
<li class="chapter" data-level="1.2" data-path="why-use-archr.html"><a href="why-use-archr.html"><i class="fa fa-check"></i><b>1.2</b> Why use ArchR?</a></li>
<li class="chapter" data-level="1.3" data-path="what-is-an-arrow-file-archrproject.html"><a href="what-is-an-arrow-file-archrproject.html"><i class="fa fa-check"></i><b>1.3</b> What is an <code>Arrow file</code> / <code>ArchRProject</code>?</a></li>
<li class="chapter" data-level="1.4" data-path="input-file-types-in-archr.html"><a href="input-file-types-in-archr.html"><i class="fa fa-check"></i><b>1.4</b> Input File Types in ArchR</a></li>
<li class="chapter" data-level="1.5" data-path="getting-set-up.html"><a href="getting-set-up.html"><i class="fa fa-check"></i><b>1.5</b> Getting Set Up</a></li>
<li class="chapter" data-level="1.6" data-path="setting-a-genome-and-geneannotation.html"><a href="setting-a-genome-and-geneannotation.html"><i class="fa fa-check"></i><b>1.6</b> Setting a Genome and GeneAnnotation</a></li>
<li class="chapter" data-level="1.7" data-path="creating-arrow-files.html"><a href="creating-arrow-files.html"><i class="fa fa-check"></i><b>1.7</b> Creating Arrow Files</a></li>
<li class="chapter" data-level="1.8" data-path="session-information.html"><a href="session-information.html"><i class="fa fa-check"></i><b>1.8</b> Session Information</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="doublet-inference-with-archr.html"><a href="doublet-inference-with-archr.html"><i class="fa fa-check"></i><b>2</b> Doublet Inference with ArchR</a>
<ul>
<li class="chapter" data-level="2.1" data-path="inferring-scatac-seq-doublets-with-archr.html"><a href="inferring-scatac-seq-doublets-with-archr.html"><i class="fa fa-check"></i><b>2.1</b> Inferring scATAC-seq Doublets with ArchR</a></li>
<li class="chapter" data-level="2.2" data-path="session-information-1.html"><a href="session-information-1.html"><i class="fa fa-check"></i><b>2.2</b> Session Information</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="creating-an-archrproject.html"><a href="creating-an-archrproject.html"><i class="fa fa-check"></i><b>3</b> Creating an ArchRProject</a>
<ul>
<li class="chapter" data-level="3.1" data-path="creating-an-archrproject-1.html"><a href="creating-an-archrproject-1.html"><i class="fa fa-check"></i><b>3.1</b> Creating An ArchRProject</a></li>
<li class="chapter" data-level="3.2" data-path="manipulating-an-archrproject.html"><a href="manipulating-an-archrproject.html"><i class="fa fa-check"></i><b>3.2</b> Manipulating An ArchRProject</a></li>
<li class="chapter" data-level="3.3" data-path="plotting-sample-statistics-from-an-archrproject.html"><a href="plotting-sample-statistics-from-an-archrproject.html"><i class="fa fa-check"></i><b>3.3</b> Plotting Sample Statistics from an ArchRProject</a></li>
<li class="chapter" data-level="3.4" data-path="plotting-sample-fragment-size-distribution-and-tss-enrichment-profiles-.html"><a href="plotting-sample-fragment-size-distribution-and-tss-enrichment-profiles-.html"><i class="fa fa-check"></i><b>3.4</b> Plotting Sample Fragment Size Distribution and TSS Enrichment Profiles.</a></li>
<li class="chapter" data-level="3.5" data-path="filtering-doublets-from-an-archrproject.html"><a href="filtering-doublets-from-an-archrproject.html"><i class="fa fa-check"></i><b>3.5</b> Filtering Doublets from an ArchRProject</a></li>
<li class="chapter" data-level="3.6" data-path="session-information-2.html"><a href="session-information-2.html"><i class="fa fa-check"></i><b>3.6</b> Session Information</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="dimensionality-reduction-with-archr.html"><a href="dimensionality-reduction-with-archr.html"><i class="fa fa-check"></i><b>4</b> Dimensionality Reduction with ArchR</a>
<ul>
<li class="chapter" data-level="4.1" data-path="archrs-lsi-implementation.html"><a href="archrs-lsi-implementation.html"><i class="fa fa-check"></i><b>4.1</b> ArchR’s LSI Implementation</a></li>
<li class="chapter" data-level="4.2" data-path="iterative-latent-semantic-indexing-lsi.html"><a href="iterative-latent-semantic-indexing-lsi.html"><i class="fa fa-check"></i><b>4.2</b> Iterative Latent Semantic Indexing (LSI)</a></li>
<li class="chapter" data-level="4.3" data-path="estimated-lsi.html"><a href="estimated-lsi.html"><i class="fa fa-check"></i><b>4.3</b> Estimated LSI</a></li>
<li class="chapter" data-level="4.4" data-path="batch-effect-correction-wtih-harmony.html"><a href="batch-effect-correction-wtih-harmony.html"><i class="fa fa-check"></i><b>4.4</b> Batch Effect Correction wtih Harmony</a></li>
<li class="chapter" data-level="4.5" data-path="session-information-3.html"><a href="session-information-3.html"><i class="fa fa-check"></i><b>4.5</b> Session Information</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="clustering-with-archr.html"><a href="clustering-with-archr.html"><i class="fa fa-check"></i><b>5</b> Clustering with ArchR</a>
<ul>
<li class="chapter" data-level="5.1" data-path="clustering-using-seurats-findclusters-function.html"><a href="clustering-using-seurats-findclusters-function.html"><i class="fa fa-check"></i><b>5.1</b> Clustering using Seurat’s <code>FindClusters()</code> function</a></li>
<li class="chapter" data-level="5.2" data-path="session-information-4.html"><a href="session-information-4.html"><i class="fa fa-check"></i><b>5.2</b> Session Information</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="single-cell-embeddings.html"><a href="single-cell-embeddings.html"><i class="fa fa-check"></i><b>6</b> Single Cell Embeddings</a>
<ul>
<li class="chapter" data-level="6.1" data-path="umap.html"><a href="umap.html"><i class="fa fa-check"></i><b>6.1</b> UMAP</a></li>
<li class="chapter" data-level="6.2" data-path="tsne.html"><a href="tsne.html"><i class="fa fa-check"></i><b>6.2</b> tSNE</a></li>
<li class="chapter" data-level="6.3" data-path="umap-of-previous-reduced-dimensions.html"><a href="umap-of-previous-reduced-dimensions.html"><i class="fa fa-check"></i><b>6.3</b> UMAP of previous Reduced Dimensions</a></li>
<li class="chapter" data-level="6.4" data-path="tsne-of-previous-reduced-dimensions.html"><a href="tsne-of-previous-reduced-dimensions.html"><i class="fa fa-check"></i><b>6.4</b> tSNE of previous Reduced Dimensions</a></li>
<li class="chapter" data-level="6.5" data-path="session-information-5.html"><a href="session-information-5.html"><i class="fa fa-check"></i><b>6.5</b> Session Information</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="marker-genes-with-archr.html"><a href="marker-genes-with-archr.html"><i class="fa fa-check"></i><b>7</b> Marker Genes with ArchR</a>
<ul>
<li class="chapter" data-level="7.1" data-path="identification-of-marker-genes.html"><a href="identification-of-marker-genes.html"><i class="fa fa-check"></i><b>7.1</b> Identification of Marker Genes</a></li>
<li class="chapter" data-level="7.2" data-path="marker-genes.html"><a href="marker-genes.html"><i class="fa fa-check"></i><b>7.2</b> Marker Genes</a></li>
<li class="chapter" data-level="7.3" data-path="marker-genes-imputation-with-magic.html"><a href="marker-genes-imputation-with-magic.html"><i class="fa fa-check"></i><b>7.3</b> Marker Genes Imputation with Magic</a></li>
<li class="chapter" data-level="7.4" data-path="track-plotting-with-archrbrowser.html"><a href="track-plotting-with-archrbrowser.html"><i class="fa fa-check"></i><b>7.4</b> Track Plotting with ArchRBrowser</a></li>
<li class="chapter" data-level="7.5" data-path="launching-the-archrbrowser.html"><a href="launching-the-archrbrowser.html"><i class="fa fa-check"></i><b>7.5</b> Launching the ArchRBrowser</a></li>
<li class="chapter" data-level="7.6" data-path="session-information-6.html"><a href="session-information-6.html"><i class="fa fa-check"></i><b>7.6</b> Session Information</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="labeling-clusters-with-scrna.html"><a href="labeling-clusters-with-scrna.html"><i class="fa fa-check"></i><b>8</b> Labeling Clusters with scRNA</a>
<ul>
<li class="chapter" data-level="8.1" data-path="labeling-scatac-cells-with-scrna-clusters.html"><a href="labeling-scatac-cells-with-scrna-clusters.html"><i class="fa fa-check"></i><b>8.1</b> Labeling scATAC cells with scRNA clusters</a></li>
<li class="chapter" data-level="8.2" data-path="adding-pseudo-scrna-profiles-for-each-scatac-cell.html"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html"><i class="fa fa-check"></i><b>8.2</b> Adding Pseudo-scRNA profiles for each scATAC cell</a></li>
<li class="chapter" data-level="8.3" data-path="labeling-scatac-clusters-with-scrna.html"><a href="labeling-scatac-clusters-with-scrna.html"><i class="fa fa-check"></i><b>8.3</b> Labeling scATAC clusters with scRNA</a></li>
<li class="chapter" data-level="8.4" data-path="session-information-7.html"><a href="session-information-7.html"><i class="fa fa-check"></i><b>8.4</b> Session Information</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="calling-peaks-with-archr.html"><a href="calling-peaks-with-archr.html"><i class="fa fa-check"></i><b>9</b> Calling Peaks with ArchR</a>
<ul>
<li class="chapter" data-level="9.1" data-path="calling-peaks-w-macs2.html"><a href="calling-peaks-w-macs2.html"><i class="fa fa-check"></i><b>9.1</b> Calling Peaks w/ Macs2</a></li>
<li class="chapter" data-level="9.2" data-path="calling-peaks-w-tilematrix.html"><a href="calling-peaks-w-tilematrix.html"><i class="fa fa-check"></i><b>9.2</b> Calling Peaks w/ TileMatrix</a></li>
<li class="chapter" data-level="9.3" data-path="add-peak-matrix.html"><a href="add-peak-matrix.html"><i class="fa fa-check"></i><b>9.3</b> Add Peak Matrix</a></li>
<li class="chapter" data-level="9.4" data-path="session-information-8.html"><a href="session-information-8.html"><i class="fa fa-check"></i><b>9.4</b> Session Information</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="identifying-marker-peaks-with-archr.html"><a href="identifying-marker-peaks-with-archr.html"><i class="fa fa-check"></i><b>10</b> Identifying Marker Peaks with ArchR</a>
<ul>
<li class="chapter" data-level="10.1" data-path="identifying-marker-peaks-with-archr-1.html"><a href="identifying-marker-peaks-with-archr-1.html"><i class="fa fa-check"></i><b>10.1</b> Identifying Marker Peaks with ArchR</a></li>
<li class="chapter" data-level="10.2" data-path="pairwise-testing-groups.html"><a href="pairwise-testing-groups.html"><i class="fa fa-check"></i><b>10.2</b> Pairwise Testing Groups</a></li>
<li class="chapter" data-level="10.3" data-path="session-information-9.html"><a href="session-information-9.html"><i class="fa fa-check"></i><b>10.3</b> Session Information</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="peak-annotation-enrichment-with-archr.html"><a href="peak-annotation-enrichment-with-archr.html"><i class="fa fa-check"></i><b>11</b> Peak Annotation Enrichment with ArchR</a>
<ul>
<li class="chapter" data-level="11.1" data-path="motif-enrichment.html"><a href="motif-enrichment.html"><i class="fa fa-check"></i><b>11.1</b> Motif Enrichment</a></li>
<li class="chapter" data-level="11.2" data-path="archr-enrichment.html"><a href="archr-enrichment.html"><i class="fa fa-check"></i><b>11.2</b> ArchR Enrichment</a></li>
<li class="chapter" data-level="11.3" data-path="custom-enrichment.html"><a href="custom-enrichment.html"><i class="fa fa-check"></i><b>11.3</b> Custom Enrichment</a></li>
<li class="chapter" data-level="11.4" data-path="session-information-10.html"><a href="session-information-10.html"><i class="fa fa-check"></i><b>11.4</b> Session Information</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="chromvar-deviatons-enrichment-with-archr.html"><a href="chromvar-deviatons-enrichment-with-archr.html"><i class="fa fa-check"></i><b>12</b> ChromVAR Deviatons Enrichment with ArchR</a>
<ul>
<li class="chapter" data-level="12.1" data-path="motif-deviations.html"><a href="motif-deviations.html"><i class="fa fa-check"></i><b>12.1</b> Motif Deviations</a></li>
<li class="chapter" data-level="12.2" data-path="archr-deviations.html"><a href="archr-deviations.html"><i class="fa fa-check"></i><b>12.2</b> ArchR Deviations</a></li>
<li class="chapter" data-level="12.3" data-path="custom-deviations.html"><a href="custom-deviations.html"><i class="fa fa-check"></i><b>12.3</b> Custom Deviations</a></li>
<li class="chapter" data-level="12.4" data-path="session-information-11.html"><a href="session-information-11.html"><i class="fa fa-check"></i><b>12.4</b> Session Information</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="footprinting-with-archr.html"><a href="footprinting-with-archr.html"><i class="fa fa-check"></i><b>13</b> Footprinting with ArchR</a>
<ul>
<li class="chapter" data-level="13.1" data-path="motif-footprinting.html"><a href="motif-footprinting.html"><i class="fa fa-check"></i><b>13.1</b> Motif Footprinting</a></li>
<li class="chapter" data-level="13.2" data-path="feature-footprinting.html"><a href="feature-footprinting.html"><i class="fa fa-check"></i><b>13.2</b> Feature Footprinting</a></li>
<li class="chapter" data-level="13.3" data-path="session-information-12.html"><a href="session-information-12.html"><i class="fa fa-check"></i><b>13.3</b> Session Information</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="integrative-analysis-with-archr.html"><a href="integrative-analysis-with-archr.html"><i class="fa fa-check"></i><b>14</b> Integrative Analysis with ArchR</a>
<ul>
<li class="chapter" data-level="14.1" data-path="co-accessibility-with-archr.html"><a href="co-accessibility-with-archr.html"><i class="fa fa-check"></i><b>14.1</b> Co-accessibility with ArchR</a></li>
<li class="chapter" data-level="14.2" data-path="peak2genelinkage-with-archr.html"><a href="peak2genelinkage-with-archr.html"><i class="fa fa-check"></i><b>14.2</b> Peak2GeneLinkage with ArchR</a></li>
<li class="chapter" data-level="14.3" data-path="identification-of-positive-tf-regulators.html"><a href="identification-of-positive-tf-regulators.html"><i class="fa fa-check"></i><b>14.3</b> Identification of Positive TF-Regulators</a></li>
<li class="chapter" data-level="14.4" data-path="session-information-13.html"><a href="session-information-13.html"><i class="fa fa-check"></i><b>14.4</b> Session Information</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="trajectory-analysis-with-archr.html"><a href="trajectory-analysis-with-archr.html"><i class="fa fa-check"></i><b>15</b> Trajectory Analysis with ArchR</a>
<ul>
<li class="chapter" data-level="15.1" data-path="myeloid-trajectory.html"><a href="myeloid-trajectory.html"><i class="fa fa-check"></i><b>15.1</b> Myeloid Trajectory</a></li>
<li class="chapter" data-level="15.2" data-path="lymphoid-trajectory.html"><a href="lymphoid-trajectory.html"><i class="fa fa-check"></i><b>15.2</b> Lymphoid Trajectory</a></li>
<li class="chapter" data-level="15.3" data-path="session-information-14.html"><a href="session-information-14.html"><i class="fa fa-check"></i><b>15.3</b> Session Information</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://bookdown.org" target="_blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">ArchR: Robust and scaleable analysis of single-cell chromatin accessibility data.</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="archr-deviations" class="section level2" number="12.2">
<h2><span class="header-section-number">12.2</span> ArchR Deviations</h2>
<div id="encode-tfbs-1" class="section level3" number="12.2.1">
<h3><span class="header-section-number">12.2.1</span> Encode TFBS</h3>
<p>If you have not added motif annotations</p>
<div class="sourceCode" id="cb265"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb265-1"><a href="archr-deviations.html#cb265-1"></a><span class="cf">if</span>(<span class="st">&quot;EncodeTFBS&quot;</span> <span class="op">%ni%</span><span class="st"> </span><span class="kw">names</span>(projHeme5<span class="op">@</span>peakAnnotation)){</span>
<span id="cb265-2"><a href="archr-deviations.html#cb265-2"></a>    projHeme5 &lt;-<span class="st"> </span><span class="kw">addArchRAnnotations</span>(<span class="dt">ArchRProj =</span> projHeme5, <span class="dt">collection =</span> <span class="st">&quot;EncodeTFBS&quot;</span>)</span>
<span id="cb265-3"><a href="archr-deviations.html#cb265-3"></a>}</span></code></pre></div>
<p>Add ChromVAR Deviations for Motifs</p>
<div class="sourceCode" id="cb266"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb266-1"><a href="archr-deviations.html#cb266-1"></a>projHeme5 &lt;-<span class="st"> </span><span class="kw">addDeviationsMatrix</span>(</span>
<span id="cb266-2"><a href="archr-deviations.html#cb266-2"></a>  <span class="dt">ArchRProj =</span> projHeme5, </span>
<span id="cb266-3"><a href="archr-deviations.html#cb266-3"></a>  <span class="dt">peakAnnotation =</span> <span class="st">&quot;EncodeTFBS&quot;</span>,</span>
<span id="cb266-4"><a href="archr-deviations.html#cb266-4"></a>  <span class="dt">force =</span> <span class="ot">TRUE</span></span>
<span id="cb266-5"><a href="archr-deviations.html#cb266-5"></a>)</span></code></pre></div>
<blockquote>
<p>## Using Previous Background Peaks!<br />
## ArchR logging to : ArchRLogs/ArchR-addDeviationsMatrix-10374737762a7-Date-2020-04-15_Time-10-59-20.log<br />
## If there is an issue, please report to github with logFile!<br />
## NULL<br />
## 2020-04-15 10:59:26 : Batch Execution w/ safelapply!, 0 mins elapsed.<br />
## ###########<br />
## 2020-04-15 11:08:11 : Completed Computing Deviations!, 8.864 mins elapsed.<br />
## ###########<br />
## ArchR logging successful to : ArchRLogs/ArchR-addDeviationsMatrix-10374737762a7-Date-2020-04-15_Time-10-59-20.log</p>
</blockquote>
<p>Plot Variable Motif Annotations</p>
<div class="sourceCode" id="cb267"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb267-1"><a href="archr-deviations.html#cb267-1"></a>plotVarDev &lt;-<span class="st"> </span><span class="kw">getVarDeviations</span>(projHeme5, <span class="dt">plot =</span> <span class="ot">TRUE</span>, <span class="dt">name =</span> <span class="st">&quot;EncodeTFBSMatrix&quot;</span>)</span></code></pre></div>
<blockquote>
<p>## DataFrame with 6 rows and 6 columns<br />
## seqnames idx name combinedVars<br />
## <Rle> <array> <array> <numeric><br />
## f222 z 222 222.GATA2_S-K562… 13.6382687713858<br />
## f41 z 41 41.EZH2_39-NHEK… 12.9150554355261<br />
## f542 z 542 542.TAL1_SC-K562… 11.5490744327685<br />
## f584 z 584 584.GATA_1-PBDEFetal… 10.7653053700024<br />
## f498 z 498 498.GATA_2-K562… 10.2384173319845<br />
## f44 z 44 44.EZH2_39-NHLF… 9.92292044310219<br />
## combinedMeans rank<br />
## <numeric> <integer><br />
## f222 -0.0312421757573656 1<br />
## f41 0.113221166775704 2<br />
## f542 -0.0155915927183644 3<br />
## f584 -0.00997025022971285 4<br />
## f498 -0.0317577572129743 5<br />
## f44 0.0707460230932926 6</p>
</blockquote>
<div class="sourceCode" id="cb268"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb268-1"><a href="archr-deviations.html#cb268-1"></a>plotVarDev</span></code></pre></div>
<blockquote>
<p>## Warning: Removed 7 rows containing missing values (geom_point).</p>
</blockquote>
<p><img src="images/HemeWalkthrough/PNG/Variable-EncodeTFBS-Deviation-Scores_1.png" width="500" height="500" /></p>
<p>We can then plot this as a nice PDF</p>
<div class="sourceCode" id="cb269"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb269-1"><a href="archr-deviations.html#cb269-1"></a><span class="kw">plotPDF</span>(plotVarDev, <span class="dt">name =</span> <span class="st">&quot;Variable-EncodeTFBS-Deviation-Scores&quot;</span>, <span class="dt">width =</span> <span class="dv">5</span>, <span class="dt">height =</span> <span class="dv">5</span>, <span class="dt">ArchRProj =</span> projHeme5, <span class="dt">addDOC =</span> <span class="ot">FALSE</span>)</span></code></pre></div>
<blockquote>
<p>## [1] “plotting ggplot!”<br />
## Warning: Removed 7 rows containing missing values (geom_point).<br />
## Warning: Removed 7 rows containing missing values (geom_point).<br />
## [1] 0</p>
</blockquote>
<p>If we want to get TF feature names for downstream plotting</p>
<div class="sourceCode" id="cb270"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb270-1"><a href="archr-deviations.html#cb270-1"></a>tfs &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;GATA_1&quot;</span>, <span class="st">&quot;CEBPB&quot;</span>, <span class="st">&quot;EBF1&quot;</span>, <span class="st">&quot;IRF4&quot;</span>, <span class="st">&quot;TBX21&quot;</span>, <span class="st">&quot;PAX5&quot;</span>)</span>
<span id="cb270-2"><a href="archr-deviations.html#cb270-2"></a><span class="kw">getFeatures</span>(projHeme5, <span class="dt">select =</span> <span class="kw">paste</span>(tfs, <span class="dt">collapse=</span><span class="st">&quot;|&quot;</span>), <span class="dt">useMatrix =</span> <span class="st">&quot;EncodeTFBSMatrix&quot;</span>)</span></code></pre></div>
<blockquote>
<p>## [1] “z:584.GATA_1-PBDEFetal…” “z:582.GATA_1-PBDE…”<br />
## [3] “z:497.GATA_1-K562…” “z:477.CEBPB-K562…”<br />
## [5] “z:462.CEBPB-IMR90…” “z:427.CEBPB-HepG2…”<br />
## [7] “z:426.CEBPB-HepG2…” “z:379.CEBPB-HeLa_S3…”<br />
## [9] “z:344.CEBPB-H1_hESC…” “z:293.EBF1_SC-GM12878…”<br />
## [11] “z:278.CEBPB-A549…” “z:213.CEBPB_S-K562…”<br />
## [13] “z:173.CEBPB_S-HepG2…” “z:130.PAX5_C2-GM12892…”<br />
## [15] “z:123.PAX5_C2-GM12891…” “z:102.PAX5_N1-GM12878…”<br />
## [17] “z:101.PAX5_C2-GM12878…” “z:93.IRF4_SC-GM12878…”<br />
## [19] “z:87.EBF1_SC-GM12878…” “z:86.CEBPB_S-GM12878…”<br />
## [21] “deviations:584.GATA_1-PBDEFetal…” “deviations:582.GATA_1-PBDE…”<br />
## [23] “deviations:497.GATA_1-K562…” “deviations:477.CEBPB-K562…”<br />
## [25] “deviations:462.CEBPB-IMR90…” “deviations:427.CEBPB-HepG2…”<br />
## [27] “deviations:426.CEBPB-HepG2…” “deviations:379.CEBPB-HeLa_S3…”<br />
## [29] “deviations:344.CEBPB-H1_hESC…” “deviations:293.EBF1_SC-GM12878…”<br />
## [31] “deviations:278.CEBPB-A549…” “deviations:213.CEBPB_S-K562…”<br />
## [33] “deviations:173.CEBPB_S-HepG2…” “deviations:130.PAX5_C2-GM12892…”<br />
## [35] “deviations:123.PAX5_C2-GM12891…” “deviations:102.PAX5_N1-GM12878…”<br />
## [37] “deviations:101.PAX5_C2-GM12878…” “deviations:93.IRF4_SC-GM12878…”<br />
## [39] “deviations:87.EBF1_SC-GM12878…” “deviations:86.CEBPB_S-GM12878…”</p>
</blockquote>
<p>To get deviation z-scores</p>
<div class="sourceCode" id="cb271"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb271-1"><a href="archr-deviations.html#cb271-1"></a>markerTFs &lt;-<span class="st"> </span><span class="kw">getFeatures</span>(projHeme5, <span class="dt">select =</span> <span class="kw">paste</span>(tfs, <span class="dt">collapse=</span><span class="st">&quot;|&quot;</span>), <span class="dt">useMatrix =</span> <span class="st">&quot;EncodeTFBSMatrix&quot;</span>)</span>
<span id="cb271-2"><a href="archr-deviations.html#cb271-2"></a>markerTFs &lt;-<span class="st"> </span><span class="kw">sort</span>(<span class="kw">grep</span>(<span class="st">&quot;z:&quot;</span>, markerTFs, <span class="dt">value =</span> <span class="ot">TRUE</span>))</span>
<span id="cb271-3"><a href="archr-deviations.html#cb271-3"></a>TFnames &lt;-<span class="st"> </span>stringr<span class="op">::</span><span class="kw">str_split</span>(stringr<span class="op">::</span><span class="kw">str_split</span>(markerTFs, <span class="dt">pattern =</span> <span class="st">&quot;</span><span class="ch">\\</span><span class="st">.&quot;</span>, <span class="dt">simplify=</span><span class="ot">TRUE</span>)[,<span class="dv">2</span>], <span class="dt">pattern =</span> <span class="st">&quot;-&quot;</span>, <span class="dt">simplify =</span> <span class="ot">TRUE</span>)[,<span class="dv">1</span>]</span>
<span id="cb271-4"><a href="archr-deviations.html#cb271-4"></a>markerTFs &lt;-<span class="st"> </span>markerTFs[<span class="op">!</span><span class="kw">duplicated</span>(TFnames)]</span>
<span id="cb271-5"><a href="archr-deviations.html#cb271-5"></a>markerTFs</span></code></pre></div>
<blockquote>
<p>## [1] “z:101.PAX5_C2-GM12878…” “z:102.PAX5_N1-GM12878…”<br />
## [3] “z:173.CEBPB_S-HepG2…” “z:278.CEBPB-A549…”<br />
## [5] “z:293.EBF1_SC-GM12878…” “z:497.GATA_1-K562…”<br />
## [7] “z:93.IRF4_SC-GM12878…”</p>
</blockquote>
<div class="sourceCode" id="cb272"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb272-1"><a href="archr-deviations.html#cb272-1"></a>p &lt;-<span class="st"> </span><span class="kw">plotEmbedding</span>(</span>
<span id="cb272-2"><a href="archr-deviations.html#cb272-2"></a>    <span class="dt">ArchRProj =</span> projHeme5, </span>
<span id="cb272-3"><a href="archr-deviations.html#cb272-3"></a>    <span class="dt">colorBy =</span> <span class="st">&quot;EncodeTFBSMatrix&quot;</span>, </span>
<span id="cb272-4"><a href="archr-deviations.html#cb272-4"></a>    <span class="dt">name =</span> markerTFs, </span>
<span id="cb272-5"><a href="archr-deviations.html#cb272-5"></a>    <span class="dt">embedding =</span> <span class="st">&quot;UMAP&quot;</span>,</span>
<span id="cb272-6"><a href="archr-deviations.html#cb272-6"></a>    <span class="dt">imputeWeights =</span> <span class="kw">getImputeWeights</span>(projHeme5)</span>
<span id="cb272-7"><a href="archr-deviations.html#cb272-7"></a>)</span></code></pre></div>
<blockquote>
<p>## Getting ImputeWeights<br />
## ArchR logging to : ArchRLogs/ArchR-plotEmbedding-103747b681a1c-Date-2020-04-15_Time-11-08-26.log<br />
## If there is an issue, please report to github with logFile!<br />
## Getting UMAP Embedding<br />
## ColorBy = EncodeTFBSMatrix<br />
## Getting Matrix Values…<br />
## Getting Matrix Values…<br />
##<br />
## Imputing Matrix<br />
## Using weights on disk<br />
## Using weights on disk<br />
## Plotting Embedding<br />
## 1 2 3 4 5 6 7<br />
## ArchR logging successful to : ArchRLogs/ArchR-plotEmbedding-103747b681a1c-Date-2020-04-15_Time-11-08-26.log</p>
</blockquote>
<p>To plot all motifs we can use cowplot</p>
<div class="sourceCode" id="cb273"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb273-1"><a href="archr-deviations.html#cb273-1"></a><span class="co">#Rearrange for grid plotting</span></span>
<span id="cb273-2"><a href="archr-deviations.html#cb273-2"></a>p2 &lt;-<span class="st"> </span><span class="kw">lapply</span>(p, <span class="cf">function</span>(x){</span>
<span id="cb273-3"><a href="archr-deviations.html#cb273-3"></a>    x <span class="op">+</span><span class="st"> </span><span class="kw">guides</span>(<span class="dt">color =</span> <span class="ot">FALSE</span>, <span class="dt">fill =</span> <span class="ot">FALSE</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb273-4"><a href="archr-deviations.html#cb273-4"></a><span class="st">    </span><span class="kw">theme_ArchR</span>(<span class="dt">baseSize =</span> <span class="fl">6.5</span>) <span class="op">+</span></span>
<span id="cb273-5"><a href="archr-deviations.html#cb273-5"></a><span class="st">    </span><span class="kw">theme</span>(<span class="dt">plot.margin =</span> <span class="kw">unit</span>(<span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>), <span class="st">&quot;cm&quot;</span>)) <span class="op">+</span></span>
<span id="cb273-6"><a href="archr-deviations.html#cb273-6"></a><span class="st">    </span><span class="kw">theme</span>(</span>
<span id="cb273-7"><a href="archr-deviations.html#cb273-7"></a>        <span class="dt">axis.text.x=</span><span class="kw">element_blank</span>(), </span>
<span id="cb273-8"><a href="archr-deviations.html#cb273-8"></a>        <span class="dt">axis.ticks.x=</span><span class="kw">element_blank</span>(), </span>
<span id="cb273-9"><a href="archr-deviations.html#cb273-9"></a>        <span class="dt">axis.text.y=</span><span class="kw">element_blank</span>(), </span>
<span id="cb273-10"><a href="archr-deviations.html#cb273-10"></a>        <span class="dt">axis.ticks.y=</span><span class="kw">element_blank</span>()</span>
<span id="cb273-11"><a href="archr-deviations.html#cb273-11"></a>    )</span>
<span id="cb273-12"><a href="archr-deviations.html#cb273-12"></a>})</span>
<span id="cb273-13"><a href="archr-deviations.html#cb273-13"></a><span class="kw">do.call</span>(cowplot<span class="op">::</span>plot_grid, <span class="kw">c</span>(<span class="kw">list</span>(<span class="dt">ncol =</span> <span class="dv">3</span>),p2))</span></code></pre></div>
<p><img src="images/HemeWalkthrough/PNG/Plot-UMAP-EncodeTFBS-W-Imputation_1.png" width="800" height="800" /></p>
<p>We can also plot the gene expression profiles for each of these TFs</p>
<div class="sourceCode" id="cb274"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb274-1"><a href="archr-deviations.html#cb274-1"></a>markerRNA &lt;-<span class="st"> </span><span class="kw">getFeatures</span>(projHeme5, <span class="dt">select =</span> <span class="kw">paste</span>(<span class="kw">gsub</span>(<span class="st">&quot;_&quot;</span>,<span class="st">&quot;&quot;</span>,tfs), <span class="dt">collapse=</span><span class="st">&quot;|&quot;</span>), <span class="dt">useMatrix =</span> <span class="st">&quot;GeneScoreMatrix&quot;</span>)</span>
<span id="cb274-2"><a href="archr-deviations.html#cb274-2"></a>markerRNA &lt;-<span class="st"> </span>markerRNA[markerRNA <span class="op">%ni%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;SREBF1&quot;</span>)]</span>
<span id="cb274-3"><a href="archr-deviations.html#cb274-3"></a>markerRNA</span></code></pre></div>
<blockquote>
<p>## [1] “TBX21” “CEBPB” “EBF1” “IRF4” “PAX5” “GATA1”</p>
</blockquote>
<div class="sourceCode" id="cb275"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb275-1"><a href="archr-deviations.html#cb275-1"></a>p &lt;-<span class="st"> </span><span class="kw">plotEmbedding</span>(</span>
<span id="cb275-2"><a href="archr-deviations.html#cb275-2"></a>    <span class="dt">ArchRProj =</span> projHeme5, </span>
<span id="cb275-3"><a href="archr-deviations.html#cb275-3"></a>    <span class="dt">colorBy =</span> <span class="st">&quot;GeneScoreMatrix&quot;</span>, </span>
<span id="cb275-4"><a href="archr-deviations.html#cb275-4"></a>    <span class="dt">name =</span> <span class="kw">sort</span>(markerRNA), </span>
<span id="cb275-5"><a href="archr-deviations.html#cb275-5"></a>    <span class="dt">embedding =</span> <span class="st">&quot;UMAP&quot;</span>,</span>
<span id="cb275-6"><a href="archr-deviations.html#cb275-6"></a>    <span class="dt">imputeWeights =</span> <span class="kw">getImputeWeights</span>(projHeme5)</span>
<span id="cb275-7"><a href="archr-deviations.html#cb275-7"></a>)</span></code></pre></div>
<blockquote>
<p>## Getting ImputeWeights<br />
## ArchR logging to : ArchRLogs/ArchR-plotEmbedding-103747bae5e9f-Date-2020-04-15_Time-11-08-53.log<br />
## If there is an issue, please report to github with logFile!<br />
## Getting UMAP Embedding<br />
## ColorBy = GeneScoreMatrix<br />
## Getting Matrix Values…<br />
## Getting Matrix Values…<br />
##<br />
## Imputing Matrix<br />
## Using weights on disk<br />
## Using weights on disk<br />
## Plotting Embedding<br />
## 1 2 3 4 5 6<br />
## ArchR logging successful to : ArchRLogs/ArchR-plotEmbedding-103747bae5e9f-Date-2020-04-15_Time-11-08-53.log</p>
</blockquote>
<div class="sourceCode" id="cb276"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb276-1"><a href="archr-deviations.html#cb276-1"></a><span class="co">#Rearrange for grid plotting</span></span>
<span id="cb276-2"><a href="archr-deviations.html#cb276-2"></a>p2 &lt;-<span class="st"> </span><span class="kw">lapply</span>(p, <span class="cf">function</span>(x){</span>
<span id="cb276-3"><a href="archr-deviations.html#cb276-3"></a>    x <span class="op">+</span><span class="st"> </span><span class="kw">guides</span>(<span class="dt">color =</span> <span class="ot">FALSE</span>, <span class="dt">fill =</span> <span class="ot">FALSE</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb276-4"><a href="archr-deviations.html#cb276-4"></a><span class="st">    </span><span class="kw">theme_ArchR</span>(<span class="dt">baseSize =</span> <span class="fl">6.5</span>) <span class="op">+</span></span>
<span id="cb276-5"><a href="archr-deviations.html#cb276-5"></a><span class="st">    </span><span class="kw">theme</span>(<span class="dt">plot.margin =</span> <span class="kw">unit</span>(<span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>), <span class="st">&quot;cm&quot;</span>)) <span class="op">+</span></span>
<span id="cb276-6"><a href="archr-deviations.html#cb276-6"></a><span class="st">    </span><span class="kw">theme</span>(</span>
<span id="cb276-7"><a href="archr-deviations.html#cb276-7"></a>        <span class="dt">axis.text.x=</span><span class="kw">element_blank</span>(), </span>
<span id="cb276-8"><a href="archr-deviations.html#cb276-8"></a>        <span class="dt">axis.ticks.x=</span><span class="kw">element_blank</span>(), </span>
<span id="cb276-9"><a href="archr-deviations.html#cb276-9"></a>        <span class="dt">axis.text.y=</span><span class="kw">element_blank</span>(), </span>
<span id="cb276-10"><a href="archr-deviations.html#cb276-10"></a>        <span class="dt">axis.ticks.y=</span><span class="kw">element_blank</span>()</span>
<span id="cb276-11"><a href="archr-deviations.html#cb276-11"></a>    )</span>
<span id="cb276-12"><a href="archr-deviations.html#cb276-12"></a>})</span>
<span id="cb276-13"><a href="archr-deviations.html#cb276-13"></a><span class="kw">do.call</span>(cowplot<span class="op">::</span>plot_grid, <span class="kw">c</span>(<span class="kw">list</span>(<span class="dt">ncol =</span> <span class="dv">3</span>),p2))</span></code></pre></div>
<p><img src="images/HemeWalkthrough/PNG/Plot-UMAP-EncodeTFBSGS-W-Imputation_1.png" width="800" height="800" /></p>
<p>We can also plot the gene expression profiles for each of these TFs</p>
<div class="sourceCode" id="cb277"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb277-1"><a href="archr-deviations.html#cb277-1"></a>markerRNA &lt;-<span class="st"> </span><span class="kw">getFeatures</span>(projHeme5, <span class="dt">select =</span> <span class="kw">paste</span>(<span class="kw">gsub</span>(<span class="st">&quot;_&quot;</span>,<span class="st">&quot;&quot;</span>,tfs), <span class="dt">collapse=</span><span class="st">&quot;|&quot;</span>), <span class="dt">useMatrix =</span> <span class="st">&quot;GeneIntegrationMatrix&quot;</span>)</span>
<span id="cb277-2"><a href="archr-deviations.html#cb277-2"></a>markerRNA &lt;-<span class="st"> </span>markerRNA[markerRNA <span class="op">%ni%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;SREBF1&quot;</span>,<span class="st">&quot;CEBPA-DT&quot;</span>)]</span>
<span id="cb277-3"><a href="archr-deviations.html#cb277-3"></a>markerRNA</span></code></pre></div>
<blockquote>
<p>## [1] “TBX21” “CEBPB” “EBF1” “IRF4” “PAX5” “GATA1”</p>
</blockquote>
<div class="sourceCode" id="cb278"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb278-1"><a href="archr-deviations.html#cb278-1"></a>p &lt;-<span class="st"> </span><span class="kw">plotEmbedding</span>(</span>
<span id="cb278-2"><a href="archr-deviations.html#cb278-2"></a>    <span class="dt">ArchRProj =</span> projHeme5, </span>
<span id="cb278-3"><a href="archr-deviations.html#cb278-3"></a>    <span class="dt">colorBy =</span> <span class="st">&quot;GeneIntegrationMatrix&quot;</span>, </span>
<span id="cb278-4"><a href="archr-deviations.html#cb278-4"></a>    <span class="dt">name =</span> <span class="kw">sort</span>(markerRNA), </span>
<span id="cb278-5"><a href="archr-deviations.html#cb278-5"></a>    <span class="dt">embedding =</span> <span class="st">&quot;UMAP&quot;</span>,</span>
<span id="cb278-6"><a href="archr-deviations.html#cb278-6"></a>    <span class="dt">continuousSet =</span> <span class="st">&quot;blueYellow&quot;</span>,</span>
<span id="cb278-7"><a href="archr-deviations.html#cb278-7"></a>    <span class="dt">imputeWeights =</span> <span class="kw">getImputeWeights</span>(projHeme5)</span>
<span id="cb278-8"><a href="archr-deviations.html#cb278-8"></a>)</span></code></pre></div>
<blockquote>
<p>## Getting ImputeWeights<br />
## ArchR logging to : ArchRLogs/ArchR-plotEmbedding-103746f7eeb66-Date-2020-04-15_Time-11-09-21.log<br />
## If there is an issue, please report to github with logFile!<br />
## Getting UMAP Embedding<br />
## ColorBy = GeneIntegrationMatrix<br />
## Getting Matrix Values…<br />
## Getting Matrix Values…<br />
##<br />
## Imputing Matrix<br />
## Using weights on disk<br />
## Using weights on disk<br />
## Plotting Embedding<br />
## 1 2 3 4 5 6<br />
## ArchR logging successful to : ArchRLogs/ArchR-plotEmbedding-103746f7eeb66-Date-2020-04-15_Time-11-09-21.log</p>
</blockquote>
<div class="sourceCode" id="cb279"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb279-1"><a href="archr-deviations.html#cb279-1"></a><span class="co">#Rearrange for grid plotting</span></span>
<span id="cb279-2"><a href="archr-deviations.html#cb279-2"></a>p2 &lt;-<span class="st"> </span><span class="kw">lapply</span>(p, <span class="cf">function</span>(x){</span>
<span id="cb279-3"><a href="archr-deviations.html#cb279-3"></a>    x <span class="op">+</span><span class="st"> </span><span class="kw">guides</span>(<span class="dt">color =</span> <span class="ot">FALSE</span>, <span class="dt">fill =</span> <span class="ot">FALSE</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb279-4"><a href="archr-deviations.html#cb279-4"></a><span class="st">    </span><span class="kw">theme_ArchR</span>(<span class="dt">baseSize =</span> <span class="fl">6.5</span>) <span class="op">+</span></span>
<span id="cb279-5"><a href="archr-deviations.html#cb279-5"></a><span class="st">    </span><span class="kw">theme</span>(<span class="dt">plot.margin =</span> <span class="kw">unit</span>(<span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>), <span class="st">&quot;cm&quot;</span>)) <span class="op">+</span></span>
<span id="cb279-6"><a href="archr-deviations.html#cb279-6"></a><span class="st">    </span><span class="kw">theme</span>(</span>
<span id="cb279-7"><a href="archr-deviations.html#cb279-7"></a>        <span class="dt">axis.text.x=</span><span class="kw">element_blank</span>(), </span>
<span id="cb279-8"><a href="archr-deviations.html#cb279-8"></a>        <span class="dt">axis.ticks.x=</span><span class="kw">element_blank</span>(), </span>
<span id="cb279-9"><a href="archr-deviations.html#cb279-9"></a>        <span class="dt">axis.text.y=</span><span class="kw">element_blank</span>(), </span>
<span id="cb279-10"><a href="archr-deviations.html#cb279-10"></a>        <span class="dt">axis.ticks.y=</span><span class="kw">element_blank</span>()</span>
<span id="cb279-11"><a href="archr-deviations.html#cb279-11"></a>    )</span>
<span id="cb279-12"><a href="archr-deviations.html#cb279-12"></a>})</span>
<span id="cb279-13"><a href="archr-deviations.html#cb279-13"></a><span class="kw">do.call</span>(cowplot<span class="op">::</span>plot_grid, <span class="kw">c</span>(<span class="kw">list</span>(<span class="dt">ncol =</span> <span class="dv">3</span>),p2))</span></code></pre></div>
<p><img src="images/HemeWalkthrough/PNG/Plot-UMAP-EncodeTFBSRNA-W-Imputation_1.png" width="800" height="800" /></p>
</div>
<div id="bulk-atac-seq-1" class="section level3" number="12.2.2">
<h3><span class="header-section-number">12.2.2</span> Bulk ATAC-seq</h3>
<p>If you have not added motif annotations</p>
<div class="sourceCode" id="cb280"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb280-1"><a href="archr-deviations.html#cb280-1"></a><span class="cf">if</span>(<span class="st">&quot;ATAC&quot;</span> <span class="op">%ni%</span><span class="st"> </span><span class="kw">names</span>(projHeme5<span class="op">@</span>peakAnnotation)){</span>
<span id="cb280-2"><a href="archr-deviations.html#cb280-2"></a>    projHeme5 &lt;-<span class="st"> </span><span class="kw">addArchRAnnotations</span>(<span class="dt">ArchRProj =</span> projHeme5, <span class="dt">collection =</span> <span class="st">&quot;ATAC&quot;</span>)</span>
<span id="cb280-3"><a href="archr-deviations.html#cb280-3"></a>}</span></code></pre></div>
<p>Add ChromVAR Deviations for Motifs</p>
<div class="sourceCode" id="cb281"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb281-1"><a href="archr-deviations.html#cb281-1"></a>projHeme5 &lt;-<span class="st"> </span><span class="kw">addDeviationsMatrix</span>(</span>
<span id="cb281-2"><a href="archr-deviations.html#cb281-2"></a>  <span class="dt">ArchRProj =</span> projHeme5, </span>
<span id="cb281-3"><a href="archr-deviations.html#cb281-3"></a>  <span class="dt">peakAnnotation =</span> <span class="st">&quot;ATAC&quot;</span>,</span>
<span id="cb281-4"><a href="archr-deviations.html#cb281-4"></a>  <span class="dt">force =</span> <span class="ot">TRUE</span></span>
<span id="cb281-5"><a href="archr-deviations.html#cb281-5"></a>)</span></code></pre></div>
<blockquote>
<p>## Using Previous Background Peaks!<br />
## ArchR logging to : ArchRLogs/ArchR-addDeviationsMatrix-103743481be2a-Date-2020-04-15_Time-11-09-46.log<br />
## If there is an issue, please report to github with logFile!<br />
## NULL<br />
## 2020-04-15 11:09:52 : Batch Execution w/ safelapply!, 0 mins elapsed.<br />
## ###########<br />
## 2020-04-15 11:13:09 : Completed Computing Deviations!, 3.376 mins elapsed.<br />
## ###########<br />
## ArchR logging successful to : ArchRLogs/ArchR-addDeviationsMatrix-103743481be2a-Date-2020-04-15_Time-11-09-46.log</p>
</blockquote>
<p>Plot Variable Motif Annotations</p>
<div class="sourceCode" id="cb282"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb282-1"><a href="archr-deviations.html#cb282-1"></a>plotVarDev &lt;-<span class="st"> </span><span class="kw">getVarDeviations</span>(projHeme5, <span class="dt">plot =</span> <span class="ot">TRUE</span>, <span class="dt">name =</span> <span class="st">&quot;ATACMatrix&quot;</span>)</span></code></pre></div>
<blockquote>
<p>## DataFrame with 6 rows and 6 columns<br />
## seqnames idx name combinedVars combinedMeans<br />
## <Rle> <array> <array> <numeric> <numeric><br />
## f22 z 22 IAtlas_T_CD8posCenMem 12.8413703976471 -0.0972946574781075<br />
## f86 z 86 Heme_CD8 12.5070480860108 -0.079462324469343<br />
## f85 z 85 Heme_CD4 12.2753260246254 -0.0561621531675016<br />
## f23 z 23 IAtlas_T_CD8posEffMem 12.2250185804317 -0.104389732164243<br />
## f21 z 21 IAtlas_T_CD8pos 12.1998071900673 -0.09178088897301<br />
## f33 z 33 IAtlas_T_Th1Precursor 12.1352640996498 -0.0819315630754632<br />
## rank<br />
## <integer><br />
## f22 1<br />
## f86 2<br />
## f85 3<br />
## f23 4<br />
## f21 5<br />
## f33 6</p>
</blockquote>
<div class="sourceCode" id="cb283"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb283-1"><a href="archr-deviations.html#cb283-1"></a>plotVarDev</span></code></pre></div>
<p><img src="images/HemeWalkthrough/PNG/Variable-ATAC-Deviation-Scores_1.png" width="500" height="500" /></p>
<p>We can then plot this as a nice PDF</p>
<div class="sourceCode" id="cb284"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb284-1"><a href="archr-deviations.html#cb284-1"></a><span class="kw">plotPDF</span>(plotVarDev, <span class="dt">name =</span> <span class="st">&quot;Variable-ATAC-Deviation-Scores&quot;</span>, <span class="dt">width =</span> <span class="dv">5</span>, <span class="dt">height =</span> <span class="dv">5</span>, <span class="dt">ArchRProj =</span> projHeme5, <span class="dt">addDOC =</span> <span class="ot">FALSE</span>)</span></code></pre></div>
<blockquote>
<p>## [1] “plotting ggplot!”<br />
## [1] 0</p>
</blockquote>
<p>If we want to get TF feature names for downstream plotting</p>
<div class="sourceCode" id="cb285"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb285-1"><a href="archr-deviations.html#cb285-1"></a>ATACPeaks &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Heme_HSC&quot;</span>, <span class="st">&quot;Heme_LMPP&quot;</span>, <span class="st">&quot;Heme_Ery&quot;</span>, <span class="st">&quot;Heme_Mono&quot;</span>, <span class="st">&quot;Heme_CD4&quot;</span>, <span class="st">&quot;Heme_CD8&quot;</span>, <span class="st">&quot;Heme_B&quot;</span>, <span class="st">&quot;Heme_NK&quot;</span>, <span class="st">&quot;IAtlas_DC_Plasmacytoid&quot;</span>)</span>
<span id="cb285-2"><a href="archr-deviations.html#cb285-2"></a><span class="kw">getFeatures</span>(projHeme5, <span class="dt">select =</span> <span class="kw">paste</span>(ATACPeaks, <span class="dt">collapse=</span><span class="st">&quot;|&quot;</span>), <span class="dt">useMatrix =</span> <span class="st">&quot;ATACMatrix&quot;</span>)</span></code></pre></div>
<blockquote>
<p>## [1] “z:Heme_NK” “z:Heme_Mono”<br />
## [3] “z:Heme_LMPP” “z:Heme_HSC”<br />
## [5] “z:Heme_Ery” “z:Heme_CD8”<br />
## [7] “z:Heme_CD4” “z:Heme_B”<br />
## [9] “z:IAtlas_DC_Plasmacytoid” “deviations:Heme_NK”<br />
## [11] “deviations:Heme_Mono” “deviations:Heme_LMPP”<br />
## [13] “deviations:Heme_HSC” “deviations:Heme_Ery”<br />
## [15] “deviations:Heme_CD8” “deviations:Heme_CD4”<br />
## [17] “deviations:Heme_B” “deviations:IAtlas_DC_Plasmacytoid”</p>
</blockquote>
<p>To get deviation z-scores</p>
<div class="sourceCode" id="cb286"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb286-1"><a href="archr-deviations.html#cb286-1"></a>markerATAC &lt;-<span class="st"> </span><span class="kw">getFeatures</span>(projHeme5, <span class="dt">select =</span> <span class="kw">paste</span>(ATACPeaks, <span class="dt">collapse=</span><span class="st">&quot;|&quot;</span>), <span class="dt">useMatrix =</span> <span class="st">&quot;ATACMatrix&quot;</span>)</span>
<span id="cb286-2"><a href="archr-deviations.html#cb286-2"></a>markerATAC &lt;-<span class="st"> </span><span class="kw">sort</span>(<span class="kw">grep</span>(<span class="st">&quot;z:&quot;</span>, markerATAC, <span class="dt">value =</span> <span class="ot">TRUE</span>))</span>
<span id="cb286-3"><a href="archr-deviations.html#cb286-3"></a>markerATAC</span></code></pre></div>
<blockquote>
<p>## [1] “z:Heme_B” “z:Heme_CD4”<br />
## [3] “z:Heme_CD8” “z:Heme_Ery”<br />
## [5] “z:Heme_HSC” “z:Heme_LMPP”<br />
## [7] “z:Heme_Mono” “z:Heme_NK”<br />
## [9] “z:IAtlas_DC_Plasmacytoid”</p>
</blockquote>
<div class="sourceCode" id="cb287"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb287-1"><a href="archr-deviations.html#cb287-1"></a>p &lt;-<span class="st"> </span><span class="kw">plotEmbedding</span>(</span>
<span id="cb287-2"><a href="archr-deviations.html#cb287-2"></a>    <span class="dt">ArchRProj =</span> projHeme5, </span>
<span id="cb287-3"><a href="archr-deviations.html#cb287-3"></a>    <span class="dt">colorBy =</span> <span class="st">&quot;ATACMatrix&quot;</span>, </span>
<span id="cb287-4"><a href="archr-deviations.html#cb287-4"></a>    <span class="dt">name =</span> markerATAC, </span>
<span id="cb287-5"><a href="archr-deviations.html#cb287-5"></a>    <span class="dt">embedding =</span> <span class="st">&quot;UMAP&quot;</span>,</span>
<span id="cb287-6"><a href="archr-deviations.html#cb287-6"></a>    <span class="dt">imputeWeights =</span> <span class="kw">getImputeWeights</span>(projHeme5)</span>
<span id="cb287-7"><a href="archr-deviations.html#cb287-7"></a>)</span></code></pre></div>
<blockquote>
<p>## Getting ImputeWeights<br />
## ArchR logging to : ArchRLogs/ArchR-plotEmbedding-1037416f3cb56-Date-2020-04-15_Time-11-13-23.log<br />
## If there is an issue, please report to github with logFile!<br />
## Getting UMAP Embedding<br />
## ColorBy = ATACMatrix<br />
## Getting Matrix Values…<br />
## Getting Matrix Values…<br />
##<br />
## Imputing Matrix<br />
## Using weights on disk<br />
## Using weights on disk<br />
## Plotting Embedding<br />
## 1 2 3 4 5 6 7 8 9<br />
## ArchR logging successful to : ArchRLogs/ArchR-plotEmbedding-1037416f3cb56-Date-2020-04-15_Time-11-13-23.log</p>
</blockquote>
<p>To plot all motifs we can use cowplot</p>
<div class="sourceCode" id="cb288"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb288-1"><a href="archr-deviations.html#cb288-1"></a><span class="co">#Rearrange for grid plotting</span></span>
<span id="cb288-2"><a href="archr-deviations.html#cb288-2"></a>p2 &lt;-<span class="st"> </span><span class="kw">lapply</span>(p, <span class="cf">function</span>(x){</span>
<span id="cb288-3"><a href="archr-deviations.html#cb288-3"></a>    x <span class="op">+</span><span class="st"> </span><span class="kw">guides</span>(<span class="dt">color =</span> <span class="ot">FALSE</span>, <span class="dt">fill =</span> <span class="ot">FALSE</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb288-4"><a href="archr-deviations.html#cb288-4"></a><span class="st">    </span><span class="kw">theme_ArchR</span>(<span class="dt">baseSize =</span> <span class="fl">6.5</span>) <span class="op">+</span></span>
<span id="cb288-5"><a href="archr-deviations.html#cb288-5"></a><span class="st">    </span><span class="kw">theme</span>(<span class="dt">plot.margin =</span> <span class="kw">unit</span>(<span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>), <span class="st">&quot;cm&quot;</span>)) <span class="op">+</span></span>
<span id="cb288-6"><a href="archr-deviations.html#cb288-6"></a><span class="st">    </span><span class="kw">theme</span>(</span>
<span id="cb288-7"><a href="archr-deviations.html#cb288-7"></a>        <span class="dt">axis.text.x=</span><span class="kw">element_blank</span>(), </span>
<span id="cb288-8"><a href="archr-deviations.html#cb288-8"></a>        <span class="dt">axis.ticks.x=</span><span class="kw">element_blank</span>(), </span>
<span id="cb288-9"><a href="archr-deviations.html#cb288-9"></a>        <span class="dt">axis.text.y=</span><span class="kw">element_blank</span>(), </span>
<span id="cb288-10"><a href="archr-deviations.html#cb288-10"></a>        <span class="dt">axis.ticks.y=</span><span class="kw">element_blank</span>()</span>
<span id="cb288-11"><a href="archr-deviations.html#cb288-11"></a>    )</span>
<span id="cb288-12"><a href="archr-deviations.html#cb288-12"></a>})</span>
<span id="cb288-13"><a href="archr-deviations.html#cb288-13"></a><span class="kw">do.call</span>(cowplot<span class="op">::</span>plot_grid, <span class="kw">c</span>(<span class="kw">list</span>(<span class="dt">ncol =</span> <span class="dv">3</span>),p2))</span></code></pre></div>
<p><img src="images/HemeWalkthrough/PNG/Plot-UMAP-MarkerATAC-W-Imputation_1.png" width="800" height="800" /></p>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="motif-deviations.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="custom-deviations.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": null
},
"fontsettings": null,
"edit": null,
"history": {
"link": "https://github.com/GreenleafLab/ArchR_Website/commits/dev/12_Deviations.Rmd",
"text": "Chapter edit history"
},
"view": {
"link": "https://github.com/GreenleafLab/ArchR_Website/view/dev/12_Deviations.Rmd",
"text": "View source on GitHub"
},
"download": ["ArchR.pdf"],
"toc": {
"collapse": "none"
},
"search": true,
"info": false
});
});
</script>

</body>

</html>
