<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>12.2 Calling Peaks w/ Macs2 | ArchR: Robust and scaleable analysis of single-cell chromatin accessibility data.</title>
  <meta name="description" content="A guide to ArchR" />
  <meta name="generator" content="bookdown 0.31 and GitBook 2.6.7" />

  <meta property="og:title" content="12.2 Calling Peaks w/ Macs2 | ArchR: Robust and scaleable analysis of single-cell chromatin accessibility data." />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="A guide to ArchR" />
  <meta name="github-repo" content="GreenleafLab/ArchR" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="12.2 Calling Peaks w/ Macs2 | ArchR: Robust and scaleable analysis of single-cell chromatin accessibility data." />
  
  <meta name="twitter:description" content="A guide to ArchR" />
  

<meta name="author" content="Jeffrey Granja and Ryan Corces" />


<meta name="date" content="2022-12-23" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="images/favicon.ico" type="image/x-icon" />
<link rel="prev" href="the-iterative-overlap-peak-merging-procedure.html"/>
<link rel="next" href="calling-peaks-w-tilematrix.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<link href="libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="../index.html">Back to ArchR Home</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><img src="images/ArchR_Logo_Integrated.png" alt="" width="200"></a></li>
<li class="chapter" data-level="1" data-path="debugging-and-troubleshooting-in-archr.html"><a href="debugging-and-troubleshooting-in-archr.html"><i class="fa fa-check"></i><b>1</b> Debugging and Troubleshooting in ArchR</a></li>
<li class="chapter" data-level="2" data-path="manage-archrs-dependencies.html"><a href="manage-archrs-dependencies.html"><i class="fa fa-check"></i><b>2</b> Manage ArchR’s Dependencies</a>
<ul>
<li class="chapter" data-level="2.1" data-path="using-renv-to-manage-dependencies.html"><a href="using-renv-to-manage-dependencies.html"><i class="fa fa-check"></i><b>2.1</b> Using renv to manage dependencies</a></li>
<li class="chapter" data-level="2.2" data-path="other-environment-managers.html"><a href="other-environment-managers.html"><i class="fa fa-check"></i><b>2.2</b> Other environment managers</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="getting-started-with-archr.html"><a href="getting-started-with-archr.html"><i class="fa fa-check"></i><b>3</b> Getting Started with ArchR</a>
<ul>
<li class="chapter" data-level="3.1" data-path="a-brief-primer-on-atac-seq-terminology.html"><a href="a-brief-primer-on-atac-seq-terminology.html"><i class="fa fa-check"></i><b>3.1</b> A Brief Primer on ATAC-seq Terminology</a></li>
<li class="chapter" data-level="3.2" data-path="why-use-archr.html"><a href="why-use-archr.html"><i class="fa fa-check"></i><b>3.2</b> Why use ArchR?</a></li>
<li class="chapter" data-level="3.3" data-path="what-is-an-arrow-file-archrproject.html"><a href="what-is-an-arrow-file-archrproject.html"><i class="fa fa-check"></i><b>3.3</b> What is an Arrow file / <code>ArchRProject</code>?</a></li>
<li class="chapter" data-level="3.4" data-path="input-file-types-in-archr.html"><a href="input-file-types-in-archr.html"><i class="fa fa-check"></i><b>3.4</b> Input File Types in ArchR</a></li>
<li class="chapter" data-level="3.5" data-path="getting-set-up.html"><a href="getting-set-up.html"><i class="fa fa-check"></i><b>3.5</b> Getting Set Up</a></li>
<li class="chapter" data-level="3.6" data-path="creating-arrow-files.html"><a href="creating-arrow-files.html"><i class="fa fa-check"></i><b>3.6</b> Creating Arrow Files</a></li>
<li class="chapter" data-level="3.7" data-path="per-cell-quality-control.html"><a href="per-cell-quality-control.html"><i class="fa fa-check"></i><b>3.7</b> Per-cell Quality Control</a></li>
<li class="chapter" data-level="3.8" data-path="using-bam-files-for-arrow-file-creation.html"><a href="using-bam-files-for-arrow-file-creation.html"><i class="fa fa-check"></i><b>3.8</b> Using BAM files for Arrow File creation</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="doublet-inference-with-archr.html"><a href="doublet-inference-with-archr.html"><i class="fa fa-check"></i><b>4</b> Doublet Inference with ArchR</a>
<ul>
<li class="chapter" data-level="4.1" data-path="how-does-doublet-identification-work-in-archr.html"><a href="how-does-doublet-identification-work-in-archr.html"><i class="fa fa-check"></i><b>4.1</b> How does doublet identification work in ArchR?</a></li>
<li class="chapter" data-level="4.2" data-path="inferring-scatac-seq-doublets-with-archr.html"><a href="inferring-scatac-seq-doublets-with-archr.html"><i class="fa fa-check"></i><b>4.2</b> Inferring scATAC-seq Doublets with ArchR</a></li>
<li class="chapter" data-level="4.3" data-path="using-demuxlet-with-archr.html"><a href="using-demuxlet-with-archr.html"><i class="fa fa-check"></i><b>4.3</b> Using <em>demuxlet</em> with ArchR</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="creating-an-archrproject.html"><a href="creating-an-archrproject.html"><i class="fa fa-check"></i><b>5</b> Creating an ArchRProject</a>
<ul>
<li class="chapter" data-level="5.1" data-path="creating-an-archrproject-1.html"><a href="creating-an-archrproject-1.html"><i class="fa fa-check"></i><b>5.1</b> Creating An ArchRProject</a></li>
<li class="chapter" data-level="5.2" data-path="manipulating-an-archrproject.html"><a href="manipulating-an-archrproject.html"><i class="fa fa-check"></i><b>5.2</b> Manipulating An ArchRProject</a></li>
<li class="chapter" data-level="5.3" data-path="plotting-sample-statistics-from-an-archrproject.html"><a href="plotting-sample-statistics-from-an-archrproject.html"><i class="fa fa-check"></i><b>5.3</b> Plotting Sample Statistics from an ArchRProject</a></li>
<li class="chapter" data-level="5.4" data-path="plotting-sample-fragment-size-distribution-and-tss-enrichment-profiles..html"><a href="plotting-sample-fragment-size-distribution-and-tss-enrichment-profiles..html"><i class="fa fa-check"></i><b>5.4</b> Plotting Sample Fragment Size Distribution and TSS Enrichment Profiles.</a></li>
<li class="chapter" data-level="5.5" data-path="saving-and-loading-an-archrproject.html"><a href="saving-and-loading-an-archrproject.html"><i class="fa fa-check"></i><b>5.5</b> Saving and Loading an <code>ArchRProject</code></a></li>
<li class="chapter" data-level="5.6" data-path="filtering-doublets-from-an-archrproject.html"><a href="filtering-doublets-from-an-archrproject.html"><i class="fa fa-check"></i><b>5.6</b> Filtering Doublets from an ArchRProject</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="dimensionality-reduction-with-archr.html"><a href="dimensionality-reduction-with-archr.html"><i class="fa fa-check"></i><b>6</b> Dimensionality Reduction with ArchR</a>
<ul>
<li class="chapter" data-level="6.1" data-path="archrs-lsi-implementation.html"><a href="archrs-lsi-implementation.html"><i class="fa fa-check"></i><b>6.1</b> ArchR’s LSI Implementation</a></li>
<li class="chapter" data-level="6.2" data-path="iterative-latent-semantic-indexing-lsi.html"><a href="iterative-latent-semantic-indexing-lsi.html"><i class="fa fa-check"></i><b>6.2</b> Iterative Latent Semantic Indexing (LSI)</a></li>
<li class="chapter" data-level="6.3" data-path="estimated-lsi.html"><a href="estimated-lsi.html"><i class="fa fa-check"></i><b>6.3</b> Estimated LSI</a></li>
<li class="chapter" data-level="6.4" data-path="batch-effect-correction-wtih-harmony.html"><a href="batch-effect-correction-wtih-harmony.html"><i class="fa fa-check"></i><b>6.4</b> Batch Effect Correction wtih Harmony</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="clustering-with-archr.html"><a href="clustering-with-archr.html"><i class="fa fa-check"></i><b>7</b> Clustering with ArchR</a>
<ul>
<li class="chapter" data-level="7.1" data-path="clustering-using-seurats-findclusters-function.html"><a href="clustering-using-seurats-findclusters-function.html"><i class="fa fa-check"></i><b>7.1</b> Clustering using Seurat’s <code>FindClusters()</code> function</a></li>
<li class="chapter" data-level="7.2" data-path="clustering-using-scran.html"><a href="clustering-using-scran.html"><i class="fa fa-check"></i><b>7.2</b> Clustering using <code>scran</code></a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="single-cell-embeddings.html"><a href="single-cell-embeddings.html"><i class="fa fa-check"></i><b>8</b> Single-cell Embeddings</a>
<ul>
<li class="chapter" data-level="8.1" data-path="uniform-manifold-approximation-and-projection-umap.html"><a href="uniform-manifold-approximation-and-projection-umap.html"><i class="fa fa-check"></i><b>8.1</b> Uniform Manifold Approximation and Projection (UMAP)</a></li>
<li class="chapter" data-level="8.2" data-path="t-stocastic-neighbor-embedding-t-sne.html"><a href="t-stocastic-neighbor-embedding-t-sne.html"><i class="fa fa-check"></i><b>8.2</b> t-Stocastic Neighbor Embedding (t-SNE)</a></li>
<li class="chapter" data-level="8.3" data-path="dimensionality-reduction-after-harmony.html"><a href="dimensionality-reduction-after-harmony.html"><i class="fa fa-check"></i><b>8.3</b> Dimensionality Reduction After Harmony</a></li>
<li class="chapter" data-level="8.4" data-path="highlighting-specific-cells-on-an-embedding.html"><a href="highlighting-specific-cells-on-an-embedding.html"><i class="fa fa-check"></i><b>8.4</b> Highlighting specific cells on an embedding</a></li>
<li class="chapter" data-level="8.5" data-path="importing-an-embedding-from-external-software.html"><a href="importing-an-embedding-from-external-software.html"><i class="fa fa-check"></i><b>8.5</b> Importing an embedding from external software</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="gene-scores-and-marker-genes-with-archr.html"><a href="gene-scores-and-marker-genes-with-archr.html"><i class="fa fa-check"></i><b>9</b> Gene Scores and Marker Genes with ArchR</a>
<ul>
<li class="chapter" data-level="9.1" data-path="calculating-gene-scores-in-archr.html"><a href="calculating-gene-scores-in-archr.html"><i class="fa fa-check"></i><b>9.1</b> Calculating Gene Scores in ArchR</a></li>
<li class="chapter" data-level="9.2" data-path="identification-of-marker-features.html"><a href="identification-of-marker-features.html"><i class="fa fa-check"></i><b>9.2</b> Identification of Marker Features</a></li>
<li class="chapter" data-level="9.3" data-path="identifying-marker-genes.html"><a href="identifying-marker-genes.html"><i class="fa fa-check"></i><b>9.3</b> Identifying Marker Genes</a></li>
<li class="chapter" data-level="9.4" data-path="visualizing-marker-genes-on-an-embedding.html"><a href="visualizing-marker-genes-on-an-embedding.html"><i class="fa fa-check"></i><b>9.4</b> Visualizing Marker Genes on an Embedding</a></li>
<li class="chapter" data-level="9.5" data-path="marker-genes-imputation-with-magic.html"><a href="marker-genes-imputation-with-magic.html"><i class="fa fa-check"></i><b>9.5</b> Marker Genes Imputation with MAGIC</a></li>
<li class="chapter" data-level="9.6" data-path="module-scores.html"><a href="module-scores.html"><i class="fa fa-check"></i><b>9.6</b> Module Scores</a></li>
<li class="chapter" data-level="9.7" data-path="track-plotting-with-archrbrowser.html"><a href="track-plotting-with-archrbrowser.html"><i class="fa fa-check"></i><b>9.7</b> Track Plotting with ArchRBrowser</a></li>
<li class="chapter" data-level="9.8" data-path="launching-the-interactive-archrbrowser.html"><a href="launching-the-interactive-archrbrowser.html"><i class="fa fa-check"></i><b>9.8</b> Launching the Interactive ArchRBrowser</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="defining-cluster-identity-with-scrna-seq.html"><a href="defining-cluster-identity-with-scrna-seq.html"><i class="fa fa-check"></i><b>10</b> Defining Cluster Identity with scRNA-seq</a>
<ul>
<li class="chapter" data-level="10.1" data-path="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html"><i class="fa fa-check"></i><b>10.1</b> Cross-platform linkage of scATAC-seq cells with scRNA-seq cells</a></li>
<li class="chapter" data-level="10.2" data-path="adding-pseudo-scrna-seq-profiles-for-each-scatac-seq-cell.html"><a href="adding-pseudo-scrna-seq-profiles-for-each-scatac-seq-cell.html"><i class="fa fa-check"></i><b>10.2</b> Adding Pseudo-scRNA-seq profiles for each scATAC-seq cell</a></li>
<li class="chapter" data-level="10.3" data-path="labeling-scatac-seq-clusters-with-scrna-seq-information.html"><a href="labeling-scatac-seq-clusters-with-scrna-seq-information.html"><i class="fa fa-check"></i><b>10.3</b> Labeling scATAC-seq clusters with scRNA-seq information</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="pseudo-bulk-replicates-in-archr.html"><a href="pseudo-bulk-replicates-in-archr.html"><i class="fa fa-check"></i><b>11</b> Pseudo-bulk Replicates in ArchR</a>
<ul>
<li class="chapter" data-level="11.1" data-path="how-does-archr-make-pseudo-bulk-replicates.html"><a href="how-does-archr-make-pseudo-bulk-replicates.html"><i class="fa fa-check"></i><b>11.1</b> How Does ArchR Make Pseudo-bulk Replicates?</a></li>
<li class="chapter" data-level="11.2" data-path="making-pseudo-bulk-replicates.html"><a href="making-pseudo-bulk-replicates.html"><i class="fa fa-check"></i><b>11.2</b> Making Pseudo-bulk Replicates</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="calling-peaks-with-archr.html"><a href="calling-peaks-with-archr.html"><i class="fa fa-check"></i><b>12</b> Calling Peaks with ArchR</a>
<ul>
<li class="chapter" data-level="12.1" data-path="the-iterative-overlap-peak-merging-procedure.html"><a href="the-iterative-overlap-peak-merging-procedure.html"><i class="fa fa-check"></i><b>12.1</b> The Iterative Overlap Peak Merging Procedure</a></li>
<li class="chapter" data-level="12.2" data-path="calling-peaks-w-macs2.html"><a href="calling-peaks-w-macs2.html"><i class="fa fa-check"></i><b>12.2</b> Calling Peaks w/ Macs2</a></li>
<li class="chapter" data-level="12.3" data-path="calling-peaks-w-tilematrix.html"><a href="calling-peaks-w-tilematrix.html"><i class="fa fa-check"></i><b>12.3</b> Calling Peaks w/ TileMatrix</a></li>
<li class="chapter" data-level="12.4" data-path="adding-a-peak-matrix.html"><a href="adding-a-peak-matrix.html"><i class="fa fa-check"></i><b>12.4</b> Adding a Peak Matrix</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="identifying-marker-peaks.html"><a href="identifying-marker-peaks.html"><i class="fa fa-check"></i><b>13</b> Identifying Marker Peaks</a>
<ul>
<li class="chapter" data-level="13.1" data-path="identifying-marker-peaks-with-archr.html"><a href="identifying-marker-peaks-with-archr.html"><i class="fa fa-check"></i><b>13.1</b> Identifying Marker Peaks with ArchR</a></li>
<li class="chapter" data-level="13.2" data-path="plotting-marker-peaks-in-archr.html"><a href="plotting-marker-peaks-in-archr.html"><i class="fa fa-check"></i><b>13.2</b> Plotting Marker Peaks in ArchR</a></li>
<li class="chapter" data-level="13.3" data-path="pairwise-testing-between-groups.html"><a href="pairwise-testing-between-groups.html"><i class="fa fa-check"></i><b>13.3</b> Pairwise Testing Between Groups</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="motif-and-feature-enrichment-with-archr.html"><a href="motif-and-feature-enrichment-with-archr.html"><i class="fa fa-check"></i><b>14</b> Motif and Feature Enrichment with ArchR</a>
<ul>
<li class="chapter" data-level="14.1" data-path="motif-enrichment.html"><a href="motif-enrichment.html"><i class="fa fa-check"></i><b>14.1</b> Motif Enrichment</a></li>
<li class="chapter" data-level="14.2" data-path="archr-enrichment.html"><a href="archr-enrichment.html"><i class="fa fa-check"></i><b>14.2</b> ArchR Enrichment</a></li>
<li class="chapter" data-level="14.3" data-path="custom-enrichment.html"><a href="custom-enrichment.html"><i class="fa fa-check"></i><b>14.3</b> Custom Enrichment</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="chromvar-deviatons-enrichment-with-archr.html"><a href="chromvar-deviatons-enrichment-with-archr.html"><i class="fa fa-check"></i><b>15</b> ChromVAR Deviatons Enrichment with ArchR</a>
<ul>
<li class="chapter" data-level="15.1" data-path="motif-deviations.html"><a href="motif-deviations.html"><i class="fa fa-check"></i><b>15.1</b> Motif Deviations</a></li>
<li class="chapter" data-level="15.2" data-path="archr-and-custom-deviations.html"><a href="archr-and-custom-deviations.html"><i class="fa fa-check"></i><b>15.2</b> ArchR and Custom Deviations</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="footprinting-with-archr.html"><a href="footprinting-with-archr.html"><i class="fa fa-check"></i><b>16</b> Footprinting with ArchR</a>
<ul>
<li class="chapter" data-level="16.1" data-path="motif-footprinting.html"><a href="motif-footprinting.html"><i class="fa fa-check"></i><b>16.1</b> Motif Footprinting</a></li>
<li class="chapter" data-level="16.2" data-path="normalization-of-footprints-for-tn5-bias.html"><a href="normalization-of-footprints-for-tn5-bias.html"><i class="fa fa-check"></i><b>16.2</b> Normalization of Footprints for Tn5 Bias</a></li>
<li class="chapter" data-level="16.3" data-path="feature-footprinting.html"><a href="feature-footprinting.html"><i class="fa fa-check"></i><b>16.3</b> Feature Footprinting</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="integrative-analysis-with-archr.html"><a href="integrative-analysis-with-archr.html"><i class="fa fa-check"></i><b>17</b> Integrative Analysis with ArchR</a>
<ul>
<li class="chapter" data-level="17.1" data-path="creating-low-overlapping-aggregates-of-cells.html"><a href="creating-low-overlapping-aggregates-of-cells.html"><i class="fa fa-check"></i><b>17.1</b> Creating Low-Overlapping Aggregates of Cells</a></li>
<li class="chapter" data-level="17.2" data-path="co-accessibility-with-archr.html"><a href="co-accessibility-with-archr.html"><i class="fa fa-check"></i><b>17.2</b> Co-accessibility with ArchR</a></li>
<li class="chapter" data-level="17.3" data-path="peak2genelinkage-with-archr.html"><a href="peak2genelinkage-with-archr.html"><i class="fa fa-check"></i><b>17.3</b> Peak2GeneLinkage with ArchR</a></li>
<li class="chapter" data-level="17.4" data-path="identification-of-positive-tf-regulators.html"><a href="identification-of-positive-tf-regulators.html"><i class="fa fa-check"></i><b>17.4</b> Identification of Positive TF-Regulators</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="trajectory-analysis-with-archr.html"><a href="trajectory-analysis-with-archr.html"><i class="fa fa-check"></i><b>18</b> Trajectory Analysis with ArchR</a>
<ul>
<li class="chapter" data-level="18.1" data-path="myeloid-trajectory---monocyte-differentiation.html"><a href="myeloid-trajectory---monocyte-differentiation.html"><i class="fa fa-check"></i><b>18.1</b> Myeloid Trajectory - Monocyte Differentiation</a></li>
<li class="chapter" data-level="18.2" data-path="lymphoid-trajectory---b-cell-cifferentiation.html"><a href="lymphoid-trajectory---b-cell-cifferentiation.html"><i class="fa fa-check"></i><b>18.2</b> Lymphoid Trajectory - B Cell Cifferentiation</a></li>
<li class="chapter" data-level="18.3" data-path="monocle3-trajectories.html"><a href="monocle3-trajectories.html"><i class="fa fa-check"></i><b>18.3</b> Monocle3 Trajectories</a></li>
<li class="chapter" data-level="18.4" data-path="slingshot-trajectories.html"><a href="slingshot-trajectories.html"><i class="fa fa-check"></i><b>18.4</b> Slingshot Trajectories</a></li>
<li class="chapter" data-level="18.5" data-path="comparing-the-different-trajectories.html"><a href="comparing-the-different-trajectories.html"><i class="fa fa-check"></i><b>18.5</b> Comparing the different trajectories</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="handling-archr-output.html"><a href="handling-archr-output.html"><i class="fa fa-check"></i><b>19</b> Handling ArchR Output</a>
<ul>
<li class="chapter" data-level="19.1" data-path="exporting-fragment-level-data.html"><a href="exporting-fragment-level-data.html"><i class="fa fa-check"></i><b>19.1</b> Exporting fragment-level data</a></li>
<li class="chapter" data-level="19.2" data-path="exporting-matrix-level-data.html"><a href="exporting-matrix-level-data.html"><i class="fa fa-check"></i><b>19.2</b> Exporting matrix-level data</a></li>
<li class="chapter" data-level="19.3" data-path="exporting-pseudo-bulked-data-to-a-summarizedexperiment.html"><a href="exporting-pseudo-bulked-data-to-a-summarizedexperiment.html"><i class="fa fa-check"></i><b>19.3</b> Exporting pseudo-bulked data to a <code>SummarizedExperiment</code></a></li>
<li class="chapter" data-level="19.4" data-path="exporting-pseudo-bulked-data-to-a-bigwig-file.html"><a href="exporting-pseudo-bulked-data-to-a-bigwig-file.html"><i class="fa fa-check"></i><b>19.4</b> Exporting pseudo-bulked data to a bigWig file</a></li>
</ul></li>
<li class="chapter" data-level="20" data-path="integration-with-bulk-atac-seq.html"><a href="integration-with-bulk-atac-seq.html"><i class="fa fa-check"></i><b>20</b> Integration with bulk ATAC-seq</a>
<ul>
<li class="chapter" data-level="20.1" data-path="projecting-bulk-atac-seq-data.html"><a href="projecting-bulk-atac-seq-data.html"><i class="fa fa-check"></i><b>20.1</b> Projecting bulk ATAC-seq data</a></li>
</ul></li>
<li class="chapter" data-level="21" data-path="multiomic-data-analysis-in-archr.html"><a href="multiomic-data-analysis-in-archr.html"><i class="fa fa-check"></i><b>21</b> Multiomic data analysis in ArchR</a>
<ul>
<li class="chapter" data-level="21.1" data-path="importing-data-and-setting-up-a-multiome-project.html"><a href="importing-data-and-setting-up-a-multiome-project.html"><i class="fa fa-check"></i><b>21.1</b> Importing data and setting up a Multiome project</a></li>
<li class="chapter" data-level="21.2" data-path="analysis-of-multiome-data-in-archr.html"><a href="analysis-of-multiome-data-in-archr.html"><i class="fa fa-check"></i><b>21.2</b> Analysis of multiome data in ArchR</a></li>
</ul></li>
<li class="chapter" data-level="22" data-path="plot-aesthetics-in-archr.html"><a href="plot-aesthetics-in-archr.html"><i class="fa fa-check"></i><b>22</b> Plot aesthetics in ArchR</a>
<ul>
<li class="chapter" data-level="22.1" data-path="using-and-manipulating-palettes-in-archr.html"><a href="using-and-manipulating-palettes-in-archr.html"><i class="fa fa-check"></i><b>22.1</b> Using and manipulating palettes in ArchR</a></li>
</ul></li>
<li class="chapter" data-level="23" data-path="testing-your-archr-installation.html"><a href="testing-your-archr-installation.html"><i class="fa fa-check"></i><b>23</b> Testing Your ArchR Installation</a></li>
<li class="chapter" data-level="24" data-path="session-information.html"><a href="session-information.html"><i class="fa fa-check"></i><b>24</b> Session Information</a></li>
<li class="divider"></li>
<li><a href="https://bookdown.org" target="_blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">ArchR: Robust and scaleable analysis of single-cell chromatin accessibility data.</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="calling-peaks-w-macs2" class="section level2 hasAnchor" number="12.2">
<h2><span class="header-section-number">12.2</span> Calling Peaks w/ Macs2<a href="calling-peaks-w-macs2.html#calling-peaks-w-macs2" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Prior to calling peaks in ArchR, you must run the <code>addGroupCoverages()</code> function! ArchR relies on these group coverage objects to perform peak calling.</p>
<p>As mentioned above, we generate a reproducible peak set in ArchR using the <code>addReproduciblePeakSet()</code> function. By default ArchR attempts to call peaks using MACS2 and this is absolutely the recommended peak caller. ArchR also implements its own native peak caller but we do not recommend using it - this alternative peak calling method is described in the next section.</p>
<p>To call peaks using MACS2, ArchR must be able to find the MACS2 executable. First, ArchR looks in your <code>PATH</code> environment variable. If this is unsuccessful, ArchR attempts to determine if you have installed MACS2 with either <code>pip</code> or <code>pip3</code>. If neither of these is successful, ArchR gives up and provides an error message. If you have MACS2 installed and ArchR cannot find it, you should provide the path to the <code>addReproduciblePeakSet()</code> function via the <code>pathToMacs2</code> parameter. It is important to make sure that the MACS2 executable that ArchR is using is what you expect it to use. Some users have multiple instances of MACS2 installed and ArchR may not automatically select the one that you expect.</p>
<div class="sourceCode" id="cb188"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb188-1"><a href="calling-peaks-w-macs2.html#cb188-1" aria-hidden="true" tabindex="-1"></a>pathToMacs2 <span class="ot">&lt;-</span> <span class="fu">findMacs2</span>()</span></code></pre></div>
<p>You could, of course, manually designate the path to the MACS2 executable instead. This may be necessary on some computational environments.</p>
<!-- We must manually define the path to macs2 because findMacs2() fails given the RStudioServer setup on our server -->
<p>With the path to MACS2 identified, we can then create a reproducible merged peak set w/ MACS2 (5-10 minutes). To avoid bias from pseudo-bulk replicates that have very few cells, we can provide a cutoff for the upper limit of the number of peaks called per cell via the <code>peaksPerCell</code> parameter. This prevents clusters with very few cells from contributing lots of low quality peaks to the merged peak set. There are many other parameters that can be tweaked in <code>addReproduciblePeakSet()</code> - try <code>?addReproduciblePeakSet</code> or check out <a href="https://www.archrproject.com/reference/addReproduciblePeakSet.html">the online function definition</a> for more information. Of particular note, the <code>reproducibility</code> parameter defines what is meant by a “reproducible peak”. For example, <code>reproducibility = "2"</code> means at least 2 pseudo-bulk replicates must have a peak call at this locus and <code>reproducibility = "(n+1)/2"</code> means that the majority of pseudo-bulk replicates must have a peak call at this locus.</p>
<div class="sourceCode" id="cb189"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb189-1"><a href="calling-peaks-w-macs2.html#cb189-1" aria-hidden="true" tabindex="-1"></a>projHeme4 <span class="ot">&lt;-</span> <span class="fu">addReproduciblePeakSet</span>(</span>
<span id="cb189-2"><a href="calling-peaks-w-macs2.html#cb189-2" aria-hidden="true" tabindex="-1"></a>    <span class="at">ArchRProj =</span> projHeme4, </span>
<span id="cb189-3"><a href="calling-peaks-w-macs2.html#cb189-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">groupBy =</span> <span class="st">&quot;Clusters2&quot;</span>, </span>
<span id="cb189-4"><a href="calling-peaks-w-macs2.html#cb189-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">pathToMacs2 =</span> pathToMacs2</span>
<span id="cb189-5"><a href="calling-peaks-w-macs2.html#cb189-5" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb189-6"><a href="calling-peaks-w-macs2.html#cb189-6" aria-hidden="true" tabindex="-1"></a><span class="do">## ArchR logging to : ArchRLogs/ArchR-addReproduciblePeakSet-371b0f34d5c2-Date-2022-12-23_Time-07-30-35.log</span></span>
<span id="cb189-7"><a href="calling-peaks-w-macs2.html#cb189-7" aria-hidden="true" tabindex="-1"></a><span class="do">## If there is an issue, please report to github with logFile!</span></span>
<span id="cb189-8"><a href="calling-peaks-w-macs2.html#cb189-8" aria-hidden="true" tabindex="-1"></a><span class="do">## Calling Peaks with Macs2</span></span>
<span id="cb189-9"><a href="calling-peaks-w-macs2.html#cb189-9" aria-hidden="true" tabindex="-1"></a><span class="do">## 2022-12-23 07:30:37 : Peak Calling Parameters!, 0.037 mins elapsed.</span></span>
<span id="cb189-10"><a href="calling-peaks-w-macs2.html#cb189-10" aria-hidden="true" tabindex="-1"></a><span class="do">##                 Group nCells nCellsUsed nReplicates nMin nMax maxPeaks</span></span>
<span id="cb189-11"><a href="calling-peaks-w-macs2.html#cb189-11" aria-hidden="true" tabindex="-1"></a><span class="do">## B                   B    428        423           2  166  257   150000</span></span>
<span id="cb189-12"><a href="calling-peaks-w-macs2.html#cb189-12" aria-hidden="true" tabindex="-1"></a><span class="do">## CD4.M           CD4.M    614        584           2   84  500   150000</span></span>
<span id="cb189-13"><a href="calling-peaks-w-macs2.html#cb189-13" aria-hidden="true" tabindex="-1"></a><span class="do">## CD4.N           CD4.N   1239        540           2   40  500   150000</span></span>
<span id="cb189-14"><a href="calling-peaks-w-macs2.html#cb189-14" aria-hidden="true" tabindex="-1"></a><span class="do">## CLP               CLP    383        383           2   87  296   150000</span></span>
<span id="cb189-15"><a href="calling-peaks-w-macs2.html#cb189-15" aria-hidden="true" tabindex="-1"></a><span class="do">## Erythroid   Erythroid    709        670           2  170  500   150000</span></span>
<span id="cb189-16"><a href="calling-peaks-w-macs2.html#cb189-16" aria-hidden="true" tabindex="-1"></a><span class="do">## GMP               GMP   1200        796           2  296  500   150000</span></span>
<span id="cb189-17"><a href="calling-peaks-w-macs2.html#cb189-17" aria-hidden="true" tabindex="-1"></a><span class="do">## Mono             Mono   2672       1000           2  500  500   150000</span></span>
<span id="cb189-18"><a href="calling-peaks-w-macs2.html#cb189-18" aria-hidden="true" tabindex="-1"></a><span class="do">## NK                 NK    932        830           2  330  500   150000</span></span>
<span id="cb189-19"><a href="calling-peaks-w-macs2.html#cb189-19" aria-hidden="true" tabindex="-1"></a><span class="do">## pDC               pDC    315        305           2  151  154   150000</span></span>
<span id="cb189-20"><a href="calling-peaks-w-macs2.html#cb189-20" aria-hidden="true" tabindex="-1"></a><span class="do">## PreB             PreB    354        354           2   40  314   150000</span></span>
<span id="cb189-21"><a href="calling-peaks-w-macs2.html#cb189-21" aria-hidden="true" tabindex="-1"></a><span class="do">## Progenitor Progenitor   1404        638           2  138  500   150000</span></span>
<span id="cb189-22"><a href="calling-peaks-w-macs2.html#cb189-22" aria-hidden="true" tabindex="-1"></a><span class="do">## 2022-12-23 07:30:37 : Batching Peak Calls!, 0.037 mins elapsed.</span></span>
<span id="cb189-23"><a href="calling-peaks-w-macs2.html#cb189-23" aria-hidden="true" tabindex="-1"></a><span class="do">## 2022-12-23 07:30:37 : Batch Execution w/ safelapply!, 0 mins elapsed.</span></span>
<span id="cb189-24"><a href="calling-peaks-w-macs2.html#cb189-24" aria-hidden="true" tabindex="-1"></a><span class="do">## 2022-12-23 07:32:20 : Identifying Reproducible Peaks!, 1.751 mins elapsed.</span></span>
<span id="cb189-25"><a href="calling-peaks-w-macs2.html#cb189-25" aria-hidden="true" tabindex="-1"></a><span class="do">## 2022-12-23 07:32:30 : Creating Union Peak Set!, 1.912 mins elapsed.</span></span>
<span id="cb189-26"><a href="calling-peaks-w-macs2.html#cb189-26" aria-hidden="true" tabindex="-1"></a><span class="do">## Converged after 8 iterations!</span></span>
<span id="cb189-27"><a href="calling-peaks-w-macs2.html#cb189-27" aria-hidden="true" tabindex="-1"></a><span class="do">## Plotting Ggplot!</span></span>
<span id="cb189-28"><a href="calling-peaks-w-macs2.html#cb189-28" aria-hidden="true" tabindex="-1"></a><span class="do">## 2022-12-23 07:32:36 : Finished Creating Union Peak Set (142475)!, 2.021 mins elapsed.</span></span></code></pre></div>
<p>The iterative overlap procedure that ArchR uses to create this reproducible peak set occurs in two stages. During the first step of <code>addReproduciblePeakSet()</code>, a reproducible peak set is created for each group indicated by <code>groupBy</code>. Using the psuedobulk replicates generated by <code>addGroupCoverages()</code>, ArchR identifies peaks that are reproducible across the replicates for each group indicated by the <code>groupBy</code> argument. The score of those peak calls is stored in the <code>score</code> column and the quantile rank of that score for the individual pseudobulk replicate that it was identified in is stored in <code>replicateScoreQuantile</code>. It is this quantile normalization that accounts for differences in number of fragments per group etc. that affect peak calling. For each of these groups, ArchR saves the group-specific peak set as a <code>GRanges</code> object in the <code>PeakCalls</code> subdirectory of the <code>ArchRProject</code>. Additionally, ArchR saves a a <code>GRanges</code> object that contains the summit calls for each pseudobulk replicate within the <code>./PeakCalls/ReplicateCalls</code> directory.</p>
<!-- Sleep for 20 seconds because it seems like the files dont finish saving before the next list.files command -->
<div class="sourceCode" id="cb190"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb190-1"><a href="calling-peaks-w-macs2.html#cb190-1" aria-hidden="true" tabindex="-1"></a><span class="fu">list.files</span>(<span class="at">path=</span><span class="fu">paste0</span>(<span class="fu">getOutputDirectory</span>(<span class="at">ArchRProj =</span> projHeme4),<span class="st">&quot;/PeakCalls&quot;</span>))</span>
<span id="cb190-2"><a href="calling-peaks-w-macs2.html#cb190-2" aria-hidden="true" tabindex="-1"></a><span class="do">##  [1] &quot;B-reproduciblePeaks.gr.rds&quot;          &quot;CD4.M-reproduciblePeaks.gr.rds&quot;     </span></span>
<span id="cb190-3"><a href="calling-peaks-w-macs2.html#cb190-3" aria-hidden="true" tabindex="-1"></a><span class="do">##  [3] &quot;CD4.N-reproduciblePeaks.gr.rds&quot;      &quot;CLP-reproduciblePeaks.gr.rds&quot;       </span></span>
<span id="cb190-4"><a href="calling-peaks-w-macs2.html#cb190-4" aria-hidden="true" tabindex="-1"></a><span class="do">##  [5] &quot;Clusters2&quot;                           &quot;Erythroid-reproduciblePeaks.gr.rds&quot; </span></span>
<span id="cb190-5"><a href="calling-peaks-w-macs2.html#cb190-5" aria-hidden="true" tabindex="-1"></a><span class="do">##  [7] &quot;GMP-reproduciblePeaks.gr.rds&quot;        &quot;Mono-reproduciblePeaks.gr.rds&quot;      </span></span>
<span id="cb190-6"><a href="calling-peaks-w-macs2.html#cb190-6" aria-hidden="true" tabindex="-1"></a><span class="do">##  [9] &quot;NK-reproduciblePeaks.gr.rds&quot;         &quot;pDC-reproduciblePeaks.gr.rds&quot;       </span></span>
<span id="cb190-7"><a href="calling-peaks-w-macs2.html#cb190-7" aria-hidden="true" tabindex="-1"></a><span class="do">## [11] &quot;PreB-reproduciblePeaks.gr.rds&quot;       &quot;Progenitor-reproduciblePeaks.gr.rds&quot;</span></span></code></pre></div>
<p>During the second step of <code>addReproduciblePeakSet()</code>, ArchR merges these group-specific peak sets into a single union peak set. The quantile rank of the peak <code>score</code> across the group is stored in <code>groupScoreQuantile</code> and the group it comes from is stored in <code>GroupReplicate</code>. <code>GroupReplicate</code> is an annotation for the group from which each peak originated during the process of iterative overlap peak merging. It is important to note that these annotations do not inherently mean that the given peak was only called in that cell group, rather that the annotated group had the highest normalized significance for that peak call.</p>
<p>Each <code>ArchRProject</code> object can only contain a single peak set at a time. As such, we assign the output of <code>addReproduciblePeakSet()</code> to our desired <code>ArchRProject</code>. If you would like to experiment with different peak sets, you have two options. Option #1 - you can save multiple copies of your <code>ArchRProject</code> using <code>saveArchRProject()</code>. Option #2 - you can create multiple peak sets, store them as <code>GenomicRanges</code> objects, and change your peak set using the <code>addPeakSet()</code> function.</p>
<p>The <code>addPeakSet()</code> function can be used to set any arbitrary peak set as the peak set for your <code>ArchRProject</code>.</p>
<p>To retrieve the peak set stored in your <code>ArchRProject</code> as a <code>GRanges</code> object, we use the <code>getPeakSet()</code> function.</p>
<div class="sourceCode" id="cb191"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb191-1"><a href="calling-peaks-w-macs2.html#cb191-1" aria-hidden="true" tabindex="-1"></a><span class="fu">getPeakSet</span>(projHeme4)</span>
<span id="cb191-2"><a href="calling-peaks-w-macs2.html#cb191-2" aria-hidden="true" tabindex="-1"></a><span class="do">## GRanges object with 142475 ranges and 13 metadata columns:</span></span>
<span id="cb191-3"><a href="calling-peaks-w-macs2.html#cb191-3" aria-hidden="true" tabindex="-1"></a><span class="do">##         seqnames              ranges strand |     score replicateScoreQuantile</span></span>
<span id="cb191-4"><a href="calling-peaks-w-macs2.html#cb191-4" aria-hidden="true" tabindex="-1"></a><span class="do">##            &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt;              &lt;numeric&gt;</span></span>
<span id="cb191-5"><a href="calling-peaks-w-macs2.html#cb191-5" aria-hidden="true" tabindex="-1"></a><span class="do">##    Mono     chr1       752494-752994      * |   35.0551                  0.829</span></span>
<span id="cb191-6"><a href="calling-peaks-w-macs2.html#cb191-6" aria-hidden="true" tabindex="-1"></a><span class="do">##       B     chr1       762696-763196      * |   48.7983                  0.939</span></span>
<span id="cb191-7"><a href="calling-peaks-w-macs2.html#cb191-7" aria-hidden="true" tabindex="-1"></a><span class="do">##       B     chr1       801002-801502      * |   11.4240                  0.609</span></span>
<span id="cb191-8"><a href="calling-peaks-w-macs2.html#cb191-8" aria-hidden="true" tabindex="-1"></a><span class="do">##     GMP     chr1       805065-805565      * |   30.1400                  0.953</span></span>
<span id="cb191-9"><a href="calling-peaks-w-macs2.html#cb191-9" aria-hidden="true" tabindex="-1"></a><span class="do">##     CLP     chr1       845326-845826      * |    2.7681                  0.704</span></span>
<span id="cb191-10"><a href="calling-peaks-w-macs2.html#cb191-10" aria-hidden="true" tabindex="-1"></a><span class="do">##     ...      ...                 ...    ... .       ...                    ...</span></span>
<span id="cb191-11"><a href="calling-peaks-w-macs2.html#cb191-11" aria-hidden="true" tabindex="-1"></a><span class="do">##   CD4.M     chrX 154493043-154493543      * |   7.04640                  0.334</span></span>
<span id="cb191-12"><a href="calling-peaks-w-macs2.html#cb191-12" aria-hidden="true" tabindex="-1"></a><span class="do">##      NK     chrX 154493558-154494058      * |  28.07310                  0.831</span></span>
<span id="cb191-13"><a href="calling-peaks-w-macs2.html#cb191-13" aria-hidden="true" tabindex="-1"></a><span class="do">##      NK     chrX 154807254-154807754      * |   4.58050                  0.405</span></span>
<span id="cb191-14"><a href="calling-peaks-w-macs2.html#cb191-14" aria-hidden="true" tabindex="-1"></a><span class="do">##     GMP     chrX 154842383-154842883      * |   8.14111                  0.750</span></span>
<span id="cb191-15"><a href="calling-peaks-w-macs2.html#cb191-15" aria-hidden="true" tabindex="-1"></a><span class="do">##     GMP     chrX 154996993-154997493      * |   4.89826                  0.507</span></span>
<span id="cb191-16"><a href="calling-peaks-w-macs2.html#cb191-16" aria-hidden="true" tabindex="-1"></a><span class="do">##         groupScoreQuantile Reproducibility         GroupReplicate</span></span>
<span id="cb191-17"><a href="calling-peaks-w-macs2.html#cb191-17" aria-hidden="true" tabindex="-1"></a><span class="do">##                  &lt;numeric&gt;       &lt;numeric&gt;            &lt;character&gt;</span></span>
<span id="cb191-18"><a href="calling-peaks-w-macs2.html#cb191-18" aria-hidden="true" tabindex="-1"></a><span class="do">##    Mono              0.685               2  Mono._.scATAC_BMMC_R1</span></span>
<span id="cb191-19"><a href="calling-peaks-w-macs2.html#cb191-19" aria-hidden="true" tabindex="-1"></a><span class="do">##       B              0.861               2     B._.scATAC_PBMC_R1</span></span>
<span id="cb191-20"><a href="calling-peaks-w-macs2.html#cb191-20" aria-hidden="true" tabindex="-1"></a><span class="do">##       B              0.316               2     B._.scATAC_PBMC_R1</span></span>
<span id="cb191-21"><a href="calling-peaks-w-macs2.html#cb191-21" aria-hidden="true" tabindex="-1"></a><span class="do">##     GMP              0.892               2   GMP._.scATAC_BMMC_R1</span></span>
<span id="cb191-22"><a href="calling-peaks-w-macs2.html#cb191-22" aria-hidden="true" tabindex="-1"></a><span class="do">##     CLP              0.310               2   CLP._.scATAC_BMMC_R1</span></span>
<span id="cb191-23"><a href="calling-peaks-w-macs2.html#cb191-23" aria-hidden="true" tabindex="-1"></a><span class="do">##     ...                ...             ...                    ...</span></span>
<span id="cb191-24"><a href="calling-peaks-w-macs2.html#cb191-24" aria-hidden="true" tabindex="-1"></a><span class="do">##   CD4.M              0.061               2 CD4.M._.scATAC_PBMC_R1</span></span>
<span id="cb191-25"><a href="calling-peaks-w-macs2.html#cb191-25" aria-hidden="true" tabindex="-1"></a><span class="do">##      NK              0.709               2    NK._.scATAC_BMMC_R1</span></span>
<span id="cb191-26"><a href="calling-peaks-w-macs2.html#cb191-26" aria-hidden="true" tabindex="-1"></a><span class="do">##      NK              0.133               2    NK._.scATAC_BMMC_R1</span></span>
<span id="cb191-27"><a href="calling-peaks-w-macs2.html#cb191-27" aria-hidden="true" tabindex="-1"></a><span class="do">##     GMP              0.475               2   GMP._.scATAC_BMMC_R1</span></span>
<span id="cb191-28"><a href="calling-peaks-w-macs2.html#cb191-28" aria-hidden="true" tabindex="-1"></a><span class="do">##     GMP              0.200               2 GMP._.scATAC_CD34_BM..</span></span>
<span id="cb191-29"><a href="calling-peaks-w-macs2.html#cb191-29" aria-hidden="true" tabindex="-1"></a><span class="do">##         distToGeneStart nearestGene    peakType distToTSS  nearestTSS        GC</span></span>
<span id="cb191-30"><a href="calling-peaks-w-macs2.html#cb191-30" aria-hidden="true" tabindex="-1"></a><span class="do">##               &lt;integer&gt; &lt;character&gt; &lt;character&gt; &lt;integer&gt; &lt;character&gt; &lt;numeric&gt;</span></span>
<span id="cb191-31"><a href="calling-peaks-w-macs2.html#cb191-31" aria-hidden="true" tabindex="-1"></a><span class="do">##    Mono           10157   LINC00115      Distal     10157  uc010nxx.2    0.4850</span></span>
<span id="cb191-32"><a href="calling-peaks-w-macs2.html#cb191-32" aria-hidden="true" tabindex="-1"></a><span class="do">##       B              24   LINC01128    Promoter        24  uc031pjj.1    0.6966</span></span>
<span id="cb191-33"><a href="calling-peaks-w-macs2.html#cb191-33" aria-hidden="true" tabindex="-1"></a><span class="do">##       B           10929      FAM41C      Distal     10929  uc001abt.4    0.4371</span></span>
<span id="cb191-34"><a href="calling-peaks-w-macs2.html#cb191-34" aria-hidden="true" tabindex="-1"></a><span class="do">##     GMP            6866      FAM41C    Intronic      6866  uc001abt.4    0.7345</span></span>
<span id="cb191-35"><a href="calling-peaks-w-macs2.html#cb191-35" aria-hidden="true" tabindex="-1"></a><span class="do">##     CLP            9240   LINC02593      Distal      1238  uc001abu.1    0.7904</span></span>
<span id="cb191-36"><a href="calling-peaks-w-macs2.html#cb191-36" aria-hidden="true" tabindex="-1"></a><span class="do">##     ...             ...         ...         ...       ...         ...       ...</span></span>
<span id="cb191-37"><a href="calling-peaks-w-macs2.html#cb191-37" aria-hidden="true" tabindex="-1"></a><span class="do">##   CD4.M             558      RAB39B    Intronic       558  uc004fne.3    0.5609</span></span>
<span id="cb191-38"><a href="calling-peaks-w-macs2.html#cb191-38" aria-hidden="true" tabindex="-1"></a><span class="do">##      NK              43      RAB39B    Promoter        43  uc004fne.3    0.5828</span></span>
<span id="cb191-39"><a href="calling-peaks-w-macs2.html#cb191-39" aria-hidden="true" tabindex="-1"></a><span class="do">##      NK           35117       TMLHE    Intronic     35117  uc004fnn.3    0.4990</span></span>
<span id="cb191-40"><a href="calling-peaks-w-macs2.html#cb191-40" aria-hidden="true" tabindex="-1"></a><span class="do">##     GMP              10       TMLHE    Promoter        10  uc004fnn.3    0.6088</span></span>
<span id="cb191-41"><a href="calling-peaks-w-macs2.html#cb191-41" aria-hidden="true" tabindex="-1"></a><span class="do">##     GMP          154620       TMLHE      Distal       207  uc004fnq.1    0.5449</span></span>
<span id="cb191-42"><a href="calling-peaks-w-macs2.html#cb191-42" aria-hidden="true" tabindex="-1"></a><span class="do">##               idx         N</span></span>
<span id="cb191-43"><a href="calling-peaks-w-macs2.html#cb191-43" aria-hidden="true" tabindex="-1"></a><span class="do">##         &lt;integer&gt; &lt;numeric&gt;</span></span>
<span id="cb191-44"><a href="calling-peaks-w-macs2.html#cb191-44" aria-hidden="true" tabindex="-1"></a><span class="do">##    Mono         1         0</span></span>
<span id="cb191-45"><a href="calling-peaks-w-macs2.html#cb191-45" aria-hidden="true" tabindex="-1"></a><span class="do">##       B         2         0</span></span>
<span id="cb191-46"><a href="calling-peaks-w-macs2.html#cb191-46" aria-hidden="true" tabindex="-1"></a><span class="do">##       B         3         0</span></span>
<span id="cb191-47"><a href="calling-peaks-w-macs2.html#cb191-47" aria-hidden="true" tabindex="-1"></a><span class="do">##     GMP         4         0</span></span>
<span id="cb191-48"><a href="calling-peaks-w-macs2.html#cb191-48" aria-hidden="true" tabindex="-1"></a><span class="do">##     CLP         5         0</span></span>
<span id="cb191-49"><a href="calling-peaks-w-macs2.html#cb191-49" aria-hidden="true" tabindex="-1"></a><span class="do">##     ...       ...       ...</span></span>
<span id="cb191-50"><a href="calling-peaks-w-macs2.html#cb191-50" aria-hidden="true" tabindex="-1"></a><span class="do">##   CD4.M      3416         0</span></span>
<span id="cb191-51"><a href="calling-peaks-w-macs2.html#cb191-51" aria-hidden="true" tabindex="-1"></a><span class="do">##      NK      3417         0</span></span>
<span id="cb191-52"><a href="calling-peaks-w-macs2.html#cb191-52" aria-hidden="true" tabindex="-1"></a><span class="do">##      NK      3418         0</span></span>
<span id="cb191-53"><a href="calling-peaks-w-macs2.html#cb191-53" aria-hidden="true" tabindex="-1"></a><span class="do">##     GMP      3419         0</span></span>
<span id="cb191-54"><a href="calling-peaks-w-macs2.html#cb191-54" aria-hidden="true" tabindex="-1"></a><span class="do">##     GMP      3420         0</span></span>
<span id="cb191-55"><a href="calling-peaks-w-macs2.html#cb191-55" aria-hidden="true" tabindex="-1"></a><span class="do">##   -------</span></span>
<span id="cb191-56"><a href="calling-peaks-w-macs2.html#cb191-56" aria-hidden="true" tabindex="-1"></a><span class="do">##   seqinfo: 23 sequences from an unspecified genome; no seqlengths</span></span></code></pre></div>
<p>The peak sets from <code>addReproduciblePeakSet()</code> are annotated with lots of metadata about each peak region. The <code>score</code> represents the normalized <code>-log10(pval)</code> returned by MACS2. There are other annotations such as <code>distToGeneStart</code> and <code>distToTSS</code> which are determined based on the contents of <code>getGeneAnnotation(ArchRProj)$genes</code> and <code>getGeneAnnotation(ArchRProj)$TSS</code> respectively. While it may seem like these should be identical, a single gene can have multiple TSSs and ArchR needs to deal with this multi-mapping issue, which we have done by including both annotations. For example, in ArchR’s default hg38 <code>geneAnnotation</code>, there are 23,274 genes and 49,052 TSSs. One confusing piece of information annotated in the peak set is the <code>idx</code> column which does <strong>not</strong> correspond to a unique peak identifier. Instead the <code>idx</code> column is actually only unique per-chromosome, so for a peak located on “chr1” with and <code>idx</code> of “5”, this means that this is the 5th peak in sequential order on “chr1” (but there would also be a peak with <code>idx</code> of “5” on “chr2”).</p>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="the-iterative-overlap-peak-merging-procedure.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="calling-peaks-w-tilematrix.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": null
},
"fontsettings": null,
"edit": null,
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "none"
},
"info": false
});
});
</script>

</body>

</html>
