<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>5.1 Creating An ArchRProject | ArchR: Robust and scaleable analysis of single-cell chromatin accessibility data.</title>
  <meta name="description" content="A guide to ArchR" />
  <meta name="generator" content="bookdown 0.41 and GitBook 2.6.7" />

  <meta property="og:title" content="5.1 Creating An ArchRProject | ArchR: Robust and scaleable analysis of single-cell chromatin accessibility data." />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="A guide to ArchR" />
  <meta name="github-repo" content="GreenleafLab/ArchR" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="5.1 Creating An ArchRProject | ArchR: Robust and scaleable analysis of single-cell chromatin accessibility data." />
  
  <meta name="twitter:description" content="A guide to ArchR" />
  

<meta name="author" content="Jeffrey Granja and Ryan Corces" />


<meta name="date" content="2025-02-06" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="images/favicon.ico" type="image/x-icon" />
<link rel="prev" href="creating-an-archrproject.html"/>
<link rel="next" href="manipulating-an-archrproject.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<link href="libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="../index.html">Back to ArchR Home</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><img src="images/ArchR_Logo_Integrated.png" alt="" width="200"></a></li>
<li class="chapter" data-level="1" data-path="debugging-and-troubleshooting-in-archr.html"><a href="debugging-and-troubleshooting-in-archr.html"><i class="fa fa-check"></i><b>1</b> Debugging and Troubleshooting in ArchR</a></li>
<li class="chapter" data-level="2" data-path="manage-archrs-dependencies.html"><a href="manage-archrs-dependencies.html"><i class="fa fa-check"></i><b>2</b> Manage ArchR’s Dependencies</a>
<ul>
<li class="chapter" data-level="2.1" data-path="using-renv-to-manage-dependencies.html"><a href="using-renv-to-manage-dependencies.html"><i class="fa fa-check"></i><b>2.1</b> Using renv to manage dependencies</a></li>
<li class="chapter" data-level="2.2" data-path="other-environment-managers.html"><a href="other-environment-managers.html"><i class="fa fa-check"></i><b>2.2</b> Other environment managers</a></li>
<li class="chapter" data-level="2.3" data-path="using-docker-for-pre-compiled-environment-setup.html"><a href="using-docker-for-pre-compiled-environment-setup.html"><i class="fa fa-check"></i><b>2.3</b> Using Docker for pre-compiled environment setup</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="getting-started-with-archr.html"><a href="getting-started-with-archr.html"><i class="fa fa-check"></i><b>3</b> Getting Started with ArchR</a>
<ul>
<li class="chapter" data-level="3.1" data-path="a-brief-primer-on-atac-seq-terminology.html"><a href="a-brief-primer-on-atac-seq-terminology.html"><i class="fa fa-check"></i><b>3.1</b> A Brief Primer on ATAC-seq Terminology</a></li>
<li class="chapter" data-level="3.2" data-path="why-use-archr.html"><a href="why-use-archr.html"><i class="fa fa-check"></i><b>3.2</b> Why use ArchR?</a></li>
<li class="chapter" data-level="3.3" data-path="what-is-an-arrow-file-archrproject.html"><a href="what-is-an-arrow-file-archrproject.html"><i class="fa fa-check"></i><b>3.3</b> What is an Arrow file / <code>ArchRProject</code>?</a></li>
<li class="chapter" data-level="3.4" data-path="input-file-types-in-archr.html"><a href="input-file-types-in-archr.html"><i class="fa fa-check"></i><b>3.4</b> Input File Types in ArchR</a></li>
<li class="chapter" data-level="3.5" data-path="getting-set-up.html"><a href="getting-set-up.html"><i class="fa fa-check"></i><b>3.5</b> Getting Set Up</a></li>
<li class="chapter" data-level="3.6" data-path="creating-arrow-files.html"><a href="creating-arrow-files.html"><i class="fa fa-check"></i><b>3.6</b> Creating Arrow Files</a></li>
<li class="chapter" data-level="3.7" data-path="per-cell-quality-control.html"><a href="per-cell-quality-control.html"><i class="fa fa-check"></i><b>3.7</b> Per-cell Quality Control</a></li>
<li class="chapter" data-level="3.8" data-path="using-bam-files-for-arrow-file-creation.html"><a href="using-bam-files-for-arrow-file-creation.html"><i class="fa fa-check"></i><b>3.8</b> Using BAM files for Arrow File creation</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="doublet-inference-with-archr.html"><a href="doublet-inference-with-archr.html"><i class="fa fa-check"></i><b>4</b> Doublet Inference with ArchR</a>
<ul>
<li class="chapter" data-level="4.1" data-path="how-does-doublet-identification-work-in-archr.html"><a href="how-does-doublet-identification-work-in-archr.html"><i class="fa fa-check"></i><b>4.1</b> How does doublet identification work in ArchR?</a></li>
<li class="chapter" data-level="4.2" data-path="inferring-scatac-seq-doublets-with-archr.html"><a href="inferring-scatac-seq-doublets-with-archr.html"><i class="fa fa-check"></i><b>4.2</b> Inferring scATAC-seq Doublets with ArchR</a></li>
<li class="chapter" data-level="4.3" data-path="using-demuxlet-with-archr.html"><a href="using-demuxlet-with-archr.html"><i class="fa fa-check"></i><b>4.3</b> Using <em>demuxlet</em> with ArchR</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="creating-an-archrproject.html"><a href="creating-an-archrproject.html"><i class="fa fa-check"></i><b>5</b> Creating an ArchRProject</a>
<ul>
<li class="chapter" data-level="5.1" data-path="creating-an-archrproject-1.html"><a href="creating-an-archrproject-1.html"><i class="fa fa-check"></i><b>5.1</b> Creating An ArchRProject</a></li>
<li class="chapter" data-level="5.2" data-path="manipulating-an-archrproject.html"><a href="manipulating-an-archrproject.html"><i class="fa fa-check"></i><b>5.2</b> Manipulating An ArchRProject</a></li>
<li class="chapter" data-level="5.3" data-path="plotting-sample-statistics-from-an-archrproject.html"><a href="plotting-sample-statistics-from-an-archrproject.html"><i class="fa fa-check"></i><b>5.3</b> Plotting Sample Statistics from an ArchRProject</a></li>
<li class="chapter" data-level="5.4" data-path="plotting-sample-fragment-size-distribution-and-tss-enrichment-profiles..html"><a href="plotting-sample-fragment-size-distribution-and-tss-enrichment-profiles..html"><i class="fa fa-check"></i><b>5.4</b> Plotting Sample Fragment Size Distribution and TSS Enrichment Profiles.</a></li>
<li class="chapter" data-level="5.5" data-path="saving-and-loading-an-archrproject.html"><a href="saving-and-loading-an-archrproject.html"><i class="fa fa-check"></i><b>5.5</b> Saving and Loading an <code>ArchRProject</code></a></li>
<li class="chapter" data-level="5.6" data-path="filtering-doublets-from-an-archrproject.html"><a href="filtering-doublets-from-an-archrproject.html"><i class="fa fa-check"></i><b>5.6</b> Filtering Doublets from an ArchRProject</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="dimensionality-reduction-with-archr.html"><a href="dimensionality-reduction-with-archr.html"><i class="fa fa-check"></i><b>6</b> Dimensionality Reduction with ArchR</a>
<ul>
<li class="chapter" data-level="6.1" data-path="archrs-lsi-implementation.html"><a href="archrs-lsi-implementation.html"><i class="fa fa-check"></i><b>6.1</b> ArchR’s LSI Implementation</a></li>
<li class="chapter" data-level="6.2" data-path="iterative-latent-semantic-indexing-lsi.html"><a href="iterative-latent-semantic-indexing-lsi.html"><i class="fa fa-check"></i><b>6.2</b> Iterative Latent Semantic Indexing (LSI)</a></li>
<li class="chapter" data-level="6.3" data-path="estimated-lsi.html"><a href="estimated-lsi.html"><i class="fa fa-check"></i><b>6.3</b> Estimated LSI</a></li>
<li class="chapter" data-level="6.4" data-path="batch-effect-correction-wtih-harmony.html"><a href="batch-effect-correction-wtih-harmony.html"><i class="fa fa-check"></i><b>6.4</b> Batch Effect Correction wtih Harmony</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="clustering-with-archr.html"><a href="clustering-with-archr.html"><i class="fa fa-check"></i><b>7</b> Clustering with ArchR</a>
<ul>
<li class="chapter" data-level="7.1" data-path="clustering-using-seurats-findclusters-function.html"><a href="clustering-using-seurats-findclusters-function.html"><i class="fa fa-check"></i><b>7.1</b> Clustering using Seurat’s <code>FindClusters()</code> function</a></li>
<li class="chapter" data-level="7.2" data-path="clustering-using-scran.html"><a href="clustering-using-scran.html"><i class="fa fa-check"></i><b>7.2</b> Clustering using <code>scran</code></a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="single-cell-embeddings.html"><a href="single-cell-embeddings.html"><i class="fa fa-check"></i><b>8</b> Single-cell Embeddings</a>
<ul>
<li class="chapter" data-level="8.1" data-path="uniform-manifold-approximation-and-projection-umap.html"><a href="uniform-manifold-approximation-and-projection-umap.html"><i class="fa fa-check"></i><b>8.1</b> Uniform Manifold Approximation and Projection (UMAP)</a></li>
<li class="chapter" data-level="8.2" data-path="t-stocastic-neighbor-embedding-t-sne.html"><a href="t-stocastic-neighbor-embedding-t-sne.html"><i class="fa fa-check"></i><b>8.2</b> t-Stocastic Neighbor Embedding (t-SNE)</a></li>
<li class="chapter" data-level="8.3" data-path="dimensionality-reduction-after-harmony.html"><a href="dimensionality-reduction-after-harmony.html"><i class="fa fa-check"></i><b>8.3</b> Dimensionality Reduction After Harmony</a></li>
<li class="chapter" data-level="8.4" data-path="highlighting-specific-cells-on-an-embedding.html"><a href="highlighting-specific-cells-on-an-embedding.html"><i class="fa fa-check"></i><b>8.4</b> Highlighting specific cells on an embedding</a></li>
<li class="chapter" data-level="8.5" data-path="importing-an-embedding-from-external-software.html"><a href="importing-an-embedding-from-external-software.html"><i class="fa fa-check"></i><b>8.5</b> Importing an embedding from external software</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="gene-scores-and-marker-genes-with-archr.html"><a href="gene-scores-and-marker-genes-with-archr.html"><i class="fa fa-check"></i><b>9</b> Gene Scores and Marker Genes with ArchR</a>
<ul>
<li class="chapter" data-level="9.1" data-path="calculating-gene-scores-in-archr.html"><a href="calculating-gene-scores-in-archr.html"><i class="fa fa-check"></i><b>9.1</b> Calculating Gene Scores in ArchR</a></li>
<li class="chapter" data-level="9.2" data-path="identification-of-marker-features.html"><a href="identification-of-marker-features.html"><i class="fa fa-check"></i><b>9.2</b> Identification of Marker Features</a></li>
<li class="chapter" data-level="9.3" data-path="identifying-marker-genes.html"><a href="identifying-marker-genes.html"><i class="fa fa-check"></i><b>9.3</b> Identifying Marker Genes</a></li>
<li class="chapter" data-level="9.4" data-path="visualizing-marker-genes-on-an-embedding.html"><a href="visualizing-marker-genes-on-an-embedding.html"><i class="fa fa-check"></i><b>9.4</b> Visualizing Marker Genes on an Embedding</a></li>
<li class="chapter" data-level="9.5" data-path="marker-genes-imputation-with-magic.html"><a href="marker-genes-imputation-with-magic.html"><i class="fa fa-check"></i><b>9.5</b> Marker Genes Imputation with MAGIC</a></li>
<li class="chapter" data-level="9.6" data-path="module-scores.html"><a href="module-scores.html"><i class="fa fa-check"></i><b>9.6</b> Module Scores</a></li>
<li class="chapter" data-level="9.7" data-path="track-plotting-with-archrbrowser.html"><a href="track-plotting-with-archrbrowser.html"><i class="fa fa-check"></i><b>9.7</b> Track Plotting with ArchRBrowser</a></li>
<li class="chapter" data-level="9.8" data-path="launching-the-interactive-archrbrowser.html"><a href="launching-the-interactive-archrbrowser.html"><i class="fa fa-check"></i><b>9.8</b> Launching the Interactive ArchRBrowser</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="defining-cluster-identity-with-scrna-seq.html"><a href="defining-cluster-identity-with-scrna-seq.html"><i class="fa fa-check"></i><b>10</b> Defining Cluster Identity with scRNA-seq</a>
<ul>
<li class="chapter" data-level="10.1" data-path="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html"><i class="fa fa-check"></i><b>10.1</b> Cross-platform linkage of scATAC-seq cells with scRNA-seq cells</a></li>
<li class="chapter" data-level="10.2" data-path="adding-pseudo-scrna-seq-profiles-for-each-scatac-seq-cell.html"><a href="adding-pseudo-scrna-seq-profiles-for-each-scatac-seq-cell.html"><i class="fa fa-check"></i><b>10.2</b> Adding Pseudo-scRNA-seq profiles for each scATAC-seq cell</a></li>
<li class="chapter" data-level="10.3" data-path="labeling-scatac-seq-clusters-with-scrna-seq-information.html"><a href="labeling-scatac-seq-clusters-with-scrna-seq-information.html"><i class="fa fa-check"></i><b>10.3</b> Labeling scATAC-seq clusters with scRNA-seq information</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="pseudo-bulk-replicates-in-archr.html"><a href="pseudo-bulk-replicates-in-archr.html"><i class="fa fa-check"></i><b>11</b> Pseudo-bulk Replicates in ArchR</a>
<ul>
<li class="chapter" data-level="11.1" data-path="how-does-archr-make-pseudo-bulk-replicates.html"><a href="how-does-archr-make-pseudo-bulk-replicates.html"><i class="fa fa-check"></i><b>11.1</b> How Does ArchR Make Pseudo-bulk Replicates?</a></li>
<li class="chapter" data-level="11.2" data-path="making-pseudo-bulk-replicates.html"><a href="making-pseudo-bulk-replicates.html"><i class="fa fa-check"></i><b>11.2</b> Making Pseudo-bulk Replicates</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="calling-peaks-with-archr.html"><a href="calling-peaks-with-archr.html"><i class="fa fa-check"></i><b>12</b> Calling Peaks with ArchR</a>
<ul>
<li class="chapter" data-level="12.1" data-path="the-iterative-overlap-peak-merging-procedure.html"><a href="the-iterative-overlap-peak-merging-procedure.html"><i class="fa fa-check"></i><b>12.1</b> The Iterative Overlap Peak Merging Procedure</a></li>
<li class="chapter" data-level="12.2" data-path="calling-peaks-w-macs2.html"><a href="calling-peaks-w-macs2.html"><i class="fa fa-check"></i><b>12.2</b> Calling Peaks w/ Macs2</a></li>
<li class="chapter" data-level="12.3" data-path="calling-peaks-w-tilematrix.html"><a href="calling-peaks-w-tilematrix.html"><i class="fa fa-check"></i><b>12.3</b> Calling Peaks w/ TileMatrix</a></li>
<li class="chapter" data-level="12.4" data-path="adding-a-peak-matrix.html"><a href="adding-a-peak-matrix.html"><i class="fa fa-check"></i><b>12.4</b> Adding a Peak Matrix</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="identifying-marker-peaks.html"><a href="identifying-marker-peaks.html"><i class="fa fa-check"></i><b>13</b> Identifying Marker Peaks</a>
<ul>
<li class="chapter" data-level="13.1" data-path="identifying-marker-peaks-with-archr.html"><a href="identifying-marker-peaks-with-archr.html"><i class="fa fa-check"></i><b>13.1</b> Identifying Marker Peaks with ArchR</a></li>
<li class="chapter" data-level="13.2" data-path="plotting-marker-peaks-in-archr.html"><a href="plotting-marker-peaks-in-archr.html"><i class="fa fa-check"></i><b>13.2</b> Plotting Marker Peaks in ArchR</a></li>
<li class="chapter" data-level="13.3" data-path="pairwise-testing-between-groups.html"><a href="pairwise-testing-between-groups.html"><i class="fa fa-check"></i><b>13.3</b> Pairwise Testing Between Groups</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="motif-and-feature-enrichment-with-archr.html"><a href="motif-and-feature-enrichment-with-archr.html"><i class="fa fa-check"></i><b>14</b> Motif and Feature Enrichment with ArchR</a>
<ul>
<li class="chapter" data-level="14.1" data-path="motif-enrichment.html"><a href="motif-enrichment.html"><i class="fa fa-check"></i><b>14.1</b> Motif Enrichment</a></li>
<li class="chapter" data-level="14.2" data-path="archr-enrichment.html"><a href="archr-enrichment.html"><i class="fa fa-check"></i><b>14.2</b> ArchR Enrichment</a></li>
<li class="chapter" data-level="14.3" data-path="custom-enrichment.html"><a href="custom-enrichment.html"><i class="fa fa-check"></i><b>14.3</b> Custom Enrichment</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="chromvar-deviatons-enrichment-with-archr.html"><a href="chromvar-deviatons-enrichment-with-archr.html"><i class="fa fa-check"></i><b>15</b> ChromVAR Deviatons Enrichment with ArchR</a>
<ul>
<li class="chapter" data-level="15.1" data-path="motif-deviations.html"><a href="motif-deviations.html"><i class="fa fa-check"></i><b>15.1</b> Motif Deviations</a></li>
<li class="chapter" data-level="15.2" data-path="archr-and-custom-deviations.html"><a href="archr-and-custom-deviations.html"><i class="fa fa-check"></i><b>15.2</b> ArchR and Custom Deviations</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="footprinting-with-archr.html"><a href="footprinting-with-archr.html"><i class="fa fa-check"></i><b>16</b> Footprinting with ArchR</a>
<ul>
<li class="chapter" data-level="16.1" data-path="motif-footprinting.html"><a href="motif-footprinting.html"><i class="fa fa-check"></i><b>16.1</b> Motif Footprinting</a></li>
<li class="chapter" data-level="16.2" data-path="normalization-of-footprints-for-tn5-bias.html"><a href="normalization-of-footprints-for-tn5-bias.html"><i class="fa fa-check"></i><b>16.2</b> Normalization of Footprints for Tn5 Bias</a></li>
<li class="chapter" data-level="16.3" data-path="feature-footprinting.html"><a href="feature-footprinting.html"><i class="fa fa-check"></i><b>16.3</b> Feature Footprinting</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="integrative-analysis-with-archr.html"><a href="integrative-analysis-with-archr.html"><i class="fa fa-check"></i><b>17</b> Integrative Analysis with ArchR</a>
<ul>
<li class="chapter" data-level="17.1" data-path="creating-low-overlapping-aggregates-of-cells.html"><a href="creating-low-overlapping-aggregates-of-cells.html"><i class="fa fa-check"></i><b>17.1</b> Creating Low-Overlapping Aggregates of Cells</a></li>
<li class="chapter" data-level="17.2" data-path="co-accessibility-with-archr.html"><a href="co-accessibility-with-archr.html"><i class="fa fa-check"></i><b>17.2</b> Co-accessibility with ArchR</a></li>
<li class="chapter" data-level="17.3" data-path="peak2genelinkage-with-archr.html"><a href="peak2genelinkage-with-archr.html"><i class="fa fa-check"></i><b>17.3</b> Peak2GeneLinkage with ArchR</a></li>
<li class="chapter" data-level="17.4" data-path="identification-of-positive-tf-regulators.html"><a href="identification-of-positive-tf-regulators.html"><i class="fa fa-check"></i><b>17.4</b> Identification of Positive TF-Regulators</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="trajectory-analysis-with-archr.html"><a href="trajectory-analysis-with-archr.html"><i class="fa fa-check"></i><b>18</b> Trajectory Analysis with ArchR</a>
<ul>
<li class="chapter" data-level="18.1" data-path="myeloid-trajectory---monocyte-differentiation.html"><a href="myeloid-trajectory---monocyte-differentiation.html"><i class="fa fa-check"></i><b>18.1</b> Myeloid Trajectory - Monocyte Differentiation</a></li>
<li class="chapter" data-level="18.2" data-path="lymphoid-trajectory---b-cell-differentiation.html"><a href="lymphoid-trajectory---b-cell-differentiation.html"><i class="fa fa-check"></i><b>18.2</b> Lymphoid Trajectory - B Cell Differentiation</a></li>
<li class="chapter" data-level="18.3" data-path="monocle3-trajectories.html"><a href="monocle3-trajectories.html"><i class="fa fa-check"></i><b>18.3</b> Monocle3 Trajectories</a></li>
<li class="chapter" data-level="18.4" data-path="slingshot-trajectories.html"><a href="slingshot-trajectories.html"><i class="fa fa-check"></i><b>18.4</b> Slingshot Trajectories</a></li>
<li class="chapter" data-level="18.5" data-path="comparing-the-different-trajectories.html"><a href="comparing-the-different-trajectories.html"><i class="fa fa-check"></i><b>18.5</b> Comparing the different trajectories</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="handling-archr-output.html"><a href="handling-archr-output.html"><i class="fa fa-check"></i><b>19</b> Handling ArchR Output</a>
<ul>
<li class="chapter" data-level="19.1" data-path="exporting-fragment-level-data.html"><a href="exporting-fragment-level-data.html"><i class="fa fa-check"></i><b>19.1</b> Exporting fragment-level data</a></li>
<li class="chapter" data-level="19.2" data-path="exporting-matrix-level-data.html"><a href="exporting-matrix-level-data.html"><i class="fa fa-check"></i><b>19.2</b> Exporting matrix-level data</a></li>
<li class="chapter" data-level="19.3" data-path="exporting-pseudo-bulked-data-to-a-summarizedexperiment.html"><a href="exporting-pseudo-bulked-data-to-a-summarizedexperiment.html"><i class="fa fa-check"></i><b>19.3</b> Exporting pseudo-bulked data to a <code>SummarizedExperiment</code></a></li>
<li class="chapter" data-level="19.4" data-path="exporting-pseudo-bulked-data-to-a-bigwig-file.html"><a href="exporting-pseudo-bulked-data-to-a-bigwig-file.html"><i class="fa fa-check"></i><b>19.4</b> Exporting pseudo-bulked data to a bigWig file</a></li>
</ul></li>
<li class="chapter" data-level="20" data-path="integration-with-bulk-atac-seq.html"><a href="integration-with-bulk-atac-seq.html"><i class="fa fa-check"></i><b>20</b> Integration with bulk ATAC-seq</a>
<ul>
<li class="chapter" data-level="20.1" data-path="projecting-bulk-atac-seq-data.html"><a href="projecting-bulk-atac-seq-data.html"><i class="fa fa-check"></i><b>20.1</b> Projecting bulk ATAC-seq data</a></li>
</ul></li>
<li class="chapter" data-level="21" data-path="multiomic-data-analysis-in-archr.html"><a href="multiomic-data-analysis-in-archr.html"><i class="fa fa-check"></i><b>21</b> Multiomic data analysis in ArchR</a>
<ul>
<li class="chapter" data-level="21.1" data-path="importing-data-and-setting-up-a-multiome-project.html"><a href="importing-data-and-setting-up-a-multiome-project.html"><i class="fa fa-check"></i><b>21.1</b> Importing data and setting up a Multiome project</a></li>
<li class="chapter" data-level="21.2" data-path="analysis-of-multiome-data-in-archr.html"><a href="analysis-of-multiome-data-in-archr.html"><i class="fa fa-check"></i><b>21.2</b> Analysis of multiome data in ArchR</a></li>
</ul></li>
<li class="chapter" data-level="22" data-path="plot-aesthetics-in-archr.html"><a href="plot-aesthetics-in-archr.html"><i class="fa fa-check"></i><b>22</b> Plot aesthetics in ArchR</a>
<ul>
<li class="chapter" data-level="22.1" data-path="using-and-manipulating-palettes-in-archr.html"><a href="using-and-manipulating-palettes-in-archr.html"><i class="fa fa-check"></i><b>22.1</b> Using and manipulating palettes in ArchR</a></li>
</ul></li>
<li class="chapter" data-level="23" data-path="testing-your-archr-installation.html"><a href="testing-your-archr-installation.html"><i class="fa fa-check"></i><b>23</b> Testing Your ArchR Installation</a></li>
<li class="chapter" data-level="24" data-path="session-information.html"><a href="session-information.html"><i class="fa fa-check"></i><b>24</b> Session Information</a></li>
<li class="divider"></li>
<li><a href="https://bookdown.org" target="_blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">ArchR: Robust and scaleable analysis of single-cell chromatin accessibility data.</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="creating-an-archrproject-1" class="section level2 hasAnchor" number="5.1">
<h2><span class="header-section-number">5.1</span> Creating An ArchRProject<a href="creating-an-archrproject-1.html#creating-an-archrproject-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>First, we must create our <code>ArchRProject</code> by providing a list of Arrow files and a few other parameters. The <code>outputDirectory</code> here describes where all downstream analyses and plots will be saved. ArchR will automatically associate the previously provided <code>geneAnnotation</code> and <code>genomeAnnotation</code> with the new <code>ArchRProject</code>. These were stored when we ran <code>addArchRGenome("hg19")</code> in a previous chapter. Importantly, multiple <code>ArchRProject</code> objects <strong>cannot</strong> be combined later so any samples that you want to analyze must be included as Arrow files at this project creation step. You’ll note that we set the parameter <code>copyArrows = TRUE</code> which is recommended because we will modify the Arrow files in downstream operations and this preserves an original copy of the Arrow files for future usage as necessary.</p>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="creating-an-archrproject-1.html#cb42-1" tabindex="-1"></a>projHeme1 <span class="ot">&lt;-</span> <span class="fu">ArchRProject</span>(</span>
<span id="cb42-2"><a href="creating-an-archrproject-1.html#cb42-2" tabindex="-1"></a>  <span class="at">ArrowFiles =</span> ArrowFiles, </span>
<span id="cb42-3"><a href="creating-an-archrproject-1.html#cb42-3" tabindex="-1"></a>  <span class="at">outputDirectory =</span> <span class="st">&quot;HemeTutorial&quot;</span>,</span>
<span id="cb42-4"><a href="creating-an-archrproject-1.html#cb42-4" tabindex="-1"></a>  <span class="at">copyArrows =</span> <span class="cn">TRUE</span></span>
<span id="cb42-5"><a href="creating-an-archrproject-1.html#cb42-5" tabindex="-1"></a>)</span>
<span id="cb42-6"><a href="creating-an-archrproject-1.html#cb42-6" tabindex="-1"></a><span class="do">## Using GeneAnnotation set by addArchRGenome(Hg19)!</span></span>
<span id="cb42-7"><a href="creating-an-archrproject-1.html#cb42-7" tabindex="-1"></a><span class="do">## Using GeneAnnotation set by addArchRGenome(Hg19)!</span></span>
<span id="cb42-8"><a href="creating-an-archrproject-1.html#cb42-8" tabindex="-1"></a><span class="do">## Validating Arrows...</span></span>
<span id="cb42-9"><a href="creating-an-archrproject-1.html#cb42-9" tabindex="-1"></a><span class="do">## Getting SampleNames...</span></span>
<span id="cb42-10"><a href="creating-an-archrproject-1.html#cb42-10" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb42-11"><a href="creating-an-archrproject-1.html#cb42-11" tabindex="-1"></a><span class="do">## Copying ArrowFiles to Ouptut Directory! If you want to save disk space set copyArrows = FALSE</span></span>
<span id="cb42-12"><a href="creating-an-archrproject-1.html#cb42-12" tabindex="-1"></a><span class="do">## 1 2 3 </span></span>
<span id="cb42-13"><a href="creating-an-archrproject-1.html#cb42-13" tabindex="-1"></a><span class="do">## Getting Cell Metadata...</span></span>
<span id="cb42-14"><a href="creating-an-archrproject-1.html#cb42-14" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb42-15"><a href="creating-an-archrproject-1.html#cb42-15" tabindex="-1"></a><span class="do">## Merging Cell Metadata...</span></span>
<span id="cb42-16"><a href="creating-an-archrproject-1.html#cb42-16" tabindex="-1"></a><span class="do">## Initializing ArchRProject...</span></span>
<span id="cb42-17"><a href="creating-an-archrproject-1.html#cb42-17" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb42-18"><a href="creating-an-archrproject-1.html#cb42-18" tabindex="-1"></a><span class="do">##                                                    / |</span></span>
<span id="cb42-19"><a href="creating-an-archrproject-1.html#cb42-19" tabindex="-1"></a><span class="do">##                                                  /    \</span></span>
<span id="cb42-20"><a href="creating-an-archrproject-1.html#cb42-20" tabindex="-1"></a><span class="do">##             .                                  /      |.</span></span>
<span id="cb42-21"><a href="creating-an-archrproject-1.html#cb42-21" tabindex="-1"></a><span class="do">##             \\\                              /        |.</span></span>
<span id="cb42-22"><a href="creating-an-archrproject-1.html#cb42-22" tabindex="-1"></a><span class="do">##               \\\                          /           `|.</span></span>
<span id="cb42-23"><a href="creating-an-archrproject-1.html#cb42-23" tabindex="-1"></a><span class="do">##                 \\\                      /              |.</span></span>
<span id="cb42-24"><a href="creating-an-archrproject-1.html#cb42-24" tabindex="-1"></a><span class="do">##                   \                    /                |\</span></span>
<span id="cb42-25"><a href="creating-an-archrproject-1.html#cb42-25" tabindex="-1"></a><span class="do">##                   \\#####\           /                  ||</span></span>
<span id="cb42-26"><a href="creating-an-archrproject-1.html#cb42-26" tabindex="-1"></a><span class="do">##                 ==###########&gt;      /                   ||</span></span>
<span id="cb42-27"><a href="creating-an-archrproject-1.html#cb42-27" tabindex="-1"></a><span class="do">##                  \\##==......\    /                     ||</span></span>
<span id="cb42-28"><a href="creating-an-archrproject-1.html#cb42-28" tabindex="-1"></a><span class="do">##             ______ =       =|__ /__                     ||      \\\</span></span>
<span id="cb42-29"><a href="creating-an-archrproject-1.html#cb42-29" tabindex="-1"></a><span class="do">##         ,--&#39; ,----`-,__ ___/&#39;  --,-`-===================##========&gt;</span></span>
<span id="cb42-30"><a href="creating-an-archrproject-1.html#cb42-30" tabindex="-1"></a><span class="do">##        \               &#39;        ##_______ _____ ,--,__,=##,__   ///</span></span>
<span id="cb42-31"><a href="creating-an-archrproject-1.html#cb42-31" tabindex="-1"></a><span class="do">##         ,    __==    ___,-,__,--&#39;#&#39;  ===&#39;      `-&#39;    | ##,-/</span></span>
<span id="cb42-32"><a href="creating-an-archrproject-1.html#cb42-32" tabindex="-1"></a><span class="do">##         -,____,---&#39;       \\####\\________________,--\\_##,/</span></span>
<span id="cb42-33"><a href="creating-an-archrproject-1.html#cb42-33" tabindex="-1"></a><span class="do">##            ___      .______        ______  __    __  .______      </span></span>
<span id="cb42-34"><a href="creating-an-archrproject-1.html#cb42-34" tabindex="-1"></a><span class="do">##           /   \     |   _  \      /      ||  |  |  | |   _  \     </span></span>
<span id="cb42-35"><a href="creating-an-archrproject-1.html#cb42-35" tabindex="-1"></a><span class="do">##          /  ^  \    |  |_)  |    |  ,----&#39;|  |__|  | |  |_)  |    </span></span>
<span id="cb42-36"><a href="creating-an-archrproject-1.html#cb42-36" tabindex="-1"></a><span class="do">##         /  /_\  \   |      /     |  |     |   __   | |      /     </span></span>
<span id="cb42-37"><a href="creating-an-archrproject-1.html#cb42-37" tabindex="-1"></a><span class="do">##        /  _____  \  |  |\  \\___ |  `----.|  |  |  | |  |\  \\___.</span></span>
<span id="cb42-38"><a href="creating-an-archrproject-1.html#cb42-38" tabindex="-1"></a><span class="do">##       /__/     \__\ | _| `._____| \______||__|  |__| | _| `._____|</span></span>
<span id="cb42-39"><a href="creating-an-archrproject-1.html#cb42-39" tabindex="-1"></a><span class="do">## </span></span></code></pre></div>
<p>We call this <code>ArchRProject</code> “projHeme1” because it is the first iteration of our hematopoiesis project. Throughout this walkthrough we will modify and update this <code>ArchRProject</code> and keep track of which version of the project we are using by iterating the project number (i.e. “projHeme2”).</p>
<p>We can examine the contents of our <code>ArchRProject</code>:</p>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb43-1"><a href="creating-an-archrproject-1.html#cb43-1" tabindex="-1"></a>projHeme1</span>
<span id="cb43-2"><a href="creating-an-archrproject-1.html#cb43-2" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb43-3"><a href="creating-an-archrproject-1.html#cb43-3" tabindex="-1"></a><span class="do">##            ___      .______        ______  __    __  .______      </span></span>
<span id="cb43-4"><a href="creating-an-archrproject-1.html#cb43-4" tabindex="-1"></a><span class="do">##           /   \     |   _  \      /      ||  |  |  | |   _  \     </span></span>
<span id="cb43-5"><a href="creating-an-archrproject-1.html#cb43-5" tabindex="-1"></a><span class="do">##          /  ^  \    |  |_)  |    |  ,----&#39;|  |__|  | |  |_)  |    </span></span>
<span id="cb43-6"><a href="creating-an-archrproject-1.html#cb43-6" tabindex="-1"></a><span class="do">##         /  /_\  \   |      /     |  |     |   __   | |      /     </span></span>
<span id="cb43-7"><a href="creating-an-archrproject-1.html#cb43-7" tabindex="-1"></a><span class="do">##        /  _____  \  |  |\  \\___ |  `----.|  |  |  | |  |\  \\___.</span></span>
<span id="cb43-8"><a href="creating-an-archrproject-1.html#cb43-8" tabindex="-1"></a><span class="do">##       /__/     \__\ | _| `._____| \______||__|  |__| | _| `._____|</span></span>
<span id="cb43-9"><a href="creating-an-archrproject-1.html#cb43-9" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb43-10"><a href="creating-an-archrproject-1.html#cb43-10" tabindex="-1"></a><span class="do">## class: ArchRProject </span></span>
<span id="cb43-11"><a href="creating-an-archrproject-1.html#cb43-11" tabindex="-1"></a><span class="do">## outputDirectory: /workspace/ArchR/ArchR_Website_Testing/bookdown/HemeTutorial </span></span>
<span id="cb43-12"><a href="creating-an-archrproject-1.html#cb43-12" tabindex="-1"></a><span class="do">## samples(3): scATAC_BMMC_R1 scATAC_CD34_BMMC_R1 scATAC_PBMC_R1</span></span>
<span id="cb43-13"><a href="creating-an-archrproject-1.html#cb43-13" tabindex="-1"></a><span class="do">## sampleColData names(1): ArrowFiles</span></span>
<span id="cb43-14"><a href="creating-an-archrproject-1.html#cb43-14" tabindex="-1"></a><span class="do">## cellColData names(15): Sample TSSEnrichment ... DoubletEnrichment</span></span>
<span id="cb43-15"><a href="creating-an-archrproject-1.html#cb43-15" tabindex="-1"></a><span class="do">##   BlacklistRatio</span></span>
<span id="cb43-16"><a href="creating-an-archrproject-1.html#cb43-16" tabindex="-1"></a><span class="do">## numberOfCells(1): 10660</span></span>
<span id="cb43-17"><a href="creating-an-archrproject-1.html#cb43-17" tabindex="-1"></a><span class="do">## medianTSS(1): 16.815</span></span>
<span id="cb43-18"><a href="creating-an-archrproject-1.html#cb43-18" tabindex="-1"></a><span class="do">## medianFrags(1): 3049.5</span></span></code></pre></div>
<p>We can see from the above that our <code>ArchRProject</code> has been initialized with a few important attributes:</p>
<ol style="list-style-type: decimal">
<li>The specified <code>outputDirectory</code>.</li>
<li>The <code>sampleNames</code> of each sample which were obtained from the Arrow files.</li>
<li>A matrix called <code>sampleColData</code> which contains data associated with each sample.</li>
<li>A matrix called <code>cellColData</code> which contains data associated with each cell. Because we already computed doublet enrichment scores using <code>addDoubletScores()</code>, which added those values to each cell in the Arrow files, we can see columns corresponding to the “DoubletEnrichment” and “DoubletScore” in the <code>cellColData</code> matrix.</li>
<li>The total number of cells in our project which represents all samples after doublet identification and removal.</li>
<li>The median TSS enrichment score and the median number of fragments across all cells and all samples.</li>
</ol>
<p>We can check how much memory is used to store the <code>ArchRProject</code> in memory within R:</p>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb44-1"><a href="creating-an-archrproject-1.html#cb44-1" tabindex="-1"></a><span class="fu">paste0</span>(<span class="st">&quot;Memory Size = &quot;</span>, <span class="fu">round</span>(<span class="fu">object.size</span>(projHeme1) <span class="sc">/</span> <span class="dv">10</span><span class="sc">^</span><span class="dv">6</span>, <span class="dv">3</span>), <span class="st">&quot; MB&quot;</span>)</span>
<span id="cb44-2"><a href="creating-an-archrproject-1.html#cb44-2" tabindex="-1"></a><span class="do">## [1] &quot;Memory Size = 37.477 MB&quot;</span></span></code></pre></div>
<p>We can also ask which data matrices are available within the <code>ArchRProject</code> which will be useful downstream once we start adding to this project:</p>
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb45-1"><a href="creating-an-archrproject-1.html#cb45-1" tabindex="-1"></a><span class="fu">getAvailableMatrices</span>(projHeme1)</span>
<span id="cb45-2"><a href="creating-an-archrproject-1.html#cb45-2" tabindex="-1"></a><span class="do">## [1] &quot;GeneScoreMatrix&quot; &quot;TileMatrix&quot;</span></span></code></pre></div>
<p>Taking a closer look at <code>cellColData</code>, we can see all of the metadata that is stored here. As the name suggests, this is metadata that applies to each individual cell.</p>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb46-1"><a href="creating-an-archrproject-1.html#cb46-1" tabindex="-1"></a><span class="fu">head</span>(projHeme1<span class="sc">@</span>cellColData)</span>
<span id="cb46-2"><a href="creating-an-archrproject-1.html#cb46-2" tabindex="-1"></a><span class="do">## DataFrame with 6 rows and 15 columns</span></span>
<span id="cb46-3"><a href="creating-an-archrproject-1.html#cb46-3" tabindex="-1"></a><span class="do">##                                           Sample TSSEnrichment ReadsInTSS</span></span>
<span id="cb46-4"><a href="creating-an-archrproject-1.html#cb46-4" tabindex="-1"></a><span class="do">##                                            &lt;Rle&gt;       &lt;array&gt;    &lt;array&gt;</span></span>
<span id="cb46-5"><a href="creating-an-archrproject-1.html#cb46-5" tabindex="-1"></a><span class="do">## scATAC_BMMC_R1#TTATGTCAGTGATTAG-1 scATAC_BMMC_R1         7.204       1146</span></span>
<span id="cb46-6"><a href="creating-an-archrproject-1.html#cb46-6" tabindex="-1"></a><span class="do">## scATAC_BMMC_R1#AAGATAGTCACCGCGA-1 scATAC_BMMC_R1         7.949        831</span></span>
<span id="cb46-7"><a href="creating-an-archrproject-1.html#cb46-7" tabindex="-1"></a><span class="do">## scATAC_BMMC_R1#GCATTGAAGATTCCGT-1 scATAC_BMMC_R1         4.447        384</span></span>
<span id="cb46-8"><a href="creating-an-archrproject-1.html#cb46-8" tabindex="-1"></a><span class="do">## scATAC_BMMC_R1#TATGTTCAGGGTTCCC-1 scATAC_BMMC_R1         6.941        659</span></span>
<span id="cb46-9"><a href="creating-an-archrproject-1.html#cb46-9" tabindex="-1"></a><span class="do">## scATAC_BMMC_R1#TCCATCGGTCCCGTGA-1 scATAC_BMMC_R1         4.771        412</span></span>
<span id="cb46-10"><a href="creating-an-archrproject-1.html#cb46-10" tabindex="-1"></a><span class="do">## scATAC_BMMC_R1#AGTTACGAGAACGTCG-1 scATAC_BMMC_R1         9.185       1104</span></span>
<span id="cb46-11"><a href="creating-an-archrproject-1.html#cb46-11" tabindex="-1"></a><span class="do">##                                   ReadsInPromoter ReadsInBlacklist</span></span>
<span id="cb46-12"><a href="creating-an-archrproject-1.html#cb46-12" tabindex="-1"></a><span class="do">##                                           &lt;array&gt;          &lt;array&gt;</span></span>
<span id="cb46-13"><a href="creating-an-archrproject-1.html#cb46-13" tabindex="-1"></a><span class="do">## scATAC_BMMC_R1#TTATGTCAGTGATTAG-1            4306              611</span></span>
<span id="cb46-14"><a href="creating-an-archrproject-1.html#cb46-14" tabindex="-1"></a><span class="do">## scATAC_BMMC_R1#AAGATAGTCACCGCGA-1            3542              502</span></span>
<span id="cb46-15"><a href="creating-an-archrproject-1.html#cb46-15" tabindex="-1"></a><span class="do">## scATAC_BMMC_R1#GCATTGAAGATTCCGT-1            1686              311</span></span>
<span id="cb46-16"><a href="creating-an-archrproject-1.html#cb46-16" tabindex="-1"></a><span class="do">## scATAC_BMMC_R1#TATGTTCAGGGTTCCC-1            2811              475</span></span>
<span id="cb46-17"><a href="creating-an-archrproject-1.html#cb46-17" tabindex="-1"></a><span class="do">## scATAC_BMMC_R1#TCCATCGGTCCCGTGA-1            2108              331</span></span>
<span id="cb46-18"><a href="creating-an-archrproject-1.html#cb46-18" tabindex="-1"></a><span class="do">## scATAC_BMMC_R1#AGTTACGAGAACGTCG-1            4457              341</span></span>
<span id="cb46-19"><a href="creating-an-archrproject-1.html#cb46-19" tabindex="-1"></a><span class="do">##                                        PromoterRatio  PassQC  NucleosomeRatio</span></span>
<span id="cb46-20"><a href="creating-an-archrproject-1.html#cb46-20" tabindex="-1"></a><span class="do">##                                              &lt;array&gt; &lt;array&gt;          &lt;array&gt;</span></span>
<span id="cb46-21"><a href="creating-an-archrproject-1.html#cb46-21" tabindex="-1"></a><span class="do">## scATAC_BMMC_R1#TTATGTCAGTGATTAG-1 0.0822100882049716       1 3.22675919948354</span></span>
<span id="cb46-22"><a href="creating-an-archrproject-1.html#cb46-22" tabindex="-1"></a><span class="do">## scATAC_BMMC_R1#AAGATAGTCACCGCGA-1 0.0857710189848896       1 1.23149248892251</span></span>
<span id="cb46-23"><a href="creating-an-archrproject-1.html#cb46-23" tabindex="-1"></a><span class="do">## scATAC_BMMC_R1#GCATTGAAGATTCCGT-1  0.044391785150079       1 3.69933184855234</span></span>
<span id="cb46-24"><a href="creating-an-archrproject-1.html#cb46-24" tabindex="-1"></a><span class="do">## scATAC_BMMC_R1#TATGTTCAGGGTTCCC-1 0.0768200699606471       1 4.27870744373918</span></span>
<span id="cb46-25"><a href="creating-an-archrproject-1.html#cb46-25" tabindex="-1"></a><span class="do">## scATAC_BMMC_R1#TCCATCGGTCCCGTGA-1 0.0603734677511742       1 4.03838383838384</span></span>
<span id="cb46-26"><a href="creating-an-archrproject-1.html#cb46-26" tabindex="-1"></a><span class="do">## scATAC_BMMC_R1#AGTTACGAGAACGTCG-1  0.130253083172599       1 2.58453802639849</span></span>
<span id="cb46-27"><a href="creating-an-archrproject-1.html#cb46-27" tabindex="-1"></a><span class="do">##                                   nMultiFrags nMonoFrags  nFrags nDiFrags</span></span>
<span id="cb46-28"><a href="creating-an-archrproject-1.html#cb46-28" tabindex="-1"></a><span class="do">##                                       &lt;array&gt;    &lt;array&gt; &lt;array&gt;  &lt;array&gt;</span></span>
<span id="cb46-29"><a href="creating-an-archrproject-1.html#cb46-29" tabindex="-1"></a><span class="do">## scATAC_BMMC_R1#TTATGTCAGTGATTAG-1        3801       6196   26189    16192</span></span>
<span id="cb46-30"><a href="creating-an-archrproject-1.html#cb46-30" tabindex="-1"></a><span class="do">## scATAC_BMMC_R1#AAGATAGTCACCGCGA-1        3448       9253   20648     7947</span></span>
<span id="cb46-31"><a href="creating-an-archrproject-1.html#cb46-31" tabindex="-1"></a><span class="do">## scATAC_BMMC_R1#GCATTGAAGATTCCGT-1        3119       4041   18990    11830</span></span>
<span id="cb46-32"><a href="creating-an-archrproject-1.html#cb46-32" tabindex="-1"></a><span class="do">## scATAC_BMMC_R1#TATGTTCAGGGTTCCC-1        3853       3466   18296    10977</span></span>
<span id="cb46-33"><a href="creating-an-archrproject-1.html#cb46-33" tabindex="-1"></a><span class="do">## scATAC_BMMC_R1#TCCATCGGTCCCGTGA-1        3564       3465   17458    10429</span></span>
<span id="cb46-34"><a href="creating-an-archrproject-1.html#cb46-34" tabindex="-1"></a><span class="do">## scATAC_BMMC_R1#AGTTACGAGAACGTCG-1        3605       4773   17109     8731</span></span>
<span id="cb46-35"><a href="creating-an-archrproject-1.html#cb46-35" tabindex="-1"></a><span class="do">##                                       DoubletScore DoubletEnrichment</span></span>
<span id="cb46-36"><a href="creating-an-archrproject-1.html#cb46-36" tabindex="-1"></a><span class="do">##                                            &lt;array&gt;           &lt;array&gt;</span></span>
<span id="cb46-37"><a href="creating-an-archrproject-1.html#cb46-37" tabindex="-1"></a><span class="do">## scATAC_BMMC_R1#TTATGTCAGTGATTAG-1                0              1.55</span></span>
<span id="cb46-38"><a href="creating-an-archrproject-1.html#cb46-38" tabindex="-1"></a><span class="do">## scATAC_BMMC_R1#AAGATAGTCACCGCGA-1 91.1760813271055              5.75</span></span>
<span id="cb46-39"><a href="creating-an-archrproject-1.html#cb46-39" tabindex="-1"></a><span class="do">## scATAC_BMMC_R1#GCATTGAAGATTCCGT-1                0               1.3</span></span>
<span id="cb46-40"><a href="creating-an-archrproject-1.html#cb46-40" tabindex="-1"></a><span class="do">## scATAC_BMMC_R1#TATGTTCAGGGTTCCC-1  70.630505425003              5.05</span></span>
<span id="cb46-41"><a href="creating-an-archrproject-1.html#cb46-41" tabindex="-1"></a><span class="do">## scATAC_BMMC_R1#TCCATCGGTCCCGTGA-1 163.664669675752               7.9</span></span>
<span id="cb46-42"><a href="creating-an-archrproject-1.html#cb46-42" tabindex="-1"></a><span class="do">## scATAC_BMMC_R1#AGTTACGAGAACGTCG-1 120.537764456854              6.65</span></span>
<span id="cb46-43"><a href="creating-an-archrproject-1.html#cb46-43" tabindex="-1"></a><span class="do">##                                        BlacklistRatio</span></span>
<span id="cb46-44"><a href="creating-an-archrproject-1.html#cb46-44" tabindex="-1"></a><span class="do">##                                               &lt;array&gt;</span></span>
<span id="cb46-45"><a href="creating-an-archrproject-1.html#cb46-45" tabindex="-1"></a><span class="do">## scATAC_BMMC_R1#TTATGTCAGTGATTAG-1  0.0116652029478025</span></span>
<span id="cb46-46"><a href="creating-an-archrproject-1.html#cb46-46" tabindex="-1"></a><span class="do">## scATAC_BMMC_R1#AAGATAGTCACCGCGA-1  0.0121561410306083</span></span>
<span id="cb46-47"><a href="creating-an-archrproject-1.html#cb46-47" tabindex="-1"></a><span class="do">## scATAC_BMMC_R1#GCATTGAAGATTCCGT-1 0.00818852027382833</span></span>
<span id="cb46-48"><a href="creating-an-archrproject-1.html#cb46-48" tabindex="-1"></a><span class="do">## scATAC_BMMC_R1#TATGTTCAGGGTTCCC-1  0.0129809794490599</span></span>
<span id="cb46-49"><a href="creating-an-archrproject-1.html#cb46-49" tabindex="-1"></a><span class="do">## scATAC_BMMC_R1#TCCATCGGTCCCGTGA-1 0.00947989460419292</span></span>
<span id="cb46-50"><a href="creating-an-archrproject-1.html#cb46-50" tabindex="-1"></a><span class="do">## scATAC_BMMC_R1#AGTTACGAGAACGTCG-1 0.00996551522590449</span></span></code></pre></div>
<p>You’ll notice that many of these columns correspond to per-cell quality control information that was calculated during Arrow file creation. These columns are:</p>
<ol style="list-style-type: decimal">
<li>TSSEnrichment - the per-cell TSS enrichment score</li>
<li>ReadsInTSS - the number of reads that fall within TSS regions (default is 100 bp around TSS)</li>
<li>ReadsInPromoter - the number of reads that fall in promoter regions (default is -2000 to +100 from the TSS)</li>
<li>PromoterRatio - the ratio of reads in promoters to reads outside of promoters</li>
<li>ReadsInBlacklist - the number of reads that fall in blacklist regions</li>
<li>BlacklistRatio - the ratio of reads in blacklist regions to reads outside of blacklist regions</li>
<li>NucleosomeRatio - similar to above but for reads mapping to nucleosome-sized reads, defined as: <code>(nDiFrags + nMultiFrags) / nMonoFrags)</code></li>
<li>nFrags - the number of fragments recovered per cell</li>
<li>nMonoFrags - the number of fragments that have length less than <code>2 * nucLength</code> (nucLength is the length of DNA that wraps around a nucleosome (147 bp by default))</li>
<li>nDiFrags - the number of fragments that have length less than <code>3 * nucLength</code> but greater than <code>2 * nucLength</code></li>
<li>nMultiFrags - the number of fragments that have length &gt;= <code>3*nucLength</code></li>
<li>PassQC - equal to <code>1</code> if the cell passed QC filters or <code>0</code> if not</li>
</ol>
<p>Throughout the tutorial, other data will get added to <code>cellColData</code>, for example the number of ReadsInPeaks or FRiP when we add a peak set to the project.</p>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="creating-an-archrproject.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="manipulating-an-archrproject.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": null
},
"fontsettings": null,
"edit": null,
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "none"
},
"info": false
});
});
</script>

</body>

</html>
