<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>10.1 Cross-platform linkage of scATAC-seq cells with scRNA-seq cells | ArchR: Robust and scaleable analysis of single-cell chromatin accessibility data.</title>
  <meta name="description" content="A guide to ArchR" />
  <meta name="generator" content="bookdown 0.41 and GitBook 2.6.7" />

  <meta property="og:title" content="10.1 Cross-platform linkage of scATAC-seq cells with scRNA-seq cells | ArchR: Robust and scaleable analysis of single-cell chromatin accessibility data." />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="A guide to ArchR" />
  <meta name="github-repo" content="GreenleafLab/ArchR" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="10.1 Cross-platform linkage of scATAC-seq cells with scRNA-seq cells | ArchR: Robust and scaleable analysis of single-cell chromatin accessibility data." />
  
  <meta name="twitter:description" content="A guide to ArchR" />
  

<meta name="author" content="Jeffrey Granja and Ryan Corces" />


<meta name="date" content="2025-01-23" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="images/favicon.ico" type="image/x-icon" />
<link rel="prev" href="defining-cluster-identity-with-scrna-seq.html"/>
<link rel="next" href="adding-pseudo-scrna-seq-profiles-for-each-scatac-seq-cell.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<link href="libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="../index.html">Back to ArchR Home</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><img src="images/ArchR_Logo_Integrated.png" alt="" width="200"></a></li>
<li class="chapter" data-level="1" data-path="debugging-and-troubleshooting-in-archr.html"><a href="debugging-and-troubleshooting-in-archr.html"><i class="fa fa-check"></i><b>1</b> Debugging and Troubleshooting in ArchR</a></li>
<li class="chapter" data-level="2" data-path="manage-archrs-dependencies.html"><a href="manage-archrs-dependencies.html"><i class="fa fa-check"></i><b>2</b> Manage ArchR’s Dependencies</a>
<ul>
<li class="chapter" data-level="2.1" data-path="using-renv-to-manage-dependencies.html"><a href="using-renv-to-manage-dependencies.html"><i class="fa fa-check"></i><b>2.1</b> Using renv to manage dependencies</a></li>
<li class="chapter" data-level="2.2" data-path="other-environment-managers.html"><a href="other-environment-managers.html"><i class="fa fa-check"></i><b>2.2</b> Other environment managers</a></li>
<li class="chapter" data-level="2.3" data-path="using-docker-for-pre-compiled-environment-setup.html"><a href="using-docker-for-pre-compiled-environment-setup.html"><i class="fa fa-check"></i><b>2.3</b> Using Docker for pre-compiled environment setup</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="getting-started-with-archr.html"><a href="getting-started-with-archr.html"><i class="fa fa-check"></i><b>3</b> Getting Started with ArchR</a>
<ul>
<li class="chapter" data-level="3.1" data-path="a-brief-primer-on-atac-seq-terminology.html"><a href="a-brief-primer-on-atac-seq-terminology.html"><i class="fa fa-check"></i><b>3.1</b> A Brief Primer on ATAC-seq Terminology</a></li>
<li class="chapter" data-level="3.2" data-path="why-use-archr.html"><a href="why-use-archr.html"><i class="fa fa-check"></i><b>3.2</b> Why use ArchR?</a></li>
<li class="chapter" data-level="3.3" data-path="what-is-an-arrow-file-archrproject.html"><a href="what-is-an-arrow-file-archrproject.html"><i class="fa fa-check"></i><b>3.3</b> What is an Arrow file / <code>ArchRProject</code>?</a></li>
<li class="chapter" data-level="3.4" data-path="input-file-types-in-archr.html"><a href="input-file-types-in-archr.html"><i class="fa fa-check"></i><b>3.4</b> Input File Types in ArchR</a></li>
<li class="chapter" data-level="3.5" data-path="getting-set-up.html"><a href="getting-set-up.html"><i class="fa fa-check"></i><b>3.5</b> Getting Set Up</a></li>
<li class="chapter" data-level="3.6" data-path="creating-arrow-files.html"><a href="creating-arrow-files.html"><i class="fa fa-check"></i><b>3.6</b> Creating Arrow Files</a></li>
<li class="chapter" data-level="3.7" data-path="per-cell-quality-control.html"><a href="per-cell-quality-control.html"><i class="fa fa-check"></i><b>3.7</b> Per-cell Quality Control</a></li>
<li class="chapter" data-level="3.8" data-path="using-bam-files-for-arrow-file-creation.html"><a href="using-bam-files-for-arrow-file-creation.html"><i class="fa fa-check"></i><b>3.8</b> Using BAM files for Arrow File creation</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="doublet-inference-with-archr.html"><a href="doublet-inference-with-archr.html"><i class="fa fa-check"></i><b>4</b> Doublet Inference with ArchR</a>
<ul>
<li class="chapter" data-level="4.1" data-path="how-does-doublet-identification-work-in-archr.html"><a href="how-does-doublet-identification-work-in-archr.html"><i class="fa fa-check"></i><b>4.1</b> How does doublet identification work in ArchR?</a></li>
<li class="chapter" data-level="4.2" data-path="inferring-scatac-seq-doublets-with-archr.html"><a href="inferring-scatac-seq-doublets-with-archr.html"><i class="fa fa-check"></i><b>4.2</b> Inferring scATAC-seq Doublets with ArchR</a></li>
<li class="chapter" data-level="4.3" data-path="using-demuxlet-with-archr.html"><a href="using-demuxlet-with-archr.html"><i class="fa fa-check"></i><b>4.3</b> Using <em>demuxlet</em> with ArchR</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="creating-an-archrproject.html"><a href="creating-an-archrproject.html"><i class="fa fa-check"></i><b>5</b> Creating an ArchRProject</a>
<ul>
<li class="chapter" data-level="5.1" data-path="creating-an-archrproject-1.html"><a href="creating-an-archrproject-1.html"><i class="fa fa-check"></i><b>5.1</b> Creating An ArchRProject</a></li>
<li class="chapter" data-level="5.2" data-path="manipulating-an-archrproject.html"><a href="manipulating-an-archrproject.html"><i class="fa fa-check"></i><b>5.2</b> Manipulating An ArchRProject</a></li>
<li class="chapter" data-level="5.3" data-path="plotting-sample-statistics-from-an-archrproject.html"><a href="plotting-sample-statistics-from-an-archrproject.html"><i class="fa fa-check"></i><b>5.3</b> Plotting Sample Statistics from an ArchRProject</a></li>
<li class="chapter" data-level="5.4" data-path="plotting-sample-fragment-size-distribution-and-tss-enrichment-profiles..html"><a href="plotting-sample-fragment-size-distribution-and-tss-enrichment-profiles..html"><i class="fa fa-check"></i><b>5.4</b> Plotting Sample Fragment Size Distribution and TSS Enrichment Profiles.</a></li>
<li class="chapter" data-level="5.5" data-path="saving-and-loading-an-archrproject.html"><a href="saving-and-loading-an-archrproject.html"><i class="fa fa-check"></i><b>5.5</b> Saving and Loading an <code>ArchRProject</code></a></li>
<li class="chapter" data-level="5.6" data-path="filtering-doublets-from-an-archrproject.html"><a href="filtering-doublets-from-an-archrproject.html"><i class="fa fa-check"></i><b>5.6</b> Filtering Doublets from an ArchRProject</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="dimensionality-reduction-with-archr.html"><a href="dimensionality-reduction-with-archr.html"><i class="fa fa-check"></i><b>6</b> Dimensionality Reduction with ArchR</a>
<ul>
<li class="chapter" data-level="6.1" data-path="archrs-lsi-implementation.html"><a href="archrs-lsi-implementation.html"><i class="fa fa-check"></i><b>6.1</b> ArchR’s LSI Implementation</a></li>
<li class="chapter" data-level="6.2" data-path="iterative-latent-semantic-indexing-lsi.html"><a href="iterative-latent-semantic-indexing-lsi.html"><i class="fa fa-check"></i><b>6.2</b> Iterative Latent Semantic Indexing (LSI)</a></li>
<li class="chapter" data-level="6.3" data-path="estimated-lsi.html"><a href="estimated-lsi.html"><i class="fa fa-check"></i><b>6.3</b> Estimated LSI</a></li>
<li class="chapter" data-level="6.4" data-path="batch-effect-correction-wtih-harmony.html"><a href="batch-effect-correction-wtih-harmony.html"><i class="fa fa-check"></i><b>6.4</b> Batch Effect Correction wtih Harmony</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="clustering-with-archr.html"><a href="clustering-with-archr.html"><i class="fa fa-check"></i><b>7</b> Clustering with ArchR</a>
<ul>
<li class="chapter" data-level="7.1" data-path="clustering-using-seurats-findclusters-function.html"><a href="clustering-using-seurats-findclusters-function.html"><i class="fa fa-check"></i><b>7.1</b> Clustering using Seurat’s <code>FindClusters()</code> function</a></li>
<li class="chapter" data-level="7.2" data-path="clustering-using-scran.html"><a href="clustering-using-scran.html"><i class="fa fa-check"></i><b>7.2</b> Clustering using <code>scran</code></a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="single-cell-embeddings.html"><a href="single-cell-embeddings.html"><i class="fa fa-check"></i><b>8</b> Single-cell Embeddings</a>
<ul>
<li class="chapter" data-level="8.1" data-path="uniform-manifold-approximation-and-projection-umap.html"><a href="uniform-manifold-approximation-and-projection-umap.html"><i class="fa fa-check"></i><b>8.1</b> Uniform Manifold Approximation and Projection (UMAP)</a></li>
<li class="chapter" data-level="8.2" data-path="t-stocastic-neighbor-embedding-t-sne.html"><a href="t-stocastic-neighbor-embedding-t-sne.html"><i class="fa fa-check"></i><b>8.2</b> t-Stocastic Neighbor Embedding (t-SNE)</a></li>
<li class="chapter" data-level="8.3" data-path="dimensionality-reduction-after-harmony.html"><a href="dimensionality-reduction-after-harmony.html"><i class="fa fa-check"></i><b>8.3</b> Dimensionality Reduction After Harmony</a></li>
<li class="chapter" data-level="8.4" data-path="highlighting-specific-cells-on-an-embedding.html"><a href="highlighting-specific-cells-on-an-embedding.html"><i class="fa fa-check"></i><b>8.4</b> Highlighting specific cells on an embedding</a></li>
<li class="chapter" data-level="8.5" data-path="importing-an-embedding-from-external-software.html"><a href="importing-an-embedding-from-external-software.html"><i class="fa fa-check"></i><b>8.5</b> Importing an embedding from external software</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="gene-scores-and-marker-genes-with-archr.html"><a href="gene-scores-and-marker-genes-with-archr.html"><i class="fa fa-check"></i><b>9</b> Gene Scores and Marker Genes with ArchR</a>
<ul>
<li class="chapter" data-level="9.1" data-path="calculating-gene-scores-in-archr.html"><a href="calculating-gene-scores-in-archr.html"><i class="fa fa-check"></i><b>9.1</b> Calculating Gene Scores in ArchR</a></li>
<li class="chapter" data-level="9.2" data-path="identification-of-marker-features.html"><a href="identification-of-marker-features.html"><i class="fa fa-check"></i><b>9.2</b> Identification of Marker Features</a></li>
<li class="chapter" data-level="9.3" data-path="identifying-marker-genes.html"><a href="identifying-marker-genes.html"><i class="fa fa-check"></i><b>9.3</b> Identifying Marker Genes</a></li>
<li class="chapter" data-level="9.4" data-path="visualizing-marker-genes-on-an-embedding.html"><a href="visualizing-marker-genes-on-an-embedding.html"><i class="fa fa-check"></i><b>9.4</b> Visualizing Marker Genes on an Embedding</a></li>
<li class="chapter" data-level="9.5" data-path="marker-genes-imputation-with-magic.html"><a href="marker-genes-imputation-with-magic.html"><i class="fa fa-check"></i><b>9.5</b> Marker Genes Imputation with MAGIC</a></li>
<li class="chapter" data-level="9.6" data-path="module-scores.html"><a href="module-scores.html"><i class="fa fa-check"></i><b>9.6</b> Module Scores</a></li>
<li class="chapter" data-level="9.7" data-path="track-plotting-with-archrbrowser.html"><a href="track-plotting-with-archrbrowser.html"><i class="fa fa-check"></i><b>9.7</b> Track Plotting with ArchRBrowser</a></li>
<li class="chapter" data-level="9.8" data-path="launching-the-interactive-archrbrowser.html"><a href="launching-the-interactive-archrbrowser.html"><i class="fa fa-check"></i><b>9.8</b> Launching the Interactive ArchRBrowser</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="defining-cluster-identity-with-scrna-seq.html"><a href="defining-cluster-identity-with-scrna-seq.html"><i class="fa fa-check"></i><b>10</b> Defining Cluster Identity with scRNA-seq</a>
<ul>
<li class="chapter" data-level="10.1" data-path="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html"><i class="fa fa-check"></i><b>10.1</b> Cross-platform linkage of scATAC-seq cells with scRNA-seq cells</a></li>
<li class="chapter" data-level="10.2" data-path="adding-pseudo-scrna-seq-profiles-for-each-scatac-seq-cell.html"><a href="adding-pseudo-scrna-seq-profiles-for-each-scatac-seq-cell.html"><i class="fa fa-check"></i><b>10.2</b> Adding Pseudo-scRNA-seq profiles for each scATAC-seq cell</a></li>
<li class="chapter" data-level="10.3" data-path="labeling-scatac-seq-clusters-with-scrna-seq-information.html"><a href="labeling-scatac-seq-clusters-with-scrna-seq-information.html"><i class="fa fa-check"></i><b>10.3</b> Labeling scATAC-seq clusters with scRNA-seq information</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="pseudo-bulk-replicates-in-archr.html"><a href="pseudo-bulk-replicates-in-archr.html"><i class="fa fa-check"></i><b>11</b> Pseudo-bulk Replicates in ArchR</a>
<ul>
<li class="chapter" data-level="11.1" data-path="how-does-archr-make-pseudo-bulk-replicates.html"><a href="how-does-archr-make-pseudo-bulk-replicates.html"><i class="fa fa-check"></i><b>11.1</b> How Does ArchR Make Pseudo-bulk Replicates?</a></li>
<li class="chapter" data-level="11.2" data-path="making-pseudo-bulk-replicates.html"><a href="making-pseudo-bulk-replicates.html"><i class="fa fa-check"></i><b>11.2</b> Making Pseudo-bulk Replicates</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="calling-peaks-with-archr.html"><a href="calling-peaks-with-archr.html"><i class="fa fa-check"></i><b>12</b> Calling Peaks with ArchR</a>
<ul>
<li class="chapter" data-level="12.1" data-path="the-iterative-overlap-peak-merging-procedure.html"><a href="the-iterative-overlap-peak-merging-procedure.html"><i class="fa fa-check"></i><b>12.1</b> The Iterative Overlap Peak Merging Procedure</a></li>
<li class="chapter" data-level="12.2" data-path="calling-peaks-w-macs2.html"><a href="calling-peaks-w-macs2.html"><i class="fa fa-check"></i><b>12.2</b> Calling Peaks w/ Macs2</a></li>
<li class="chapter" data-level="12.3" data-path="calling-peaks-w-tilematrix.html"><a href="calling-peaks-w-tilematrix.html"><i class="fa fa-check"></i><b>12.3</b> Calling Peaks w/ TileMatrix</a></li>
<li class="chapter" data-level="12.4" data-path="adding-a-peak-matrix.html"><a href="adding-a-peak-matrix.html"><i class="fa fa-check"></i><b>12.4</b> Adding a Peak Matrix</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="identifying-marker-peaks.html"><a href="identifying-marker-peaks.html"><i class="fa fa-check"></i><b>13</b> Identifying Marker Peaks</a>
<ul>
<li class="chapter" data-level="13.1" data-path="identifying-marker-peaks-with-archr.html"><a href="identifying-marker-peaks-with-archr.html"><i class="fa fa-check"></i><b>13.1</b> Identifying Marker Peaks with ArchR</a></li>
<li class="chapter" data-level="13.2" data-path="plotting-marker-peaks-in-archr.html"><a href="plotting-marker-peaks-in-archr.html"><i class="fa fa-check"></i><b>13.2</b> Plotting Marker Peaks in ArchR</a></li>
<li class="chapter" data-level="13.3" data-path="pairwise-testing-between-groups.html"><a href="pairwise-testing-between-groups.html"><i class="fa fa-check"></i><b>13.3</b> Pairwise Testing Between Groups</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="motif-and-feature-enrichment-with-archr.html"><a href="motif-and-feature-enrichment-with-archr.html"><i class="fa fa-check"></i><b>14</b> Motif and Feature Enrichment with ArchR</a>
<ul>
<li class="chapter" data-level="14.1" data-path="motif-enrichment.html"><a href="motif-enrichment.html"><i class="fa fa-check"></i><b>14.1</b> Motif Enrichment</a></li>
<li class="chapter" data-level="14.2" data-path="archr-enrichment.html"><a href="archr-enrichment.html"><i class="fa fa-check"></i><b>14.2</b> ArchR Enrichment</a></li>
<li class="chapter" data-level="14.3" data-path="custom-enrichment.html"><a href="custom-enrichment.html"><i class="fa fa-check"></i><b>14.3</b> Custom Enrichment</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="chromvar-deviatons-enrichment-with-archr.html"><a href="chromvar-deviatons-enrichment-with-archr.html"><i class="fa fa-check"></i><b>15</b> ChromVAR Deviatons Enrichment with ArchR</a>
<ul>
<li class="chapter" data-level="15.1" data-path="motif-deviations.html"><a href="motif-deviations.html"><i class="fa fa-check"></i><b>15.1</b> Motif Deviations</a></li>
<li class="chapter" data-level="15.2" data-path="archr-and-custom-deviations.html"><a href="archr-and-custom-deviations.html"><i class="fa fa-check"></i><b>15.2</b> ArchR and Custom Deviations</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="footprinting-with-archr.html"><a href="footprinting-with-archr.html"><i class="fa fa-check"></i><b>16</b> Footprinting with ArchR</a>
<ul>
<li class="chapter" data-level="16.1" data-path="motif-footprinting.html"><a href="motif-footprinting.html"><i class="fa fa-check"></i><b>16.1</b> Motif Footprinting</a></li>
<li class="chapter" data-level="16.2" data-path="normalization-of-footprints-for-tn5-bias.html"><a href="normalization-of-footprints-for-tn5-bias.html"><i class="fa fa-check"></i><b>16.2</b> Normalization of Footprints for Tn5 Bias</a></li>
<li class="chapter" data-level="16.3" data-path="feature-footprinting.html"><a href="feature-footprinting.html"><i class="fa fa-check"></i><b>16.3</b> Feature Footprinting</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="integrative-analysis-with-archr.html"><a href="integrative-analysis-with-archr.html"><i class="fa fa-check"></i><b>17</b> Integrative Analysis with ArchR</a>
<ul>
<li class="chapter" data-level="17.1" data-path="creating-low-overlapping-aggregates-of-cells.html"><a href="creating-low-overlapping-aggregates-of-cells.html"><i class="fa fa-check"></i><b>17.1</b> Creating Low-Overlapping Aggregates of Cells</a></li>
<li class="chapter" data-level="17.2" data-path="co-accessibility-with-archr.html"><a href="co-accessibility-with-archr.html"><i class="fa fa-check"></i><b>17.2</b> Co-accessibility with ArchR</a></li>
<li class="chapter" data-level="17.3" data-path="peak2genelinkage-with-archr.html"><a href="peak2genelinkage-with-archr.html"><i class="fa fa-check"></i><b>17.3</b> Peak2GeneLinkage with ArchR</a></li>
<li class="chapter" data-level="17.4" data-path="identification-of-positive-tf-regulators.html"><a href="identification-of-positive-tf-regulators.html"><i class="fa fa-check"></i><b>17.4</b> Identification of Positive TF-Regulators</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="trajectory-analysis-with-archr.html"><a href="trajectory-analysis-with-archr.html"><i class="fa fa-check"></i><b>18</b> Trajectory Analysis with ArchR</a>
<ul>
<li class="chapter" data-level="18.1" data-path="myeloid-trajectory---monocyte-differentiation.html"><a href="myeloid-trajectory---monocyte-differentiation.html"><i class="fa fa-check"></i><b>18.1</b> Myeloid Trajectory - Monocyte Differentiation</a></li>
<li class="chapter" data-level="18.2" data-path="lymphoid-trajectory---b-cell-cifferentiation.html"><a href="lymphoid-trajectory---b-cell-cifferentiation.html"><i class="fa fa-check"></i><b>18.2</b> Lymphoid Trajectory - B Cell Cifferentiation</a></li>
<li class="chapter" data-level="18.3" data-path="monocle3-trajectories.html"><a href="monocle3-trajectories.html"><i class="fa fa-check"></i><b>18.3</b> Monocle3 Trajectories</a></li>
<li class="chapter" data-level="18.4" data-path="slingshot-trajectories.html"><a href="slingshot-trajectories.html"><i class="fa fa-check"></i><b>18.4</b> Slingshot Trajectories</a></li>
<li class="chapter" data-level="18.5" data-path="comparing-the-different-trajectories.html"><a href="comparing-the-different-trajectories.html"><i class="fa fa-check"></i><b>18.5</b> Comparing the different trajectories</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="handling-archr-output.html"><a href="handling-archr-output.html"><i class="fa fa-check"></i><b>19</b> Handling ArchR Output</a>
<ul>
<li class="chapter" data-level="19.1" data-path="exporting-fragment-level-data.html"><a href="exporting-fragment-level-data.html"><i class="fa fa-check"></i><b>19.1</b> Exporting fragment-level data</a></li>
<li class="chapter" data-level="19.2" data-path="exporting-matrix-level-data.html"><a href="exporting-matrix-level-data.html"><i class="fa fa-check"></i><b>19.2</b> Exporting matrix-level data</a></li>
<li class="chapter" data-level="19.3" data-path="exporting-pseudo-bulked-data-to-a-summarizedexperiment.html"><a href="exporting-pseudo-bulked-data-to-a-summarizedexperiment.html"><i class="fa fa-check"></i><b>19.3</b> Exporting pseudo-bulked data to a <code>SummarizedExperiment</code></a></li>
<li class="chapter" data-level="19.4" data-path="exporting-pseudo-bulked-data-to-a-bigwig-file.html"><a href="exporting-pseudo-bulked-data-to-a-bigwig-file.html"><i class="fa fa-check"></i><b>19.4</b> Exporting pseudo-bulked data to a bigWig file</a></li>
</ul></li>
<li class="chapter" data-level="20" data-path="integration-with-bulk-atac-seq.html"><a href="integration-with-bulk-atac-seq.html"><i class="fa fa-check"></i><b>20</b> Integration with bulk ATAC-seq</a>
<ul>
<li class="chapter" data-level="20.1" data-path="projecting-bulk-atac-seq-data.html"><a href="projecting-bulk-atac-seq-data.html"><i class="fa fa-check"></i><b>20.1</b> Projecting bulk ATAC-seq data</a></li>
</ul></li>
<li class="chapter" data-level="21" data-path="multiomic-data-analysis-in-archr.html"><a href="multiomic-data-analysis-in-archr.html"><i class="fa fa-check"></i><b>21</b> Multiomic data analysis in ArchR</a>
<ul>
<li class="chapter" data-level="21.1" data-path="importing-data-and-setting-up-a-multiome-project.html"><a href="importing-data-and-setting-up-a-multiome-project.html"><i class="fa fa-check"></i><b>21.1</b> Importing data and setting up a Multiome project</a></li>
<li class="chapter" data-level="21.2" data-path="analysis-of-multiome-data-in-archr.html"><a href="analysis-of-multiome-data-in-archr.html"><i class="fa fa-check"></i><b>21.2</b> Analysis of multiome data in ArchR</a></li>
</ul></li>
<li class="chapter" data-level="22" data-path="plot-aesthetics-in-archr.html"><a href="plot-aesthetics-in-archr.html"><i class="fa fa-check"></i><b>22</b> Plot aesthetics in ArchR</a>
<ul>
<li class="chapter" data-level="22.1" data-path="using-and-manipulating-palettes-in-archr.html"><a href="using-and-manipulating-palettes-in-archr.html"><i class="fa fa-check"></i><b>22.1</b> Using and manipulating palettes in ArchR</a></li>
</ul></li>
<li class="chapter" data-level="23" data-path="testing-your-archr-installation.html"><a href="testing-your-archr-installation.html"><i class="fa fa-check"></i><b>23</b> Testing Your ArchR Installation</a></li>
<li class="chapter" data-level="24" data-path="session-information.html"><a href="session-information.html"><i class="fa fa-check"></i><b>24</b> Session Information</a></li>
<li class="divider"></li>
<li><a href="https://bookdown.org" target="_blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">ArchR: Robust and scaleable analysis of single-cell chromatin accessibility data.</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells" class="section level2 hasAnchor" number="10.1">
<h2><span class="header-section-number">10.1</span> Cross-platform linkage of scATAC-seq cells with scRNA-seq cells<a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>In order to integrate our tutorial scATAC-seq data with matched scRNA-seq data, we will use scRNA-seq data derived from the same hematopoietic cell types from Granja* et al (2019).</p>
<p>We have stored this scRNA-seq data as a 111 MB <code>RangedSummarizedExperiment</code> object. However, ArchR also accepts unmodified <code>Seurat</code> objects as input to the integration workflow. However, Seurat SCTransform-normalized data is not suitable for this integration workflow. Downloading and examining this object, we see that it has a gene expression counts matrix and associated metadata.</p>
<div class="sourceCode" id="cb144"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb144-1"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb144-1" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">file.exists</span>(<span class="st">&quot;scRNA-Hematopoiesis-Granja-2019.rds&quot;</span>)){</span>
<span id="cb144-2"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb144-2" tabindex="-1"></a>    <span class="fu">download.file</span>(</span>
<span id="cb144-3"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb144-3" tabindex="-1"></a>        <span class="at">url =</span> <span class="st">&quot;https://jeffgranja.s3.amazonaws.com/ArchR/TestData/scRNA-Hematopoiesis-Granja-2019.rds&quot;</span>,</span>
<span id="cb144-4"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb144-4" tabindex="-1"></a>        <span class="at">destfile =</span> <span class="st">&quot;scRNA-Hematopoiesis-Granja-2019.rds&quot;</span></span>
<span id="cb144-5"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb144-5" tabindex="-1"></a>    )</span>
<span id="cb144-6"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb144-6" tabindex="-1"></a>}</span>
<span id="cb144-7"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb144-7" tabindex="-1"></a></span>
<span id="cb144-8"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb144-8" tabindex="-1"></a>seRNA <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;scRNA-Hematopoiesis-Granja-2019.rds&quot;</span>)</span>
<span id="cb144-9"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb144-9" tabindex="-1"></a>seRNA</span>
<span id="cb144-10"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb144-10" tabindex="-1"></a><span class="do">## class: RangedSummarizedExperiment </span></span>
<span id="cb144-11"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb144-11" tabindex="-1"></a><span class="do">## dim: 20287 35582 </span></span>
<span id="cb144-12"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb144-12" tabindex="-1"></a><span class="do">## metadata(0):</span></span>
<span id="cb144-13"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb144-13" tabindex="-1"></a><span class="do">## assays(1): counts</span></span>
<span id="cb144-14"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb144-14" tabindex="-1"></a><span class="do">## rownames(20287): FAM138A OR4F5 ... S100B PRMT2</span></span>
<span id="cb144-15"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb144-15" tabindex="-1"></a><span class="do">## rowData names(3): gene_name gene_id exonLength</span></span>
<span id="cb144-16"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb144-16" tabindex="-1"></a><span class="do">## colnames(35582): CD34_32_R5:AAACCTGAGTATCGAA-1</span></span>
<span id="cb144-17"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb144-17" tabindex="-1"></a><span class="do">##   CD34_32_R5:AAACCTGAGTCGTTTG-1 ...</span></span>
<span id="cb144-18"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb144-18" tabindex="-1"></a><span class="do">##   BMMC_10x_GREENLEAF_REP2:TTTGTTGCATGTGTCA-1</span></span>
<span id="cb144-19"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb144-19" tabindex="-1"></a><span class="do">##   BMMC_10x_GREENLEAF_REP2:TTTGTTGCATTGAAAG-1</span></span>
<span id="cb144-20"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb144-20" tabindex="-1"></a><span class="do">## colData names(10): Group nUMI_pre ... BioClassification Barcode</span></span></code></pre></div>
<p>The metadata contains a column called <code>BioClassification</code> which contains the cell type classifications for each cell in the scRNA-seq dataset.</p>
<div class="sourceCode" id="cb145"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb145-1"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb145-1" tabindex="-1"></a><span class="fu">colnames</span>(<span class="fu">colData</span>(seRNA))</span>
<span id="cb145-2"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb145-2" tabindex="-1"></a><span class="do">##  [1] &quot;Group&quot;             &quot;nUMI_pre&quot;          &quot;nUMI&quot;             </span></span>
<span id="cb145-3"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb145-3" tabindex="-1"></a><span class="do">##  [4] &quot;nGene&quot;             &quot;initialClusters&quot;   &quot;UMAP1&quot;            </span></span>
<span id="cb145-4"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb145-4" tabindex="-1"></a><span class="do">##  [7] &quot;UMAP2&quot;             &quot;Clusters&quot;          &quot;BioClassification&quot;</span></span>
<span id="cb145-5"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb145-5" tabindex="-1"></a><span class="do">## [10] &quot;Barcode&quot;</span></span></code></pre></div>
<p>Using <code>table()</code> we can see how many cells are in each of the scRNA-seq cell type classifications.</p>
<div class="sourceCode" id="cb146"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb146-1"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb146-1" tabindex="-1"></a><span class="fu">table</span>(<span class="fu">colData</span>(seRNA)<span class="sc">$</span>BioClassification)</span>
<span id="cb146-2"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb146-2" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb146-3"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb146-3" tabindex="-1"></a><span class="do">##         01_HSC 02_Early.Eryth  03_Late.Eryth  04_Early.Baso    05_CMP.LMPP </span></span>
<span id="cb146-4"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb146-4" tabindex="-1"></a><span class="do">##           1425           1653            446            111           2260 </span></span>
<span id="cb146-5"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb146-5" tabindex="-1"></a><span class="do">##       06_CLP.1         07_GMP    08_GMP.Neut         09_pDC         10_cDC </span></span>
<span id="cb146-6"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb146-6" tabindex="-1"></a><span class="do">##            903           2097           1050            544            325 </span></span>
<span id="cb146-7"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb146-7" tabindex="-1"></a><span class="do">## 11_CD14.Mono.1 12_CD14.Mono.2   13_CD16.Mono         14_Unk       15_CLP.2 </span></span>
<span id="cb146-8"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb146-8" tabindex="-1"></a><span class="do">##           1800           4222            292            520            377 </span></span>
<span id="cb146-9"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb146-9" tabindex="-1"></a><span class="do">##       16_Pre.B           17_B      18_Plasma       19_CD8.N      20_CD4.N1 </span></span>
<span id="cb146-10"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb146-10" tabindex="-1"></a><span class="do">##            710           1711             62           1521           2470 </span></span>
<span id="cb146-11"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb146-11" tabindex="-1"></a><span class="do">##      21_CD4.N2       22_CD4.M      23_CD8.EM      24_CD8.CM          25_NK </span></span>
<span id="cb146-12"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb146-12" tabindex="-1"></a><span class="do">##           2364           3539            796           2080           2143 </span></span>
<span id="cb146-13"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb146-13" tabindex="-1"></a><span class="do">##         26_Unk </span></span>
<span id="cb146-14"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb146-14" tabindex="-1"></a><span class="do">##            161</span></span></code></pre></div>
<p>There are two types of integration approaches that we can perform. <strong>Unconstrained integration</strong> is a completely agnostic approach that would take all of the cells in your scATAC-seq experiment and attempt to align them to any of the cells in the scRNA-seq experiment. While this is a feasible preliminary solution, we can improve the quality of our cross-platform alignment by <em>constraining</em> the integration process. To perform a <strong>constrained integration</strong> we use prior knowledge of the cell types to limit the search space of the alignment. For example, if we knew that Clusters A, B, and C in the scATAC-seq data corresponded to 3 different T cell clusters, and we knew that Clusters X and Y in the scRNA-seq data corresponded to 2 different T cell clusters, we could tell ArchR to specifically try to align cells from scATAC-seq Clusters A, B, and C to cell from scRNA-seq clusters X and Y. We illustrate these two approaches below, first performing an unconstrained integration to acheive preliminary cluster identities and then using this prior knowledge to perform a more refined constrained integration. However, please note that your own application must guide which type of integration you perform. Constrained integration may not be suited for all applications so the below sections are merely a guide for how to accomplish this and should not be viewed as a “tutorial” on how to integrate your own data.</p>
<div id="unconstrained-integration" class="section level3 hasAnchor" number="10.1.1">
<h3><span class="header-section-number">10.1.1</span> Unconstrained Integration<a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#unconstrained-integration" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>To integrate scATAC-seq with scRNA-seq, we use the <code>addGeneIntegrationMatrix()</code> function. As mentioned above, this function accepts either a <code>Seurat</code> object or a <code>RangedSummarizedExperiment</code> object via the <code>seRNA</code> parameter. The first round of integration we perform will be a preliminary <strong>unconstrained</strong> integration and we will not store this in the Arrow files (<code>addToArrow = FALSE</code>). We provide a name for the integration matrix that will be stored in the <code>ArchRProject</code> via the <code>matrixName</code> parameter. The other key parameters for this function provide column names in <code>cellColData</code> where certain information will be stored: <code>nameCell</code> will store the cell ID from the matched scRNA-seq cell, <code>nameGroup</code> will store the group ID from the scRNA-seq cell, and <code>nameScore</code> will store the cross-platform integration score.</p>
<div class="sourceCode" id="cb147"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb147-1"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb147-1" tabindex="-1"></a>projHeme2 <span class="ot">&lt;-</span> <span class="fu">addGeneIntegrationMatrix</span>(</span>
<span id="cb147-2"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb147-2" tabindex="-1"></a>    <span class="at">ArchRProj =</span> projHeme2, </span>
<span id="cb147-3"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb147-3" tabindex="-1"></a>    <span class="at">useMatrix =</span> <span class="st">&quot;GeneScoreMatrix&quot;</span>,</span>
<span id="cb147-4"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb147-4" tabindex="-1"></a>    <span class="at">matrixName =</span> <span class="st">&quot;GeneIntegrationMatrix&quot;</span>,</span>
<span id="cb147-5"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb147-5" tabindex="-1"></a>    <span class="at">reducedDims =</span> <span class="st">&quot;IterativeLSI&quot;</span>,</span>
<span id="cb147-6"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb147-6" tabindex="-1"></a>    <span class="at">seRNA =</span> seRNA,</span>
<span id="cb147-7"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb147-7" tabindex="-1"></a>    <span class="at">addToArrow =</span> <span class="cn">FALSE</span>,</span>
<span id="cb147-8"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb147-8" tabindex="-1"></a>    <span class="at">groupRNA =</span> <span class="st">&quot;BioClassification&quot;</span>,</span>
<span id="cb147-9"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb147-9" tabindex="-1"></a>    <span class="at">nameCell =</span> <span class="st">&quot;predictedCell_Un&quot;</span>,</span>
<span id="cb147-10"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb147-10" tabindex="-1"></a>    <span class="at">nameGroup =</span> <span class="st">&quot;predictedGroup_Un&quot;</span>,</span>
<span id="cb147-11"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb147-11" tabindex="-1"></a>    <span class="at">nameScore =</span> <span class="st">&quot;predictedScore_Un&quot;</span></span>
<span id="cb147-12"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb147-12" tabindex="-1"></a>)</span>
<span id="cb147-13"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb147-13" tabindex="-1"></a><span class="do">## ArchR logging to : ArchRLogs/ArchR-addGeneIntegrationMatrix-1a11a7b24cd-Date-2025-01-23_Time-20-20-12.9844.log</span></span>
<span id="cb147-14"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb147-14" tabindex="-1"></a><span class="do">## If there is an issue, please report to github with logFile!</span></span>
<span id="cb147-15"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb147-15" tabindex="-1"></a><span class="do">## 2025-01-23 20:20:13.163894 : Running Seurat&#39;s Integration Stuart* et al 2019, 0.003 mins elapsed.</span></span>
<span id="cb147-16"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb147-16" tabindex="-1"></a><span class="do">## 2025-01-23 20:20:13.759011 : Checking ATAC Input, 0.013 mins elapsed.</span></span>
<span id="cb147-17"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb147-17" tabindex="-1"></a><span class="do">## 2025-01-23 20:20:14.299231 : Checking RNA Input, 0.022 mins elapsed.</span></span>
<span id="cb147-18"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb147-18" tabindex="-1"></a><span class="do">## 2025-01-23 20:20:27.755453 : Found 18601 overlapping gene names from gene scores and rna matrix!, 0.246 mins elapsed.</span></span>
<span id="cb147-19"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb147-19" tabindex="-1"></a><span class="do">## 2025-01-23 20:20:27.76262 : Creating Integration Blocks, 0.246 mins elapsed.</span></span>
<span id="cb147-20"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb147-20" tabindex="-1"></a><span class="do">## 2025-01-23 20:20:27.955851 : Prepping Interation Data, 0.25 mins elapsed.</span></span>
<span id="cb147-21"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb147-21" tabindex="-1"></a><span class="do">## subThreading Disabled since ArchRLocking is TRUE see `addArchRLocking`</span></span>
<span id="cb147-22"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb147-22" tabindex="-1"></a><span class="do">## 2025-01-23 20:20:28.685436 : Computing Integration in 1 Integration Blocks!, 0 mins elapsed.</span></span>
<span id="cb147-23"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb147-23" tabindex="-1"></a><span class="do">## 2025-01-23 20:20:28.72401 : Block (1 of 1) : Computing Integration, 0.001 mins elapsed.</span></span>
<span id="cb147-24"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb147-24" tabindex="-1"></a><span class="do">## 2025-01-23 20:20:30.100461 : Block (1 of 1) : Identifying Variable Genes, 0.024 mins elapsed.</span></span>
<span id="cb147-25"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb147-25" tabindex="-1"></a><span class="do">## 2025-01-23 20:20:34.107292 : Block (1 of 1) : Getting GeneScoreMatrix, 0.09 mins elapsed.</span></span>
<span id="cb147-26"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb147-26" tabindex="-1"></a><span class="do">## 2025-01-23 20:20:43.429565 : Block (1 of 1) : Imputing GeneScoreMatrix, 0.246 mins elapsed.</span></span>
<span id="cb147-27"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb147-27" tabindex="-1"></a><span class="do">## Getting ImputeWeights</span></span>
<span id="cb147-28"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb147-28" tabindex="-1"></a><span class="do">## 2025-01-23 20:21:14.660182 : Block (1 of 1) : Seurat FindTransferAnchors, 0.766 mins elapsed.</span></span>
<span id="cb147-29"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb147-29" tabindex="-1"></a><span class="do">## 2025-01-23 20:22:54.939764 : Block (1 of 1) : Seurat TransferData Cell Group Labels, 2.438 mins elapsed.</span></span>
<span id="cb147-30"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb147-30" tabindex="-1"></a><span class="do">## 2025-01-23 20:23:16.331755 : Block (1 of 1) : Seurat TransferData Cell Names Labels, 2.794 mins elapsed.</span></span>
<span id="cb147-31"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb147-31" tabindex="-1"></a><span class="do">## 2025-01-23 20:23:51.071236 : Block (1 of 1) : Saving TransferAnchors Joint CCA, 3.373 mins elapsed.</span></span>
<span id="cb147-32"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb147-32" tabindex="-1"></a><span class="do">## 2025-01-23 20:23:53.643257 : Block (1 of 1) : Completed Integration, 3.416 mins elapsed.</span></span>
<span id="cb147-33"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb147-33" tabindex="-1"></a><span class="do">## 2025-01-23 20:23:55.801247 : Block (1 of 1) : Plotting Joint UMAP, 3.452 mins elapsed.</span></span>
<span id="cb147-34"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb147-34" tabindex="-1"></a><span class="do">## Length of unique values greater than palette, interpolating..</span></span>
<span id="cb147-35"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb147-35" tabindex="-1"></a><span class="do">## 2025-01-23 20:24:19.785256 : Completed Integration with RNA Matrix, 3.852 mins elapsed.</span></span>
<span id="cb147-36"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb147-36" tabindex="-1"></a><span class="do">## ArchR logging successful to : ArchRLogs/ArchR-addGeneIntegrationMatrix-1a11a7b24cd-Date-2025-01-23_Time-20-20-12.9844.log</span></span></code></pre></div>
<p>This unconstrained integration can be used as shown below to perform a more refined constrained integration.</p>
<p>You may want to assess the “quality” of the integration between your scATAC-seq and scRNA-seq data. ArchR provides two general ways to do this. The first is the option to ouput a UMAP plot of the joint CCA subspace. This is enabled through the <code>plotUMAP</code> and <code>UMAPParams</code> parameters to <code>addGeneIntegrationMatrix()</code>. If <code>plotUMAP = TRUE</code>, then a UMAP plot is saved for each block of the integration in a subfolder (based on <code>matrixName</code>) of the “RNAIntegration” folder. These plots can be difficult to interpret, especially with the intercellular heterogeneity isn’t very strong. As an alternative, ArchR also stores the <code>NAME.score</code> value from <code>Seurat::TransferData()</code> in a column of <code>cellColData</code> indicated by what you pass to the <code>nameScore</code> parameter.</p>
</div>
<div id="constrained-integration" class="section level3 hasAnchor" number="10.1.2">
<h3><span class="header-section-number">10.1.2</span> Constrained Integration<a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#constrained-integration" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Now that we have our preliminary unconstrained integration, we will identify general cell types to profide a framework to further refine the integration results.</p>
<p>Given that this tutorial data is from hematopoietic cells, we would ideally constrain the integration to associate similar cell types togther. First, we will identify which cell types from the scRNA-seq data are most abundant in each of our scATAC-seq clusters. The goal of this is to identify the cells in both the scATAC-seq and scRNA-seq data that correspond to T cells and NK cells using the unconstrained integration so that we can perform a constrained integration using this information. To do this, we will create a <code>confusionMatrix</code> that looks at the intersection of <code>Clusters</code> and <code>predictedGroup_Un</code> which contains the cell types as identified by scRNA-seq.</p>
<div class="sourceCode" id="cb148"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb148-1"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb148-1" tabindex="-1"></a>cM <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">confusionMatrix</span>(projHeme2<span class="sc">$</span>Clusters, projHeme2<span class="sc">$</span>predictedGroup_Un))</span>
<span id="cb148-2"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb148-2" tabindex="-1"></a>preClust <span class="ot">&lt;-</span> <span class="fu">colnames</span>(cM)[<span class="fu">apply</span>(cM, <span class="dv">1</span> , which.max)]</span>
<span id="cb148-3"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb148-3" tabindex="-1"></a><span class="fu">cbind</span>(preClust, <span class="fu">rownames</span>(cM)) <span class="co">#Assignments</span></span>
<span id="cb148-4"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb148-4" tabindex="-1"></a><span class="do">##       preClust              </span></span>
<span id="cb148-5"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb148-5" tabindex="-1"></a><span class="do">##  [1,] &quot;17_B&quot;           &quot;C9&quot; </span></span>
<span id="cb148-6"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb148-6" tabindex="-1"></a><span class="do">##  [2,] &quot;20_CD4.N1&quot;      &quot;C11&quot;</span></span>
<span id="cb148-7"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb148-7" tabindex="-1"></a><span class="do">##  [3,] &quot;16_Pre.B&quot;       &quot;C4&quot; </span></span>
<span id="cb148-8"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb148-8" tabindex="-1"></a><span class="do">##  [4,] &quot;08_GMP.Neut&quot;    &quot;C7&quot; </span></span>
<span id="cb148-9"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb148-9" tabindex="-1"></a><span class="do">##  [5,] &quot;12_CD14.Mono.2&quot; &quot;C1&quot; </span></span>
<span id="cb148-10"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb148-10" tabindex="-1"></a><span class="do">##  [6,] &quot;02_Early.Eryth&quot; &quot;C6&quot; </span></span>
<span id="cb148-11"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb148-11" tabindex="-1"></a><span class="do">##  [7,] &quot;01_HSC&quot;         &quot;C8&quot; </span></span>
<span id="cb148-12"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb148-12" tabindex="-1"></a><span class="do">##  [8,] &quot;22_CD4.M&quot;       &quot;C12&quot;</span></span>
<span id="cb148-13"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb148-13" tabindex="-1"></a><span class="do">##  [9,] &quot;09_pDC&quot;         &quot;C3&quot; </span></span>
<span id="cb148-14"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb148-14" tabindex="-1"></a><span class="do">## [10,] &quot;25_NK&quot;          &quot;C10&quot;</span></span>
<span id="cb148-15"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb148-15" tabindex="-1"></a><span class="do">## [11,] &quot;12_CD14.Mono.2&quot; &quot;C2&quot; </span></span>
<span id="cb148-16"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb148-16" tabindex="-1"></a><span class="do">## [12,] &quot;15_CLP.2&quot;       &quot;C5&quot;</span></span></code></pre></div>
<p>The above list shows which scRNA-seq cell type is most abundant in each of the 12 scATAC-seq clusters.</p>
<p>First, lets look at the cell type labels from our scRNA-seq data that were used in our unconstrained integration:</p>
<div class="sourceCode" id="cb149"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb149-1"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb149-1" tabindex="-1"></a><span class="fu">unique</span>(<span class="fu">unique</span>(projHeme2<span class="sc">$</span>predictedGroup_Un))</span>
<span id="cb149-2"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb149-2" tabindex="-1"></a><span class="do">##  [1] &quot;17_B&quot;           &quot;22_CD4.M&quot;       &quot;16_Pre.B&quot;       &quot;06_CLP.1&quot;      </span></span>
<span id="cb149-3"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb149-3" tabindex="-1"></a><span class="do">##  [5] &quot;08_GMP.Neut&quot;    &quot;24_CD8.CM&quot;      &quot;11_CD14.Mono.1&quot; &quot;12_CD14.Mono.2&quot;</span></span>
<span id="cb149-4"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb149-4" tabindex="-1"></a><span class="do">##  [9] &quot;07_GMP&quot;         &quot;02_Early.Eryth&quot; &quot;05_CMP.LMPP&quot;    &quot;03_Late.Eryth&quot; </span></span>
<span id="cb149-5"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb149-5" tabindex="-1"></a><span class="do">## [13] &quot;04_Early.Baso&quot;  &quot;09_pDC&quot;         &quot;20_CD4.N1&quot;      &quot;10_cDC&quot;        </span></span>
<span id="cb149-6"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb149-6" tabindex="-1"></a><span class="do">## [17] &quot;23_CD8.EM&quot;      &quot;19_CD8.N&quot;       &quot;25_NK&quot;          &quot;15_CLP.2&quot;      </span></span>
<span id="cb149-7"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb149-7" tabindex="-1"></a><span class="do">## [21] &quot;21_CD4.N2&quot;      &quot;01_HSC&quot;         &quot;14_Unk&quot;         &quot;13_CD16.Mono&quot;</span></span></code></pre></div>
<p>From the above list, we can see that the clusters in the scRNA-seq data that correspond to T cells and NK cells are Clusters 19-25. We will create a string-based representation of these clusters to use in the downstream constrained integration.</p>
<div class="sourceCode" id="cb150"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb150-1"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb150-1" tabindex="-1"></a><span class="co">#From scRNA</span></span>
<span id="cb150-2"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb150-2" tabindex="-1"></a>cTNK <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="fu">paste0</span>(<span class="dv">19</span><span class="sc">:</span><span class="dv">25</span>), <span class="at">collapse=</span><span class="st">&quot;|&quot;</span>)</span>
<span id="cb150-3"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb150-3" tabindex="-1"></a>cTNK</span>
<span id="cb150-4"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb150-4" tabindex="-1"></a><span class="do">## [1] &quot;19|20|21|22|23|24|25&quot;</span></span></code></pre></div>
<p>We can then take all of the other clusters and create a string-based representation of all “Non-T cell, Non-NK cell” clusters (i.e. Cluster 1 - 18).</p>
<div class="sourceCode" id="cb151"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb151-1"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb151-1" tabindex="-1"></a>cNonTNK <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="fu">c</span>(<span class="fu">paste0</span>(<span class="st">&quot;0&quot;</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">9</span>), <span class="dv">10</span><span class="sc">:</span><span class="dv">13</span>, <span class="dv">15</span><span class="sc">:</span><span class="dv">18</span>), <span class="at">collapse=</span><span class="st">&quot;|&quot;</span>)</span>
<span id="cb151-2"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb151-2" tabindex="-1"></a>cNonTNK</span>
<span id="cb151-3"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb151-3" tabindex="-1"></a><span class="do">## [1] &quot;01|02|03|04|05|06|07|08|09|10|11|12|13|15|16|17|18&quot;</span></span></code></pre></div>
<p>These string-based representations are pattern matching strings that we will use with <code>grep</code> to extract the scATAC-seq clusters that correspond to these scRNA-seq cell types. The <code>|</code> in the string acts as an <code>or</code> statement so we end up searching for any row in the <code>preClust</code> column of our confusion matrix that matches one of the scRNA-seq cluster numbers provided in the pattern match string.</p>
<p>For T cells and NK cells, this identifies scATAC-seq clusters C7, C8, and C9:</p>
<div class="sourceCode" id="cb152"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb152-1"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb152-1" tabindex="-1"></a>clustTNK <span class="ot">&lt;-</span> <span class="fu">rownames</span>(cM)[<span class="fu">grep</span>(cTNK, preClust)]</span>
<span id="cb152-2"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb152-2" tabindex="-1"></a>clustTNK</span>
<span id="cb152-3"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb152-3" tabindex="-1"></a><span class="do">## [1] &quot;C11&quot; &quot;C12&quot; &quot;C10&quot;</span></span></code></pre></div>
<p>For Non-T cells and Non-NK cells, this identifies the remaining scATAC-seq clusters:</p>
<div class="sourceCode" id="cb153"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb153-1"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb153-1" tabindex="-1"></a>clustNonTNK <span class="ot">&lt;-</span> <span class="fu">rownames</span>(cM)[<span class="fu">grep</span>(cNonTNK, preClust)]</span>
<span id="cb153-2"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb153-2" tabindex="-1"></a>clustNonTNK</span>
<span id="cb153-3"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb153-3" tabindex="-1"></a><span class="do">## [1] &quot;C9&quot; &quot;C4&quot; &quot;C7&quot; &quot;C1&quot; &quot;C6&quot; &quot;C8&quot; &quot;C3&quot; &quot;C2&quot; &quot;C5&quot;</span></span></code></pre></div>
<p>We then perform a similar opperation to identify the scRNA-seq cells that correspond to these same cell types.
First, we identify the T cell and NK cells in the scRNA-seq data</p>
<div class="sourceCode" id="cb154"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb154-1"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb154-1" tabindex="-1"></a>rnaTNK <span class="ot">&lt;-</span> <span class="fu">colnames</span>(seRNA)[<span class="fu">grep</span>(cTNK, <span class="fu">colData</span>(seRNA)<span class="sc">$</span>BioClassification)]</span>
<span id="cb154-2"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb154-2" tabindex="-1"></a><span class="fu">head</span>(rnaTNK)</span>
<span id="cb154-3"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb154-3" tabindex="-1"></a><span class="do">## [1] &quot;PBMC_10x_GREENLEAF_REP1:AAACCCAGTCGTCATA-1&quot;</span></span>
<span id="cb154-4"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb154-4" tabindex="-1"></a><span class="do">## [2] &quot;PBMC_10x_GREENLEAF_REP1:AAACCCATCCGATGTA-1&quot;</span></span>
<span id="cb154-5"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb154-5" tabindex="-1"></a><span class="do">## [3] &quot;PBMC_10x_GREENLEAF_REP1:AAACCCATCTCAACGA-1&quot;</span></span>
<span id="cb154-6"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb154-6" tabindex="-1"></a><span class="do">## [4] &quot;PBMC_10x_GREENLEAF_REP1:AAACCCATCTCTCGAC-1&quot;</span></span>
<span id="cb154-7"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb154-7" tabindex="-1"></a><span class="do">## [5] &quot;PBMC_10x_GREENLEAF_REP1:AAACGAACAATCGTCA-1&quot;</span></span>
<span id="cb154-8"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb154-8" tabindex="-1"></a><span class="do">## [6] &quot;PBMC_10x_GREENLEAF_REP1:AAACGAACACGATTCA-1&quot;</span></span></code></pre></div>
<p>Then, we identify the Non-T cell Non-NK cell cells in the scRNA-seq data.</p>
<div class="sourceCode" id="cb155"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb155-1"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb155-1" tabindex="-1"></a>rnaNonTNK <span class="ot">&lt;-</span> <span class="fu">colnames</span>(seRNA)[<span class="fu">grep</span>(cNonTNK, <span class="fu">colData</span>(seRNA)<span class="sc">$</span>BioClassification)]</span>
<span id="cb155-2"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb155-2" tabindex="-1"></a><span class="fu">head</span>(rnaNonTNK)</span>
<span id="cb155-3"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb155-3" tabindex="-1"></a><span class="do">## [1] &quot;CD34_32_R5:AAACCTGAGTATCGAA-1&quot; &quot;CD34_32_R5:AAACCTGAGTCGTTTG-1&quot;</span></span>
<span id="cb155-4"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb155-4" tabindex="-1"></a><span class="do">## [3] &quot;CD34_32_R5:AAACCTGGTTCCACAA-1&quot; &quot;CD34_32_R5:AAACGGGAGCTTCGCG-1&quot;</span></span>
<span id="cb155-5"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb155-5" tabindex="-1"></a><span class="do">## [5] &quot;CD34_32_R5:AAACGGGAGGGAGTAA-1&quot; &quot;CD34_32_R5:AAACGGGAGTTACGGG-1&quot;</span></span></code></pre></div>
<p>To prepare for this constrained integration, we create a nested list. This is a <code>SimpleList</code> of multiple <code>SimpleList</code> objects, one for each group that we would like to constrain. In this example, we have two groups: one group called <code>TNK</code> that identifies the T cells and NK cells across the two platforms, and a second group called <code>NonTNK</code> that identifies the Non-T cell Non-NK cell cells across the two platforms. Each of these <code>SimpleList</code> objects has two vectors of cell IDs, one called <code>ATAC</code> and one called <code>RNA</code> as shown below:</p>
<div class="sourceCode" id="cb156"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb156-1"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb156-1" tabindex="-1"></a>groupList <span class="ot">&lt;-</span> <span class="fu">SimpleList</span>(</span>
<span id="cb156-2"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb156-2" tabindex="-1"></a>    <span class="at">TNK =</span> <span class="fu">SimpleList</span>(</span>
<span id="cb156-3"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb156-3" tabindex="-1"></a>        <span class="at">ATAC =</span> projHeme2<span class="sc">$</span>cellNames[projHeme2<span class="sc">$</span>Clusters <span class="sc">%in%</span> clustTNK],</span>
<span id="cb156-4"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb156-4" tabindex="-1"></a>        <span class="at">RNA =</span> rnaTNK</span>
<span id="cb156-5"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb156-5" tabindex="-1"></a>    ),</span>
<span id="cb156-6"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb156-6" tabindex="-1"></a>    <span class="at">NonTNK =</span> <span class="fu">SimpleList</span>(</span>
<span id="cb156-7"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb156-7" tabindex="-1"></a>        <span class="at">ATAC =</span> projHeme2<span class="sc">$</span>cellNames[projHeme2<span class="sc">$</span>Clusters <span class="sc">%in%</span> clustNonTNK],</span>
<span id="cb156-8"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb156-8" tabindex="-1"></a>        <span class="at">RNA =</span> rnaNonTNK</span>
<span id="cb156-9"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb156-9" tabindex="-1"></a>    )    </span>
<span id="cb156-10"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb156-10" tabindex="-1"></a>)</span></code></pre></div>
<p>We pass this list to the <code>groupList</code> parameter of the <code>addGeneIntegrationMatrix()</code> function to constrain our integration. Note that, in this case, we are still not adding these results to the Arrow files (<code>addToArrow = FALSE</code>). We recommend checking the results of the integration thoroughly against your expectations prior to saving the results in the Arrow files. We illustrate this process in the next section of the book.</p>
<div class="sourceCode" id="cb157"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb157-1"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb157-1" tabindex="-1"></a><span class="co">#~5 minutes</span></span>
<span id="cb157-2"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb157-2" tabindex="-1"></a>projHeme2 <span class="ot">&lt;-</span> <span class="fu">addGeneIntegrationMatrix</span>(</span>
<span id="cb157-3"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb157-3" tabindex="-1"></a>    <span class="at">ArchRProj =</span> projHeme2, </span>
<span id="cb157-4"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb157-4" tabindex="-1"></a>    <span class="at">useMatrix =</span> <span class="st">&quot;GeneScoreMatrix&quot;</span>,</span>
<span id="cb157-5"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb157-5" tabindex="-1"></a>    <span class="at">matrixName =</span> <span class="st">&quot;GeneIntegrationMatrix&quot;</span>,</span>
<span id="cb157-6"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb157-6" tabindex="-1"></a>    <span class="at">reducedDims =</span> <span class="st">&quot;IterativeLSI&quot;</span>,</span>
<span id="cb157-7"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb157-7" tabindex="-1"></a>    <span class="at">seRNA =</span> seRNA,</span>
<span id="cb157-8"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb157-8" tabindex="-1"></a>    <span class="at">addToArrow =</span> <span class="cn">FALSE</span>, </span>
<span id="cb157-9"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb157-9" tabindex="-1"></a>    <span class="at">groupList =</span> groupList,</span>
<span id="cb157-10"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb157-10" tabindex="-1"></a>    <span class="at">groupRNA =</span> <span class="st">&quot;BioClassification&quot;</span>,</span>
<span id="cb157-11"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb157-11" tabindex="-1"></a>    <span class="at">nameCell =</span> <span class="st">&quot;predictedCell_Co&quot;</span>,</span>
<span id="cb157-12"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb157-12" tabindex="-1"></a>    <span class="at">nameGroup =</span> <span class="st">&quot;predictedGroup_Co&quot;</span>,</span>
<span id="cb157-13"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb157-13" tabindex="-1"></a>    <span class="at">nameScore =</span> <span class="st">&quot;predictedScore_Co&quot;</span></span>
<span id="cb157-14"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb157-14" tabindex="-1"></a>)</span>
<span id="cb157-15"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb157-15" tabindex="-1"></a><span class="do">## ArchR logging to : ArchRLogs/ArchR-addGeneIntegrationMatrix-1a17935ef4a-Date-2025-01-23_Time-20-24-20.231422.log</span></span>
<span id="cb157-16"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb157-16" tabindex="-1"></a><span class="do">## If there is an issue, please report to github with logFile!</span></span>
<span id="cb157-17"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb157-17" tabindex="-1"></a><span class="do">## 2025-01-23 20:24:20.412121 : Running Seurat&#39;s Integration Stuart* et al 2019, 0.003 mins elapsed.</span></span>
<span id="cb157-18"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb157-18" tabindex="-1"></a><span class="do">## 2025-01-23 20:24:20.949329 : Checking ATAC Input, 0.012 mins elapsed.</span></span>
<span id="cb157-19"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb157-19" tabindex="-1"></a><span class="do">## 2025-01-23 20:24:21.550167 : Checking RNA Input, 0.022 mins elapsed.</span></span>
<span id="cb157-20"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb157-20" tabindex="-1"></a><span class="do">## 2025-01-23 20:24:29.562689 : Found 18601 overlapping gene names from gene scores and rna matrix!, 0.156 mins elapsed.</span></span>
<span id="cb157-21"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb157-21" tabindex="-1"></a><span class="do">## 2025-01-23 20:24:29.56959 : Creating Integration Blocks, 0.156 mins elapsed.</span></span>
<span id="cb157-22"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb157-22" tabindex="-1"></a><span class="do">## 2025-01-23 20:24:29.703166 : Prepping Interation Data, 0.158 mins elapsed.</span></span>
<span id="cb157-23"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb157-23" tabindex="-1"></a><span class="do">## subThreading Disabled since ArchRLocking is TRUE see `addArchRLocking`</span></span>
<span id="cb157-24"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb157-24" tabindex="-1"></a><span class="do">## 2025-01-23 20:24:30.483271 : Computing Integration in 2 Integration Blocks!, 0 mins elapsed.</span></span>
<span id="cb157-25"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb157-25" tabindex="-1"></a><span class="do">## 2025-01-23 20:24:30.489394 : Block (1 of 2) : Computing Integration, 0 mins elapsed.</span></span>
<span id="cb157-26"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb157-26" tabindex="-1"></a><span class="do">## 2025-01-23 20:24:32.047863 : Block (1 of 2) : Identifying Variable Genes, 0.026 mins elapsed.</span></span>
<span id="cb157-27"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb157-27" tabindex="-1"></a><span class="do">## 2025-01-23 20:24:33.437231 : Block (1 of 2) : Getting GeneScoreMatrix, 0.049 mins elapsed.</span></span>
<span id="cb157-28"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb157-28" tabindex="-1"></a><span class="do">## 2025-01-23 20:24:41.794688 : Block (1 of 2) : Imputing GeneScoreMatrix, 0.189 mins elapsed.</span></span>
<span id="cb157-29"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb157-29" tabindex="-1"></a><span class="do">## Getting ImputeWeights</span></span>
<span id="cb157-30"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb157-30" tabindex="-1"></a><span class="do">## 2025-01-23 20:24:48.033978 : Block (1 of 2) : Seurat FindTransferAnchors, 0.293 mins elapsed.</span></span>
<span id="cb157-31"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb157-31" tabindex="-1"></a><span class="do">## 2025-01-23 20:26:16.492843 : Block (1 of 2) : Seurat TransferData Cell Group Labels, 1.767 mins elapsed.</span></span>
<span id="cb157-32"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb157-32" tabindex="-1"></a><span class="do">## 2025-01-23 20:26:35.146521 : Block (1 of 2) : Seurat TransferData Cell Names Labels, 2.078 mins elapsed.</span></span>
<span id="cb157-33"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb157-33" tabindex="-1"></a><span class="do">## 2025-01-23 20:27:00.798186 : Block (1 of 2) : Saving TransferAnchors Joint CCA, 2.505 mins elapsed.</span></span>
<span id="cb157-34"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb157-34" tabindex="-1"></a><span class="do">## 2025-01-23 20:27:03.144284 : Block (1 of 2) : Completed Integration, 2.544 mins elapsed.</span></span>
<span id="cb157-35"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb157-35" tabindex="-1"></a><span class="do">## 2025-01-23 20:27:05.178293 : Block (2 of 2) : Computing Integration, 2.578 mins elapsed.</span></span>
<span id="cb157-36"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb157-36" tabindex="-1"></a><span class="do">## 2025-01-23 20:27:06.914584 : Block (2 of 2) : Identifying Variable Genes, 2.607 mins elapsed.</span></span>
<span id="cb157-37"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb157-37" tabindex="-1"></a><span class="do">## 2025-01-23 20:27:08.455123 : Block (2 of 2) : Getting GeneScoreMatrix, 2.633 mins elapsed.</span></span>
<span id="cb157-38"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb157-38" tabindex="-1"></a><span class="do">## 2025-01-23 20:27:17.502101 : Block (2 of 2) : Imputing GeneScoreMatrix, 2.784 mins elapsed.</span></span>
<span id="cb157-39"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb157-39" tabindex="-1"></a><span class="do">## Getting ImputeWeights</span></span>
<span id="cb157-40"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb157-40" tabindex="-1"></a><span class="do">## 2025-01-23 20:27:39.386951 : Block (2 of 2) : Seurat FindTransferAnchors, 3.148 mins elapsed.</span></span>
<span id="cb157-41"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb157-41" tabindex="-1"></a><span class="do">## 2025-01-23 20:29:14.002704 : Block (2 of 2) : Seurat TransferData Cell Group Labels, 4.725 mins elapsed.</span></span>
<span id="cb157-42"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb157-42" tabindex="-1"></a><span class="do">## 2025-01-23 20:29:35.106172 : Block (2 of 2) : Seurat TransferData Cell Names Labels, 5.077 mins elapsed.</span></span>
<span id="cb157-43"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb157-43" tabindex="-1"></a><span class="do">## 2025-01-23 20:30:05.494531 : Block (2 of 2) : Saving TransferAnchors Joint CCA, 5.584 mins elapsed.</span></span>
<span id="cb157-44"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb157-44" tabindex="-1"></a><span class="do">## 2025-01-23 20:30:07.963272 : Block (2 of 2) : Completed Integration, 5.625 mins elapsed.</span></span>
<span id="cb157-45"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb157-45" tabindex="-1"></a><span class="do">## 2025-01-23 20:30:10.018843 : Block (1 of 2) : Plotting Joint UMAP, 5.659 mins elapsed.</span></span>
<span id="cb157-46"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb157-46" tabindex="-1"></a><span class="do">## 2025-01-23 20:30:26.290658 : Block (2 of 2) : Plotting Joint UMAP, 5.93 mins elapsed.</span></span>
<span id="cb157-47"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb157-47" tabindex="-1"></a><span class="do">## 2025-01-23 20:30:49.725359 : Completed Integration with RNA Matrix, 6.321 mins elapsed.</span></span>
<span id="cb157-48"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb157-48" tabindex="-1"></a><span class="do">## ArchR logging successful to : ArchRLogs/ArchR-addGeneIntegrationMatrix-1a17935ef4a-Date-2025-01-23_Time-20-24-20.231422.log</span></span></code></pre></div>
</div>
<div id="comparing-unconstrained-and-constrained-integrations" class="section level3 hasAnchor" number="10.1.3">
<h3><span class="header-section-number">10.1.3</span> Comparing Unconstrained and Constrained Integrations<a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#comparing-unconstrained-and-constrained-integrations" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>To compare our unconstrained and constrained integrations, we will color the cells in our scATAC-seq data based on the scRNA-seq cell type identified through integration. To do this, we will create a color palette using the built-in ArchR function <code>paletteDiscrete()</code>.</p>
<div class="sourceCode" id="cb158"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb158-1"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb158-1" tabindex="-1"></a>pal <span class="ot">&lt;-</span> <span class="fu">paletteDiscrete</span>(<span class="at">values =</span> <span class="fu">colData</span>(seRNA)<span class="sc">$</span>BioClassification)</span>
<span id="cb158-2"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb158-2" tabindex="-1"></a><span class="do">## Length of unique values greater than palette, interpolating..</span></span></code></pre></div>
<p>In ArchR, a palette is essentially a named vector where the values are hex codes corresponding to the color to be associated with the names.</p>
<div class="sourceCode" id="cb159"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb159-1"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb159-1" tabindex="-1"></a>pal</span>
<span id="cb159-2"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb159-2" tabindex="-1"></a><span class="do">##         01_HSC 02_Early.Eryth  03_Late.Eryth  04_Early.Baso    05_CMP.LMPP </span></span>
<span id="cb159-3"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb159-3" tabindex="-1"></a><span class="do">##      &quot;#D51F26&quot;      &quot;#502A59&quot;      &quot;#235D55&quot;      &quot;#3D6E57&quot;      &quot;#8D2B8B&quot; </span></span>
<span id="cb159-4"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb159-4" tabindex="-1"></a><span class="do">##       06_CLP.1         07_GMP    08_GMP.Neut         09_pDC         10_cDC </span></span>
<span id="cb159-5"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb159-5" tabindex="-1"></a><span class="do">##      &quot;#DE6C3E&quot;      &quot;#F9B712&quot;      &quot;#D8CE42&quot;      &quot;#8E9ACD&quot;      &quot;#B774B1&quot; </span></span>
<span id="cb159-6"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb159-6" tabindex="-1"></a><span class="do">## 11_CD14.Mono.1 12_CD14.Mono.2   13_CD16.Mono         14_Unk       15_CLP.2 </span></span>
<span id="cb159-7"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb159-7" tabindex="-1"></a><span class="do">##      &quot;#D69FC8&quot;      &quot;#C7C8DE&quot;      &quot;#8FD3D4&quot;      &quot;#89C86E&quot;      &quot;#CC9672&quot; </span></span>
<span id="cb159-8"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb159-8" tabindex="-1"></a><span class="do">##       16_Pre.B           17_B      18_Plasma       19_CD8.N      20_CD4.N1 </span></span>
<span id="cb159-9"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb159-9" tabindex="-1"></a><span class="do">##      &quot;#CF7E96&quot;      &quot;#A27AA4&quot;      &quot;#CD4F32&quot;      &quot;#6B977E&quot;      &quot;#518AA3&quot; </span></span>
<span id="cb159-10"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb159-10" tabindex="-1"></a><span class="do">##      21_CD4.N2       22_CD4.M      23_CD8.EM      24_CD8.CM          25_NK </span></span>
<span id="cb159-11"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb159-11" tabindex="-1"></a><span class="do">##      &quot;#5A5297&quot;      &quot;#0F707D&quot;      &quot;#5E2E32&quot;      &quot;#A95A3C&quot;      &quot;#B28D5C&quot; </span></span>
<span id="cb159-12"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb159-12" tabindex="-1"></a><span class="do">##         26_Unk </span></span>
<span id="cb159-13"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb159-13" tabindex="-1"></a><span class="do">##      &quot;#3D3D3D&quot;</span></span></code></pre></div>
<p>We can now visualize the integration by overlaying the scRNA-seq cell types on our scATAC-seq data based on the unconstrained integration.</p>
<div class="sourceCode" id="cb160"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb160-1"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb160-1" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fu">plotEmbedding</span>(</span>
<span id="cb160-2"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb160-2" tabindex="-1"></a>    projHeme2, </span>
<span id="cb160-3"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb160-3" tabindex="-1"></a>    <span class="at">colorBy =</span> <span class="st">&quot;cellColData&quot;</span>, </span>
<span id="cb160-4"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb160-4" tabindex="-1"></a>    <span class="at">name =</span> <span class="st">&quot;predictedGroup_Un&quot;</span>, </span>
<span id="cb160-5"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb160-5" tabindex="-1"></a>    <span class="at">pal =</span> pal</span>
<span id="cb160-6"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb160-6" tabindex="-1"></a>)</span>
<span id="cb160-7"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb160-7" tabindex="-1"></a><span class="do">## ArchR logging to : ArchRLogs/ArchR-plotEmbedding-1a14b444c24-Date-2025-01-23_Time-20-30-49.875203.log</span></span>
<span id="cb160-8"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb160-8" tabindex="-1"></a><span class="do">## If there is an issue, please report to github with logFile!</span></span>
<span id="cb160-9"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb160-9" tabindex="-1"></a><span class="do">## Getting UMAP Embedding</span></span>
<span id="cb160-10"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb160-10" tabindex="-1"></a><span class="do">## ColorBy = cellColData</span></span>
<span id="cb160-11"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb160-11" tabindex="-1"></a><span class="do">## Plotting Embedding</span></span>
<span id="cb160-12"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb160-12" tabindex="-1"></a><span class="do">## 1 </span></span>
<span id="cb160-13"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb160-13" tabindex="-1"></a><span class="do">## ArchR logging successful to : ArchRLogs/ArchR-plotEmbedding-1a14b444c24-Date-2025-01-23_Time-20-30-49.875203.log</span></span></code></pre></div>
<div class="sourceCode" id="cb161"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb161-1"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb161-1" tabindex="-1"></a>p1</span></code></pre></div>
<p><img src="ArchR_files/figure-html/unnamed-chunk-157-1.svg" width="672" /></p>
<p>Similarly, we can visualize the integration by overlaying the scRNA-seq cell types on our scATAC-seq data based on the constrained integration.</p>
<div class="sourceCode" id="cb162"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb162-1"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb162-1" tabindex="-1"></a>p2 <span class="ot">&lt;-</span> <span class="fu">plotEmbedding</span>(</span>
<span id="cb162-2"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb162-2" tabindex="-1"></a>    projHeme2, </span>
<span id="cb162-3"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb162-3" tabindex="-1"></a>    <span class="at">colorBy =</span> <span class="st">&quot;cellColData&quot;</span>, </span>
<span id="cb162-4"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb162-4" tabindex="-1"></a>    <span class="at">name =</span> <span class="st">&quot;predictedGroup_Co&quot;</span>, </span>
<span id="cb162-5"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb162-5" tabindex="-1"></a>    <span class="at">pal =</span> pal</span>
<span id="cb162-6"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb162-6" tabindex="-1"></a>)</span>
<span id="cb162-7"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb162-7" tabindex="-1"></a><span class="do">## ArchR logging to : ArchRLogs/ArchR-plotEmbedding-1a16d5099e8-Date-2025-01-23_Time-20-30-52.194744.log</span></span>
<span id="cb162-8"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb162-8" tabindex="-1"></a><span class="do">## If there is an issue, please report to github with logFile!</span></span>
<span id="cb162-9"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb162-9" tabindex="-1"></a><span class="do">## Getting UMAP Embedding</span></span>
<span id="cb162-10"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb162-10" tabindex="-1"></a><span class="do">## ColorBy = cellColData</span></span>
<span id="cb162-11"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb162-11" tabindex="-1"></a><span class="do">## Plotting Embedding</span></span>
<span id="cb162-12"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb162-12" tabindex="-1"></a><span class="do">## 1 </span></span>
<span id="cb162-13"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb162-13" tabindex="-1"></a><span class="do">## ArchR logging successful to : ArchRLogs/ArchR-plotEmbedding-1a16d5099e8-Date-2025-01-23_Time-20-30-52.194744.log</span></span></code></pre></div>
<div class="sourceCode" id="cb163"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb163-1"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb163-1" tabindex="-1"></a>p2</span></code></pre></div>
<p><img src="ArchR_files/figure-html/unnamed-chunk-159-1.svg" width="672" /></p>
<p>The differences between these the unconstrained and constrained integration is very subtle in this example, largely because the cell types of interest are already very distinct. However, you should notice differences, especially in the T cell clusters (Clusters 17-22).</p>
<p>To save an editable vectorized version of this plot, we use the <code>plotPDF()</code> function.</p>
<div class="sourceCode" id="cb164"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb164-1"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb164-1" tabindex="-1"></a><span class="fu">plotPDF</span>(p1,p2, <span class="at">name =</span> <span class="st">&quot;Plot-UMAP-RNA-Integration.pdf&quot;</span>, <span class="at">ArchRProj =</span> projHeme2, <span class="at">addDOC =</span> <span class="cn">FALSE</span>, <span class="at">width =</span> <span class="dv">5</span>, <span class="at">height =</span> <span class="dv">5</span>)</span>
<span id="cb164-2"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb164-2" tabindex="-1"></a><span class="do">## Plotting Ggplot!</span></span>
<span id="cb164-3"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb164-3" tabindex="-1"></a><span class="do">## Plotting Ggplot!</span></span></code></pre></div>
<p>We can now save our <code>projHeme2</code> using the <code>saveArchRProject()</code> function. Here, we set <code>load = TRUE</code> to make sure that the <code>projHeme2</code> object tracks with the project that we just saved.</p>
<div class="sourceCode" id="cb165"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb165-1"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb165-1" tabindex="-1"></a>projHeme2 <span class="ot">&lt;-</span> <span class="fu">saveArchRProject</span>(<span class="at">ArchRProj =</span> projHeme2, <span class="at">outputDirectory =</span> <span class="st">&quot;Save-ProjHeme2&quot;</span>, <span class="at">load =</span> <span class="cn">TRUE</span>)</span>
<span id="cb165-2"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb165-2" tabindex="-1"></a><span class="do">## Copying ArchRProject to new outputDirectory : /workspace/ArchR/ArchR_Website_Testing/bookdown/Save-ProjHeme2</span></span>
<span id="cb165-3"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb165-3" tabindex="-1"></a><span class="do">## Copying Arrow Files...</span></span>
<span id="cb165-4"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb165-4" tabindex="-1"></a><span class="do">## Copying Arrow Files (1 of 3)</span></span>
<span id="cb165-5"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb165-5" tabindex="-1"></a><span class="do">## Copying Arrow Files (2 of 3)</span></span>
<span id="cb165-6"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb165-6" tabindex="-1"></a><span class="do">## Copying Arrow Files (3 of 3)</span></span>
<span id="cb165-7"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb165-7" tabindex="-1"></a><span class="do">## Getting ImputeWeights</span></span>
<span id="cb165-8"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb165-8" tabindex="-1"></a><span class="do">## Dropping ImputeWeights...</span></span>
<span id="cb165-9"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb165-9" tabindex="-1"></a><span class="do">## Copying Other Files...</span></span>
<span id="cb165-10"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb165-10" tabindex="-1"></a><span class="do">## Copying Other Files (1 of 6): Embeddings</span></span>
<span id="cb165-11"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb165-11" tabindex="-1"></a><span class="do">## Copying Other Files (2 of 6): IterativeLSI</span></span>
<span id="cb165-12"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb165-12" tabindex="-1"></a><span class="do">## Copying Other Files (3 of 6): IterativeLSI2</span></span>
<span id="cb165-13"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb165-13" tabindex="-1"></a><span class="do">## Copying Other Files (4 of 6): IterativeLSI3</span></span>
<span id="cb165-14"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb165-14" tabindex="-1"></a><span class="do">## Copying Other Files (5 of 6): Plots</span></span>
<span id="cb165-15"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb165-15" tabindex="-1"></a><span class="do">## Copying Other Files (6 of 6): RNAIntegration</span></span>
<span id="cb165-16"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb165-16" tabindex="-1"></a><span class="do">## Saving ArchRProject...</span></span>
<span id="cb165-17"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb165-17" tabindex="-1"></a><span class="do">## Loading ArchRProject...</span></span>
<span id="cb165-18"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb165-18" tabindex="-1"></a><span class="do">## Successfully loaded ArchRProject!</span></span>
<span id="cb165-19"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb165-19" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb165-20"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb165-20" tabindex="-1"></a><span class="do">##                                                    / |</span></span>
<span id="cb165-21"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb165-21" tabindex="-1"></a><span class="do">##                                                  /    \</span></span>
<span id="cb165-22"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb165-22" tabindex="-1"></a><span class="do">##             .                                  /      |.</span></span>
<span id="cb165-23"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb165-23" tabindex="-1"></a><span class="do">##             \\\                              /        |.</span></span>
<span id="cb165-24"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb165-24" tabindex="-1"></a><span class="do">##               \\\                          /           `|.</span></span>
<span id="cb165-25"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb165-25" tabindex="-1"></a><span class="do">##                 \\\                      /              |.</span></span>
<span id="cb165-26"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb165-26" tabindex="-1"></a><span class="do">##                   \                    /                |\</span></span>
<span id="cb165-27"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb165-27" tabindex="-1"></a><span class="do">##                   \\#####\           /                  ||</span></span>
<span id="cb165-28"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb165-28" tabindex="-1"></a><span class="do">##                 ==###########&gt;      /                   ||</span></span>
<span id="cb165-29"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb165-29" tabindex="-1"></a><span class="do">##                  \\##==......\    /                     ||</span></span>
<span id="cb165-30"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb165-30" tabindex="-1"></a><span class="do">##             ______ =       =|__ /__                     ||      \\\</span></span>
<span id="cb165-31"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb165-31" tabindex="-1"></a><span class="do">##         ,--&#39; ,----`-,__ ___/&#39;  --,-`-===================##========&gt;</span></span>
<span id="cb165-32"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb165-32" tabindex="-1"></a><span class="do">##        \               &#39;        ##_______ _____ ,--,__,=##,__   ///</span></span>
<span id="cb165-33"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb165-33" tabindex="-1"></a><span class="do">##         ,    __==    ___,-,__,--&#39;#&#39;  ===&#39;      `-&#39;    | ##,-/</span></span>
<span id="cb165-34"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb165-34" tabindex="-1"></a><span class="do">##         -,____,---&#39;       \\####\\________________,--\\_##,/</span></span>
<span id="cb165-35"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb165-35" tabindex="-1"></a><span class="do">##            ___      .______        ______  __    __  .______      </span></span>
<span id="cb165-36"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb165-36" tabindex="-1"></a><span class="do">##           /   \     |   _  \      /      ||  |  |  | |   _  \     </span></span>
<span id="cb165-37"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb165-37" tabindex="-1"></a><span class="do">##          /  ^  \    |  |_)  |    |  ,----&#39;|  |__|  | |  |_)  |    </span></span>
<span id="cb165-38"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb165-38" tabindex="-1"></a><span class="do">##         /  /_\  \   |      /     |  |     |   __   | |      /     </span></span>
<span id="cb165-39"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb165-39" tabindex="-1"></a><span class="do">##        /  _____  \  |  |\  \\___ |  `----.|  |  |  | |  |\  \\___.</span></span>
<span id="cb165-40"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb165-40" tabindex="-1"></a><span class="do">##       /__/     \__\ | _| `._____| \______||__|  |__| | _| `._____|</span></span>
<span id="cb165-41"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html#cb165-41" tabindex="-1"></a><span class="do">## </span></span></code></pre></div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="defining-cluster-identity-with-scrna-seq.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="adding-pseudo-scrna-seq-profiles-for-each-scatac-seq-cell.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": null
},
"fontsettings": null,
"edit": null,
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "none"
},
"info": false
});
});
</script>

</body>

</html>
