<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>3.5 Getting Set Up | ArchR: Robust and scaleable analysis of single-cell chromatin accessibility data.</title>
  <meta name="description" content="A guide to ArchR" />
  <meta name="generator" content="bookdown 0.41 and GitBook 2.6.7" />

  <meta property="og:title" content="3.5 Getting Set Up | ArchR: Robust and scaleable analysis of single-cell chromatin accessibility data." />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="A guide to ArchR" />
  <meta name="github-repo" content="GreenleafLab/ArchR" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="3.5 Getting Set Up | ArchR: Robust and scaleable analysis of single-cell chromatin accessibility data." />
  
  <meta name="twitter:description" content="A guide to ArchR" />
  

<meta name="author" content="Jeffrey Granja and Ryan Corces" />


<meta name="date" content="2025-02-06" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="images/favicon.ico" type="image/x-icon" />
<link rel="prev" href="input-file-types-in-archr.html"/>
<link rel="next" href="creating-arrow-files.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<link href="libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="../index.html">Back to ArchR Home</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><img src="images/ArchR_Logo_Integrated.png" alt="" width="200"></a></li>
<li class="chapter" data-level="1" data-path="debugging-and-troubleshooting-in-archr.html"><a href="debugging-and-troubleshooting-in-archr.html"><i class="fa fa-check"></i><b>1</b> Debugging and Troubleshooting in ArchR</a></li>
<li class="chapter" data-level="2" data-path="manage-archrs-dependencies.html"><a href="manage-archrs-dependencies.html"><i class="fa fa-check"></i><b>2</b> Manage ArchR’s Dependencies</a>
<ul>
<li class="chapter" data-level="2.1" data-path="using-renv-to-manage-dependencies.html"><a href="using-renv-to-manage-dependencies.html"><i class="fa fa-check"></i><b>2.1</b> Using renv to manage dependencies</a></li>
<li class="chapter" data-level="2.2" data-path="other-environment-managers.html"><a href="other-environment-managers.html"><i class="fa fa-check"></i><b>2.2</b> Other environment managers</a></li>
<li class="chapter" data-level="2.3" data-path="using-docker-for-pre-compiled-environment-setup.html"><a href="using-docker-for-pre-compiled-environment-setup.html"><i class="fa fa-check"></i><b>2.3</b> Using Docker for pre-compiled environment setup</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="getting-started-with-archr.html"><a href="getting-started-with-archr.html"><i class="fa fa-check"></i><b>3</b> Getting Started with ArchR</a>
<ul>
<li class="chapter" data-level="3.1" data-path="a-brief-primer-on-atac-seq-terminology.html"><a href="a-brief-primer-on-atac-seq-terminology.html"><i class="fa fa-check"></i><b>3.1</b> A Brief Primer on ATAC-seq Terminology</a></li>
<li class="chapter" data-level="3.2" data-path="why-use-archr.html"><a href="why-use-archr.html"><i class="fa fa-check"></i><b>3.2</b> Why use ArchR?</a></li>
<li class="chapter" data-level="3.3" data-path="what-is-an-arrow-file-archrproject.html"><a href="what-is-an-arrow-file-archrproject.html"><i class="fa fa-check"></i><b>3.3</b> What is an Arrow file / <code>ArchRProject</code>?</a></li>
<li class="chapter" data-level="3.4" data-path="input-file-types-in-archr.html"><a href="input-file-types-in-archr.html"><i class="fa fa-check"></i><b>3.4</b> Input File Types in ArchR</a></li>
<li class="chapter" data-level="3.5" data-path="getting-set-up.html"><a href="getting-set-up.html"><i class="fa fa-check"></i><b>3.5</b> Getting Set Up</a></li>
<li class="chapter" data-level="3.6" data-path="creating-arrow-files.html"><a href="creating-arrow-files.html"><i class="fa fa-check"></i><b>3.6</b> Creating Arrow Files</a></li>
<li class="chapter" data-level="3.7" data-path="per-cell-quality-control.html"><a href="per-cell-quality-control.html"><i class="fa fa-check"></i><b>3.7</b> Per-cell Quality Control</a></li>
<li class="chapter" data-level="3.8" data-path="using-bam-files-for-arrow-file-creation.html"><a href="using-bam-files-for-arrow-file-creation.html"><i class="fa fa-check"></i><b>3.8</b> Using BAM files for Arrow File creation</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="doublet-inference-with-archr.html"><a href="doublet-inference-with-archr.html"><i class="fa fa-check"></i><b>4</b> Doublet Inference with ArchR</a>
<ul>
<li class="chapter" data-level="4.1" data-path="how-does-doublet-identification-work-in-archr.html"><a href="how-does-doublet-identification-work-in-archr.html"><i class="fa fa-check"></i><b>4.1</b> How does doublet identification work in ArchR?</a></li>
<li class="chapter" data-level="4.2" data-path="inferring-scatac-seq-doublets-with-archr.html"><a href="inferring-scatac-seq-doublets-with-archr.html"><i class="fa fa-check"></i><b>4.2</b> Inferring scATAC-seq Doublets with ArchR</a></li>
<li class="chapter" data-level="4.3" data-path="using-demuxlet-with-archr.html"><a href="using-demuxlet-with-archr.html"><i class="fa fa-check"></i><b>4.3</b> Using <em>demuxlet</em> with ArchR</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="creating-an-archrproject.html"><a href="creating-an-archrproject.html"><i class="fa fa-check"></i><b>5</b> Creating an ArchRProject</a>
<ul>
<li class="chapter" data-level="5.1" data-path="creating-an-archrproject-1.html"><a href="creating-an-archrproject-1.html"><i class="fa fa-check"></i><b>5.1</b> Creating An ArchRProject</a></li>
<li class="chapter" data-level="5.2" data-path="manipulating-an-archrproject.html"><a href="manipulating-an-archrproject.html"><i class="fa fa-check"></i><b>5.2</b> Manipulating An ArchRProject</a></li>
<li class="chapter" data-level="5.3" data-path="plotting-sample-statistics-from-an-archrproject.html"><a href="plotting-sample-statistics-from-an-archrproject.html"><i class="fa fa-check"></i><b>5.3</b> Plotting Sample Statistics from an ArchRProject</a></li>
<li class="chapter" data-level="5.4" data-path="plotting-sample-fragment-size-distribution-and-tss-enrichment-profiles..html"><a href="plotting-sample-fragment-size-distribution-and-tss-enrichment-profiles..html"><i class="fa fa-check"></i><b>5.4</b> Plotting Sample Fragment Size Distribution and TSS Enrichment Profiles.</a></li>
<li class="chapter" data-level="5.5" data-path="saving-and-loading-an-archrproject.html"><a href="saving-and-loading-an-archrproject.html"><i class="fa fa-check"></i><b>5.5</b> Saving and Loading an <code>ArchRProject</code></a></li>
<li class="chapter" data-level="5.6" data-path="filtering-doublets-from-an-archrproject.html"><a href="filtering-doublets-from-an-archrproject.html"><i class="fa fa-check"></i><b>5.6</b> Filtering Doublets from an ArchRProject</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="dimensionality-reduction-with-archr.html"><a href="dimensionality-reduction-with-archr.html"><i class="fa fa-check"></i><b>6</b> Dimensionality Reduction with ArchR</a>
<ul>
<li class="chapter" data-level="6.1" data-path="archrs-lsi-implementation.html"><a href="archrs-lsi-implementation.html"><i class="fa fa-check"></i><b>6.1</b> ArchR’s LSI Implementation</a></li>
<li class="chapter" data-level="6.2" data-path="iterative-latent-semantic-indexing-lsi.html"><a href="iterative-latent-semantic-indexing-lsi.html"><i class="fa fa-check"></i><b>6.2</b> Iterative Latent Semantic Indexing (LSI)</a></li>
<li class="chapter" data-level="6.3" data-path="estimated-lsi.html"><a href="estimated-lsi.html"><i class="fa fa-check"></i><b>6.3</b> Estimated LSI</a></li>
<li class="chapter" data-level="6.4" data-path="batch-effect-correction-wtih-harmony.html"><a href="batch-effect-correction-wtih-harmony.html"><i class="fa fa-check"></i><b>6.4</b> Batch Effect Correction wtih Harmony</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="clustering-with-archr.html"><a href="clustering-with-archr.html"><i class="fa fa-check"></i><b>7</b> Clustering with ArchR</a>
<ul>
<li class="chapter" data-level="7.1" data-path="clustering-using-seurats-findclusters-function.html"><a href="clustering-using-seurats-findclusters-function.html"><i class="fa fa-check"></i><b>7.1</b> Clustering using Seurat’s <code>FindClusters()</code> function</a></li>
<li class="chapter" data-level="7.2" data-path="clustering-using-scran.html"><a href="clustering-using-scran.html"><i class="fa fa-check"></i><b>7.2</b> Clustering using <code>scran</code></a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="single-cell-embeddings.html"><a href="single-cell-embeddings.html"><i class="fa fa-check"></i><b>8</b> Single-cell Embeddings</a>
<ul>
<li class="chapter" data-level="8.1" data-path="uniform-manifold-approximation-and-projection-umap.html"><a href="uniform-manifold-approximation-and-projection-umap.html"><i class="fa fa-check"></i><b>8.1</b> Uniform Manifold Approximation and Projection (UMAP)</a></li>
<li class="chapter" data-level="8.2" data-path="t-stocastic-neighbor-embedding-t-sne.html"><a href="t-stocastic-neighbor-embedding-t-sne.html"><i class="fa fa-check"></i><b>8.2</b> t-Stocastic Neighbor Embedding (t-SNE)</a></li>
<li class="chapter" data-level="8.3" data-path="dimensionality-reduction-after-harmony.html"><a href="dimensionality-reduction-after-harmony.html"><i class="fa fa-check"></i><b>8.3</b> Dimensionality Reduction After Harmony</a></li>
<li class="chapter" data-level="8.4" data-path="highlighting-specific-cells-on-an-embedding.html"><a href="highlighting-specific-cells-on-an-embedding.html"><i class="fa fa-check"></i><b>8.4</b> Highlighting specific cells on an embedding</a></li>
<li class="chapter" data-level="8.5" data-path="importing-an-embedding-from-external-software.html"><a href="importing-an-embedding-from-external-software.html"><i class="fa fa-check"></i><b>8.5</b> Importing an embedding from external software</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="gene-scores-and-marker-genes-with-archr.html"><a href="gene-scores-and-marker-genes-with-archr.html"><i class="fa fa-check"></i><b>9</b> Gene Scores and Marker Genes with ArchR</a>
<ul>
<li class="chapter" data-level="9.1" data-path="calculating-gene-scores-in-archr.html"><a href="calculating-gene-scores-in-archr.html"><i class="fa fa-check"></i><b>9.1</b> Calculating Gene Scores in ArchR</a></li>
<li class="chapter" data-level="9.2" data-path="identification-of-marker-features.html"><a href="identification-of-marker-features.html"><i class="fa fa-check"></i><b>9.2</b> Identification of Marker Features</a></li>
<li class="chapter" data-level="9.3" data-path="identifying-marker-genes.html"><a href="identifying-marker-genes.html"><i class="fa fa-check"></i><b>9.3</b> Identifying Marker Genes</a></li>
<li class="chapter" data-level="9.4" data-path="visualizing-marker-genes-on-an-embedding.html"><a href="visualizing-marker-genes-on-an-embedding.html"><i class="fa fa-check"></i><b>9.4</b> Visualizing Marker Genes on an Embedding</a></li>
<li class="chapter" data-level="9.5" data-path="marker-genes-imputation-with-magic.html"><a href="marker-genes-imputation-with-magic.html"><i class="fa fa-check"></i><b>9.5</b> Marker Genes Imputation with MAGIC</a></li>
<li class="chapter" data-level="9.6" data-path="module-scores.html"><a href="module-scores.html"><i class="fa fa-check"></i><b>9.6</b> Module Scores</a></li>
<li class="chapter" data-level="9.7" data-path="track-plotting-with-archrbrowser.html"><a href="track-plotting-with-archrbrowser.html"><i class="fa fa-check"></i><b>9.7</b> Track Plotting with ArchRBrowser</a></li>
<li class="chapter" data-level="9.8" data-path="launching-the-interactive-archrbrowser.html"><a href="launching-the-interactive-archrbrowser.html"><i class="fa fa-check"></i><b>9.8</b> Launching the Interactive ArchRBrowser</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="defining-cluster-identity-with-scrna-seq.html"><a href="defining-cluster-identity-with-scrna-seq.html"><i class="fa fa-check"></i><b>10</b> Defining Cluster Identity with scRNA-seq</a>
<ul>
<li class="chapter" data-level="10.1" data-path="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html"><i class="fa fa-check"></i><b>10.1</b> Cross-platform linkage of scATAC-seq cells with scRNA-seq cells</a></li>
<li class="chapter" data-level="10.2" data-path="adding-pseudo-scrna-seq-profiles-for-each-scatac-seq-cell.html"><a href="adding-pseudo-scrna-seq-profiles-for-each-scatac-seq-cell.html"><i class="fa fa-check"></i><b>10.2</b> Adding Pseudo-scRNA-seq profiles for each scATAC-seq cell</a></li>
<li class="chapter" data-level="10.3" data-path="labeling-scatac-seq-clusters-with-scrna-seq-information.html"><a href="labeling-scatac-seq-clusters-with-scrna-seq-information.html"><i class="fa fa-check"></i><b>10.3</b> Labeling scATAC-seq clusters with scRNA-seq information</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="pseudo-bulk-replicates-in-archr.html"><a href="pseudo-bulk-replicates-in-archr.html"><i class="fa fa-check"></i><b>11</b> Pseudo-bulk Replicates in ArchR</a>
<ul>
<li class="chapter" data-level="11.1" data-path="how-does-archr-make-pseudo-bulk-replicates.html"><a href="how-does-archr-make-pseudo-bulk-replicates.html"><i class="fa fa-check"></i><b>11.1</b> How Does ArchR Make Pseudo-bulk Replicates?</a></li>
<li class="chapter" data-level="11.2" data-path="making-pseudo-bulk-replicates.html"><a href="making-pseudo-bulk-replicates.html"><i class="fa fa-check"></i><b>11.2</b> Making Pseudo-bulk Replicates</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="calling-peaks-with-archr.html"><a href="calling-peaks-with-archr.html"><i class="fa fa-check"></i><b>12</b> Calling Peaks with ArchR</a>
<ul>
<li class="chapter" data-level="12.1" data-path="the-iterative-overlap-peak-merging-procedure.html"><a href="the-iterative-overlap-peak-merging-procedure.html"><i class="fa fa-check"></i><b>12.1</b> The Iterative Overlap Peak Merging Procedure</a></li>
<li class="chapter" data-level="12.2" data-path="calling-peaks-w-macs2.html"><a href="calling-peaks-w-macs2.html"><i class="fa fa-check"></i><b>12.2</b> Calling Peaks w/ Macs2</a></li>
<li class="chapter" data-level="12.3" data-path="calling-peaks-w-tilematrix.html"><a href="calling-peaks-w-tilematrix.html"><i class="fa fa-check"></i><b>12.3</b> Calling Peaks w/ TileMatrix</a></li>
<li class="chapter" data-level="12.4" data-path="adding-a-peak-matrix.html"><a href="adding-a-peak-matrix.html"><i class="fa fa-check"></i><b>12.4</b> Adding a Peak Matrix</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="identifying-marker-peaks.html"><a href="identifying-marker-peaks.html"><i class="fa fa-check"></i><b>13</b> Identifying Marker Peaks</a>
<ul>
<li class="chapter" data-level="13.1" data-path="identifying-marker-peaks-with-archr.html"><a href="identifying-marker-peaks-with-archr.html"><i class="fa fa-check"></i><b>13.1</b> Identifying Marker Peaks with ArchR</a></li>
<li class="chapter" data-level="13.2" data-path="plotting-marker-peaks-in-archr.html"><a href="plotting-marker-peaks-in-archr.html"><i class="fa fa-check"></i><b>13.2</b> Plotting Marker Peaks in ArchR</a></li>
<li class="chapter" data-level="13.3" data-path="pairwise-testing-between-groups.html"><a href="pairwise-testing-between-groups.html"><i class="fa fa-check"></i><b>13.3</b> Pairwise Testing Between Groups</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="motif-and-feature-enrichment-with-archr.html"><a href="motif-and-feature-enrichment-with-archr.html"><i class="fa fa-check"></i><b>14</b> Motif and Feature Enrichment with ArchR</a>
<ul>
<li class="chapter" data-level="14.1" data-path="motif-enrichment.html"><a href="motif-enrichment.html"><i class="fa fa-check"></i><b>14.1</b> Motif Enrichment</a></li>
<li class="chapter" data-level="14.2" data-path="archr-enrichment.html"><a href="archr-enrichment.html"><i class="fa fa-check"></i><b>14.2</b> ArchR Enrichment</a></li>
<li class="chapter" data-level="14.3" data-path="custom-enrichment.html"><a href="custom-enrichment.html"><i class="fa fa-check"></i><b>14.3</b> Custom Enrichment</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="chromvar-deviatons-enrichment-with-archr.html"><a href="chromvar-deviatons-enrichment-with-archr.html"><i class="fa fa-check"></i><b>15</b> ChromVAR Deviatons Enrichment with ArchR</a>
<ul>
<li class="chapter" data-level="15.1" data-path="motif-deviations.html"><a href="motif-deviations.html"><i class="fa fa-check"></i><b>15.1</b> Motif Deviations</a></li>
<li class="chapter" data-level="15.2" data-path="archr-and-custom-deviations.html"><a href="archr-and-custom-deviations.html"><i class="fa fa-check"></i><b>15.2</b> ArchR and Custom Deviations</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="footprinting-with-archr.html"><a href="footprinting-with-archr.html"><i class="fa fa-check"></i><b>16</b> Footprinting with ArchR</a>
<ul>
<li class="chapter" data-level="16.1" data-path="motif-footprinting.html"><a href="motif-footprinting.html"><i class="fa fa-check"></i><b>16.1</b> Motif Footprinting</a></li>
<li class="chapter" data-level="16.2" data-path="normalization-of-footprints-for-tn5-bias.html"><a href="normalization-of-footprints-for-tn5-bias.html"><i class="fa fa-check"></i><b>16.2</b> Normalization of Footprints for Tn5 Bias</a></li>
<li class="chapter" data-level="16.3" data-path="feature-footprinting.html"><a href="feature-footprinting.html"><i class="fa fa-check"></i><b>16.3</b> Feature Footprinting</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="integrative-analysis-with-archr.html"><a href="integrative-analysis-with-archr.html"><i class="fa fa-check"></i><b>17</b> Integrative Analysis with ArchR</a>
<ul>
<li class="chapter" data-level="17.1" data-path="creating-low-overlapping-aggregates-of-cells.html"><a href="creating-low-overlapping-aggregates-of-cells.html"><i class="fa fa-check"></i><b>17.1</b> Creating Low-Overlapping Aggregates of Cells</a></li>
<li class="chapter" data-level="17.2" data-path="co-accessibility-with-archr.html"><a href="co-accessibility-with-archr.html"><i class="fa fa-check"></i><b>17.2</b> Co-accessibility with ArchR</a></li>
<li class="chapter" data-level="17.3" data-path="peak2genelinkage-with-archr.html"><a href="peak2genelinkage-with-archr.html"><i class="fa fa-check"></i><b>17.3</b> Peak2GeneLinkage with ArchR</a></li>
<li class="chapter" data-level="17.4" data-path="identification-of-positive-tf-regulators.html"><a href="identification-of-positive-tf-regulators.html"><i class="fa fa-check"></i><b>17.4</b> Identification of Positive TF-Regulators</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="trajectory-analysis-with-archr.html"><a href="trajectory-analysis-with-archr.html"><i class="fa fa-check"></i><b>18</b> Trajectory Analysis with ArchR</a>
<ul>
<li class="chapter" data-level="18.1" data-path="myeloid-trajectory---monocyte-differentiation.html"><a href="myeloid-trajectory---monocyte-differentiation.html"><i class="fa fa-check"></i><b>18.1</b> Myeloid Trajectory - Monocyte Differentiation</a></li>
<li class="chapter" data-level="18.2" data-path="lymphoid-trajectory---b-cell-differentiation.html"><a href="lymphoid-trajectory---b-cell-differentiation.html"><i class="fa fa-check"></i><b>18.2</b> Lymphoid Trajectory - B Cell Differentiation</a></li>
<li class="chapter" data-level="18.3" data-path="monocle3-trajectories.html"><a href="monocle3-trajectories.html"><i class="fa fa-check"></i><b>18.3</b> Monocle3 Trajectories</a></li>
<li class="chapter" data-level="18.4" data-path="slingshot-trajectories.html"><a href="slingshot-trajectories.html"><i class="fa fa-check"></i><b>18.4</b> Slingshot Trajectories</a></li>
<li class="chapter" data-level="18.5" data-path="comparing-the-different-trajectories.html"><a href="comparing-the-different-trajectories.html"><i class="fa fa-check"></i><b>18.5</b> Comparing the different trajectories</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="handling-archr-output.html"><a href="handling-archr-output.html"><i class="fa fa-check"></i><b>19</b> Handling ArchR Output</a>
<ul>
<li class="chapter" data-level="19.1" data-path="exporting-fragment-level-data.html"><a href="exporting-fragment-level-data.html"><i class="fa fa-check"></i><b>19.1</b> Exporting fragment-level data</a></li>
<li class="chapter" data-level="19.2" data-path="exporting-matrix-level-data.html"><a href="exporting-matrix-level-data.html"><i class="fa fa-check"></i><b>19.2</b> Exporting matrix-level data</a></li>
<li class="chapter" data-level="19.3" data-path="exporting-pseudo-bulked-data-to-a-summarizedexperiment.html"><a href="exporting-pseudo-bulked-data-to-a-summarizedexperiment.html"><i class="fa fa-check"></i><b>19.3</b> Exporting pseudo-bulked data to a <code>SummarizedExperiment</code></a></li>
<li class="chapter" data-level="19.4" data-path="exporting-pseudo-bulked-data-to-a-bigwig-file.html"><a href="exporting-pseudo-bulked-data-to-a-bigwig-file.html"><i class="fa fa-check"></i><b>19.4</b> Exporting pseudo-bulked data to a bigWig file</a></li>
</ul></li>
<li class="chapter" data-level="20" data-path="integration-with-bulk-atac-seq.html"><a href="integration-with-bulk-atac-seq.html"><i class="fa fa-check"></i><b>20</b> Integration with bulk ATAC-seq</a>
<ul>
<li class="chapter" data-level="20.1" data-path="projecting-bulk-atac-seq-data.html"><a href="projecting-bulk-atac-seq-data.html"><i class="fa fa-check"></i><b>20.1</b> Projecting bulk ATAC-seq data</a></li>
</ul></li>
<li class="chapter" data-level="21" data-path="multiomic-data-analysis-in-archr.html"><a href="multiomic-data-analysis-in-archr.html"><i class="fa fa-check"></i><b>21</b> Multiomic data analysis in ArchR</a>
<ul>
<li class="chapter" data-level="21.1" data-path="importing-data-and-setting-up-a-multiome-project.html"><a href="importing-data-and-setting-up-a-multiome-project.html"><i class="fa fa-check"></i><b>21.1</b> Importing data and setting up a Multiome project</a></li>
<li class="chapter" data-level="21.2" data-path="analysis-of-multiome-data-in-archr.html"><a href="analysis-of-multiome-data-in-archr.html"><i class="fa fa-check"></i><b>21.2</b> Analysis of multiome data in ArchR</a></li>
</ul></li>
<li class="chapter" data-level="22" data-path="plot-aesthetics-in-archr.html"><a href="plot-aesthetics-in-archr.html"><i class="fa fa-check"></i><b>22</b> Plot aesthetics in ArchR</a>
<ul>
<li class="chapter" data-level="22.1" data-path="using-and-manipulating-palettes-in-archr.html"><a href="using-and-manipulating-palettes-in-archr.html"><i class="fa fa-check"></i><b>22.1</b> Using and manipulating palettes in ArchR</a></li>
</ul></li>
<li class="chapter" data-level="23" data-path="testing-your-archr-installation.html"><a href="testing-your-archr-installation.html"><i class="fa fa-check"></i><b>23</b> Testing Your ArchR Installation</a></li>
<li class="chapter" data-level="24" data-path="session-information.html"><a href="session-information.html"><i class="fa fa-check"></i><b>24</b> Session Information</a></li>
<li class="divider"></li>
<li><a href="https://bookdown.org" target="_blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">ArchR: Robust and scaleable analysis of single-cell chromatin accessibility data.</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="getting-set-up" class="section level2 hasAnchor" number="3.5">
<h2><span class="header-section-number">3.5</span> Getting Set Up<a href="getting-set-up.html#getting-set-up" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The first thing we do is change to our desired our working directory, set the number of threads we would like to use, and load our gene and genome annotations. Depending on the configuration of your local environment, you may need to modify the number of <code>threads</code> used below in <code>addArchRThreads()</code>. By default ArchR uses half of the total number of <code>threads</code> available but you can adjust this manually as you see fit. As a reminder, ArchR is not supported on Windows.</p>
<p>For the purposes of this tutorial, we are assuming that you followed the instructions in the chapter on handling ArchR’s dependencies using <code>renv</code> and that you have set up a working directory for running this tutorial.</p>
<p>First, we load the ArchR library. If this fails, you have not properly installed ArchR and should <a href="https://www.archrproject.com/index.html">revisit the installation instructions</a>. We will also set our seed for randomized opperations for consistency. You can set your seed to whatever you would like (here we use <code>1</code>), but we encourage you to be consistent and always set the same seed.</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="getting-set-up.html#cb16-1" tabindex="-1"></a><span class="fu">library</span>(ArchR)</span>
<span id="cb16-2"><a href="getting-set-up.html#cb16-2" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">1</span>)</span></code></pre></div>
<p>Next, we either enable or disable file locking for HDF5 files. File locking allows for processes to take advantage of subthreading but may not work on certain network-based file systems, such as network-based lustre file systems, that are often present on high-performance computing clusters. This has caused a lot of problems for specific users, in the past (perhaps best summarized <a href="https://github.com/GreenleafLab/ArchR/issues/248#issuecomment-789453997">here</a>). Thus, it is extremely important to consistently enable or disable file locking based on your particular computational environment. Enabling file locking will prevent subthreading, disabling file locking will permit it. Regardless, whatever you chose for file locking is controlled by environment variables and only persists during the current <code>R</code> session so you should include a call to <code>addArchRFileLocking()</code> in the beginning of each new R session (just like <code>set.seed()</code> or <code>addArchRThreads()</code> etc.). The filesystem used to generate this tutorial works best with <code>locking = TRUE</code> so that is what we set below but you should use whatever you think is best for your system. If you have trouble with Arrow File generation on the tutorial data or with various HDF5 file accessibility errors, try setting <code>locking = TRUE</code> at the beginning of your project (before Arrow File generation).</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="getting-set-up.html#cb17-1" tabindex="-1"></a><span class="fu">addArchRLocking</span>(<span class="at">locking =</span> <span class="cn">TRUE</span>)</span>
<span id="cb17-2"><a href="getting-set-up.html#cb17-2" tabindex="-1"></a><span class="do">## Setting ArchRLocking to TRUE.</span></span></code></pre></div>
<p>Next, we set the default number of threads for ArchR functions. This is something you will have to do during each new R session. We recommend setting <code>threads</code> to 1/2 to 3/4 of the total available cores. The memory usage in ArchR will often scale with the number of threads used so allowing ArchR to use more threads will also lead to higher memory usage.</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="getting-set-up.html#cb18-1" tabindex="-1"></a><span class="fu">addArchRThreads</span>(<span class="at">threads =</span> <span class="dv">1</span>) </span>
<span id="cb18-2"><a href="getting-set-up.html#cb18-2" tabindex="-1"></a><span class="do">## Setting default number of Parallel threads to 1.</span></span></code></pre></div>
<p>Then, we set the genome to be used for gene and genome annotations. As above, this is something you will have to do during each new R session. Of course, this genome version must match the genome version that was used for alignment. For the data used in this tutorial, we will use the hg19 reference genome but ArchR natively supports additional genome annotations and custome genome annotations as outlined in the next section.</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="getting-set-up.html#cb19-1" tabindex="-1"></a><span class="fu">addArchRGenome</span>(<span class="st">&quot;hg19&quot;</span>)</span>
<span id="cb19-2"><a href="getting-set-up.html#cb19-2" tabindex="-1"></a><span class="do">## Setting default genome to Hg19.</span></span></code></pre></div>
<p>ArchR requires gene and genome annotations to do things such as calculate TSS enrichment scores, nucleotide content, and gene activity scores. Because our tutorial dataset uses scATAC-seq data that has already been aligned to the hg19 reference genome, we have set “hg19” as the default genome in the previous section. However, ArchR supports “hg19”, “hg38”, “mm9”, and “mm10” natively and you can create your own genome and gene annotations using the <code>createGeneAnnotation()</code> and <code>createGenomeAnnotation()</code> functions.</p>
<p>Providing this information to ArchR is streamlined through the <code>addArchRGenome()</code> function. This function tells ArchR that, for all analyses in the current session, it should use the <code>genomeAnnotation</code> and <code>geneAnnotation</code> associated with the defined <code>ArchRGenome</code>. Each of the natively supported genomes are composed of a <code>BSgenome</code> object that defines the genomic coordinates and sequence of each chromosome, a <code>GRanges</code> object containing a set of blacklisted regions, a <code>TxDb</code> object that defines the positions and structures of all genes, and an <code>OrgDb</code> object that provides a central gene identifier and contains mappings between this identifier and other kinds of identifiers.</p>
<p>Below are examples of how to load gene and genome annotations for the natively supported genomes as well as information on their <code>BSgenome</code> and blacklist components.</p>
<hr>
<p>The precompiled version of the <strong>hg19</strong> genome in ArchR uses <code>BSgenome.Hsapiens.UCSC.hg19</code>, <code>TxDb.Hsapiens.UCSC.hg19.knownGene</code>, <code>org.Hs.eg.db</code>, and a blacklist that was merged using <code>ArchR::mergeGR()</code> from the <a href="https://github.com/Boyle-Lab/Blacklist/blob/master/lists/hg19-blacklist.v2.bed.gz">hg19 v2 blacklist regions</a> and from <a href="https://github.com/caleblareau/mitoblacklist/blob/master/peaks/hg19_peaks.narrowPeak">mitochondrial regions that show high mappability to the hg19 nuclear genome</a> from Caleb Lareau and Jason Buenrostro. To set a global genome default to the precompiled hg19 genome:</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="getting-set-up.html#cb20-1" tabindex="-1"></a><span class="fu">addArchRGenome</span>(<span class="st">&quot;hg19&quot;</span>)</span></code></pre></div>
<hr>
<p>The precompiled version of the <strong>hg38</strong> genome in ArchR uses <code>BSgenome.Hsapiens.UCSC.hg38</code>, <code>TxDb.Hsapiens.UCSC.hg38.knownGene</code>, <code>org.Hs.eg.db</code>, and a blacklist that was merged using <code>ArchR::mergeGR()</code> from the <a href="https://github.com/Boyle-Lab/Blacklist/blob/master/lists/hg38-blacklist.v2.bed.gz">hg38 v2 blacklist regions</a> and from <a href="https://github.com/caleblareau/mitoblacklist/blob/master/peaks/hg38_peaks.narrowPeak">mitochondrial regions that show high mappability to the hg38 nuclear genome</a> from Caleb Lareau and Jason Buenrostro. To set a global genome default to the precompiled hg38 genome:</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="getting-set-up.html#cb21-1" tabindex="-1"></a><span class="fu">addArchRGenome</span>(<span class="st">&quot;hg38&quot;</span>)</span></code></pre></div>
<hr>
<p>The precompiled version of the <strong>mm9</strong> genome in ArchR uses <code>BSgenome.Mmusculus.UCSC.mm9</code>, <code>TxDb.Mmusculus.UCSC.mm9.knownGene</code>, <code>org.Mm.eg.db</code>, and a blacklist that was merged using <code>ArchR::mergeGR()</code> from the <a href="http://mitra.stanford.edu/kundaje/akundaje/release/blacklists/mm9-mouse/mm9-blacklist.bed.gz">mm9 v1 blacklist regions</a> from Anshul Kundaje and from <a href="https://github.com/caleblareau/mitoblacklist/blob/master/peaks/mm9_peaks.narrowPeak">mitochondrial regions that show high mappability to the mm9 nuclear genome</a> from Caleb Lareau and Jason Buenrostro. To set a global genome default to the precompiled mm9 genome:</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="getting-set-up.html#cb22-1" tabindex="-1"></a><span class="fu">addArchRGenome</span>(<span class="st">&quot;mm9&quot;</span>)</span></code></pre></div>
<hr>
<p>The precompiled version of the <strong>mm10</strong> genome in ArchR uses <code>BSgenome.Mmusculus.UCSC.mm10</code>, <code>TxDb.Mmusculus.UCSC.mm10.knownGene</code>, <code>org.Mm.eg.db</code>, and a blacklist that was merged using <code>ArchR::mergeGR()</code> from the <a href="https://github.com/Boyle-Lab/Blacklist/blob/master/lists/mm10-blacklist.v2.bed.gz">mm10 v2 blacklist regions</a> and from <a href="https://github.com/caleblareau/mitoblacklist/blob/master/peaks/mm10_peaks.narrowPeak">mitochondrial regions that show high mappability to the mm10 nuclear genome</a> from Caleb Lareau and Jason Buenrostro. To set a global genome default to the precompiled mm10 genome:</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="getting-set-up.html#cb23-1" tabindex="-1"></a><span class="fu">addArchRGenome</span>(<span class="st">&quot;mm10&quot;</span>)</span></code></pre></div>
<hr>
<div id="creating-a-custom-archrgenome" class="section level3 hasAnchor" number="3.5.1">
<h3><span class="header-section-number">3.5.1</span> Creating a Custom <code>ArchRGenome</code><a href="getting-set-up.html#creating-a-custom-archrgenome" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>As described above, an <code>ArchRGenome</code> consists of a genome annotation and a gene annotation. If you create your own genome and gene annotations, then you would pass these directly to <code>createArrowFiles()</code> using the <code>genomeAnnotation</code> and <code>geneAnnotation</code> parameters, rather than calling <code>addArchRGenome()</code>.</p>
<p>To create a <strong>custom genome annotation</strong>, we use <code>createGenomeAnnotation()</code>. To do this, you will need the following information:</p>
<ol style="list-style-type: decimal">
<li>A <code>BSgenome</code> object which contains the sequence information for a genome. These are commonly Bioconductor packages (for example, <code>BSgenome.Hsapiens.UCSC.hg38</code>) that can be easily found with google.</li>
<li>A <code>GRanges</code> genomic ranges object containing a set of blacklisted regions that will be used to filter out unwanted regions from downstream analysis. This is not required but is recommended. For information on how blacklists are created, see the <a href="https://www.ncbi.nlm.nih.gov/pubmed/31249361">publication on the ENCODE blacklists</a>.</li>
</ol>
<p>For example, if we wanted to create a custom genome annotation from Drosophila melanogaster, we would first identify and install and load the relevant <code>BSgenome</code> object.</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="getting-set-up.html#cb24-1" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">&quot;BSgenome.Dmelanogaster.UCSC.dm6&quot;</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>)){</span>
<span id="cb24-2"><a href="getting-set-up.html#cb24-2" tabindex="-1"></a>  BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">&quot;BSgenome.Dmelanogaster.UCSC.dm6&quot;</span>, <span class="at">update =</span> <span class="cn">FALSE</span>)</span>
<span id="cb24-3"><a href="getting-set-up.html#cb24-3" tabindex="-1"></a>}</span>
<span id="cb24-4"><a href="getting-set-up.html#cb24-4" tabindex="-1"></a><span class="fu">library</span>(BSgenome.Dmelanogaster.UCSC.dm6)</span>
<span id="cb24-5"><a href="getting-set-up.html#cb24-5" tabindex="-1"></a><span class="do">## Loading required package: BSgenome</span></span>
<span id="cb24-6"><a href="getting-set-up.html#cb24-6" tabindex="-1"></a><span class="do">## Loading required package: Biostrings</span></span>
<span id="cb24-7"><a href="getting-set-up.html#cb24-7" tabindex="-1"></a><span class="do">## Loading required package: XVector</span></span>
<span id="cb24-8"><a href="getting-set-up.html#cb24-8" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb24-9"><a href="getting-set-up.html#cb24-9" tabindex="-1"></a><span class="do">## Attaching package: &#39;XVector&#39;</span></span>
<span id="cb24-10"><a href="getting-set-up.html#cb24-10" tabindex="-1"></a><span class="do">## The following object is masked from &#39;package:plyr&#39;:</span></span>
<span id="cb24-11"><a href="getting-set-up.html#cb24-11" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb24-12"><a href="getting-set-up.html#cb24-12" tabindex="-1"></a><span class="do">##     compact</span></span>
<span id="cb24-13"><a href="getting-set-up.html#cb24-13" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb24-14"><a href="getting-set-up.html#cb24-14" tabindex="-1"></a><span class="do">## Attaching package: &#39;Biostrings&#39;</span></span>
<span id="cb24-15"><a href="getting-set-up.html#cb24-15" tabindex="-1"></a><span class="do">## The following object is masked from &#39;package:grid&#39;:</span></span>
<span id="cb24-16"><a href="getting-set-up.html#cb24-16" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb24-17"><a href="getting-set-up.html#cb24-17" tabindex="-1"></a><span class="do">##     pattern</span></span>
<span id="cb24-18"><a href="getting-set-up.html#cb24-18" tabindex="-1"></a><span class="do">## The following object is masked from &#39;package:base&#39;:</span></span>
<span id="cb24-19"><a href="getting-set-up.html#cb24-19" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb24-20"><a href="getting-set-up.html#cb24-20" tabindex="-1"></a><span class="do">##     strsplit</span></span>
<span id="cb24-21"><a href="getting-set-up.html#cb24-21" tabindex="-1"></a><span class="do">## Loading required package: BiocIO</span></span>
<span id="cb24-22"><a href="getting-set-up.html#cb24-22" tabindex="-1"></a><span class="do">## Loading required package: rtracklayer</span></span>
<span id="cb24-23"><a href="getting-set-up.html#cb24-23" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb24-24"><a href="getting-set-up.html#cb24-24" tabindex="-1"></a><span class="do">## Attaching package: &#39;rtracklayer&#39;</span></span>
<span id="cb24-25"><a href="getting-set-up.html#cb24-25" tabindex="-1"></a><span class="do">## The following object is masked from &#39;package:BiocIO&#39;:</span></span>
<span id="cb24-26"><a href="getting-set-up.html#cb24-26" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb24-27"><a href="getting-set-up.html#cb24-27" tabindex="-1"></a><span class="do">##     FileForFormat</span></span></code></pre></div>
<p>Then we create a genome annotation from this <code>BSgenome</code> object.</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="getting-set-up.html#cb25-1" tabindex="-1"></a>genomeAnnotation <span class="ot">&lt;-</span> <span class="fu">createGenomeAnnotation</span>(<span class="at">genome =</span> BSgenome.Dmelanogaster.UCSC.dm6)</span>
<span id="cb25-2"><a href="getting-set-up.html#cb25-2" tabindex="-1"></a><span class="do">## Getting genome..</span></span>
<span id="cb25-3"><a href="getting-set-up.html#cb25-3" tabindex="-1"></a><span class="do">## Attempting to infer chromSizes..</span></span>
<span id="cb25-4"><a href="getting-set-up.html#cb25-4" tabindex="-1"></a><span class="do">## Attempting to infer blacklist..</span></span>
<span id="cb25-5"><a href="getting-set-up.html#cb25-5" tabindex="-1"></a><span class="do">## Blacklist not downloaded! Continuing without, be careful for downstream biases..</span></span></code></pre></div>
<p>Examining this object shows the constituent parts of an ArchR genome annotation.</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="getting-set-up.html#cb26-1" tabindex="-1"></a>genomeAnnotation</span>
<span id="cb26-2"><a href="getting-set-up.html#cb26-2" tabindex="-1"></a><span class="do">## List of length 3</span></span>
<span id="cb26-3"><a href="getting-set-up.html#cb26-3" tabindex="-1"></a><span class="do">## names(3): genome chromSizes blacklist</span></span></code></pre></div>
<p>To create a <strong>custom gene annotation</strong>, we use <code>createGeneAnnotation()</code>. To do this, you will need the following information:</p>
<ol style="list-style-type: decimal">
<li>A <code>TxDb</code> object (transcript database) from Bioconductor which contains information for gene/transcript coordinates.</li>
<li>An <code>OrgDb</code> object (organism database) from Bioconductor which provides a unified framework to map between gene names and various gene identifiers.</li>
</ol>
<p>Continuing with our example from Drosophila melanogaster, we first install and load the relevant <code>TxDb</code> and <code>OrgDb</code> objects.</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="getting-set-up.html#cb27-1" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">&quot;TxDb.Dmelanogaster.UCSC.dm6.ensGene&quot;</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>)){</span>
<span id="cb27-2"><a href="getting-set-up.html#cb27-2" tabindex="-1"></a>  BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">&quot;TxDb.Dmelanogaster.UCSC.dm6.ensGene&quot;</span>, <span class="at">update =</span> <span class="cn">FALSE</span>)</span>
<span id="cb27-3"><a href="getting-set-up.html#cb27-3" tabindex="-1"></a>}</span>
<span id="cb27-4"><a href="getting-set-up.html#cb27-4" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">&quot;org.Dm.eg.db&quot;</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>)){</span>
<span id="cb27-5"><a href="getting-set-up.html#cb27-5" tabindex="-1"></a>  BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">&quot;org.Dm.eg.db&quot;</span>, <span class="at">update =</span> <span class="cn">FALSE</span>)</span>
<span id="cb27-6"><a href="getting-set-up.html#cb27-6" tabindex="-1"></a>}</span>
<span id="cb27-7"><a href="getting-set-up.html#cb27-7" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb27-8"><a href="getting-set-up.html#cb27-8" tabindex="-1"></a><span class="fu">library</span>(TxDb.Dmelanogaster.UCSC.dm6.ensGene)</span>
<span id="cb27-9"><a href="getting-set-up.html#cb27-9" tabindex="-1"></a><span class="do">## Loading required package: GenomicFeatures</span></span>
<span id="cb27-10"><a href="getting-set-up.html#cb27-10" tabindex="-1"></a><span class="do">## Loading required package: AnnotationDbi</span></span>
<span id="cb27-11"><a href="getting-set-up.html#cb27-11" tabindex="-1"></a><span class="fu">library</span>(org.Dm.eg.db)</span></code></pre></div>
<p>Then we create the gene annotation object.</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="getting-set-up.html#cb28-1" tabindex="-1"></a>geneAnnotation <span class="ot">&lt;-</span> <span class="fu">createGeneAnnotation</span>(<span class="at">TxDb =</span> TxDb.Dmelanogaster.UCSC.dm6.ensGene, <span class="at">OrgDb =</span> org.Dm.eg.db)</span>
<span id="cb28-2"><a href="getting-set-up.html#cb28-2" tabindex="-1"></a><span class="do">## Getting Genes..</span></span>
<span id="cb28-3"><a href="getting-set-up.html#cb28-3" tabindex="-1"></a><span class="do">##   1 gene was dropped because it has exons located on both strands of the</span></span>
<span id="cb28-4"><a href="getting-set-up.html#cb28-4" tabindex="-1"></a><span class="do">##   same reference sequence or on more than one reference sequence, so</span></span>
<span id="cb28-5"><a href="getting-set-up.html#cb28-5" tabindex="-1"></a><span class="do">##   cannot be represented by a single genomic range.</span></span>
<span id="cb28-6"><a href="getting-set-up.html#cb28-6" tabindex="-1"></a><span class="do">##   Use &#39;single.strand.genes.only=FALSE&#39; to get all the genes in a</span></span>
<span id="cb28-7"><a href="getting-set-up.html#cb28-7" tabindex="-1"></a><span class="do">##   GRangesList object, or use suppressMessages() to suppress this message.</span></span>
<span id="cb28-8"><a href="getting-set-up.html#cb28-8" tabindex="-1"></a><span class="do">## Determined Annotation Style = ENSEMBL</span></span>
<span id="cb28-9"><a href="getting-set-up.html#cb28-9" tabindex="-1"></a><span class="do">## Getting Exons..</span></span>
<span id="cb28-10"><a href="getting-set-up.html#cb28-10" tabindex="-1"></a><span class="do">## Getting TSS..</span></span></code></pre></div>
<p>Examining this object shows the constituent parts of an ArchR gene annotation.</p>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="getting-set-up.html#cb29-1" tabindex="-1"></a>geneAnnotation</span>
<span id="cb29-2"><a href="getting-set-up.html#cb29-2" tabindex="-1"></a><span class="do">## List of length 3</span></span>
<span id="cb29-3"><a href="getting-set-up.html#cb29-3" tabindex="-1"></a><span class="do">## names(3): genes exons TSS</span></span></code></pre></div>
<p>In some cases, information contained within the <code>OrgDb</code> object and the <code>TxDb</code> object might not match as expected. ArchR will attempt to default to an annotation style / keytype of “ENTREZID” or “ENSEMBL” to map between gene names within the <code>TxDb</code> object and various gene identifiers within the <code>OrgDb</code> object. For some organisms, (for ex. <a href="https://github.com/GreenleafLab/ArchR/issues/318#issuecomment-717786201">arabidopsis</a>), you may need to specify a particular annotation style to the <code>annoStyle</code> parameter of <code>createGeneAnnotation()</code> to tell ArchR what the keytype is of the gene names contained within your <code>TxDb</code> object.</p>
<p>Alternatively, if you dont have a <code>TxDb</code> and <code>OrgDb</code> object, you can create a <code>geneAnnotation</code> object from the following information :</p>
<ol style="list-style-type: decimal">
<li>A “genes” object - a <code>GRanges</code> object containing gene coordinates (start to end). This object must have a symbols column matching the symbols column of the “exons” object described below.</li>
<li>An “exons” object - <code>GRanges</code> object containing gene exon coordinates. Must have a symbols column matching the symbols column of the “genes” object described above.</li>
<li>A <code>GRanges</code> object containing standed transcription start site (<code>TSS</code>) coordinates.</li>
</ol>
<!-- #JJJ Is the TSS object used for anything other than TSS scores? If so, shouldnt it have a symbols column as well? Either way, we should add that info to point #3 -->
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="getting-set-up.html#cb30-1" tabindex="-1"></a>geneAnnotation <span class="ot">&lt;-</span> <span class="fu">createGeneAnnotation</span>(</span>
<span id="cb30-2"><a href="getting-set-up.html#cb30-2" tabindex="-1"></a>  <span class="at">TSS =</span> geneAnnotation<span class="sc">$</span>TSS, </span>
<span id="cb30-3"><a href="getting-set-up.html#cb30-3" tabindex="-1"></a>  <span class="at">exons =</span> geneAnnotation<span class="sc">$</span>exons, </span>
<span id="cb30-4"><a href="getting-set-up.html#cb30-4" tabindex="-1"></a>  <span class="at">genes =</span> geneAnnotation<span class="sc">$</span>genes</span>
<span id="cb30-5"><a href="getting-set-up.html#cb30-5" tabindex="-1"></a>)</span></code></pre></div>
<p>In general, we cannot provide much support for non-standard genomes. However, some ArchR users have created a repository to hold custom genome annotations for some popular model organisms. We encourage you to check out and contribute to that <a href="https://github.com/pfenninglab/custom_ArchR_genomes_and_annotations">user-developed custom genomes repository</a>. If you are looking for information on how to forge your own custom gene annotation, you should read through <a href="https://github.com/GreenleafLab/ArchR/issues/112">previous issue posts</a> for inspiration. For species with lots of small scaffolds, please make sure to remove genes within 2 kb of scaffold ends from the gene annotation, remove unassembled scaffolds from the genome annotation, and remove any scaffolds that lack genes as these will interfere with downstream analyses. When creating a custom genome for use with ArchR, it is imperative that all of the provided information matches. More explicitly, the chromosomes/<code>seqnames</code> present in your <code>genomeAnnotation</code> must match those present in your <code>geneAnnotation</code> and both must match the genome used for alignment. If you have contigs with no mapped reads, these may cause problems for ArchR and you should remove them entirely from your data and annotations.</p>
</div>
<div id="using-non-standard-genomes-in-archr" class="section level3 hasAnchor" number="3.5.2">
<h3><span class="header-section-number">3.5.2</span> Using Non-standard Genomes in ArchR<a href="getting-set-up.html#using-non-standard-genomes-in-archr" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>ArchR implements some checks to prevent you from doing things that we consider “out of the norm”. One of these checks forces the <code>seqnames</code> of your genome annotation to start with “chr”. This is true in most applications but there are some genomes (for example Maize) which do not use “chr” as a prefix for the different chromosomes. To perform any ArchR analyses on such a genome, you have to tell ArchR to ignore the chromosome prefixes. To do this, you must run <code>addArchRChrPrefix(chrPrefix = FALSE)</code> prior to creating your Arrow files. This will turn off the requirement for “chr” prefixes on seqnames globally in the current R session. Please keep in mind ArchR converts chromosome/seqnames to character by default. Thus, if your <code>seqnames</code> are just integers, you will need to supply them as characters i.e. <code>useSeqnames = c("1", "2", "3")</code> instead of <code>useSeqnames = c(1, 2, 3)</code>. You can always check if chromosome prefixes are required in the current R session using <code>getArchRChrPrefix()</code>.</p>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="input-file-types-in-archr.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="creating-arrow-files.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": null
},
"fontsettings": null,
"edit": null,
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "none"
},
"info": false
});
});
</script>

</body>

</html>
