<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>9.3 Identifying Marker Genes | ArchR: Robust and scaleable analysis of single-cell chromatin accessibility data.</title>
  <meta name="description" content="A guide to ArchR" />
  <meta name="generator" content="bookdown 0.41 and GitBook 2.6.7" />

  <meta property="og:title" content="9.3 Identifying Marker Genes | ArchR: Robust and scaleable analysis of single-cell chromatin accessibility data." />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="A guide to ArchR" />
  <meta name="github-repo" content="GreenleafLab/ArchR" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="9.3 Identifying Marker Genes | ArchR: Robust and scaleable analysis of single-cell chromatin accessibility data." />
  
  <meta name="twitter:description" content="A guide to ArchR" />
  

<meta name="author" content="Jeffrey Granja and Ryan Corces" />


<meta name="date" content="2025-01-23" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="images/favicon.ico" type="image/x-icon" />
<link rel="prev" href="identification-of-marker-features.html"/>
<link rel="next" href="visualizing-marker-genes-on-an-embedding.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<link href="libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="../index.html">Back to ArchR Home</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><img src="images/ArchR_Logo_Integrated.png" alt="" width="200"></a></li>
<li class="chapter" data-level="1" data-path="debugging-and-troubleshooting-in-archr.html"><a href="debugging-and-troubleshooting-in-archr.html"><i class="fa fa-check"></i><b>1</b> Debugging and Troubleshooting in ArchR</a></li>
<li class="chapter" data-level="2" data-path="manage-archrs-dependencies.html"><a href="manage-archrs-dependencies.html"><i class="fa fa-check"></i><b>2</b> Manage ArchR’s Dependencies</a>
<ul>
<li class="chapter" data-level="2.1" data-path="using-renv-to-manage-dependencies.html"><a href="using-renv-to-manage-dependencies.html"><i class="fa fa-check"></i><b>2.1</b> Using renv to manage dependencies</a></li>
<li class="chapter" data-level="2.2" data-path="other-environment-managers.html"><a href="other-environment-managers.html"><i class="fa fa-check"></i><b>2.2</b> Other environment managers</a></li>
<li class="chapter" data-level="2.3" data-path="using-docker-for-pre-compiled-environment-setup.html"><a href="using-docker-for-pre-compiled-environment-setup.html"><i class="fa fa-check"></i><b>2.3</b> Using Docker for pre-compiled environment setup</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="getting-started-with-archr.html"><a href="getting-started-with-archr.html"><i class="fa fa-check"></i><b>3</b> Getting Started with ArchR</a>
<ul>
<li class="chapter" data-level="3.1" data-path="a-brief-primer-on-atac-seq-terminology.html"><a href="a-brief-primer-on-atac-seq-terminology.html"><i class="fa fa-check"></i><b>3.1</b> A Brief Primer on ATAC-seq Terminology</a></li>
<li class="chapter" data-level="3.2" data-path="why-use-archr.html"><a href="why-use-archr.html"><i class="fa fa-check"></i><b>3.2</b> Why use ArchR?</a></li>
<li class="chapter" data-level="3.3" data-path="what-is-an-arrow-file-archrproject.html"><a href="what-is-an-arrow-file-archrproject.html"><i class="fa fa-check"></i><b>3.3</b> What is an Arrow file / <code>ArchRProject</code>?</a></li>
<li class="chapter" data-level="3.4" data-path="input-file-types-in-archr.html"><a href="input-file-types-in-archr.html"><i class="fa fa-check"></i><b>3.4</b> Input File Types in ArchR</a></li>
<li class="chapter" data-level="3.5" data-path="getting-set-up.html"><a href="getting-set-up.html"><i class="fa fa-check"></i><b>3.5</b> Getting Set Up</a></li>
<li class="chapter" data-level="3.6" data-path="creating-arrow-files.html"><a href="creating-arrow-files.html"><i class="fa fa-check"></i><b>3.6</b> Creating Arrow Files</a></li>
<li class="chapter" data-level="3.7" data-path="per-cell-quality-control.html"><a href="per-cell-quality-control.html"><i class="fa fa-check"></i><b>3.7</b> Per-cell Quality Control</a></li>
<li class="chapter" data-level="3.8" data-path="using-bam-files-for-arrow-file-creation.html"><a href="using-bam-files-for-arrow-file-creation.html"><i class="fa fa-check"></i><b>3.8</b> Using BAM files for Arrow File creation</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="doublet-inference-with-archr.html"><a href="doublet-inference-with-archr.html"><i class="fa fa-check"></i><b>4</b> Doublet Inference with ArchR</a>
<ul>
<li class="chapter" data-level="4.1" data-path="how-does-doublet-identification-work-in-archr.html"><a href="how-does-doublet-identification-work-in-archr.html"><i class="fa fa-check"></i><b>4.1</b> How does doublet identification work in ArchR?</a></li>
<li class="chapter" data-level="4.2" data-path="inferring-scatac-seq-doublets-with-archr.html"><a href="inferring-scatac-seq-doublets-with-archr.html"><i class="fa fa-check"></i><b>4.2</b> Inferring scATAC-seq Doublets with ArchR</a></li>
<li class="chapter" data-level="4.3" data-path="using-demuxlet-with-archr.html"><a href="using-demuxlet-with-archr.html"><i class="fa fa-check"></i><b>4.3</b> Using <em>demuxlet</em> with ArchR</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="creating-an-archrproject.html"><a href="creating-an-archrproject.html"><i class="fa fa-check"></i><b>5</b> Creating an ArchRProject</a>
<ul>
<li class="chapter" data-level="5.1" data-path="creating-an-archrproject-1.html"><a href="creating-an-archrproject-1.html"><i class="fa fa-check"></i><b>5.1</b> Creating An ArchRProject</a></li>
<li class="chapter" data-level="5.2" data-path="manipulating-an-archrproject.html"><a href="manipulating-an-archrproject.html"><i class="fa fa-check"></i><b>5.2</b> Manipulating An ArchRProject</a></li>
<li class="chapter" data-level="5.3" data-path="plotting-sample-statistics-from-an-archrproject.html"><a href="plotting-sample-statistics-from-an-archrproject.html"><i class="fa fa-check"></i><b>5.3</b> Plotting Sample Statistics from an ArchRProject</a></li>
<li class="chapter" data-level="5.4" data-path="plotting-sample-fragment-size-distribution-and-tss-enrichment-profiles..html"><a href="plotting-sample-fragment-size-distribution-and-tss-enrichment-profiles..html"><i class="fa fa-check"></i><b>5.4</b> Plotting Sample Fragment Size Distribution and TSS Enrichment Profiles.</a></li>
<li class="chapter" data-level="5.5" data-path="saving-and-loading-an-archrproject.html"><a href="saving-and-loading-an-archrproject.html"><i class="fa fa-check"></i><b>5.5</b> Saving and Loading an <code>ArchRProject</code></a></li>
<li class="chapter" data-level="5.6" data-path="filtering-doublets-from-an-archrproject.html"><a href="filtering-doublets-from-an-archrproject.html"><i class="fa fa-check"></i><b>5.6</b> Filtering Doublets from an ArchRProject</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="dimensionality-reduction-with-archr.html"><a href="dimensionality-reduction-with-archr.html"><i class="fa fa-check"></i><b>6</b> Dimensionality Reduction with ArchR</a>
<ul>
<li class="chapter" data-level="6.1" data-path="archrs-lsi-implementation.html"><a href="archrs-lsi-implementation.html"><i class="fa fa-check"></i><b>6.1</b> ArchR’s LSI Implementation</a></li>
<li class="chapter" data-level="6.2" data-path="iterative-latent-semantic-indexing-lsi.html"><a href="iterative-latent-semantic-indexing-lsi.html"><i class="fa fa-check"></i><b>6.2</b> Iterative Latent Semantic Indexing (LSI)</a></li>
<li class="chapter" data-level="6.3" data-path="estimated-lsi.html"><a href="estimated-lsi.html"><i class="fa fa-check"></i><b>6.3</b> Estimated LSI</a></li>
<li class="chapter" data-level="6.4" data-path="batch-effect-correction-wtih-harmony.html"><a href="batch-effect-correction-wtih-harmony.html"><i class="fa fa-check"></i><b>6.4</b> Batch Effect Correction wtih Harmony</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="clustering-with-archr.html"><a href="clustering-with-archr.html"><i class="fa fa-check"></i><b>7</b> Clustering with ArchR</a>
<ul>
<li class="chapter" data-level="7.1" data-path="clustering-using-seurats-findclusters-function.html"><a href="clustering-using-seurats-findclusters-function.html"><i class="fa fa-check"></i><b>7.1</b> Clustering using Seurat’s <code>FindClusters()</code> function</a></li>
<li class="chapter" data-level="7.2" data-path="clustering-using-scran.html"><a href="clustering-using-scran.html"><i class="fa fa-check"></i><b>7.2</b> Clustering using <code>scran</code></a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="single-cell-embeddings.html"><a href="single-cell-embeddings.html"><i class="fa fa-check"></i><b>8</b> Single-cell Embeddings</a>
<ul>
<li class="chapter" data-level="8.1" data-path="uniform-manifold-approximation-and-projection-umap.html"><a href="uniform-manifold-approximation-and-projection-umap.html"><i class="fa fa-check"></i><b>8.1</b> Uniform Manifold Approximation and Projection (UMAP)</a></li>
<li class="chapter" data-level="8.2" data-path="t-stocastic-neighbor-embedding-t-sne.html"><a href="t-stocastic-neighbor-embedding-t-sne.html"><i class="fa fa-check"></i><b>8.2</b> t-Stocastic Neighbor Embedding (t-SNE)</a></li>
<li class="chapter" data-level="8.3" data-path="dimensionality-reduction-after-harmony.html"><a href="dimensionality-reduction-after-harmony.html"><i class="fa fa-check"></i><b>8.3</b> Dimensionality Reduction After Harmony</a></li>
<li class="chapter" data-level="8.4" data-path="highlighting-specific-cells-on-an-embedding.html"><a href="highlighting-specific-cells-on-an-embedding.html"><i class="fa fa-check"></i><b>8.4</b> Highlighting specific cells on an embedding</a></li>
<li class="chapter" data-level="8.5" data-path="importing-an-embedding-from-external-software.html"><a href="importing-an-embedding-from-external-software.html"><i class="fa fa-check"></i><b>8.5</b> Importing an embedding from external software</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="gene-scores-and-marker-genes-with-archr.html"><a href="gene-scores-and-marker-genes-with-archr.html"><i class="fa fa-check"></i><b>9</b> Gene Scores and Marker Genes with ArchR</a>
<ul>
<li class="chapter" data-level="9.1" data-path="calculating-gene-scores-in-archr.html"><a href="calculating-gene-scores-in-archr.html"><i class="fa fa-check"></i><b>9.1</b> Calculating Gene Scores in ArchR</a></li>
<li class="chapter" data-level="9.2" data-path="identification-of-marker-features.html"><a href="identification-of-marker-features.html"><i class="fa fa-check"></i><b>9.2</b> Identification of Marker Features</a></li>
<li class="chapter" data-level="9.3" data-path="identifying-marker-genes.html"><a href="identifying-marker-genes.html"><i class="fa fa-check"></i><b>9.3</b> Identifying Marker Genes</a></li>
<li class="chapter" data-level="9.4" data-path="visualizing-marker-genes-on-an-embedding.html"><a href="visualizing-marker-genes-on-an-embedding.html"><i class="fa fa-check"></i><b>9.4</b> Visualizing Marker Genes on an Embedding</a></li>
<li class="chapter" data-level="9.5" data-path="marker-genes-imputation-with-magic.html"><a href="marker-genes-imputation-with-magic.html"><i class="fa fa-check"></i><b>9.5</b> Marker Genes Imputation with MAGIC</a></li>
<li class="chapter" data-level="9.6" data-path="module-scores.html"><a href="module-scores.html"><i class="fa fa-check"></i><b>9.6</b> Module Scores</a></li>
<li class="chapter" data-level="9.7" data-path="track-plotting-with-archrbrowser.html"><a href="track-plotting-with-archrbrowser.html"><i class="fa fa-check"></i><b>9.7</b> Track Plotting with ArchRBrowser</a></li>
<li class="chapter" data-level="9.8" data-path="launching-the-interactive-archrbrowser.html"><a href="launching-the-interactive-archrbrowser.html"><i class="fa fa-check"></i><b>9.8</b> Launching the Interactive ArchRBrowser</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="defining-cluster-identity-with-scrna-seq.html"><a href="defining-cluster-identity-with-scrna-seq.html"><i class="fa fa-check"></i><b>10</b> Defining Cluster Identity with scRNA-seq</a>
<ul>
<li class="chapter" data-level="10.1" data-path="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html"><i class="fa fa-check"></i><b>10.1</b> Cross-platform linkage of scATAC-seq cells with scRNA-seq cells</a></li>
<li class="chapter" data-level="10.2" data-path="adding-pseudo-scrna-seq-profiles-for-each-scatac-seq-cell.html"><a href="adding-pseudo-scrna-seq-profiles-for-each-scatac-seq-cell.html"><i class="fa fa-check"></i><b>10.2</b> Adding Pseudo-scRNA-seq profiles for each scATAC-seq cell</a></li>
<li class="chapter" data-level="10.3" data-path="labeling-scatac-seq-clusters-with-scrna-seq-information.html"><a href="labeling-scatac-seq-clusters-with-scrna-seq-information.html"><i class="fa fa-check"></i><b>10.3</b> Labeling scATAC-seq clusters with scRNA-seq information</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="pseudo-bulk-replicates-in-archr.html"><a href="pseudo-bulk-replicates-in-archr.html"><i class="fa fa-check"></i><b>11</b> Pseudo-bulk Replicates in ArchR</a>
<ul>
<li class="chapter" data-level="11.1" data-path="how-does-archr-make-pseudo-bulk-replicates.html"><a href="how-does-archr-make-pseudo-bulk-replicates.html"><i class="fa fa-check"></i><b>11.1</b> How Does ArchR Make Pseudo-bulk Replicates?</a></li>
<li class="chapter" data-level="11.2" data-path="making-pseudo-bulk-replicates.html"><a href="making-pseudo-bulk-replicates.html"><i class="fa fa-check"></i><b>11.2</b> Making Pseudo-bulk Replicates</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="calling-peaks-with-archr.html"><a href="calling-peaks-with-archr.html"><i class="fa fa-check"></i><b>12</b> Calling Peaks with ArchR</a>
<ul>
<li class="chapter" data-level="12.1" data-path="the-iterative-overlap-peak-merging-procedure.html"><a href="the-iterative-overlap-peak-merging-procedure.html"><i class="fa fa-check"></i><b>12.1</b> The Iterative Overlap Peak Merging Procedure</a></li>
<li class="chapter" data-level="12.2" data-path="calling-peaks-w-macs2.html"><a href="calling-peaks-w-macs2.html"><i class="fa fa-check"></i><b>12.2</b> Calling Peaks w/ Macs2</a></li>
<li class="chapter" data-level="12.3" data-path="calling-peaks-w-tilematrix.html"><a href="calling-peaks-w-tilematrix.html"><i class="fa fa-check"></i><b>12.3</b> Calling Peaks w/ TileMatrix</a></li>
<li class="chapter" data-level="12.4" data-path="adding-a-peak-matrix.html"><a href="adding-a-peak-matrix.html"><i class="fa fa-check"></i><b>12.4</b> Adding a Peak Matrix</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="identifying-marker-peaks.html"><a href="identifying-marker-peaks.html"><i class="fa fa-check"></i><b>13</b> Identifying Marker Peaks</a>
<ul>
<li class="chapter" data-level="13.1" data-path="identifying-marker-peaks-with-archr.html"><a href="identifying-marker-peaks-with-archr.html"><i class="fa fa-check"></i><b>13.1</b> Identifying Marker Peaks with ArchR</a></li>
<li class="chapter" data-level="13.2" data-path="plotting-marker-peaks-in-archr.html"><a href="plotting-marker-peaks-in-archr.html"><i class="fa fa-check"></i><b>13.2</b> Plotting Marker Peaks in ArchR</a></li>
<li class="chapter" data-level="13.3" data-path="pairwise-testing-between-groups.html"><a href="pairwise-testing-between-groups.html"><i class="fa fa-check"></i><b>13.3</b> Pairwise Testing Between Groups</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="motif-and-feature-enrichment-with-archr.html"><a href="motif-and-feature-enrichment-with-archr.html"><i class="fa fa-check"></i><b>14</b> Motif and Feature Enrichment with ArchR</a>
<ul>
<li class="chapter" data-level="14.1" data-path="motif-enrichment.html"><a href="motif-enrichment.html"><i class="fa fa-check"></i><b>14.1</b> Motif Enrichment</a></li>
<li class="chapter" data-level="14.2" data-path="archr-enrichment.html"><a href="archr-enrichment.html"><i class="fa fa-check"></i><b>14.2</b> ArchR Enrichment</a></li>
<li class="chapter" data-level="14.3" data-path="custom-enrichment.html"><a href="custom-enrichment.html"><i class="fa fa-check"></i><b>14.3</b> Custom Enrichment</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="chromvar-deviatons-enrichment-with-archr.html"><a href="chromvar-deviatons-enrichment-with-archr.html"><i class="fa fa-check"></i><b>15</b> ChromVAR Deviatons Enrichment with ArchR</a>
<ul>
<li class="chapter" data-level="15.1" data-path="motif-deviations.html"><a href="motif-deviations.html"><i class="fa fa-check"></i><b>15.1</b> Motif Deviations</a></li>
<li class="chapter" data-level="15.2" data-path="archr-and-custom-deviations.html"><a href="archr-and-custom-deviations.html"><i class="fa fa-check"></i><b>15.2</b> ArchR and Custom Deviations</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="footprinting-with-archr.html"><a href="footprinting-with-archr.html"><i class="fa fa-check"></i><b>16</b> Footprinting with ArchR</a>
<ul>
<li class="chapter" data-level="16.1" data-path="motif-footprinting.html"><a href="motif-footprinting.html"><i class="fa fa-check"></i><b>16.1</b> Motif Footprinting</a></li>
<li class="chapter" data-level="16.2" data-path="normalization-of-footprints-for-tn5-bias.html"><a href="normalization-of-footprints-for-tn5-bias.html"><i class="fa fa-check"></i><b>16.2</b> Normalization of Footprints for Tn5 Bias</a></li>
<li class="chapter" data-level="16.3" data-path="feature-footprinting.html"><a href="feature-footprinting.html"><i class="fa fa-check"></i><b>16.3</b> Feature Footprinting</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="integrative-analysis-with-archr.html"><a href="integrative-analysis-with-archr.html"><i class="fa fa-check"></i><b>17</b> Integrative Analysis with ArchR</a>
<ul>
<li class="chapter" data-level="17.1" data-path="creating-low-overlapping-aggregates-of-cells.html"><a href="creating-low-overlapping-aggregates-of-cells.html"><i class="fa fa-check"></i><b>17.1</b> Creating Low-Overlapping Aggregates of Cells</a></li>
<li class="chapter" data-level="17.2" data-path="co-accessibility-with-archr.html"><a href="co-accessibility-with-archr.html"><i class="fa fa-check"></i><b>17.2</b> Co-accessibility with ArchR</a></li>
<li class="chapter" data-level="17.3" data-path="peak2genelinkage-with-archr.html"><a href="peak2genelinkage-with-archr.html"><i class="fa fa-check"></i><b>17.3</b> Peak2GeneLinkage with ArchR</a></li>
<li class="chapter" data-level="17.4" data-path="identification-of-positive-tf-regulators.html"><a href="identification-of-positive-tf-regulators.html"><i class="fa fa-check"></i><b>17.4</b> Identification of Positive TF-Regulators</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="trajectory-analysis-with-archr.html"><a href="trajectory-analysis-with-archr.html"><i class="fa fa-check"></i><b>18</b> Trajectory Analysis with ArchR</a>
<ul>
<li class="chapter" data-level="18.1" data-path="myeloid-trajectory---monocyte-differentiation.html"><a href="myeloid-trajectory---monocyte-differentiation.html"><i class="fa fa-check"></i><b>18.1</b> Myeloid Trajectory - Monocyte Differentiation</a></li>
<li class="chapter" data-level="18.2" data-path="lymphoid-trajectory---b-cell-cifferentiation.html"><a href="lymphoid-trajectory---b-cell-cifferentiation.html"><i class="fa fa-check"></i><b>18.2</b> Lymphoid Trajectory - B Cell Cifferentiation</a></li>
<li class="chapter" data-level="18.3" data-path="monocle3-trajectories.html"><a href="monocle3-trajectories.html"><i class="fa fa-check"></i><b>18.3</b> Monocle3 Trajectories</a></li>
<li class="chapter" data-level="18.4" data-path="slingshot-trajectories.html"><a href="slingshot-trajectories.html"><i class="fa fa-check"></i><b>18.4</b> Slingshot Trajectories</a></li>
<li class="chapter" data-level="18.5" data-path="comparing-the-different-trajectories.html"><a href="comparing-the-different-trajectories.html"><i class="fa fa-check"></i><b>18.5</b> Comparing the different trajectories</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="handling-archr-output.html"><a href="handling-archr-output.html"><i class="fa fa-check"></i><b>19</b> Handling ArchR Output</a>
<ul>
<li class="chapter" data-level="19.1" data-path="exporting-fragment-level-data.html"><a href="exporting-fragment-level-data.html"><i class="fa fa-check"></i><b>19.1</b> Exporting fragment-level data</a></li>
<li class="chapter" data-level="19.2" data-path="exporting-matrix-level-data.html"><a href="exporting-matrix-level-data.html"><i class="fa fa-check"></i><b>19.2</b> Exporting matrix-level data</a></li>
<li class="chapter" data-level="19.3" data-path="exporting-pseudo-bulked-data-to-a-summarizedexperiment.html"><a href="exporting-pseudo-bulked-data-to-a-summarizedexperiment.html"><i class="fa fa-check"></i><b>19.3</b> Exporting pseudo-bulked data to a <code>SummarizedExperiment</code></a></li>
<li class="chapter" data-level="19.4" data-path="exporting-pseudo-bulked-data-to-a-bigwig-file.html"><a href="exporting-pseudo-bulked-data-to-a-bigwig-file.html"><i class="fa fa-check"></i><b>19.4</b> Exporting pseudo-bulked data to a bigWig file</a></li>
</ul></li>
<li class="chapter" data-level="20" data-path="integration-with-bulk-atac-seq.html"><a href="integration-with-bulk-atac-seq.html"><i class="fa fa-check"></i><b>20</b> Integration with bulk ATAC-seq</a>
<ul>
<li class="chapter" data-level="20.1" data-path="projecting-bulk-atac-seq-data.html"><a href="projecting-bulk-atac-seq-data.html"><i class="fa fa-check"></i><b>20.1</b> Projecting bulk ATAC-seq data</a></li>
</ul></li>
<li class="chapter" data-level="21" data-path="multiomic-data-analysis-in-archr.html"><a href="multiomic-data-analysis-in-archr.html"><i class="fa fa-check"></i><b>21</b> Multiomic data analysis in ArchR</a>
<ul>
<li class="chapter" data-level="21.1" data-path="importing-data-and-setting-up-a-multiome-project.html"><a href="importing-data-and-setting-up-a-multiome-project.html"><i class="fa fa-check"></i><b>21.1</b> Importing data and setting up a Multiome project</a></li>
<li class="chapter" data-level="21.2" data-path="analysis-of-multiome-data-in-archr.html"><a href="analysis-of-multiome-data-in-archr.html"><i class="fa fa-check"></i><b>21.2</b> Analysis of multiome data in ArchR</a></li>
</ul></li>
<li class="chapter" data-level="22" data-path="plot-aesthetics-in-archr.html"><a href="plot-aesthetics-in-archr.html"><i class="fa fa-check"></i><b>22</b> Plot aesthetics in ArchR</a>
<ul>
<li class="chapter" data-level="22.1" data-path="using-and-manipulating-palettes-in-archr.html"><a href="using-and-manipulating-palettes-in-archr.html"><i class="fa fa-check"></i><b>22.1</b> Using and manipulating palettes in ArchR</a></li>
</ul></li>
<li class="chapter" data-level="23" data-path="testing-your-archr-installation.html"><a href="testing-your-archr-installation.html"><i class="fa fa-check"></i><b>23</b> Testing Your ArchR Installation</a></li>
<li class="chapter" data-level="24" data-path="session-information.html"><a href="session-information.html"><i class="fa fa-check"></i><b>24</b> Session Information</a></li>
<li class="divider"></li>
<li><a href="https://bookdown.org" target="_blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">ArchR: Robust and scaleable analysis of single-cell chromatin accessibility data.</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="identifying-marker-genes" class="section level2 hasAnchor" number="9.3">
<h2><span class="header-section-number">9.3</span> Identifying Marker Genes<a href="identifying-marker-genes.html#identifying-marker-genes" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>To identify marker genes based on gene scores, we call the <code>getMarkerFeatures()</code> function with <code>useMatrix = "GeneScoreMatrix"</code>. We specify that we want to know the cluster-specific features with <code>groupBy = "Clusters"</code> which tells ArchR to use the “Clusters” column in <code>cellColData</code> to stratify cell groups.</p>
<div class="sourceCode" id="cb123"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb123-1"><a href="identifying-marker-genes.html#cb123-1" tabindex="-1"></a>markersGS <span class="ot">&lt;-</span> <span class="fu">getMarkerFeatures</span>(</span>
<span id="cb123-2"><a href="identifying-marker-genes.html#cb123-2" tabindex="-1"></a>    <span class="at">ArchRProj =</span> projHeme2, </span>
<span id="cb123-3"><a href="identifying-marker-genes.html#cb123-3" tabindex="-1"></a>    <span class="at">useMatrix =</span> <span class="st">&quot;GeneScoreMatrix&quot;</span>, </span>
<span id="cb123-4"><a href="identifying-marker-genes.html#cb123-4" tabindex="-1"></a>    <span class="at">groupBy =</span> <span class="st">&quot;Clusters&quot;</span>,</span>
<span id="cb123-5"><a href="identifying-marker-genes.html#cb123-5" tabindex="-1"></a>    <span class="at">bias =</span> <span class="fu">c</span>(<span class="st">&quot;TSSEnrichment&quot;</span>, <span class="st">&quot;log10(nFrags)&quot;</span>),</span>
<span id="cb123-6"><a href="identifying-marker-genes.html#cb123-6" tabindex="-1"></a>    <span class="at">testMethod =</span> <span class="st">&quot;wilcoxon&quot;</span></span>
<span id="cb123-7"><a href="identifying-marker-genes.html#cb123-7" tabindex="-1"></a>)</span>
<span id="cb123-8"><a href="identifying-marker-genes.html#cb123-8" tabindex="-1"></a><span class="do">## ArchR logging to : ArchRLogs/ArchR-getMarkerFeatures-1a140af79ea-Date-2025-01-23_Time-20-16-01.415005.log</span></span>
<span id="cb123-9"><a href="identifying-marker-genes.html#cb123-9" tabindex="-1"></a><span class="do">## If there is an issue, please report to github with logFile!</span></span>
<span id="cb123-10"><a href="identifying-marker-genes.html#cb123-10" tabindex="-1"></a><span class="do">## MatrixClass = Sparse.Double.Matrix</span></span>
<span id="cb123-11"><a href="identifying-marker-genes.html#cb123-11" tabindex="-1"></a><span class="do">## 2025-01-23 20:16:02.335737 : Matching Known Biases, 0.004 mins elapsed.</span></span>
<span id="cb123-12"><a href="identifying-marker-genes.html#cb123-12" tabindex="-1"></a><span class="do">## ###########</span></span>
<span id="cb123-13"><a href="identifying-marker-genes.html#cb123-13" tabindex="-1"></a><span class="do">## 2025-01-23 20:17:51.255706 : Completed Pairwise Tests, 1.819 mins elapsed.</span></span>
<span id="cb123-14"><a href="identifying-marker-genes.html#cb123-14" tabindex="-1"></a><span class="do">## ###########</span></span>
<span id="cb123-15"><a href="identifying-marker-genes.html#cb123-15" tabindex="-1"></a><span class="do">## ArchR logging successful to : ArchRLogs/ArchR-getMarkerFeatures-1a140af79ea-Date-2025-01-23_Time-20-16-01.415005.log</span></span></code></pre></div>
<p>This function returns a <code>SummarizedExperiment</code> object containing relevant information on the marker features identified. This type of return value is common in ArchR and is one of the key ways that ArchR enables downstream data analysis. <code>SummarizedExperiment</code> objects are similar to a matrix where rows represent features of interest (i.e. genes) and columns represent samples. A <code>SummarizedExperiment</code> object contains one or more assays, each represented by a matrix-like object of numeric data, and metadata that applies to the rows or columns of the assays matrices. It is beyond the scope of this tutorial to dive into <code>SummarizedExperiment</code> objects but check out the <a href="https://bioconductor.org/packages/release/bioc/html/SummarizedExperiment.html">bioconductor page</a> if you need more information.</p>
<p>We can get a list of <code>DataFrame</code> objects, one for each of our clusters, containing the relevant marker features using the <code>getMarkers()</code> function:</p>
<div class="sourceCode" id="cb124"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb124-1"><a href="identifying-marker-genes.html#cb124-1" tabindex="-1"></a>markerList <span class="ot">&lt;-</span> <span class="fu">getMarkers</span>(markersGS, <span class="at">cutOff =</span> <span class="st">&quot;FDR &lt;= 0.01 &amp; Log2FC &gt;= 1.25&quot;</span>)</span>
<span id="cb124-2"><a href="identifying-marker-genes.html#cb124-2" tabindex="-1"></a>markerList<span class="sc">$</span>C6</span>
<span id="cb124-3"><a href="identifying-marker-genes.html#cb124-3" tabindex="-1"></a><span class="do">## DataFrame with 460 rows and 9 columns</span></span>
<span id="cb124-4"><a href="identifying-marker-genes.html#cb124-4" tabindex="-1"></a><span class="do">##       seqnames     start       end    strand        name       idx    Log2FC</span></span>
<span id="cb124-5"><a href="identifying-marker-genes.html#cb124-5" tabindex="-1"></a><span class="do">##          &lt;Rle&gt; &lt;integer&gt; &lt;integer&gt; &lt;integer&gt; &lt;character&gt; &lt;integer&gt; &lt;numeric&gt;</span></span>
<span id="cb124-6"><a href="identifying-marker-genes.html#cb124-6" tabindex="-1"></a><span class="do">## 14724    chr22  33454377  32908540         2        SYN3       266   1.50383</span></span>
<span id="cb124-7"><a href="identifying-marker-genes.html#cb124-7" tabindex="-1"></a><span class="do">## 7180     chr15  33603177  34158303         1        RYR3       143   1.61610</span></span>
<span id="cb124-8"><a href="identifying-marker-genes.html#cb124-8" tabindex="-1"></a><span class="do">## 15180     chr3  37493813  37861281         1       ITGA9       190   1.42296</span></span>
<span id="cb124-9"><a href="identifying-marker-genes.html#cb124-9" tabindex="-1"></a><span class="do">## 730       chr1  47655771  47649261         2    PDZK1IP1       730   2.31953</span></span>
<span id="cb124-10"><a href="identifying-marker-genes.html#cb124-10" tabindex="-1"></a><span class="do">## 4691     chr12   6055398   5671817         2        ANO2        53   1.66399</span></span>
<span id="cb124-11"><a href="identifying-marker-genes.html#cb124-11" tabindex="-1"></a><span class="do">## ...        ...       ...       ...       ...         ...       ...       ...</span></span>
<span id="cb124-12"><a href="identifying-marker-genes.html#cb124-12" tabindex="-1"></a><span class="do">## 4722     chr12   6949118   6956557         1        GNB3        84   1.72832</span></span>
<span id="cb124-13"><a href="identifying-marker-genes.html#cb124-13" tabindex="-1"></a><span class="do">## 12015    chr19  58180303  58190520         1      ZSCAN4      1563   1.30289</span></span>
<span id="cb124-14"><a href="identifying-marker-genes.html#cb124-14" tabindex="-1"></a><span class="do">## 17723     chr5 135528851 135527156         2      SMIM32       583   1.57534</span></span>
<span id="cb124-15"><a href="identifying-marker-genes.html#cb124-15" tabindex="-1"></a><span class="do">## 18899     chr6  78173120  78171948         2       HTR1B       740   1.25133</span></span>
<span id="cb124-16"><a href="identifying-marker-genes.html#cb124-16" tabindex="-1"></a><span class="do">## 22727     chrX 101112549 101087085         2        NXF5       510   1.82215</span></span>
<span id="cb124-17"><a href="identifying-marker-genes.html#cb124-17" tabindex="-1"></a><span class="do">##               FDR  MeanDiff</span></span>
<span id="cb124-18"><a href="identifying-marker-genes.html#cb124-18" tabindex="-1"></a><span class="do">##         &lt;numeric&gt; &lt;numeric&gt;</span></span>
<span id="cb124-19"><a href="identifying-marker-genes.html#cb124-19" tabindex="-1"></a><span class="do">## 14724 1.00235e-28  1.258169</span></span>
<span id="cb124-20"><a href="identifying-marker-genes.html#cb124-20" tabindex="-1"></a><span class="do">## 7180  1.12058e-22  0.913549</span></span>
<span id="cb124-21"><a href="identifying-marker-genes.html#cb124-21" tabindex="-1"></a><span class="do">## 15180 6.17469e-21  0.908384</span></span>
<span id="cb124-22"><a href="identifying-marker-genes.html#cb124-22" tabindex="-1"></a><span class="do">## 730   1.52549e-20  0.758720</span></span>
<span id="cb124-23"><a href="identifying-marker-genes.html#cb124-23" tabindex="-1"></a><span class="do">## 4691  1.52549e-20  0.588930</span></span>
<span id="cb124-24"><a href="identifying-marker-genes.html#cb124-24" tabindex="-1"></a><span class="do">## ...           ...       ...</span></span>
<span id="cb124-25"><a href="identifying-marker-genes.html#cb124-25" tabindex="-1"></a><span class="do">## 4722   0.00934267 0.2275534</span></span>
<span id="cb124-26"><a href="identifying-marker-genes.html#cb124-26" tabindex="-1"></a><span class="do">## 12015  0.00940705 0.0876258</span></span>
<span id="cb124-27"><a href="identifying-marker-genes.html#cb124-27" tabindex="-1"></a><span class="do">## 17723  0.00941627 0.2378894</span></span>
<span id="cb124-28"><a href="identifying-marker-genes.html#cb124-28" tabindex="-1"></a><span class="do">## 18899  0.00960201 0.4154427</span></span>
<span id="cb124-29"><a href="identifying-marker-genes.html#cb124-29" tabindex="-1"></a><span class="do">## 22727  0.00987594 0.0780624</span></span></code></pre></div>
<p>To visualize all of the marker features simultaneously, we can create a heatmap using the <code>markerHeatmap()</code> function, optionally supplying some marker genes to label on the heatmap via the <code>labelMarkers</code> parameter.</p>
<div class="sourceCode" id="cb125"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb125-1"><a href="identifying-marker-genes.html#cb125-1" tabindex="-1"></a>markerGenes  <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb125-2"><a href="identifying-marker-genes.html#cb125-2" tabindex="-1"></a>    <span class="st">&quot;CD34&quot;</span>, <span class="co">#Early Progenitor</span></span>
<span id="cb125-3"><a href="identifying-marker-genes.html#cb125-3" tabindex="-1"></a>    <span class="st">&quot;GATA1&quot;</span>, <span class="co">#Erythroid</span></span>
<span id="cb125-4"><a href="identifying-marker-genes.html#cb125-4" tabindex="-1"></a>    <span class="st">&quot;PAX5&quot;</span>, <span class="st">&quot;MS4A1&quot;</span>, <span class="st">&quot;EBF1&quot;</span>, <span class="st">&quot;MME&quot;</span>, <span class="co">#B-Cell Trajectory</span></span>
<span id="cb125-5"><a href="identifying-marker-genes.html#cb125-5" tabindex="-1"></a>    <span class="st">&quot;CD14&quot;</span>, <span class="st">&quot;CEBPB&quot;</span>, <span class="st">&quot;MPO&quot;</span>, <span class="co">#Monocytes</span></span>
<span id="cb125-6"><a href="identifying-marker-genes.html#cb125-6" tabindex="-1"></a>    <span class="st">&quot;IRF8&quot;</span>, </span>
<span id="cb125-7"><a href="identifying-marker-genes.html#cb125-7" tabindex="-1"></a>    <span class="st">&quot;CD3D&quot;</span>, <span class="st">&quot;CD8A&quot;</span>, <span class="st">&quot;TBX21&quot;</span>, <span class="st">&quot;IL7R&quot;</span> <span class="co">#TCells</span></span>
<span id="cb125-8"><a href="identifying-marker-genes.html#cb125-8" tabindex="-1"></a>  )</span>
<span id="cb125-9"><a href="identifying-marker-genes.html#cb125-9" tabindex="-1"></a></span>
<span id="cb125-10"><a href="identifying-marker-genes.html#cb125-10" tabindex="-1"></a>heatmapGS <span class="ot">&lt;-</span> <span class="fu">markerHeatmap</span>(</span>
<span id="cb125-11"><a href="identifying-marker-genes.html#cb125-11" tabindex="-1"></a>  <span class="at">seMarker =</span> markersGS, </span>
<span id="cb125-12"><a href="identifying-marker-genes.html#cb125-12" tabindex="-1"></a>  <span class="at">cutOff =</span> <span class="st">&quot;FDR &lt;= 0.01 &amp; Log2FC &gt;= 1.25&quot;</span>, </span>
<span id="cb125-13"><a href="identifying-marker-genes.html#cb125-13" tabindex="-1"></a>  <span class="at">labelMarkers =</span> markerGenes,</span>
<span id="cb125-14"><a href="identifying-marker-genes.html#cb125-14" tabindex="-1"></a>  <span class="at">transpose =</span> <span class="cn">TRUE</span></span>
<span id="cb125-15"><a href="identifying-marker-genes.html#cb125-15" tabindex="-1"></a>)</span>
<span id="cb125-16"><a href="identifying-marker-genes.html#cb125-16" tabindex="-1"></a><span class="do">## Warning in markerHeatmap(seMarker = markersGS, cutOff = &quot;FDR &lt;= 0.01 &amp; Log2FC &gt;= 1.25&quot;, : &#39;markerHeatmap&#39; is deprecated.</span></span>
<span id="cb125-17"><a href="identifying-marker-genes.html#cb125-17" tabindex="-1"></a><span class="do">## Use &#39;plotMarkerHeatmap&#39; instead.</span></span>
<span id="cb125-18"><a href="identifying-marker-genes.html#cb125-18" tabindex="-1"></a><span class="do">## See help(&quot;Deprecated&quot;)</span></span>
<span id="cb125-19"><a href="identifying-marker-genes.html#cb125-19" tabindex="-1"></a><span class="do">## ArchR logging to : ArchRLogs/ArchR-plotMarkerHeatmap-1a168e860a9-Date-2025-01-23_Time-20-17-52.300759.log</span></span>
<span id="cb125-20"><a href="identifying-marker-genes.html#cb125-20" tabindex="-1"></a><span class="do">## If there is an issue, please report to github with logFile!</span></span>
<span id="cb125-21"><a href="identifying-marker-genes.html#cb125-21" tabindex="-1"></a><span class="do">## Printing Top Marker Genes:</span></span>
<span id="cb125-22"><a href="identifying-marker-genes.html#cb125-22" tabindex="-1"></a><span class="do">## C1:</span></span>
<span id="cb125-23"><a href="identifying-marker-genes.html#cb125-23" tabindex="-1"></a><span class="do">##  LOC100288069, RBP7, ARHGEF10L, CELA3A, LINC01226, C1orf185, ZFYVE9, MIR1262, F3, HSD3BP4, SRGAP2D, SNX27, S100A9, FCGR2C, MIR556</span></span>
<span id="cb125-24"><a href="identifying-marker-genes.html#cb125-24" tabindex="-1"></a><span class="do">## C2:</span></span>
<span id="cb125-25"><a href="identifying-marker-genes.html#cb125-25" tabindex="-1"></a><span class="do">##  PERM1, RHD, FGR, PPIEL, BTBD19, TCTEX1D4, L1TD1, MIR3671, GPR88, CASQ2, NHLH2, SLC22A15, IGSF3, FCGR1CP, LINC00869</span></span>
<span id="cb125-26"><a href="identifying-marker-genes.html#cb125-26" tabindex="-1"></a><span class="do">## C3:</span></span>
<span id="cb125-27"><a href="identifying-marker-genes.html#cb125-27" tabindex="-1"></a><span class="do">##  FAM131C, PLA2G5, IFNLR1, NCMAP, TSPAN1, TTC39A, SLC25A3P1, GBP1P1, GFI1, LOC100129046, S100A4, S100A16, BCAN, FCRL5, FCRL4</span></span>
<span id="cb125-28"><a href="identifying-marker-genes.html#cb125-28" tabindex="-1"></a><span class="do">## C4:</span></span>
<span id="cb125-29"><a href="identifying-marker-genes.html#cb125-29" tabindex="-1"></a><span class="do">##  LINC01342, MAD2L2, LDLRAP1, ITGB3BP, PGM1, ROR1, TTC24, FCRLB, MIR1231, KLHDC8A, LYPLAL1, PGBD5, OPN3, OR2L13, ZNF22</span></span>
<span id="cb125-30"><a href="identifying-marker-genes.html#cb125-30" tabindex="-1"></a><span class="do">## C5:</span></span>
<span id="cb125-31"><a href="identifying-marker-genes.html#cb125-31" tabindex="-1"></a><span class="do">##  TTLL10, KAZN, TFAP2E, RIMS3, TRABD2B, PGLYRP3, UCK2, TNR, KIAA1614, LEMD1, SERTAD4-AS1, CCDC3, DEPP1, PRKG1-AS1, SH2D4B</span></span>
<span id="cb125-32"><a href="identifying-marker-genes.html#cb125-32" tabindex="-1"></a><span class="do">## C6:</span></span>
<span id="cb125-33"><a href="identifying-marker-genes.html#cb125-33" tabindex="-1"></a><span class="do">##  PAX7, RPS14P3, FAM43B, MIR378F, GRHL3, TRIM63, GJA4, RHBDL2, PPIE, MPL, PDZK1IP1, FOXE3, DNAJC6, OLFM3, PHTF1</span></span>
<span id="cb125-34"><a href="identifying-marker-genes.html#cb125-34" tabindex="-1"></a><span class="do">## C7:</span></span>
<span id="cb125-35"><a href="identifying-marker-genes.html#cb125-35" tabindex="-1"></a><span class="do">##  CFAP74, MIR34A, DMRTB1, MGAT4EP, LOC148709, EIF5AL1, RASSF10, SLC6A5, MIR4488, LRRC10B, SHANK2-AS1, CACNA1C-AS4, DCLK1, HTR2A, LRRC36</span></span>
<span id="cb125-36"><a href="identifying-marker-genes.html#cb125-36" tabindex="-1"></a><span class="do">## C8:</span></span>
<span id="cb125-37"><a href="identifying-marker-genes.html#cb125-37" tabindex="-1"></a><span class="do">##  LINC02593, PRDM16, MIR4251, MIR4689, TMEM51-AS1, PADI3, NBL1, RAP1GAP, LDLRAD2, HSPG2, MIR4253, CATSPER4, FAM229A, EPHA10, EDN2</span></span>
<span id="cb125-38"><a href="identifying-marker-genes.html#cb125-38" tabindex="-1"></a><span class="do">## C9:</span></span>
<span id="cb125-39"><a href="identifying-marker-genes.html#cb125-39" tabindex="-1"></a><span class="do">##  ID3, CNR2, BEND5, SHISAL2A, CLCA3P, LOC729970, TMEM56, MFSD14A, LRRC39, SLC16A4, DRAM2, PIFO, FCRL3, FCRL2, DNM3OS</span></span>
<span id="cb125-40"><a href="identifying-marker-genes.html#cb125-40" tabindex="-1"></a><span class="do">## C10:</span></span>
<span id="cb125-41"><a href="identifying-marker-genes.html#cb125-41" tabindex="-1"></a><span class="do">##  MMP23B, RUNX3, IL12RB2, GBP2, GBP5, LOC729930, TGFBR3, VCAM1, HIST2H2BA, FCGR1B, CD160, PDZK1, SH2D2A, NTRK1, PYHIN1</span></span>
<span id="cb125-42"><a href="identifying-marker-genes.html#cb125-42" tabindex="-1"></a><span class="do">## C11:</span></span>
<span id="cb125-43"><a href="identifying-marker-genes.html#cb125-43" tabindex="-1"></a><span class="do">##  RCAN3AS, CRYBG2, PPP1R8, LCK, HOOK1, SYDE2, SNORD21, CHI3L2, CD2, HSPA7, BLZF1, SELL, RNU6-72P, PRKCQ-AS1, TAF3</span></span>
<span id="cb125-44"><a href="identifying-marker-genes.html#cb125-44" tabindex="-1"></a><span class="do">## C12:</span></span>
<span id="cb125-45"><a href="identifying-marker-genes.html#cb125-45" tabindex="-1"></a><span class="do">##  SNORA59B, C1orf194, AMIGO1, GPR61, CRABP2, GAS5-AS1, POLR2G, ZPR1, RPL13P5, PIP4K2C, DTX3, LINC02291, MTRNR2L4, TNK1, HOXB2</span></span>
<span id="cb125-46"><a href="identifying-marker-genes.html#cb125-46" tabindex="-1"></a><span class="do">## Identified 2736 markers!</span></span>
<span id="cb125-47"><a href="identifying-marker-genes.html#cb125-47" tabindex="-1"></a><span class="do">##  [1] &quot;CD34&quot;  &quot;GATA1&quot; &quot;PAX5&quot;  &quot;MS4A1&quot; &quot;EBF1&quot;  &quot;MME&quot;   &quot;CD14&quot;  &quot;CEBPB&quot; &quot;MPO&quot;  </span></span>
<span id="cb125-48"><a href="identifying-marker-genes.html#cb125-48" tabindex="-1"></a><span class="do">## [10] &quot;IRF8&quot;  &quot;CD3D&quot;  &quot;CD8A&quot;  &quot;TBX21&quot; &quot;IL7R&quot;</span></span>
<span id="cb125-49"><a href="identifying-marker-genes.html#cb125-49" tabindex="-1"></a><span class="do">## Adding Annotations..</span></span>
<span id="cb125-50"><a href="identifying-marker-genes.html#cb125-50" tabindex="-1"></a><span class="do">## Preparing Main Heatmap..</span></span>
<span id="cb125-51"><a href="identifying-marker-genes.html#cb125-51" tabindex="-1"></a><span class="do">## &#39;magick&#39; package is suggested to install to give better rasterization.</span></span>
<span id="cb125-52"><a href="identifying-marker-genes.html#cb125-52" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb125-53"><a href="identifying-marker-genes.html#cb125-53" tabindex="-1"></a><span class="do">## Set `ht_opt$message = FALSE` to turn off this message.</span></span>
<span id="cb125-54"><a href="identifying-marker-genes.html#cb125-54" tabindex="-1"></a><span class="do">## ArchR logging successful to : ArchRLogs/ArchR-plotMarkerHeatmap-1a168e860a9-Date-2025-01-23_Time-20-17-52.300759.log</span></span></code></pre></div>
<p>To plot this heatmap, we can use the <code>ComplexHeatmap::draw()</code> function because the <code>heatmapGS</code> object is actually a list of heatmaps:</p>
<div class="sourceCode" id="cb126"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb126-1"><a href="identifying-marker-genes.html#cb126-1" tabindex="-1"></a>ComplexHeatmap<span class="sc">::</span><span class="fu">draw</span>(heatmapGS, <span class="at">heatmap_legend_side =</span> <span class="st">&quot;bot&quot;</span>, <span class="at">annotation_legend_side =</span> <span class="st">&quot;bot&quot;</span>)</span></code></pre></div>
<p><img src="ArchR_files/figure-html/unnamed-chunk-123-1.svg" width="672" /></p>
<p>Many heatmaps in ArchR utilize the <code>ComplexHeatmap</code> package which provides a very versatile and flexible framework for plotting heatmaps. If you need to change something about how the heatmap looks when plotted, you should consult the documentation for <code>ComplexHeatmap</code>. For example, if you want to change the order in which the heatmap rows are plotted, you can edit the <code>row_order</code> slot of the heatmap object and re-plot it:</p>
<div class="sourceCode" id="cb127"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb127-1"><a href="identifying-marker-genes.html#cb127-1" tabindex="-1"></a>heatmapGS<span class="sc">@</span>row_order <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">10</span>,<span class="dv">11</span>,<span class="dv">12</span>,<span class="dv">3</span>,<span class="dv">4</span>,<span class="dv">5</span>,<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">6</span>,<span class="dv">8</span>,<span class="dv">7</span>,<span class="dv">9</span>)</span>
<span id="cb127-2"><a href="identifying-marker-genes.html#cb127-2" tabindex="-1"></a>ComplexHeatmap<span class="sc">::</span><span class="fu">draw</span>(heatmapGS, <span class="at">heatmap_legend_side =</span> <span class="st">&quot;bot&quot;</span>, <span class="at">annotation_legend_side =</span> <span class="st">&quot;bot&quot;</span>)</span></code></pre></div>
<p><img src="ArchR_files/figure-html/unnamed-chunk-124-1.svg" width="672" /></p>
<p>To save an editable vectorized version of this plot, we use <code>plotPDF()</code>.</p>
<div class="sourceCode" id="cb128"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb128-1"><a href="identifying-marker-genes.html#cb128-1" tabindex="-1"></a><span class="fu">plotPDF</span>(heatmapGS, <span class="at">name =</span> <span class="st">&quot;GeneScores-Marker-Heatmap&quot;</span>, <span class="at">width =</span> <span class="dv">8</span>, <span class="at">height =</span> <span class="dv">6</span>, <span class="at">ArchRProj =</span> projHeme2, <span class="at">addDOC =</span> <span class="cn">FALSE</span>)</span>
<span id="cb128-2"><a href="identifying-marker-genes.html#cb128-2" tabindex="-1"></a><span class="do">## Plotting ComplexHeatmap!</span></span></code></pre></div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="identification-of-marker-features.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="visualizing-marker-genes-on-an-embedding.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": null
},
"fontsettings": null,
"edit": null,
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "none"
},
"info": false
});
});
</script>

</body>

</html>
