<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>21.1 Importing data and setting up a Multiome project | ArchR: Robust and scaleable analysis of single-cell chromatin accessibility data.</title>
  <meta name="description" content="A guide to ArchR" />
  <meta name="generator" content="bookdown 0.41 and GitBook 2.6.7" />

  <meta property="og:title" content="21.1 Importing data and setting up a Multiome project | ArchR: Robust and scaleable analysis of single-cell chromatin accessibility data." />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="A guide to ArchR" />
  <meta name="github-repo" content="GreenleafLab/ArchR" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="21.1 Importing data and setting up a Multiome project | ArchR: Robust and scaleable analysis of single-cell chromatin accessibility data." />
  
  <meta name="twitter:description" content="A guide to ArchR" />
  

<meta name="author" content="Jeffrey Granja and Ryan Corces" />


<meta name="date" content="2025-02-06" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="images/favicon.ico" type="image/x-icon" />
<link rel="prev" href="multiomic-data-analysis-in-archr.html"/>
<link rel="next" href="analysis-of-multiome-data-in-archr.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<link href="libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="../index.html">Back to ArchR Home</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><img src="images/ArchR_Logo_Integrated.png" alt="" width="200"></a></li>
<li class="chapter" data-level="1" data-path="debugging-and-troubleshooting-in-archr.html"><a href="debugging-and-troubleshooting-in-archr.html"><i class="fa fa-check"></i><b>1</b> Debugging and Troubleshooting in ArchR</a></li>
<li class="chapter" data-level="2" data-path="manage-archrs-dependencies.html"><a href="manage-archrs-dependencies.html"><i class="fa fa-check"></i><b>2</b> Manage ArchR’s Dependencies</a>
<ul>
<li class="chapter" data-level="2.1" data-path="using-renv-to-manage-dependencies.html"><a href="using-renv-to-manage-dependencies.html"><i class="fa fa-check"></i><b>2.1</b> Using renv to manage dependencies</a></li>
<li class="chapter" data-level="2.2" data-path="other-environment-managers.html"><a href="other-environment-managers.html"><i class="fa fa-check"></i><b>2.2</b> Other environment managers</a></li>
<li class="chapter" data-level="2.3" data-path="using-docker-for-pre-compiled-environment-setup.html"><a href="using-docker-for-pre-compiled-environment-setup.html"><i class="fa fa-check"></i><b>2.3</b> Using Docker for pre-compiled environment setup</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="getting-started-with-archr.html"><a href="getting-started-with-archr.html"><i class="fa fa-check"></i><b>3</b> Getting Started with ArchR</a>
<ul>
<li class="chapter" data-level="3.1" data-path="a-brief-primer-on-atac-seq-terminology.html"><a href="a-brief-primer-on-atac-seq-terminology.html"><i class="fa fa-check"></i><b>3.1</b> A Brief Primer on ATAC-seq Terminology</a></li>
<li class="chapter" data-level="3.2" data-path="why-use-archr.html"><a href="why-use-archr.html"><i class="fa fa-check"></i><b>3.2</b> Why use ArchR?</a></li>
<li class="chapter" data-level="3.3" data-path="what-is-an-arrow-file-archrproject.html"><a href="what-is-an-arrow-file-archrproject.html"><i class="fa fa-check"></i><b>3.3</b> What is an Arrow file / <code>ArchRProject</code>?</a></li>
<li class="chapter" data-level="3.4" data-path="input-file-types-in-archr.html"><a href="input-file-types-in-archr.html"><i class="fa fa-check"></i><b>3.4</b> Input File Types in ArchR</a></li>
<li class="chapter" data-level="3.5" data-path="getting-set-up.html"><a href="getting-set-up.html"><i class="fa fa-check"></i><b>3.5</b> Getting Set Up</a></li>
<li class="chapter" data-level="3.6" data-path="creating-arrow-files.html"><a href="creating-arrow-files.html"><i class="fa fa-check"></i><b>3.6</b> Creating Arrow Files</a></li>
<li class="chapter" data-level="3.7" data-path="per-cell-quality-control.html"><a href="per-cell-quality-control.html"><i class="fa fa-check"></i><b>3.7</b> Per-cell Quality Control</a></li>
<li class="chapter" data-level="3.8" data-path="using-bam-files-for-arrow-file-creation.html"><a href="using-bam-files-for-arrow-file-creation.html"><i class="fa fa-check"></i><b>3.8</b> Using BAM files for Arrow File creation</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="doublet-inference-with-archr.html"><a href="doublet-inference-with-archr.html"><i class="fa fa-check"></i><b>4</b> Doublet Inference with ArchR</a>
<ul>
<li class="chapter" data-level="4.1" data-path="how-does-doublet-identification-work-in-archr.html"><a href="how-does-doublet-identification-work-in-archr.html"><i class="fa fa-check"></i><b>4.1</b> How does doublet identification work in ArchR?</a></li>
<li class="chapter" data-level="4.2" data-path="inferring-scatac-seq-doublets-with-archr.html"><a href="inferring-scatac-seq-doublets-with-archr.html"><i class="fa fa-check"></i><b>4.2</b> Inferring scATAC-seq Doublets with ArchR</a></li>
<li class="chapter" data-level="4.3" data-path="using-demuxlet-with-archr.html"><a href="using-demuxlet-with-archr.html"><i class="fa fa-check"></i><b>4.3</b> Using <em>demuxlet</em> with ArchR</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="creating-an-archrproject.html"><a href="creating-an-archrproject.html"><i class="fa fa-check"></i><b>5</b> Creating an ArchRProject</a>
<ul>
<li class="chapter" data-level="5.1" data-path="creating-an-archrproject-1.html"><a href="creating-an-archrproject-1.html"><i class="fa fa-check"></i><b>5.1</b> Creating An ArchRProject</a></li>
<li class="chapter" data-level="5.2" data-path="manipulating-an-archrproject.html"><a href="manipulating-an-archrproject.html"><i class="fa fa-check"></i><b>5.2</b> Manipulating An ArchRProject</a></li>
<li class="chapter" data-level="5.3" data-path="plotting-sample-statistics-from-an-archrproject.html"><a href="plotting-sample-statistics-from-an-archrproject.html"><i class="fa fa-check"></i><b>5.3</b> Plotting Sample Statistics from an ArchRProject</a></li>
<li class="chapter" data-level="5.4" data-path="plotting-sample-fragment-size-distribution-and-tss-enrichment-profiles..html"><a href="plotting-sample-fragment-size-distribution-and-tss-enrichment-profiles..html"><i class="fa fa-check"></i><b>5.4</b> Plotting Sample Fragment Size Distribution and TSS Enrichment Profiles.</a></li>
<li class="chapter" data-level="5.5" data-path="saving-and-loading-an-archrproject.html"><a href="saving-and-loading-an-archrproject.html"><i class="fa fa-check"></i><b>5.5</b> Saving and Loading an <code>ArchRProject</code></a></li>
<li class="chapter" data-level="5.6" data-path="filtering-doublets-from-an-archrproject.html"><a href="filtering-doublets-from-an-archrproject.html"><i class="fa fa-check"></i><b>5.6</b> Filtering Doublets from an ArchRProject</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="dimensionality-reduction-with-archr.html"><a href="dimensionality-reduction-with-archr.html"><i class="fa fa-check"></i><b>6</b> Dimensionality Reduction with ArchR</a>
<ul>
<li class="chapter" data-level="6.1" data-path="archrs-lsi-implementation.html"><a href="archrs-lsi-implementation.html"><i class="fa fa-check"></i><b>6.1</b> ArchR’s LSI Implementation</a></li>
<li class="chapter" data-level="6.2" data-path="iterative-latent-semantic-indexing-lsi.html"><a href="iterative-latent-semantic-indexing-lsi.html"><i class="fa fa-check"></i><b>6.2</b> Iterative Latent Semantic Indexing (LSI)</a></li>
<li class="chapter" data-level="6.3" data-path="estimated-lsi.html"><a href="estimated-lsi.html"><i class="fa fa-check"></i><b>6.3</b> Estimated LSI</a></li>
<li class="chapter" data-level="6.4" data-path="batch-effect-correction-wtih-harmony.html"><a href="batch-effect-correction-wtih-harmony.html"><i class="fa fa-check"></i><b>6.4</b> Batch Effect Correction wtih Harmony</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="clustering-with-archr.html"><a href="clustering-with-archr.html"><i class="fa fa-check"></i><b>7</b> Clustering with ArchR</a>
<ul>
<li class="chapter" data-level="7.1" data-path="clustering-using-seurats-findclusters-function.html"><a href="clustering-using-seurats-findclusters-function.html"><i class="fa fa-check"></i><b>7.1</b> Clustering using Seurat’s <code>FindClusters()</code> function</a></li>
<li class="chapter" data-level="7.2" data-path="clustering-using-scran.html"><a href="clustering-using-scran.html"><i class="fa fa-check"></i><b>7.2</b> Clustering using <code>scran</code></a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="single-cell-embeddings.html"><a href="single-cell-embeddings.html"><i class="fa fa-check"></i><b>8</b> Single-cell Embeddings</a>
<ul>
<li class="chapter" data-level="8.1" data-path="uniform-manifold-approximation-and-projection-umap.html"><a href="uniform-manifold-approximation-and-projection-umap.html"><i class="fa fa-check"></i><b>8.1</b> Uniform Manifold Approximation and Projection (UMAP)</a></li>
<li class="chapter" data-level="8.2" data-path="t-stocastic-neighbor-embedding-t-sne.html"><a href="t-stocastic-neighbor-embedding-t-sne.html"><i class="fa fa-check"></i><b>8.2</b> t-Stocastic Neighbor Embedding (t-SNE)</a></li>
<li class="chapter" data-level="8.3" data-path="dimensionality-reduction-after-harmony.html"><a href="dimensionality-reduction-after-harmony.html"><i class="fa fa-check"></i><b>8.3</b> Dimensionality Reduction After Harmony</a></li>
<li class="chapter" data-level="8.4" data-path="highlighting-specific-cells-on-an-embedding.html"><a href="highlighting-specific-cells-on-an-embedding.html"><i class="fa fa-check"></i><b>8.4</b> Highlighting specific cells on an embedding</a></li>
<li class="chapter" data-level="8.5" data-path="importing-an-embedding-from-external-software.html"><a href="importing-an-embedding-from-external-software.html"><i class="fa fa-check"></i><b>8.5</b> Importing an embedding from external software</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="gene-scores-and-marker-genes-with-archr.html"><a href="gene-scores-and-marker-genes-with-archr.html"><i class="fa fa-check"></i><b>9</b> Gene Scores and Marker Genes with ArchR</a>
<ul>
<li class="chapter" data-level="9.1" data-path="calculating-gene-scores-in-archr.html"><a href="calculating-gene-scores-in-archr.html"><i class="fa fa-check"></i><b>9.1</b> Calculating Gene Scores in ArchR</a></li>
<li class="chapter" data-level="9.2" data-path="identification-of-marker-features.html"><a href="identification-of-marker-features.html"><i class="fa fa-check"></i><b>9.2</b> Identification of Marker Features</a></li>
<li class="chapter" data-level="9.3" data-path="identifying-marker-genes.html"><a href="identifying-marker-genes.html"><i class="fa fa-check"></i><b>9.3</b> Identifying Marker Genes</a></li>
<li class="chapter" data-level="9.4" data-path="visualizing-marker-genes-on-an-embedding.html"><a href="visualizing-marker-genes-on-an-embedding.html"><i class="fa fa-check"></i><b>9.4</b> Visualizing Marker Genes on an Embedding</a></li>
<li class="chapter" data-level="9.5" data-path="marker-genes-imputation-with-magic.html"><a href="marker-genes-imputation-with-magic.html"><i class="fa fa-check"></i><b>9.5</b> Marker Genes Imputation with MAGIC</a></li>
<li class="chapter" data-level="9.6" data-path="module-scores.html"><a href="module-scores.html"><i class="fa fa-check"></i><b>9.6</b> Module Scores</a></li>
<li class="chapter" data-level="9.7" data-path="track-plotting-with-archrbrowser.html"><a href="track-plotting-with-archrbrowser.html"><i class="fa fa-check"></i><b>9.7</b> Track Plotting with ArchRBrowser</a></li>
<li class="chapter" data-level="9.8" data-path="launching-the-interactive-archrbrowser.html"><a href="launching-the-interactive-archrbrowser.html"><i class="fa fa-check"></i><b>9.8</b> Launching the Interactive ArchRBrowser</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="defining-cluster-identity-with-scrna-seq.html"><a href="defining-cluster-identity-with-scrna-seq.html"><i class="fa fa-check"></i><b>10</b> Defining Cluster Identity with scRNA-seq</a>
<ul>
<li class="chapter" data-level="10.1" data-path="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html"><i class="fa fa-check"></i><b>10.1</b> Cross-platform linkage of scATAC-seq cells with scRNA-seq cells</a></li>
<li class="chapter" data-level="10.2" data-path="adding-pseudo-scrna-seq-profiles-for-each-scatac-seq-cell.html"><a href="adding-pseudo-scrna-seq-profiles-for-each-scatac-seq-cell.html"><i class="fa fa-check"></i><b>10.2</b> Adding Pseudo-scRNA-seq profiles for each scATAC-seq cell</a></li>
<li class="chapter" data-level="10.3" data-path="labeling-scatac-seq-clusters-with-scrna-seq-information.html"><a href="labeling-scatac-seq-clusters-with-scrna-seq-information.html"><i class="fa fa-check"></i><b>10.3</b> Labeling scATAC-seq clusters with scRNA-seq information</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="pseudo-bulk-replicates-in-archr.html"><a href="pseudo-bulk-replicates-in-archr.html"><i class="fa fa-check"></i><b>11</b> Pseudo-bulk Replicates in ArchR</a>
<ul>
<li class="chapter" data-level="11.1" data-path="how-does-archr-make-pseudo-bulk-replicates.html"><a href="how-does-archr-make-pseudo-bulk-replicates.html"><i class="fa fa-check"></i><b>11.1</b> How Does ArchR Make Pseudo-bulk Replicates?</a></li>
<li class="chapter" data-level="11.2" data-path="making-pseudo-bulk-replicates.html"><a href="making-pseudo-bulk-replicates.html"><i class="fa fa-check"></i><b>11.2</b> Making Pseudo-bulk Replicates</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="calling-peaks-with-archr.html"><a href="calling-peaks-with-archr.html"><i class="fa fa-check"></i><b>12</b> Calling Peaks with ArchR</a>
<ul>
<li class="chapter" data-level="12.1" data-path="the-iterative-overlap-peak-merging-procedure.html"><a href="the-iterative-overlap-peak-merging-procedure.html"><i class="fa fa-check"></i><b>12.1</b> The Iterative Overlap Peak Merging Procedure</a></li>
<li class="chapter" data-level="12.2" data-path="calling-peaks-w-macs2.html"><a href="calling-peaks-w-macs2.html"><i class="fa fa-check"></i><b>12.2</b> Calling Peaks w/ Macs2</a></li>
<li class="chapter" data-level="12.3" data-path="calling-peaks-w-tilematrix.html"><a href="calling-peaks-w-tilematrix.html"><i class="fa fa-check"></i><b>12.3</b> Calling Peaks w/ TileMatrix</a></li>
<li class="chapter" data-level="12.4" data-path="adding-a-peak-matrix.html"><a href="adding-a-peak-matrix.html"><i class="fa fa-check"></i><b>12.4</b> Adding a Peak Matrix</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="identifying-marker-peaks.html"><a href="identifying-marker-peaks.html"><i class="fa fa-check"></i><b>13</b> Identifying Marker Peaks</a>
<ul>
<li class="chapter" data-level="13.1" data-path="identifying-marker-peaks-with-archr.html"><a href="identifying-marker-peaks-with-archr.html"><i class="fa fa-check"></i><b>13.1</b> Identifying Marker Peaks with ArchR</a></li>
<li class="chapter" data-level="13.2" data-path="plotting-marker-peaks-in-archr.html"><a href="plotting-marker-peaks-in-archr.html"><i class="fa fa-check"></i><b>13.2</b> Plotting Marker Peaks in ArchR</a></li>
<li class="chapter" data-level="13.3" data-path="pairwise-testing-between-groups.html"><a href="pairwise-testing-between-groups.html"><i class="fa fa-check"></i><b>13.3</b> Pairwise Testing Between Groups</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="motif-and-feature-enrichment-with-archr.html"><a href="motif-and-feature-enrichment-with-archr.html"><i class="fa fa-check"></i><b>14</b> Motif and Feature Enrichment with ArchR</a>
<ul>
<li class="chapter" data-level="14.1" data-path="motif-enrichment.html"><a href="motif-enrichment.html"><i class="fa fa-check"></i><b>14.1</b> Motif Enrichment</a></li>
<li class="chapter" data-level="14.2" data-path="archr-enrichment.html"><a href="archr-enrichment.html"><i class="fa fa-check"></i><b>14.2</b> ArchR Enrichment</a></li>
<li class="chapter" data-level="14.3" data-path="custom-enrichment.html"><a href="custom-enrichment.html"><i class="fa fa-check"></i><b>14.3</b> Custom Enrichment</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="chromvar-deviatons-enrichment-with-archr.html"><a href="chromvar-deviatons-enrichment-with-archr.html"><i class="fa fa-check"></i><b>15</b> ChromVAR Deviatons Enrichment with ArchR</a>
<ul>
<li class="chapter" data-level="15.1" data-path="motif-deviations.html"><a href="motif-deviations.html"><i class="fa fa-check"></i><b>15.1</b> Motif Deviations</a></li>
<li class="chapter" data-level="15.2" data-path="archr-and-custom-deviations.html"><a href="archr-and-custom-deviations.html"><i class="fa fa-check"></i><b>15.2</b> ArchR and Custom Deviations</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="footprinting-with-archr.html"><a href="footprinting-with-archr.html"><i class="fa fa-check"></i><b>16</b> Footprinting with ArchR</a>
<ul>
<li class="chapter" data-level="16.1" data-path="motif-footprinting.html"><a href="motif-footprinting.html"><i class="fa fa-check"></i><b>16.1</b> Motif Footprinting</a></li>
<li class="chapter" data-level="16.2" data-path="normalization-of-footprints-for-tn5-bias.html"><a href="normalization-of-footprints-for-tn5-bias.html"><i class="fa fa-check"></i><b>16.2</b> Normalization of Footprints for Tn5 Bias</a></li>
<li class="chapter" data-level="16.3" data-path="feature-footprinting.html"><a href="feature-footprinting.html"><i class="fa fa-check"></i><b>16.3</b> Feature Footprinting</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="integrative-analysis-with-archr.html"><a href="integrative-analysis-with-archr.html"><i class="fa fa-check"></i><b>17</b> Integrative Analysis with ArchR</a>
<ul>
<li class="chapter" data-level="17.1" data-path="creating-low-overlapping-aggregates-of-cells.html"><a href="creating-low-overlapping-aggregates-of-cells.html"><i class="fa fa-check"></i><b>17.1</b> Creating Low-Overlapping Aggregates of Cells</a></li>
<li class="chapter" data-level="17.2" data-path="co-accessibility-with-archr.html"><a href="co-accessibility-with-archr.html"><i class="fa fa-check"></i><b>17.2</b> Co-accessibility with ArchR</a></li>
<li class="chapter" data-level="17.3" data-path="peak2genelinkage-with-archr.html"><a href="peak2genelinkage-with-archr.html"><i class="fa fa-check"></i><b>17.3</b> Peak2GeneLinkage with ArchR</a></li>
<li class="chapter" data-level="17.4" data-path="identification-of-positive-tf-regulators.html"><a href="identification-of-positive-tf-regulators.html"><i class="fa fa-check"></i><b>17.4</b> Identification of Positive TF-Regulators</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="trajectory-analysis-with-archr.html"><a href="trajectory-analysis-with-archr.html"><i class="fa fa-check"></i><b>18</b> Trajectory Analysis with ArchR</a>
<ul>
<li class="chapter" data-level="18.1" data-path="myeloid-trajectory---monocyte-differentiation.html"><a href="myeloid-trajectory---monocyte-differentiation.html"><i class="fa fa-check"></i><b>18.1</b> Myeloid Trajectory - Monocyte Differentiation</a></li>
<li class="chapter" data-level="18.2" data-path="lymphoid-trajectory---b-cell-differentiation.html"><a href="lymphoid-trajectory---b-cell-differentiation.html"><i class="fa fa-check"></i><b>18.2</b> Lymphoid Trajectory - B Cell Differentiation</a></li>
<li class="chapter" data-level="18.3" data-path="monocle3-trajectories.html"><a href="monocle3-trajectories.html"><i class="fa fa-check"></i><b>18.3</b> Monocle3 Trajectories</a></li>
<li class="chapter" data-level="18.4" data-path="slingshot-trajectories.html"><a href="slingshot-trajectories.html"><i class="fa fa-check"></i><b>18.4</b> Slingshot Trajectories</a></li>
<li class="chapter" data-level="18.5" data-path="comparing-the-different-trajectories.html"><a href="comparing-the-different-trajectories.html"><i class="fa fa-check"></i><b>18.5</b> Comparing the different trajectories</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="handling-archr-output.html"><a href="handling-archr-output.html"><i class="fa fa-check"></i><b>19</b> Handling ArchR Output</a>
<ul>
<li class="chapter" data-level="19.1" data-path="exporting-fragment-level-data.html"><a href="exporting-fragment-level-data.html"><i class="fa fa-check"></i><b>19.1</b> Exporting fragment-level data</a></li>
<li class="chapter" data-level="19.2" data-path="exporting-matrix-level-data.html"><a href="exporting-matrix-level-data.html"><i class="fa fa-check"></i><b>19.2</b> Exporting matrix-level data</a></li>
<li class="chapter" data-level="19.3" data-path="exporting-pseudo-bulked-data-to-a-summarizedexperiment.html"><a href="exporting-pseudo-bulked-data-to-a-summarizedexperiment.html"><i class="fa fa-check"></i><b>19.3</b> Exporting pseudo-bulked data to a <code>SummarizedExperiment</code></a></li>
<li class="chapter" data-level="19.4" data-path="exporting-pseudo-bulked-data-to-a-bigwig-file.html"><a href="exporting-pseudo-bulked-data-to-a-bigwig-file.html"><i class="fa fa-check"></i><b>19.4</b> Exporting pseudo-bulked data to a bigWig file</a></li>
</ul></li>
<li class="chapter" data-level="20" data-path="integration-with-bulk-atac-seq.html"><a href="integration-with-bulk-atac-seq.html"><i class="fa fa-check"></i><b>20</b> Integration with bulk ATAC-seq</a>
<ul>
<li class="chapter" data-level="20.1" data-path="projecting-bulk-atac-seq-data.html"><a href="projecting-bulk-atac-seq-data.html"><i class="fa fa-check"></i><b>20.1</b> Projecting bulk ATAC-seq data</a></li>
</ul></li>
<li class="chapter" data-level="21" data-path="multiomic-data-analysis-in-archr.html"><a href="multiomic-data-analysis-in-archr.html"><i class="fa fa-check"></i><b>21</b> Multiomic data analysis in ArchR</a>
<ul>
<li class="chapter" data-level="21.1" data-path="importing-data-and-setting-up-a-multiome-project.html"><a href="importing-data-and-setting-up-a-multiome-project.html"><i class="fa fa-check"></i><b>21.1</b> Importing data and setting up a Multiome project</a></li>
<li class="chapter" data-level="21.2" data-path="analysis-of-multiome-data-in-archr.html"><a href="analysis-of-multiome-data-in-archr.html"><i class="fa fa-check"></i><b>21.2</b> Analysis of multiome data in ArchR</a></li>
</ul></li>
<li class="chapter" data-level="22" data-path="plot-aesthetics-in-archr.html"><a href="plot-aesthetics-in-archr.html"><i class="fa fa-check"></i><b>22</b> Plot aesthetics in ArchR</a>
<ul>
<li class="chapter" data-level="22.1" data-path="using-and-manipulating-palettes-in-archr.html"><a href="using-and-manipulating-palettes-in-archr.html"><i class="fa fa-check"></i><b>22.1</b> Using and manipulating palettes in ArchR</a></li>
</ul></li>
<li class="chapter" data-level="23" data-path="testing-your-archr-installation.html"><a href="testing-your-archr-installation.html"><i class="fa fa-check"></i><b>23</b> Testing Your ArchR Installation</a></li>
<li class="chapter" data-level="24" data-path="session-information.html"><a href="session-information.html"><i class="fa fa-check"></i><b>24</b> Session Information</a></li>
<li class="divider"></li>
<li><a href="https://bookdown.org" target="_blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">ArchR: Robust and scaleable analysis of single-cell chromatin accessibility data.</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="importing-data-and-setting-up-a-multiome-project" class="section level2 hasAnchor" number="21.1">
<h2><span class="header-section-number">21.1</span> Importing data and setting up a Multiome project<a href="importing-data-and-setting-up-a-multiome-project.html#importing-data-and-setting-up-a-multiome-project" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>First, it is worth noting that the datasets used in this chapter are different from the datasets used in the other chapters and we will be creating a new <code>ArchRProject</code> here. For this reason, and to make this chapter a standalone, we will repeat operations from other parts of the manual.</p>
<p>Lets start by loading ArchR and setting up our genome (<code>hg38</code> in this case), our threads (which are system dependent so you should change that based on your computational resources), and our random seed.</p>
<div class="sourceCode" id="cb455"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb455-1"><a href="importing-data-and-setting-up-a-multiome-project.html#cb455-1" tabindex="-1"></a><span class="fu">library</span>(ArchR)</span>
<span id="cb455-2"><a href="importing-data-and-setting-up-a-multiome-project.html#cb455-2" tabindex="-1"></a><span class="fu">addArchRGenome</span>(<span class="st">&quot;hg38&quot;</span>)</span>
<span id="cb455-3"><a href="importing-data-and-setting-up-a-multiome-project.html#cb455-3" tabindex="-1"></a><span class="do">## Setting default genome to Hg38.</span></span>
<span id="cb455-4"><a href="importing-data-and-setting-up-a-multiome-project.html#cb455-4" tabindex="-1"></a><span class="fu">addArchRThreads</span>(<span class="dv">8</span>)</span>
<span id="cb455-5"><a href="importing-data-and-setting-up-a-multiome-project.html#cb455-5" tabindex="-1"></a><span class="do">## Setting default number of Parallel threads to 8.</span></span>
<span id="cb455-6"><a href="importing-data-and-setting-up-a-multiome-project.html#cb455-6" tabindex="-1"></a><span class="fu">addArchRLocking</span>(<span class="at">locking =</span> <span class="cn">TRUE</span>)</span>
<span id="cb455-7"><a href="importing-data-and-setting-up-a-multiome-project.html#cb455-7" tabindex="-1"></a><span class="do">## Setting ArchRLocking to TRUE.</span></span>
<span id="cb455-8"><a href="importing-data-and-setting-up-a-multiome-project.html#cb455-8" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">1</span>)</span></code></pre></div>
<p>We can download the files associated with the multiome tutorial just like we did for the main tutorial but using <code>tutorial = "Multiome"</code>.</p>
<div class="sourceCode" id="cb456"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb456-1"><a href="importing-data-and-setting-up-a-multiome-project.html#cb456-1" tabindex="-1"></a>inputFiles <span class="ot">&lt;-</span> <span class="fu">getTutorialData</span>(<span class="at">tutorial =</span> <span class="st">&quot;Multiome&quot;</span>)</span>
<span id="cb456-2"><a href="importing-data-and-setting-up-a-multiome-project.html#cb456-2" tabindex="-1"></a><span class="do">## Downloading files to Multiome...</span></span></code></pre></div>
<p>If we inspect <code>inputFiles</code> you can see that we have a <code>.fragments.tsv.gz</code> file and a <code>filtered_feature_bc_matrix.h5</code> file for each sample. These two files are both standard outputs of the 10x Genomics <code>cellranger-arc</code> pipeline.</p>
<div class="sourceCode" id="cb457"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb457-1"><a href="importing-data-and-setting-up-a-multiome-project.html#cb457-1" tabindex="-1"></a>inputFiles</span>
<span id="cb457-2"><a href="importing-data-and-setting-up-a-multiome-project.html#cb457-2" tabindex="-1"></a><span class="do">##                                            pbmc_sorted_3k </span></span>
<span id="cb457-3"><a href="importing-data-and-setting-up-a-multiome-project.html#cb457-3" tabindex="-1"></a><span class="do">##                &quot;Multiome/pbmc_sorted_3k.fragments.tsv.gz&quot; </span></span>
<span id="cb457-4"><a href="importing-data-and-setting-up-a-multiome-project.html#cb457-4" tabindex="-1"></a><span class="do">##                                          pbmc_unsorted_3k </span></span>
<span id="cb457-5"><a href="importing-data-and-setting-up-a-multiome-project.html#cb457-5" tabindex="-1"></a><span class="do">##              &quot;Multiome/pbmc_unsorted_3k.fragments.tsv.gz&quot; </span></span>
<span id="cb457-6"><a href="importing-data-and-setting-up-a-multiome-project.html#cb457-6" tabindex="-1"></a><span class="do">##                                            pbmc_sorted_3k </span></span>
<span id="cb457-7"><a href="importing-data-and-setting-up-a-multiome-project.html#cb457-7" tabindex="-1"></a><span class="do">##   &quot;Multiome/pbmc_sorted_3k.filtered_feature_bc_matrix.h5&quot; </span></span>
<span id="cb457-8"><a href="importing-data-and-setting-up-a-multiome-project.html#cb457-8" tabindex="-1"></a><span class="do">##                                          pbmc_unsorted_3k </span></span>
<span id="cb457-9"><a href="importing-data-and-setting-up-a-multiome-project.html#cb457-9" tabindex="-1"></a><span class="do">## &quot;Multiome/pbmc_unsorted_3k.filtered_feature_bc_matrix.h5&quot;</span></span></code></pre></div>
<p>The first thing we need to do is separate out the scATAC-seq input files from the scRNA-seq input files. Here, we are doing that using <code>grep()</code> but you can also do it manually however you want. The important thing is to end up with a named vector of files corresponding to the scATAC-seq data and a named vector of files corresponding to the scRNA-seq data <strong>and that the names of the corresponding samples match</strong>!</p>
<div class="sourceCode" id="cb458"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb458-1"><a href="importing-data-and-setting-up-a-multiome-project.html#cb458-1" tabindex="-1"></a>atacFiles <span class="ot">&lt;-</span> inputFiles[<span class="fu">grep</span>(<span class="at">pattern =</span> <span class="st">&quot;</span><span class="sc">\\</span><span class="st">.fragments.tsv.gz$&quot;</span>, <span class="at">x =</span> inputFiles)]</span>
<span id="cb458-2"><a href="importing-data-and-setting-up-a-multiome-project.html#cb458-2" tabindex="-1"></a>rnaFiles <span class="ot">&lt;-</span> inputFiles[<span class="fu">grep</span>(<span class="at">pattern =</span> <span class="st">&quot;</span><span class="sc">\\</span><span class="st">.filtered_feature_bc_matrix.h5$&quot;</span>, <span class="at">x =</span> inputFiles)]</span></code></pre></div>
<p>For clarity, the above code is the same as doing this manually:</p>
<div class="sourceCode" id="cb459"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb459-1"><a href="importing-data-and-setting-up-a-multiome-project.html#cb459-1" tabindex="-1"></a>atacFiles <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;pbmc_sorted_3k&quot;</span> <span class="ot">=</span> <span class="st">&quot;Multiome/pbmc_sorted_3k.fragments.tsv.gz&quot;</span>, <span class="st">&quot;pbmc_unsorted_3k&quot;</span> <span class="ot">=</span> <span class="st">&quot;Multiome/pbmc_unsorted_3k.fragments.tsv.gz&quot;</span>)</span>
<span id="cb459-2"><a href="importing-data-and-setting-up-a-multiome-project.html#cb459-2" tabindex="-1"></a>rnaFiles <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;pbmc_sorted_3k&quot;</span> <span class="ot">=</span> <span class="st">&quot;Multiome/pbmc_sorted_3k.filtered_feature_bc_matrix.h5&quot;</span>, <span class="st">&quot;pbmc_unsorted_3k&quot;</span> <span class="ot">=</span> <span class="st">&quot;Multiome/pbmc_unsorted_3k.filtered_feature_bc_matrix.h5&quot;</span>)</span></code></pre></div>
<p>You can see that the names in these vectors match which is what will be used to link the files together upon import.</p>
<div class="sourceCode" id="cb460"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb460-1"><a href="importing-data-and-setting-up-a-multiome-project.html#cb460-1" tabindex="-1"></a><span class="fu">names</span>(atacFiles)</span>
<span id="cb460-2"><a href="importing-data-and-setting-up-a-multiome-project.html#cb460-2" tabindex="-1"></a><span class="do">## [1] &quot;pbmc_sorted_3k&quot;   &quot;pbmc_unsorted_3k&quot;</span></span>
<span id="cb460-3"><a href="importing-data-and-setting-up-a-multiome-project.html#cb460-3" tabindex="-1"></a><span class="fu">names</span>(rnaFiles)</span>
<span id="cb460-4"><a href="importing-data-and-setting-up-a-multiome-project.html#cb460-4" tabindex="-1"></a><span class="do">## [1] &quot;pbmc_sorted_3k&quot;   &quot;pbmc_unsorted_3k&quot;</span></span>
<span id="cb460-5"><a href="importing-data-and-setting-up-a-multiome-project.html#cb460-5" tabindex="-1"></a><span class="fu">all.equal</span>(<span class="fu">names</span>(atacFiles), <span class="fu">names</span>(rnaFiles))</span>
<span id="cb460-6"><a href="importing-data-and-setting-up-a-multiome-project.html#cb460-6" tabindex="-1"></a><span class="do">## [1] TRUE</span></span></code></pre></div>
<p>As in the main tutorial, the first step is to create ArrowFiles from the scATAC-seq fragment files.</p>
<div class="sourceCode" id="cb461"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb461-1"><a href="importing-data-and-setting-up-a-multiome-project.html#cb461-1" tabindex="-1"></a>ArrowFiles <span class="ot">&lt;-</span> <span class="fu">createArrowFiles</span>(</span>
<span id="cb461-2"><a href="importing-data-and-setting-up-a-multiome-project.html#cb461-2" tabindex="-1"></a>  <span class="at">inputFiles =</span> atacFiles,</span>
<span id="cb461-3"><a href="importing-data-and-setting-up-a-multiome-project.html#cb461-3" tabindex="-1"></a>  <span class="at">sampleNames =</span> <span class="fu">names</span>(atacFiles),</span>
<span id="cb461-4"><a href="importing-data-and-setting-up-a-multiome-project.html#cb461-4" tabindex="-1"></a>  <span class="at">minTSS =</span> <span class="dv">4</span>,</span>
<span id="cb461-5"><a href="importing-data-and-setting-up-a-multiome-project.html#cb461-5" tabindex="-1"></a>  <span class="at">minFrags =</span> <span class="dv">1000</span>,</span>
<span id="cb461-6"><a href="importing-data-and-setting-up-a-multiome-project.html#cb461-6" tabindex="-1"></a>  <span class="at">addTileMat =</span> <span class="cn">TRUE</span>,</span>
<span id="cb461-7"><a href="importing-data-and-setting-up-a-multiome-project.html#cb461-7" tabindex="-1"></a>  <span class="at">addGeneScoreMat =</span> <span class="cn">TRUE</span></span>
<span id="cb461-8"><a href="importing-data-and-setting-up-a-multiome-project.html#cb461-8" tabindex="-1"></a>)</span>
<span id="cb461-9"><a href="importing-data-and-setting-up-a-multiome-project.html#cb461-9" tabindex="-1"></a><span class="do">## Using GeneAnnotation set by addArchRGenome(Hg38)!</span></span>
<span id="cb461-10"><a href="importing-data-and-setting-up-a-multiome-project.html#cb461-10" tabindex="-1"></a><span class="do">## Using GeneAnnotation set by addArchRGenome(Hg38)!</span></span>
<span id="cb461-11"><a href="importing-data-and-setting-up-a-multiome-project.html#cb461-11" tabindex="-1"></a><span class="do">## ArchR logging to : ArchRLogs/ArchR-createArrows-96aeb8df5-Date-2025-02-06_Time-03-35-55.533788.log</span></span>
<span id="cb461-12"><a href="importing-data-and-setting-up-a-multiome-project.html#cb461-12" tabindex="-1"></a><span class="do">## If there is an issue, please report to github with logFile!</span></span>
<span id="cb461-13"><a href="importing-data-and-setting-up-a-multiome-project.html#cb461-13" tabindex="-1"></a><span class="do">## Cleaning Temporary Files</span></span>
<span id="cb461-14"><a href="importing-data-and-setting-up-a-multiome-project.html#cb461-14" tabindex="-1"></a><span class="do">## subThreading Disabled since ArchRLocking is TRUE see `addArchRLocking`</span></span>
<span id="cb461-15"><a href="importing-data-and-setting-up-a-multiome-project.html#cb461-15" tabindex="-1"></a><span class="do">## 2025-02-06 03:35:56.187222 : Batch Execution w/ safelapply!, 0 mins elapsed.</span></span>
<span id="cb461-16"><a href="importing-data-and-setting-up-a-multiome-project.html#cb461-16" tabindex="-1"></a><span class="do">## ArchR logging successful to : ArchRLogs/ArchR-createArrows-96aeb8df5-Date-2025-02-06_Time-03-35-55.533788.log</span></span></code></pre></div>
<p>Then we create an <code>ArchRProject</code> object from those ArrowFiles</p>
<div class="sourceCode" id="cb462"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb462-1"><a href="importing-data-and-setting-up-a-multiome-project.html#cb462-1" tabindex="-1"></a>projMulti1 <span class="ot">&lt;-</span> <span class="fu">ArchRProject</span>(<span class="at">ArrowFiles =</span> ArrowFiles)</span>
<span id="cb462-2"><a href="importing-data-and-setting-up-a-multiome-project.html#cb462-2" tabindex="-1"></a><span class="do">## Using GeneAnnotation set by addArchRGenome(Hg38)!</span></span>
<span id="cb462-3"><a href="importing-data-and-setting-up-a-multiome-project.html#cb462-3" tabindex="-1"></a><span class="do">## Using GeneAnnotation set by addArchRGenome(Hg38)!</span></span>
<span id="cb462-4"><a href="importing-data-and-setting-up-a-multiome-project.html#cb462-4" tabindex="-1"></a><span class="do">## Validating Arrows...</span></span>
<span id="cb462-5"><a href="importing-data-and-setting-up-a-multiome-project.html#cb462-5" tabindex="-1"></a><span class="do">## Getting SampleNames...</span></span>
<span id="cb462-6"><a href="importing-data-and-setting-up-a-multiome-project.html#cb462-6" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb462-7"><a href="importing-data-and-setting-up-a-multiome-project.html#cb462-7" tabindex="-1"></a><span class="do">## Copying ArrowFiles to Ouptut Directory! If you want to save disk space set copyArrows = FALSE</span></span>
<span id="cb462-8"><a href="importing-data-and-setting-up-a-multiome-project.html#cb462-8" tabindex="-1"></a><span class="do">## 1 2 </span></span>
<span id="cb462-9"><a href="importing-data-and-setting-up-a-multiome-project.html#cb462-9" tabindex="-1"></a><span class="do">## Getting Cell Metadata...</span></span>
<span id="cb462-10"><a href="importing-data-and-setting-up-a-multiome-project.html#cb462-10" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb462-11"><a href="importing-data-and-setting-up-a-multiome-project.html#cb462-11" tabindex="-1"></a><span class="do">## Merging Cell Metadata...</span></span>
<span id="cb462-12"><a href="importing-data-and-setting-up-a-multiome-project.html#cb462-12" tabindex="-1"></a><span class="do">## Initializing ArchRProject...</span></span>
<span id="cb462-13"><a href="importing-data-and-setting-up-a-multiome-project.html#cb462-13" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb462-14"><a href="importing-data-and-setting-up-a-multiome-project.html#cb462-14" tabindex="-1"></a><span class="do">##                                                    / |</span></span>
<span id="cb462-15"><a href="importing-data-and-setting-up-a-multiome-project.html#cb462-15" tabindex="-1"></a><span class="do">##                                                  /    \</span></span>
<span id="cb462-16"><a href="importing-data-and-setting-up-a-multiome-project.html#cb462-16" tabindex="-1"></a><span class="do">##             .                                  /      |.</span></span>
<span id="cb462-17"><a href="importing-data-and-setting-up-a-multiome-project.html#cb462-17" tabindex="-1"></a><span class="do">##             \\\                              /        |.</span></span>
<span id="cb462-18"><a href="importing-data-and-setting-up-a-multiome-project.html#cb462-18" tabindex="-1"></a><span class="do">##               \\\                          /           `|.</span></span>
<span id="cb462-19"><a href="importing-data-and-setting-up-a-multiome-project.html#cb462-19" tabindex="-1"></a><span class="do">##                 \\\                      /              |.</span></span>
<span id="cb462-20"><a href="importing-data-and-setting-up-a-multiome-project.html#cb462-20" tabindex="-1"></a><span class="do">##                   \                    /                |\</span></span>
<span id="cb462-21"><a href="importing-data-and-setting-up-a-multiome-project.html#cb462-21" tabindex="-1"></a><span class="do">##                   \\#####\           /                  ||</span></span>
<span id="cb462-22"><a href="importing-data-and-setting-up-a-multiome-project.html#cb462-22" tabindex="-1"></a><span class="do">##                 ==###########&gt;      /                   ||</span></span>
<span id="cb462-23"><a href="importing-data-and-setting-up-a-multiome-project.html#cb462-23" tabindex="-1"></a><span class="do">##                  \\##==......\    /                     ||</span></span>
<span id="cb462-24"><a href="importing-data-and-setting-up-a-multiome-project.html#cb462-24" tabindex="-1"></a><span class="do">##             ______ =       =|__ /__                     ||      \\\</span></span>
<span id="cb462-25"><a href="importing-data-and-setting-up-a-multiome-project.html#cb462-25" tabindex="-1"></a><span class="do">##         ,--&#39; ,----`-,__ ___/&#39;  --,-`-===================##========&gt;</span></span>
<span id="cb462-26"><a href="importing-data-and-setting-up-a-multiome-project.html#cb462-26" tabindex="-1"></a><span class="do">##        \               &#39;        ##_______ _____ ,--,__,=##,__   ///</span></span>
<span id="cb462-27"><a href="importing-data-and-setting-up-a-multiome-project.html#cb462-27" tabindex="-1"></a><span class="do">##         ,    __==    ___,-,__,--&#39;#&#39;  ===&#39;      `-&#39;    | ##,-/</span></span>
<span id="cb462-28"><a href="importing-data-and-setting-up-a-multiome-project.html#cb462-28" tabindex="-1"></a><span class="do">##         -,____,---&#39;       \\####\\________________,--\\_##,/</span></span>
<span id="cb462-29"><a href="importing-data-and-setting-up-a-multiome-project.html#cb462-29" tabindex="-1"></a><span class="do">##            ___      .______        ______  __    __  .______      </span></span>
<span id="cb462-30"><a href="importing-data-and-setting-up-a-multiome-project.html#cb462-30" tabindex="-1"></a><span class="do">##           /   \     |   _  \      /      ||  |  |  | |   _  \     </span></span>
<span id="cb462-31"><a href="importing-data-and-setting-up-a-multiome-project.html#cb462-31" tabindex="-1"></a><span class="do">##          /  ^  \    |  |_)  |    |  ,----&#39;|  |__|  | |  |_)  |    </span></span>
<span id="cb462-32"><a href="importing-data-and-setting-up-a-multiome-project.html#cb462-32" tabindex="-1"></a><span class="do">##         /  /_\  \   |      /     |  |     |   __   | |      /     </span></span>
<span id="cb462-33"><a href="importing-data-and-setting-up-a-multiome-project.html#cb462-33" tabindex="-1"></a><span class="do">##        /  _____  \  |  |\  \\___ |  `----.|  |  |  | |  |\  \\___.</span></span>
<span id="cb462-34"><a href="importing-data-and-setting-up-a-multiome-project.html#cb462-34" tabindex="-1"></a><span class="do">##       /__/     \__\ | _| `._____| \______||__|  |__| | _| `._____|</span></span>
<span id="cb462-35"><a href="importing-data-and-setting-up-a-multiome-project.html#cb462-35" tabindex="-1"></a><span class="do">## </span></span></code></pre></div>
<p>Now that we’ve handled the scATAC-seq data, we turn out attention to the scRNA-seq data. ArchR provides the <code>import10xFeatureMatrix()</code> function which will perform this automatically for all of the <code>filtered_feature_bc_matrix.h5</code> files in your data set. While this process may seem straightforward, there are a few things going on under-the-hood which are important to understand. First, the <code>filtered_feature_bc_matrix.h5</code> created by <code>cellranger-arc</code> are not always perfectly matched across samples. For example, ArchR will throw an error if your various scRNA-seq input files do not match based on the gene names or the metadata columns because we view this as an unforseen incompatibility between data, potentially caused by alignment to different reference genomes or something similar. There are also more permissible mismatches that occur across these input files. Relatively frequently, the transcript-level data that is associated with each gene can vary slightly across samples for reasons that aren’t completely clear. For more information, see <a href="https://github.com/GreenleafLab/ArchR/issues/507#issuecomment-1063139673">this post</a>. Because of this, you have to decide what to do when these minor conflicts arise. We provide two options which are controlled by the <code>strictMatch</code> parameter. If you want to try to keep the genes with mis-matched metadata information, set <code>strictMatch = FALSE</code> which will coerce all samples to match the metadata information of the first sample from your <code>input</code>. Alternatively, if you would prefer to remove genes whose metadata is mis-matched across samples, you can set <code>strictMatch = TRUE</code> which will remove the offending genes from all samples. This is often a very small number of genes so it is unlikely to affect your analysis either way.</p>
<div class="sourceCode" id="cb463"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb463-1"><a href="importing-data-and-setting-up-a-multiome-project.html#cb463-1" tabindex="-1"></a>seRNA <span class="ot">&lt;-</span> <span class="fu">import10xFeatureMatrix</span>(</span>
<span id="cb463-2"><a href="importing-data-and-setting-up-a-multiome-project.html#cb463-2" tabindex="-1"></a>  <span class="at">input =</span> rnaFiles,</span>
<span id="cb463-3"><a href="importing-data-and-setting-up-a-multiome-project.html#cb463-3" tabindex="-1"></a>  <span class="at">names =</span> <span class="fu">names</span>(rnaFiles),</span>
<span id="cb463-4"><a href="importing-data-and-setting-up-a-multiome-project.html#cb463-4" tabindex="-1"></a>  <span class="at">strictMatch =</span> <span class="cn">TRUE</span></span>
<span id="cb463-5"><a href="importing-data-and-setting-up-a-multiome-project.html#cb463-5" tabindex="-1"></a>)</span>
<span id="cb463-6"><a href="importing-data-and-setting-up-a-multiome-project.html#cb463-6" tabindex="-1"></a><span class="do">## Importing Feature Matrix 1 of 2</span></span>
<span id="cb463-7"><a href="importing-data-and-setting-up-a-multiome-project.html#cb463-7" tabindex="-1"></a><span class="do">## Merging individual RNA objects...</span></span>
<span id="cb463-8"><a href="importing-data-and-setting-up-a-multiome-project.html#cb463-8" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb463-9"><a href="importing-data-and-setting-up-a-multiome-project.html#cb463-9" tabindex="-1"></a><span class="do">## Merging pbmc_unsorted_3k</span></span>
<span id="cb463-10"><a href="importing-data-and-setting-up-a-multiome-project.html#cb463-10" tabindex="-1"></a><span class="do">## Importing Feature Matrix 2 of 2</span></span>
<span id="cb463-11"><a href="importing-data-and-setting-up-a-multiome-project.html#cb463-11" tabindex="-1"></a><span class="do">## Warning in import10xFeatureMatrix(input = rnaFiles, names = names(rnaFiles), : These features had no interval present and were given a fake GRanges location on `chrNA`:</span></span>
<span id="cb463-12"><a href="importing-data-and-setting-up-a-multiome-project.html#cb463-12" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb463-13"><a href="importing-data-and-setting-up-a-multiome-project.html#cb463-13" tabindex="-1"></a><span class="do">## MT-ND1,MT-ND2,MT-CO1,MT-CO2,MT-ATP8,MT-ATP6,MT-CO3,MT-ND3,MT-ND4L,MT-ND4,MT-ND5,MT-ND6,MT-CYB</span></span>
<span id="cb463-14"><a href="importing-data-and-setting-up-a-multiome-project.html#cb463-14" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb463-15"><a href="importing-data-and-setting-up-a-multiome-project.html#cb463-15" tabindex="-1"></a><span class="do">## Either modify the output SE or see `features` input to use a reference GRanges to add to these features!</span></span></code></pre></div>
<p>In this tutorial data, there actually aren’t any examples of this transcript-level metadata mismatch. But if there were, ArchR would output warnings to tell you about the genes that were being excluded or coerced. However, what you do see is a warning that some of the features (genes) did not have an “interval” (or genomic location) present in the 10x input file. This is most frequently the case for genes encoded on the mitochondrial DNA but could in theory happen for other genes in other species as well. This causes problems for ArchR because we assume that every gene has a position in the genome. To get around this problem, ArchR will default to assigning these genes a fake genomic position on “chrNA”. All of these genes will then get excluded from downstream analyses. If you want to rescue these genes, then you must provide this information via the <code>features</code> parameter. This is most easily done using an Ensembl database object from BioConductor. The below code assumes that you have installed the <code>EnsDb.Hsapiens.v86</code> package from BioConductor.</p>
<div class="sourceCode" id="cb464"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb464-1"><a href="importing-data-and-setting-up-a-multiome-project.html#cb464-1" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">&quot;EnsDb.Hsapiens.v86&quot;</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>)) {</span>
<span id="cb464-2"><a href="importing-data-and-setting-up-a-multiome-project.html#cb464-2" tabindex="-1"></a>  BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">&quot;EnsDb.Hsapiens.v86&quot;</span>, <span class="at">update =</span> <span class="cn">FALSE</span>)</span>
<span id="cb464-3"><a href="importing-data-and-setting-up-a-multiome-project.html#cb464-3" tabindex="-1"></a>}</span>
<span id="cb464-4"><a href="importing-data-and-setting-up-a-multiome-project.html#cb464-4" tabindex="-1"></a><span class="fu">library</span>(EnsDb.Hsapiens.v86)</span>
<span id="cb464-5"><a href="importing-data-and-setting-up-a-multiome-project.html#cb464-5" tabindex="-1"></a><span class="do">## Loading required package: ensembldb</span></span>
<span id="cb464-6"><a href="importing-data-and-setting-up-a-multiome-project.html#cb464-6" tabindex="-1"></a><span class="do">## Loading required package: AnnotationFilter</span></span>
<span id="cb464-7"><a href="importing-data-and-setting-up-a-multiome-project.html#cb464-7" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb464-8"><a href="importing-data-and-setting-up-a-multiome-project.html#cb464-8" tabindex="-1"></a><span class="do">## Attaching package: &#39;AnnotationFilter&#39;</span></span>
<span id="cb464-9"><a href="importing-data-and-setting-up-a-multiome-project.html#cb464-9" tabindex="-1"></a><span class="do">## The following object is masked from &#39;package:magrittr&#39;:</span></span>
<span id="cb464-10"><a href="importing-data-and-setting-up-a-multiome-project.html#cb464-10" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb464-11"><a href="importing-data-and-setting-up-a-multiome-project.html#cb464-11" tabindex="-1"></a><span class="do">##     not</span></span>
<span id="cb464-12"><a href="importing-data-and-setting-up-a-multiome-project.html#cb464-12" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb464-13"><a href="importing-data-and-setting-up-a-multiome-project.html#cb464-13" tabindex="-1"></a><span class="do">## Attaching package: &#39;ensembldb&#39;</span></span>
<span id="cb464-14"><a href="importing-data-and-setting-up-a-multiome-project.html#cb464-14" tabindex="-1"></a><span class="do">## The following object is masked from &#39;package:stats&#39;:</span></span>
<span id="cb464-15"><a href="importing-data-and-setting-up-a-multiome-project.html#cb464-15" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb464-16"><a href="importing-data-and-setting-up-a-multiome-project.html#cb464-16" tabindex="-1"></a><span class="do">##     filter</span></span>
<span id="cb464-17"><a href="importing-data-and-setting-up-a-multiome-project.html#cb464-17" tabindex="-1"></a>seRNA <span class="ot">&lt;-</span> <span class="fu">import10xFeatureMatrix</span>(</span>
<span id="cb464-18"><a href="importing-data-and-setting-up-a-multiome-project.html#cb464-18" tabindex="-1"></a>  <span class="at">input =</span> rnaFiles,</span>
<span id="cb464-19"><a href="importing-data-and-setting-up-a-multiome-project.html#cb464-19" tabindex="-1"></a>  <span class="at">names =</span> <span class="fu">names</span>(rnaFiles),</span>
<span id="cb464-20"><a href="importing-data-and-setting-up-a-multiome-project.html#cb464-20" tabindex="-1"></a>  <span class="at">strictMatch =</span> <span class="cn">TRUE</span>,</span>
<span id="cb464-21"><a href="importing-data-and-setting-up-a-multiome-project.html#cb464-21" tabindex="-1"></a>  <span class="at">features =</span> <span class="fu">genes</span>(EnsDb.Hsapiens.v86)</span>
<span id="cb464-22"><a href="importing-data-and-setting-up-a-multiome-project.html#cb464-22" tabindex="-1"></a>)</span>
<span id="cb464-23"><a href="importing-data-and-setting-up-a-multiome-project.html#cb464-23" tabindex="-1"></a><span class="do">## Importing Feature Matrix 1 of 2</span></span>
<span id="cb464-24"><a href="importing-data-and-setting-up-a-multiome-project.html#cb464-24" tabindex="-1"></a><span class="do">## Correcting missing intervals...</span></span>
<span id="cb464-25"><a href="importing-data-and-setting-up-a-multiome-project.html#cb464-25" tabindex="-1"></a><span class="do">## Merging individual RNA objects...</span></span>
<span id="cb464-26"><a href="importing-data-and-setting-up-a-multiome-project.html#cb464-26" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb464-27"><a href="importing-data-and-setting-up-a-multiome-project.html#cb464-27" tabindex="-1"></a><span class="do">## Merging pbmc_unsorted_3k</span></span>
<span id="cb464-28"><a href="importing-data-and-setting-up-a-multiome-project.html#cb464-28" tabindex="-1"></a><span class="do">## Importing Feature Matrix 2 of 2</span></span>
<span id="cb464-29"><a href="importing-data-and-setting-up-a-multiome-project.html#cb464-29" tabindex="-1"></a><span class="do">## Correcting missing intervals...</span></span></code></pre></div>
<p>After supplying the genes to the <code>features</code> param, you can see that all of the genes encoded on the mitochondrial genome are rescued.</p>
<p>The next thing that we want to do is add this scRNA-seq data to our <code>ArchRProject</code> via the <code>addGeneExpressionMatrix()</code> function. However, there is some tidying up that we need to do before we are ready to do that. Inherent in how the multiomic data is generated, you will likely have cells that pass scRNA-seq quality control but not scATAC-seq quality control and vice versa. This can cause problems downstream with other ArchR functions that expect every cell to have every data type. For example, if you were to add gene expression data for only a subset of cells and then try to perform <code>addIterativeLSI()</code> on the corresponding <code>GeneExpressionMatrix</code>, ArchR would not know what to do with the cells that were missing data in the <code>GeneExpressionMatrix</code>. Because our <code>ArchRProject</code> is anchored in the scATAC-seq data, we first check to see how many cells in our project (the ones that have passed scATAC-seq quality control) are not also in our scRNA-seq data</p>
<div class="sourceCode" id="cb465"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb465-1"><a href="importing-data-and-setting-up-a-multiome-project.html#cb465-1" tabindex="-1"></a><span class="fu">length</span>(<span class="fu">which</span>(<span class="fu">getCellNames</span>(projMulti1) <span class="sc">%ni%</span> <span class="fu">colnames</span>(seRNA)))</span>
<span id="cb465-2"><a href="importing-data-and-setting-up-a-multiome-project.html#cb465-2" tabindex="-1"></a><span class="do">## [1] 119</span></span></code></pre></div>
<p>Though not many cells in this case, we need to remove these cells from our project before proceeding. We do this using the <code>subsetArchRProject()</code> function and saving this as a new <code>ArchRProject</code> called <code>projMulti2</code>. Again, this removal isn’t explicitly necessary but there are downstream ArchR functions that could break if all cells dont have both data types so the safest thing to do is remove them.</p>
<div class="sourceCode" id="cb466"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb466-1"><a href="importing-data-and-setting-up-a-multiome-project.html#cb466-1" tabindex="-1"></a>cellsToKeep <span class="ot">&lt;-</span> <span class="fu">which</span>(<span class="fu">getCellNames</span>(projMulti1) <span class="sc">%in%</span> <span class="fu">colnames</span>(seRNA))</span>
<span id="cb466-2"><a href="importing-data-and-setting-up-a-multiome-project.html#cb466-2" tabindex="-1"></a>projMulti2 <span class="ot">&lt;-</span> <span class="fu">subsetArchRProject</span>(<span class="at">ArchRProj =</span> projMulti1, <span class="at">cells =</span> <span class="fu">getCellNames</span>(projMulti1)[cellsToKeep], <span class="at">outputDirectory =</span> <span class="st">&quot;Save-ProjMulti2&quot;</span>, <span class="at">force =</span> <span class="cn">TRUE</span>)</span>
<span id="cb466-3"><a href="importing-data-and-setting-up-a-multiome-project.html#cb466-3" tabindex="-1"></a><span class="do">## Copying ArchRProject to new outputDirectory : /workspace/ArchR/ArchR_Website_Testing/bookdown/Save-ProjMulti2</span></span>
<span id="cb466-4"><a href="importing-data-and-setting-up-a-multiome-project.html#cb466-4" tabindex="-1"></a><span class="do">## Copying Arrow Files...</span></span>
<span id="cb466-5"><a href="importing-data-and-setting-up-a-multiome-project.html#cb466-5" tabindex="-1"></a><span class="do">## Getting ImputeWeights</span></span>
<span id="cb466-6"><a href="importing-data-and-setting-up-a-multiome-project.html#cb466-6" tabindex="-1"></a><span class="do">## No imputeWeights found, returning NULL</span></span>
<span id="cb466-7"><a href="importing-data-and-setting-up-a-multiome-project.html#cb466-7" tabindex="-1"></a><span class="do">## Copying Other Files...</span></span>
<span id="cb466-8"><a href="importing-data-and-setting-up-a-multiome-project.html#cb466-8" tabindex="-1"></a><span class="do">## Saving ArchRProject...</span></span>
<span id="cb466-9"><a href="importing-data-and-setting-up-a-multiome-project.html#cb466-9" tabindex="-1"></a><span class="do">## Loading ArchRProject...</span></span>
<span id="cb466-10"><a href="importing-data-and-setting-up-a-multiome-project.html#cb466-10" tabindex="-1"></a><span class="do">## Successfully loaded ArchRProject!</span></span>
<span id="cb466-11"><a href="importing-data-and-setting-up-a-multiome-project.html#cb466-11" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb466-12"><a href="importing-data-and-setting-up-a-multiome-project.html#cb466-12" tabindex="-1"></a><span class="do">##                                                    / |</span></span>
<span id="cb466-13"><a href="importing-data-and-setting-up-a-multiome-project.html#cb466-13" tabindex="-1"></a><span class="do">##                                                  /    \</span></span>
<span id="cb466-14"><a href="importing-data-and-setting-up-a-multiome-project.html#cb466-14" tabindex="-1"></a><span class="do">##             .                                  /      |.</span></span>
<span id="cb466-15"><a href="importing-data-and-setting-up-a-multiome-project.html#cb466-15" tabindex="-1"></a><span class="do">##             \\\                              /        |.</span></span>
<span id="cb466-16"><a href="importing-data-and-setting-up-a-multiome-project.html#cb466-16" tabindex="-1"></a><span class="do">##               \\\                          /           `|.</span></span>
<span id="cb466-17"><a href="importing-data-and-setting-up-a-multiome-project.html#cb466-17" tabindex="-1"></a><span class="do">##                 \\\                      /              |.</span></span>
<span id="cb466-18"><a href="importing-data-and-setting-up-a-multiome-project.html#cb466-18" tabindex="-1"></a><span class="do">##                   \                    /                |\</span></span>
<span id="cb466-19"><a href="importing-data-and-setting-up-a-multiome-project.html#cb466-19" tabindex="-1"></a><span class="do">##                   \\#####\           /                  ||</span></span>
<span id="cb466-20"><a href="importing-data-and-setting-up-a-multiome-project.html#cb466-20" tabindex="-1"></a><span class="do">##                 ==###########&gt;      /                   ||</span></span>
<span id="cb466-21"><a href="importing-data-and-setting-up-a-multiome-project.html#cb466-21" tabindex="-1"></a><span class="do">##                  \\##==......\    /                     ||</span></span>
<span id="cb466-22"><a href="importing-data-and-setting-up-a-multiome-project.html#cb466-22" tabindex="-1"></a><span class="do">##             ______ =       =|__ /__                     ||      \\\</span></span>
<span id="cb466-23"><a href="importing-data-and-setting-up-a-multiome-project.html#cb466-23" tabindex="-1"></a><span class="do">##         ,--&#39; ,----`-,__ ___/&#39;  --,-`-===================##========&gt;</span></span>
<span id="cb466-24"><a href="importing-data-and-setting-up-a-multiome-project.html#cb466-24" tabindex="-1"></a><span class="do">##        \               &#39;        ##_______ _____ ,--,__,=##,__   ///</span></span>
<span id="cb466-25"><a href="importing-data-and-setting-up-a-multiome-project.html#cb466-25" tabindex="-1"></a><span class="do">##         ,    __==    ___,-,__,--&#39;#&#39;  ===&#39;      `-&#39;    | ##,-/</span></span>
<span id="cb466-26"><a href="importing-data-and-setting-up-a-multiome-project.html#cb466-26" tabindex="-1"></a><span class="do">##         -,____,---&#39;       \\####\\________________,--\\_##,/</span></span>
<span id="cb466-27"><a href="importing-data-and-setting-up-a-multiome-project.html#cb466-27" tabindex="-1"></a><span class="do">##            ___      .______        ______  __    __  .______      </span></span>
<span id="cb466-28"><a href="importing-data-and-setting-up-a-multiome-project.html#cb466-28" tabindex="-1"></a><span class="do">##           /   \     |   _  \      /      ||  |  |  | |   _  \     </span></span>
<span id="cb466-29"><a href="importing-data-and-setting-up-a-multiome-project.html#cb466-29" tabindex="-1"></a><span class="do">##          /  ^  \    |  |_)  |    |  ,----&#39;|  |__|  | |  |_)  |    </span></span>
<span id="cb466-30"><a href="importing-data-and-setting-up-a-multiome-project.html#cb466-30" tabindex="-1"></a><span class="do">##         /  /_\  \   |      /     |  |     |   __   | |      /     </span></span>
<span id="cb466-31"><a href="importing-data-and-setting-up-a-multiome-project.html#cb466-31" tabindex="-1"></a><span class="do">##        /  _____  \  |  |\  \\___ |  `----.|  |  |  | |  |\  \\___.</span></span>
<span id="cb466-32"><a href="importing-data-and-setting-up-a-multiome-project.html#cb466-32" tabindex="-1"></a><span class="do">##       /__/     \__\ | _| `._____| \______||__|  |__| | _| `._____|</span></span>
<span id="cb466-33"><a href="importing-data-and-setting-up-a-multiome-project.html#cb466-33" tabindex="-1"></a><span class="do">## </span></span></code></pre></div>
<p>Finally, we are ready to add the gene expression data to our project. You’ll notice here another parameter called <code>strictMatch</code>. When <code>strictMatch = TRUE</code>, this will ensure that all cells in the <code>ArchRProject</code> are also represented in the <code>seRNA</code> object. This is a nice fail-safe to make sure that the above project subsetting went smoothly. If <code>strictMatch = FALSE</code> (the default), then this function will merely throw a warning telling you that not all of your cells have scRNA-seq information and that this could cause problems downstream.</p>
<div class="sourceCode" id="cb467"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb467-1"><a href="importing-data-and-setting-up-a-multiome-project.html#cb467-1" tabindex="-1"></a>projMulti2 <span class="ot">&lt;-</span> <span class="fu">addGeneExpressionMatrix</span>(<span class="at">input =</span> projMulti2, <span class="at">seRNA =</span> seRNA, <span class="at">strictMatch =</span> <span class="cn">TRUE</span>, <span class="at">force =</span> <span class="cn">TRUE</span>)</span>
<span id="cb467-2"><a href="importing-data-and-setting-up-a-multiome-project.html#cb467-2" tabindex="-1"></a><span class="do">## ArchR logging to : ArchRLogs/ArchR-addGeneExpressionMatrix-936437d2-Date-2025-02-06_Time-03-40-14.18066.log</span></span>
<span id="cb467-3"><a href="importing-data-and-setting-up-a-multiome-project.html#cb467-3" tabindex="-1"></a><span class="do">## If there is an issue, please report to github with logFile!</span></span>
<span id="cb467-4"><a href="importing-data-and-setting-up-a-multiome-project.html#cb467-4" tabindex="-1"></a><span class="do">## Overlap w/ scATAC = 1</span></span>
<span id="cb467-5"><a href="importing-data-and-setting-up-a-multiome-project.html#cb467-5" tabindex="-1"></a><span class="do">## 2025-02-06 03:40:14.741087 :</span></span>
<span id="cb467-6"><a href="importing-data-and-setting-up-a-multiome-project.html#cb467-6" tabindex="-1"></a><span class="do">## Overlap Per Sample w/ scATAC : pbmc_sorted_3k=2626,pbmc_unsorted_3k=2845</span></span>
<span id="cb467-7"><a href="importing-data-and-setting-up-a-multiome-project.html#cb467-7" tabindex="-1"></a><span class="do">## 2025-02-06 03:40:14.755301 :</span></span>
<span id="cb467-8"><a href="importing-data-and-setting-up-a-multiome-project.html#cb467-8" tabindex="-1"></a><span class="do">## 2025-02-06 03:40:16.021819 : Batch Execution w/ safelapply!, 0 mins elapsed.</span></span>
<span id="cb467-9"><a href="importing-data-and-setting-up-a-multiome-project.html#cb467-9" tabindex="-1"></a><span class="do">## ArchR logging successful to : ArchRLogs/ArchR-addGeneExpressionMatrix-936437d2-Date-2025-02-06_Time-03-40-14.18066.log</span></span></code></pre></div>
<p>The last thing that we will do during this project setup phase is filter out any doublets.</p>
<div class="sourceCode" id="cb468"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb468-1"><a href="importing-data-and-setting-up-a-multiome-project.html#cb468-1" tabindex="-1"></a>projMulti2 <span class="ot">&lt;-</span> <span class="fu">addDoubletScores</span>(projMulti2, <span class="at">force =</span> <span class="cn">TRUE</span>)</span>
<span id="cb468-2"><a href="importing-data-and-setting-up-a-multiome-project.html#cb468-2" tabindex="-1"></a><span class="do">## ArchR logging to : ArchRLogs/ArchR-addDoubletScores-95eccb65e-Date-2025-02-06_Time-03-42-52.270117.log</span></span>
<span id="cb468-3"><a href="importing-data-and-setting-up-a-multiome-project.html#cb468-3" tabindex="-1"></a><span class="do">## If there is an issue, please report to github with logFile!</span></span>
<span id="cb468-4"><a href="importing-data-and-setting-up-a-multiome-project.html#cb468-4" tabindex="-1"></a><span class="do">## 2025-02-06 03:42:52.703694 : Batch Execution w/ safelapply!, 0 mins elapsed.</span></span>
<span id="cb468-5"><a href="importing-data-and-setting-up-a-multiome-project.html#cb468-5" tabindex="-1"></a><span class="do">## 2025-02-06 03:42:52.726859 : pbmc_sorted_3k (1 of 2) :  Computing Doublet Statistics, 0 mins elapsed.</span></span>
<span id="cb468-6"><a href="importing-data-and-setting-up-a-multiome-project.html#cb468-6" tabindex="-1"></a><span class="do">## pbmc_sorted_3k (1 of 2) : UMAP Projection R^2 = 0.99786</span></span>
<span id="cb468-7"><a href="importing-data-and-setting-up-a-multiome-project.html#cb468-7" tabindex="-1"></a><span class="do">## pbmc_sorted_3k (1 of 2) : UMAP Projection R^2 = 0.99786</span></span>
<span id="cb468-8"><a href="importing-data-and-setting-up-a-multiome-project.html#cb468-8" tabindex="-1"></a><span class="do">## 2025-02-06 03:45:40.92925 : pbmc_unsorted_3k (2 of 2) :  Computing Doublet Statistics, 2.804 mins elapsed.</span></span>
<span id="cb468-9"><a href="importing-data-and-setting-up-a-multiome-project.html#cb468-9" tabindex="-1"></a><span class="do">## pbmc_unsorted_3k (2 of 2) : UMAP Projection R^2 = 0.99594</span></span>
<span id="cb468-10"><a href="importing-data-and-setting-up-a-multiome-project.html#cb468-10" tabindex="-1"></a><span class="do">## pbmc_unsorted_3k (2 of 2) : UMAP Projection R^2 = 0.99594</span></span>
<span id="cb468-11"><a href="importing-data-and-setting-up-a-multiome-project.html#cb468-11" tabindex="-1"></a><span class="do">## ArchR logging successful to : ArchRLogs/ArchR-addDoubletScores-95eccb65e-Date-2025-02-06_Time-03-42-52.270117.log</span></span>
<span id="cb468-12"><a href="importing-data-and-setting-up-a-multiome-project.html#cb468-12" tabindex="-1"></a>projMulti2 <span class="ot">&lt;-</span> <span class="fu">filterDoublets</span>(projMulti2)</span>
<span id="cb468-13"><a href="importing-data-and-setting-up-a-multiome-project.html#cb468-13" tabindex="-1"></a><span class="do">## Filtering 148 cells from ArchRProject!</span></span>
<span id="cb468-14"><a href="importing-data-and-setting-up-a-multiome-project.html#cb468-14" tabindex="-1"></a><span class="do">##  pbmc_unsorted_3k : 80 of 2845 (2.8%)</span></span>
<span id="cb468-15"><a href="importing-data-and-setting-up-a-multiome-project.html#cb468-15" tabindex="-1"></a><span class="do">##  pbmc_sorted_3k : 68 of 2626 (2.6%)</span></span></code></pre></div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="multiomic-data-analysis-in-archr.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="analysis-of-multiome-data-in-archr.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": null
},
"fontsettings": null,
"edit": null,
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "none"
},
"info": false
});
});
</script>

</body>

</html>
