<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>8.1 Labeling scATAC cells with scRNA clusters | ArchR: Robust and scaleable analysis of single-cell chromatin accessibility data.</title>
  <meta name="description" content="A guide to ArchR" />
  <meta name="generator" content="bookdown 0.17 and GitBook 2.6.7" />

  <meta property="og:title" content="8.1 Labeling scATAC cells with scRNA clusters | ArchR: Robust and scaleable analysis of single-cell chromatin accessibility data." />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="A guide to ArchR" />
  <meta name="github-repo" content="GreenleafLab/ArchR" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="8.1 Labeling scATAC cells with scRNA clusters | ArchR: Robust and scaleable analysis of single-cell chromatin accessibility data." />
  
  <meta name="twitter:description" content="A guide to ArchR" />
  

<meta name="author" content="Jeffrey Granja and Ryan Corces" />


<meta name="date" content="2020-04-21" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="images/favicon.ico" type="image/x-icon" />
<link rel="prev" href="labeling-clusters-with-scrna.html"/>
<link rel="next" href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<link href="libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="../index.html">Back to ArchR Home</a></li>

<li class="divider"></li>
<li><a href="index.html#section"><img src="images/ArchR_Logo_Integrated.png" alt="" width="200"></a></li>
<li class="chapter" data-level="1" data-path="getting-started-with-archr.html"><a href="getting-started-with-archr.html"><i class="fa fa-check"></i><b>1</b> Getting Started with ArchR</a>
<ul>
<li class="chapter" data-level="1.1" data-path="a-brief-primer-on-atac-seq-terminology.html"><a href="a-brief-primer-on-atac-seq-terminology.html"><i class="fa fa-check"></i><b>1.1</b> A Brief Primer on ATAC-seq Terminology</a></li>
<li class="chapter" data-level="1.2" data-path="why-use-archr.html"><a href="why-use-archr.html"><i class="fa fa-check"></i><b>1.2</b> Why use ArchR?</a></li>
<li class="chapter" data-level="1.3" data-path="what-is-an-arrow-file-archrproject.html"><a href="what-is-an-arrow-file-archrproject.html"><i class="fa fa-check"></i><b>1.3</b> What is an Arrow file / <code>ArchRProject</code>?</a></li>
<li class="chapter" data-level="1.4" data-path="input-file-types-in-archr.html"><a href="input-file-types-in-archr.html"><i class="fa fa-check"></i><b>1.4</b> Input File Types in ArchR</a></li>
<li class="chapter" data-level="1.5" data-path="getting-set-up.html"><a href="getting-set-up.html"><i class="fa fa-check"></i><b>1.5</b> Getting Set Up</a></li>
<li class="chapter" data-level="1.6" data-path="creating-arrow-files.html"><a href="creating-arrow-files.html"><i class="fa fa-check"></i><b>1.6</b> Creating Arrow Files</a></li>
<li class="chapter" data-level="1.7" data-path="per-cell-quality-control.html"><a href="per-cell-quality-control.html"><i class="fa fa-check"></i><b>1.7</b> Per-cell Quality Control</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="doublet-inference-with-archr.html"><a href="doublet-inference-with-archr.html"><i class="fa fa-check"></i><b>2</b> Doublet Inference with ArchR</a>
<ul>
<li class="chapter" data-level="2.1" data-path="how-does-doublet-identification-work-in-archr.html"><a href="how-does-doublet-identification-work-in-archr.html"><i class="fa fa-check"></i><b>2.1</b> How does doublet identification work in ArchR?</a></li>
<li class="chapter" data-level="2.2" data-path="inferring-scatac-seq-doublets-with-archr.html"><a href="inferring-scatac-seq-doublets-with-archr.html"><i class="fa fa-check"></i><b>2.2</b> Inferring scATAC-seq Doublets with ArchR</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="creating-an-archrproject.html"><a href="creating-an-archrproject.html"><i class="fa fa-check"></i><b>3</b> Creating an ArchRProject</a>
<ul>
<li class="chapter" data-level="3.1" data-path="creating-an-archrproject-1.html"><a href="creating-an-archrproject-1.html"><i class="fa fa-check"></i><b>3.1</b> Creating An ArchRProject</a></li>
<li class="chapter" data-level="3.2" data-path="manipulating-an-archrproject.html"><a href="manipulating-an-archrproject.html"><i class="fa fa-check"></i><b>3.2</b> Manipulating An ArchRProject</a></li>
<li class="chapter" data-level="3.3" data-path="plotting-sample-statistics-from-an-archrproject.html"><a href="plotting-sample-statistics-from-an-archrproject.html"><i class="fa fa-check"></i><b>3.3</b> Plotting Sample Statistics from an ArchRProject</a></li>
<li class="chapter" data-level="3.4" data-path="plotting-sample-fragment-size-distribution-and-tss-enrichment-profiles-.html"><a href="plotting-sample-fragment-size-distribution-and-tss-enrichment-profiles-.html"><i class="fa fa-check"></i><b>3.4</b> Plotting Sample Fragment Size Distribution and TSS Enrichment Profiles.</a></li>
<li class="chapter" data-level="3.5" data-path="saving-and-loading-an-archrproject.html"><a href="saving-and-loading-an-archrproject.html"><i class="fa fa-check"></i><b>3.5</b> Saving and Loading an <code>ArchRProject</code></a></li>
<li class="chapter" data-level="3.6" data-path="filtering-doublets-from-an-archrproject.html"><a href="filtering-doublets-from-an-archrproject.html"><i class="fa fa-check"></i><b>3.6</b> Filtering Doublets from an ArchRProject</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="dimensionality-reduction-with-archr.html"><a href="dimensionality-reduction-with-archr.html"><i class="fa fa-check"></i><b>4</b> Dimensionality Reduction with ArchR</a>
<ul>
<li class="chapter" data-level="4.1" data-path="archrs-lsi-implementation.html"><a href="archrs-lsi-implementation.html"><i class="fa fa-check"></i><b>4.1</b> ArchR’s LSI Implementation</a></li>
<li class="chapter" data-level="4.2" data-path="iterative-latent-semantic-indexing-lsi.html"><a href="iterative-latent-semantic-indexing-lsi.html"><i class="fa fa-check"></i><b>4.2</b> Iterative Latent Semantic Indexing (LSI)</a></li>
<li class="chapter" data-level="4.3" data-path="estimated-lsi.html"><a href="estimated-lsi.html"><i class="fa fa-check"></i><b>4.3</b> Estimated LSI</a></li>
<li class="chapter" data-level="4.4" data-path="batch-effect-correction-wtih-harmony.html"><a href="batch-effect-correction-wtih-harmony.html"><i class="fa fa-check"></i><b>4.4</b> Batch Effect Correction wtih Harmony</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="clustering-with-archr.html"><a href="clustering-with-archr.html"><i class="fa fa-check"></i><b>5</b> Clustering with ArchR</a>
<ul>
<li class="chapter" data-level="5.1" data-path="clustering-using-seurats-findclusters-function.html"><a href="clustering-using-seurats-findclusters-function.html"><i class="fa fa-check"></i><b>5.1</b> Clustering using Seurat’s <code>FindClusters()</code> function</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="single-cell-embeddings.html"><a href="single-cell-embeddings.html"><i class="fa fa-check"></i><b>6</b> Single Cell Embeddings</a>
<ul>
<li class="chapter" data-level="6.1" data-path="uniform-manifold-approximation-and-projection-umap.html"><a href="uniform-manifold-approximation-and-projection-umap.html"><i class="fa fa-check"></i><b>6.1</b> Uniform Manifold Approximation and Projection (UMAP)</a></li>
<li class="chapter" data-level="6.2" data-path="t-stocastic-neighbor-embedding-t-sne.html"><a href="t-stocastic-neighbor-embedding-t-sne.html"><i class="fa fa-check"></i><b>6.2</b> t-Stocastic Neighbor Embedding (t-SNE)</a></li>
<li class="chapter" data-level="6.3" data-path="dimensionality-reduction-after-harmony.html"><a href="dimensionality-reduction-after-harmony.html"><i class="fa fa-check"></i><b>6.3</b> Dimensionality Reduction After Harmony</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="gene-scores-and-marker-genes-with-archr.html"><a href="gene-scores-and-marker-genes-with-archr.html"><i class="fa fa-check"></i><b>7</b> Gene Scores and Marker Genes with ArchR</a>
<ul>
<li class="chapter" data-level="7.1" data-path="calculating-gene-scores-in-archr.html"><a href="calculating-gene-scores-in-archr.html"><i class="fa fa-check"></i><b>7.1</b> Calculating Gene Scores in ArchR</a></li>
<li class="chapter" data-level="7.2" data-path="identification-of-marker-genes.html"><a href="identification-of-marker-genes.html"><i class="fa fa-check"></i><b>7.2</b> Identification of Marker Genes</a></li>
<li class="chapter" data-level="7.3" data-path="visualizing-marker-genes-on-an-embedding.html"><a href="visualizing-marker-genes-on-an-embedding.html"><i class="fa fa-check"></i><b>7.3</b> Visualizing Marker Genes on an Embedding</a></li>
<li class="chapter" data-level="7.4" data-path="marker-genes-imputation-with-magic.html"><a href="marker-genes-imputation-with-magic.html"><i class="fa fa-check"></i><b>7.4</b> Marker Genes Imputation with MAGIC</a></li>
<li class="chapter" data-level="7.5" data-path="track-plotting-with-archrbrowser.html"><a href="track-plotting-with-archrbrowser.html"><i class="fa fa-check"></i><b>7.5</b> Track Plotting with ArchRBrowser</a></li>
<li class="chapter" data-level="7.6" data-path="launching-the-archrbrowser.html"><a href="launching-the-archrbrowser.html"><i class="fa fa-check"></i><b>7.6</b> Launching the ArchRBrowser</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="labeling-clusters-with-scrna.html"><a href="labeling-clusters-with-scrna.html"><i class="fa fa-check"></i><b>8</b> Labeling Clusters with scRNA</a>
<ul>
<li class="chapter" data-level="8.1" data-path="labeling-scatac-cells-with-scrna-clusters.html"><a href="labeling-scatac-cells-with-scrna-clusters.html"><i class="fa fa-check"></i><b>8.1</b> Labeling scATAC cells with scRNA clusters</a></li>
<li class="chapter" data-level="8.2" data-path="adding-pseudo-scrna-profiles-for-each-scatac-cell.html"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html"><i class="fa fa-check"></i><b>8.2</b> Adding Pseudo-scRNA profiles for each scATAC cell</a></li>
<li class="chapter" data-level="8.3" data-path="labeling-scatac-clusters-with-scrna.html"><a href="labeling-scatac-clusters-with-scrna.html"><i class="fa fa-check"></i><b>8.3</b> Labeling scATAC clusters with scRNA</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="pseudo-bulk-replicates-in-archr.html"><a href="pseudo-bulk-replicates-in-archr.html"><i class="fa fa-check"></i><b>9</b> Pseudo-bulk Replicates in ArchR</a>
<ul>
<li class="chapter" data-level="9.1" data-path="how-does-archr-make-pseudo-bulk-replicates.html"><a href="how-does-archr-make-pseudo-bulk-replicates.html"><i class="fa fa-check"></i><b>9.1</b> How does ArchR make Pseudo-bulk Replicates?</a></li>
<li class="chapter" data-level="9.2" data-path="example.html"><a href="example.html"><i class="fa fa-check"></i><b>9.2</b> Example</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="calling-peaks-with-archr.html"><a href="calling-peaks-with-archr.html"><i class="fa fa-check"></i><b>10</b> Calling Peaks with ArchR</a>
<ul>
<li class="chapter" data-level="10.1" data-path="calling-peaks-w-macs2.html"><a href="calling-peaks-w-macs2.html"><i class="fa fa-check"></i><b>10.1</b> Calling Peaks w/ Macs2</a></li>
<li class="chapter" data-level="10.2" data-path="calling-peaks-w-tilematrix.html"><a href="calling-peaks-w-tilematrix.html"><i class="fa fa-check"></i><b>10.2</b> Calling Peaks w/ TileMatrix</a></li>
<li class="chapter" data-level="10.3" data-path="add-peak-matrix.html"><a href="add-peak-matrix.html"><i class="fa fa-check"></i><b>10.3</b> Add Peak Matrix</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="identifying-marker-peaks-with-archr.html"><a href="identifying-marker-peaks-with-archr.html"><i class="fa fa-check"></i><b>11</b> Identifying Marker Peaks with ArchR</a>
<ul>
<li class="chapter" data-level="11.1" data-path="identifying-marker-peaks-with-archr-1.html"><a href="identifying-marker-peaks-with-archr-1.html"><i class="fa fa-check"></i><b>11.1</b> Identifying Marker Peaks with ArchR</a></li>
<li class="chapter" data-level="11.2" data-path="pairwise-testing-groups.html"><a href="pairwise-testing-groups.html"><i class="fa fa-check"></i><b>11.2</b> Pairwise Testing Groups</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="peak-annotation-enrichment-with-archr.html"><a href="peak-annotation-enrichment-with-archr.html"><i class="fa fa-check"></i><b>12</b> Peak Annotation Enrichment with ArchR</a>
<ul>
<li class="chapter" data-level="12.1" data-path="motif-enrichment.html"><a href="motif-enrichment.html"><i class="fa fa-check"></i><b>12.1</b> Motif Enrichment</a></li>
<li class="chapter" data-level="12.2" data-path="archr-enrichment.html"><a href="archr-enrichment.html"><i class="fa fa-check"></i><b>12.2</b> ArchR Enrichment</a></li>
<li class="chapter" data-level="12.3" data-path="custom-enrichment.html"><a href="custom-enrichment.html"><i class="fa fa-check"></i><b>12.3</b> Custom Enrichment</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="chromvar-deviatons-enrichment-with-archr.html"><a href="chromvar-deviatons-enrichment-with-archr.html"><i class="fa fa-check"></i><b>13</b> ChromVAR Deviatons Enrichment with ArchR</a>
<ul>
<li class="chapter" data-level="13.1" data-path="motif-deviations.html"><a href="motif-deviations.html"><i class="fa fa-check"></i><b>13.1</b> Motif Deviations</a></li>
<li class="chapter" data-level="13.2" data-path="archr-deviations.html"><a href="archr-deviations.html"><i class="fa fa-check"></i><b>13.2</b> ArchR Deviations</a></li>
<li class="chapter" data-level="13.3" data-path="custom-deviations.html"><a href="custom-deviations.html"><i class="fa fa-check"></i><b>13.3</b> Custom Deviations</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="footprinting-with-archr.html"><a href="footprinting-with-archr.html"><i class="fa fa-check"></i><b>14</b> Footprinting with ArchR</a>
<ul>
<li class="chapter" data-level="14.1" data-path="motif-footprinting.html"><a href="motif-footprinting.html"><i class="fa fa-check"></i><b>14.1</b> Motif Footprinting</a></li>
<li class="chapter" data-level="14.2" data-path="feature-footprinting.html"><a href="feature-footprinting.html"><i class="fa fa-check"></i><b>14.2</b> Feature Footprinting</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="integrative-analysis-with-archr.html"><a href="integrative-analysis-with-archr.html"><i class="fa fa-check"></i><b>15</b> Integrative Analysis with ArchR</a>
<ul>
<li class="chapter" data-level="15.1" data-path="co-accessibility-with-archr.html"><a href="co-accessibility-with-archr.html"><i class="fa fa-check"></i><b>15.1</b> Co-accessibility with ArchR</a></li>
<li class="chapter" data-level="15.2" data-path="peak2genelinkage-with-archr.html"><a href="peak2genelinkage-with-archr.html"><i class="fa fa-check"></i><b>15.2</b> Peak2GeneLinkage with ArchR</a></li>
<li class="chapter" data-level="15.3" data-path="identification-of-positive-tf-regulators.html"><a href="identification-of-positive-tf-regulators.html"><i class="fa fa-check"></i><b>15.3</b> Identification of Positive TF-Regulators</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="trajectory-analysis-with-archr.html"><a href="trajectory-analysis-with-archr.html"><i class="fa fa-check"></i><b>16</b> Trajectory Analysis with ArchR</a>
<ul>
<li class="chapter" data-level="16.1" data-path="myeloid-trajectory.html"><a href="myeloid-trajectory.html"><i class="fa fa-check"></i><b>16.1</b> Myeloid Trajectory</a></li>
<li class="chapter" data-level="16.2" data-path="lymphoid-trajectory.html"><a href="lymphoid-trajectory.html"><i class="fa fa-check"></i><b>16.2</b> Lymphoid Trajectory</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="plot-aesthetics-in-archr.html"><a href="plot-aesthetics-in-archr.html"><i class="fa fa-check"></i><b>17</b> Plot aesthetics in ArchR</a></li>
<li class="chapter" data-level="18" data-path="session-information.html"><a href="session-information.html"><i class="fa fa-check"></i><b>18</b> Session Information</a></li>
<li class="divider"></li>
<li><a href="https://bookdown.org" target="_blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">ArchR: Robust and scaleable analysis of single-cell chromatin accessibility data.</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="labeling-scatac-cells-with-scrna-clusters" class="section level2" number="8.1">
<h2><span class="header-section-number">8.1</span> Labeling scATAC cells with scRNA clusters</h2>
<p>First we need to download scRNA data for hematopoiesis from Granja* et al (2019).</p>
<div class="sourceCode" id="cb113"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb113-1"><a href="labeling-scatac-cells-with-scrna-clusters.html#cb113-1"></a><span class="co">#111 MB Download</span></span>
<span id="cb113-2"><a href="labeling-scatac-cells-with-scrna-clusters.html#cb113-2"></a><span class="cf">if</span>(<span class="op">!</span><span class="kw">file.exists</span>(<span class="st">&quot;scRNA-Hematopoiesis-Granja-2019.rds&quot;</span>)){</span>
<span id="cb113-3"><a href="labeling-scatac-cells-with-scrna-clusters.html#cb113-3"></a>    <span class="kw">download.file</span>(</span>
<span id="cb113-4"><a href="labeling-scatac-cells-with-scrna-clusters.html#cb113-4"></a>        <span class="dt">url =</span> <span class="st">&quot;https://jeffgranja.s3.amazonaws.com/ArchR/TestData/scRNA-Hematopoiesis-Granja-2019.rds&quot;</span>,</span>
<span id="cb113-5"><a href="labeling-scatac-cells-with-scrna-clusters.html#cb113-5"></a>        <span class="dt">destfile =</span> <span class="st">&quot;scRNA-Hematopoiesis-Granja-2019.rds&quot;</span></span>
<span id="cb113-6"><a href="labeling-scatac-cells-with-scrna-clusters.html#cb113-6"></a>    )</span>
<span id="cb113-7"><a href="labeling-scatac-cells-with-scrna-clusters.html#cb113-7"></a>}</span>
<span id="cb113-8"><a href="labeling-scatac-cells-with-scrna-clusters.html#cb113-8"></a></span>
<span id="cb113-9"><a href="labeling-scatac-cells-with-scrna-clusters.html#cb113-9"></a>seRNA &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="st">&quot;scRNA-Hematopoiesis-Granja-2019.rds&quot;</span>)</span>
<span id="cb113-10"><a href="labeling-scatac-cells-with-scrna-clusters.html#cb113-10"></a>seRNA</span></code></pre></div>
<blockquote>
<p>## class: RangedSummarizedExperiment<br />
## dim: 20287 35582<br />
## metadata(0):<br />
## assays(1): counts<br />
## rownames(20287): FAM138A OR4F5 … S100B PRMT2<br />
## rowData names(3): gene_name gene_id exonLength<br />
## colnames(35582): CD34_32_R5:AAACCTGAGTATCGAA-1<br />
## CD34_32_R5:AAACCTGAGTCGTTTG-1 …<br />
## BMMC_10x_GREENLEAF_REP2:TTTGTTGCATGTGTCA-1<br />
## BMMC_10x_GREENLEAF_REP2:TTTGTTGCATTGAAAG-1<br />
## colData names(10): Group nUMI_pre … BioClassification Barcode</p>
</blockquote>
<div class="sourceCode" id="cb114"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb114-1"><a href="labeling-scatac-cells-with-scrna-clusters.html#cb114-1"></a><span class="kw">colnames</span>(<span class="kw">colData</span>(seRNA))</span></code></pre></div>
<blockquote>
<p>## [1] “Group” “nUMI_pre” “nUMI”<br />
## [4] “nGene” “initialClusters” “UMAP1”<br />
## [7] “UMAP2” “Clusters” “BioClassification”<br />
## [10] “Barcode”</p>
</blockquote>
<div class="sourceCode" id="cb115"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb115-1"><a href="labeling-scatac-cells-with-scrna-clusters.html#cb115-1"></a><span class="kw">table</span>(<span class="kw">colData</span>(seRNA)<span class="op">$</span>BioClassification)</span></code></pre></div>
<blockquote>
<p>## 01_HSC 02_Early.Eryth 03_Late.Eryth 04_Early.Baso 05_CMP.LMPP<br />
## 1425 1653 446 111 2260<br />
## 06_CLP.1 07_GMP 08_GMP.Neut 09_pDC 10_cDC<br />
## 903 2097 1050 544 325<br />
## 11_CD14.Mono.1 12_CD14.Mono.2 13_CD16.Mono 14_Unk 15_CLP.2<br />
## 1800 4222 292 520 377<br />
## 16_Pre.B 17_B 18_Plasma 19_CD8.N 20_CD4.N1<br />
## 710 1711 62 1521 2470<br />
## 21_CD4.N2 22_CD4.M 23_CD8.EM 24_CD8.CM 25_NK<br />
## 2364 3539 796 2080 2143<br />
## 26_Unk<br />
## 161</p>
</blockquote>
<p>Unconstrained Integration (not-recommended)</p>
<div class="sourceCode" id="cb116"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb116-1"><a href="labeling-scatac-cells-with-scrna-clusters.html#cb116-1"></a><span class="co">#~5 minutes</span></span>
<span id="cb116-2"><a href="labeling-scatac-cells-with-scrna-clusters.html#cb116-2"></a>projHeme2 &lt;-<span class="st"> </span><span class="kw">addGeneIntegrationMatrix</span>(</span>
<span id="cb116-3"><a href="labeling-scatac-cells-with-scrna-clusters.html#cb116-3"></a>    <span class="dt">ArchRProj =</span> projHeme2, </span>
<span id="cb116-4"><a href="labeling-scatac-cells-with-scrna-clusters.html#cb116-4"></a>    <span class="dt">useMatrix =</span> <span class="st">&quot;GeneScoreMatrix&quot;</span>,</span>
<span id="cb116-5"><a href="labeling-scatac-cells-with-scrna-clusters.html#cb116-5"></a>    <span class="dt">matrixName =</span> <span class="st">&quot;GeneIntegrationMatrix&quot;</span>,</span>
<span id="cb116-6"><a href="labeling-scatac-cells-with-scrna-clusters.html#cb116-6"></a>    <span class="dt">reducedDims =</span> <span class="st">&quot;IterativeLSI&quot;</span>,</span>
<span id="cb116-7"><a href="labeling-scatac-cells-with-scrna-clusters.html#cb116-7"></a>    <span class="dt">seRNA =</span> seRNA,</span>
<span id="cb116-8"><a href="labeling-scatac-cells-with-scrna-clusters.html#cb116-8"></a>    <span class="dt">addToArrow =</span> <span class="ot">FALSE</span>,</span>
<span id="cb116-9"><a href="labeling-scatac-cells-with-scrna-clusters.html#cb116-9"></a>    <span class="dt">groupRNA =</span> <span class="st">&quot;BioClassification&quot;</span>,</span>
<span id="cb116-10"><a href="labeling-scatac-cells-with-scrna-clusters.html#cb116-10"></a>    <span class="dt">nameCell =</span> <span class="st">&quot;predictedCell_Un&quot;</span>, <span class="co">#Name of column where cell from scRNA is matched to each cell</span></span>
<span id="cb116-11"><a href="labeling-scatac-cells-with-scrna-clusters.html#cb116-11"></a>    <span class="dt">nameGroup =</span> <span class="st">&quot;predictedGroup_Un&quot;</span>, <span class="co">#Name of column where group from scRNA is matched to each cell</span></span>
<span id="cb116-12"><a href="labeling-scatac-cells-with-scrna-clusters.html#cb116-12"></a>    <span class="dt">nameScore =</span> <span class="st">&quot;predictedScore_Un&quot;</span> <span class="co">#Name of column where prediction score from scRNA</span></span>
<span id="cb116-13"><a href="labeling-scatac-cells-with-scrna-clusters.html#cb116-13"></a>)</span></code></pre></div>
<blockquote>
<p>## ArchR logging to : ArchRLogs/ArchR-addGeneIntegrationMatrix-f6635e453cfd-Date-2020-04-15_Time-10-08-55.log<br />
## If there is an issue, please report to github with logFile!<br />
## 2020-04-15 10:08:56 : Running Seurat’s Integration Stuart* et al 2019, 0.005 mins elapsed.<br />
## 2020-04-15 10:09:06 : Checking ATAC Input, 0.172 mins elapsed.<br />
## 2020-04-15 10:09:06 : Checking RNA Input, 0.173 mins elapsed.<br />
## 2020-04-15 10:09:17 : Creating Integration Blocks, 0.367 mins elapsed.<br />
## 2020-04-15 10:09:18 : Prepping Interation Data, 0.372 mins elapsed.<br />
## 2020-04-15 10:09:19 : Computing Integration in 1 Integration Blocks!, 0 mins elapsed.<br />
## 2020-04-15 10:09:19 : Block (1 of 1) : Computing Integration, 0 mins elapsed.<br />
## 2020-04-15 10:09:23 : Block (1 of 1) : Identifying Variable Genes, 0.071 mins elapsed.<br />
## 2020-04-15 10:09:28 : Block (1 of 1) : Getting GeneScoreMatrix, 0.146 mins elapsed.<br />
## 2020-04-15 10:09:37 : Block (1 of 1) : Imputing GeneScoreMatrix, 0.298 mins elapsed.<br />
## 2020-04-15 10:09:37 : Computing Impute Weights Using Magic (Cell 2018), 0 mins elapsed.<br />
## 2020-04-15 10:09:37 : Computing Partial Diffusion Matrix with Magic (1 of 2), 0 mins elapsed.<br />
## 2020-04-15 10:09:44 : Computing Partial Diffusion Matrix with Magic (2 of 2), 0.122 mins elapsed.<br />
## 2020-04-15 10:09:51 : Completed Getting Magic Weights!, 0.242 mins elapsed.<br />
## Getting ImputeWeights<br />
## Using weights on disk<br />
## Using weights on disk<br />
## Getting ImputeWeights<br />
## 2020-04-15 10:10:17 : Block (1 of 1) : Seurat FindTransferAnchors, 0.968 mins elapsed.<br />
## 2020-04-15 10:12:00 : Block (1 of 1) : Seurat TransferData Cell Labels, 2.681 mins elapsed.<br />
## 2020-04-15 10:12:34 : Block (1 of 1) : Completed Integration, 3.243 mins elapsed.<br />
## 2020-04-15 10:12:34 : Completed Integration with RNA Matrix, 3.256 mins elapsed.<br />
## ArchR logging successful to : ArchRLogs/ArchR-addGeneIntegrationMatrix-f6635e453cfd-Date-2020-04-15_Time-10-08-55.log</p>
</blockquote>
<p>Constrained Integration (recommended)</p>
<div class="sourceCode" id="cb117"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb117-1"><a href="labeling-scatac-cells-with-scrna-clusters.html#cb117-1"></a><span class="co">#We can use the unconstrained fits to help identify which are T/NK Cells</span></span>
<span id="cb117-2"><a href="labeling-scatac-cells-with-scrna-clusters.html#cb117-2"></a>cM &lt;-<span class="st"> </span><span class="kw">as.matrix</span>(<span class="kw">confusionMatrix</span>(projHeme2<span class="op">$</span>Clusters, projHeme2<span class="op">$</span>predictedGroup_Un))</span>
<span id="cb117-3"><a href="labeling-scatac-cells-with-scrna-clusters.html#cb117-3"></a>preClust &lt;-<span class="st"> </span><span class="kw">colnames</span>(cM)[<span class="kw">apply</span>(cM, <span class="dv">1</span> , which.max)]</span>
<span id="cb117-4"><a href="labeling-scatac-cells-with-scrna-clusters.html#cb117-4"></a><span class="kw">cbind</span>(preClust, <span class="kw">rownames</span>(cM)) <span class="co">#Assignments</span></span></code></pre></div>
<blockquote>
<p>## preClust<br />
## [1,] “17_B” “C3”<br />
## [2,] “20_CD4.N1” “C8”<br />
## [3,] “16_Pre.B” “C4”<br />
## [4,] “08_GMP.Neut” “C11”<br />
## [5,] “11_CD14.Mono.1” “C1”<br />
## [6,] “01_HSC” “C12”<br />
## [7,] “03_Late.Eryth” “C10”<br />
## [8,] “22_CD4.M” “C9”<br />
## [9,] “25_NK” “C7”<br />
## [10,] “09_pDC” “C5”<br />
## [11,] “12_CD14.Mono.2” “C2”<br />
## [12,] “15_CLP.2” “C6”</p>
</blockquote>
<div class="sourceCode" id="cb118"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb118-1"><a href="labeling-scatac-cells-with-scrna-clusters.html#cb118-1"></a><span class="co">#From scRNA</span></span>
<span id="cb118-2"><a href="labeling-scatac-cells-with-scrna-clusters.html#cb118-2"></a>cTNK &lt;-<span class="st"> </span><span class="kw">paste0</span>(<span class="kw">paste0</span>(<span class="dv">19</span><span class="op">:</span><span class="dv">25</span>), <span class="dt">collapse=</span><span class="st">&quot;|&quot;</span>)</span>
<span id="cb118-3"><a href="labeling-scatac-cells-with-scrna-clusters.html#cb118-3"></a>cTNK</span></code></pre></div>
<blockquote>
<p>## [1] “19|20|21|22|23|24|25”</p>
</blockquote>
<div class="sourceCode" id="cb119"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb119-1"><a href="labeling-scatac-cells-with-scrna-clusters.html#cb119-1"></a>cNonTNK &lt;-<span class="st"> </span><span class="kw">paste0</span>(<span class="kw">c</span>(<span class="kw">paste0</span>(<span class="st">&quot;0&quot;</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">9</span>), <span class="dv">10</span><span class="op">:</span><span class="dv">13</span>, <span class="dv">15</span><span class="op">:</span><span class="dv">18</span>), <span class="dt">collapse=</span><span class="st">&quot;|&quot;</span>)</span>
<span id="cb119-2"><a href="labeling-scatac-cells-with-scrna-clusters.html#cb119-2"></a>cNonTNK</span></code></pre></div>
<blockquote>
<p>## [1] “01|02|03|04|05|06|07|08|09|10|11|12|13|15|16|17|18”</p>
</blockquote>
<div class="sourceCode" id="cb120"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb120-1"><a href="labeling-scatac-cells-with-scrna-clusters.html#cb120-1"></a><span class="co">#Assign scATAC to these categories</span></span>
<span id="cb120-2"><a href="labeling-scatac-cells-with-scrna-clusters.html#cb120-2"></a>clustTNK &lt;-<span class="st"> </span><span class="kw">rownames</span>(cM)[<span class="kw">grep</span>(cTNK, preClust)]</span>
<span id="cb120-3"><a href="labeling-scatac-cells-with-scrna-clusters.html#cb120-3"></a>clustTNK</span></code></pre></div>
<blockquote>
<p>## [1] “C8” “C9” “C7”</p>
</blockquote>
<div class="sourceCode" id="cb121"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb121-1"><a href="labeling-scatac-cells-with-scrna-clusters.html#cb121-1"></a>clustNonTNK &lt;-<span class="st"> </span><span class="kw">rownames</span>(cM)[<span class="kw">grep</span>(cNonTNK, preClust)]</span>
<span id="cb121-2"><a href="labeling-scatac-cells-with-scrna-clusters.html#cb121-2"></a>clustNonTNK</span></code></pre></div>
<blockquote>
<p>## [1] “C3” “C4” “C11” “C1” “C12” “C10” “C5” “C2” “C6”</p>
</blockquote>
<div class="sourceCode" id="cb122"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb122-1"><a href="labeling-scatac-cells-with-scrna-clusters.html#cb122-1"></a><span class="co">#RNA get cells in these categories</span></span>
<span id="cb122-2"><a href="labeling-scatac-cells-with-scrna-clusters.html#cb122-2"></a>rnaTNK &lt;-<span class="st"> </span><span class="kw">colnames</span>(seRNA)[<span class="kw">grep</span>(cTNK, <span class="kw">colData</span>(seRNA)<span class="op">$</span>BioClassification)]</span>
<span id="cb122-3"><a href="labeling-scatac-cells-with-scrna-clusters.html#cb122-3"></a><span class="kw">head</span>(rnaTNK)</span></code></pre></div>
<blockquote>
<p>## [1] “PBMC_10x_GREENLEAF_REP1:AAACCCAGTCGTCATA-1”<br />
## [2] “PBMC_10x_GREENLEAF_REP1:AAACCCATCCGATGTA-1”<br />
## [3] “PBMC_10x_GREENLEAF_REP1:AAACCCATCTCAACGA-1”<br />
## [4] “PBMC_10x_GREENLEAF_REP1:AAACCCATCTCTCGAC-1”<br />
## [5] “PBMC_10x_GREENLEAF_REP1:AAACGAACAATCGTCA-1”<br />
## [6] “PBMC_10x_GREENLEAF_REP1:AAACGAACACGATTCA-1”</p>
</blockquote>
<div class="sourceCode" id="cb123"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb123-1"><a href="labeling-scatac-cells-with-scrna-clusters.html#cb123-1"></a>rnaNonTNK &lt;-<span class="st"> </span><span class="kw">colnames</span>(seRNA)[<span class="kw">grep</span>(cNonTNK, <span class="kw">colData</span>(seRNA)<span class="op">$</span>BioClassification)]</span>
<span id="cb123-2"><a href="labeling-scatac-cells-with-scrna-clusters.html#cb123-2"></a><span class="kw">head</span>(rnaNonTNK)</span></code></pre></div>
<blockquote>
<p>## [1] “CD34_32_R5:AAACCTGAGTATCGAA-1” “CD34_32_R5:AAACCTGAGTCGTTTG-1”<br />
## [3] “CD34_32_R5:AAACCTGGTTCCACAA-1” “CD34_32_R5:AAACGGGAGCTTCGCG-1”<br />
## [5] “CD34_32_R5:AAACGGGAGGGAGTAA-1” “CD34_32_R5:AAACGGGAGTTACGGG-1”</p>
</blockquote>
<div class="sourceCode" id="cb124"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb124-1"><a href="labeling-scatac-cells-with-scrna-clusters.html#cb124-1"></a>groupList &lt;-<span class="st"> </span><span class="kw">SimpleList</span>(</span>
<span id="cb124-2"><a href="labeling-scatac-cells-with-scrna-clusters.html#cb124-2"></a>    <span class="dt">TNK =</span> <span class="kw">SimpleList</span>(</span>
<span id="cb124-3"><a href="labeling-scatac-cells-with-scrna-clusters.html#cb124-3"></a>        <span class="dt">ATAC =</span> projHeme2<span class="op">$</span>cellNames[projHeme2<span class="op">$</span>Clusters <span class="op">%in%</span><span class="st"> </span>clustTNK],</span>
<span id="cb124-4"><a href="labeling-scatac-cells-with-scrna-clusters.html#cb124-4"></a>        <span class="dt">RNA =</span> rnaTNK</span>
<span id="cb124-5"><a href="labeling-scatac-cells-with-scrna-clusters.html#cb124-5"></a>    ),</span>
<span id="cb124-6"><a href="labeling-scatac-cells-with-scrna-clusters.html#cb124-6"></a>    <span class="dt">NonTNK =</span> <span class="kw">SimpleList</span>(</span>
<span id="cb124-7"><a href="labeling-scatac-cells-with-scrna-clusters.html#cb124-7"></a>        <span class="dt">ATAC =</span> projHeme2<span class="op">$</span>cellNames[projHeme2<span class="op">$</span>Clusters <span class="op">%in%</span><span class="st"> </span>clustNonTNK],</span>
<span id="cb124-8"><a href="labeling-scatac-cells-with-scrna-clusters.html#cb124-8"></a>        <span class="dt">RNA =</span> rnaNonTNK</span>
<span id="cb124-9"><a href="labeling-scatac-cells-with-scrna-clusters.html#cb124-9"></a>    )    </span>
<span id="cb124-10"><a href="labeling-scatac-cells-with-scrna-clusters.html#cb124-10"></a>)</span>
<span id="cb124-11"><a href="labeling-scatac-cells-with-scrna-clusters.html#cb124-11"></a></span>
<span id="cb124-12"><a href="labeling-scatac-cells-with-scrna-clusters.html#cb124-12"></a><span class="co">#~5 minutes</span></span>
<span id="cb124-13"><a href="labeling-scatac-cells-with-scrna-clusters.html#cb124-13"></a>projHeme2 &lt;-<span class="st"> </span><span class="kw">addGeneIntegrationMatrix</span>(</span>
<span id="cb124-14"><a href="labeling-scatac-cells-with-scrna-clusters.html#cb124-14"></a>    <span class="dt">ArchRProj =</span> projHeme2, </span>
<span id="cb124-15"><a href="labeling-scatac-cells-with-scrna-clusters.html#cb124-15"></a>    <span class="dt">useMatrix =</span> <span class="st">&quot;GeneScoreMatrix&quot;</span>,</span>
<span id="cb124-16"><a href="labeling-scatac-cells-with-scrna-clusters.html#cb124-16"></a>    <span class="dt">matrixName =</span> <span class="st">&quot;GeneIntegrationMatrix&quot;</span>,</span>
<span id="cb124-17"><a href="labeling-scatac-cells-with-scrna-clusters.html#cb124-17"></a>    <span class="dt">reducedDims =</span> <span class="st">&quot;IterativeLSI&quot;</span>,</span>
<span id="cb124-18"><a href="labeling-scatac-cells-with-scrna-clusters.html#cb124-18"></a>    <span class="dt">seRNA =</span> seRNA,</span>
<span id="cb124-19"><a href="labeling-scatac-cells-with-scrna-clusters.html#cb124-19"></a>    <span class="dt">addToArrow =</span> <span class="ot">FALSE</span>, </span>
<span id="cb124-20"><a href="labeling-scatac-cells-with-scrna-clusters.html#cb124-20"></a>    <span class="dt">groupList =</span> groupList, <span class="co">#Constrain List</span></span>
<span id="cb124-21"><a href="labeling-scatac-cells-with-scrna-clusters.html#cb124-21"></a>    <span class="dt">groupRNA =</span> <span class="st">&quot;BioClassification&quot;</span>,</span>
<span id="cb124-22"><a href="labeling-scatac-cells-with-scrna-clusters.html#cb124-22"></a>    <span class="dt">nameCell =</span> <span class="st">&quot;predictedCell_Co&quot;</span>, <span class="co">#Name of column where cell from scRNA is matched to each cell</span></span>
<span id="cb124-23"><a href="labeling-scatac-cells-with-scrna-clusters.html#cb124-23"></a>    <span class="dt">nameGroup =</span> <span class="st">&quot;predictedGroup_Co&quot;</span>, <span class="co">#Name of column where group from scRNA is matched to each cell</span></span>
<span id="cb124-24"><a href="labeling-scatac-cells-with-scrna-clusters.html#cb124-24"></a>    <span class="dt">nameScore =</span> <span class="st">&quot;predictedScore_Co&quot;</span> <span class="co">#Name of column where prediction score from scRNA</span></span>
<span id="cb124-25"><a href="labeling-scatac-cells-with-scrna-clusters.html#cb124-25"></a>)</span></code></pre></div>
<blockquote>
<p>## ArchR logging to : ArchRLogs/ArchR-addGeneIntegrationMatrix-f663618b2629-Date-2020-04-15_Time-10-12-35.log<br />
## If there is an issue, please report to github with logFile!<br />
## 2020-04-15 10:12:36 : Running Seurat’s Integration Stuart* et al 2019, 0.014 mins elapsed.<br />
## 2020-04-15 10:12:36 : Checking ATAC Input, 0.027 mins elapsed.<br />
## 2020-04-15 10:12:36 : Checking RNA Input, 0.027 mins elapsed.<br />
## 2020-04-15 10:12:48 : Creating Integration Blocks, 0.218 mins elapsed.<br />
## 2020-04-15 10:12:48 : Prepping Interation Data, 0.222 mins elapsed.<br />
## 2020-04-15 10:12:49 : Computing Integration in 2 Integration Blocks!, 0 mins elapsed.<br />
## 2020-04-15 10:15:24 : Completed Integration with RNA Matrix, 2.574 mins elapsed.<br />
## ArchR logging successful to : ArchRLogs/ArchR-addGeneIntegrationMatrix-f663618b2629-Date-2020-04-15_Time-10-12-35.log</p>
</blockquote>
<p>Compare results</p>
<div class="sourceCode" id="cb125"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb125-1"><a href="labeling-scatac-cells-with-scrna-clusters.html#cb125-1"></a>pal &lt;-<span class="st"> </span><span class="kw">paletteDiscrete</span>(<span class="dt">values =</span> <span class="kw">colData</span>(seRNA)<span class="op">$</span>BioClassification)</span></code></pre></div>
<blockquote>
<p>## Length of unique values greater than palette, interpolating..</p>
</blockquote>
<div class="sourceCode" id="cb126"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb126-1"><a href="labeling-scatac-cells-with-scrna-clusters.html#cb126-1"></a>pal</span></code></pre></div>
<blockquote>
<p>## 01_HSC 02_Early.Eryth 03_Late.Eryth 04_Early.Baso 05_CMP.LMPP<br />
## “#D51F26” “#502A59” “#235D55” “#3D6E57” “#8D2B8B”<br />
## 06_CLP.1 07_GMP 08_GMP.Neut 09_pDC 10_cDC<br />
## “#DE6C3E” “#F9B712” “#D8CE42” “#8E9ACD” “#B774B1”<br />
## 11_CD14.Mono.1 12_CD14.Mono.2 13_CD16.Mono 14_Unk 15_CLP.2<br />
## “#D69FC8” “#C7C8DE” “#8FD3D4” “#89C86E” “#CC9672”<br />
## 16_Pre.B 17_B 18_Plasma 19_CD8.N 20_CD4.N1<br />
## “#CF7E96” “#A27AA4” “#CD4F32” “#6B977E” “#518AA3”<br />
## 21_CD4.N2 22_CD4.M 23_CD8.EM 24_CD8.CM 25_NK<br />
## “#5A5297” “#0F707D” “#5E2E32” “#A95A3C” “#B28D5C”<br />
## 26_Unk<br />
## “#3D3D3D”</p>
</blockquote>
<div class="sourceCode" id="cb127"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb127-1"><a href="labeling-scatac-cells-with-scrna-clusters.html#cb127-1"></a>p1 &lt;-<span class="st"> </span><span class="kw">plotEmbedding</span>(</span>
<span id="cb127-2"><a href="labeling-scatac-cells-with-scrna-clusters.html#cb127-2"></a>    projHeme2, </span>
<span id="cb127-3"><a href="labeling-scatac-cells-with-scrna-clusters.html#cb127-3"></a>    <span class="dt">colorBy =</span> <span class="st">&quot;cellColData&quot;</span>, </span>
<span id="cb127-4"><a href="labeling-scatac-cells-with-scrna-clusters.html#cb127-4"></a>    <span class="dt">name =</span> <span class="st">&quot;predictedGroup_Un&quot;</span>, </span>
<span id="cb127-5"><a href="labeling-scatac-cells-with-scrna-clusters.html#cb127-5"></a>    <span class="dt">pal =</span> pal</span>
<span id="cb127-6"><a href="labeling-scatac-cells-with-scrna-clusters.html#cb127-6"></a>)</span></code></pre></div>
<blockquote>
<p>## ArchR logging to : ArchRLogs/ArchR-plotEmbedding-f66351b51d5d-Date-2020-04-15_Time-10-15-31.log<br />
## If there is an issue, please report to github with logFile!<br />
## Getting UMAP Embedding<br />
## ColorBy = cellColData<br />
## Plotting Embedding<br />
## 1<br />
## ArchR logging successful to : ArchRLogs/ArchR-plotEmbedding-f66351b51d5d-Date-2020-04-15_Time-10-15-31.log</p>
</blockquote>
<div class="sourceCode" id="cb128"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb128-1"><a href="labeling-scatac-cells-with-scrna-clusters.html#cb128-1"></a>p1</span></code></pre></div>
<p><img src="images/HemeWalkthrough/PNG/Plot-UMAP-RNA-Integration_1.png" width="600" height="600" /></p>
<div class="sourceCode" id="cb129"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb129-1"><a href="labeling-scatac-cells-with-scrna-clusters.html#cb129-1"></a>p2 &lt;-<span class="st"> </span><span class="kw">plotEmbedding</span>(</span>
<span id="cb129-2"><a href="labeling-scatac-cells-with-scrna-clusters.html#cb129-2"></a>    projHeme2, </span>
<span id="cb129-3"><a href="labeling-scatac-cells-with-scrna-clusters.html#cb129-3"></a>    <span class="dt">colorBy =</span> <span class="st">&quot;cellColData&quot;</span>, </span>
<span id="cb129-4"><a href="labeling-scatac-cells-with-scrna-clusters.html#cb129-4"></a>    <span class="dt">name =</span> <span class="st">&quot;predictedGroup_Co&quot;</span>, </span>
<span id="cb129-5"><a href="labeling-scatac-cells-with-scrna-clusters.html#cb129-5"></a>    <span class="dt">pal =</span> pal</span>
<span id="cb129-6"><a href="labeling-scatac-cells-with-scrna-clusters.html#cb129-6"></a>)</span></code></pre></div>
<blockquote>
<p>## ArchR logging to : ArchRLogs/ArchR-plotEmbedding-f6632ef266a4-Date-2020-04-15_Time-10-15-49.log<br />
## If there is an issue, please report to github with logFile!<br />
## Getting UMAP Embedding<br />
## ColorBy = cellColData<br />
## Plotting Embedding<br />
## 1<br />
## ArchR logging successful to : ArchRLogs/ArchR-plotEmbedding-f6632ef266a4-Date-2020-04-15_Time-10-15-49.log</p>
</blockquote>
<div class="sourceCode" id="cb130"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb130-1"><a href="labeling-scatac-cells-with-scrna-clusters.html#cb130-1"></a>p2</span></code></pre></div>
<p><img src="images/HemeWalkthrough/PNG/Plot-UMAP-RNA-Integration_2.png" width="600" height="600" /></p>
<p>To save a nice looking pdf we use plotPDF which removes white pages and tries to make the plots
nice looking.</p>
<div class="sourceCode" id="cb131"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb131-1"><a href="labeling-scatac-cells-with-scrna-clusters.html#cb131-1"></a><span class="kw">plotPDF</span>(p1,p2, <span class="dt">name =</span> <span class="st">&quot;Plot-UMAP-RNA-Integration.pdf&quot;</span>, <span class="dt">ArchRProj =</span> projHeme2, <span class="dt">addDOC =</span> <span class="ot">FALSE</span>, <span class="dt">width =</span> <span class="dv">5</span>, <span class="dt">height =</span> <span class="dv">5</span>)</span></code></pre></div>
<blockquote>
<p>## [1] “plotting ggplot!”<br />
## [1] “plotting ggplot!”<br />
## [1] 0</p>
</blockquote>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="labeling-clusters-with-scrna.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": null
},
"fontsettings": null,
"edit": null,
"history": {
"link": "https://github.com/GreenleafLab/ArchR_Website/commits/dev/08_scRNA_Labeling.Rmd",
"text": "Chapter edit history"
},
"view": {
"link": "https://github.com/GreenleafLab/ArchR_Website/view/dev/08_scRNA_Labeling.Rmd",
"text": "View source on GitHub"
},
"download": ["ArchR.pdf"],
"toc": {
"collapse": "none"
},
"search": true,
"info": false
});
});
</script>

</body>

</html>
