<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>15.2 Lymphoid Trajectory | Chapter 15 - Cellular Trajectory Analysis with ArchR</title>
  <meta name="description" content="A guide to ArchR" />
  <meta name="generator" content="bookdown 0.17 and GitBook 2.6.7" />

  <meta property="og:title" content="15.2 Lymphoid Trajectory | Chapter 15 - Cellular Trajectory Analysis with ArchR" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="A guide to ArchR" />
  <meta name="github-repo" content="GreenleafLab/ArchR" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="15.2 Lymphoid Trajectory | Chapter 15 - Cellular Trajectory Analysis with ArchR" />
  
  <meta name="twitter:description" content="A guide to ArchR" />
  

<meta name="author" content="Jeffrey Granja and Ryan Corces" />


<meta name="date" content="2020-04-16" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="images/favicon.ico" type="image/x-icon" />
<link rel="prev" href="myeloid-trajectory.html"/>
<link rel="next" href="session-information-14.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<link href="libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="../index.html">Back to ArchR Home</a></li>

<li class="divider"></li>
<li><a href="index.html#section"><img src="images/ArchR_Logo_Integrated.png" alt="" width="200"></a></li>
<li class="chapter" data-level="1" data-path="getting-started-with-archr.html"><a href="getting-started-with-archr.html"><i class="fa fa-check"></i><b>1</b> Getting Started with ArchR</a>
<ul>
<li class="chapter" data-level="1.1" data-path="a-brief-primer-on-atac-seq-terminology.html"><a href="a-brief-primer-on-atac-seq-terminology.html"><i class="fa fa-check"></i><b>1.1</b> A Brief Primer on ATAC-seq Terminology</a></li>
<li class="chapter" data-level="1.2" data-path="why-use-archr.html"><a href="why-use-archr.html"><i class="fa fa-check"></i><b>1.2</b> Why use ArchR?</a></li>
<li class="chapter" data-level="1.3" data-path="what-is-an-arrow-file-archrproject.html"><a href="what-is-an-arrow-file-archrproject.html"><i class="fa fa-check"></i><b>1.3</b> What is an <code>Arrow file</code> / <code>ArchRProject</code>?</a></li>
<li class="chapter" data-level="1.4" data-path="input-file-types-in-archr.html"><a href="input-file-types-in-archr.html"><i class="fa fa-check"></i><b>1.4</b> Input File Types in ArchR</a></li>
<li class="chapter" data-level="1.5" data-path="getting-set-up.html"><a href="getting-set-up.html"><i class="fa fa-check"></i><b>1.5</b> Getting Set Up</a></li>
<li class="chapter" data-level="1.6" data-path="setting-a-genome-and-geneannotation.html"><a href="setting-a-genome-and-geneannotation.html"><i class="fa fa-check"></i><b>1.6</b> Setting a Genome and GeneAnnotation</a></li>
<li class="chapter" data-level="1.7" data-path="creating-arrow-files.html"><a href="creating-arrow-files.html"><i class="fa fa-check"></i><b>1.7</b> Creating Arrow Files</a></li>
<li class="chapter" data-level="1.8" data-path="session-information.html"><a href="session-information.html"><i class="fa fa-check"></i><b>1.8</b> Session Information</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="doublet-inference-with-archr.html"><a href="doublet-inference-with-archr.html"><i class="fa fa-check"></i><b>2</b> Doublet Inference with ArchR</a>
<ul>
<li class="chapter" data-level="2.1" data-path="inferring-scatac-seq-doublets-with-archr.html"><a href="inferring-scatac-seq-doublets-with-archr.html"><i class="fa fa-check"></i><b>2.1</b> Inferring scATAC-seq Doublets with ArchR</a></li>
<li class="chapter" data-level="2.2" data-path="session-information-1.html"><a href="session-information-1.html"><i class="fa fa-check"></i><b>2.2</b> Session Information</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="creating-an-archrproject.html"><a href="creating-an-archrproject.html"><i class="fa fa-check"></i><b>3</b> Creating an ArchRProject</a>
<ul>
<li class="chapter" data-level="3.1" data-path="creating-an-archrproject-1.html"><a href="creating-an-archrproject-1.html"><i class="fa fa-check"></i><b>3.1</b> Creating An ArchRProject</a></li>
<li class="chapter" data-level="3.2" data-path="manipulating-an-archrproject.html"><a href="manipulating-an-archrproject.html"><i class="fa fa-check"></i><b>3.2</b> Manipulating An ArchRProject</a></li>
<li class="chapter" data-level="3.3" data-path="plotting-sample-statistics-from-an-archrproject.html"><a href="plotting-sample-statistics-from-an-archrproject.html"><i class="fa fa-check"></i><b>3.3</b> Plotting Sample Statistics from an ArchRProject</a></li>
<li class="chapter" data-level="3.4" data-path="plotting-sample-fragment-size-distribution-and-tss-enrichment-profiles-.html"><a href="plotting-sample-fragment-size-distribution-and-tss-enrichment-profiles-.html"><i class="fa fa-check"></i><b>3.4</b> Plotting Sample Fragment Size Distribution and TSS Enrichment Profiles.</a></li>
<li class="chapter" data-level="3.5" data-path="filtering-doublets-from-an-archrproject.html"><a href="filtering-doublets-from-an-archrproject.html"><i class="fa fa-check"></i><b>3.5</b> Filtering Doublets from an ArchRProject</a></li>
<li class="chapter" data-level="3.6" data-path="session-information-2.html"><a href="session-information-2.html"><i class="fa fa-check"></i><b>3.6</b> Session Information</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="dimensionality-reduction-with-archr.html"><a href="dimensionality-reduction-with-archr.html"><i class="fa fa-check"></i><b>4</b> Dimensionality Reduction with ArchR</a>
<ul>
<li class="chapter" data-level="4.1" data-path="archrs-lsi-implementation.html"><a href="archrs-lsi-implementation.html"><i class="fa fa-check"></i><b>4.1</b> ArchRâ€™s LSI Implementation</a></li>
<li class="chapter" data-level="4.2" data-path="iterative-latent-semantic-indexing-lsi.html"><a href="iterative-latent-semantic-indexing-lsi.html"><i class="fa fa-check"></i><b>4.2</b> Iterative Latent Semantic Indexing (LSI)</a></li>
<li class="chapter" data-level="4.3" data-path="batch-effect-correction-wtih-harmony.html"><a href="batch-effect-correction-wtih-harmony.html"><i class="fa fa-check"></i><b>4.3</b> Batch Effect Correction wtih Harmony</a></li>
<li class="chapter" data-level="4.4" data-path="session-information-3.html"><a href="session-information-3.html"><i class="fa fa-check"></i><b>4.4</b> Session Information</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="clustering-with-archr.html"><a href="clustering-with-archr.html"><i class="fa fa-check"></i><b>5</b> Clustering with ArchR</a>
<ul>
<li class="chapter" data-level="5.1" data-path="session-information-4.html"><a href="session-information-4.html"><i class="fa fa-check"></i><b>5.1</b> Session Information</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="single-cell-embeddings.html"><a href="single-cell-embeddings.html"><i class="fa fa-check"></i><b>6</b> Single Cell Embeddings</a>
<ul>
<li class="chapter" data-level="6.1" data-path="umap.html"><a href="umap.html"><i class="fa fa-check"></i><b>6.1</b> UMAP</a></li>
<li class="chapter" data-level="6.2" data-path="tsne.html"><a href="tsne.html"><i class="fa fa-check"></i><b>6.2</b> tSNE</a></li>
<li class="chapter" data-level="6.3" data-path="umap-of-previous-reduced-dimensions.html"><a href="umap-of-previous-reduced-dimensions.html"><i class="fa fa-check"></i><b>6.3</b> UMAP of previous Reduced Dimensions</a></li>
<li class="chapter" data-level="6.4" data-path="tsne-of-previous-reduced-dimensions.html"><a href="tsne-of-previous-reduced-dimensions.html"><i class="fa fa-check"></i><b>6.4</b> tSNE of previous Reduced Dimensions</a></li>
<li class="chapter" data-level="6.5" data-path="session-information-5.html"><a href="session-information-5.html"><i class="fa fa-check"></i><b>6.5</b> Session Information</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="marker-genes-with-archr.html"><a href="marker-genes-with-archr.html"><i class="fa fa-check"></i><b>7</b> Marker Genes with ArchR</a>
<ul>
<li class="chapter" data-level="7.1" data-path="identification-of-marker-genes.html"><a href="identification-of-marker-genes.html"><i class="fa fa-check"></i><b>7.1</b> Identification of Marker Genes</a></li>
<li class="chapter" data-level="7.2" data-path="marker-genes.html"><a href="marker-genes.html"><i class="fa fa-check"></i><b>7.2</b> Marker Genes</a></li>
<li class="chapter" data-level="7.3" data-path="marker-genes-imputation-with-magic.html"><a href="marker-genes-imputation-with-magic.html"><i class="fa fa-check"></i><b>7.3</b> Marker Genes Imputation with Magic</a></li>
<li class="chapter" data-level="7.4" data-path="track-plotting-with-archrbrowser.html"><a href="track-plotting-with-archrbrowser.html"><i class="fa fa-check"></i><b>7.4</b> Track Plotting with ArchRBrowser</a></li>
<li class="chapter" data-level="7.5" data-path="launching-the-archrbrowser.html"><a href="launching-the-archrbrowser.html"><i class="fa fa-check"></i><b>7.5</b> Launching the ArchRBrowser</a></li>
<li class="chapter" data-level="7.6" data-path="session-information-6.html"><a href="session-information-6.html"><i class="fa fa-check"></i><b>7.6</b> Session Information</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="labeling-clusters-with-scrna.html"><a href="labeling-clusters-with-scrna.html"><i class="fa fa-check"></i><b>8</b> Labeling Clusters with scRNA</a>
<ul>
<li class="chapter" data-level="8.1" data-path="labeling-scatac-cells-with-scrna-clusters.html"><a href="labeling-scatac-cells-with-scrna-clusters.html"><i class="fa fa-check"></i><b>8.1</b> Labeling scATAC cells with scRNA clusters</a></li>
<li class="chapter" data-level="8.2" data-path="adding-pseudo-scrna-profiles-for-each-scatac-cell.html"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html"><i class="fa fa-check"></i><b>8.2</b> Adding Pseudo-scRNA profiles for each scATAC cell</a></li>
<li class="chapter" data-level="8.3" data-path="labeling-scatac-clusters-with-scrna.html"><a href="labeling-scatac-clusters-with-scrna.html"><i class="fa fa-check"></i><b>8.3</b> Labeling scATAC clusters with scRNA</a></li>
<li class="chapter" data-level="8.4" data-path="session-information-7.html"><a href="session-information-7.html"><i class="fa fa-check"></i><b>8.4</b> Session Information</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="calling-peaks-with-archr.html"><a href="calling-peaks-with-archr.html"><i class="fa fa-check"></i><b>9</b> Calling Peaks with ArchR</a>
<ul>
<li class="chapter" data-level="9.1" data-path="calling-peaks-w-macs2.html"><a href="calling-peaks-w-macs2.html"><i class="fa fa-check"></i><b>9.1</b> Calling Peaks w/ Macs2</a></li>
<li class="chapter" data-level="9.2" data-path="calling-peaks-w-tilematrix.html"><a href="calling-peaks-w-tilematrix.html"><i class="fa fa-check"></i><b>9.2</b> Calling Peaks w/ TileMatrix</a></li>
<li class="chapter" data-level="9.3" data-path="add-peak-matrix.html"><a href="add-peak-matrix.html"><i class="fa fa-check"></i><b>9.3</b> Add Peak Matrix</a></li>
<li class="chapter" data-level="9.4" data-path="session-information-8.html"><a href="session-information-8.html"><i class="fa fa-check"></i><b>9.4</b> Session Information</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="identifying-marker-peaks-with-archr.html"><a href="identifying-marker-peaks-with-archr.html"><i class="fa fa-check"></i><b>10</b> Identifying Marker Peaks with ArchR</a>
<ul>
<li class="chapter" data-level="10.1" data-path="identifying-marker-peaks-with-archr-1.html"><a href="identifying-marker-peaks-with-archr-1.html"><i class="fa fa-check"></i><b>10.1</b> Identifying Marker Peaks with ArchR</a></li>
<li class="chapter" data-level="10.2" data-path="pairwise-testing-groups.html"><a href="pairwise-testing-groups.html"><i class="fa fa-check"></i><b>10.2</b> Pairwise Testing Groups</a></li>
<li class="chapter" data-level="10.3" data-path="session-information-9.html"><a href="session-information-9.html"><i class="fa fa-check"></i><b>10.3</b> Session Information</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="peak-annotation-enrichment-with-archr.html"><a href="peak-annotation-enrichment-with-archr.html"><i class="fa fa-check"></i><b>11</b> Peak Annotation Enrichment with ArchR</a>
<ul>
<li class="chapter" data-level="11.1" data-path="motif-enrichment.html"><a href="motif-enrichment.html"><i class="fa fa-check"></i><b>11.1</b> Motif Enrichment</a></li>
<li class="chapter" data-level="11.2" data-path="archr-enrichment.html"><a href="archr-enrichment.html"><i class="fa fa-check"></i><b>11.2</b> ArchR Enrichment</a></li>
<li class="chapter" data-level="11.3" data-path="custom-enrichment.html"><a href="custom-enrichment.html"><i class="fa fa-check"></i><b>11.3</b> Custom Enrichment</a></li>
<li class="chapter" data-level="11.4" data-path="session-information-10.html"><a href="session-information-10.html"><i class="fa fa-check"></i><b>11.4</b> Session Information</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="chromvar-deviatons-enrichment-with-archr.html"><a href="chromvar-deviatons-enrichment-with-archr.html"><i class="fa fa-check"></i><b>12</b> ChromVAR Deviatons Enrichment with ArchR</a>
<ul>
<li class="chapter" data-level="12.1" data-path="motif-deviations.html"><a href="motif-deviations.html"><i class="fa fa-check"></i><b>12.1</b> Motif Deviations</a></li>
<li class="chapter" data-level="12.2" data-path="archr-deviations.html"><a href="archr-deviations.html"><i class="fa fa-check"></i><b>12.2</b> ArchR Deviations</a></li>
<li class="chapter" data-level="12.3" data-path="custom-deviations.html"><a href="custom-deviations.html"><i class="fa fa-check"></i><b>12.3</b> Custom Deviations</a></li>
<li class="chapter" data-level="12.4" data-path="session-information-11.html"><a href="session-information-11.html"><i class="fa fa-check"></i><b>12.4</b> Session Information</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="footprinting-with-archr.html"><a href="footprinting-with-archr.html"><i class="fa fa-check"></i><b>13</b> Footprinting with ArchR</a>
<ul>
<li class="chapter" data-level="13.1" data-path="motif-footprinting.html"><a href="motif-footprinting.html"><i class="fa fa-check"></i><b>13.1</b> Motif Footprinting</a></li>
<li class="chapter" data-level="13.2" data-path="feature-footprinting.html"><a href="feature-footprinting.html"><i class="fa fa-check"></i><b>13.2</b> Feature Footprinting</a></li>
<li class="chapter" data-level="13.3" data-path="session-information-12.html"><a href="session-information-12.html"><i class="fa fa-check"></i><b>13.3</b> Session Information</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="integrative-analysis-with-archr.html"><a href="integrative-analysis-with-archr.html"><i class="fa fa-check"></i><b>14</b> Integrative Analysis with ArchR</a>
<ul>
<li class="chapter" data-level="14.1" data-path="coaccessibility-with-archr.html"><a href="coaccessibility-with-archr.html"><i class="fa fa-check"></i><b>14.1</b> CoAccessibility with ArchR</a></li>
<li class="chapter" data-level="14.2" data-path="peak2genelinkage-with-archr.html"><a href="peak2genelinkage-with-archr.html"><i class="fa fa-check"></i><b>14.2</b> Peak2GeneLinkage with ArchR</a></li>
<li class="chapter" data-level="14.3" data-path="identification-of-positive-tf-regulators.html"><a href="identification-of-positive-tf-regulators.html"><i class="fa fa-check"></i><b>14.3</b> Identification of Positive TF-Regulators</a></li>
<li class="chapter" data-level="14.4" data-path="session-information-13.html"><a href="session-information-13.html"><i class="fa fa-check"></i><b>14.4</b> Session Information</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="trajectory-analysis-with-archr.html"><a href="trajectory-analysis-with-archr.html"><i class="fa fa-check"></i><b>15</b> Trajectory Analysis with ArchR</a>
<ul>
<li class="chapter" data-level="15.1" data-path="myeloid-trajectory.html"><a href="myeloid-trajectory.html"><i class="fa fa-check"></i><b>15.1</b> Myeloid Trajectory</a></li>
<li class="chapter" data-level="15.2" data-path="lymphoid-trajectory.html"><a href="lymphoid-trajectory.html"><i class="fa fa-check"></i><b>15.2</b> Lymphoid Trajectory</a></li>
<li class="chapter" data-level="15.3" data-path="session-information-14.html"><a href="session-information-14.html"><i class="fa fa-check"></i><b>15.3</b> Session Information</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://bookdown.org" target="_blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Chapter 15 - Cellular Trajectory Analysis with ArchR</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="lymphoid-trajectory" class="section level2" number="15.2">
<h2><span class="header-section-number">15.2</span> Lymphoid Trajectory</h2>
<div class="sourceCode" id="cb242"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb242-1"><a href="lymphoid-trajectory.html#cb242-1"></a>p1 &lt;-<span class="st"> </span><span class="kw">plotEmbedding</span>(<span class="dt">ArchRProj =</span> projHeme5, <span class="dt">colorBy =</span> <span class="st">&quot;cellColData&quot;</span>, <span class="dt">name =</span> <span class="st">&quot;Clusters&quot;</span>, <span class="dt">embedding =</span> <span class="st">&quot;UMAP&quot;</span>)</span>
<span id="cb242-2"><a href="lymphoid-trajectory.html#cb242-2"></a>p2 &lt;-<span class="st"> </span><span class="kw">plotEmbedding</span>(<span class="dt">ArchRProj =</span> projHeme5, <span class="dt">colorBy =</span> <span class="st">&quot;cellColData&quot;</span>, <span class="dt">name =</span> <span class="st">&quot;Clusters2&quot;</span>, <span class="dt">embedding =</span> <span class="st">&quot;UMAP&quot;</span>)</span>
<span id="cb242-3"><a href="lymphoid-trajectory.html#cb242-3"></a><span class="kw">ggAlignPlots</span>(p1, p2, <span class="dt">type =</span> <span class="st">&quot;h&quot;</span>)</span></code></pre></div>
<p><img src="images/HemeWalkthrough/PNG/Plot-UMAP-Clusters12-Combined_1.png" width="800" height="400" /></p>
<div class="sourceCode" id="cb243"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb243-1"><a href="lymphoid-trajectory.html#cb243-1"></a><span class="co">#Trajectory to add</span></span>
<span id="cb243-2"><a href="lymphoid-trajectory.html#cb243-2"></a>trajectory &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Progenitor&quot;</span>, <span class="st">&quot;CLP&quot;</span>, <span class="st">&quot;PreB&quot;</span>, <span class="st">&quot;B&quot;</span>)</span>
<span id="cb243-3"><a href="lymphoid-trajectory.html#cb243-3"></a>trajectory</span>
<span id="cb243-4"><a href="lymphoid-trajectory.html#cb243-4"></a><span class="co"># [1] &quot;Cluster6&quot;  &quot;Cluster13&quot; &quot;Cluster12&quot; &quot;Cluster11&quot;</span></span>
<span id="cb243-5"><a href="lymphoid-trajectory.html#cb243-5"></a></span>
<span id="cb243-6"><a href="lymphoid-trajectory.html#cb243-6"></a><span class="co">#First we need to create a Trajectory and add it to ArchRProj cellColData</span></span>
<span id="cb243-7"><a href="lymphoid-trajectory.html#cb243-7"></a>projHeme5 &lt;-<span class="st"> </span><span class="kw">addTrajectory</span>(</span>
<span id="cb243-8"><a href="lymphoid-trajectory.html#cb243-8"></a>    <span class="dt">ArchRProj =</span> projHeme5, </span>
<span id="cb243-9"><a href="lymphoid-trajectory.html#cb243-9"></a>    <span class="dt">name =</span> <span class="st">&quot;LymphoidU&quot;</span>, </span>
<span id="cb243-10"><a href="lymphoid-trajectory.html#cb243-10"></a>    <span class="dt">groupBy =</span> <span class="st">&quot;Clusters2&quot;</span>,</span>
<span id="cb243-11"><a href="lymphoid-trajectory.html#cb243-11"></a>    <span class="dt">trajectory =</span> trajectory, </span>
<span id="cb243-12"><a href="lymphoid-trajectory.html#cb243-12"></a>    <span class="dt">embedding =</span> <span class="st">&quot;UMAP&quot;</span>, </span>
<span id="cb243-13"><a href="lymphoid-trajectory.html#cb243-13"></a>    <span class="dt">force =</span> <span class="ot">TRUE</span></span>
<span id="cb243-14"><a href="lymphoid-trajectory.html#cb243-14"></a>)</span>
<span id="cb243-15"><a href="lymphoid-trajectory.html#cb243-15"></a></span>
<span id="cb243-16"><a href="lymphoid-trajectory.html#cb243-16"></a><span class="kw">head</span>(projHeme5<span class="op">$</span>LymphoidU[<span class="op">!</span><span class="kw">is.na</span>(projHeme5<span class="op">$</span>LymphoidU)]) <span class="co">#NA means not in trajectory</span></span>
<span id="cb243-17"><a href="lymphoid-trajectory.html#cb243-17"></a><span class="co"># [1]       NA       NA 80.56020       NA 79.47324       NA</span></span></code></pre></div>
<div class="sourceCode" id="cb244"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb244-1"><a href="lymphoid-trajectory.html#cb244-1"></a>p &lt;-<span class="st"> </span><span class="kw">plotTrajectory</span>(projHeme5, <span class="dt">trajectory =</span> <span class="st">&quot;LymphoidU&quot;</span>, <span class="dt">colorBy =</span> <span class="st">&quot;colData&quot;</span>, <span class="dt">name =</span> <span class="st">&quot;LymphoidU&quot;</span>)</span>
<span id="cb244-2"><a href="lymphoid-trajectory.html#cb244-2"></a>p[[<span class="dv">1</span>]]</span></code></pre></div>
<p><img src="images/HemeWalkthrough/PNG/Plot-LymphoidU-Traj-UMAP_1.png" width="500" height="500" /></p>
<p>To save a nice looking pdf we use plotPDF which removes white pages and tries to make the plots
nice looking.</p>
<div class="sourceCode" id="cb245"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb245-1"><a href="lymphoid-trajectory.html#cb245-1"></a><span class="kw">plotPDF</span>(p, <span class="dt">name =</span> <span class="st">&quot;Plot-LymphoidU-Traj-UMAP.pdf&quot;</span>, <span class="dt">ArchRProj =</span> projHeme5, <span class="dt">addDOC =</span> <span class="ot">FALSE</span>, <span class="dt">width =</span> <span class="dv">5</span>, <span class="dt">height =</span> <span class="dv">5</span>)</span></code></pre></div>
<p>We can overlay other features on the UMAP</p>
<div class="sourceCode" id="cb246"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb246-1"><a href="lymphoid-trajectory.html#cb246-1"></a>p1 &lt;-<span class="st"> </span><span class="kw">plotTrajectory</span>(projHeme5, <span class="dt">trajectory =</span> <span class="st">&quot;LymphoidU&quot;</span>, <span class="dt">colorBy =</span> <span class="st">&quot;GeneScoreMatrix&quot;</span>, <span class="dt">name =</span> <span class="st">&quot;PAX5&quot;</span>, <span class="dt">continuousSet =</span> <span class="st">&quot;horizonExtra&quot;</span>)</span>
<span id="cb246-2"><a href="lymphoid-trajectory.html#cb246-2"></a>p2 &lt;-<span class="st"> </span><span class="kw">plotTrajectory</span>(projHeme5, <span class="dt">trajectory =</span> <span class="st">&quot;LymphoidU&quot;</span>, <span class="dt">colorBy =</span> <span class="st">&quot;GeneIntegrationMatrix&quot;</span>, <span class="dt">name =</span> <span class="st">&quot;PAX5&quot;</span>, <span class="dt">continuousSet =</span> <span class="st">&quot;blueYellow&quot;</span>)</span>
<span id="cb246-3"><a href="lymphoid-trajectory.html#cb246-3"></a><span class="kw">ggAlignPlots</span>(p1[[<span class="dv">1</span>]], p2[[<span class="dv">1</span>]], <span class="dt">type =</span> <span class="st">&quot;h&quot;</span>)</span></code></pre></div>
<p><img src="images/HemeWalkthrough/PNG/Plot-UMAP-PAX5-Combined_1.png" width="800" height="400" /></p>
<div class="sourceCode" id="cb247"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb247-1"><a href="lymphoid-trajectory.html#cb247-1"></a><span class="kw">ggAlignPlots</span>(p1[[<span class="dv">2</span>]], p2[[<span class="dv">2</span>]], <span class="dt">type =</span> <span class="st">&quot;h&quot;</span>)</span></code></pre></div>
<p><img src="images/HemeWalkthrough/PNG/Plot-UMAP-PAX5-Combined2_1.png" width="800" height="400" /></p>
<p>We can then make a heatmap for</p>
<div class="sourceCode" id="cb248"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb248-1"><a href="lymphoid-trajectory.html#cb248-1"></a>trajMM  &lt;-<span class="st"> </span><span class="kw">getTrajectory</span>(<span class="dt">ArchRProj =</span> projHeme5, <span class="dt">name =</span> <span class="st">&quot;LymphoidU&quot;</span>, <span class="dt">useMatrix =</span> <span class="st">&quot;MotifMatrix&quot;</span>, <span class="dt">log2Norm =</span> <span class="ot">FALSE</span>)</span>
<span id="cb248-2"><a href="lymphoid-trajectory.html#cb248-2"></a>p1 &lt;-<span class="st"> </span><span class="kw">trajectoryHeatmap</span>(trajMM, <span class="dt">pal =</span> <span class="kw">paletteContinuous</span>(<span class="dt">set =</span> <span class="st">&quot;solarExtra&quot;</span>))</span></code></pre></div>
<p><img src="images/HemeWalkthrough/PNG/Plot-LymphoidU-Traj-Heatmaps_1.png" width="600" height="800" /></p>
<div class="sourceCode" id="cb249"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb249-1"><a href="lymphoid-trajectory.html#cb249-1"></a>trajGSM &lt;-<span class="st"> </span><span class="kw">getTrajectory</span>(<span class="dt">ArchRProj =</span> projHeme5, <span class="dt">name =</span> <span class="st">&quot;LymphoidU&quot;</span>, <span class="dt">useMatrix =</span> <span class="st">&quot;GeneScoreMatrix&quot;</span>, <span class="dt">log2Norm =</span> <span class="ot">TRUE</span>)</span>
<span id="cb249-2"><a href="lymphoid-trajectory.html#cb249-2"></a>p2 &lt;-<span class="st"> </span><span class="kw">trajectoryHeatmap</span>(trajGSM,  <span class="dt">pal =</span> <span class="kw">paletteContinuous</span>(<span class="dt">set =</span> <span class="st">&quot;horizonExtra&quot;</span>))</span></code></pre></div>
<p><img src="images/HemeWalkthrough/PNG/Plot-LymphoidU-Traj-Heatmaps_2.png" width="600" height="800" /></p>
<div class="sourceCode" id="cb250"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb250-1"><a href="lymphoid-trajectory.html#cb250-1"></a>trajGIM &lt;-<span class="st"> </span><span class="kw">getTrajectory</span>(<span class="dt">ArchRProj =</span> projHeme5, <span class="dt">name =</span> <span class="st">&quot;LymphoidU&quot;</span>, <span class="dt">useMatrix =</span> <span class="st">&quot;GeneIntegrationMatrix&quot;</span>, <span class="dt">log2Norm =</span> <span class="ot">FALSE</span>)</span>
<span id="cb250-2"><a href="lymphoid-trajectory.html#cb250-2"></a>p3 &lt;-<span class="st"> </span><span class="kw">trajectoryHeatmap</span>(trajGIM,  <span class="dt">pal =</span> <span class="kw">paletteContinuous</span>(<span class="dt">set =</span> <span class="st">&quot;blueYellow&quot;</span>))</span></code></pre></div>
<p><img src="images/HemeWalkthrough/PNG/Plot-LymphoidU-Traj-Heatmaps_3.png" width="600" height="800" /></p>
<div class="sourceCode" id="cb251"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb251-1"><a href="lymphoid-trajectory.html#cb251-1"></a>trajPM  &lt;-<span class="st"> </span><span class="kw">getTrajectory</span>(<span class="dt">ArchRProj =</span> projHeme5, <span class="dt">name =</span> <span class="st">&quot;LymphoidU&quot;</span>, <span class="dt">useMatrix =</span> <span class="st">&quot;PeakMatrix&quot;</span>, <span class="dt">log2Norm =</span> <span class="ot">TRUE</span>)</span>
<span id="cb251-2"><a href="lymphoid-trajectory.html#cb251-2"></a>p4 &lt;-<span class="st"> </span><span class="kw">trajectoryHeatmap</span>(trajPM, <span class="dt">pal =</span> <span class="kw">paletteContinuous</span>(<span class="dt">set =</span> <span class="st">&quot;solarExtra&quot;</span>))</span></code></pre></div>
<p><img src="images/HemeWalkthrough/PNG/Plot-LymphoidU-Traj-Heatmaps_4.png" width="600" height="800" /></p>
<p>To save a nice looking pdf we use plotPDF which removes white pages and tries to make the plots
nice looking.</p>
<div class="sourceCode" id="cb252"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb252-1"><a href="lymphoid-trajectory.html#cb252-1"></a><span class="kw">plotPDF</span>(p1, p2, p3, p4, <span class="dt">name =</span> <span class="st">&quot;Plot-LymphoidU-Traj-Heatmaps.pdf&quot;</span>, <span class="dt">ArchRProj =</span> projHeme5, <span class="dt">addDOC =</span> <span class="ot">FALSE</span>, <span class="dt">width =</span> <span class="dv">6</span>, <span class="dt">height =</span> <span class="dv">8</span>)</span></code></pre></div>
<p>We can also perform integrative analyses such as identifying positive regulators with RNA</p>
<div class="sourceCode" id="cb253"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb253-1"><a href="lymphoid-trajectory.html#cb253-1"></a><span class="co">#Gene Scores</span></span>
<span id="cb253-2"><a href="lymphoid-trajectory.html#cb253-2"></a>corGSM_MM &lt;-<span class="st"> </span><span class="kw">correlateTrajectories</span>(trajGSM, trajMM)</span>
<span id="cb253-3"><a href="lymphoid-trajectory.html#cb253-3"></a>corGSM_MM[[<span class="dv">1</span>]]<span class="op">$</span>matchname1</span>
<span id="cb253-4"><a href="lymphoid-trajectory.html#cb253-4"></a>corGSM_MM[[<span class="dv">1</span>]]</span>
<span id="cb253-5"><a href="lymphoid-trajectory.html#cb253-5"></a></span>
<span id="cb253-6"><a href="lymphoid-trajectory.html#cb253-6"></a><span class="co">#We can plot these identified regulators</span></span>
<span id="cb253-7"><a href="lymphoid-trajectory.html#cb253-7"></a>trajGSM2 &lt;-<span class="st"> </span>trajGSM[corGSM_MM[[<span class="dv">1</span>]]<span class="op">$</span>name1, ]</span>
<span id="cb253-8"><a href="lymphoid-trajectory.html#cb253-8"></a>trajMM2 &lt;-<span class="st"> </span>trajMM[corGSM_MM[[<span class="dv">1</span>]]<span class="op">$</span>name2, ]</span>
<span id="cb253-9"><a href="lymphoid-trajectory.html#cb253-9"></a></span>
<span id="cb253-10"><a href="lymphoid-trajectory.html#cb253-10"></a><span class="co">#To Order We Can Create a Trajectory of these assays Multiplied</span></span>
<span id="cb253-11"><a href="lymphoid-trajectory.html#cb253-11"></a>trajCombined &lt;-<span class="st"> </span>trajGSM2</span>
<span id="cb253-12"><a href="lymphoid-trajectory.html#cb253-12"></a><span class="kw">assay</span>(trajCombined) &lt;-<span class="st"> </span><span class="kw">t</span>(<span class="kw">apply</span>(<span class="kw">assay</span>(trajGSM2), <span class="dv">1</span>, scale)) <span class="op">+</span><span class="st"> </span><span class="kw">t</span>(<span class="kw">apply</span>(<span class="kw">assay</span>(trajMM2), <span class="dv">1</span>, scale))</span>
<span id="cb253-13"><a href="lymphoid-trajectory.html#cb253-13"></a></span>
<span id="cb253-14"><a href="lymphoid-trajectory.html#cb253-14"></a><span class="co">#We can get the row order from the heatmap function</span></span>
<span id="cb253-15"><a href="lymphoid-trajectory.html#cb253-15"></a>combinedMat &lt;-<span class="st"> </span><span class="kw">trajectoryHeatmap</span>(trajCombined, <span class="dt">returnMat =</span> <span class="ot">TRUE</span>, <span class="dt">varCutOff =</span> <span class="dv">0</span>)</span>
<span id="cb253-16"><a href="lymphoid-trajectory.html#cb253-16"></a>rowOrder &lt;-<span class="st"> </span><span class="kw">match</span>(<span class="kw">rownames</span>(combinedMat), <span class="kw">rownames</span>(trajGSM2))</span>
<span id="cb253-17"><a href="lymphoid-trajectory.html#cb253-17"></a></span>
<span id="cb253-18"><a href="lymphoid-trajectory.html#cb253-18"></a>ht1 &lt;-<span class="st"> </span><span class="kw">trajectoryHeatmap</span>(trajGSM2,  <span class="dt">pal =</span> <span class="kw">paletteContinuous</span>(<span class="dt">set =</span> <span class="st">&quot;horizonExtra&quot;</span>),  <span class="dt">varCutOff =</span> <span class="dv">0</span>, <span class="dt">rowOrder =</span> rowOrder)</span>
<span id="cb253-19"><a href="lymphoid-trajectory.html#cb253-19"></a>ht2 &lt;-<span class="st"> </span><span class="kw">trajectoryHeatmap</span>(trajMM2,  <span class="dt">pal =</span> <span class="kw">paletteContinuous</span>(<span class="dt">set =</span> <span class="st">&quot;solarExtra&quot;</span>), <span class="dt">varCutOff =</span> <span class="dv">0</span>, <span class="dt">rowOrder =</span> rowOrder)</span>
<span id="cb253-20"><a href="lymphoid-trajectory.html#cb253-20"></a></span>
<span id="cb253-21"><a href="lymphoid-trajectory.html#cb253-21"></a>ht1 <span class="op">+</span><span class="st"> </span>ht2</span></code></pre></div>
<p><img src="images/HemeWalkthrough/PNG/Plot-LymphoidU-GS-TF_1.png" width="800" height="600" /></p>
<div class="sourceCode" id="cb254"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb254-1"><a href="lymphoid-trajectory.html#cb254-1"></a><span class="co">#Gene Expression</span></span>
<span id="cb254-2"><a href="lymphoid-trajectory.html#cb254-2"></a>corGIM_MM &lt;-<span class="st"> </span><span class="kw">correlateTrajectories</span>(trajGIM, trajMM)</span>
<span id="cb254-3"><a href="lymphoid-trajectory.html#cb254-3"></a>corGIM_MM[[<span class="dv">1</span>]]<span class="op">$</span>matchname1</span>
<span id="cb254-4"><a href="lymphoid-trajectory.html#cb254-4"></a>corGIM_MM[[<span class="dv">1</span>]]</span>
<span id="cb254-5"><a href="lymphoid-trajectory.html#cb254-5"></a></span>
<span id="cb254-6"><a href="lymphoid-trajectory.html#cb254-6"></a><span class="co">#We can plot these identified regulators</span></span>
<span id="cb254-7"><a href="lymphoid-trajectory.html#cb254-7"></a>trajGIM2 &lt;-<span class="st"> </span>trajGIM[corGIM_MM[[<span class="dv">1</span>]]<span class="op">$</span>name1, ]</span>
<span id="cb254-8"><a href="lymphoid-trajectory.html#cb254-8"></a>trajMM2 &lt;-<span class="st"> </span>trajMM[corGIM_MM[[<span class="dv">1</span>]]<span class="op">$</span>name2, ]</span>
<span id="cb254-9"><a href="lymphoid-trajectory.html#cb254-9"></a></span>
<span id="cb254-10"><a href="lymphoid-trajectory.html#cb254-10"></a><span class="co">#To Order We Can Create a Trajectory of these assays Multiplied</span></span>
<span id="cb254-11"><a href="lymphoid-trajectory.html#cb254-11"></a>trajCombined &lt;-<span class="st"> </span>trajGIM2</span>
<span id="cb254-12"><a href="lymphoid-trajectory.html#cb254-12"></a><span class="kw">assay</span>(trajCombined) &lt;-<span class="st"> </span><span class="kw">t</span>(<span class="kw">apply</span>(<span class="kw">assay</span>(trajGIM2), <span class="dv">1</span>, scale)) <span class="op">+</span><span class="st"> </span><span class="kw">t</span>(<span class="kw">apply</span>(<span class="kw">assay</span>(trajMM2), <span class="dv">1</span>, scale))</span>
<span id="cb254-13"><a href="lymphoid-trajectory.html#cb254-13"></a></span>
<span id="cb254-14"><a href="lymphoid-trajectory.html#cb254-14"></a><span class="co">#We can get the row order from the heatmap function</span></span>
<span id="cb254-15"><a href="lymphoid-trajectory.html#cb254-15"></a>combinedMat &lt;-<span class="st"> </span><span class="kw">trajectoryHeatmap</span>(trajCombined, <span class="dt">returnMat =</span> <span class="ot">TRUE</span>, <span class="dt">varCutOff =</span> <span class="dv">0</span>)</span>
<span id="cb254-16"><a href="lymphoid-trajectory.html#cb254-16"></a>rowOrder &lt;-<span class="st"> </span><span class="kw">match</span>(<span class="kw">rownames</span>(combinedMat), <span class="kw">rownames</span>(trajGIM2))</span>
<span id="cb254-17"><a href="lymphoid-trajectory.html#cb254-17"></a></span>
<span id="cb254-18"><a href="lymphoid-trajectory.html#cb254-18"></a><span class="co">#Now we can plot the heatmaps</span></span>
<span id="cb254-19"><a href="lymphoid-trajectory.html#cb254-19"></a>ht1 &lt;-<span class="st"> </span><span class="kw">trajectoryHeatmap</span>(trajGIM2,  <span class="dt">pal =</span> <span class="kw">paletteContinuous</span>(<span class="dt">set =</span> <span class="st">&quot;blueYellow&quot;</span>),  <span class="dt">varCutOff =</span> <span class="dv">0</span>, <span class="dt">rowOrder =</span> rowOrder)</span>
<span id="cb254-20"><a href="lymphoid-trajectory.html#cb254-20"></a>ht2 &lt;-<span class="st"> </span><span class="kw">trajectoryHeatmap</span>(trajMM2, <span class="dt">pal =</span> <span class="kw">paletteContinuous</span>(<span class="dt">set =</span> <span class="st">&quot;solarExtra&quot;</span>), <span class="dt">varCutOff =</span> <span class="dv">0</span>, <span class="dt">rowOrder =</span> rowOrder)</span>
<span id="cb254-21"><a href="lymphoid-trajectory.html#cb254-21"></a></span>
<span id="cb254-22"><a href="lymphoid-trajectory.html#cb254-22"></a>ht1 <span class="op">+</span><span class="st"> </span>ht2</span></code></pre></div>
<p><img src="images/HemeWalkthrough/PNG/Plot-LymphoidU-GEx-TF_1.png" width="800" height="600" /></p>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="myeloid-trajectory.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="session-information-14.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": null
},
"fontsettings": null,
"edit": null,
"history": {
"link": "https://github.com/GreenleafLab/ArchR_Website/commits/dev/15_Trajectory.Rmd",
"text": "Chapter edit history"
},
"view": {
"link": "https://github.com/GreenleafLab/ArchR_Website/view/dev/15_Trajectory.Rmd",
"text": "View source on GitHub"
},
"download": ["ArchR.pdf"],
"toc": {
"collapse": "none"
},
"search": true,
"info": false
});
});
</script>

</body>

</html>
