<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>18.3 Monocle3 Trajectories | ArchR: Robust and scaleable analysis of single-cell chromatin accessibility data.</title>
  <meta name="description" content="A guide to ArchR" />
  <meta name="generator" content="bookdown 0.41 and GitBook 2.6.7" />

  <meta property="og:title" content="18.3 Monocle3 Trajectories | ArchR: Robust and scaleable analysis of single-cell chromatin accessibility data." />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="A guide to ArchR" />
  <meta name="github-repo" content="GreenleafLab/ArchR" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="18.3 Monocle3 Trajectories | ArchR: Robust and scaleable analysis of single-cell chromatin accessibility data." />
  
  <meta name="twitter:description" content="A guide to ArchR" />
  

<meta name="author" content="Jeffrey Granja and Ryan Corces" />


<meta name="date" content="2025-01-23" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="images/favicon.ico" type="image/x-icon" />
<link rel="prev" href="lymphoid-trajectory---b-cell-cifferentiation.html"/>
<link rel="next" href="slingshot-trajectories.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<link href="libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="../index.html">Back to ArchR Home</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><img src="images/ArchR_Logo_Integrated.png" alt="" width="200"></a></li>
<li class="chapter" data-level="1" data-path="debugging-and-troubleshooting-in-archr.html"><a href="debugging-and-troubleshooting-in-archr.html"><i class="fa fa-check"></i><b>1</b> Debugging and Troubleshooting in ArchR</a></li>
<li class="chapter" data-level="2" data-path="manage-archrs-dependencies.html"><a href="manage-archrs-dependencies.html"><i class="fa fa-check"></i><b>2</b> Manage ArchR’s Dependencies</a>
<ul>
<li class="chapter" data-level="2.1" data-path="using-renv-to-manage-dependencies.html"><a href="using-renv-to-manage-dependencies.html"><i class="fa fa-check"></i><b>2.1</b> Using renv to manage dependencies</a></li>
<li class="chapter" data-level="2.2" data-path="other-environment-managers.html"><a href="other-environment-managers.html"><i class="fa fa-check"></i><b>2.2</b> Other environment managers</a></li>
<li class="chapter" data-level="2.3" data-path="using-docker-for-pre-compiled-environment-setup.html"><a href="using-docker-for-pre-compiled-environment-setup.html"><i class="fa fa-check"></i><b>2.3</b> Using Docker for pre-compiled environment setup</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="getting-started-with-archr.html"><a href="getting-started-with-archr.html"><i class="fa fa-check"></i><b>3</b> Getting Started with ArchR</a>
<ul>
<li class="chapter" data-level="3.1" data-path="a-brief-primer-on-atac-seq-terminology.html"><a href="a-brief-primer-on-atac-seq-terminology.html"><i class="fa fa-check"></i><b>3.1</b> A Brief Primer on ATAC-seq Terminology</a></li>
<li class="chapter" data-level="3.2" data-path="why-use-archr.html"><a href="why-use-archr.html"><i class="fa fa-check"></i><b>3.2</b> Why use ArchR?</a></li>
<li class="chapter" data-level="3.3" data-path="what-is-an-arrow-file-archrproject.html"><a href="what-is-an-arrow-file-archrproject.html"><i class="fa fa-check"></i><b>3.3</b> What is an Arrow file / <code>ArchRProject</code>?</a></li>
<li class="chapter" data-level="3.4" data-path="input-file-types-in-archr.html"><a href="input-file-types-in-archr.html"><i class="fa fa-check"></i><b>3.4</b> Input File Types in ArchR</a></li>
<li class="chapter" data-level="3.5" data-path="getting-set-up.html"><a href="getting-set-up.html"><i class="fa fa-check"></i><b>3.5</b> Getting Set Up</a></li>
<li class="chapter" data-level="3.6" data-path="creating-arrow-files.html"><a href="creating-arrow-files.html"><i class="fa fa-check"></i><b>3.6</b> Creating Arrow Files</a></li>
<li class="chapter" data-level="3.7" data-path="per-cell-quality-control.html"><a href="per-cell-quality-control.html"><i class="fa fa-check"></i><b>3.7</b> Per-cell Quality Control</a></li>
<li class="chapter" data-level="3.8" data-path="using-bam-files-for-arrow-file-creation.html"><a href="using-bam-files-for-arrow-file-creation.html"><i class="fa fa-check"></i><b>3.8</b> Using BAM files for Arrow File creation</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="doublet-inference-with-archr.html"><a href="doublet-inference-with-archr.html"><i class="fa fa-check"></i><b>4</b> Doublet Inference with ArchR</a>
<ul>
<li class="chapter" data-level="4.1" data-path="how-does-doublet-identification-work-in-archr.html"><a href="how-does-doublet-identification-work-in-archr.html"><i class="fa fa-check"></i><b>4.1</b> How does doublet identification work in ArchR?</a></li>
<li class="chapter" data-level="4.2" data-path="inferring-scatac-seq-doublets-with-archr.html"><a href="inferring-scatac-seq-doublets-with-archr.html"><i class="fa fa-check"></i><b>4.2</b> Inferring scATAC-seq Doublets with ArchR</a></li>
<li class="chapter" data-level="4.3" data-path="using-demuxlet-with-archr.html"><a href="using-demuxlet-with-archr.html"><i class="fa fa-check"></i><b>4.3</b> Using <em>demuxlet</em> with ArchR</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="creating-an-archrproject.html"><a href="creating-an-archrproject.html"><i class="fa fa-check"></i><b>5</b> Creating an ArchRProject</a>
<ul>
<li class="chapter" data-level="5.1" data-path="creating-an-archrproject-1.html"><a href="creating-an-archrproject-1.html"><i class="fa fa-check"></i><b>5.1</b> Creating An ArchRProject</a></li>
<li class="chapter" data-level="5.2" data-path="manipulating-an-archrproject.html"><a href="manipulating-an-archrproject.html"><i class="fa fa-check"></i><b>5.2</b> Manipulating An ArchRProject</a></li>
<li class="chapter" data-level="5.3" data-path="plotting-sample-statistics-from-an-archrproject.html"><a href="plotting-sample-statistics-from-an-archrproject.html"><i class="fa fa-check"></i><b>5.3</b> Plotting Sample Statistics from an ArchRProject</a></li>
<li class="chapter" data-level="5.4" data-path="plotting-sample-fragment-size-distribution-and-tss-enrichment-profiles..html"><a href="plotting-sample-fragment-size-distribution-and-tss-enrichment-profiles..html"><i class="fa fa-check"></i><b>5.4</b> Plotting Sample Fragment Size Distribution and TSS Enrichment Profiles.</a></li>
<li class="chapter" data-level="5.5" data-path="saving-and-loading-an-archrproject.html"><a href="saving-and-loading-an-archrproject.html"><i class="fa fa-check"></i><b>5.5</b> Saving and Loading an <code>ArchRProject</code></a></li>
<li class="chapter" data-level="5.6" data-path="filtering-doublets-from-an-archrproject.html"><a href="filtering-doublets-from-an-archrproject.html"><i class="fa fa-check"></i><b>5.6</b> Filtering Doublets from an ArchRProject</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="dimensionality-reduction-with-archr.html"><a href="dimensionality-reduction-with-archr.html"><i class="fa fa-check"></i><b>6</b> Dimensionality Reduction with ArchR</a>
<ul>
<li class="chapter" data-level="6.1" data-path="archrs-lsi-implementation.html"><a href="archrs-lsi-implementation.html"><i class="fa fa-check"></i><b>6.1</b> ArchR’s LSI Implementation</a></li>
<li class="chapter" data-level="6.2" data-path="iterative-latent-semantic-indexing-lsi.html"><a href="iterative-latent-semantic-indexing-lsi.html"><i class="fa fa-check"></i><b>6.2</b> Iterative Latent Semantic Indexing (LSI)</a></li>
<li class="chapter" data-level="6.3" data-path="estimated-lsi.html"><a href="estimated-lsi.html"><i class="fa fa-check"></i><b>6.3</b> Estimated LSI</a></li>
<li class="chapter" data-level="6.4" data-path="batch-effect-correction-wtih-harmony.html"><a href="batch-effect-correction-wtih-harmony.html"><i class="fa fa-check"></i><b>6.4</b> Batch Effect Correction wtih Harmony</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="clustering-with-archr.html"><a href="clustering-with-archr.html"><i class="fa fa-check"></i><b>7</b> Clustering with ArchR</a>
<ul>
<li class="chapter" data-level="7.1" data-path="clustering-using-seurats-findclusters-function.html"><a href="clustering-using-seurats-findclusters-function.html"><i class="fa fa-check"></i><b>7.1</b> Clustering using Seurat’s <code>FindClusters()</code> function</a></li>
<li class="chapter" data-level="7.2" data-path="clustering-using-scran.html"><a href="clustering-using-scran.html"><i class="fa fa-check"></i><b>7.2</b> Clustering using <code>scran</code></a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="single-cell-embeddings.html"><a href="single-cell-embeddings.html"><i class="fa fa-check"></i><b>8</b> Single-cell Embeddings</a>
<ul>
<li class="chapter" data-level="8.1" data-path="uniform-manifold-approximation-and-projection-umap.html"><a href="uniform-manifold-approximation-and-projection-umap.html"><i class="fa fa-check"></i><b>8.1</b> Uniform Manifold Approximation and Projection (UMAP)</a></li>
<li class="chapter" data-level="8.2" data-path="t-stocastic-neighbor-embedding-t-sne.html"><a href="t-stocastic-neighbor-embedding-t-sne.html"><i class="fa fa-check"></i><b>8.2</b> t-Stocastic Neighbor Embedding (t-SNE)</a></li>
<li class="chapter" data-level="8.3" data-path="dimensionality-reduction-after-harmony.html"><a href="dimensionality-reduction-after-harmony.html"><i class="fa fa-check"></i><b>8.3</b> Dimensionality Reduction After Harmony</a></li>
<li class="chapter" data-level="8.4" data-path="highlighting-specific-cells-on-an-embedding.html"><a href="highlighting-specific-cells-on-an-embedding.html"><i class="fa fa-check"></i><b>8.4</b> Highlighting specific cells on an embedding</a></li>
<li class="chapter" data-level="8.5" data-path="importing-an-embedding-from-external-software.html"><a href="importing-an-embedding-from-external-software.html"><i class="fa fa-check"></i><b>8.5</b> Importing an embedding from external software</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="gene-scores-and-marker-genes-with-archr.html"><a href="gene-scores-and-marker-genes-with-archr.html"><i class="fa fa-check"></i><b>9</b> Gene Scores and Marker Genes with ArchR</a>
<ul>
<li class="chapter" data-level="9.1" data-path="calculating-gene-scores-in-archr.html"><a href="calculating-gene-scores-in-archr.html"><i class="fa fa-check"></i><b>9.1</b> Calculating Gene Scores in ArchR</a></li>
<li class="chapter" data-level="9.2" data-path="identification-of-marker-features.html"><a href="identification-of-marker-features.html"><i class="fa fa-check"></i><b>9.2</b> Identification of Marker Features</a></li>
<li class="chapter" data-level="9.3" data-path="identifying-marker-genes.html"><a href="identifying-marker-genes.html"><i class="fa fa-check"></i><b>9.3</b> Identifying Marker Genes</a></li>
<li class="chapter" data-level="9.4" data-path="visualizing-marker-genes-on-an-embedding.html"><a href="visualizing-marker-genes-on-an-embedding.html"><i class="fa fa-check"></i><b>9.4</b> Visualizing Marker Genes on an Embedding</a></li>
<li class="chapter" data-level="9.5" data-path="marker-genes-imputation-with-magic.html"><a href="marker-genes-imputation-with-magic.html"><i class="fa fa-check"></i><b>9.5</b> Marker Genes Imputation with MAGIC</a></li>
<li class="chapter" data-level="9.6" data-path="module-scores.html"><a href="module-scores.html"><i class="fa fa-check"></i><b>9.6</b> Module Scores</a></li>
<li class="chapter" data-level="9.7" data-path="track-plotting-with-archrbrowser.html"><a href="track-plotting-with-archrbrowser.html"><i class="fa fa-check"></i><b>9.7</b> Track Plotting with ArchRBrowser</a></li>
<li class="chapter" data-level="9.8" data-path="launching-the-interactive-archrbrowser.html"><a href="launching-the-interactive-archrbrowser.html"><i class="fa fa-check"></i><b>9.8</b> Launching the Interactive ArchRBrowser</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="defining-cluster-identity-with-scrna-seq.html"><a href="defining-cluster-identity-with-scrna-seq.html"><i class="fa fa-check"></i><b>10</b> Defining Cluster Identity with scRNA-seq</a>
<ul>
<li class="chapter" data-level="10.1" data-path="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html"><i class="fa fa-check"></i><b>10.1</b> Cross-platform linkage of scATAC-seq cells with scRNA-seq cells</a></li>
<li class="chapter" data-level="10.2" data-path="adding-pseudo-scrna-seq-profiles-for-each-scatac-seq-cell.html"><a href="adding-pseudo-scrna-seq-profiles-for-each-scatac-seq-cell.html"><i class="fa fa-check"></i><b>10.2</b> Adding Pseudo-scRNA-seq profiles for each scATAC-seq cell</a></li>
<li class="chapter" data-level="10.3" data-path="labeling-scatac-seq-clusters-with-scrna-seq-information.html"><a href="labeling-scatac-seq-clusters-with-scrna-seq-information.html"><i class="fa fa-check"></i><b>10.3</b> Labeling scATAC-seq clusters with scRNA-seq information</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="pseudo-bulk-replicates-in-archr.html"><a href="pseudo-bulk-replicates-in-archr.html"><i class="fa fa-check"></i><b>11</b> Pseudo-bulk Replicates in ArchR</a>
<ul>
<li class="chapter" data-level="11.1" data-path="how-does-archr-make-pseudo-bulk-replicates.html"><a href="how-does-archr-make-pseudo-bulk-replicates.html"><i class="fa fa-check"></i><b>11.1</b> How Does ArchR Make Pseudo-bulk Replicates?</a></li>
<li class="chapter" data-level="11.2" data-path="making-pseudo-bulk-replicates.html"><a href="making-pseudo-bulk-replicates.html"><i class="fa fa-check"></i><b>11.2</b> Making Pseudo-bulk Replicates</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="calling-peaks-with-archr.html"><a href="calling-peaks-with-archr.html"><i class="fa fa-check"></i><b>12</b> Calling Peaks with ArchR</a>
<ul>
<li class="chapter" data-level="12.1" data-path="the-iterative-overlap-peak-merging-procedure.html"><a href="the-iterative-overlap-peak-merging-procedure.html"><i class="fa fa-check"></i><b>12.1</b> The Iterative Overlap Peak Merging Procedure</a></li>
<li class="chapter" data-level="12.2" data-path="calling-peaks-w-macs2.html"><a href="calling-peaks-w-macs2.html"><i class="fa fa-check"></i><b>12.2</b> Calling Peaks w/ Macs2</a></li>
<li class="chapter" data-level="12.3" data-path="calling-peaks-w-tilematrix.html"><a href="calling-peaks-w-tilematrix.html"><i class="fa fa-check"></i><b>12.3</b> Calling Peaks w/ TileMatrix</a></li>
<li class="chapter" data-level="12.4" data-path="adding-a-peak-matrix.html"><a href="adding-a-peak-matrix.html"><i class="fa fa-check"></i><b>12.4</b> Adding a Peak Matrix</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="identifying-marker-peaks.html"><a href="identifying-marker-peaks.html"><i class="fa fa-check"></i><b>13</b> Identifying Marker Peaks</a>
<ul>
<li class="chapter" data-level="13.1" data-path="identifying-marker-peaks-with-archr.html"><a href="identifying-marker-peaks-with-archr.html"><i class="fa fa-check"></i><b>13.1</b> Identifying Marker Peaks with ArchR</a></li>
<li class="chapter" data-level="13.2" data-path="plotting-marker-peaks-in-archr.html"><a href="plotting-marker-peaks-in-archr.html"><i class="fa fa-check"></i><b>13.2</b> Plotting Marker Peaks in ArchR</a></li>
<li class="chapter" data-level="13.3" data-path="pairwise-testing-between-groups.html"><a href="pairwise-testing-between-groups.html"><i class="fa fa-check"></i><b>13.3</b> Pairwise Testing Between Groups</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="motif-and-feature-enrichment-with-archr.html"><a href="motif-and-feature-enrichment-with-archr.html"><i class="fa fa-check"></i><b>14</b> Motif and Feature Enrichment with ArchR</a>
<ul>
<li class="chapter" data-level="14.1" data-path="motif-enrichment.html"><a href="motif-enrichment.html"><i class="fa fa-check"></i><b>14.1</b> Motif Enrichment</a></li>
<li class="chapter" data-level="14.2" data-path="archr-enrichment.html"><a href="archr-enrichment.html"><i class="fa fa-check"></i><b>14.2</b> ArchR Enrichment</a></li>
<li class="chapter" data-level="14.3" data-path="custom-enrichment.html"><a href="custom-enrichment.html"><i class="fa fa-check"></i><b>14.3</b> Custom Enrichment</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="chromvar-deviatons-enrichment-with-archr.html"><a href="chromvar-deviatons-enrichment-with-archr.html"><i class="fa fa-check"></i><b>15</b> ChromVAR Deviatons Enrichment with ArchR</a>
<ul>
<li class="chapter" data-level="15.1" data-path="motif-deviations.html"><a href="motif-deviations.html"><i class="fa fa-check"></i><b>15.1</b> Motif Deviations</a></li>
<li class="chapter" data-level="15.2" data-path="archr-and-custom-deviations.html"><a href="archr-and-custom-deviations.html"><i class="fa fa-check"></i><b>15.2</b> ArchR and Custom Deviations</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="footprinting-with-archr.html"><a href="footprinting-with-archr.html"><i class="fa fa-check"></i><b>16</b> Footprinting with ArchR</a>
<ul>
<li class="chapter" data-level="16.1" data-path="motif-footprinting.html"><a href="motif-footprinting.html"><i class="fa fa-check"></i><b>16.1</b> Motif Footprinting</a></li>
<li class="chapter" data-level="16.2" data-path="normalization-of-footprints-for-tn5-bias.html"><a href="normalization-of-footprints-for-tn5-bias.html"><i class="fa fa-check"></i><b>16.2</b> Normalization of Footprints for Tn5 Bias</a></li>
<li class="chapter" data-level="16.3" data-path="feature-footprinting.html"><a href="feature-footprinting.html"><i class="fa fa-check"></i><b>16.3</b> Feature Footprinting</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="integrative-analysis-with-archr.html"><a href="integrative-analysis-with-archr.html"><i class="fa fa-check"></i><b>17</b> Integrative Analysis with ArchR</a>
<ul>
<li class="chapter" data-level="17.1" data-path="creating-low-overlapping-aggregates-of-cells.html"><a href="creating-low-overlapping-aggregates-of-cells.html"><i class="fa fa-check"></i><b>17.1</b> Creating Low-Overlapping Aggregates of Cells</a></li>
<li class="chapter" data-level="17.2" data-path="co-accessibility-with-archr.html"><a href="co-accessibility-with-archr.html"><i class="fa fa-check"></i><b>17.2</b> Co-accessibility with ArchR</a></li>
<li class="chapter" data-level="17.3" data-path="peak2genelinkage-with-archr.html"><a href="peak2genelinkage-with-archr.html"><i class="fa fa-check"></i><b>17.3</b> Peak2GeneLinkage with ArchR</a></li>
<li class="chapter" data-level="17.4" data-path="identification-of-positive-tf-regulators.html"><a href="identification-of-positive-tf-regulators.html"><i class="fa fa-check"></i><b>17.4</b> Identification of Positive TF-Regulators</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="trajectory-analysis-with-archr.html"><a href="trajectory-analysis-with-archr.html"><i class="fa fa-check"></i><b>18</b> Trajectory Analysis with ArchR</a>
<ul>
<li class="chapter" data-level="18.1" data-path="myeloid-trajectory---monocyte-differentiation.html"><a href="myeloid-trajectory---monocyte-differentiation.html"><i class="fa fa-check"></i><b>18.1</b> Myeloid Trajectory - Monocyte Differentiation</a></li>
<li class="chapter" data-level="18.2" data-path="lymphoid-trajectory---b-cell-cifferentiation.html"><a href="lymphoid-trajectory---b-cell-cifferentiation.html"><i class="fa fa-check"></i><b>18.2</b> Lymphoid Trajectory - B Cell Cifferentiation</a></li>
<li class="chapter" data-level="18.3" data-path="monocle3-trajectories.html"><a href="monocle3-trajectories.html"><i class="fa fa-check"></i><b>18.3</b> Monocle3 Trajectories</a></li>
<li class="chapter" data-level="18.4" data-path="slingshot-trajectories.html"><a href="slingshot-trajectories.html"><i class="fa fa-check"></i><b>18.4</b> Slingshot Trajectories</a></li>
<li class="chapter" data-level="18.5" data-path="comparing-the-different-trajectories.html"><a href="comparing-the-different-trajectories.html"><i class="fa fa-check"></i><b>18.5</b> Comparing the different trajectories</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="handling-archr-output.html"><a href="handling-archr-output.html"><i class="fa fa-check"></i><b>19</b> Handling ArchR Output</a>
<ul>
<li class="chapter" data-level="19.1" data-path="exporting-fragment-level-data.html"><a href="exporting-fragment-level-data.html"><i class="fa fa-check"></i><b>19.1</b> Exporting fragment-level data</a></li>
<li class="chapter" data-level="19.2" data-path="exporting-matrix-level-data.html"><a href="exporting-matrix-level-data.html"><i class="fa fa-check"></i><b>19.2</b> Exporting matrix-level data</a></li>
<li class="chapter" data-level="19.3" data-path="exporting-pseudo-bulked-data-to-a-summarizedexperiment.html"><a href="exporting-pseudo-bulked-data-to-a-summarizedexperiment.html"><i class="fa fa-check"></i><b>19.3</b> Exporting pseudo-bulked data to a <code>SummarizedExperiment</code></a></li>
<li class="chapter" data-level="19.4" data-path="exporting-pseudo-bulked-data-to-a-bigwig-file.html"><a href="exporting-pseudo-bulked-data-to-a-bigwig-file.html"><i class="fa fa-check"></i><b>19.4</b> Exporting pseudo-bulked data to a bigWig file</a></li>
</ul></li>
<li class="chapter" data-level="20" data-path="integration-with-bulk-atac-seq.html"><a href="integration-with-bulk-atac-seq.html"><i class="fa fa-check"></i><b>20</b> Integration with bulk ATAC-seq</a>
<ul>
<li class="chapter" data-level="20.1" data-path="projecting-bulk-atac-seq-data.html"><a href="projecting-bulk-atac-seq-data.html"><i class="fa fa-check"></i><b>20.1</b> Projecting bulk ATAC-seq data</a></li>
</ul></li>
<li class="chapter" data-level="21" data-path="multiomic-data-analysis-in-archr.html"><a href="multiomic-data-analysis-in-archr.html"><i class="fa fa-check"></i><b>21</b> Multiomic data analysis in ArchR</a>
<ul>
<li class="chapter" data-level="21.1" data-path="importing-data-and-setting-up-a-multiome-project.html"><a href="importing-data-and-setting-up-a-multiome-project.html"><i class="fa fa-check"></i><b>21.1</b> Importing data and setting up a Multiome project</a></li>
<li class="chapter" data-level="21.2" data-path="analysis-of-multiome-data-in-archr.html"><a href="analysis-of-multiome-data-in-archr.html"><i class="fa fa-check"></i><b>21.2</b> Analysis of multiome data in ArchR</a></li>
</ul></li>
<li class="chapter" data-level="22" data-path="plot-aesthetics-in-archr.html"><a href="plot-aesthetics-in-archr.html"><i class="fa fa-check"></i><b>22</b> Plot aesthetics in ArchR</a>
<ul>
<li class="chapter" data-level="22.1" data-path="using-and-manipulating-palettes-in-archr.html"><a href="using-and-manipulating-palettes-in-archr.html"><i class="fa fa-check"></i><b>22.1</b> Using and manipulating palettes in ArchR</a></li>
</ul></li>
<li class="chapter" data-level="23" data-path="testing-your-archr-installation.html"><a href="testing-your-archr-installation.html"><i class="fa fa-check"></i><b>23</b> Testing Your ArchR Installation</a></li>
<li class="chapter" data-level="24" data-path="session-information.html"><a href="session-information.html"><i class="fa fa-check"></i><b>24</b> Session Information</a></li>
<li class="divider"></li>
<li><a href="https://bookdown.org" target="_blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">ArchR: Robust and scaleable analysis of single-cell chromatin accessibility data.</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="monocle3-trajectories" class="section level2 hasAnchor" number="18.3">
<h2><span class="header-section-number">18.3</span> Monocle3 Trajectories<a href="monocle3-trajectories.html#monocle3-trajectories" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>ArchR can also handle trajectories derived in <a href="https://cole-trapnell-lab.github.io/monocle3/">Monocle3</a>. This is handled by the <code>getMonocleTrajectories()</code> and <code>addMonocleTrajectory()</code> functions. Before diving in, it is worth noting that much of the infrastructure that ArchR uses for storing and visualizing trajectories is based on the assumption that the trajectory is linear and the trajectories determined by Monocle3 are not necessarily linear. Because of this, we either have to (i) force Monocle3 to create a linear trajectory by limiting the cell groups that it uses, or (ii) to treat the resulting Monocle3 trajectory as a collection of individual trajectories coming from a single cellular group. We illustrate these two workflows below.</p>
<p>Though this is slightly counter to the typical ArchR workflow, we first run <code>getMonocleTrajectories()</code> to retrieve the trajectory from Monocle3. In this function call, <code>principalGroup</code> indicates the apex group from which the trajectory will emanate and <code>useGroups</code> tells Monocle3 all of the groups that should be mapped on the trajectory.</p>
<p>First, we will create a linear trajectory of the lymphoid lineage. The same workflow would apply for the myeloid lineage but we are just going to create the lymphoid trajectory here to enable downstream comparison of ArchR, Monocle3, and Slingshot trajectories in a subsequent secion of this book.</p>
<div class="sourceCode" id="cb416"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb416-1"><a href="monocle3-trajectories.html#cb416-1" tabindex="-1"></a>monocle_lymphoid_solo <span class="ot">&lt;-</span> <span class="fu">getMonocleTrajectories</span>(</span>
<span id="cb416-2"><a href="monocle3-trajectories.html#cb416-2" tabindex="-1"></a>    <span class="at">ArchRProj =</span> projHeme5, </span>
<span id="cb416-3"><a href="monocle3-trajectories.html#cb416-3" tabindex="-1"></a>    <span class="at">name =</span> <span class="st">&quot;Monocle_Lymphoid_solo&quot;</span>,</span>
<span id="cb416-4"><a href="monocle3-trajectories.html#cb416-4" tabindex="-1"></a>    <span class="at">useGroups =</span> <span class="fu">c</span>(<span class="st">&quot;Progenitor&quot;</span>,<span class="st">&quot;CLP&quot;</span>,<span class="st">&quot;PreB&quot;</span>,<span class="st">&quot;B&quot;</span>),</span>
<span id="cb416-5"><a href="monocle3-trajectories.html#cb416-5" tabindex="-1"></a>    <span class="at">principalGroup =</span> <span class="st">&quot;Progenitor&quot;</span>,</span>
<span id="cb416-6"><a href="monocle3-trajectories.html#cb416-6" tabindex="-1"></a>    <span class="at">groupBy =</span> <span class="st">&quot;Clusters2&quot;</span>,</span>
<span id="cb416-7"><a href="monocle3-trajectories.html#cb416-7" tabindex="-1"></a>    <span class="at">embedding =</span> <span class="st">&quot;UMAP&quot;</span>,</span>
<span id="cb416-8"><a href="monocle3-trajectories.html#cb416-8" tabindex="-1"></a>    <span class="at">clusterParams =</span> <span class="fu">list</span>(<span class="at">k =</span> <span class="dv">50</span>),</span>
<span id="cb416-9"><a href="monocle3-trajectories.html#cb416-9" tabindex="-1"></a>    <span class="at">seed =</span> <span class="dv">1</span></span>
<span id="cb416-10"><a href="monocle3-trajectories.html#cb416-10" tabindex="-1"></a>)</span>
<span id="cb416-11"><a href="monocle3-trajectories.html#cb416-11" tabindex="-1"></a><span class="do">## Running Monocole3 Trajectory Infrastructure!</span></span>
<span id="cb416-12"><a href="monocle3-trajectories.html#cb416-12" tabindex="-1"></a><span class="do">## Adding Embedding</span></span>
<span id="cb416-13"><a href="monocle3-trajectories.html#cb416-13" tabindex="-1"></a><span class="do">## Clustering Embedding</span></span>
<span id="cb416-14"><a href="monocle3-trajectories.html#cb416-14" tabindex="-1"></a><span class="do">## Learning Graphs</span></span>
<span id="cb416-15"><a href="monocle3-trajectories.html#cb416-15" tabindex="-1"></a><span class="do">##   |                                                                              |                                                                      |   0%  |                                                                              |======================================================================| 100%</span></span>
<span id="cb416-16"><a href="monocle3-trajectories.html#cb416-16" tabindex="-1"></a><span class="do">## Getting Principal Node</span></span>
<span id="cb416-17"><a href="monocle3-trajectories.html#cb416-17" tabindex="-1"></a><span class="do">## Ordering Cells</span></span>
<span id="cb416-18"><a href="monocle3-trajectories.html#cb416-18" tabindex="-1"></a><span class="do">## Plotting Results - /workspace/ArchR/ArchR_Website_Testing/bookdown/Save-ProjHeme4/Monocole3/Plot-Results-Monocle_Lymphoid_solo.pdf</span></span></code></pre></div>
<p>You’ll notice that during the above function call, ArchR tells you that the trajectory results were plotted to a PDF called <code>Monocole3/Plot-Results-Monocle_Lymphoid_solo.pdf</code> which you can reference to see the trajectory results.</p>
<p>After we have retrieved this trajectory object from Monocle3 (called a “cell data set” or CDS), we need to add this information to our <code>ArchRProject</code> object. To do this, we use <code>addMonocleTrajectory()</code> and pass it the output of <code>getMonocleTrajectories()</code> to the <code>monocleCDS</code> parameter.</p>
<div class="sourceCode" id="cb417"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb417-1"><a href="monocle3-trajectories.html#cb417-1" tabindex="-1"></a>projHeme5 <span class="ot">&lt;-</span> <span class="fu">addMonocleTrajectory</span>(</span>
<span id="cb417-2"><a href="monocle3-trajectories.html#cb417-2" tabindex="-1"></a>    <span class="at">ArchRProj =</span> projHeme5,</span>
<span id="cb417-3"><a href="monocle3-trajectories.html#cb417-3" tabindex="-1"></a>    <span class="at">name =</span> <span class="st">&quot;Monocle_Lymphoid_solo&quot;</span>,</span>
<span id="cb417-4"><a href="monocle3-trajectories.html#cb417-4" tabindex="-1"></a>    <span class="at">useGroups =</span> <span class="fu">c</span>(<span class="st">&quot;Progenitor&quot;</span>,<span class="st">&quot;CLP&quot;</span>,<span class="st">&quot;PreB&quot;</span>,<span class="st">&quot;B&quot;</span>),</span>
<span id="cb417-5"><a href="monocle3-trajectories.html#cb417-5" tabindex="-1"></a>    <span class="at">groupBy =</span> <span class="st">&quot;Clusters2&quot;</span>,</span>
<span id="cb417-6"><a href="monocle3-trajectories.html#cb417-6" tabindex="-1"></a>    <span class="at">monocleCDS =</span> monocle_lymphoid_solo,</span>
<span id="cb417-7"><a href="monocle3-trajectories.html#cb417-7" tabindex="-1"></a>    <span class="at">force =</span> <span class="cn">TRUE</span></span>
<span id="cb417-8"><a href="monocle3-trajectories.html#cb417-8" tabindex="-1"></a>)</span></code></pre></div>
<p>Now, we can use <code>plotTrajectory()</code> to plot as before. You’ll notice that we’ve set the parameter <code>addArrow = FALSE</code> because the fitted lines often look a little off for these non-ArchR trajectories.</p>
<div class="sourceCode" id="cb418"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb418-1"><a href="monocle3-trajectories.html#cb418-1" tabindex="-1"></a>p_lymphoid_solo <span class="ot">&lt;-</span> <span class="fu">plotTrajectory</span>(projHeme5, <span class="at">trajectory =</span> <span class="st">&quot;Monocle_Lymphoid_solo&quot;</span>, <span class="at">colorBy =</span> <span class="st">&quot;cellColData&quot;</span>, <span class="at">name =</span> <span class="st">&quot;Monocle_Lymphoid_solo&quot;</span>, <span class="at">addArrow =</span> <span class="cn">FALSE</span>)</span>
<span id="cb418-2"><a href="monocle3-trajectories.html#cb418-2" tabindex="-1"></a><span class="do">## ArchR logging to : ArchRLogs/ArchR-plotTrajectory-1a140888767-Date-2025-01-23_Time-23-06-28.428608.log</span></span>
<span id="cb418-3"><a href="monocle3-trajectories.html#cb418-3" tabindex="-1"></a><span class="do">## If there is an issue, please report to github with logFile!</span></span>
<span id="cb418-4"><a href="monocle3-trajectories.html#cb418-4" tabindex="-1"></a><span class="do">## Plotting</span></span>
<span id="cb418-5"><a href="monocle3-trajectories.html#cb418-5" tabindex="-1"></a><span class="do">## Plotting Trajectory</span></span>
<span id="cb418-6"><a href="monocle3-trajectories.html#cb418-6" tabindex="-1"></a><span class="do">## ArchR logging successful to : ArchRLogs/ArchR-plotTrajectory-1a140888767-Date-2025-01-23_Time-23-06-28.428608.log</span></span>
<span id="cb418-7"><a href="monocle3-trajectories.html#cb418-7" tabindex="-1"></a>p_lymphoid_solo[[<span class="dv">1</span>]]</span>
<span id="cb418-8"><a href="monocle3-trajectories.html#cb418-8" tabindex="-1"></a><span class="do">## Warning: Removed 7715 rows containing non-finite outside the scale range</span></span>
<span id="cb418-9"><a href="monocle3-trajectories.html#cb418-9" tabindex="-1"></a><span class="do">## (`stat_summary_hex()`).</span></span></code></pre></div>
<p><img src="ArchR_files/figure-html/unnamed-chunk-412-1.svg" width="672" /></p>
<p>Alternatively, we could create a single trajectory encompassing both the myeloid and lymphoid branches with a single call to <code>getMonocleTrajectories()</code>, providing all of the relevant groups to <code>useGroups</code>. For certain applications, this may be a useful implementation so it is worth thinking about the best way to do this for your particular biological question.</p>
<p>First, we create a new <code>monocleCDS</code> object using <code>getMonocleTrajectories()</code>, but passing the cell groups corresponding to both the lymphoid and myeloid cell types to <code>useGroups</code>.</p>
<div class="sourceCode" id="cb419"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb419-1"><a href="monocle3-trajectories.html#cb419-1" tabindex="-1"></a>monocle_progenitor <span class="ot">&lt;-</span> <span class="fu">getMonocleTrajectories</span>(</span>
<span id="cb419-2"><a href="monocle3-trajectories.html#cb419-2" tabindex="-1"></a>    <span class="at">ArchRProj =</span> projHeme5, </span>
<span id="cb419-3"><a href="monocle3-trajectories.html#cb419-3" tabindex="-1"></a>    <span class="at">name =</span> <span class="st">&quot;Monocle_Progenitor&quot;</span>,</span>
<span id="cb419-4"><a href="monocle3-trajectories.html#cb419-4" tabindex="-1"></a>    <span class="at">useGroups =</span> <span class="fu">c</span>(<span class="st">&quot;Progenitor&quot;</span>, <span class="st">&quot;GMP&quot;</span>,<span class="st">&quot;Mono&quot;</span>,<span class="st">&quot;CLP&quot;</span>,<span class="st">&quot;PreB&quot;</span>,<span class="st">&quot;B&quot;</span>),</span>
<span id="cb419-5"><a href="monocle3-trajectories.html#cb419-5" tabindex="-1"></a>    <span class="at">principalGroup =</span> <span class="st">&quot;Progenitor&quot;</span>,</span>
<span id="cb419-6"><a href="monocle3-trajectories.html#cb419-6" tabindex="-1"></a>    <span class="at">groupBy =</span> <span class="st">&quot;Clusters2&quot;</span>,</span>
<span id="cb419-7"><a href="monocle3-trajectories.html#cb419-7" tabindex="-1"></a>    <span class="at">embedding =</span> <span class="st">&quot;UMAP&quot;</span>,</span>
<span id="cb419-8"><a href="monocle3-trajectories.html#cb419-8" tabindex="-1"></a>    <span class="at">clusterParams =</span> <span class="fu">list</span>(<span class="at">k =</span> <span class="dv">50</span>),</span>
<span id="cb419-9"><a href="monocle3-trajectories.html#cb419-9" tabindex="-1"></a>    <span class="at">seed =</span> <span class="dv">1</span></span>
<span id="cb419-10"><a href="monocle3-trajectories.html#cb419-10" tabindex="-1"></a>)</span>
<span id="cb419-11"><a href="monocle3-trajectories.html#cb419-11" tabindex="-1"></a><span class="do">## Running Monocole3 Trajectory Infrastructure!</span></span>
<span id="cb419-12"><a href="monocle3-trajectories.html#cb419-12" tabindex="-1"></a><span class="do">## Adding Embedding</span></span>
<span id="cb419-13"><a href="monocle3-trajectories.html#cb419-13" tabindex="-1"></a><span class="do">## Clustering Embedding</span></span>
<span id="cb419-14"><a href="monocle3-trajectories.html#cb419-14" tabindex="-1"></a><span class="do">## Learning Graphs</span></span>
<span id="cb419-15"><a href="monocle3-trajectories.html#cb419-15" tabindex="-1"></a><span class="do">##   |                                                                              |                                                                      |   0%  |                                                                              |======================================================================| 100%</span></span>
<span id="cb419-16"><a href="monocle3-trajectories.html#cb419-16" tabindex="-1"></a><span class="do">## Getting Principal Node</span></span>
<span id="cb419-17"><a href="monocle3-trajectories.html#cb419-17" tabindex="-1"></a><span class="do">## Ordering Cells</span></span>
<span id="cb419-18"><a href="monocle3-trajectories.html#cb419-18" tabindex="-1"></a><span class="do">## Plotting Results - /workspace/ArchR/ArchR_Website_Testing/bookdown/Save-ProjHeme4/Monocole3/Plot-Results-Monocle_Progenitor.pdf</span></span></code></pre></div>
<p>As mentioned above, the resulting Monocle trajectory contains multiple branches so we need to add each branch individually as its own trajectory. First, we will add the myeloid trajectory, by specifying the myeloid cell groupings to <code>useGroups</code>.</p>
<div class="sourceCode" id="cb420"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb420-1"><a href="monocle3-trajectories.html#cb420-1" tabindex="-1"></a>projHeme5 <span class="ot">&lt;-</span> <span class="fu">addMonocleTrajectory</span>(</span>
<span id="cb420-2"><a href="monocle3-trajectories.html#cb420-2" tabindex="-1"></a>    <span class="at">ArchRProj =</span> projHeme5,</span>
<span id="cb420-3"><a href="monocle3-trajectories.html#cb420-3" tabindex="-1"></a>    <span class="at">name =</span> <span class="st">&quot;Monocle_Myeloid&quot;</span>,</span>
<span id="cb420-4"><a href="monocle3-trajectories.html#cb420-4" tabindex="-1"></a>    <span class="at">useGroups =</span> <span class="fu">c</span>(<span class="st">&quot;Progenitor&quot;</span>,<span class="st">&quot;GMP&quot;</span>,<span class="st">&quot;Mono&quot;</span>),</span>
<span id="cb420-5"><a href="monocle3-trajectories.html#cb420-5" tabindex="-1"></a>    <span class="at">groupBy =</span> <span class="st">&quot;Clusters2&quot;</span>,</span>
<span id="cb420-6"><a href="monocle3-trajectories.html#cb420-6" tabindex="-1"></a>    <span class="at">monocleCDS =</span> monocle_progenitor,</span>
<span id="cb420-7"><a href="monocle3-trajectories.html#cb420-7" tabindex="-1"></a>    <span class="at">force =</span> <span class="cn">TRUE</span></span>
<span id="cb420-8"><a href="monocle3-trajectories.html#cb420-8" tabindex="-1"></a>)</span></code></pre></div>
<p>Plotting this trajectory is the same as above.</p>
<div class="sourceCode" id="cb421"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb421-1"><a href="monocle3-trajectories.html#cb421-1" tabindex="-1"></a>p_myeloid <span class="ot">&lt;-</span> <span class="fu">plotTrajectory</span>(projHeme5, <span class="at">trajectory =</span> <span class="st">&quot;Monocle_Myeloid&quot;</span>, <span class="at">colorBy =</span> <span class="st">&quot;cellColData&quot;</span>, <span class="at">name =</span> <span class="st">&quot;Monocle_Myeloid&quot;</span>, <span class="at">addArrow =</span> <span class="cn">FALSE</span>)</span>
<span id="cb421-2"><a href="monocle3-trajectories.html#cb421-2" tabindex="-1"></a><span class="do">## ArchR logging to : ArchRLogs/ArchR-plotTrajectory-1a12e8201d2-Date-2025-01-23_Time-23-07-02.007601.log</span></span>
<span id="cb421-3"><a href="monocle3-trajectories.html#cb421-3" tabindex="-1"></a><span class="do">## If there is an issue, please report to github with logFile!</span></span>
<span id="cb421-4"><a href="monocle3-trajectories.html#cb421-4" tabindex="-1"></a><span class="do">## Plotting</span></span>
<span id="cb421-5"><a href="monocle3-trajectories.html#cb421-5" tabindex="-1"></a><span class="do">## Plotting Trajectory</span></span>
<span id="cb421-6"><a href="monocle3-trajectories.html#cb421-6" tabindex="-1"></a><span class="do">## ArchR logging successful to : ArchRLogs/ArchR-plotTrajectory-1a12e8201d2-Date-2025-01-23_Time-23-07-02.007601.log</span></span>
<span id="cb421-7"><a href="monocle3-trajectories.html#cb421-7" tabindex="-1"></a>p_myeloid[[<span class="dv">1</span>]]</span>
<span id="cb421-8"><a href="monocle3-trajectories.html#cb421-8" tabindex="-1"></a><span class="do">## Warning: Removed 4960 rows containing non-finite outside the scale range</span></span>
<span id="cb421-9"><a href="monocle3-trajectories.html#cb421-9" tabindex="-1"></a><span class="do">## (`stat_summary_hex()`).</span></span></code></pre></div>
<p><img src="ArchR_files/figure-html/unnamed-chunk-415-1.svg" width="672" /></p>
<p>Similarly, we can add the lymphoid trajectory.</p>
<div class="sourceCode" id="cb422"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb422-1"><a href="monocle3-trajectories.html#cb422-1" tabindex="-1"></a>projHeme5 <span class="ot">&lt;-</span> <span class="fu">addMonocleTrajectory</span>(</span>
<span id="cb422-2"><a href="monocle3-trajectories.html#cb422-2" tabindex="-1"></a>    <span class="at">ArchRProj =</span> projHeme5,</span>
<span id="cb422-3"><a href="monocle3-trajectories.html#cb422-3" tabindex="-1"></a>    <span class="at">name =</span> <span class="st">&quot;Monocle_Lymphoid&quot;</span>,</span>
<span id="cb422-4"><a href="monocle3-trajectories.html#cb422-4" tabindex="-1"></a>    <span class="at">useGroups =</span> <span class="fu">c</span>(<span class="st">&quot;Progenitor&quot;</span>,<span class="st">&quot;CLP&quot;</span>,<span class="st">&quot;PreB&quot;</span>,<span class="st">&quot;B&quot;</span>),</span>
<span id="cb422-5"><a href="monocle3-trajectories.html#cb422-5" tabindex="-1"></a>    <span class="at">groupBy =</span> <span class="st">&quot;Clusters2&quot;</span>,</span>
<span id="cb422-6"><a href="monocle3-trajectories.html#cb422-6" tabindex="-1"></a>    <span class="at">monocleCDS =</span> monocle_progenitor,</span>
<span id="cb422-7"><a href="monocle3-trajectories.html#cb422-7" tabindex="-1"></a>    <span class="at">force =</span> <span class="cn">TRUE</span></span>
<span id="cb422-8"><a href="monocle3-trajectories.html#cb422-8" tabindex="-1"></a>)</span></code></pre></div>
<p>And then plot that lymphoid trajectory.</p>
<div class="sourceCode" id="cb423"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb423-1"><a href="monocle3-trajectories.html#cb423-1" tabindex="-1"></a>p_lymphoid <span class="ot">&lt;-</span> <span class="fu">plotTrajectory</span>(projHeme5, <span class="at">trajectory =</span> <span class="st">&quot;Monocle_Lymphoid&quot;</span>, <span class="at">colorBy =</span> <span class="st">&quot;cellColData&quot;</span>, <span class="at">name =</span> <span class="st">&quot;Monocle_Lymphoid&quot;</span>, <span class="at">addArrow =</span> <span class="cn">FALSE</span>)</span>
<span id="cb423-2"><a href="monocle3-trajectories.html#cb423-2" tabindex="-1"></a><span class="do">## ArchR logging to : ArchRLogs/ArchR-plotTrajectory-1a15e3f703-Date-2025-01-23_Time-23-07-04.417621.log</span></span>
<span id="cb423-3"><a href="monocle3-trajectories.html#cb423-3" tabindex="-1"></a><span class="do">## If there is an issue, please report to github with logFile!</span></span>
<span id="cb423-4"><a href="monocle3-trajectories.html#cb423-4" tabindex="-1"></a><span class="do">## Plotting</span></span>
<span id="cb423-5"><a href="monocle3-trajectories.html#cb423-5" tabindex="-1"></a><span class="do">## Plotting Trajectory</span></span>
<span id="cb423-6"><a href="monocle3-trajectories.html#cb423-6" tabindex="-1"></a><span class="do">## ArchR logging successful to : ArchRLogs/ArchR-plotTrajectory-1a15e3f703-Date-2025-01-23_Time-23-07-04.417621.log</span></span>
<span id="cb423-7"><a href="monocle3-trajectories.html#cb423-7" tabindex="-1"></a>p_lymphoid[[<span class="dv">1</span>]]</span>
<span id="cb423-8"><a href="monocle3-trajectories.html#cb423-8" tabindex="-1"></a><span class="do">## Warning: Removed 7715 rows containing non-finite outside the scale range</span></span>
<span id="cb423-9"><a href="monocle3-trajectories.html#cb423-9" tabindex="-1"></a><span class="do">## (`stat_summary_hex()`).</span></span></code></pre></div>
<p><img src="ArchR_files/figure-html/unnamed-chunk-417-1.svg" width="672" /></p>
<p>If you look closely, you’ll notice slight differences between this lymphoid trajectory identified in isolation compared to the lymphoid branch of the larger joint monocle trajectory calculated at the beginning of this section. In particular, the “Progenitor” cells have a different pattern.</p>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="lymphoid-trajectory---b-cell-cifferentiation.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="slingshot-trajectories.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": null
},
"fontsettings": null,
"edit": null,
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "none"
},
"info": false
});
});
</script>

</body>

</html>
