<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>15.1 Motif Deviations | ArchR: Robust and scaleable analysis of single-cell chromatin accessibility data.</title>
  <meta name="description" content="A guide to ArchR" />
  <meta name="generator" content="bookdown 0.41 and GitBook 2.6.7" />

  <meta property="og:title" content="15.1 Motif Deviations | ArchR: Robust and scaleable analysis of single-cell chromatin accessibility data." />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="A guide to ArchR" />
  <meta name="github-repo" content="GreenleafLab/ArchR" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="15.1 Motif Deviations | ArchR: Robust and scaleable analysis of single-cell chromatin accessibility data." />
  
  <meta name="twitter:description" content="A guide to ArchR" />
  

<meta name="author" content="Jeffrey Granja and Ryan Corces" />


<meta name="date" content="2025-02-06" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="images/favicon.ico" type="image/x-icon" />
<link rel="prev" href="chromvar-deviatons-enrichment-with-archr.html"/>
<link rel="next" href="archr-and-custom-deviations.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<link href="libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="../index.html">Back to ArchR Home</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><img src="images/ArchR_Logo_Integrated.png" alt="" width="200"></a></li>
<li class="chapter" data-level="1" data-path="debugging-and-troubleshooting-in-archr.html"><a href="debugging-and-troubleshooting-in-archr.html"><i class="fa fa-check"></i><b>1</b> Debugging and Troubleshooting in ArchR</a></li>
<li class="chapter" data-level="2" data-path="manage-archrs-dependencies.html"><a href="manage-archrs-dependencies.html"><i class="fa fa-check"></i><b>2</b> Manage ArchR’s Dependencies</a>
<ul>
<li class="chapter" data-level="2.1" data-path="using-renv-to-manage-dependencies.html"><a href="using-renv-to-manage-dependencies.html"><i class="fa fa-check"></i><b>2.1</b> Using renv to manage dependencies</a></li>
<li class="chapter" data-level="2.2" data-path="other-environment-managers.html"><a href="other-environment-managers.html"><i class="fa fa-check"></i><b>2.2</b> Other environment managers</a></li>
<li class="chapter" data-level="2.3" data-path="using-docker-for-pre-compiled-environment-setup.html"><a href="using-docker-for-pre-compiled-environment-setup.html"><i class="fa fa-check"></i><b>2.3</b> Using Docker for pre-compiled environment setup</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="getting-started-with-archr.html"><a href="getting-started-with-archr.html"><i class="fa fa-check"></i><b>3</b> Getting Started with ArchR</a>
<ul>
<li class="chapter" data-level="3.1" data-path="a-brief-primer-on-atac-seq-terminology.html"><a href="a-brief-primer-on-atac-seq-terminology.html"><i class="fa fa-check"></i><b>3.1</b> A Brief Primer on ATAC-seq Terminology</a></li>
<li class="chapter" data-level="3.2" data-path="why-use-archr.html"><a href="why-use-archr.html"><i class="fa fa-check"></i><b>3.2</b> Why use ArchR?</a></li>
<li class="chapter" data-level="3.3" data-path="what-is-an-arrow-file-archrproject.html"><a href="what-is-an-arrow-file-archrproject.html"><i class="fa fa-check"></i><b>3.3</b> What is an Arrow file / <code>ArchRProject</code>?</a></li>
<li class="chapter" data-level="3.4" data-path="input-file-types-in-archr.html"><a href="input-file-types-in-archr.html"><i class="fa fa-check"></i><b>3.4</b> Input File Types in ArchR</a></li>
<li class="chapter" data-level="3.5" data-path="getting-set-up.html"><a href="getting-set-up.html"><i class="fa fa-check"></i><b>3.5</b> Getting Set Up</a></li>
<li class="chapter" data-level="3.6" data-path="creating-arrow-files.html"><a href="creating-arrow-files.html"><i class="fa fa-check"></i><b>3.6</b> Creating Arrow Files</a></li>
<li class="chapter" data-level="3.7" data-path="per-cell-quality-control.html"><a href="per-cell-quality-control.html"><i class="fa fa-check"></i><b>3.7</b> Per-cell Quality Control</a></li>
<li class="chapter" data-level="3.8" data-path="using-bam-files-for-arrow-file-creation.html"><a href="using-bam-files-for-arrow-file-creation.html"><i class="fa fa-check"></i><b>3.8</b> Using BAM files for Arrow File creation</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="doublet-inference-with-archr.html"><a href="doublet-inference-with-archr.html"><i class="fa fa-check"></i><b>4</b> Doublet Inference with ArchR</a>
<ul>
<li class="chapter" data-level="4.1" data-path="how-does-doublet-identification-work-in-archr.html"><a href="how-does-doublet-identification-work-in-archr.html"><i class="fa fa-check"></i><b>4.1</b> How does doublet identification work in ArchR?</a></li>
<li class="chapter" data-level="4.2" data-path="inferring-scatac-seq-doublets-with-archr.html"><a href="inferring-scatac-seq-doublets-with-archr.html"><i class="fa fa-check"></i><b>4.2</b> Inferring scATAC-seq Doublets with ArchR</a></li>
<li class="chapter" data-level="4.3" data-path="using-demuxlet-with-archr.html"><a href="using-demuxlet-with-archr.html"><i class="fa fa-check"></i><b>4.3</b> Using <em>demuxlet</em> with ArchR</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="creating-an-archrproject.html"><a href="creating-an-archrproject.html"><i class="fa fa-check"></i><b>5</b> Creating an ArchRProject</a>
<ul>
<li class="chapter" data-level="5.1" data-path="creating-an-archrproject-1.html"><a href="creating-an-archrproject-1.html"><i class="fa fa-check"></i><b>5.1</b> Creating An ArchRProject</a></li>
<li class="chapter" data-level="5.2" data-path="manipulating-an-archrproject.html"><a href="manipulating-an-archrproject.html"><i class="fa fa-check"></i><b>5.2</b> Manipulating An ArchRProject</a></li>
<li class="chapter" data-level="5.3" data-path="plotting-sample-statistics-from-an-archrproject.html"><a href="plotting-sample-statistics-from-an-archrproject.html"><i class="fa fa-check"></i><b>5.3</b> Plotting Sample Statistics from an ArchRProject</a></li>
<li class="chapter" data-level="5.4" data-path="plotting-sample-fragment-size-distribution-and-tss-enrichment-profiles..html"><a href="plotting-sample-fragment-size-distribution-and-tss-enrichment-profiles..html"><i class="fa fa-check"></i><b>5.4</b> Plotting Sample Fragment Size Distribution and TSS Enrichment Profiles.</a></li>
<li class="chapter" data-level="5.5" data-path="saving-and-loading-an-archrproject.html"><a href="saving-and-loading-an-archrproject.html"><i class="fa fa-check"></i><b>5.5</b> Saving and Loading an <code>ArchRProject</code></a></li>
<li class="chapter" data-level="5.6" data-path="filtering-doublets-from-an-archrproject.html"><a href="filtering-doublets-from-an-archrproject.html"><i class="fa fa-check"></i><b>5.6</b> Filtering Doublets from an ArchRProject</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="dimensionality-reduction-with-archr.html"><a href="dimensionality-reduction-with-archr.html"><i class="fa fa-check"></i><b>6</b> Dimensionality Reduction with ArchR</a>
<ul>
<li class="chapter" data-level="6.1" data-path="archrs-lsi-implementation.html"><a href="archrs-lsi-implementation.html"><i class="fa fa-check"></i><b>6.1</b> ArchR’s LSI Implementation</a></li>
<li class="chapter" data-level="6.2" data-path="iterative-latent-semantic-indexing-lsi.html"><a href="iterative-latent-semantic-indexing-lsi.html"><i class="fa fa-check"></i><b>6.2</b> Iterative Latent Semantic Indexing (LSI)</a></li>
<li class="chapter" data-level="6.3" data-path="estimated-lsi.html"><a href="estimated-lsi.html"><i class="fa fa-check"></i><b>6.3</b> Estimated LSI</a></li>
<li class="chapter" data-level="6.4" data-path="batch-effect-correction-wtih-harmony.html"><a href="batch-effect-correction-wtih-harmony.html"><i class="fa fa-check"></i><b>6.4</b> Batch Effect Correction wtih Harmony</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="clustering-with-archr.html"><a href="clustering-with-archr.html"><i class="fa fa-check"></i><b>7</b> Clustering with ArchR</a>
<ul>
<li class="chapter" data-level="7.1" data-path="clustering-using-seurats-findclusters-function.html"><a href="clustering-using-seurats-findclusters-function.html"><i class="fa fa-check"></i><b>7.1</b> Clustering using Seurat’s <code>FindClusters()</code> function</a></li>
<li class="chapter" data-level="7.2" data-path="clustering-using-scran.html"><a href="clustering-using-scran.html"><i class="fa fa-check"></i><b>7.2</b> Clustering using <code>scran</code></a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="single-cell-embeddings.html"><a href="single-cell-embeddings.html"><i class="fa fa-check"></i><b>8</b> Single-cell Embeddings</a>
<ul>
<li class="chapter" data-level="8.1" data-path="uniform-manifold-approximation-and-projection-umap.html"><a href="uniform-manifold-approximation-and-projection-umap.html"><i class="fa fa-check"></i><b>8.1</b> Uniform Manifold Approximation and Projection (UMAP)</a></li>
<li class="chapter" data-level="8.2" data-path="t-stocastic-neighbor-embedding-t-sne.html"><a href="t-stocastic-neighbor-embedding-t-sne.html"><i class="fa fa-check"></i><b>8.2</b> t-Stocastic Neighbor Embedding (t-SNE)</a></li>
<li class="chapter" data-level="8.3" data-path="dimensionality-reduction-after-harmony.html"><a href="dimensionality-reduction-after-harmony.html"><i class="fa fa-check"></i><b>8.3</b> Dimensionality Reduction After Harmony</a></li>
<li class="chapter" data-level="8.4" data-path="highlighting-specific-cells-on-an-embedding.html"><a href="highlighting-specific-cells-on-an-embedding.html"><i class="fa fa-check"></i><b>8.4</b> Highlighting specific cells on an embedding</a></li>
<li class="chapter" data-level="8.5" data-path="importing-an-embedding-from-external-software.html"><a href="importing-an-embedding-from-external-software.html"><i class="fa fa-check"></i><b>8.5</b> Importing an embedding from external software</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="gene-scores-and-marker-genes-with-archr.html"><a href="gene-scores-and-marker-genes-with-archr.html"><i class="fa fa-check"></i><b>9</b> Gene Scores and Marker Genes with ArchR</a>
<ul>
<li class="chapter" data-level="9.1" data-path="calculating-gene-scores-in-archr.html"><a href="calculating-gene-scores-in-archr.html"><i class="fa fa-check"></i><b>9.1</b> Calculating Gene Scores in ArchR</a></li>
<li class="chapter" data-level="9.2" data-path="identification-of-marker-features.html"><a href="identification-of-marker-features.html"><i class="fa fa-check"></i><b>9.2</b> Identification of Marker Features</a></li>
<li class="chapter" data-level="9.3" data-path="identifying-marker-genes.html"><a href="identifying-marker-genes.html"><i class="fa fa-check"></i><b>9.3</b> Identifying Marker Genes</a></li>
<li class="chapter" data-level="9.4" data-path="visualizing-marker-genes-on-an-embedding.html"><a href="visualizing-marker-genes-on-an-embedding.html"><i class="fa fa-check"></i><b>9.4</b> Visualizing Marker Genes on an Embedding</a></li>
<li class="chapter" data-level="9.5" data-path="marker-genes-imputation-with-magic.html"><a href="marker-genes-imputation-with-magic.html"><i class="fa fa-check"></i><b>9.5</b> Marker Genes Imputation with MAGIC</a></li>
<li class="chapter" data-level="9.6" data-path="module-scores.html"><a href="module-scores.html"><i class="fa fa-check"></i><b>9.6</b> Module Scores</a></li>
<li class="chapter" data-level="9.7" data-path="track-plotting-with-archrbrowser.html"><a href="track-plotting-with-archrbrowser.html"><i class="fa fa-check"></i><b>9.7</b> Track Plotting with ArchRBrowser</a></li>
<li class="chapter" data-level="9.8" data-path="launching-the-interactive-archrbrowser.html"><a href="launching-the-interactive-archrbrowser.html"><i class="fa fa-check"></i><b>9.8</b> Launching the Interactive ArchRBrowser</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="defining-cluster-identity-with-scrna-seq.html"><a href="defining-cluster-identity-with-scrna-seq.html"><i class="fa fa-check"></i><b>10</b> Defining Cluster Identity with scRNA-seq</a>
<ul>
<li class="chapter" data-level="10.1" data-path="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html"><i class="fa fa-check"></i><b>10.1</b> Cross-platform linkage of scATAC-seq cells with scRNA-seq cells</a></li>
<li class="chapter" data-level="10.2" data-path="adding-pseudo-scrna-seq-profiles-for-each-scatac-seq-cell.html"><a href="adding-pseudo-scrna-seq-profiles-for-each-scatac-seq-cell.html"><i class="fa fa-check"></i><b>10.2</b> Adding Pseudo-scRNA-seq profiles for each scATAC-seq cell</a></li>
<li class="chapter" data-level="10.3" data-path="labeling-scatac-seq-clusters-with-scrna-seq-information.html"><a href="labeling-scatac-seq-clusters-with-scrna-seq-information.html"><i class="fa fa-check"></i><b>10.3</b> Labeling scATAC-seq clusters with scRNA-seq information</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="pseudo-bulk-replicates-in-archr.html"><a href="pseudo-bulk-replicates-in-archr.html"><i class="fa fa-check"></i><b>11</b> Pseudo-bulk Replicates in ArchR</a>
<ul>
<li class="chapter" data-level="11.1" data-path="how-does-archr-make-pseudo-bulk-replicates.html"><a href="how-does-archr-make-pseudo-bulk-replicates.html"><i class="fa fa-check"></i><b>11.1</b> How Does ArchR Make Pseudo-bulk Replicates?</a></li>
<li class="chapter" data-level="11.2" data-path="making-pseudo-bulk-replicates.html"><a href="making-pseudo-bulk-replicates.html"><i class="fa fa-check"></i><b>11.2</b> Making Pseudo-bulk Replicates</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="calling-peaks-with-archr.html"><a href="calling-peaks-with-archr.html"><i class="fa fa-check"></i><b>12</b> Calling Peaks with ArchR</a>
<ul>
<li class="chapter" data-level="12.1" data-path="the-iterative-overlap-peak-merging-procedure.html"><a href="the-iterative-overlap-peak-merging-procedure.html"><i class="fa fa-check"></i><b>12.1</b> The Iterative Overlap Peak Merging Procedure</a></li>
<li class="chapter" data-level="12.2" data-path="calling-peaks-w-macs2.html"><a href="calling-peaks-w-macs2.html"><i class="fa fa-check"></i><b>12.2</b> Calling Peaks w/ Macs2</a></li>
<li class="chapter" data-level="12.3" data-path="calling-peaks-w-tilematrix.html"><a href="calling-peaks-w-tilematrix.html"><i class="fa fa-check"></i><b>12.3</b> Calling Peaks w/ TileMatrix</a></li>
<li class="chapter" data-level="12.4" data-path="adding-a-peak-matrix.html"><a href="adding-a-peak-matrix.html"><i class="fa fa-check"></i><b>12.4</b> Adding a Peak Matrix</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="identifying-marker-peaks.html"><a href="identifying-marker-peaks.html"><i class="fa fa-check"></i><b>13</b> Identifying Marker Peaks</a>
<ul>
<li class="chapter" data-level="13.1" data-path="identifying-marker-peaks-with-archr.html"><a href="identifying-marker-peaks-with-archr.html"><i class="fa fa-check"></i><b>13.1</b> Identifying Marker Peaks with ArchR</a></li>
<li class="chapter" data-level="13.2" data-path="plotting-marker-peaks-in-archr.html"><a href="plotting-marker-peaks-in-archr.html"><i class="fa fa-check"></i><b>13.2</b> Plotting Marker Peaks in ArchR</a></li>
<li class="chapter" data-level="13.3" data-path="pairwise-testing-between-groups.html"><a href="pairwise-testing-between-groups.html"><i class="fa fa-check"></i><b>13.3</b> Pairwise Testing Between Groups</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="motif-and-feature-enrichment-with-archr.html"><a href="motif-and-feature-enrichment-with-archr.html"><i class="fa fa-check"></i><b>14</b> Motif and Feature Enrichment with ArchR</a>
<ul>
<li class="chapter" data-level="14.1" data-path="motif-enrichment.html"><a href="motif-enrichment.html"><i class="fa fa-check"></i><b>14.1</b> Motif Enrichment</a></li>
<li class="chapter" data-level="14.2" data-path="archr-enrichment.html"><a href="archr-enrichment.html"><i class="fa fa-check"></i><b>14.2</b> ArchR Enrichment</a></li>
<li class="chapter" data-level="14.3" data-path="custom-enrichment.html"><a href="custom-enrichment.html"><i class="fa fa-check"></i><b>14.3</b> Custom Enrichment</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="chromvar-deviatons-enrichment-with-archr.html"><a href="chromvar-deviatons-enrichment-with-archr.html"><i class="fa fa-check"></i><b>15</b> ChromVAR Deviatons Enrichment with ArchR</a>
<ul>
<li class="chapter" data-level="15.1" data-path="motif-deviations.html"><a href="motif-deviations.html"><i class="fa fa-check"></i><b>15.1</b> Motif Deviations</a></li>
<li class="chapter" data-level="15.2" data-path="archr-and-custom-deviations.html"><a href="archr-and-custom-deviations.html"><i class="fa fa-check"></i><b>15.2</b> ArchR and Custom Deviations</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="footprinting-with-archr.html"><a href="footprinting-with-archr.html"><i class="fa fa-check"></i><b>16</b> Footprinting with ArchR</a>
<ul>
<li class="chapter" data-level="16.1" data-path="motif-footprinting.html"><a href="motif-footprinting.html"><i class="fa fa-check"></i><b>16.1</b> Motif Footprinting</a></li>
<li class="chapter" data-level="16.2" data-path="normalization-of-footprints-for-tn5-bias.html"><a href="normalization-of-footprints-for-tn5-bias.html"><i class="fa fa-check"></i><b>16.2</b> Normalization of Footprints for Tn5 Bias</a></li>
<li class="chapter" data-level="16.3" data-path="feature-footprinting.html"><a href="feature-footprinting.html"><i class="fa fa-check"></i><b>16.3</b> Feature Footprinting</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="integrative-analysis-with-archr.html"><a href="integrative-analysis-with-archr.html"><i class="fa fa-check"></i><b>17</b> Integrative Analysis with ArchR</a>
<ul>
<li class="chapter" data-level="17.1" data-path="creating-low-overlapping-aggregates-of-cells.html"><a href="creating-low-overlapping-aggregates-of-cells.html"><i class="fa fa-check"></i><b>17.1</b> Creating Low-Overlapping Aggregates of Cells</a></li>
<li class="chapter" data-level="17.2" data-path="co-accessibility-with-archr.html"><a href="co-accessibility-with-archr.html"><i class="fa fa-check"></i><b>17.2</b> Co-accessibility with ArchR</a></li>
<li class="chapter" data-level="17.3" data-path="peak2genelinkage-with-archr.html"><a href="peak2genelinkage-with-archr.html"><i class="fa fa-check"></i><b>17.3</b> Peak2GeneLinkage with ArchR</a></li>
<li class="chapter" data-level="17.4" data-path="identification-of-positive-tf-regulators.html"><a href="identification-of-positive-tf-regulators.html"><i class="fa fa-check"></i><b>17.4</b> Identification of Positive TF-Regulators</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="trajectory-analysis-with-archr.html"><a href="trajectory-analysis-with-archr.html"><i class="fa fa-check"></i><b>18</b> Trajectory Analysis with ArchR</a>
<ul>
<li class="chapter" data-level="18.1" data-path="myeloid-trajectory---monocyte-differentiation.html"><a href="myeloid-trajectory---monocyte-differentiation.html"><i class="fa fa-check"></i><b>18.1</b> Myeloid Trajectory - Monocyte Differentiation</a></li>
<li class="chapter" data-level="18.2" data-path="lymphoid-trajectory---b-cell-differentiation.html"><a href="lymphoid-trajectory---b-cell-differentiation.html"><i class="fa fa-check"></i><b>18.2</b> Lymphoid Trajectory - B Cell Differentiation</a></li>
<li class="chapter" data-level="18.3" data-path="monocle3-trajectories.html"><a href="monocle3-trajectories.html"><i class="fa fa-check"></i><b>18.3</b> Monocle3 Trajectories</a></li>
<li class="chapter" data-level="18.4" data-path="slingshot-trajectories.html"><a href="slingshot-trajectories.html"><i class="fa fa-check"></i><b>18.4</b> Slingshot Trajectories</a></li>
<li class="chapter" data-level="18.5" data-path="comparing-the-different-trajectories.html"><a href="comparing-the-different-trajectories.html"><i class="fa fa-check"></i><b>18.5</b> Comparing the different trajectories</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="handling-archr-output.html"><a href="handling-archr-output.html"><i class="fa fa-check"></i><b>19</b> Handling ArchR Output</a>
<ul>
<li class="chapter" data-level="19.1" data-path="exporting-fragment-level-data.html"><a href="exporting-fragment-level-data.html"><i class="fa fa-check"></i><b>19.1</b> Exporting fragment-level data</a></li>
<li class="chapter" data-level="19.2" data-path="exporting-matrix-level-data.html"><a href="exporting-matrix-level-data.html"><i class="fa fa-check"></i><b>19.2</b> Exporting matrix-level data</a></li>
<li class="chapter" data-level="19.3" data-path="exporting-pseudo-bulked-data-to-a-summarizedexperiment.html"><a href="exporting-pseudo-bulked-data-to-a-summarizedexperiment.html"><i class="fa fa-check"></i><b>19.3</b> Exporting pseudo-bulked data to a <code>SummarizedExperiment</code></a></li>
<li class="chapter" data-level="19.4" data-path="exporting-pseudo-bulked-data-to-a-bigwig-file.html"><a href="exporting-pseudo-bulked-data-to-a-bigwig-file.html"><i class="fa fa-check"></i><b>19.4</b> Exporting pseudo-bulked data to a bigWig file</a></li>
</ul></li>
<li class="chapter" data-level="20" data-path="integration-with-bulk-atac-seq.html"><a href="integration-with-bulk-atac-seq.html"><i class="fa fa-check"></i><b>20</b> Integration with bulk ATAC-seq</a>
<ul>
<li class="chapter" data-level="20.1" data-path="projecting-bulk-atac-seq-data.html"><a href="projecting-bulk-atac-seq-data.html"><i class="fa fa-check"></i><b>20.1</b> Projecting bulk ATAC-seq data</a></li>
</ul></li>
<li class="chapter" data-level="21" data-path="multiomic-data-analysis-in-archr.html"><a href="multiomic-data-analysis-in-archr.html"><i class="fa fa-check"></i><b>21</b> Multiomic data analysis in ArchR</a>
<ul>
<li class="chapter" data-level="21.1" data-path="importing-data-and-setting-up-a-multiome-project.html"><a href="importing-data-and-setting-up-a-multiome-project.html"><i class="fa fa-check"></i><b>21.1</b> Importing data and setting up a Multiome project</a></li>
<li class="chapter" data-level="21.2" data-path="analysis-of-multiome-data-in-archr.html"><a href="analysis-of-multiome-data-in-archr.html"><i class="fa fa-check"></i><b>21.2</b> Analysis of multiome data in ArchR</a></li>
</ul></li>
<li class="chapter" data-level="22" data-path="plot-aesthetics-in-archr.html"><a href="plot-aesthetics-in-archr.html"><i class="fa fa-check"></i><b>22</b> Plot aesthetics in ArchR</a>
<ul>
<li class="chapter" data-level="22.1" data-path="using-and-manipulating-palettes-in-archr.html"><a href="using-and-manipulating-palettes-in-archr.html"><i class="fa fa-check"></i><b>22.1</b> Using and manipulating palettes in ArchR</a></li>
</ul></li>
<li class="chapter" data-level="23" data-path="testing-your-archr-installation.html"><a href="testing-your-archr-installation.html"><i class="fa fa-check"></i><b>23</b> Testing Your ArchR Installation</a></li>
<li class="chapter" data-level="24" data-path="session-information.html"><a href="session-information.html"><i class="fa fa-check"></i><b>24</b> Session Information</a></li>
<li class="divider"></li>
<li><a href="https://bookdown.org" target="_blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">ArchR: Robust and scaleable analysis of single-cell chromatin accessibility data.</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="motif-deviations" class="section level2 hasAnchor" number="15.1">
<h2><span class="header-section-number">15.1</span> Motif Deviations<a href="motif-deviations.html#motif-deviations" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>First, lets make sure we have added motif annotations to our <code>ArchRProject</code>.</p>
<div class="sourceCode" id="cb281"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb281-1"><a href="motif-deviations.html#cb281-1" tabindex="-1"></a><span class="cf">if</span>(<span class="st">&quot;Motif&quot;</span> <span class="sc">%ni%</span> <span class="fu">names</span>(projHeme5<span class="sc">@</span>peakAnnotation)){</span>
<span id="cb281-2"><a href="motif-deviations.html#cb281-2" tabindex="-1"></a>    projHeme5 <span class="ot">&lt;-</span> <span class="fu">addMotifAnnotations</span>(<span class="at">ArchRProj =</span> projHeme5, <span class="at">motifSet =</span> <span class="st">&quot;cisbp&quot;</span>, <span class="at">name =</span> <span class="st">&quot;Motif&quot;</span>)</span>
<span id="cb281-3"><a href="motif-deviations.html#cb281-3" tabindex="-1"></a>}</span></code></pre></div>
<p>We also need to add a set of background peaks which are used in computing deviations. Background peaks are chosen using the <code>chromVAR::getBackgroundPeaks()</code> function which samples peaks based on similarity in GC-content and number of fragments across all samples using the Mahalanobis distance.</p>
<div class="sourceCode" id="cb282"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb282-1"><a href="motif-deviations.html#cb282-1" tabindex="-1"></a>projHeme5 <span class="ot">&lt;-</span> <span class="fu">addBgdPeaks</span>(projHeme5)</span>
<span id="cb282-2"><a href="motif-deviations.html#cb282-2" tabindex="-1"></a><span class="do">## Identifying Background Peaks!</span></span></code></pre></div>
<p>We are now ready to compute per-cell deviations accross all of our motif annotations using the <code>addDeviationsMatrix()</code> function. This function has an optional parameter called <code>matrixName</code> that allows us to define the name of the deviations matrix that will be stored in the Arrow files. If we do not provide a value to this parameter, as in the example below, this function creates a matrix name by adding the word “Matrix” to the name of the <code>peakAnnotation</code>. The example below creates a deviations matrix in each of our Arrow files called “MotifMatrix”.</p>
<div class="sourceCode" id="cb283"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb283-1"><a href="motif-deviations.html#cb283-1" tabindex="-1"></a>projHeme5 <span class="ot">&lt;-</span> <span class="fu">addDeviationsMatrix</span>(</span>
<span id="cb283-2"><a href="motif-deviations.html#cb283-2" tabindex="-1"></a>  <span class="at">ArchRProj =</span> projHeme5, </span>
<span id="cb283-3"><a href="motif-deviations.html#cb283-3" tabindex="-1"></a>  <span class="at">peakAnnotation =</span> <span class="st">&quot;Motif&quot;</span>,</span>
<span id="cb283-4"><a href="motif-deviations.html#cb283-4" tabindex="-1"></a>  <span class="at">force =</span> <span class="cn">TRUE</span></span>
<span id="cb283-5"><a href="motif-deviations.html#cb283-5" tabindex="-1"></a>)</span>
<span id="cb283-6"><a href="motif-deviations.html#cb283-6" tabindex="-1"></a><span class="do">## Using Previous Background Peaks!</span></span>
<span id="cb283-7"><a href="motif-deviations.html#cb283-7" tabindex="-1"></a><span class="do">## ArchR logging to : ArchRLogs/ArchR-addDeviationsMatrix-945abde0c-Date-2025-02-06_Time-02-14-59.781655.log</span></span>
<span id="cb283-8"><a href="motif-deviations.html#cb283-8" tabindex="-1"></a><span class="do">## If there is an issue, please report to github with logFile!</span></span>
<span id="cb283-9"><a href="motif-deviations.html#cb283-9" tabindex="-1"></a><span class="do">## 2025-02-06 02:15:07.011044 : Batch Execution w/ safelapply!, 0 mins elapsed.</span></span>
<span id="cb283-10"><a href="motif-deviations.html#cb283-10" tabindex="-1"></a><span class="do">## ###########</span></span>
<span id="cb283-11"><a href="motif-deviations.html#cb283-11" tabindex="-1"></a><span class="do">## 2025-02-06 02:20:56.583871 : Completed Computing Deviations!, 5.947 mins elapsed.</span></span>
<span id="cb283-12"><a href="motif-deviations.html#cb283-12" tabindex="-1"></a><span class="do">## ###########</span></span>
<span id="cb283-13"><a href="motif-deviations.html#cb283-13" tabindex="-1"></a><span class="do">## ArchR logging successful to : ArchRLogs/ArchR-addDeviationsMatrix-945abde0c-Date-2025-02-06_Time-02-14-59.781655.log</span></span></code></pre></div>
<p>To access these deviations, we use the <code>getVarDeviations()</code> function. If we want this function to return a <code>ggplot</code> object, we set <code>plot = TRUE</code> otherwise, this function would return the <code>DataFrame</code> object. The <code>head</code> of that <code>DataFrame</code> object is displayed by default when the function is run.</p>
<div class="sourceCode" id="cb284"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb284-1"><a href="motif-deviations.html#cb284-1" tabindex="-1"></a>plotVarDev <span class="ot">&lt;-</span> <span class="fu">getVarDeviations</span>(projHeme5, <span class="at">name =</span> <span class="st">&quot;MotifMatrix&quot;</span>, <span class="at">plot =</span> <span class="cn">TRUE</span>)</span>
<span id="cb284-2"><a href="motif-deviations.html#cb284-2" tabindex="-1"></a><span class="do">## DataFrame with 6 rows and 6 columns</span></span>
<span id="cb284-3"><a href="motif-deviations.html#cb284-3" tabindex="-1"></a><span class="do">##      seqnames       idx        name combinedVars combinedMeans      rank</span></span>
<span id="cb284-4"><a href="motif-deviations.html#cb284-4" tabindex="-1"></a><span class="do">##         &lt;Rle&gt; &lt;integer&gt; &lt;character&gt;    &lt;numeric&gt;     &lt;numeric&gt; &lt;integer&gt;</span></span>
<span id="cb284-5"><a href="motif-deviations.html#cb284-5" tabindex="-1"></a><span class="do">## f336        z       336    SPIB_336      13.9954    -0.0810304         1</span></span>
<span id="cb284-6"><a href="motif-deviations.html#cb284-6" tabindex="-1"></a><span class="do">## f155        z       155   CEBPA_155      11.9525    -0.1800011         2</span></span>
<span id="cb284-7"><a href="motif-deviations.html#cb284-7" tabindex="-1"></a><span class="do">## f140        z       140   CEBPB_140      10.4502    -0.1199047         3</span></span>
<span id="cb284-8"><a href="motif-deviations.html#cb284-8" tabindex="-1"></a><span class="do">## f384        z       384   GATA3_384      10.3798    -0.0476146         4</span></span>
<span id="cb284-9"><a href="motif-deviations.html#cb284-9" tabindex="-1"></a><span class="do">## f388        z       388   GATA2_388      10.2313    -0.0428557         5</span></span>
<span id="cb284-10"><a href="motif-deviations.html#cb284-10" tabindex="-1"></a><span class="do">## f383        z       383   GATA1_383      10.1460    -0.0430076         6</span></span></code></pre></div>
<p>From the above snapshot of the <code>DataFrame</code>, you can see that the <code>seqnames</code> of the <code>MotifMatrix</code> are not chromosomes. Typically, in matrices like the <code>TileMatrix</code>, <code>PeakMatrix</code>, and <code>GeneScoreMatrix</code>, we store the chromosome information in <code>seqnames</code>. The <code>MotifMatrix</code> does not have any corresponding position information but, instead, stores both the “devations” and “z-scores” from chromVAR into the same matrix using two different seqnames - <code>deviations</code> and <code>z</code>. This becomes important later on if you try to use the <code>MotifMatrix</code> (which is of class <code>Sparse.Assays.Matrix</code>) in functions such as <code>getMarkerFeatures()</code>. In these types of operations, ArchR will expect you to subset <code>MotifMatrix</code> to one of the two <code>seqnames</code> (i.e. select either z-scores or deviations to perform calculations).</p>
<p>We can then plot these variable deviations.</p>
<div class="sourceCode" id="cb285"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb285-1"><a href="motif-deviations.html#cb285-1" tabindex="-1"></a>plotVarDev</span>
<span id="cb285-2"><a href="motif-deviations.html#cb285-2" tabindex="-1"></a><span class="do">## Warning: ggrepel: 15 unlabeled data points (too many overlaps). Consider</span></span>
<span id="cb285-3"><a href="motif-deviations.html#cb285-3" tabindex="-1"></a><span class="do">## increasing max.overlaps</span></span></code></pre></div>
<p><img src="ArchR_files/figure-html/unnamed-chunk-280-1.svg" width="672" /></p>
<p>To save an editable vectorized version of this plot, we use the <code>plotPDF()</code> function.</p>
<div class="sourceCode" id="cb286"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb286-1"><a href="motif-deviations.html#cb286-1" tabindex="-1"></a><span class="fu">plotPDF</span>(plotVarDev, <span class="at">name =</span> <span class="st">&quot;Variable-Motif-Deviation-Scores&quot;</span>, <span class="at">width =</span> <span class="dv">5</span>, <span class="at">height =</span> <span class="dv">5</span>, <span class="at">ArchRProj =</span> projHeme5, <span class="at">addDOC =</span> <span class="cn">FALSE</span>)</span>
<span id="cb286-2"><a href="motif-deviations.html#cb286-2" tabindex="-1"></a><span class="do">## Plotting Ggplot!</span></span></code></pre></div>
<p>What if we want to extract a subset of motifs for downstream analysis? We can do this using the <code>getFeatures()</code> function. The <code>paste(motifs, collapse="|")</code> statement below creates a concatenated <code>or</code> statement that enables selection of all of the motifs.</p>
<div class="sourceCode" id="cb287"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb287-1"><a href="motif-deviations.html#cb287-1" tabindex="-1"></a>motifs <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;GATA1&quot;</span>, <span class="st">&quot;CEBPA&quot;</span>, <span class="st">&quot;EBF1&quot;</span>, <span class="st">&quot;IRF4&quot;</span>, <span class="st">&quot;TBX21&quot;</span>, <span class="st">&quot;PAX5&quot;</span>)</span>
<span id="cb287-2"><a href="motif-deviations.html#cb287-2" tabindex="-1"></a>markerMotifs <span class="ot">&lt;-</span> <span class="fu">getFeatures</span>(projHeme5, <span class="at">select =</span> <span class="fu">paste</span>(motifs, <span class="at">collapse=</span><span class="st">&quot;|&quot;</span>), <span class="at">useMatrix =</span> <span class="st">&quot;MotifMatrix&quot;</span>)</span>
<span id="cb287-3"><a href="motif-deviations.html#cb287-3" tabindex="-1"></a>markerMotifs</span>
<span id="cb287-4"><a href="motif-deviations.html#cb287-4" tabindex="-1"></a><span class="do">##  [1] &quot;z:TBX21_780&quot;          &quot;z:PAX5_709&quot;           &quot;z:IRF4_632&quot;          </span></span>
<span id="cb287-5"><a href="motif-deviations.html#cb287-5" tabindex="-1"></a><span class="do">##  [4] &quot;z:GATA1_383&quot;          &quot;z:CEBPA_155&quot;          &quot;z:EBF1_67&quot;           </span></span>
<span id="cb287-6"><a href="motif-deviations.html#cb287-6" tabindex="-1"></a><span class="do">##  [7] &quot;z:SREBF1_22&quot;          &quot;deviations:TBX21_780&quot; &quot;deviations:PAX5_709&quot; </span></span>
<span id="cb287-7"><a href="motif-deviations.html#cb287-7" tabindex="-1"></a><span class="do">## [10] &quot;deviations:IRF4_632&quot;  &quot;deviations:GATA1_383&quot; &quot;deviations:CEBPA_155&quot;</span></span>
<span id="cb287-8"><a href="motif-deviations.html#cb287-8" tabindex="-1"></a><span class="do">## [13] &quot;deviations:EBF1_67&quot;   &quot;deviations:SREBF1_22&quot;</span></span></code></pre></div>
<p>As mentioned above, <code>MotifMatrix</code> contains <code>seqnames</code> for both z-scores and deviations, shown above by “z:” and “deviations:”. To get just the features corresponding to z-scores, we can use <code>grep</code>. Unfortunately, in the example motifs shown above, you can see that in addition to “EBF1”, we also selected “SREBF1” which we do not want to analyze. Because of this, we remove it below using the <code>%ni%</code> expression which is an ArchR helper function that provides the opposite of <code>%in%</code> from base R.</p>
<div class="sourceCode" id="cb288"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb288-1"><a href="motif-deviations.html#cb288-1" tabindex="-1"></a>markerMotifs <span class="ot">&lt;-</span> <span class="fu">grep</span>(<span class="st">&quot;z:&quot;</span>, markerMotifs, <span class="at">value =</span> <span class="cn">TRUE</span>)</span>
<span id="cb288-2"><a href="motif-deviations.html#cb288-2" tabindex="-1"></a>markerMotifs <span class="ot">&lt;-</span> markerMotifs[markerMotifs <span class="sc">%ni%</span> <span class="st">&quot;z:SREBF1_22&quot;</span>]</span>
<span id="cb288-3"><a href="motif-deviations.html#cb288-3" tabindex="-1"></a>markerMotifs</span>
<span id="cb288-4"><a href="motif-deviations.html#cb288-4" tabindex="-1"></a><span class="do">## [1] &quot;z:TBX21_780&quot; &quot;z:PAX5_709&quot;  &quot;z:IRF4_632&quot;  &quot;z:GATA1_383&quot; &quot;z:CEBPA_155&quot;</span></span>
<span id="cb288-5"><a href="motif-deviations.html#cb288-5" tabindex="-1"></a><span class="do">## [6] &quot;z:EBF1_67&quot;</span></span></code></pre></div>
<p>Now that we have the names of the features that we are interested in, we can plot the distribution of chromVAR deviation scores for each cluster. Notice that we supply the impute weights that we calculated previously during our gene score analyses. As a reminder, these impute weights allow us to smooth the signal across nearby cells which is helpful in the context of our sparse scATAC-seq data.</p>
<div class="sourceCode" id="cb289"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb289-1"><a href="motif-deviations.html#cb289-1" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">plotGroups</span>(<span class="at">ArchRProj =</span> projHeme5, </span>
<span id="cb289-2"><a href="motif-deviations.html#cb289-2" tabindex="-1"></a>  <span class="at">groupBy =</span> <span class="st">&quot;Clusters2&quot;</span>, </span>
<span id="cb289-3"><a href="motif-deviations.html#cb289-3" tabindex="-1"></a>  <span class="at">colorBy =</span> <span class="st">&quot;MotifMatrix&quot;</span>, </span>
<span id="cb289-4"><a href="motif-deviations.html#cb289-4" tabindex="-1"></a>  <span class="at">name =</span> markerMotifs,</span>
<span id="cb289-5"><a href="motif-deviations.html#cb289-5" tabindex="-1"></a>  <span class="at">imputeWeights =</span> <span class="fu">getImputeWeights</span>(projHeme5)</span>
<span id="cb289-6"><a href="motif-deviations.html#cb289-6" tabindex="-1"></a>)</span>
<span id="cb289-7"><a href="motif-deviations.html#cb289-7" tabindex="-1"></a><span class="do">## Getting ImputeWeights</span></span>
<span id="cb289-8"><a href="motif-deviations.html#cb289-8" tabindex="-1"></a><span class="do">## No imputeWeights found, returning NULL</span></span>
<span id="cb289-9"><a href="motif-deviations.html#cb289-9" tabindex="-1"></a><span class="do">## Getting Matrix Values...</span></span>
<span id="cb289-10"><a href="motif-deviations.html#cb289-10" tabindex="-1"></a><span class="do">## 2025-02-06 02:21:05.135753 :</span></span>
<span id="cb289-11"><a href="motif-deviations.html#cb289-11" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb289-12"><a href="motif-deviations.html#cb289-12" tabindex="-1"></a><span class="do">## 1 2 3 4 5 6</span></span></code></pre></div>
<p>We can use <code>cowplot</code> to plot the distributions of all of these motifs in a single plot.</p>
<div class="sourceCode" id="cb290"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb290-1"><a href="motif-deviations.html#cb290-1" tabindex="-1"></a>p2 <span class="ot">&lt;-</span> <span class="fu">lapply</span>(<span class="fu">seq_along</span>(p), <span class="cf">function</span>(x){</span>
<span id="cb290-2"><a href="motif-deviations.html#cb290-2" tabindex="-1"></a>  <span class="cf">if</span>(x <span class="sc">!=</span> <span class="dv">1</span>){</span>
<span id="cb290-3"><a href="motif-deviations.html#cb290-3" tabindex="-1"></a>    p[[x]] <span class="sc">+</span> <span class="fu">guides</span>(<span class="at">color =</span> <span class="st">&quot;none&quot;</span>, <span class="at">fill =</span> <span class="st">&quot;none&quot;</span>) <span class="sc">+</span> </span>
<span id="cb290-4"><a href="motif-deviations.html#cb290-4" tabindex="-1"></a>    <span class="fu">theme_ArchR</span>(<span class="at">baseSize =</span> <span class="dv">6</span>) <span class="sc">+</span></span>
<span id="cb290-5"><a href="motif-deviations.html#cb290-5" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">plot.margin =</span> <span class="fu">unit</span>(<span class="fu">c</span>(<span class="fl">0.1</span>, <span class="fl">0.1</span>, <span class="fl">0.1</span>, <span class="fl">0.1</span>), <span class="st">&quot;cm&quot;</span>)) <span class="sc">+</span></span>
<span id="cb290-6"><a href="motif-deviations.html#cb290-6" tabindex="-1"></a>    <span class="fu">theme</span>(</span>
<span id="cb290-7"><a href="motif-deviations.html#cb290-7" tabindex="-1"></a>        <span class="at">axis.text.y=</span><span class="fu">element_blank</span>(), </span>
<span id="cb290-8"><a href="motif-deviations.html#cb290-8" tabindex="-1"></a>        <span class="at">axis.ticks.y=</span><span class="fu">element_blank</span>(),</span>
<span id="cb290-9"><a href="motif-deviations.html#cb290-9" tabindex="-1"></a>        <span class="at">axis.title.y=</span><span class="fu">element_blank</span>()</span>
<span id="cb290-10"><a href="motif-deviations.html#cb290-10" tabindex="-1"></a>    ) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">&quot;&quot;</span>)</span>
<span id="cb290-11"><a href="motif-deviations.html#cb290-11" tabindex="-1"></a>  }<span class="cf">else</span>{</span>
<span id="cb290-12"><a href="motif-deviations.html#cb290-12" tabindex="-1"></a>    p[[x]] <span class="sc">+</span> <span class="fu">guides</span>(<span class="at">color =</span> <span class="st">&quot;none&quot;</span>, <span class="at">fill =</span> <span class="st">&quot;none&quot;</span>) <span class="sc">+</span> </span>
<span id="cb290-13"><a href="motif-deviations.html#cb290-13" tabindex="-1"></a>    <span class="fu">theme_ArchR</span>(<span class="at">baseSize =</span> <span class="dv">6</span>) <span class="sc">+</span></span>
<span id="cb290-14"><a href="motif-deviations.html#cb290-14" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">plot.margin =</span> <span class="fu">unit</span>(<span class="fu">c</span>(<span class="fl">0.1</span>, <span class="fl">0.1</span>, <span class="fl">0.1</span>, <span class="fl">0.1</span>), <span class="st">&quot;cm&quot;</span>)) <span class="sc">+</span></span>
<span id="cb290-15"><a href="motif-deviations.html#cb290-15" tabindex="-1"></a>    <span class="fu">theme</span>(</span>
<span id="cb290-16"><a href="motif-deviations.html#cb290-16" tabindex="-1"></a>        <span class="at">axis.ticks.y=</span><span class="fu">element_blank</span>(),</span>
<span id="cb290-17"><a href="motif-deviations.html#cb290-17" tabindex="-1"></a>        <span class="at">axis.title.y=</span><span class="fu">element_blank</span>()</span>
<span id="cb290-18"><a href="motif-deviations.html#cb290-18" tabindex="-1"></a>    ) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">&quot;&quot;</span>)</span>
<span id="cb290-19"><a href="motif-deviations.html#cb290-19" tabindex="-1"></a>  }</span>
<span id="cb290-20"><a href="motif-deviations.html#cb290-20" tabindex="-1"></a>})</span>
<span id="cb290-21"><a href="motif-deviations.html#cb290-21" tabindex="-1"></a><span class="fu">do.call</span>(cowplot<span class="sc">::</span>plot_grid, <span class="fu">c</span>(<span class="fu">list</span>(<span class="at">nrow =</span> <span class="dv">1</span>, <span class="at">rel_widths =</span> <span class="fu">c</span>(<span class="dv">2</span>, <span class="fu">rep</span>(<span class="dv">1</span>, <span class="fu">length</span>(p2) <span class="sc">-</span> <span class="dv">1</span>))),p2))</span>
<span id="cb290-22"><a href="motif-deviations.html#cb290-22" tabindex="-1"></a><span class="do">## Picking joint bandwidth of 0.248</span></span>
<span id="cb290-23"><a href="motif-deviations.html#cb290-23" tabindex="-1"></a><span class="do">## Picking joint bandwidth of 0.259</span></span>
<span id="cb290-24"><a href="motif-deviations.html#cb290-24" tabindex="-1"></a><span class="do">## Picking joint bandwidth of 0.294</span></span>
<span id="cb290-25"><a href="motif-deviations.html#cb290-25" tabindex="-1"></a><span class="do">## Picking joint bandwidth of 0.339</span></span>
<span id="cb290-26"><a href="motif-deviations.html#cb290-26" tabindex="-1"></a><span class="do">## Picking joint bandwidth of 0.315</span></span>
<span id="cb290-27"><a href="motif-deviations.html#cb290-27" tabindex="-1"></a><span class="do">## Picking joint bandwidth of 0.281</span></span></code></pre></div>
<p><img src="ArchR_files/figure-html/unnamed-chunk-284-1.svg" width="672" /></p>
<p>To save an editable vectorized version of this plot, we use the <code>plotPDF()</code> function.</p>
<div class="sourceCode" id="cb291"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb291-1"><a href="motif-deviations.html#cb291-1" tabindex="-1"></a><span class="fu">plotPDF</span>(p, <span class="at">name =</span> <span class="st">&quot;Plot-Groups-Deviations-w-Imputation&quot;</span>, <span class="at">width =</span> <span class="dv">5</span>, <span class="at">height =</span> <span class="dv">5</span>, <span class="at">ArchRProj =</span> projHeme5, <span class="at">addDOC =</span> <span class="cn">FALSE</span>)</span>
<span id="cb291-2"><a href="motif-deviations.html#cb291-2" tabindex="-1"></a><span class="do">## Plotting Ggplot!</span></span>
<span id="cb291-3"><a href="motif-deviations.html#cb291-3" tabindex="-1"></a><span class="do">## Picking joint bandwidth of 0.248</span></span>
<span id="cb291-4"><a href="motif-deviations.html#cb291-4" tabindex="-1"></a><span class="do">## Picking joint bandwidth of 0.248</span></span>
<span id="cb291-5"><a href="motif-deviations.html#cb291-5" tabindex="-1"></a><span class="do">## Plotting Ggplot!</span></span>
<span id="cb291-6"><a href="motif-deviations.html#cb291-6" tabindex="-1"></a><span class="do">## Picking joint bandwidth of 0.259</span></span>
<span id="cb291-7"><a href="motif-deviations.html#cb291-7" tabindex="-1"></a><span class="do">## Picking joint bandwidth of 0.259</span></span>
<span id="cb291-8"><a href="motif-deviations.html#cb291-8" tabindex="-1"></a><span class="do">## Plotting Ggplot!</span></span>
<span id="cb291-9"><a href="motif-deviations.html#cb291-9" tabindex="-1"></a><span class="do">## Picking joint bandwidth of 0.294</span></span>
<span id="cb291-10"><a href="motif-deviations.html#cb291-10" tabindex="-1"></a><span class="do">## Picking joint bandwidth of 0.294</span></span>
<span id="cb291-11"><a href="motif-deviations.html#cb291-11" tabindex="-1"></a><span class="do">## Plotting Ggplot!</span></span>
<span id="cb291-12"><a href="motif-deviations.html#cb291-12" tabindex="-1"></a><span class="do">## Picking joint bandwidth of 0.339</span></span>
<span id="cb291-13"><a href="motif-deviations.html#cb291-13" tabindex="-1"></a><span class="do">## Picking joint bandwidth of 0.339</span></span>
<span id="cb291-14"><a href="motif-deviations.html#cb291-14" tabindex="-1"></a><span class="do">## Plotting Ggplot!</span></span>
<span id="cb291-15"><a href="motif-deviations.html#cb291-15" tabindex="-1"></a><span class="do">## Picking joint bandwidth of 0.315</span></span>
<span id="cb291-16"><a href="motif-deviations.html#cb291-16" tabindex="-1"></a><span class="do">## Picking joint bandwidth of 0.315</span></span>
<span id="cb291-17"><a href="motif-deviations.html#cb291-17" tabindex="-1"></a><span class="do">## Plotting Ggplot!</span></span>
<span id="cb291-18"><a href="motif-deviations.html#cb291-18" tabindex="-1"></a><span class="do">## Picking joint bandwidth of 0.281</span></span>
<span id="cb291-19"><a href="motif-deviations.html#cb291-19" tabindex="-1"></a><span class="do">## Picking joint bandwidth of 0.281</span></span></code></pre></div>
<p>Instead of looking at the distributions of these z-scores, we can overlay the z-scores on our UMAP embedding as we’ve done previously for gene scores.</p>
<div class="sourceCode" id="cb292"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb292-1"><a href="motif-deviations.html#cb292-1" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">plotEmbedding</span>(</span>
<span id="cb292-2"><a href="motif-deviations.html#cb292-2" tabindex="-1"></a>    <span class="at">ArchRProj =</span> projHeme5, </span>
<span id="cb292-3"><a href="motif-deviations.html#cb292-3" tabindex="-1"></a>    <span class="at">colorBy =</span> <span class="st">&quot;MotifMatrix&quot;</span>, </span>
<span id="cb292-4"><a href="motif-deviations.html#cb292-4" tabindex="-1"></a>    <span class="at">name =</span> <span class="fu">sort</span>(markerMotifs), </span>
<span id="cb292-5"><a href="motif-deviations.html#cb292-5" tabindex="-1"></a>    <span class="at">embedding =</span> <span class="st">&quot;UMAP&quot;</span>,</span>
<span id="cb292-6"><a href="motif-deviations.html#cb292-6" tabindex="-1"></a>    <span class="at">imputeWeights =</span> <span class="fu">getImputeWeights</span>(projHeme5)</span>
<span id="cb292-7"><a href="motif-deviations.html#cb292-7" tabindex="-1"></a>)</span>
<span id="cb292-8"><a href="motif-deviations.html#cb292-8" tabindex="-1"></a><span class="do">## Getting ImputeWeights</span></span>
<span id="cb292-9"><a href="motif-deviations.html#cb292-9" tabindex="-1"></a><span class="do">## No imputeWeights found, returning NULL</span></span>
<span id="cb292-10"><a href="motif-deviations.html#cb292-10" tabindex="-1"></a><span class="do">## ArchR logging to : ArchRLogs/ArchR-plotEmbedding-9409e4810-Date-2025-02-06_Time-02-21-15.83702.log</span></span>
<span id="cb292-11"><a href="motif-deviations.html#cb292-11" tabindex="-1"></a><span class="do">## If there is an issue, please report to github with logFile!</span></span>
<span id="cb292-12"><a href="motif-deviations.html#cb292-12" tabindex="-1"></a><span class="do">## Getting UMAP Embedding</span></span>
<span id="cb292-13"><a href="motif-deviations.html#cb292-13" tabindex="-1"></a><span class="do">## ColorBy = MotifMatrix</span></span>
<span id="cb292-14"><a href="motif-deviations.html#cb292-14" tabindex="-1"></a><span class="do">## Getting Matrix Values...</span></span>
<span id="cb292-15"><a href="motif-deviations.html#cb292-15" tabindex="-1"></a><span class="do">## 2025-02-06 02:21:17.125601 :</span></span>
<span id="cb292-16"><a href="motif-deviations.html#cb292-16" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb292-17"><a href="motif-deviations.html#cb292-17" tabindex="-1"></a><span class="do">## Plotting Embedding</span></span>
<span id="cb292-18"><a href="motif-deviations.html#cb292-18" tabindex="-1"></a><span class="do">## 1 2 3 4 5 6 </span></span>
<span id="cb292-19"><a href="motif-deviations.html#cb292-19" tabindex="-1"></a><span class="do">## ArchR logging successful to : ArchRLogs/ArchR-plotEmbedding-9409e4810-Date-2025-02-06_Time-02-21-15.83702.log</span></span></code></pre></div>
<p>We can plot all of these motif UMAPs using <code>cowplot</code>.</p>
<div class="sourceCode" id="cb293"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb293-1"><a href="motif-deviations.html#cb293-1" tabindex="-1"></a>p2 <span class="ot">&lt;-</span> <span class="fu">lapply</span>(p, <span class="cf">function</span>(x){</span>
<span id="cb293-2"><a href="motif-deviations.html#cb293-2" tabindex="-1"></a>    x <span class="sc">+</span> <span class="fu">guides</span>(<span class="at">color =</span> <span class="st">&quot;none&quot;</span>, <span class="at">fill =</span> <span class="st">&quot;none&quot;</span>) <span class="sc">+</span> </span>
<span id="cb293-3"><a href="motif-deviations.html#cb293-3" tabindex="-1"></a>    <span class="fu">theme_ArchR</span>(<span class="at">baseSize =</span> <span class="fl">6.5</span>) <span class="sc">+</span></span>
<span id="cb293-4"><a href="motif-deviations.html#cb293-4" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">plot.margin =</span> <span class="fu">unit</span>(<span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>), <span class="st">&quot;cm&quot;</span>)) <span class="sc">+</span></span>
<span id="cb293-5"><a href="motif-deviations.html#cb293-5" tabindex="-1"></a>    <span class="fu">theme</span>(</span>
<span id="cb293-6"><a href="motif-deviations.html#cb293-6" tabindex="-1"></a>        <span class="at">axis.text.x=</span><span class="fu">element_blank</span>(), </span>
<span id="cb293-7"><a href="motif-deviations.html#cb293-7" tabindex="-1"></a>        <span class="at">axis.ticks.x=</span><span class="fu">element_blank</span>(), </span>
<span id="cb293-8"><a href="motif-deviations.html#cb293-8" tabindex="-1"></a>        <span class="at">axis.text.y=</span><span class="fu">element_blank</span>(), </span>
<span id="cb293-9"><a href="motif-deviations.html#cb293-9" tabindex="-1"></a>        <span class="at">axis.ticks.y=</span><span class="fu">element_blank</span>()</span>
<span id="cb293-10"><a href="motif-deviations.html#cb293-10" tabindex="-1"></a>    )</span>
<span id="cb293-11"><a href="motif-deviations.html#cb293-11" tabindex="-1"></a>})</span>
<span id="cb293-12"><a href="motif-deviations.html#cb293-12" tabindex="-1"></a><span class="fu">do.call</span>(cowplot<span class="sc">::</span>plot_grid, <span class="fu">c</span>(<span class="fu">list</span>(<span class="at">ncol =</span> <span class="dv">3</span>),p2))</span></code></pre></div>
<p><img src="ArchR_files/figure-html/unnamed-chunk-287-1.svg" width="672" /></p>
<p>To see how these TF deviation z-scores compare to the inferred gene expression via gene scores of the corresponding TF genes, we can overlay the gene scores for each of these TFs on the UMAP embedding.</p>
<div class="sourceCode" id="cb294"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb294-1"><a href="motif-deviations.html#cb294-1" tabindex="-1"></a>markerGS <span class="ot">&lt;-</span> <span class="fu">getFeatures</span>(projHeme5, <span class="at">select =</span> <span class="fu">paste</span>(motifs, <span class="at">collapse=</span><span class="st">&quot;|&quot;</span>), <span class="at">useMatrix =</span> <span class="st">&quot;GeneScoreMatrix&quot;</span>)</span>
<span id="cb294-2"><a href="motif-deviations.html#cb294-2" tabindex="-1"></a>markerGS <span class="ot">&lt;-</span> markerGS[markerGS <span class="sc">%ni%</span> <span class="fu">c</span>(<span class="st">&quot;SREBF1&quot;</span>,<span class="st">&quot;CEBPA-DT&quot;</span>)]</span>
<span id="cb294-3"><a href="motif-deviations.html#cb294-3" tabindex="-1"></a>markerGS</span>
<span id="cb294-4"><a href="motif-deviations.html#cb294-4" tabindex="-1"></a><span class="do">## [1] &quot;TBX21&quot; &quot;CEBPA&quot; &quot;EBF1&quot;  &quot;IRF4&quot;  &quot;PAX5&quot;  &quot;GATA1&quot;</span></span></code></pre></div>
<div class="sourceCode" id="cb295"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb295-1"><a href="motif-deviations.html#cb295-1" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">plotEmbedding</span>(</span>
<span id="cb295-2"><a href="motif-deviations.html#cb295-2" tabindex="-1"></a>    <span class="at">ArchRProj =</span> projHeme5, </span>
<span id="cb295-3"><a href="motif-deviations.html#cb295-3" tabindex="-1"></a>    <span class="at">colorBy =</span> <span class="st">&quot;GeneScoreMatrix&quot;</span>, </span>
<span id="cb295-4"><a href="motif-deviations.html#cb295-4" tabindex="-1"></a>    <span class="at">name =</span> <span class="fu">sort</span>(markerGS), </span>
<span id="cb295-5"><a href="motif-deviations.html#cb295-5" tabindex="-1"></a>    <span class="at">embedding =</span> <span class="st">&quot;UMAP&quot;</span>,</span>
<span id="cb295-6"><a href="motif-deviations.html#cb295-6" tabindex="-1"></a>    <span class="at">imputeWeights =</span> <span class="fu">getImputeWeights</span>(projHeme5)</span>
<span id="cb295-7"><a href="motif-deviations.html#cb295-7" tabindex="-1"></a>)</span>
<span id="cb295-8"><a href="motif-deviations.html#cb295-8" tabindex="-1"></a><span class="do">## Getting ImputeWeights</span></span>
<span id="cb295-9"><a href="motif-deviations.html#cb295-9" tabindex="-1"></a><span class="do">## No imputeWeights found, returning NULL</span></span>
<span id="cb295-10"><a href="motif-deviations.html#cb295-10" tabindex="-1"></a><span class="do">## ArchR logging to : ArchRLogs/ArchR-plotEmbedding-9740e407f-Date-2025-02-06_Time-02-21-24.516978.log</span></span>
<span id="cb295-11"><a href="motif-deviations.html#cb295-11" tabindex="-1"></a><span class="do">## If there is an issue, please report to github with logFile!</span></span>
<span id="cb295-12"><a href="motif-deviations.html#cb295-12" tabindex="-1"></a><span class="do">## Getting UMAP Embedding</span></span>
<span id="cb295-13"><a href="motif-deviations.html#cb295-13" tabindex="-1"></a><span class="do">## ColorBy = GeneScoreMatrix</span></span>
<span id="cb295-14"><a href="motif-deviations.html#cb295-14" tabindex="-1"></a><span class="do">## Getting Matrix Values...</span></span>
<span id="cb295-15"><a href="motif-deviations.html#cb295-15" tabindex="-1"></a><span class="do">## 2025-02-06 02:21:25.818279 :</span></span>
<span id="cb295-16"><a href="motif-deviations.html#cb295-16" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb295-17"><a href="motif-deviations.html#cb295-17" tabindex="-1"></a><span class="do">## Plotting Embedding</span></span>
<span id="cb295-18"><a href="motif-deviations.html#cb295-18" tabindex="-1"></a><span class="do">## 1 2 3 4 5 6 </span></span>
<span id="cb295-19"><a href="motif-deviations.html#cb295-19" tabindex="-1"></a><span class="do">## ArchR logging successful to : ArchRLogs/ArchR-plotEmbedding-9740e407f-Date-2025-02-06_Time-02-21-24.516978.log</span></span></code></pre></div>
<div class="sourceCode" id="cb296"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb296-1"><a href="motif-deviations.html#cb296-1" tabindex="-1"></a>p2 <span class="ot">&lt;-</span> <span class="fu">lapply</span>(p, <span class="cf">function</span>(x){</span>
<span id="cb296-2"><a href="motif-deviations.html#cb296-2" tabindex="-1"></a>    x <span class="sc">+</span> <span class="fu">guides</span>(<span class="at">color =</span> <span class="st">&quot;none&quot;</span>, <span class="at">fill =</span> <span class="st">&quot;none&quot;</span>) <span class="sc">+</span> </span>
<span id="cb296-3"><a href="motif-deviations.html#cb296-3" tabindex="-1"></a>    <span class="fu">theme_ArchR</span>(<span class="at">baseSize =</span> <span class="fl">6.5</span>) <span class="sc">+</span></span>
<span id="cb296-4"><a href="motif-deviations.html#cb296-4" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">plot.margin =</span> <span class="fu">unit</span>(<span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>), <span class="st">&quot;cm&quot;</span>)) <span class="sc">+</span></span>
<span id="cb296-5"><a href="motif-deviations.html#cb296-5" tabindex="-1"></a>    <span class="fu">theme</span>(</span>
<span id="cb296-6"><a href="motif-deviations.html#cb296-6" tabindex="-1"></a>        <span class="at">axis.text.x=</span><span class="fu">element_blank</span>(), </span>
<span id="cb296-7"><a href="motif-deviations.html#cb296-7" tabindex="-1"></a>        <span class="at">axis.ticks.x=</span><span class="fu">element_blank</span>(), </span>
<span id="cb296-8"><a href="motif-deviations.html#cb296-8" tabindex="-1"></a>        <span class="at">axis.text.y=</span><span class="fu">element_blank</span>(), </span>
<span id="cb296-9"><a href="motif-deviations.html#cb296-9" tabindex="-1"></a>        <span class="at">axis.ticks.y=</span><span class="fu">element_blank</span>()</span>
<span id="cb296-10"><a href="motif-deviations.html#cb296-10" tabindex="-1"></a>    )</span>
<span id="cb296-11"><a href="motif-deviations.html#cb296-11" tabindex="-1"></a>})</span>
<span id="cb296-12"><a href="motif-deviations.html#cb296-12" tabindex="-1"></a><span class="fu">do.call</span>(cowplot<span class="sc">::</span>plot_grid, <span class="fu">c</span>(<span class="fu">list</span>(<span class="at">ncol =</span> <span class="dv">3</span>),p2))</span></code></pre></div>
<p><img src="ArchR_files/figure-html/unnamed-chunk-289-1.svg" width="672" /></p>
<p>Similarly, because we previously linked our scATAC-seq data with corresponding scRNA-seq data, we can plot the linked gene expression for each of these TFs on the UMAP embedding.</p>
<div class="sourceCode" id="cb297"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb297-1"><a href="motif-deviations.html#cb297-1" tabindex="-1"></a>markerRNA <span class="ot">&lt;-</span> <span class="fu">getFeatures</span>(projHeme5, <span class="at">select =</span> <span class="fu">paste</span>(motifs, <span class="at">collapse=</span><span class="st">&quot;|&quot;</span>), <span class="at">useMatrix =</span> <span class="st">&quot;GeneIntegrationMatrix&quot;</span>)</span>
<span id="cb297-2"><a href="motif-deviations.html#cb297-2" tabindex="-1"></a>markerRNA <span class="ot">&lt;-</span> markerRNA[markerRNA <span class="sc">%ni%</span> <span class="fu">c</span>(<span class="st">&quot;SREBF1&quot;</span>,<span class="st">&quot;CEBPA-DT&quot;</span>)]</span>
<span id="cb297-3"><a href="motif-deviations.html#cb297-3" tabindex="-1"></a>markerRNA</span>
<span id="cb297-4"><a href="motif-deviations.html#cb297-4" tabindex="-1"></a><span class="do">## [1] &quot;TBX21&quot; &quot;CEBPA&quot; &quot;EBF1&quot;  &quot;IRF4&quot;  &quot;PAX5&quot;  &quot;GATA1&quot;</span></span></code></pre></div>
<div class="sourceCode" id="cb298"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb298-1"><a href="motif-deviations.html#cb298-1" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">plotEmbedding</span>(</span>
<span id="cb298-2"><a href="motif-deviations.html#cb298-2" tabindex="-1"></a>    <span class="at">ArchRProj =</span> projHeme5, </span>
<span id="cb298-3"><a href="motif-deviations.html#cb298-3" tabindex="-1"></a>    <span class="at">colorBy =</span> <span class="st">&quot;GeneIntegrationMatrix&quot;</span>, </span>
<span id="cb298-4"><a href="motif-deviations.html#cb298-4" tabindex="-1"></a>    <span class="at">name =</span> <span class="fu">sort</span>(markerRNA), </span>
<span id="cb298-5"><a href="motif-deviations.html#cb298-5" tabindex="-1"></a>    <span class="at">embedding =</span> <span class="st">&quot;UMAP&quot;</span>,</span>
<span id="cb298-6"><a href="motif-deviations.html#cb298-6" tabindex="-1"></a>    <span class="at">continuousSet =</span> <span class="st">&quot;blueYellow&quot;</span>,</span>
<span id="cb298-7"><a href="motif-deviations.html#cb298-7" tabindex="-1"></a>    <span class="at">imputeWeights =</span> <span class="fu">getImputeWeights</span>(projHeme5)</span>
<span id="cb298-8"><a href="motif-deviations.html#cb298-8" tabindex="-1"></a>)</span>
<span id="cb298-9"><a href="motif-deviations.html#cb298-9" tabindex="-1"></a><span class="do">## Getting ImputeWeights</span></span>
<span id="cb298-10"><a href="motif-deviations.html#cb298-10" tabindex="-1"></a><span class="do">## No imputeWeights found, returning NULL</span></span>
<span id="cb298-11"><a href="motif-deviations.html#cb298-11" tabindex="-1"></a><span class="do">## ArchR logging to : ArchRLogs/ArchR-plotEmbedding-9594b3859-Date-2025-02-06_Time-02-21-35.228827.log</span></span>
<span id="cb298-12"><a href="motif-deviations.html#cb298-12" tabindex="-1"></a><span class="do">## If there is an issue, please report to github with logFile!</span></span>
<span id="cb298-13"><a href="motif-deviations.html#cb298-13" tabindex="-1"></a><span class="do">## Getting UMAP Embedding</span></span>
<span id="cb298-14"><a href="motif-deviations.html#cb298-14" tabindex="-1"></a><span class="do">## ColorBy = GeneIntegrationMatrix</span></span>
<span id="cb298-15"><a href="motif-deviations.html#cb298-15" tabindex="-1"></a><span class="do">## Getting Matrix Values...</span></span>
<span id="cb298-16"><a href="motif-deviations.html#cb298-16" tabindex="-1"></a><span class="do">## 2025-02-06 02:21:36.506366 :</span></span>
<span id="cb298-17"><a href="motif-deviations.html#cb298-17" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb298-18"><a href="motif-deviations.html#cb298-18" tabindex="-1"></a><span class="do">## Plotting Embedding</span></span>
<span id="cb298-19"><a href="motif-deviations.html#cb298-19" tabindex="-1"></a><span class="do">## 1 2 3 4 5 6 </span></span>
<span id="cb298-20"><a href="motif-deviations.html#cb298-20" tabindex="-1"></a><span class="do">## ArchR logging successful to : ArchRLogs/ArchR-plotEmbedding-9594b3859-Date-2025-02-06_Time-02-21-35.228827.log</span></span></code></pre></div>
<div class="sourceCode" id="cb299"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb299-1"><a href="motif-deviations.html#cb299-1" tabindex="-1"></a>p2 <span class="ot">&lt;-</span> <span class="fu">lapply</span>(p, <span class="cf">function</span>(x){</span>
<span id="cb299-2"><a href="motif-deviations.html#cb299-2" tabindex="-1"></a>    x <span class="sc">+</span> <span class="fu">guides</span>(<span class="at">color =</span> <span class="st">&quot;none&quot;</span>, <span class="at">fill =</span> <span class="st">&quot;none&quot;</span>) <span class="sc">+</span> </span>
<span id="cb299-3"><a href="motif-deviations.html#cb299-3" tabindex="-1"></a>    <span class="fu">theme_ArchR</span>(<span class="at">baseSize =</span> <span class="fl">6.5</span>) <span class="sc">+</span></span>
<span id="cb299-4"><a href="motif-deviations.html#cb299-4" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">plot.margin =</span> <span class="fu">unit</span>(<span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>), <span class="st">&quot;cm&quot;</span>)) <span class="sc">+</span></span>
<span id="cb299-5"><a href="motif-deviations.html#cb299-5" tabindex="-1"></a>    <span class="fu">theme</span>(</span>
<span id="cb299-6"><a href="motif-deviations.html#cb299-6" tabindex="-1"></a>        <span class="at">axis.text.x=</span><span class="fu">element_blank</span>(), </span>
<span id="cb299-7"><a href="motif-deviations.html#cb299-7" tabindex="-1"></a>        <span class="at">axis.ticks.x=</span><span class="fu">element_blank</span>(), </span>
<span id="cb299-8"><a href="motif-deviations.html#cb299-8" tabindex="-1"></a>        <span class="at">axis.text.y=</span><span class="fu">element_blank</span>(), </span>
<span id="cb299-9"><a href="motif-deviations.html#cb299-9" tabindex="-1"></a>        <span class="at">axis.ticks.y=</span><span class="fu">element_blank</span>()</span>
<span id="cb299-10"><a href="motif-deviations.html#cb299-10" tabindex="-1"></a>    )</span>
<span id="cb299-11"><a href="motif-deviations.html#cb299-11" tabindex="-1"></a>})</span>
<span id="cb299-12"><a href="motif-deviations.html#cb299-12" tabindex="-1"></a><span class="fu">do.call</span>(cowplot<span class="sc">::</span>plot_grid, <span class="fu">c</span>(<span class="fu">list</span>(<span class="at">ncol =</span> <span class="dv">3</span>),p2))</span></code></pre></div>
<p><img src="ArchR_files/figure-html/unnamed-chunk-291-1.svg" width="672" /></p>
<p>In addition to the analyses shown above, you may want to plot heatmaps of motif deviations, for example using the <code>plotMarkerHeatmap()</code> function based on the output of <code>getMarkerFeatures()</code>. It is important to remember not to filter for marker features (via <code>cutoff</code>) using the the log2(fold change) of deviations on these heatmaps because this is not relevant for deviations which are not a counts-based measure. Instead, marker features can be identified using <code>MeanDiff</code> (for ex. <code>MeanDiff &gt;= 0.1</code>). Similarly, <code>plotLog2FC</code> should be <code>FALSE</code>.</p>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="chromvar-deviatons-enrichment-with-archr.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="archr-and-custom-deviations.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": null
},
"fontsettings": null,
"edit": null,
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "none"
},
"info": false
});
});
</script>

</body>

</html>
