<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>12.1 Motif Deviations | ArchR: Robust and scaleable analysis of single-cell chromatin accessibility data.</title>
  <meta name="description" content="A guide to ArchR" />
  <meta name="generator" content="bookdown 0.17 and GitBook 2.6.7" />

  <meta property="og:title" content="12.1 Motif Deviations | ArchR: Robust and scaleable analysis of single-cell chromatin accessibility data." />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="A guide to ArchR" />
  <meta name="github-repo" content="GreenleafLab/ArchR" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="12.1 Motif Deviations | ArchR: Robust and scaleable analysis of single-cell chromatin accessibility data." />
  
  <meta name="twitter:description" content="A guide to ArchR" />
  

<meta name="author" content="Jeffrey Granja and Ryan Corces" />


<meta name="date" content="2020-04-18" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="images/favicon.ico" type="image/x-icon" />
<link rel="prev" href="chromvar-deviatons-enrichment-with-archr.html"/>
<link rel="next" href="archr-deviations.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<link href="libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="../index.html">Back to ArchR Home</a></li>

<li class="divider"></li>
<li><a href="index.html#section"><img src="images/ArchR_Logo_Integrated.png" alt="" width="200"></a></li>
<li class="chapter" data-level="1" data-path="getting-started-with-archr.html"><a href="getting-started-with-archr.html"><i class="fa fa-check"></i><b>1</b> Getting Started with ArchR</a>
<ul>
<li class="chapter" data-level="1.1" data-path="a-brief-primer-on-atac-seq-terminology.html"><a href="a-brief-primer-on-atac-seq-terminology.html"><i class="fa fa-check"></i><b>1.1</b> A Brief Primer on ATAC-seq Terminology</a></li>
<li class="chapter" data-level="1.2" data-path="why-use-archr.html"><a href="why-use-archr.html"><i class="fa fa-check"></i><b>1.2</b> Why use ArchR?</a></li>
<li class="chapter" data-level="1.3" data-path="what-is-an-arrow-file-archrproject.html"><a href="what-is-an-arrow-file-archrproject.html"><i class="fa fa-check"></i><b>1.3</b> What is an <code>Arrow file</code> / <code>ArchRProject</code>?</a></li>
<li class="chapter" data-level="1.4" data-path="input-file-types-in-archr.html"><a href="input-file-types-in-archr.html"><i class="fa fa-check"></i><b>1.4</b> Input File Types in ArchR</a></li>
<li class="chapter" data-level="1.5" data-path="getting-set-up.html"><a href="getting-set-up.html"><i class="fa fa-check"></i><b>1.5</b> Getting Set Up</a></li>
<li class="chapter" data-level="1.6" data-path="setting-a-genome-and-geneannotation.html"><a href="setting-a-genome-and-geneannotation.html"><i class="fa fa-check"></i><b>1.6</b> Setting a Genome and GeneAnnotation</a></li>
<li class="chapter" data-level="1.7" data-path="creating-arrow-files.html"><a href="creating-arrow-files.html"><i class="fa fa-check"></i><b>1.7</b> Creating Arrow Files</a></li>
<li class="chapter" data-level="1.8" data-path="session-information.html"><a href="session-information.html"><i class="fa fa-check"></i><b>1.8</b> Session Information</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="doublet-inference-with-archr.html"><a href="doublet-inference-with-archr.html"><i class="fa fa-check"></i><b>2</b> Doublet Inference with ArchR</a>
<ul>
<li class="chapter" data-level="2.1" data-path="inferring-scatac-seq-doublets-with-archr.html"><a href="inferring-scatac-seq-doublets-with-archr.html"><i class="fa fa-check"></i><b>2.1</b> Inferring scATAC-seq Doublets with ArchR</a></li>
<li class="chapter" data-level="2.2" data-path="session-information-1.html"><a href="session-information-1.html"><i class="fa fa-check"></i><b>2.2</b> Session Information</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="creating-an-archrproject.html"><a href="creating-an-archrproject.html"><i class="fa fa-check"></i><b>3</b> Creating an ArchRProject</a>
<ul>
<li class="chapter" data-level="3.1" data-path="creating-an-archrproject-1.html"><a href="creating-an-archrproject-1.html"><i class="fa fa-check"></i><b>3.1</b> Creating An ArchRProject</a></li>
<li class="chapter" data-level="3.2" data-path="manipulating-an-archrproject.html"><a href="manipulating-an-archrproject.html"><i class="fa fa-check"></i><b>3.2</b> Manipulating An ArchRProject</a></li>
<li class="chapter" data-level="3.3" data-path="plotting-sample-statistics-from-an-archrproject.html"><a href="plotting-sample-statistics-from-an-archrproject.html"><i class="fa fa-check"></i><b>3.3</b> Plotting Sample Statistics from an ArchRProject</a></li>
<li class="chapter" data-level="3.4" data-path="plotting-sample-fragment-size-distribution-and-tss-enrichment-profiles-.html"><a href="plotting-sample-fragment-size-distribution-and-tss-enrichment-profiles-.html"><i class="fa fa-check"></i><b>3.4</b> Plotting Sample Fragment Size Distribution and TSS Enrichment Profiles.</a></li>
<li class="chapter" data-level="3.5" data-path="filtering-doublets-from-an-archrproject.html"><a href="filtering-doublets-from-an-archrproject.html"><i class="fa fa-check"></i><b>3.5</b> Filtering Doublets from an ArchRProject</a></li>
<li class="chapter" data-level="3.6" data-path="session-information-2.html"><a href="session-information-2.html"><i class="fa fa-check"></i><b>3.6</b> Session Information</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="dimensionality-reduction-with-archr.html"><a href="dimensionality-reduction-with-archr.html"><i class="fa fa-check"></i><b>4</b> Dimensionality Reduction with ArchR</a>
<ul>
<li class="chapter" data-level="4.1" data-path="archrs-lsi-implementation.html"><a href="archrs-lsi-implementation.html"><i class="fa fa-check"></i><b>4.1</b> ArchR’s LSI Implementation</a></li>
<li class="chapter" data-level="4.2" data-path="iterative-latent-semantic-indexing-lsi.html"><a href="iterative-latent-semantic-indexing-lsi.html"><i class="fa fa-check"></i><b>4.2</b> Iterative Latent Semantic Indexing (LSI)</a></li>
<li class="chapter" data-level="4.3" data-path="estimated-lsi.html"><a href="estimated-lsi.html"><i class="fa fa-check"></i><b>4.3</b> Estimated LSI</a></li>
<li class="chapter" data-level="4.4" data-path="batch-effect-correction-wtih-harmony.html"><a href="batch-effect-correction-wtih-harmony.html"><i class="fa fa-check"></i><b>4.4</b> Batch Effect Correction wtih Harmony</a></li>
<li class="chapter" data-level="4.5" data-path="session-information-3.html"><a href="session-information-3.html"><i class="fa fa-check"></i><b>4.5</b> Session Information</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="clustering-with-archr.html"><a href="clustering-with-archr.html"><i class="fa fa-check"></i><b>5</b> Clustering with ArchR</a>
<ul>
<li class="chapter" data-level="5.1" data-path="clustering-using-seurats-findclusters-function.html"><a href="clustering-using-seurats-findclusters-function.html"><i class="fa fa-check"></i><b>5.1</b> Clustering using Seurat’s <code>FindClusters()</code> function</a></li>
<li class="chapter" data-level="5.2" data-path="session-information-4.html"><a href="session-information-4.html"><i class="fa fa-check"></i><b>5.2</b> Session Information</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="single-cell-embeddings.html"><a href="single-cell-embeddings.html"><i class="fa fa-check"></i><b>6</b> Single Cell Embeddings</a>
<ul>
<li class="chapter" data-level="6.1" data-path="umap.html"><a href="umap.html"><i class="fa fa-check"></i><b>6.1</b> UMAP</a></li>
<li class="chapter" data-level="6.2" data-path="tsne.html"><a href="tsne.html"><i class="fa fa-check"></i><b>6.2</b> tSNE</a></li>
<li class="chapter" data-level="6.3" data-path="umap-of-previous-reduced-dimensions.html"><a href="umap-of-previous-reduced-dimensions.html"><i class="fa fa-check"></i><b>6.3</b> UMAP of previous Reduced Dimensions</a></li>
<li class="chapter" data-level="6.4" data-path="tsne-of-previous-reduced-dimensions.html"><a href="tsne-of-previous-reduced-dimensions.html"><i class="fa fa-check"></i><b>6.4</b> tSNE of previous Reduced Dimensions</a></li>
<li class="chapter" data-level="6.5" data-path="session-information-5.html"><a href="session-information-5.html"><i class="fa fa-check"></i><b>6.5</b> Session Information</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="marker-genes-with-archr.html"><a href="marker-genes-with-archr.html"><i class="fa fa-check"></i><b>7</b> Marker Genes with ArchR</a>
<ul>
<li class="chapter" data-level="7.1" data-path="identification-of-marker-genes.html"><a href="identification-of-marker-genes.html"><i class="fa fa-check"></i><b>7.1</b> Identification of Marker Genes</a></li>
<li class="chapter" data-level="7.2" data-path="marker-genes.html"><a href="marker-genes.html"><i class="fa fa-check"></i><b>7.2</b> Marker Genes</a></li>
<li class="chapter" data-level="7.3" data-path="marker-genes-imputation-with-magic.html"><a href="marker-genes-imputation-with-magic.html"><i class="fa fa-check"></i><b>7.3</b> Marker Genes Imputation with Magic</a></li>
<li class="chapter" data-level="7.4" data-path="track-plotting-with-archrbrowser.html"><a href="track-plotting-with-archrbrowser.html"><i class="fa fa-check"></i><b>7.4</b> Track Plotting with ArchRBrowser</a></li>
<li class="chapter" data-level="7.5" data-path="launching-the-archrbrowser.html"><a href="launching-the-archrbrowser.html"><i class="fa fa-check"></i><b>7.5</b> Launching the ArchRBrowser</a></li>
<li class="chapter" data-level="7.6" data-path="session-information-6.html"><a href="session-information-6.html"><i class="fa fa-check"></i><b>7.6</b> Session Information</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="labeling-clusters-with-scrna.html"><a href="labeling-clusters-with-scrna.html"><i class="fa fa-check"></i><b>8</b> Labeling Clusters with scRNA</a>
<ul>
<li class="chapter" data-level="8.1" data-path="labeling-scatac-cells-with-scrna-clusters.html"><a href="labeling-scatac-cells-with-scrna-clusters.html"><i class="fa fa-check"></i><b>8.1</b> Labeling scATAC cells with scRNA clusters</a></li>
<li class="chapter" data-level="8.2" data-path="adding-pseudo-scrna-profiles-for-each-scatac-cell.html"><a href="adding-pseudo-scrna-profiles-for-each-scatac-cell.html"><i class="fa fa-check"></i><b>8.2</b> Adding Pseudo-scRNA profiles for each scATAC cell</a></li>
<li class="chapter" data-level="8.3" data-path="labeling-scatac-clusters-with-scrna.html"><a href="labeling-scatac-clusters-with-scrna.html"><i class="fa fa-check"></i><b>8.3</b> Labeling scATAC clusters with scRNA</a></li>
<li class="chapter" data-level="8.4" data-path="session-information-7.html"><a href="session-information-7.html"><i class="fa fa-check"></i><b>8.4</b> Session Information</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="calling-peaks-with-archr.html"><a href="calling-peaks-with-archr.html"><i class="fa fa-check"></i><b>9</b> Calling Peaks with ArchR</a>
<ul>
<li class="chapter" data-level="9.1" data-path="calling-peaks-w-macs2.html"><a href="calling-peaks-w-macs2.html"><i class="fa fa-check"></i><b>9.1</b> Calling Peaks w/ Macs2</a></li>
<li class="chapter" data-level="9.2" data-path="calling-peaks-w-tilematrix.html"><a href="calling-peaks-w-tilematrix.html"><i class="fa fa-check"></i><b>9.2</b> Calling Peaks w/ TileMatrix</a></li>
<li class="chapter" data-level="9.3" data-path="add-peak-matrix.html"><a href="add-peak-matrix.html"><i class="fa fa-check"></i><b>9.3</b> Add Peak Matrix</a></li>
<li class="chapter" data-level="9.4" data-path="session-information-8.html"><a href="session-information-8.html"><i class="fa fa-check"></i><b>9.4</b> Session Information</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="identifying-marker-peaks-with-archr.html"><a href="identifying-marker-peaks-with-archr.html"><i class="fa fa-check"></i><b>10</b> Identifying Marker Peaks with ArchR</a>
<ul>
<li class="chapter" data-level="10.1" data-path="identifying-marker-peaks-with-archr-1.html"><a href="identifying-marker-peaks-with-archr-1.html"><i class="fa fa-check"></i><b>10.1</b> Identifying Marker Peaks with ArchR</a></li>
<li class="chapter" data-level="10.2" data-path="pairwise-testing-groups.html"><a href="pairwise-testing-groups.html"><i class="fa fa-check"></i><b>10.2</b> Pairwise Testing Groups</a></li>
<li class="chapter" data-level="10.3" data-path="session-information-9.html"><a href="session-information-9.html"><i class="fa fa-check"></i><b>10.3</b> Session Information</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="peak-annotation-enrichment-with-archr.html"><a href="peak-annotation-enrichment-with-archr.html"><i class="fa fa-check"></i><b>11</b> Peak Annotation Enrichment with ArchR</a>
<ul>
<li class="chapter" data-level="11.1" data-path="motif-enrichment.html"><a href="motif-enrichment.html"><i class="fa fa-check"></i><b>11.1</b> Motif Enrichment</a></li>
<li class="chapter" data-level="11.2" data-path="archr-enrichment.html"><a href="archr-enrichment.html"><i class="fa fa-check"></i><b>11.2</b> ArchR Enrichment</a></li>
<li class="chapter" data-level="11.3" data-path="custom-enrichment.html"><a href="custom-enrichment.html"><i class="fa fa-check"></i><b>11.3</b> Custom Enrichment</a></li>
<li class="chapter" data-level="11.4" data-path="session-information-10.html"><a href="session-information-10.html"><i class="fa fa-check"></i><b>11.4</b> Session Information</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="chromvar-deviatons-enrichment-with-archr.html"><a href="chromvar-deviatons-enrichment-with-archr.html"><i class="fa fa-check"></i><b>12</b> ChromVAR Deviatons Enrichment with ArchR</a>
<ul>
<li class="chapter" data-level="12.1" data-path="motif-deviations.html"><a href="motif-deviations.html"><i class="fa fa-check"></i><b>12.1</b> Motif Deviations</a></li>
<li class="chapter" data-level="12.2" data-path="archr-deviations.html"><a href="archr-deviations.html"><i class="fa fa-check"></i><b>12.2</b> ArchR Deviations</a></li>
<li class="chapter" data-level="12.3" data-path="custom-deviations.html"><a href="custom-deviations.html"><i class="fa fa-check"></i><b>12.3</b> Custom Deviations</a></li>
<li class="chapter" data-level="12.4" data-path="session-information-11.html"><a href="session-information-11.html"><i class="fa fa-check"></i><b>12.4</b> Session Information</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="footprinting-with-archr.html"><a href="footprinting-with-archr.html"><i class="fa fa-check"></i><b>13</b> Footprinting with ArchR</a>
<ul>
<li class="chapter" data-level="13.1" data-path="motif-footprinting.html"><a href="motif-footprinting.html"><i class="fa fa-check"></i><b>13.1</b> Motif Footprinting</a></li>
<li class="chapter" data-level="13.2" data-path="feature-footprinting.html"><a href="feature-footprinting.html"><i class="fa fa-check"></i><b>13.2</b> Feature Footprinting</a></li>
<li class="chapter" data-level="13.3" data-path="session-information-12.html"><a href="session-information-12.html"><i class="fa fa-check"></i><b>13.3</b> Session Information</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="integrative-analysis-with-archr.html"><a href="integrative-analysis-with-archr.html"><i class="fa fa-check"></i><b>14</b> Integrative Analysis with ArchR</a>
<ul>
<li class="chapter" data-level="14.1" data-path="co-accessibility-with-archr.html"><a href="co-accessibility-with-archr.html"><i class="fa fa-check"></i><b>14.1</b> Co-accessibility with ArchR</a></li>
<li class="chapter" data-level="14.2" data-path="peak2genelinkage-with-archr.html"><a href="peak2genelinkage-with-archr.html"><i class="fa fa-check"></i><b>14.2</b> Peak2GeneLinkage with ArchR</a></li>
<li class="chapter" data-level="14.3" data-path="identification-of-positive-tf-regulators.html"><a href="identification-of-positive-tf-regulators.html"><i class="fa fa-check"></i><b>14.3</b> Identification of Positive TF-Regulators</a></li>
<li class="chapter" data-level="14.4" data-path="session-information-13.html"><a href="session-information-13.html"><i class="fa fa-check"></i><b>14.4</b> Session Information</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="trajectory-analysis-with-archr.html"><a href="trajectory-analysis-with-archr.html"><i class="fa fa-check"></i><b>15</b> Trajectory Analysis with ArchR</a>
<ul>
<li class="chapter" data-level="15.1" data-path="myeloid-trajectory.html"><a href="myeloid-trajectory.html"><i class="fa fa-check"></i><b>15.1</b> Myeloid Trajectory</a></li>
<li class="chapter" data-level="15.2" data-path="lymphoid-trajectory.html"><a href="lymphoid-trajectory.html"><i class="fa fa-check"></i><b>15.2</b> Lymphoid Trajectory</a></li>
<li class="chapter" data-level="15.3" data-path="session-information-14.html"><a href="session-information-14.html"><i class="fa fa-check"></i><b>15.3</b> Session Information</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://bookdown.org" target="_blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">ArchR: Robust and scaleable analysis of single-cell chromatin accessibility data.</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="motif-deviations" class="section level2" number="12.1">
<h2><span class="header-section-number">12.1</span> Motif Deviations</h2>
<p>If you have not added motif annotations</p>
<div class="sourceCode" id="cb246"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb246-1"><a href="motif-deviations.html#cb246-1"></a><span class="cf">if</span>(<span class="st">&quot;Motif&quot;</span> <span class="op">%ni%</span><span class="st"> </span><span class="kw">names</span>(projHeme5<span class="op">@</span>peakAnnotation)){</span>
<span id="cb246-2"><a href="motif-deviations.html#cb246-2"></a>    projHeme5 &lt;-<span class="st"> </span><span class="kw">addMotifAnnotations</span>(<span class="dt">ArchRProj =</span> projHeme5, <span class="dt">motifSet =</span> <span class="st">&quot;cisbp&quot;</span>, <span class="dt">name =</span> <span class="st">&quot;Motif&quot;</span>)</span>
<span id="cb246-3"><a href="motif-deviations.html#cb246-3"></a>}</span></code></pre></div>
<p>If you have not added background peaks</p>
<div class="sourceCode" id="cb247"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb247-1"><a href="motif-deviations.html#cb247-1"></a>projHeme5 &lt;-<span class="st"> </span><span class="kw">addBgdPeaks</span>(projHeme5)</span></code></pre></div>
<blockquote>
<p>## Identifying Background Peaks!<br />
## No methods found in package ‘IRanges’ for request: ‘score’ when loading ‘TFBSTools’</p>
</blockquote>
<p>Add ChromVAR Deviations for Motifs</p>
<div class="sourceCode" id="cb248"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb248-1"><a href="motif-deviations.html#cb248-1"></a>projHeme5 &lt;-<span class="st"> </span><span class="kw">addDeviationsMatrix</span>(</span>
<span id="cb248-2"><a href="motif-deviations.html#cb248-2"></a>  <span class="dt">ArchRProj =</span> projHeme5, </span>
<span id="cb248-3"><a href="motif-deviations.html#cb248-3"></a>  <span class="dt">peakAnnotation =</span> <span class="st">&quot;Motif&quot;</span>,</span>
<span id="cb248-4"><a href="motif-deviations.html#cb248-4"></a>  <span class="dt">force =</span> <span class="ot">TRUE</span></span>
<span id="cb248-5"><a href="motif-deviations.html#cb248-5"></a>)</span></code></pre></div>
<blockquote>
<p>## Using Previous Background Peaks!<br />
## ArchR logging to : ArchRLogs/ArchR-addDeviationsMatrix-103741bce39ff-Date-2020-04-15_Time-10-45-35.log<br />
## If there is an issue, please report to github with logFile!<br />
## NULL<br />
## 2020-04-15 10:45:40 : Batch Execution w/ safelapply!, 0 mins elapsed.<br />
## ###########<br />
## 2020-04-15 10:57:19 : Completed Computing Deviations!, 11.734 mins elapsed.<br />
## ###########<br />
## ArchR logging successful to : ArchRLogs/ArchR-addDeviationsMatrix-103741bce39ff-Date-2020-04-15_Time-10-45-35.log</p>
</blockquote>
<p>Plot Variable Motif Annotations</p>
<div class="sourceCode" id="cb249"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb249-1"><a href="motif-deviations.html#cb249-1"></a>plotVarDev &lt;-<span class="st"> </span><span class="kw">getVarDeviations</span>(projHeme5, <span class="dt">plot =</span> <span class="ot">TRUE</span>)</span></code></pre></div>
<blockquote>
<p>## DataFrame with 6 rows and 6 columns<br />
## seqnames idx name combinedVars combinedMeans<br />
## <Rle> <array> <array> <numeric> <numeric><br />
## f388 z 388 GATA2_388 11.9292478607949 -0.034894792575792<br />
## f155 z 155 CEBPA_155 11.8070700579364 -0.174087405321135<br />
## f383 z 383 GATA1_383 11.8045825337775 -0.0378306234562619<br />
## f336 z 336 SPIB_336 11.3432739583017 -0.0819836042460723<br />
## f385 z 385 GATA5_385 10.8828679211543 -0.036867577013264<br />
## f651 z 651 SMARCC1_651 10.2885493109675 -0.131812047523969<br />
## rank<br />
## <integer><br />
## f388 1<br />
## f155 2<br />
## f383 3<br />
## f336 4<br />
## f385 5<br />
## f651 6</p>
</blockquote>
<div class="sourceCode" id="cb250"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb250-1"><a href="motif-deviations.html#cb250-1"></a>plotVarDev</span></code></pre></div>
<blockquote>
<p>## Warning: Removed 1 rows containing missing values (geom_point).</p>
</blockquote>
<p><img src="images/HemeWalkthrough/PNG/Variable-Motif-Deviation-Scores_1.png" width="500" height="500" /></p>
<p>We can then plot this as a nice PDF</p>
<div class="sourceCode" id="cb251"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb251-1"><a href="motif-deviations.html#cb251-1"></a><span class="kw">plotPDF</span>(plotVarDev, <span class="dt">name =</span> <span class="st">&quot;Variable-Motif-Deviation-Scores&quot;</span>, <span class="dt">width =</span> <span class="dv">5</span>, <span class="dt">height =</span> <span class="dv">5</span>, <span class="dt">ArchRProj =</span> projHeme5, <span class="dt">addDOC =</span> <span class="ot">FALSE</span>)</span></code></pre></div>
<blockquote>
<p>## [1] “plotting ggplot!”<br />
## Warning: Removed 1 rows containing missing values (geom_point).<br />
## Warning: Removed 1 rows containing missing values (geom_point).<br />
## [1] 0</p>
</blockquote>
<p>If we want to get motif feature names for downstream plotting</p>
<div class="sourceCode" id="cb252"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb252-1"><a href="motif-deviations.html#cb252-1"></a>motifs &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;GATA1&quot;</span>, <span class="st">&quot;CEBPA&quot;</span>, <span class="st">&quot;EBF1&quot;</span>, <span class="st">&quot;IRF4&quot;</span>, <span class="st">&quot;TBX21&quot;</span>, <span class="st">&quot;PAX5&quot;</span>)</span>
<span id="cb252-2"><a href="motif-deviations.html#cb252-2"></a><span class="kw">getFeatures</span>(projHeme5, <span class="dt">select =</span> <span class="kw">paste</span>(motifs, <span class="dt">collapse=</span><span class="st">&quot;|&quot;</span>), <span class="dt">useMatrix =</span> <span class="st">&quot;MotifMatrix&quot;</span>)</span></code></pre></div>
<blockquote>
<p>## [1] “z:TBX21_780” “z:PAX5_709” “z:IRF4_632”<br />
## [4] “z:GATA1_383” “z:CEBPA_155” “z:EBF1_67”<br />
## [7] “z:SREBF1_22” “deviations:TBX21_780” “deviations:PAX5_709”<br />
## [10] “deviations:IRF4_632” “deviations:GATA1_383” “deviations:CEBPA_155”<br />
## [13] “deviations:EBF1_67” “deviations:SREBF1_22”</p>
</blockquote>
<p>To get deviation z-scores</p>
<div class="sourceCode" id="cb253"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb253-1"><a href="motif-deviations.html#cb253-1"></a>markerMotifs &lt;-<span class="st"> </span><span class="kw">getFeatures</span>(projHeme5, <span class="dt">select =</span> <span class="kw">paste</span>(motifs, <span class="dt">collapse=</span><span class="st">&quot;|&quot;</span>), <span class="dt">useMatrix =</span> <span class="st">&quot;MotifMatrix&quot;</span>)</span>
<span id="cb253-2"><a href="motif-deviations.html#cb253-2"></a>markerMotifs &lt;-<span class="st"> </span><span class="kw">grep</span>(<span class="st">&quot;z:&quot;</span>, markerMotifs, <span class="dt">value =</span> <span class="ot">TRUE</span>)</span>
<span id="cb253-3"><a href="motif-deviations.html#cb253-3"></a>markerMotifs &lt;-<span class="st"> </span>markerMotifs[markerMotifs <span class="op">%ni%</span><span class="st"> &quot;z:SREBF1_22&quot;</span>]</span>
<span id="cb253-4"><a href="motif-deviations.html#cb253-4"></a>markerMotifs</span></code></pre></div>
<blockquote>
<p>## [1] “z:TBX21_780” “z:PAX5_709” “z:IRF4_632” “z:GATA1_383” “z:CEBPA_155”<br />
## [6] “z:EBF1_67”</p>
</blockquote>
<p>We can also now plot the distribution of chromVAR deviation scores for each cluster</p>
<div class="sourceCode" id="cb254"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb254-1"><a href="motif-deviations.html#cb254-1"></a>p &lt;-<span class="st"> </span><span class="kw">plotGroups</span>(<span class="dt">ArchRProj =</span> projHeme5, </span>
<span id="cb254-2"><a href="motif-deviations.html#cb254-2"></a>  <span class="dt">groupBy =</span> <span class="st">&quot;Clusters2&quot;</span>, </span>
<span id="cb254-3"><a href="motif-deviations.html#cb254-3"></a>  <span class="dt">colorBy =</span> <span class="st">&quot;MotifMatrix&quot;</span>, </span>
<span id="cb254-4"><a href="motif-deviations.html#cb254-4"></a>  <span class="dt">name =</span> markerMotifs,</span>
<span id="cb254-5"><a href="motif-deviations.html#cb254-5"></a>  <span class="dt">imputeWeights =</span> <span class="kw">getImputeWeights</span>(projHeme5)</span>
<span id="cb254-6"><a href="motif-deviations.html#cb254-6"></a>)</span></code></pre></div>
<blockquote>
<p>## Getting ImputeWeights<br />
## Getting Matrix Values…<br />
## Getting Matrix Values…<br />
##<br />
## ArchR logging to : ArchRLogs/ArchR-imputeMatrix-103746efc5207-Date-2020-04-15_Time-10-57-40.log<br />
## If there is an issue, please report to github with logFile!<br />
## Using weights on disk<br />
## Using weights on disk<br />
## 1 2 3 4 5 6</p>
</blockquote>
<p>To plot all genes we can use cowplot</p>
<div class="sourceCode" id="cb255"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb255-1"><a href="motif-deviations.html#cb255-1"></a><span class="co">#Rearrange for grid plotting</span></span>
<span id="cb255-2"><a href="motif-deviations.html#cb255-2"></a>p2 &lt;-<span class="st"> </span><span class="kw">lapply</span>(<span class="kw">seq_along</span>(p), <span class="cf">function</span>(x){</span>
<span id="cb255-3"><a href="motif-deviations.html#cb255-3"></a>  <span class="cf">if</span>(x <span class="op">!=</span><span class="st"> </span><span class="dv">1</span>){</span>
<span id="cb255-4"><a href="motif-deviations.html#cb255-4"></a>    p[[x]] <span class="op">+</span><span class="st"> </span><span class="kw">guides</span>(<span class="dt">color =</span> <span class="ot">FALSE</span>, <span class="dt">fill =</span> <span class="ot">FALSE</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb255-5"><a href="motif-deviations.html#cb255-5"></a><span class="st">    </span><span class="kw">theme_ArchR</span>(<span class="dt">baseSize =</span> <span class="dv">6</span>) <span class="op">+</span></span>
<span id="cb255-6"><a href="motif-deviations.html#cb255-6"></a><span class="st">    </span><span class="kw">theme</span>(<span class="dt">plot.margin =</span> <span class="kw">unit</span>(<span class="kw">c</span>(<span class="fl">0.1</span>, <span class="fl">0.1</span>, <span class="fl">0.1</span>, <span class="fl">0.1</span>), <span class="st">&quot;cm&quot;</span>)) <span class="op">+</span></span>
<span id="cb255-7"><a href="motif-deviations.html#cb255-7"></a><span class="st">    </span><span class="kw">theme</span>(</span>
<span id="cb255-8"><a href="motif-deviations.html#cb255-8"></a>        <span class="dt">axis.text.y=</span><span class="kw">element_blank</span>(), </span>
<span id="cb255-9"><a href="motif-deviations.html#cb255-9"></a>        <span class="dt">axis.ticks.y=</span><span class="kw">element_blank</span>(),</span>
<span id="cb255-10"><a href="motif-deviations.html#cb255-10"></a>        <span class="dt">axis.title.y=</span><span class="kw">element_blank</span>()</span>
<span id="cb255-11"><a href="motif-deviations.html#cb255-11"></a>    ) <span class="op">+</span><span class="st"> </span><span class="kw">ylab</span>(<span class="st">&quot;&quot;</span>)</span>
<span id="cb255-12"><a href="motif-deviations.html#cb255-12"></a>  }<span class="cf">else</span>{</span>
<span id="cb255-13"><a href="motif-deviations.html#cb255-13"></a>    p[[x]] <span class="op">+</span><span class="st"> </span><span class="kw">guides</span>(<span class="dt">color =</span> <span class="ot">FALSE</span>, <span class="dt">fill =</span> <span class="ot">FALSE</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb255-14"><a href="motif-deviations.html#cb255-14"></a><span class="st">    </span><span class="kw">theme_ArchR</span>(<span class="dt">baseSize =</span> <span class="dv">6</span>) <span class="op">+</span></span>
<span id="cb255-15"><a href="motif-deviations.html#cb255-15"></a><span class="st">    </span><span class="kw">theme</span>(<span class="dt">plot.margin =</span> <span class="kw">unit</span>(<span class="kw">c</span>(<span class="fl">0.1</span>, <span class="fl">0.1</span>, <span class="fl">0.1</span>, <span class="fl">0.1</span>), <span class="st">&quot;cm&quot;</span>)) <span class="op">+</span></span>
<span id="cb255-16"><a href="motif-deviations.html#cb255-16"></a><span class="st">    </span><span class="kw">theme</span>(</span>
<span id="cb255-17"><a href="motif-deviations.html#cb255-17"></a>        <span class="dt">axis.ticks.y=</span><span class="kw">element_blank</span>(),</span>
<span id="cb255-18"><a href="motif-deviations.html#cb255-18"></a>        <span class="dt">axis.title.y=</span><span class="kw">element_blank</span>()</span>
<span id="cb255-19"><a href="motif-deviations.html#cb255-19"></a>    ) <span class="op">+</span><span class="st"> </span><span class="kw">ylab</span>(<span class="st">&quot;&quot;</span>)</span>
<span id="cb255-20"><a href="motif-deviations.html#cb255-20"></a>  }</span>
<span id="cb255-21"><a href="motif-deviations.html#cb255-21"></a>})</span>
<span id="cb255-22"><a href="motif-deviations.html#cb255-22"></a><span class="kw">do.call</span>(cowplot<span class="op">::</span>plot_grid, <span class="kw">c</span>(<span class="kw">list</span>(<span class="dt">nrow =</span> <span class="dv">1</span>, <span class="dt">rel_widths =</span> <span class="kw">c</span>(<span class="dv">2</span>, <span class="kw">rep</span>(<span class="dv">1</span>, <span class="kw">length</span>(p2) <span class="op">-</span><span class="st"> </span><span class="dv">1</span>))),p2))</span></code></pre></div>
<blockquote>
<p>## Picking joint bandwidth of 0.0181<br />
## Picking joint bandwidth of 0.02<br />
## Picking joint bandwidth of 0.07<br />
## Picking joint bandwidth of 0.109<br />
## Picking joint bandwidth of 0.102<br />
## Picking joint bandwidth of 0.0703</p>
</blockquote>
<p><img src="images/HemeWalkthrough/PNG/Plot-Groups-Deviations-w-Imputation-Cowplot_1.png" width="800" height="800" /></p>
<p>We can then plot this as a nice PDF</p>
<div class="sourceCode" id="cb256"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb256-1"><a href="motif-deviations.html#cb256-1"></a><span class="kw">plotPDF</span>(p, <span class="dt">name =</span> <span class="st">&quot;Plot-Groups-Deviations-w-Imputation&quot;</span>, <span class="dt">width =</span> <span class="dv">5</span>, <span class="dt">height =</span> <span class="dv">5</span>, <span class="dt">ArchRProj =</span> projHeme5, <span class="dt">addDOC =</span> <span class="ot">FALSE</span>)</span></code></pre></div>
<blockquote>
<p>## [1] “plotting ggplot!”<br />
## Picking joint bandwidth of 0.0181<br />
## Picking joint bandwidth of 0.0181<br />
## [1] “plotting ggplot!”<br />
## Picking joint bandwidth of 0.02<br />
## Picking joint bandwidth of 0.02<br />
## [1] “plotting ggplot!”<br />
## Picking joint bandwidth of 0.07<br />
## Picking joint bandwidth of 0.07<br />
## [1] “plotting ggplot!”<br />
## Picking joint bandwidth of 0.109<br />
## Picking joint bandwidth of 0.109<br />
## [1] “plotting ggplot!”<br />
## Picking joint bandwidth of 0.102<br />
## Picking joint bandwidth of 0.102<br />
## [1] “plotting ggplot!”<br />
## Picking joint bandwidth of 0.0703<br />
## Picking joint bandwidth of 0.0703<br />
## [1] 0</p>
</blockquote>
<p>We can now overlay these motif deviation scores on the UMAP embedding</p>
<div class="sourceCode" id="cb257"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb257-1"><a href="motif-deviations.html#cb257-1"></a>p &lt;-<span class="st"> </span><span class="kw">plotEmbedding</span>(</span>
<span id="cb257-2"><a href="motif-deviations.html#cb257-2"></a>    <span class="dt">ArchRProj =</span> projHeme5, </span>
<span id="cb257-3"><a href="motif-deviations.html#cb257-3"></a>    <span class="dt">colorBy =</span> <span class="st">&quot;MotifMatrix&quot;</span>, </span>
<span id="cb257-4"><a href="motif-deviations.html#cb257-4"></a>    <span class="dt">name =</span> <span class="kw">sort</span>(markerMotifs), </span>
<span id="cb257-5"><a href="motif-deviations.html#cb257-5"></a>    <span class="dt">embedding =</span> <span class="st">&quot;UMAP&quot;</span>,</span>
<span id="cb257-6"><a href="motif-deviations.html#cb257-6"></a>    <span class="dt">imputeWeights =</span> <span class="kw">getImputeWeights</span>(projHeme5)</span>
<span id="cb257-7"><a href="motif-deviations.html#cb257-7"></a>)</span></code></pre></div>
<blockquote>
<p>## Getting ImputeWeights<br />
## ArchR logging to : ArchRLogs/ArchR-plotEmbedding-10374bec190f-Date-2020-04-15_Time-10-58-03.log<br />
## If there is an issue, please report to github with logFile!<br />
## Getting UMAP Embedding<br />
## ColorBy = MotifMatrix<br />
## Getting Matrix Values…<br />
## Getting Matrix Values…<br />
##<br />
## Imputing Matrix<br />
## Using weights on disk<br />
## Using weights on disk<br />
## Plotting Embedding<br />
## 1 2 3 4 5 6<br />
## ArchR logging successful to : ArchRLogs/ArchR-plotEmbedding-10374bec190f-Date-2020-04-15_Time-10-58-03.log</p>
</blockquote>
<p>To plot all motifs we can use cowplot</p>
<div class="sourceCode" id="cb258"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb258-1"><a href="motif-deviations.html#cb258-1"></a><span class="co">#Rearrange for grid plotting</span></span>
<span id="cb258-2"><a href="motif-deviations.html#cb258-2"></a>p2 &lt;-<span class="st"> </span><span class="kw">lapply</span>(p, <span class="cf">function</span>(x){</span>
<span id="cb258-3"><a href="motif-deviations.html#cb258-3"></a>    x <span class="op">+</span><span class="st"> </span><span class="kw">guides</span>(<span class="dt">color =</span> <span class="ot">FALSE</span>, <span class="dt">fill =</span> <span class="ot">FALSE</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb258-4"><a href="motif-deviations.html#cb258-4"></a><span class="st">    </span><span class="kw">theme_ArchR</span>(<span class="dt">baseSize =</span> <span class="fl">6.5</span>) <span class="op">+</span></span>
<span id="cb258-5"><a href="motif-deviations.html#cb258-5"></a><span class="st">    </span><span class="kw">theme</span>(<span class="dt">plot.margin =</span> <span class="kw">unit</span>(<span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>), <span class="st">&quot;cm&quot;</span>)) <span class="op">+</span></span>
<span id="cb258-6"><a href="motif-deviations.html#cb258-6"></a><span class="st">    </span><span class="kw">theme</span>(</span>
<span id="cb258-7"><a href="motif-deviations.html#cb258-7"></a>        <span class="dt">axis.text.x=</span><span class="kw">element_blank</span>(), </span>
<span id="cb258-8"><a href="motif-deviations.html#cb258-8"></a>        <span class="dt">axis.ticks.x=</span><span class="kw">element_blank</span>(), </span>
<span id="cb258-9"><a href="motif-deviations.html#cb258-9"></a>        <span class="dt">axis.text.y=</span><span class="kw">element_blank</span>(), </span>
<span id="cb258-10"><a href="motif-deviations.html#cb258-10"></a>        <span class="dt">axis.ticks.y=</span><span class="kw">element_blank</span>()</span>
<span id="cb258-11"><a href="motif-deviations.html#cb258-11"></a>    )</span>
<span id="cb258-12"><a href="motif-deviations.html#cb258-12"></a>})</span>
<span id="cb258-13"><a href="motif-deviations.html#cb258-13"></a><span class="kw">do.call</span>(cowplot<span class="op">::</span>plot_grid, <span class="kw">c</span>(<span class="kw">list</span>(<span class="dt">ncol =</span> <span class="dv">3</span>),p2))</span></code></pre></div>
<p><img src="images/HemeWalkthrough/PNG/Plot-UMAP-MarkerMotifs-W-Imputation_1.png" width="800" height="800" /></p>
<p>We can also plot the gene expression profiles for each of these TFs</p>
<div class="sourceCode" id="cb259"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb259-1"><a href="motif-deviations.html#cb259-1"></a>markerRNA &lt;-<span class="st"> </span><span class="kw">getFeatures</span>(projHeme5, <span class="dt">select =</span> <span class="kw">paste</span>(motifs, <span class="dt">collapse=</span><span class="st">&quot;|&quot;</span>), <span class="dt">useMatrix =</span> <span class="st">&quot;GeneScoreMatrix&quot;</span>)</span>
<span id="cb259-2"><a href="motif-deviations.html#cb259-2"></a>markerRNA &lt;-<span class="st"> </span>markerRNA[markerRNA <span class="op">%ni%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;SREBF1&quot;</span>,<span class="st">&quot;CEBPA-DT&quot;</span>)]</span>
<span id="cb259-3"><a href="motif-deviations.html#cb259-3"></a>markerRNA</span></code></pre></div>
<blockquote>
<p>## [1] “TBX21” “CEBPA” “EBF1” “IRF4” “PAX5” “GATA1”</p>
</blockquote>
<div class="sourceCode" id="cb260"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb260-1"><a href="motif-deviations.html#cb260-1"></a>p &lt;-<span class="st"> </span><span class="kw">plotEmbedding</span>(</span>
<span id="cb260-2"><a href="motif-deviations.html#cb260-2"></a>    <span class="dt">ArchRProj =</span> projHeme5, </span>
<span id="cb260-3"><a href="motif-deviations.html#cb260-3"></a>    <span class="dt">colorBy =</span> <span class="st">&quot;GeneScoreMatrix&quot;</span>, </span>
<span id="cb260-4"><a href="motif-deviations.html#cb260-4"></a>    <span class="dt">name =</span> <span class="kw">sort</span>(markerRNA), </span>
<span id="cb260-5"><a href="motif-deviations.html#cb260-5"></a>    <span class="dt">embedding =</span> <span class="st">&quot;UMAP&quot;</span>,</span>
<span id="cb260-6"><a href="motif-deviations.html#cb260-6"></a>    <span class="dt">imputeWeights =</span> <span class="kw">getImputeWeights</span>(projHeme5)</span>
<span id="cb260-7"><a href="motif-deviations.html#cb260-7"></a>)</span></code></pre></div>
<blockquote>
<p>## Getting ImputeWeights<br />
## ArchR logging to : ArchRLogs/ArchR-plotEmbedding-1037475e6dc52-Date-2020-04-15_Time-10-58-30.log<br />
## If there is an issue, please report to github with logFile!<br />
## Getting UMAP Embedding<br />
## ColorBy = GeneScoreMatrix<br />
## Getting Matrix Values…<br />
## Getting Matrix Values…<br />
##<br />
## Imputing Matrix<br />
## Using weights on disk<br />
## Using weights on disk<br />
## Plotting Embedding<br />
## 1 2 3 4 5 6<br />
## ArchR logging successful to : ArchRLogs/ArchR-plotEmbedding-1037475e6dc52-Date-2020-04-15_Time-10-58-30.log</p>
</blockquote>
<div class="sourceCode" id="cb261"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb261-1"><a href="motif-deviations.html#cb261-1"></a><span class="co">#Rearrange for grid plotting</span></span>
<span id="cb261-2"><a href="motif-deviations.html#cb261-2"></a>p2 &lt;-<span class="st"> </span><span class="kw">lapply</span>(p, <span class="cf">function</span>(x){</span>
<span id="cb261-3"><a href="motif-deviations.html#cb261-3"></a>    x <span class="op">+</span><span class="st"> </span><span class="kw">guides</span>(<span class="dt">color =</span> <span class="ot">FALSE</span>, <span class="dt">fill =</span> <span class="ot">FALSE</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb261-4"><a href="motif-deviations.html#cb261-4"></a><span class="st">    </span><span class="kw">theme_ArchR</span>(<span class="dt">baseSize =</span> <span class="fl">6.5</span>) <span class="op">+</span></span>
<span id="cb261-5"><a href="motif-deviations.html#cb261-5"></a><span class="st">    </span><span class="kw">theme</span>(<span class="dt">plot.margin =</span> <span class="kw">unit</span>(<span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>), <span class="st">&quot;cm&quot;</span>)) <span class="op">+</span></span>
<span id="cb261-6"><a href="motif-deviations.html#cb261-6"></a><span class="st">    </span><span class="kw">theme</span>(</span>
<span id="cb261-7"><a href="motif-deviations.html#cb261-7"></a>        <span class="dt">axis.text.x=</span><span class="kw">element_blank</span>(), </span>
<span id="cb261-8"><a href="motif-deviations.html#cb261-8"></a>        <span class="dt">axis.ticks.x=</span><span class="kw">element_blank</span>(), </span>
<span id="cb261-9"><a href="motif-deviations.html#cb261-9"></a>        <span class="dt">axis.text.y=</span><span class="kw">element_blank</span>(), </span>
<span id="cb261-10"><a href="motif-deviations.html#cb261-10"></a>        <span class="dt">axis.ticks.y=</span><span class="kw">element_blank</span>()</span>
<span id="cb261-11"><a href="motif-deviations.html#cb261-11"></a>    )</span>
<span id="cb261-12"><a href="motif-deviations.html#cb261-12"></a>})</span>
<span id="cb261-13"><a href="motif-deviations.html#cb261-13"></a><span class="kw">do.call</span>(cowplot<span class="op">::</span>plot_grid, <span class="kw">c</span>(<span class="kw">list</span>(<span class="dt">ncol =</span> <span class="dv">3</span>),p2))</span></code></pre></div>
<p><img src="images/HemeWalkthrough/PNG/Plot-UMAP-MarkerMotifsGS-W-Imputation_1.png" width="800" height="800" /></p>
<p>We can also plot the gene expression profiles for each of these TFs</p>
<div class="sourceCode" id="cb262"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb262-1"><a href="motif-deviations.html#cb262-1"></a>markerRNA &lt;-<span class="st"> </span><span class="kw">getFeatures</span>(projHeme5, <span class="dt">select =</span> <span class="kw">paste</span>(motifs, <span class="dt">collapse=</span><span class="st">&quot;|&quot;</span>), <span class="dt">useMatrix =</span> <span class="st">&quot;GeneIntegrationMatrix&quot;</span>)</span>
<span id="cb262-2"><a href="motif-deviations.html#cb262-2"></a>markerRNA &lt;-<span class="st"> </span>markerRNA[markerRNA <span class="op">%ni%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;SREBF1&quot;</span>,<span class="st">&quot;CEBPA-DT&quot;</span>)]</span>
<span id="cb262-3"><a href="motif-deviations.html#cb262-3"></a>markerRNA</span></code></pre></div>
<blockquote>
<p>## [1] “TBX21” “CEBPA” “EBF1” “IRF4” “PAX5” “GATA1”</p>
</blockquote>
<div class="sourceCode" id="cb263"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb263-1"><a href="motif-deviations.html#cb263-1"></a>p &lt;-<span class="st"> </span><span class="kw">plotEmbedding</span>(</span>
<span id="cb263-2"><a href="motif-deviations.html#cb263-2"></a>    <span class="dt">ArchRProj =</span> projHeme5, </span>
<span id="cb263-3"><a href="motif-deviations.html#cb263-3"></a>    <span class="dt">colorBy =</span> <span class="st">&quot;GeneIntegrationMatrix&quot;</span>, </span>
<span id="cb263-4"><a href="motif-deviations.html#cb263-4"></a>    <span class="dt">name =</span> <span class="kw">sort</span>(markerRNA), </span>
<span id="cb263-5"><a href="motif-deviations.html#cb263-5"></a>    <span class="dt">embedding =</span> <span class="st">&quot;UMAP&quot;</span>,</span>
<span id="cb263-6"><a href="motif-deviations.html#cb263-6"></a>    <span class="dt">continuousSet =</span> <span class="st">&quot;blueYellow&quot;</span>,</span>
<span id="cb263-7"><a href="motif-deviations.html#cb263-7"></a>    <span class="dt">imputeWeights =</span> <span class="kw">getImputeWeights</span>(projHeme5)</span>
<span id="cb263-8"><a href="motif-deviations.html#cb263-8"></a>)</span></code></pre></div>
<blockquote>
<p>## Getting ImputeWeights<br />
## ArchR logging to : ArchRLogs/ArchR-plotEmbedding-10374438d3144-Date-2020-04-15_Time-10-58-56.log<br />
## If there is an issue, please report to github with logFile!<br />
## Getting UMAP Embedding<br />
## ColorBy = GeneIntegrationMatrix<br />
## Getting Matrix Values…<br />
## Getting Matrix Values…<br />
##<br />
## Imputing Matrix<br />
## Using weights on disk<br />
## Using weights on disk<br />
## Plotting Embedding<br />
## 1 2 3 4 5 6<br />
## ArchR logging successful to : ArchRLogs/ArchR-plotEmbedding-10374438d3144-Date-2020-04-15_Time-10-58-56.log</p>
</blockquote>
<div class="sourceCode" id="cb264"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb264-1"><a href="motif-deviations.html#cb264-1"></a><span class="co">#Rearrange for grid plotting</span></span>
<span id="cb264-2"><a href="motif-deviations.html#cb264-2"></a>p2 &lt;-<span class="st"> </span><span class="kw">lapply</span>(p, <span class="cf">function</span>(x){</span>
<span id="cb264-3"><a href="motif-deviations.html#cb264-3"></a>    x <span class="op">+</span><span class="st"> </span><span class="kw">guides</span>(<span class="dt">color =</span> <span class="ot">FALSE</span>, <span class="dt">fill =</span> <span class="ot">FALSE</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb264-4"><a href="motif-deviations.html#cb264-4"></a><span class="st">    </span><span class="kw">theme_ArchR</span>(<span class="dt">baseSize =</span> <span class="fl">6.5</span>) <span class="op">+</span></span>
<span id="cb264-5"><a href="motif-deviations.html#cb264-5"></a><span class="st">    </span><span class="kw">theme</span>(<span class="dt">plot.margin =</span> <span class="kw">unit</span>(<span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>), <span class="st">&quot;cm&quot;</span>)) <span class="op">+</span></span>
<span id="cb264-6"><a href="motif-deviations.html#cb264-6"></a><span class="st">    </span><span class="kw">theme</span>(</span>
<span id="cb264-7"><a href="motif-deviations.html#cb264-7"></a>        <span class="dt">axis.text.x=</span><span class="kw">element_blank</span>(), </span>
<span id="cb264-8"><a href="motif-deviations.html#cb264-8"></a>        <span class="dt">axis.ticks.x=</span><span class="kw">element_blank</span>(), </span>
<span id="cb264-9"><a href="motif-deviations.html#cb264-9"></a>        <span class="dt">axis.text.y=</span><span class="kw">element_blank</span>(), </span>
<span id="cb264-10"><a href="motif-deviations.html#cb264-10"></a>        <span class="dt">axis.ticks.y=</span><span class="kw">element_blank</span>()</span>
<span id="cb264-11"><a href="motif-deviations.html#cb264-11"></a>    )</span>
<span id="cb264-12"><a href="motif-deviations.html#cb264-12"></a>})</span>
<span id="cb264-13"><a href="motif-deviations.html#cb264-13"></a><span class="kw">do.call</span>(cowplot<span class="op">::</span>plot_grid, <span class="kw">c</span>(<span class="kw">list</span>(<span class="dt">ncol =</span> <span class="dv">3</span>),p2))</span></code></pre></div>
<p><img src="images/HemeWalkthrough/PNG/Plot-UMAP-MarkerMotifsRNA-W-Imputation_1.png" width="800" height="800" /></p>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="chromvar-deviatons-enrichment-with-archr.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="archr-deviations.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": null
},
"fontsettings": null,
"edit": null,
"history": {
"link": "https://github.com/GreenleafLab/ArchR_Website/commits/dev/12_Deviations.Rmd",
"text": "Chapter edit history"
},
"view": {
"link": "https://github.com/GreenleafLab/ArchR_Website/view/dev/12_Deviations.Rmd",
"text": "View source on GitHub"
},
"download": ["ArchR.pdf"],
"toc": {
"collapse": "none"
},
"search": true,
"info": false
});
});
</script>

</body>

</html>
