<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>17.3 Peak2GeneLinkage with ArchR | ArchR: Robust and scaleable analysis of single-cell chromatin accessibility data.</title>
  <meta name="description" content="A guide to ArchR" />
  <meta name="generator" content="bookdown 0.41 and GitBook 2.6.7" />

  <meta property="og:title" content="17.3 Peak2GeneLinkage with ArchR | ArchR: Robust and scaleable analysis of single-cell chromatin accessibility data." />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="A guide to ArchR" />
  <meta name="github-repo" content="GreenleafLab/ArchR" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="17.3 Peak2GeneLinkage with ArchR | ArchR: Robust and scaleable analysis of single-cell chromatin accessibility data." />
  
  <meta name="twitter:description" content="A guide to ArchR" />
  

<meta name="author" content="Jeffrey Granja and Ryan Corces" />


<meta name="date" content="2025-02-06" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="images/favicon.ico" type="image/x-icon" />
<link rel="prev" href="co-accessibility-with-archr.html"/>
<link rel="next" href="identification-of-positive-tf-regulators.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<link href="libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="../index.html">Back to ArchR Home</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><img src="images/ArchR_Logo_Integrated.png" alt="" width="200"></a></li>
<li class="chapter" data-level="1" data-path="debugging-and-troubleshooting-in-archr.html"><a href="debugging-and-troubleshooting-in-archr.html"><i class="fa fa-check"></i><b>1</b> Debugging and Troubleshooting in ArchR</a></li>
<li class="chapter" data-level="2" data-path="manage-archrs-dependencies.html"><a href="manage-archrs-dependencies.html"><i class="fa fa-check"></i><b>2</b> Manage ArchR’s Dependencies</a>
<ul>
<li class="chapter" data-level="2.1" data-path="using-renv-to-manage-dependencies.html"><a href="using-renv-to-manage-dependencies.html"><i class="fa fa-check"></i><b>2.1</b> Using renv to manage dependencies</a></li>
<li class="chapter" data-level="2.2" data-path="other-environment-managers.html"><a href="other-environment-managers.html"><i class="fa fa-check"></i><b>2.2</b> Other environment managers</a></li>
<li class="chapter" data-level="2.3" data-path="using-docker-for-pre-compiled-environment-setup.html"><a href="using-docker-for-pre-compiled-environment-setup.html"><i class="fa fa-check"></i><b>2.3</b> Using Docker for pre-compiled environment setup</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="getting-started-with-archr.html"><a href="getting-started-with-archr.html"><i class="fa fa-check"></i><b>3</b> Getting Started with ArchR</a>
<ul>
<li class="chapter" data-level="3.1" data-path="a-brief-primer-on-atac-seq-terminology.html"><a href="a-brief-primer-on-atac-seq-terminology.html"><i class="fa fa-check"></i><b>3.1</b> A Brief Primer on ATAC-seq Terminology</a></li>
<li class="chapter" data-level="3.2" data-path="why-use-archr.html"><a href="why-use-archr.html"><i class="fa fa-check"></i><b>3.2</b> Why use ArchR?</a></li>
<li class="chapter" data-level="3.3" data-path="what-is-an-arrow-file-archrproject.html"><a href="what-is-an-arrow-file-archrproject.html"><i class="fa fa-check"></i><b>3.3</b> What is an Arrow file / <code>ArchRProject</code>?</a></li>
<li class="chapter" data-level="3.4" data-path="input-file-types-in-archr.html"><a href="input-file-types-in-archr.html"><i class="fa fa-check"></i><b>3.4</b> Input File Types in ArchR</a></li>
<li class="chapter" data-level="3.5" data-path="getting-set-up.html"><a href="getting-set-up.html"><i class="fa fa-check"></i><b>3.5</b> Getting Set Up</a></li>
<li class="chapter" data-level="3.6" data-path="creating-arrow-files.html"><a href="creating-arrow-files.html"><i class="fa fa-check"></i><b>3.6</b> Creating Arrow Files</a></li>
<li class="chapter" data-level="3.7" data-path="per-cell-quality-control.html"><a href="per-cell-quality-control.html"><i class="fa fa-check"></i><b>3.7</b> Per-cell Quality Control</a></li>
<li class="chapter" data-level="3.8" data-path="using-bam-files-for-arrow-file-creation.html"><a href="using-bam-files-for-arrow-file-creation.html"><i class="fa fa-check"></i><b>3.8</b> Using BAM files for Arrow File creation</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="doublet-inference-with-archr.html"><a href="doublet-inference-with-archr.html"><i class="fa fa-check"></i><b>4</b> Doublet Inference with ArchR</a>
<ul>
<li class="chapter" data-level="4.1" data-path="how-does-doublet-identification-work-in-archr.html"><a href="how-does-doublet-identification-work-in-archr.html"><i class="fa fa-check"></i><b>4.1</b> How does doublet identification work in ArchR?</a></li>
<li class="chapter" data-level="4.2" data-path="inferring-scatac-seq-doublets-with-archr.html"><a href="inferring-scatac-seq-doublets-with-archr.html"><i class="fa fa-check"></i><b>4.2</b> Inferring scATAC-seq Doublets with ArchR</a></li>
<li class="chapter" data-level="4.3" data-path="using-demuxlet-with-archr.html"><a href="using-demuxlet-with-archr.html"><i class="fa fa-check"></i><b>4.3</b> Using <em>demuxlet</em> with ArchR</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="creating-an-archrproject.html"><a href="creating-an-archrproject.html"><i class="fa fa-check"></i><b>5</b> Creating an ArchRProject</a>
<ul>
<li class="chapter" data-level="5.1" data-path="creating-an-archrproject-1.html"><a href="creating-an-archrproject-1.html"><i class="fa fa-check"></i><b>5.1</b> Creating An ArchRProject</a></li>
<li class="chapter" data-level="5.2" data-path="manipulating-an-archrproject.html"><a href="manipulating-an-archrproject.html"><i class="fa fa-check"></i><b>5.2</b> Manipulating An ArchRProject</a></li>
<li class="chapter" data-level="5.3" data-path="plotting-sample-statistics-from-an-archrproject.html"><a href="plotting-sample-statistics-from-an-archrproject.html"><i class="fa fa-check"></i><b>5.3</b> Plotting Sample Statistics from an ArchRProject</a></li>
<li class="chapter" data-level="5.4" data-path="plotting-sample-fragment-size-distribution-and-tss-enrichment-profiles..html"><a href="plotting-sample-fragment-size-distribution-and-tss-enrichment-profiles..html"><i class="fa fa-check"></i><b>5.4</b> Plotting Sample Fragment Size Distribution and TSS Enrichment Profiles.</a></li>
<li class="chapter" data-level="5.5" data-path="saving-and-loading-an-archrproject.html"><a href="saving-and-loading-an-archrproject.html"><i class="fa fa-check"></i><b>5.5</b> Saving and Loading an <code>ArchRProject</code></a></li>
<li class="chapter" data-level="5.6" data-path="filtering-doublets-from-an-archrproject.html"><a href="filtering-doublets-from-an-archrproject.html"><i class="fa fa-check"></i><b>5.6</b> Filtering Doublets from an ArchRProject</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="dimensionality-reduction-with-archr.html"><a href="dimensionality-reduction-with-archr.html"><i class="fa fa-check"></i><b>6</b> Dimensionality Reduction with ArchR</a>
<ul>
<li class="chapter" data-level="6.1" data-path="archrs-lsi-implementation.html"><a href="archrs-lsi-implementation.html"><i class="fa fa-check"></i><b>6.1</b> ArchR’s LSI Implementation</a></li>
<li class="chapter" data-level="6.2" data-path="iterative-latent-semantic-indexing-lsi.html"><a href="iterative-latent-semantic-indexing-lsi.html"><i class="fa fa-check"></i><b>6.2</b> Iterative Latent Semantic Indexing (LSI)</a></li>
<li class="chapter" data-level="6.3" data-path="estimated-lsi.html"><a href="estimated-lsi.html"><i class="fa fa-check"></i><b>6.3</b> Estimated LSI</a></li>
<li class="chapter" data-level="6.4" data-path="batch-effect-correction-wtih-harmony.html"><a href="batch-effect-correction-wtih-harmony.html"><i class="fa fa-check"></i><b>6.4</b> Batch Effect Correction wtih Harmony</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="clustering-with-archr.html"><a href="clustering-with-archr.html"><i class="fa fa-check"></i><b>7</b> Clustering with ArchR</a>
<ul>
<li class="chapter" data-level="7.1" data-path="clustering-using-seurats-findclusters-function.html"><a href="clustering-using-seurats-findclusters-function.html"><i class="fa fa-check"></i><b>7.1</b> Clustering using Seurat’s <code>FindClusters()</code> function</a></li>
<li class="chapter" data-level="7.2" data-path="clustering-using-scran.html"><a href="clustering-using-scran.html"><i class="fa fa-check"></i><b>7.2</b> Clustering using <code>scran</code></a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="single-cell-embeddings.html"><a href="single-cell-embeddings.html"><i class="fa fa-check"></i><b>8</b> Single-cell Embeddings</a>
<ul>
<li class="chapter" data-level="8.1" data-path="uniform-manifold-approximation-and-projection-umap.html"><a href="uniform-manifold-approximation-and-projection-umap.html"><i class="fa fa-check"></i><b>8.1</b> Uniform Manifold Approximation and Projection (UMAP)</a></li>
<li class="chapter" data-level="8.2" data-path="t-stocastic-neighbor-embedding-t-sne.html"><a href="t-stocastic-neighbor-embedding-t-sne.html"><i class="fa fa-check"></i><b>8.2</b> t-Stocastic Neighbor Embedding (t-SNE)</a></li>
<li class="chapter" data-level="8.3" data-path="dimensionality-reduction-after-harmony.html"><a href="dimensionality-reduction-after-harmony.html"><i class="fa fa-check"></i><b>8.3</b> Dimensionality Reduction After Harmony</a></li>
<li class="chapter" data-level="8.4" data-path="highlighting-specific-cells-on-an-embedding.html"><a href="highlighting-specific-cells-on-an-embedding.html"><i class="fa fa-check"></i><b>8.4</b> Highlighting specific cells on an embedding</a></li>
<li class="chapter" data-level="8.5" data-path="importing-an-embedding-from-external-software.html"><a href="importing-an-embedding-from-external-software.html"><i class="fa fa-check"></i><b>8.5</b> Importing an embedding from external software</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="gene-scores-and-marker-genes-with-archr.html"><a href="gene-scores-and-marker-genes-with-archr.html"><i class="fa fa-check"></i><b>9</b> Gene Scores and Marker Genes with ArchR</a>
<ul>
<li class="chapter" data-level="9.1" data-path="calculating-gene-scores-in-archr.html"><a href="calculating-gene-scores-in-archr.html"><i class="fa fa-check"></i><b>9.1</b> Calculating Gene Scores in ArchR</a></li>
<li class="chapter" data-level="9.2" data-path="identification-of-marker-features.html"><a href="identification-of-marker-features.html"><i class="fa fa-check"></i><b>9.2</b> Identification of Marker Features</a></li>
<li class="chapter" data-level="9.3" data-path="identifying-marker-genes.html"><a href="identifying-marker-genes.html"><i class="fa fa-check"></i><b>9.3</b> Identifying Marker Genes</a></li>
<li class="chapter" data-level="9.4" data-path="visualizing-marker-genes-on-an-embedding.html"><a href="visualizing-marker-genes-on-an-embedding.html"><i class="fa fa-check"></i><b>9.4</b> Visualizing Marker Genes on an Embedding</a></li>
<li class="chapter" data-level="9.5" data-path="marker-genes-imputation-with-magic.html"><a href="marker-genes-imputation-with-magic.html"><i class="fa fa-check"></i><b>9.5</b> Marker Genes Imputation with MAGIC</a></li>
<li class="chapter" data-level="9.6" data-path="module-scores.html"><a href="module-scores.html"><i class="fa fa-check"></i><b>9.6</b> Module Scores</a></li>
<li class="chapter" data-level="9.7" data-path="track-plotting-with-archrbrowser.html"><a href="track-plotting-with-archrbrowser.html"><i class="fa fa-check"></i><b>9.7</b> Track Plotting with ArchRBrowser</a></li>
<li class="chapter" data-level="9.8" data-path="launching-the-interactive-archrbrowser.html"><a href="launching-the-interactive-archrbrowser.html"><i class="fa fa-check"></i><b>9.8</b> Launching the Interactive ArchRBrowser</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="defining-cluster-identity-with-scrna-seq.html"><a href="defining-cluster-identity-with-scrna-seq.html"><i class="fa fa-check"></i><b>10</b> Defining Cluster Identity with scRNA-seq</a>
<ul>
<li class="chapter" data-level="10.1" data-path="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html"><i class="fa fa-check"></i><b>10.1</b> Cross-platform linkage of scATAC-seq cells with scRNA-seq cells</a></li>
<li class="chapter" data-level="10.2" data-path="adding-pseudo-scrna-seq-profiles-for-each-scatac-seq-cell.html"><a href="adding-pseudo-scrna-seq-profiles-for-each-scatac-seq-cell.html"><i class="fa fa-check"></i><b>10.2</b> Adding Pseudo-scRNA-seq profiles for each scATAC-seq cell</a></li>
<li class="chapter" data-level="10.3" data-path="labeling-scatac-seq-clusters-with-scrna-seq-information.html"><a href="labeling-scatac-seq-clusters-with-scrna-seq-information.html"><i class="fa fa-check"></i><b>10.3</b> Labeling scATAC-seq clusters with scRNA-seq information</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="pseudo-bulk-replicates-in-archr.html"><a href="pseudo-bulk-replicates-in-archr.html"><i class="fa fa-check"></i><b>11</b> Pseudo-bulk Replicates in ArchR</a>
<ul>
<li class="chapter" data-level="11.1" data-path="how-does-archr-make-pseudo-bulk-replicates.html"><a href="how-does-archr-make-pseudo-bulk-replicates.html"><i class="fa fa-check"></i><b>11.1</b> How Does ArchR Make Pseudo-bulk Replicates?</a></li>
<li class="chapter" data-level="11.2" data-path="making-pseudo-bulk-replicates.html"><a href="making-pseudo-bulk-replicates.html"><i class="fa fa-check"></i><b>11.2</b> Making Pseudo-bulk Replicates</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="calling-peaks-with-archr.html"><a href="calling-peaks-with-archr.html"><i class="fa fa-check"></i><b>12</b> Calling Peaks with ArchR</a>
<ul>
<li class="chapter" data-level="12.1" data-path="the-iterative-overlap-peak-merging-procedure.html"><a href="the-iterative-overlap-peak-merging-procedure.html"><i class="fa fa-check"></i><b>12.1</b> The Iterative Overlap Peak Merging Procedure</a></li>
<li class="chapter" data-level="12.2" data-path="calling-peaks-w-macs2.html"><a href="calling-peaks-w-macs2.html"><i class="fa fa-check"></i><b>12.2</b> Calling Peaks w/ Macs2</a></li>
<li class="chapter" data-level="12.3" data-path="calling-peaks-w-tilematrix.html"><a href="calling-peaks-w-tilematrix.html"><i class="fa fa-check"></i><b>12.3</b> Calling Peaks w/ TileMatrix</a></li>
<li class="chapter" data-level="12.4" data-path="adding-a-peak-matrix.html"><a href="adding-a-peak-matrix.html"><i class="fa fa-check"></i><b>12.4</b> Adding a Peak Matrix</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="identifying-marker-peaks.html"><a href="identifying-marker-peaks.html"><i class="fa fa-check"></i><b>13</b> Identifying Marker Peaks</a>
<ul>
<li class="chapter" data-level="13.1" data-path="identifying-marker-peaks-with-archr.html"><a href="identifying-marker-peaks-with-archr.html"><i class="fa fa-check"></i><b>13.1</b> Identifying Marker Peaks with ArchR</a></li>
<li class="chapter" data-level="13.2" data-path="plotting-marker-peaks-in-archr.html"><a href="plotting-marker-peaks-in-archr.html"><i class="fa fa-check"></i><b>13.2</b> Plotting Marker Peaks in ArchR</a></li>
<li class="chapter" data-level="13.3" data-path="pairwise-testing-between-groups.html"><a href="pairwise-testing-between-groups.html"><i class="fa fa-check"></i><b>13.3</b> Pairwise Testing Between Groups</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="motif-and-feature-enrichment-with-archr.html"><a href="motif-and-feature-enrichment-with-archr.html"><i class="fa fa-check"></i><b>14</b> Motif and Feature Enrichment with ArchR</a>
<ul>
<li class="chapter" data-level="14.1" data-path="motif-enrichment.html"><a href="motif-enrichment.html"><i class="fa fa-check"></i><b>14.1</b> Motif Enrichment</a></li>
<li class="chapter" data-level="14.2" data-path="archr-enrichment.html"><a href="archr-enrichment.html"><i class="fa fa-check"></i><b>14.2</b> ArchR Enrichment</a></li>
<li class="chapter" data-level="14.3" data-path="custom-enrichment.html"><a href="custom-enrichment.html"><i class="fa fa-check"></i><b>14.3</b> Custom Enrichment</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="chromvar-deviatons-enrichment-with-archr.html"><a href="chromvar-deviatons-enrichment-with-archr.html"><i class="fa fa-check"></i><b>15</b> ChromVAR Deviatons Enrichment with ArchR</a>
<ul>
<li class="chapter" data-level="15.1" data-path="motif-deviations.html"><a href="motif-deviations.html"><i class="fa fa-check"></i><b>15.1</b> Motif Deviations</a></li>
<li class="chapter" data-level="15.2" data-path="archr-and-custom-deviations.html"><a href="archr-and-custom-deviations.html"><i class="fa fa-check"></i><b>15.2</b> ArchR and Custom Deviations</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="footprinting-with-archr.html"><a href="footprinting-with-archr.html"><i class="fa fa-check"></i><b>16</b> Footprinting with ArchR</a>
<ul>
<li class="chapter" data-level="16.1" data-path="motif-footprinting.html"><a href="motif-footprinting.html"><i class="fa fa-check"></i><b>16.1</b> Motif Footprinting</a></li>
<li class="chapter" data-level="16.2" data-path="normalization-of-footprints-for-tn5-bias.html"><a href="normalization-of-footprints-for-tn5-bias.html"><i class="fa fa-check"></i><b>16.2</b> Normalization of Footprints for Tn5 Bias</a></li>
<li class="chapter" data-level="16.3" data-path="feature-footprinting.html"><a href="feature-footprinting.html"><i class="fa fa-check"></i><b>16.3</b> Feature Footprinting</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="integrative-analysis-with-archr.html"><a href="integrative-analysis-with-archr.html"><i class="fa fa-check"></i><b>17</b> Integrative Analysis with ArchR</a>
<ul>
<li class="chapter" data-level="17.1" data-path="creating-low-overlapping-aggregates-of-cells.html"><a href="creating-low-overlapping-aggregates-of-cells.html"><i class="fa fa-check"></i><b>17.1</b> Creating Low-Overlapping Aggregates of Cells</a></li>
<li class="chapter" data-level="17.2" data-path="co-accessibility-with-archr.html"><a href="co-accessibility-with-archr.html"><i class="fa fa-check"></i><b>17.2</b> Co-accessibility with ArchR</a></li>
<li class="chapter" data-level="17.3" data-path="peak2genelinkage-with-archr.html"><a href="peak2genelinkage-with-archr.html"><i class="fa fa-check"></i><b>17.3</b> Peak2GeneLinkage with ArchR</a></li>
<li class="chapter" data-level="17.4" data-path="identification-of-positive-tf-regulators.html"><a href="identification-of-positive-tf-regulators.html"><i class="fa fa-check"></i><b>17.4</b> Identification of Positive TF-Regulators</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="trajectory-analysis-with-archr.html"><a href="trajectory-analysis-with-archr.html"><i class="fa fa-check"></i><b>18</b> Trajectory Analysis with ArchR</a>
<ul>
<li class="chapter" data-level="18.1" data-path="myeloid-trajectory---monocyte-differentiation.html"><a href="myeloid-trajectory---monocyte-differentiation.html"><i class="fa fa-check"></i><b>18.1</b> Myeloid Trajectory - Monocyte Differentiation</a></li>
<li class="chapter" data-level="18.2" data-path="lymphoid-trajectory---b-cell-differentiation.html"><a href="lymphoid-trajectory---b-cell-differentiation.html"><i class="fa fa-check"></i><b>18.2</b> Lymphoid Trajectory - B Cell Differentiation</a></li>
<li class="chapter" data-level="18.3" data-path="monocle3-trajectories.html"><a href="monocle3-trajectories.html"><i class="fa fa-check"></i><b>18.3</b> Monocle3 Trajectories</a></li>
<li class="chapter" data-level="18.4" data-path="slingshot-trajectories.html"><a href="slingshot-trajectories.html"><i class="fa fa-check"></i><b>18.4</b> Slingshot Trajectories</a></li>
<li class="chapter" data-level="18.5" data-path="comparing-the-different-trajectories.html"><a href="comparing-the-different-trajectories.html"><i class="fa fa-check"></i><b>18.5</b> Comparing the different trajectories</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="handling-archr-output.html"><a href="handling-archr-output.html"><i class="fa fa-check"></i><b>19</b> Handling ArchR Output</a>
<ul>
<li class="chapter" data-level="19.1" data-path="exporting-fragment-level-data.html"><a href="exporting-fragment-level-data.html"><i class="fa fa-check"></i><b>19.1</b> Exporting fragment-level data</a></li>
<li class="chapter" data-level="19.2" data-path="exporting-matrix-level-data.html"><a href="exporting-matrix-level-data.html"><i class="fa fa-check"></i><b>19.2</b> Exporting matrix-level data</a></li>
<li class="chapter" data-level="19.3" data-path="exporting-pseudo-bulked-data-to-a-summarizedexperiment.html"><a href="exporting-pseudo-bulked-data-to-a-summarizedexperiment.html"><i class="fa fa-check"></i><b>19.3</b> Exporting pseudo-bulked data to a <code>SummarizedExperiment</code></a></li>
<li class="chapter" data-level="19.4" data-path="exporting-pseudo-bulked-data-to-a-bigwig-file.html"><a href="exporting-pseudo-bulked-data-to-a-bigwig-file.html"><i class="fa fa-check"></i><b>19.4</b> Exporting pseudo-bulked data to a bigWig file</a></li>
</ul></li>
<li class="chapter" data-level="20" data-path="integration-with-bulk-atac-seq.html"><a href="integration-with-bulk-atac-seq.html"><i class="fa fa-check"></i><b>20</b> Integration with bulk ATAC-seq</a>
<ul>
<li class="chapter" data-level="20.1" data-path="projecting-bulk-atac-seq-data.html"><a href="projecting-bulk-atac-seq-data.html"><i class="fa fa-check"></i><b>20.1</b> Projecting bulk ATAC-seq data</a></li>
</ul></li>
<li class="chapter" data-level="21" data-path="multiomic-data-analysis-in-archr.html"><a href="multiomic-data-analysis-in-archr.html"><i class="fa fa-check"></i><b>21</b> Multiomic data analysis in ArchR</a>
<ul>
<li class="chapter" data-level="21.1" data-path="importing-data-and-setting-up-a-multiome-project.html"><a href="importing-data-and-setting-up-a-multiome-project.html"><i class="fa fa-check"></i><b>21.1</b> Importing data and setting up a Multiome project</a></li>
<li class="chapter" data-level="21.2" data-path="analysis-of-multiome-data-in-archr.html"><a href="analysis-of-multiome-data-in-archr.html"><i class="fa fa-check"></i><b>21.2</b> Analysis of multiome data in ArchR</a></li>
</ul></li>
<li class="chapter" data-level="22" data-path="plot-aesthetics-in-archr.html"><a href="plot-aesthetics-in-archr.html"><i class="fa fa-check"></i><b>22</b> Plot aesthetics in ArchR</a>
<ul>
<li class="chapter" data-level="22.1" data-path="using-and-manipulating-palettes-in-archr.html"><a href="using-and-manipulating-palettes-in-archr.html"><i class="fa fa-check"></i><b>22.1</b> Using and manipulating palettes in ArchR</a></li>
</ul></li>
<li class="chapter" data-level="23" data-path="testing-your-archr-installation.html"><a href="testing-your-archr-installation.html"><i class="fa fa-check"></i><b>23</b> Testing Your ArchR Installation</a></li>
<li class="chapter" data-level="24" data-path="session-information.html"><a href="session-information.html"><i class="fa fa-check"></i><b>24</b> Session Information</a></li>
<li class="divider"></li>
<li><a href="https://bookdown.org" target="_blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">ArchR: Robust and scaleable analysis of single-cell chromatin accessibility data.</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="peak2genelinkage-with-archr" class="section level2 hasAnchor" number="17.3">
<h2><span class="header-section-number">17.3</span> Peak2GeneLinkage with ArchR<a href="peak2genelinkage-with-archr.html#peak2genelinkage-with-archr" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Similar to co-accessibility, ArchR can also identify so-called “peak-to-gene links”. The primary differences between peak-to-gene links and co-accessibility is that co-accessibility is an ATAC-seq-only analysis that looks for correlations in accessibility between two peaks while peak-to-gene linkage leverages integrated scRNA-seq data to look for correlations between peak accessibility and gene expression. These represent orthogonal approaches to a similar problem. However, because peak-to-gene linkage correlates scATAC-seq and scRNA-seq data, we often think of these links as more relevant to gene regulatory interactions.</p>
<p>To identify peak-to-gene links in ArchR, we use the <code>addPeak2GeneLinks()</code> function.</p>
<div class="sourceCode" id="cb345"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb345-1"><a href="peak2genelinkage-with-archr.html#cb345-1" tabindex="-1"></a>projHeme5 <span class="ot">&lt;-</span> <span class="fu">addPeak2GeneLinks</span>(</span>
<span id="cb345-2"><a href="peak2genelinkage-with-archr.html#cb345-2" tabindex="-1"></a>    <span class="at">ArchRProj =</span> projHeme5,</span>
<span id="cb345-3"><a href="peak2genelinkage-with-archr.html#cb345-3" tabindex="-1"></a>    <span class="at">reducedDims =</span> <span class="st">&quot;IterativeLSI&quot;</span></span>
<span id="cb345-4"><a href="peak2genelinkage-with-archr.html#cb345-4" tabindex="-1"></a>)</span>
<span id="cb345-5"><a href="peak2genelinkage-with-archr.html#cb345-5" tabindex="-1"></a><span class="do">## ArchR logging to : ArchRLogs/ArchR-addPeak2GeneLinks-9711a92c1-Date-2025-02-06_Time-02-47-26.96776.log</span></span>
<span id="cb345-6"><a href="peak2genelinkage-with-archr.html#cb345-6" tabindex="-1"></a><span class="do">## If there is an issue, please report to github with logFile!</span></span>
<span id="cb345-7"><a href="peak2genelinkage-with-archr.html#cb345-7" tabindex="-1"></a><span class="do">## 2025-02-06 02:47:27.421921 : Getting Available Matrices, 0.008 mins elapsed.</span></span>
<span id="cb345-8"><a href="peak2genelinkage-with-archr.html#cb345-8" tabindex="-1"></a><span class="do">## 2025-02-06 02:47:28.973197 : Filtered Low Prediction Score Cells (1719 of 10250, 0.168), 0.008 mins elapsed.</span></span>
<span id="cb345-9"><a href="peak2genelinkage-with-archr.html#cb345-9" tabindex="-1"></a><span class="do">## 2025-02-06 02:47:29.599043 : Computing KNN, 0.019 mins elapsed.</span></span>
<span id="cb345-10"><a href="peak2genelinkage-with-archr.html#cb345-10" tabindex="-1"></a><span class="do">## 2025-02-06 02:47:29.694643 : Identifying Non-Overlapping KNN pairs, 0.02 mins elapsed.</span></span>
<span id="cb345-11"><a href="peak2genelinkage-with-archr.html#cb345-11" tabindex="-1"></a><span class="do">## 2025-02-06 02:47:31.214207 : Identified 493 Groupings!, 0.046 mins elapsed.</span></span>
<span id="cb345-12"><a href="peak2genelinkage-with-archr.html#cb345-12" tabindex="-1"></a><span class="do">## 2025-02-06 02:47:31.244854 : Getting Group RNA Matrix, 0.046 mins elapsed.</span></span>
<span id="cb345-13"><a href="peak2genelinkage-with-archr.html#cb345-13" tabindex="-1"></a><span class="do">## 2025-02-06 02:48:04.61646 : Getting Group ATAC Matrix, 0.602 mins elapsed.</span></span>
<span id="cb345-14"><a href="peak2genelinkage-with-archr.html#cb345-14" tabindex="-1"></a><span class="do">## 2025-02-06 02:48:40.540854 : Normalizing Group Matrices, 1.201 mins elapsed.</span></span>
<span id="cb345-15"><a href="peak2genelinkage-with-archr.html#cb345-15" tabindex="-1"></a><span class="do">## 2025-02-06 02:48:45.903532 : Finding Peak Gene Pairings, 1.29 mins elapsed.</span></span>
<span id="cb345-16"><a href="peak2genelinkage-with-archr.html#cb345-16" tabindex="-1"></a><span class="do">## 2025-02-06 02:48:46.204627 : Computing Correlations, 1.295 mins elapsed.</span></span>
<span id="cb345-17"><a href="peak2genelinkage-with-archr.html#cb345-17" tabindex="-1"></a><span class="do">## 2025-02-06 02:49:02.312747 : Completed Peak2Gene Correlations!, 1.564 mins elapsed.</span></span>
<span id="cb345-18"><a href="peak2genelinkage-with-archr.html#cb345-18" tabindex="-1"></a><span class="do">## ArchR logging successful to : ArchRLogs/ArchR-addPeak2GeneLinks-9711a92c1-Date-2025-02-06_Time-02-47-26.96776.log</span></span></code></pre></div>
<p>We can then retrieve these peak-to-gene links in a similar fashion to how we retrieved co-accessibility interactions by using the <code>getPeak2GeneLinks()</code> function. As we saw previously, this function allows for a user-specified cutoff for correlation and resolution for linkages.</p>
<div class="sourceCode" id="cb346"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb346-1"><a href="peak2genelinkage-with-archr.html#cb346-1" tabindex="-1"></a>p2g <span class="ot">&lt;-</span> <span class="fu">getPeak2GeneLinks</span>(</span>
<span id="cb346-2"><a href="peak2genelinkage-with-archr.html#cb346-2" tabindex="-1"></a>    <span class="at">ArchRProj =</span> projHeme5,</span>
<span id="cb346-3"><a href="peak2genelinkage-with-archr.html#cb346-3" tabindex="-1"></a>    <span class="at">corCutOff =</span> <span class="fl">0.45</span>,</span>
<span id="cb346-4"><a href="peak2genelinkage-with-archr.html#cb346-4" tabindex="-1"></a>    <span class="at">resolution =</span> <span class="dv">1</span>,</span>
<span id="cb346-5"><a href="peak2genelinkage-with-archr.html#cb346-5" tabindex="-1"></a>    <span class="at">returnLoops =</span> <span class="cn">FALSE</span></span>
<span id="cb346-6"><a href="peak2genelinkage-with-archr.html#cb346-6" tabindex="-1"></a>)</span></code></pre></div>
<p>When <code>returnLoops</code> is set to false, this function returns a <code>DataFrame</code> object anaolgous to the <code>DataFrame</code> object returned by <code>getCoAccessibility()</code>. The primary difference is that the indexes for the scATAC-seq peaks are stored in a column called <code>idxATAC</code> and the indexes for the scRNA-seq gene are stored in a column called <code>idxRNA</code>.</p>
<div class="sourceCode" id="cb347"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb347-1"><a href="peak2genelinkage-with-archr.html#cb347-1" tabindex="-1"></a>p2g</span>
<span id="cb347-2"><a href="peak2genelinkage-with-archr.html#cb347-2" tabindex="-1"></a><span class="do">## DataFrame with 43706 rows and 6 columns</span></span>
<span id="cb347-3"><a href="peak2genelinkage-with-archr.html#cb347-3" tabindex="-1"></a><span class="do">##         idxATAC    idxRNA Correlation         FDR  VarQATAC   VarQRNA</span></span>
<span id="cb347-4"><a href="peak2genelinkage-with-archr.html#cb347-4" tabindex="-1"></a><span class="do">##       &lt;integer&gt; &lt;integer&gt;   &lt;numeric&gt;   &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;</span></span>
<span id="cb347-5"><a href="peak2genelinkage-with-archr.html#cb347-5" tabindex="-1"></a><span class="do">## 1             5         2    0.595121 6.46022e-47  0.351537  0.480619</span></span>
<span id="cb347-6"><a href="peak2genelinkage-with-archr.html#cb347-6" tabindex="-1"></a><span class="do">## 2            50         4    0.481528 9.74742e-29  0.740901  0.309768</span></span>
<span id="cb347-7"><a href="peak2genelinkage-with-archr.html#cb347-7" tabindex="-1"></a><span class="do">## 3            51         4    0.523227 1.30350e-34  0.652344  0.309768</span></span>
<span id="cb347-8"><a href="peak2genelinkage-with-archr.html#cb347-8" tabindex="-1"></a><span class="do">## 4            91         5    0.577011 1.60106e-43  0.875806  0.687221</span></span>
<span id="cb347-9"><a href="peak2genelinkage-with-archr.html#cb347-9" tabindex="-1"></a><span class="do">## 5             2         6    0.462365 2.66174e-26  0.904862  0.481049</span></span>
<span id="cb347-10"><a href="peak2genelinkage-with-archr.html#cb347-10" tabindex="-1"></a><span class="do">## ...         ...       ...         ...         ...       ...       ...</span></span>
<span id="cb347-11"><a href="peak2genelinkage-with-archr.html#cb347-11" tabindex="-1"></a><span class="do">## 43702    147359     18589    0.464330 1.52289e-26  0.692552  0.736896</span></span>
<span id="cb347-12"><a href="peak2genelinkage-with-archr.html#cb347-12" tabindex="-1"></a><span class="do">## 43703    147358     18590    0.474616 7.68618e-28  0.355919  0.957045</span></span>
<span id="cb347-13"><a href="peak2genelinkage-with-archr.html#cb347-13" tabindex="-1"></a><span class="do">## 43704    147359     18590    0.521417 2.44218e-34  0.692552  0.957045</span></span>
<span id="cb347-14"><a href="peak2genelinkage-with-archr.html#cb347-14" tabindex="-1"></a><span class="do">## 43705    147373     18590    0.599431 9.35259e-48  0.769238  0.957045</span></span>
<span id="cb347-15"><a href="peak2genelinkage-with-archr.html#cb347-15" tabindex="-1"></a><span class="do">## 43706    147373     18591    0.511117 8.04962e-33  0.769238  0.960594</span></span></code></pre></div>
<p>Here, <code>idxATAC</code> and <code>idxRNA</code> refer to the <strong>row indices</strong> of the peak or gene in the corresponding <code>geneSet</code> or <code>peakSet</code> which can be accessed via the metadata component of the <code>p2g</code> object.</p>
<div class="sourceCode" id="cb348"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb348-1"><a href="peak2genelinkage-with-archr.html#cb348-1" tabindex="-1"></a><span class="fu">metadata</span>(p2g)</span>
<span id="cb348-2"><a href="peak2genelinkage-with-archr.html#cb348-2" tabindex="-1"></a><span class="do">## $peakSet</span></span>
<span id="cb348-3"><a href="peak2genelinkage-with-archr.html#cb348-3" tabindex="-1"></a><span class="do">## GRanges object with 147407 ranges and 0 metadata columns:</span></span>
<span id="cb348-4"><a href="peak2genelinkage-with-archr.html#cb348-4" tabindex="-1"></a><span class="do">##            seqnames              ranges strand</span></span>
<span id="cb348-5"><a href="peak2genelinkage-with-archr.html#cb348-5" tabindex="-1"></a><span class="do">##               &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt;</span></span>
<span id="cb348-6"><a href="peak2genelinkage-with-archr.html#cb348-6" tabindex="-1"></a><span class="do">##        [1]     chr1       752503-753003      *</span></span>
<span id="cb348-7"><a href="peak2genelinkage-with-archr.html#cb348-7" tabindex="-1"></a><span class="do">##        [2]     chr1       762688-763188      *</span></span>
<span id="cb348-8"><a href="peak2genelinkage-with-archr.html#cb348-8" tabindex="-1"></a><span class="do">##        [3]     chr1       773648-774148      *</span></span>
<span id="cb348-9"><a href="peak2genelinkage-with-archr.html#cb348-9" tabindex="-1"></a><span class="do">##        [4]     chr1       779906-780406      *</span></span>
<span id="cb348-10"><a href="peak2genelinkage-with-archr.html#cb348-10" tabindex="-1"></a><span class="do">##        [5]     chr1       801002-801502      *</span></span>
<span id="cb348-11"><a href="peak2genelinkage-with-archr.html#cb348-11" tabindex="-1"></a><span class="do">##        ...      ...                 ...    ...</span></span>
<span id="cb348-12"><a href="peak2genelinkage-with-archr.html#cb348-12" tabindex="-1"></a><span class="do">##   [147403]     chrX 154807254-154807754      *</span></span>
<span id="cb348-13"><a href="peak2genelinkage-with-archr.html#cb348-13" tabindex="-1"></a><span class="do">##   [147404]     chrX 154840939-154841439      *</span></span>
<span id="cb348-14"><a href="peak2genelinkage-with-archr.html#cb348-14" tabindex="-1"></a><span class="do">##   [147405]     chrX 154841881-154842381      *</span></span>
<span id="cb348-15"><a href="peak2genelinkage-with-archr.html#cb348-15" tabindex="-1"></a><span class="do">##   [147406]     chrX 154842390-154842890      *</span></span>
<span id="cb348-16"><a href="peak2genelinkage-with-archr.html#cb348-16" tabindex="-1"></a><span class="do">##   [147407]     chrX 154862036-154862536      *</span></span>
<span id="cb348-17"><a href="peak2genelinkage-with-archr.html#cb348-17" tabindex="-1"></a><span class="do">##   -------</span></span>
<span id="cb348-18"><a href="peak2genelinkage-with-archr.html#cb348-18" tabindex="-1"></a><span class="do">##   seqinfo: 23 sequences from an unspecified genome; no seqlengths</span></span>
<span id="cb348-19"><a href="peak2genelinkage-with-archr.html#cb348-19" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb348-20"><a href="peak2genelinkage-with-archr.html#cb348-20" tabindex="-1"></a><span class="do">## $geneSet</span></span>
<span id="cb348-21"><a href="peak2genelinkage-with-archr.html#cb348-21" tabindex="-1"></a><span class="do">## GRanges object with 18601 ranges and 2 metadata columns:</span></span>
<span id="cb348-22"><a href="peak2genelinkage-with-archr.html#cb348-22" tabindex="-1"></a><span class="do">##           seqnames    ranges strand |        name       idx</span></span>
<span id="cb348-23"><a href="peak2genelinkage-with-archr.html#cb348-23" tabindex="-1"></a><span class="do">##              &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;integer&gt;</span></span>
<span id="cb348-24"><a href="peak2genelinkage-with-archr.html#cb348-24" tabindex="-1"></a><span class="do">##       [1]     chr1     69091      * |       OR4F5         1</span></span>
<span id="cb348-25"><a href="peak2genelinkage-with-archr.html#cb348-25" tabindex="-1"></a><span class="do">##       [2]     chr1    762902      * |   LINC00115         2</span></span>
<span id="cb348-26"><a href="peak2genelinkage-with-archr.html#cb348-26" tabindex="-1"></a><span class="do">##       [3]     chr1    812182      * |      FAM41C         3</span></span>
<span id="cb348-27"><a href="peak2genelinkage-with-archr.html#cb348-27" tabindex="-1"></a><span class="do">##       [4]     chr1    860530      * |      SAMD11         4</span></span>
<span id="cb348-28"><a href="peak2genelinkage-with-archr.html#cb348-28" tabindex="-1"></a><span class="do">##       [5]     chr1    894679      * |       NOC2L         5</span></span>
<span id="cb348-29"><a href="peak2genelinkage-with-archr.html#cb348-29" tabindex="-1"></a><span class="do">##       ...      ...       ...    ... .         ...       ...</span></span>
<span id="cb348-30"><a href="peak2genelinkage-with-archr.html#cb348-30" tabindex="-1"></a><span class="do">##   [18597]     chrX 154299695      * |       BRCC3       708</span></span>
<span id="cb348-31"><a href="peak2genelinkage-with-archr.html#cb348-31" tabindex="-1"></a><span class="do">##   [18598]     chrX 154444701      * |        VBP1       709</span></span>
<span id="cb348-32"><a href="peak2genelinkage-with-archr.html#cb348-32" tabindex="-1"></a><span class="do">##   [18599]     chrX 154493852      * |      RAB39B       710</span></span>
<span id="cb348-33"><a href="peak2genelinkage-with-archr.html#cb348-33" tabindex="-1"></a><span class="do">##   [18600]     chrX 154563986      * |       CLIC2       711</span></span>
<span id="cb348-34"><a href="peak2genelinkage-with-archr.html#cb348-34" tabindex="-1"></a><span class="do">##   [18601]     chrX 154842622      * |       TMLHE       712</span></span>
<span id="cb348-35"><a href="peak2genelinkage-with-archr.html#cb348-35" tabindex="-1"></a><span class="do">##   -------</span></span>
<span id="cb348-36"><a href="peak2genelinkage-with-archr.html#cb348-36" tabindex="-1"></a><span class="do">##   seqinfo: 23 sequences from an unspecified genome; no seqlengths</span></span>
<span id="cb348-37"><a href="peak2genelinkage-with-archr.html#cb348-37" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb348-38"><a href="peak2genelinkage-with-archr.html#cb348-38" tabindex="-1"></a><span class="do">## $seATAC</span></span>
<span id="cb348-39"><a href="peak2genelinkage-with-archr.html#cb348-39" tabindex="-1"></a><span class="do">## [1] &quot;/workspace/ArchR/ArchR_Website_Testing/bookdown/Save-ProjHeme4/Peak2GeneLinks/seATAC-Group-KNN.rds&quot;</span></span>
<span id="cb348-40"><a href="peak2genelinkage-with-archr.html#cb348-40" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb348-41"><a href="peak2genelinkage-with-archr.html#cb348-41" tabindex="-1"></a><span class="do">## $seRNA</span></span>
<span id="cb348-42"><a href="peak2genelinkage-with-archr.html#cb348-42" tabindex="-1"></a><span class="do">## [1] &quot;/workspace/ArchR/ArchR_Website_Testing/bookdown/Save-ProjHeme4/Peak2GeneLinks/seRNA-Group-KNN.rds&quot;</span></span></code></pre></div>
<p>It is important not to confuse <code>idxRNA</code> with the value in the <code>idx</code> column of the <code>geneSet</code>. Within the <code>geneSet</code>, <code>idx</code> corresponds to the chronological position of the given gene across each chromosome. As such, <code>idx</code> is only unique across an individual chromosome and is not relevant for mapping <code>idxRNA</code> to a gene name.</p>
<p>So, if we wanted to add the gene name and peak coordinates to our <code>p2g</code> <code>DataFrame</code> object, we would do the following:</p>
<div class="sourceCode" id="cb349"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb349-1"><a href="peak2genelinkage-with-archr.html#cb349-1" tabindex="-1"></a>p2g<span class="sc">$</span>geneName <span class="ot">&lt;-</span> <span class="fu">mcols</span>(<span class="fu">metadata</span>(p2g)<span class="sc">$</span>geneSet)<span class="sc">$</span>name[p2g<span class="sc">$</span>idxRNA]</span>
<span id="cb349-2"><a href="peak2genelinkage-with-archr.html#cb349-2" tabindex="-1"></a>p2g<span class="sc">$</span>peakName <span class="ot">&lt;-</span> (<span class="fu">metadata</span>(p2g)<span class="sc">$</span>peakSet <span class="sc">%&gt;%</span> {<span class="fu">paste0</span>(<span class="fu">seqnames</span>(.), <span class="st">&quot;_&quot;</span>, <span class="fu">start</span>(.), <span class="st">&quot;_&quot;</span>, <span class="fu">end</span>(.))})[p2g<span class="sc">$</span>idxATAC]</span>
<span id="cb349-3"><a href="peak2genelinkage-with-archr.html#cb349-3" tabindex="-1"></a>p2g</span>
<span id="cb349-4"><a href="peak2genelinkage-with-archr.html#cb349-4" tabindex="-1"></a><span class="do">## DataFrame with 43706 rows and 8 columns</span></span>
<span id="cb349-5"><a href="peak2genelinkage-with-archr.html#cb349-5" tabindex="-1"></a><span class="do">##         idxATAC    idxRNA Correlation         FDR  VarQATAC   VarQRNA</span></span>
<span id="cb349-6"><a href="peak2genelinkage-with-archr.html#cb349-6" tabindex="-1"></a><span class="do">##       &lt;integer&gt; &lt;integer&gt;   &lt;numeric&gt;   &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;</span></span>
<span id="cb349-7"><a href="peak2genelinkage-with-archr.html#cb349-7" tabindex="-1"></a><span class="do">## 1             5         2    0.595121 6.46022e-47  0.351537  0.480619</span></span>
<span id="cb349-8"><a href="peak2genelinkage-with-archr.html#cb349-8" tabindex="-1"></a><span class="do">## 2            50         4    0.481528 9.74742e-29  0.740901  0.309768</span></span>
<span id="cb349-9"><a href="peak2genelinkage-with-archr.html#cb349-9" tabindex="-1"></a><span class="do">## 3            51         4    0.523227 1.30350e-34  0.652344  0.309768</span></span>
<span id="cb349-10"><a href="peak2genelinkage-with-archr.html#cb349-10" tabindex="-1"></a><span class="do">## 4            91         5    0.577011 1.60106e-43  0.875806  0.687221</span></span>
<span id="cb349-11"><a href="peak2genelinkage-with-archr.html#cb349-11" tabindex="-1"></a><span class="do">## 5             2         6    0.462365 2.66174e-26  0.904862  0.481049</span></span>
<span id="cb349-12"><a href="peak2genelinkage-with-archr.html#cb349-12" tabindex="-1"></a><span class="do">## ...         ...       ...         ...         ...       ...       ...</span></span>
<span id="cb349-13"><a href="peak2genelinkage-with-archr.html#cb349-13" tabindex="-1"></a><span class="do">## 43702    147359     18589    0.464330 1.52289e-26  0.692552  0.736896</span></span>
<span id="cb349-14"><a href="peak2genelinkage-with-archr.html#cb349-14" tabindex="-1"></a><span class="do">## 43703    147358     18590    0.474616 7.68618e-28  0.355919  0.957045</span></span>
<span id="cb349-15"><a href="peak2genelinkage-with-archr.html#cb349-15" tabindex="-1"></a><span class="do">## 43704    147359     18590    0.521417 2.44218e-34  0.692552  0.957045</span></span>
<span id="cb349-16"><a href="peak2genelinkage-with-archr.html#cb349-16" tabindex="-1"></a><span class="do">## 43705    147373     18590    0.599431 9.35259e-48  0.769238  0.957045</span></span>
<span id="cb349-17"><a href="peak2genelinkage-with-archr.html#cb349-17" tabindex="-1"></a><span class="do">## 43706    147373     18591    0.511117 8.04962e-33  0.769238  0.960594</span></span>
<span id="cb349-18"><a href="peak2genelinkage-with-archr.html#cb349-18" tabindex="-1"></a><span class="do">##          geneName               peakName</span></span>
<span id="cb349-19"><a href="peak2genelinkage-with-archr.html#cb349-19" tabindex="-1"></a><span class="do">##       &lt;character&gt;            &lt;character&gt;</span></span>
<span id="cb349-20"><a href="peak2genelinkage-with-archr.html#cb349-20" tabindex="-1"></a><span class="do">## 1       LINC00115     chr1_801002_801502</span></span>
<span id="cb349-21"><a href="peak2genelinkage-with-archr.html#cb349-21" tabindex="-1"></a><span class="do">## 2          SAMD11     chr1_974038_974538</span></span>
<span id="cb349-22"><a href="peak2genelinkage-with-archr.html#cb349-22" tabindex="-1"></a><span class="do">## 3          SAMD11     chr1_974855_975355</span></span>
<span id="cb349-23"><a href="peak2genelinkage-with-archr.html#cb349-23" tabindex="-1"></a><span class="do">## 4           NOC2L   chr1_1136870_1137370</span></span>
<span id="cb349-24"><a href="peak2genelinkage-with-archr.html#cb349-24" tabindex="-1"></a><span class="do">## 5          KLHL17     chr1_762688_763188</span></span>
<span id="cb349-25"><a href="peak2genelinkage-with-archr.html#cb349-25" tabindex="-1"></a><span class="do">## ...           ...                    ...</span></span>
<span id="cb349-26"><a href="peak2genelinkage-with-archr.html#cb349-26" tabindex="-1"></a><span class="do">## 43702        GAB3 chrX_153832174_15383..</span></span>
<span id="cb349-27"><a href="peak2genelinkage-with-archr.html#cb349-27" tabindex="-1"></a><span class="do">## 43703        DKC1 chrX_153828590_15382..</span></span>
<span id="cb349-28"><a href="peak2genelinkage-with-archr.html#cb349-28" tabindex="-1"></a><span class="do">## 43704        DKC1 chrX_153832174_15383..</span></span>
<span id="cb349-29"><a href="peak2genelinkage-with-archr.html#cb349-29" tabindex="-1"></a><span class="do">## 43705        DKC1 chrX_153979960_15398..</span></span>
<span id="cb349-30"><a href="peak2genelinkage-with-archr.html#cb349-30" tabindex="-1"></a><span class="do">## 43706        MPP1 chrX_153979960_15398..</span></span></code></pre></div>
<p>You may also notice that there is other information stored within the <code>metadata()</code> of <code>p2g</code> including pointers to files called <code>seATAC</code> and <code>seRNA</code> which represent <code>SummarizedExperiment</code> objects for the ATAC-seq and RNA-seq data used to identify peak-to-gene linkages and these each include both the raw and normalized data matrices used in the analysis.</p>
<div class="sourceCode" id="cb350"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb350-1"><a href="peak2genelinkage-with-archr.html#cb350-1" tabindex="-1"></a><span class="fu">metadata</span>(p2g)<span class="sc">$</span>seATAC</span>
<span id="cb350-2"><a href="peak2genelinkage-with-archr.html#cb350-2" tabindex="-1"></a><span class="do">## [1] &quot;/workspace/ArchR/ArchR_Website_Testing/bookdown/Save-ProjHeme4/Peak2GeneLinks/seATAC-Group-KNN.rds&quot;</span></span>
<span id="cb350-3"><a href="peak2genelinkage-with-archr.html#cb350-3" tabindex="-1"></a><span class="fu">metadata</span>(p2g)<span class="sc">$</span>seRNA</span>
<span id="cb350-4"><a href="peak2genelinkage-with-archr.html#cb350-4" tabindex="-1"></a><span class="do">## [1] &quot;/workspace/ArchR/ArchR_Website_Testing/bookdown/Save-ProjHeme4/Peak2GeneLinks/seRNA-Group-KNN.rds&quot;</span></span></code></pre></div>
<p>If we set <code>returnLoops = TRUE</code>, then <code>getPeak2GeneLinks()</code> will return a loop track <code>GRanges</code> object that connects the peak and gene. As for co-accessibility, the start and end of the <code>IRanges</code> object represent the position of the peak and gene being linked. When <code>resolution = 1</code>, this links the center of the peak to the single-base TSS of the gene.</p>
<div class="sourceCode" id="cb351"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb351-1"><a href="peak2genelinkage-with-archr.html#cb351-1" tabindex="-1"></a>p2g <span class="ot">&lt;-</span> <span class="fu">getPeak2GeneLinks</span>(</span>
<span id="cb351-2"><a href="peak2genelinkage-with-archr.html#cb351-2" tabindex="-1"></a>    <span class="at">ArchRProj =</span> projHeme5,</span>
<span id="cb351-3"><a href="peak2genelinkage-with-archr.html#cb351-3" tabindex="-1"></a>    <span class="at">corCutOff =</span> <span class="fl">0.45</span>,</span>
<span id="cb351-4"><a href="peak2genelinkage-with-archr.html#cb351-4" tabindex="-1"></a>    <span class="at">resolution =</span> <span class="dv">1</span>,</span>
<span id="cb351-5"><a href="peak2genelinkage-with-archr.html#cb351-5" tabindex="-1"></a>    <span class="at">returnLoops =</span> <span class="cn">TRUE</span></span>
<span id="cb351-6"><a href="peak2genelinkage-with-archr.html#cb351-6" tabindex="-1"></a>)</span>
<span id="cb351-7"><a href="peak2genelinkage-with-archr.html#cb351-7" tabindex="-1"></a></span>
<span id="cb351-8"><a href="peak2genelinkage-with-archr.html#cb351-8" tabindex="-1"></a>p2g[[<span class="dv">1</span>]]</span>
<span id="cb351-9"><a href="peak2genelinkage-with-archr.html#cb351-9" tabindex="-1"></a><span class="do">## GRanges object with 43626 ranges and 2 metadata columns:</span></span>
<span id="cb351-10"><a href="peak2genelinkage-with-archr.html#cb351-10" tabindex="-1"></a><span class="do">##           seqnames              ranges strand |     value         FDR</span></span>
<span id="cb351-11"><a href="peak2genelinkage-with-archr.html#cb351-11" tabindex="-1"></a><span class="do">##              &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt;   &lt;numeric&gt;</span></span>
<span id="cb351-12"><a href="peak2genelinkage-with-archr.html#cb351-12" tabindex="-1"></a><span class="do">##       [1]     chr1       762902-801252      * |  0.595121 6.46022e-47</span></span>
<span id="cb351-13"><a href="peak2genelinkage-with-archr.html#cb351-13" tabindex="-1"></a><span class="do">##       [2]     chr1       762938-895967      * |  0.462365 2.66174e-26</span></span>
<span id="cb351-14"><a href="peak2genelinkage-with-archr.html#cb351-14" tabindex="-1"></a><span class="do">##       [3]     chr1       762938-948847      * |  0.550805 5.63187e-39</span></span>
<span id="cb351-15"><a href="peak2genelinkage-with-archr.html#cb351-15" tabindex="-1"></a><span class="do">##       [4]     chr1       860530-974288      * |  0.481528 9.74742e-29</span></span>
<span id="cb351-16"><a href="peak2genelinkage-with-archr.html#cb351-16" tabindex="-1"></a><span class="do">##       [5]     chr1       860530-975105      * |  0.523227 1.30350e-34</span></span>
<span id="cb351-17"><a href="peak2genelinkage-with-archr.html#cb351-17" tabindex="-1"></a><span class="do">##       ...      ...                 ...    ... .       ...         ...</span></span>
<span id="cb351-18"><a href="peak2genelinkage-with-archr.html#cb351-18" tabindex="-1"></a><span class="do">##   [43622]     chrX 153828840-153991031      * |  0.474616 7.68618e-28</span></span>
<span id="cb351-19"><a href="peak2genelinkage-with-archr.html#cb351-19" tabindex="-1"></a><span class="do">##   [43623]     chrX 153832424-153979858      * |  0.464330 1.52289e-26</span></span>
<span id="cb351-20"><a href="peak2genelinkage-with-archr.html#cb351-20" tabindex="-1"></a><span class="do">##   [43624]     chrX 153832424-153991031      * |  0.521417 2.44218e-34</span></span>
<span id="cb351-21"><a href="peak2genelinkage-with-archr.html#cb351-21" tabindex="-1"></a><span class="do">##   [43625]     chrX 153980210-153991031      * |  0.599431 9.35259e-48</span></span>
<span id="cb351-22"><a href="peak2genelinkage-with-archr.html#cb351-22" tabindex="-1"></a><span class="do">##   [43626]     chrX 153980210-154033802      * |  0.511117 8.04962e-33</span></span>
<span id="cb351-23"><a href="peak2genelinkage-with-archr.html#cb351-23" tabindex="-1"></a><span class="do">##   -------</span></span>
<span id="cb351-24"><a href="peak2genelinkage-with-archr.html#cb351-24" tabindex="-1"></a><span class="do">##   seqinfo: 23 sequences from an unspecified genome; no seqlengths</span></span></code></pre></div>
<p>We can alternatively decrease the resolution of these links by setting <code>resolution = 1000</code>. This is primarily useful for plotting the links as a browser tracks because there are instances where many nearby peaks all link to the same gene and this can be difficult to visualize.</p>
<div class="sourceCode" id="cb352"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb352-1"><a href="peak2genelinkage-with-archr.html#cb352-1" tabindex="-1"></a>p2g <span class="ot">&lt;-</span> <span class="fu">getPeak2GeneLinks</span>(</span>
<span id="cb352-2"><a href="peak2genelinkage-with-archr.html#cb352-2" tabindex="-1"></a>    <span class="at">ArchRProj =</span> projHeme5,</span>
<span id="cb352-3"><a href="peak2genelinkage-with-archr.html#cb352-3" tabindex="-1"></a>    <span class="at">corCutOff =</span> <span class="fl">0.45</span>,</span>
<span id="cb352-4"><a href="peak2genelinkage-with-archr.html#cb352-4" tabindex="-1"></a>    <span class="at">resolution =</span> <span class="dv">1000</span>,</span>
<span id="cb352-5"><a href="peak2genelinkage-with-archr.html#cb352-5" tabindex="-1"></a>    <span class="at">returnLoops =</span> <span class="cn">TRUE</span></span>
<span id="cb352-6"><a href="peak2genelinkage-with-archr.html#cb352-6" tabindex="-1"></a>)</span>
<span id="cb352-7"><a href="peak2genelinkage-with-archr.html#cb352-7" tabindex="-1"></a></span>
<span id="cb352-8"><a href="peak2genelinkage-with-archr.html#cb352-8" tabindex="-1"></a>p2g[[<span class="dv">1</span>]]</span>
<span id="cb352-9"><a href="peak2genelinkage-with-archr.html#cb352-9" tabindex="-1"></a><span class="do">## GRanges object with 42045 ranges and 2 metadata columns:</span></span>
<span id="cb352-10"><a href="peak2genelinkage-with-archr.html#cb352-10" tabindex="-1"></a><span class="do">##           seqnames              ranges strand |     value         FDR</span></span>
<span id="cb352-11"><a href="peak2genelinkage-with-archr.html#cb352-11" tabindex="-1"></a><span class="do">##              &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt;   &lt;numeric&gt;</span></span>
<span id="cb352-12"><a href="peak2genelinkage-with-archr.html#cb352-12" tabindex="-1"></a><span class="do">##       [1]     chr1       762500-801500      * |  0.595121 6.46022e-47</span></span>
<span id="cb352-13"><a href="peak2genelinkage-with-archr.html#cb352-13" tabindex="-1"></a><span class="do">##       [2]     chr1       762500-895500      * |  0.462365 2.66174e-26</span></span>
<span id="cb352-14"><a href="peak2genelinkage-with-archr.html#cb352-14" tabindex="-1"></a><span class="do">##       [3]     chr1       762500-948500      * |  0.550805 5.63187e-39</span></span>
<span id="cb352-15"><a href="peak2genelinkage-with-archr.html#cb352-15" tabindex="-1"></a><span class="do">##       [4]     chr1       860500-974500      * |  0.481528 9.74742e-29</span></span>
<span id="cb352-16"><a href="peak2genelinkage-with-archr.html#cb352-16" tabindex="-1"></a><span class="do">##       [5]     chr1       860500-975500      * |  0.523227 1.30350e-34</span></span>
<span id="cb352-17"><a href="peak2genelinkage-with-archr.html#cb352-17" tabindex="-1"></a><span class="do">##       ...      ...                 ...    ... .       ...         ...</span></span>
<span id="cb352-18"><a href="peak2genelinkage-with-archr.html#cb352-18" tabindex="-1"></a><span class="do">##   [42041]     chrX 153828500-153991500      * |  0.474616 7.68618e-28</span></span>
<span id="cb352-19"><a href="peak2genelinkage-with-archr.html#cb352-19" tabindex="-1"></a><span class="do">##   [42042]     chrX 153832500-153979500      * |  0.464330 1.52289e-26</span></span>
<span id="cb352-20"><a href="peak2genelinkage-with-archr.html#cb352-20" tabindex="-1"></a><span class="do">##   [42043]     chrX 153832500-153991500      * |  0.521417 2.44218e-34</span></span>
<span id="cb352-21"><a href="peak2genelinkage-with-archr.html#cb352-21" tabindex="-1"></a><span class="do">##   [42044]     chrX 153980500-153991500      * |  0.599431 9.35259e-48</span></span>
<span id="cb352-22"><a href="peak2genelinkage-with-archr.html#cb352-22" tabindex="-1"></a><span class="do">##   [42045]     chrX 153980500-154033500      * |  0.511117 8.04962e-33</span></span>
<span id="cb352-23"><a href="peak2genelinkage-with-archr.html#cb352-23" tabindex="-1"></a><span class="do">##   -------</span></span>
<span id="cb352-24"><a href="peak2genelinkage-with-archr.html#cb352-24" tabindex="-1"></a><span class="do">##   seqinfo: 23 sequences from an unspecified genome; no seqlengths</span></span></code></pre></div>
<p>Decreasing the resolution even further also decreases the total number of peak-to-gene links identified.</p>
<div class="sourceCode" id="cb353"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb353-1"><a href="peak2genelinkage-with-archr.html#cb353-1" tabindex="-1"></a>p2g <span class="ot">&lt;-</span> <span class="fu">getPeak2GeneLinks</span>(</span>
<span id="cb353-2"><a href="peak2genelinkage-with-archr.html#cb353-2" tabindex="-1"></a>    <span class="at">ArchRProj =</span> projHeme5,</span>
<span id="cb353-3"><a href="peak2genelinkage-with-archr.html#cb353-3" tabindex="-1"></a>    <span class="at">corCutOff =</span> <span class="fl">0.45</span>,</span>
<span id="cb353-4"><a href="peak2genelinkage-with-archr.html#cb353-4" tabindex="-1"></a>    <span class="at">resolution =</span> <span class="dv">10000</span>,</span>
<span id="cb353-5"><a href="peak2genelinkage-with-archr.html#cb353-5" tabindex="-1"></a>    <span class="at">returnLoops =</span> <span class="cn">TRUE</span></span>
<span id="cb353-6"><a href="peak2genelinkage-with-archr.html#cb353-6" tabindex="-1"></a>)</span>
<span id="cb353-7"><a href="peak2genelinkage-with-archr.html#cb353-7" tabindex="-1"></a></span>
<span id="cb353-8"><a href="peak2genelinkage-with-archr.html#cb353-8" tabindex="-1"></a>p2g[[<span class="dv">1</span>]]</span>
<span id="cb353-9"><a href="peak2genelinkage-with-archr.html#cb353-9" tabindex="-1"></a><span class="do">## GRanges object with 33725 ranges and 2 metadata columns:</span></span>
<span id="cb353-10"><a href="peak2genelinkage-with-archr.html#cb353-10" tabindex="-1"></a><span class="do">##           seqnames              ranges strand |     value         FDR</span></span>
<span id="cb353-11"><a href="peak2genelinkage-with-archr.html#cb353-11" tabindex="-1"></a><span class="do">##              &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt;   &lt;numeric&gt;</span></span>
<span id="cb353-12"><a href="peak2genelinkage-with-archr.html#cb353-12" tabindex="-1"></a><span class="do">##       [1]     chr1       765000-805000      * |  0.595121 6.46022e-47</span></span>
<span id="cb353-13"><a href="peak2genelinkage-with-archr.html#cb353-13" tabindex="-1"></a><span class="do">##       [2]     chr1       765000-895000      * |  0.462365 2.66174e-26</span></span>
<span id="cb353-14"><a href="peak2genelinkage-with-archr.html#cb353-14" tabindex="-1"></a><span class="do">##       [3]     chr1       765000-945000      * |  0.550805 5.63187e-39</span></span>
<span id="cb353-15"><a href="peak2genelinkage-with-archr.html#cb353-15" tabindex="-1"></a><span class="do">##       [4]     chr1       865000-975000      * |  0.523227 1.30350e-34</span></span>
<span id="cb353-16"><a href="peak2genelinkage-with-archr.html#cb353-16" tabindex="-1"></a><span class="do">##       [5]     chr1              895000      * |  0.474614 7.69115e-28</span></span>
<span id="cb353-17"><a href="peak2genelinkage-with-archr.html#cb353-17" tabindex="-1"></a><span class="do">##       ...      ...                 ...    ... .       ...         ...</span></span>
<span id="cb353-18"><a href="peak2genelinkage-with-archr.html#cb353-18" tabindex="-1"></a><span class="do">##   [33721]     chrX 153825000-153995000      * |  0.474616 7.68618e-28</span></span>
<span id="cb353-19"><a href="peak2genelinkage-with-archr.html#cb353-19" tabindex="-1"></a><span class="do">##   [33722]     chrX 153835000-153975000      * |  0.464330 1.52289e-26</span></span>
<span id="cb353-20"><a href="peak2genelinkage-with-archr.html#cb353-20" tabindex="-1"></a><span class="do">##   [33723]     chrX 153835000-153995000      * |  0.521417 2.44218e-34</span></span>
<span id="cb353-21"><a href="peak2genelinkage-with-archr.html#cb353-21" tabindex="-1"></a><span class="do">##   [33724]     chrX 153985000-153995000      * |  0.599431 9.35259e-48</span></span>
<span id="cb353-22"><a href="peak2genelinkage-with-archr.html#cb353-22" tabindex="-1"></a><span class="do">##   [33725]     chrX 153985000-154035000      * |  0.511117 8.04962e-33</span></span>
<span id="cb353-23"><a href="peak2genelinkage-with-archr.html#cb353-23" tabindex="-1"></a><span class="do">##   -------</span></span>
<span id="cb353-24"><a href="peak2genelinkage-with-archr.html#cb353-24" tabindex="-1"></a><span class="do">##   seqinfo: 23 sequences from an unspecified genome; no seqlengths</span></span></code></pre></div>
<p>You may find that you would like to use a statistical approach to evaluate peak-to-gene associations. Such an approach was developed in <a href="https://doi.org/10.1016/j.molcel.2021.10.013">Regner et al. 2021</a> and involves using permutation testing to empirically derive an FDR. This approach is quite computationally expensive but has been implemented as an option in <code>addPeak2GeneLinks()</code> via the <code>addPermutedPval</code> parameter. For more information on this approach, see <a href="https://github.com/GreenleafLab/ArchR/discussions/1149">this discussion post</a>.</p>
<div id="plotting-browser-tracks-with-peak-to-gene-links" class="section level3 hasAnchor" number="17.3.1">
<h3><span class="header-section-number">17.3.1</span> Plotting browser tracks with peak-to-gene links<a href="peak2genelinkage-with-archr.html#plotting-browser-tracks-with-peak-to-gene-links" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>To plot these peak-to-gene links as a browser track, we use the same workflow shown for co-accessibility in the previous section. Here we use the <code>plotBrowserTrack()</code> function</p>
<div class="sourceCode" id="cb354"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb354-1"><a href="peak2genelinkage-with-archr.html#cb354-1" tabindex="-1"></a>markerGenes  <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb354-2"><a href="peak2genelinkage-with-archr.html#cb354-2" tabindex="-1"></a>    <span class="st">&quot;CD34&quot;</span>, <span class="co">#Early Progenitor</span></span>
<span id="cb354-3"><a href="peak2genelinkage-with-archr.html#cb354-3" tabindex="-1"></a>    <span class="st">&quot;GATA1&quot;</span>, <span class="co">#Erythroid</span></span>
<span id="cb354-4"><a href="peak2genelinkage-with-archr.html#cb354-4" tabindex="-1"></a>    <span class="st">&quot;PAX5&quot;</span>, <span class="st">&quot;MS4A1&quot;</span>, <span class="co">#B-Cell Trajectory</span></span>
<span id="cb354-5"><a href="peak2genelinkage-with-archr.html#cb354-5" tabindex="-1"></a>    <span class="st">&quot;CD14&quot;</span>, <span class="co">#Monocytes</span></span>
<span id="cb354-6"><a href="peak2genelinkage-with-archr.html#cb354-6" tabindex="-1"></a>    <span class="st">&quot;CD3D&quot;</span>, <span class="st">&quot;CD8A&quot;</span>, <span class="st">&quot;TBX21&quot;</span>, <span class="st">&quot;IL7R&quot;</span> <span class="co">#TCells</span></span>
<span id="cb354-7"><a href="peak2genelinkage-with-archr.html#cb354-7" tabindex="-1"></a>  )</span>
<span id="cb354-8"><a href="peak2genelinkage-with-archr.html#cb354-8" tabindex="-1"></a></span>
<span id="cb354-9"><a href="peak2genelinkage-with-archr.html#cb354-9" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">plotBrowserTrack</span>(</span>
<span id="cb354-10"><a href="peak2genelinkage-with-archr.html#cb354-10" tabindex="-1"></a>    <span class="at">ArchRProj =</span> projHeme5, </span>
<span id="cb354-11"><a href="peak2genelinkage-with-archr.html#cb354-11" tabindex="-1"></a>    <span class="at">groupBy =</span> <span class="st">&quot;Clusters2&quot;</span>, </span>
<span id="cb354-12"><a href="peak2genelinkage-with-archr.html#cb354-12" tabindex="-1"></a>    <span class="at">geneSymbol =</span> markerGenes, </span>
<span id="cb354-13"><a href="peak2genelinkage-with-archr.html#cb354-13" tabindex="-1"></a>    <span class="at">upstream =</span> <span class="dv">50000</span>,</span>
<span id="cb354-14"><a href="peak2genelinkage-with-archr.html#cb354-14" tabindex="-1"></a>    <span class="at">downstream =</span> <span class="dv">50000</span>,</span>
<span id="cb354-15"><a href="peak2genelinkage-with-archr.html#cb354-15" tabindex="-1"></a>    <span class="at">loops =</span> <span class="fu">getPeak2GeneLinks</span>(projHeme5)</span>
<span id="cb354-16"><a href="peak2genelinkage-with-archr.html#cb354-16" tabindex="-1"></a>)</span>
<span id="cb354-17"><a href="peak2genelinkage-with-archr.html#cb354-17" tabindex="-1"></a><span class="do">## ArchR logging to : ArchRLogs/ArchR-plotBrowserTrack-925b468a1-Date-2025-02-06_Time-02-49-03.297189.log</span></span>
<span id="cb354-18"><a href="peak2genelinkage-with-archr.html#cb354-18" tabindex="-1"></a><span class="do">## If there is an issue, please report to github with logFile!</span></span>
<span id="cb354-19"><a href="peak2genelinkage-with-archr.html#cb354-19" tabindex="-1"></a><span class="do">## 2025-02-06 02:49:01.985563 : Validating Region, 0.022 mins elapsed.</span></span>
<span id="cb354-20"><a href="peak2genelinkage-with-archr.html#cb354-20" tabindex="-1"></a><span class="do">## GRanges object with 9 ranges and 2 metadata columns:</span></span>
<span id="cb354-21"><a href="peak2genelinkage-with-archr.html#cb354-21" tabindex="-1"></a><span class="do">##       seqnames              ranges strand |     gene_id      symbol</span></span>
<span id="cb354-22"><a href="peak2genelinkage-with-archr.html#cb354-22" tabindex="-1"></a><span class="do">##          &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt;</span></span>
<span id="cb354-23"><a href="peak2genelinkage-with-archr.html#cb354-23" tabindex="-1"></a><span class="do">##   [1]     chr1 208059883-208084683      - |         947        CD34</span></span>
<span id="cb354-24"><a href="peak2genelinkage-with-archr.html#cb354-24" tabindex="-1"></a><span class="do">##   [2]     chrX   48644982-48652717      + |        2623       GATA1</span></span>
<span id="cb354-25"><a href="peak2genelinkage-with-archr.html#cb354-25" tabindex="-1"></a><span class="do">##   [3]     chr9   36838531-37034476      - |        5079        PAX5</span></span>
<span id="cb354-26"><a href="peak2genelinkage-with-archr.html#cb354-26" tabindex="-1"></a><span class="do">##   [4]    chr11   60223282-60238225      + |         931       MS4A1</span></span>
<span id="cb354-27"><a href="peak2genelinkage-with-archr.html#cb354-27" tabindex="-1"></a><span class="do">##   [5]     chr5 140011313-140013286      - |         929        CD14</span></span>
<span id="cb354-28"><a href="peak2genelinkage-with-archr.html#cb354-28" tabindex="-1"></a><span class="do">##   [6]    chr11 118209789-118213459      - |         915        CD3D</span></span>
<span id="cb354-29"><a href="peak2genelinkage-with-archr.html#cb354-29" tabindex="-1"></a><span class="do">##   [7]     chr2   87011728-87035519      - |         925        CD8A</span></span>
<span id="cb354-30"><a href="peak2genelinkage-with-archr.html#cb354-30" tabindex="-1"></a><span class="do">##   [8]    chr17   45810610-45823485      + |       30009       TBX21</span></span>
<span id="cb354-31"><a href="peak2genelinkage-with-archr.html#cb354-31" tabindex="-1"></a><span class="do">##   [9]     chr5   35856977-35879705      + |        3575        IL7R</span></span>
<span id="cb354-32"><a href="peak2genelinkage-with-archr.html#cb354-32" tabindex="-1"></a><span class="do">##   -------</span></span>
<span id="cb354-33"><a href="peak2genelinkage-with-archr.html#cb354-33" tabindex="-1"></a><span class="do">##   seqinfo: 24 sequences from hg19 genome</span></span>
<span id="cb354-34"><a href="peak2genelinkage-with-archr.html#cb354-34" tabindex="-1"></a><span class="do">## 2025-02-06 02:49:02.075656 : Adding Bulk Tracks (1 of 9), 0.02 mins elapsed.</span></span>
<span id="cb354-35"><a href="peak2genelinkage-with-archr.html#cb354-35" tabindex="-1"></a><span class="do">## 2025-02-06 02:49:03.994375 : Adding Feature Tracks (1 of 9), 0.012 mins elapsed.</span></span>
<span id="cb354-36"><a href="peak2genelinkage-with-archr.html#cb354-36" tabindex="-1"></a><span class="do">## 2025-02-06 02:49:04.076611 : Adding Loop Tracks (1 of 9), 0.013 mins elapsed.</span></span>
<span id="cb354-37"><a href="peak2genelinkage-with-archr.html#cb354-37" tabindex="-1"></a><span class="do">## 2025-02-06 02:49:04.211859 : Adding Gene Tracks (1 of 9), 0.015 mins elapsed.</span></span>
<span id="cb354-38"><a href="peak2genelinkage-with-archr.html#cb354-38" tabindex="-1"></a><span class="do">## 2025-02-06 02:49:04.420231 : Plotting, 0.019 mins elapsed.</span></span>
<span id="cb354-39"><a href="peak2genelinkage-with-archr.html#cb354-39" tabindex="-1"></a><span class="do">## 2025-02-06 02:49:05.577812 : Adding Bulk Tracks (2 of 9), 0.038 mins elapsed.</span></span>
<span id="cb354-40"><a href="peak2genelinkage-with-archr.html#cb354-40" tabindex="-1"></a><span class="do">## 2025-02-06 02:49:07.455849 : Adding Feature Tracks (2 of 9), 0.069 mins elapsed.</span></span>
<span id="cb354-41"><a href="peak2genelinkage-with-archr.html#cb354-41" tabindex="-1"></a><span class="do">## 2025-02-06 02:49:07.546801 : Adding Loop Tracks (2 of 9), 0.071 mins elapsed.</span></span>
<span id="cb354-42"><a href="peak2genelinkage-with-archr.html#cb354-42" tabindex="-1"></a><span class="do">## 2025-02-06 02:49:07.6892 : Adding Gene Tracks (2 of 9), 0.073 mins elapsed.</span></span>
<span id="cb354-43"><a href="peak2genelinkage-with-archr.html#cb354-43" tabindex="-1"></a><span class="do">## 2025-02-06 02:49:07.906346 : Plotting, 0.077 mins elapsed.</span></span>
<span id="cb354-44"><a href="peak2genelinkage-with-archr.html#cb354-44" tabindex="-1"></a><span class="do">## 2025-02-06 02:49:08.691367 : Adding Bulk Tracks (3 of 9), 0.09 mins elapsed.</span></span>
<span id="cb354-45"><a href="peak2genelinkage-with-archr.html#cb354-45" tabindex="-1"></a><span class="do">## 2025-02-06 02:49:10.577964 : Adding Feature Tracks (3 of 9), 0.121 mins elapsed.</span></span>
<span id="cb354-46"><a href="peak2genelinkage-with-archr.html#cb354-46" tabindex="-1"></a><span class="do">## 2025-02-06 02:49:10.672172 : Adding Loop Tracks (3 of 9), 0.123 mins elapsed.</span></span>
<span id="cb354-47"><a href="peak2genelinkage-with-archr.html#cb354-47" tabindex="-1"></a><span class="do">## 2025-02-06 02:49:11.061905 : Adding Gene Tracks (3 of 9), 0.129 mins elapsed.</span></span>
<span id="cb354-48"><a href="peak2genelinkage-with-archr.html#cb354-48" tabindex="-1"></a><span class="do">## 2025-02-06 02:49:11.286861 : Plotting, 0.133 mins elapsed.</span></span>
<span id="cb354-49"><a href="peak2genelinkage-with-archr.html#cb354-49" tabindex="-1"></a><span class="do">## 2025-02-06 02:49:12.250556 : Adding Bulk Tracks (4 of 9), 0.149 mins elapsed.</span></span>
<span id="cb354-50"><a href="peak2genelinkage-with-archr.html#cb354-50" tabindex="-1"></a><span class="do">## 2025-02-06 02:49:14.237169 : Adding Feature Tracks (4 of 9), 0.182 mins elapsed.</span></span>
<span id="cb354-51"><a href="peak2genelinkage-with-archr.html#cb354-51" tabindex="-1"></a><span class="do">## 2025-02-06 02:49:14.323251 : Adding Loop Tracks (4 of 9), 0.184 mins elapsed.</span></span>
<span id="cb354-52"><a href="peak2genelinkage-with-archr.html#cb354-52" tabindex="-1"></a><span class="do">## 2025-02-06 02:49:14.475799 : Adding Gene Tracks (4 of 9), 0.186 mins elapsed.</span></span>
<span id="cb354-53"><a href="peak2genelinkage-with-archr.html#cb354-53" tabindex="-1"></a><span class="do">## 2025-02-06 02:49:14.680334 : Plotting, 0.19 mins elapsed.</span></span>
<span id="cb354-54"><a href="peak2genelinkage-with-archr.html#cb354-54" tabindex="-1"></a><span class="do">## 2025-02-06 02:49:15.566444 : Adding Bulk Tracks (5 of 9), 0.205 mins elapsed.</span></span>
<span id="cb354-55"><a href="peak2genelinkage-with-archr.html#cb354-55" tabindex="-1"></a><span class="do">## 2025-02-06 02:49:17.509581 : Adding Feature Tracks (5 of 9), 0.237 mins elapsed.</span></span>
<span id="cb354-56"><a href="peak2genelinkage-with-archr.html#cb354-56" tabindex="-1"></a><span class="do">## 2025-02-06 02:49:17.60025 : Adding Loop Tracks (5 of 9), 0.238 mins elapsed.</span></span>
<span id="cb354-57"><a href="peak2genelinkage-with-archr.html#cb354-57" tabindex="-1"></a><span class="do">## 2025-02-06 02:49:17.803788 : Adding Gene Tracks (5 of 9), 0.242 mins elapsed.</span></span>
<span id="cb354-58"><a href="peak2genelinkage-with-archr.html#cb354-58" tabindex="-1"></a><span class="do">## 2025-02-06 02:49:18.010258 : Plotting, 0.245 mins elapsed.</span></span>
<span id="cb354-59"><a href="peak2genelinkage-with-archr.html#cb354-59" tabindex="-1"></a><span class="do">## 2025-02-06 02:49:19.063965 : Adding Bulk Tracks (6 of 9), 0.263 mins elapsed.</span></span>
<span id="cb354-60"><a href="peak2genelinkage-with-archr.html#cb354-60" tabindex="-1"></a><span class="do">## 2025-02-06 02:49:21.100406 : Adding Feature Tracks (6 of 9), 0.297 mins elapsed.</span></span>
<span id="cb354-61"><a href="peak2genelinkage-with-archr.html#cb354-61" tabindex="-1"></a><span class="do">## 2025-02-06 02:49:21.186736 : Adding Loop Tracks (6 of 9), 0.298 mins elapsed.</span></span>
<span id="cb354-62"><a href="peak2genelinkage-with-archr.html#cb354-62" tabindex="-1"></a><span class="do">## 2025-02-06 02:49:21.669676 : Adding Gene Tracks (6 of 9), 0.306 mins elapsed.</span></span>
<span id="cb354-63"><a href="peak2genelinkage-with-archr.html#cb354-63" tabindex="-1"></a><span class="do">## 2025-02-06 02:49:21.871229 : Plotting, 0.31 mins elapsed.</span></span>
<span id="cb354-64"><a href="peak2genelinkage-with-archr.html#cb354-64" tabindex="-1"></a><span class="do">## 2025-02-06 02:49:22.830714 : Adding Bulk Tracks (7 of 9), 0.326 mins elapsed.</span></span>
<span id="cb354-65"><a href="peak2genelinkage-with-archr.html#cb354-65" tabindex="-1"></a><span class="do">## 2025-02-06 02:49:24.83812 : Adding Feature Tracks (7 of 9), 0.359 mins elapsed.</span></span>
<span id="cb354-66"><a href="peak2genelinkage-with-archr.html#cb354-66" tabindex="-1"></a><span class="do">## 2025-02-06 02:49:24.927555 : Adding Loop Tracks (7 of 9), 0.361 mins elapsed.</span></span>
<span id="cb354-67"><a href="peak2genelinkage-with-archr.html#cb354-67" tabindex="-1"></a><span class="do">## 2025-02-06 02:49:25.172896 : Adding Gene Tracks (7 of 9), 0.365 mins elapsed.</span></span>
<span id="cb354-68"><a href="peak2genelinkage-with-archr.html#cb354-68" tabindex="-1"></a><span class="do">## 2025-02-06 02:49:25.38227 : Plotting, 0.368 mins elapsed.</span></span>
<span id="cb354-69"><a href="peak2genelinkage-with-archr.html#cb354-69" tabindex="-1"></a><span class="do">## 2025-02-06 02:49:26.249319 : Adding Bulk Tracks (8 of 9), 0.383 mins elapsed.</span></span>
<span id="cb354-70"><a href="peak2genelinkage-with-archr.html#cb354-70" tabindex="-1"></a><span class="do">## 2025-02-06 02:49:28.328631 : Adding Feature Tracks (8 of 9), 0.417 mins elapsed.</span></span>
<span id="cb354-71"><a href="peak2genelinkage-with-archr.html#cb354-71" tabindex="-1"></a><span class="do">## 2025-02-06 02:49:28.411748 : Adding Loop Tracks (8 of 9), 0.419 mins elapsed.</span></span>
<span id="cb354-72"><a href="peak2genelinkage-with-archr.html#cb354-72" tabindex="-1"></a><span class="do">## 2025-02-06 02:49:28.699722 : Adding Gene Tracks (8 of 9), 0.423 mins elapsed.</span></span>
<span id="cb354-73"><a href="peak2genelinkage-with-archr.html#cb354-73" tabindex="-1"></a><span class="do">## 2025-02-06 02:49:28.925004 : Plotting, 0.427 mins elapsed.</span></span>
<span id="cb354-74"><a href="peak2genelinkage-with-archr.html#cb354-74" tabindex="-1"></a><span class="do">## 2025-02-06 02:49:29.956741 : Adding Bulk Tracks (9 of 9), 0.444 mins elapsed.</span></span>
<span id="cb354-75"><a href="peak2genelinkage-with-archr.html#cb354-75" tabindex="-1"></a><span class="do">## 2025-02-06 02:49:30.079082 : Adding Feature Tracks (9 of 9), 0.446 mins elapsed.</span></span>
<span id="cb354-76"><a href="peak2genelinkage-with-archr.html#cb354-76" tabindex="-1"></a><span class="do">## 2025-02-06 02:49:30.16675 : Adding Loop Tracks (9 of 9), 0.448 mins elapsed.</span></span>
<span id="cb354-77"><a href="peak2genelinkage-with-archr.html#cb354-77" tabindex="-1"></a><span class="do">## 2025-02-06 02:49:30.375868 : Adding Gene Tracks (9 of 9), 0.451 mins elapsed.</span></span>
<span id="cb354-78"><a href="peak2genelinkage-with-archr.html#cb354-78" tabindex="-1"></a><span class="do">## 2025-02-06 02:49:30.583511 : Plotting, 0.455 mins elapsed.</span></span>
<span id="cb354-79"><a href="peak2genelinkage-with-archr.html#cb354-79" tabindex="-1"></a><span class="do">## ArchR logging successful to : ArchRLogs/ArchR-plotBrowserTrack-925b468a1-Date-2025-02-06_Time-02-49-03.297189.log</span></span></code></pre></div>
<p>To plot our browser track we use the <code>grid.draw</code> function and select a specific marker gene by name using the <code>$</code> accessor.</p>
<div class="sourceCode" id="cb355"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb355-1"><a href="peak2genelinkage-with-archr.html#cb355-1" tabindex="-1"></a>grid<span class="sc">::</span><span class="fu">grid.newpage</span>()</span>
<span id="cb355-2"><a href="peak2genelinkage-with-archr.html#cb355-2" tabindex="-1"></a>grid<span class="sc">::</span><span class="fu">grid.draw</span>(p<span class="sc">$</span>CD14)</span></code></pre></div>
<p><img src="ArchR_files/figure-html/unnamed-chunk-345-1.svg" width="672" /></p>
<p>To save an editable vectorized version of this plot, we use <code>plotPDF()</code>.</p>
<div class="sourceCode" id="cb356"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb356-1"><a href="peak2genelinkage-with-archr.html#cb356-1" tabindex="-1"></a><span class="fu">plotPDF</span>(<span class="at">plotList =</span> p, </span>
<span id="cb356-2"><a href="peak2genelinkage-with-archr.html#cb356-2" tabindex="-1"></a>    <span class="at">name =</span> <span class="st">&quot;Plot-Tracks-Marker-Genes-with-Peak2GeneLinks.pdf&quot;</span>, </span>
<span id="cb356-3"><a href="peak2genelinkage-with-archr.html#cb356-3" tabindex="-1"></a>    <span class="at">ArchRProj =</span> projHeme5, </span>
<span id="cb356-4"><a href="peak2genelinkage-with-archr.html#cb356-4" tabindex="-1"></a>    <span class="at">addDOC =</span> <span class="cn">FALSE</span>, <span class="at">width =</span> <span class="dv">5</span>, <span class="at">height =</span> <span class="dv">5</span>)</span>
<span id="cb356-5"><a href="peak2genelinkage-with-archr.html#cb356-5" tabindex="-1"></a><span class="do">## Plotting Gtable!</span></span>
<span id="cb356-6"><a href="peak2genelinkage-with-archr.html#cb356-6" tabindex="-1"></a><span class="do">## NULL</span></span>
<span id="cb356-7"><a href="peak2genelinkage-with-archr.html#cb356-7" tabindex="-1"></a><span class="do">## Plotting Gtable!</span></span>
<span id="cb356-8"><a href="peak2genelinkage-with-archr.html#cb356-8" tabindex="-1"></a><span class="do">## NULL</span></span>
<span id="cb356-9"><a href="peak2genelinkage-with-archr.html#cb356-9" tabindex="-1"></a><span class="do">## Plotting Gtable!</span></span>
<span id="cb356-10"><a href="peak2genelinkage-with-archr.html#cb356-10" tabindex="-1"></a><span class="do">## NULL</span></span>
<span id="cb356-11"><a href="peak2genelinkage-with-archr.html#cb356-11" tabindex="-1"></a><span class="do">## Plotting Gtable!</span></span>
<span id="cb356-12"><a href="peak2genelinkage-with-archr.html#cb356-12" tabindex="-1"></a><span class="do">## NULL</span></span>
<span id="cb356-13"><a href="peak2genelinkage-with-archr.html#cb356-13" tabindex="-1"></a><span class="do">## Plotting Gtable!</span></span>
<span id="cb356-14"><a href="peak2genelinkage-with-archr.html#cb356-14" tabindex="-1"></a><span class="do">## NULL</span></span>
<span id="cb356-15"><a href="peak2genelinkage-with-archr.html#cb356-15" tabindex="-1"></a><span class="do">## Plotting Gtable!</span></span>
<span id="cb356-16"><a href="peak2genelinkage-with-archr.html#cb356-16" tabindex="-1"></a><span class="do">## NULL</span></span>
<span id="cb356-17"><a href="peak2genelinkage-with-archr.html#cb356-17" tabindex="-1"></a><span class="do">## Plotting Gtable!</span></span>
<span id="cb356-18"><a href="peak2genelinkage-with-archr.html#cb356-18" tabindex="-1"></a><span class="do">## NULL</span></span>
<span id="cb356-19"><a href="peak2genelinkage-with-archr.html#cb356-19" tabindex="-1"></a><span class="do">## Plotting Gtable!</span></span>
<span id="cb356-20"><a href="peak2genelinkage-with-archr.html#cb356-20" tabindex="-1"></a><span class="do">## NULL</span></span>
<span id="cb356-21"><a href="peak2genelinkage-with-archr.html#cb356-21" tabindex="-1"></a><span class="do">## Plotting Gtable!</span></span>
<span id="cb356-22"><a href="peak2genelinkage-with-archr.html#cb356-22" tabindex="-1"></a><span class="do">## NULL</span></span></code></pre></div>
</div>
<div id="plotting-a-heatmap-of-peak-to-gene-links" class="section level3 hasAnchor" number="17.3.2">
<h3><span class="header-section-number">17.3.2</span> Plotting a heatmap of peak-to-gene links<a href="peak2genelinkage-with-archr.html#plotting-a-heatmap-of-peak-to-gene-links" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>To visualize the correspondence of all of our peak-to-gene links, we can plot a peak-to-gene heatmap which contains two side-by-side heatmaps, one for our scATAC-seq data and one for our scRNA-seq data. To do this, we use the <code>plotPeak2GeneHeatmap()</code></p>
<div class="sourceCode" id="cb357"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb357-1"><a href="peak2genelinkage-with-archr.html#cb357-1" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">plotPeak2GeneHeatmap</span>(<span class="at">ArchRProj =</span> projHeme5, <span class="at">groupBy =</span> <span class="st">&quot;Clusters2&quot;</span>)</span>
<span id="cb357-2"><a href="peak2genelinkage-with-archr.html#cb357-2" tabindex="-1"></a><span class="do">## ArchR logging to : ArchRLogs/ArchR-plotPeak2GeneHeatmap-97c367426-Date-2025-02-06_Time-02-49-36.1348.log</span></span>
<span id="cb357-3"><a href="peak2genelinkage-with-archr.html#cb357-3" tabindex="-1"></a><span class="do">## If there is an issue, please report to github with logFile!</span></span>
<span id="cb357-4"><a href="peak2genelinkage-with-archr.html#cb357-4" tabindex="-1"></a><span class="do">## 2025-02-06 02:49:51.026259 : Determining KNN Groups!, 0.248 mins elapsed.</span></span>
<span id="cb357-5"><a href="peak2genelinkage-with-archr.html#cb357-5" tabindex="-1"></a><span class="do">## 2025-02-06 02:49:53.315845 : Ordering Peak2Gene Links!, 0.286 mins elapsed.</span></span>
<span id="cb357-6"><a href="peak2genelinkage-with-archr.html#cb357-6" tabindex="-1"></a><span class="do">## 2025-02-06 02:49:59.295731 : Constructing ATAC Heatmap!, 0.386 mins elapsed.</span></span>
<span id="cb357-7"><a href="peak2genelinkage-with-archr.html#cb357-7" tabindex="-1"></a><span class="do">## Adding Annotations..</span></span>
<span id="cb357-8"><a href="peak2genelinkage-with-archr.html#cb357-8" tabindex="-1"></a><span class="do">## Preparing Main Heatmap..</span></span>
<span id="cb357-9"><a href="peak2genelinkage-with-archr.html#cb357-9" tabindex="-1"></a><span class="do">## &#39;magick&#39; package is suggested to install to give better rasterization.</span></span>
<span id="cb357-10"><a href="peak2genelinkage-with-archr.html#cb357-10" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb357-11"><a href="peak2genelinkage-with-archr.html#cb357-11" tabindex="-1"></a><span class="do">## Set `ht_opt$message = FALSE` to turn off this message.</span></span>
<span id="cb357-12"><a href="peak2genelinkage-with-archr.html#cb357-12" tabindex="-1"></a><span class="do">## 2025-02-06 02:49:59.713415 : Constructing RNA Heatmap!, 0.393 mins elapsed.</span></span>
<span id="cb357-13"><a href="peak2genelinkage-with-archr.html#cb357-13" tabindex="-1"></a><span class="do">## Adding Annotations..</span></span>
<span id="cb357-14"><a href="peak2genelinkage-with-archr.html#cb357-14" tabindex="-1"></a><span class="do">## Preparing Main Heatmap..</span></span>
<span id="cb357-15"><a href="peak2genelinkage-with-archr.html#cb357-15" tabindex="-1"></a><span class="do">## &#39;magick&#39; package is suggested to install to give better rasterization.</span></span>
<span id="cb357-16"><a href="peak2genelinkage-with-archr.html#cb357-16" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb357-17"><a href="peak2genelinkage-with-archr.html#cb357-17" tabindex="-1"></a><span class="do">## Set `ht_opt$message = FALSE` to turn off this message.</span></span>
<span id="cb357-18"><a href="peak2genelinkage-with-archr.html#cb357-18" tabindex="-1"></a><span class="do">## ArchR logging successful to : ArchRLogs/ArchR-plotPeak2GeneHeatmap-97c367426-Date-2025-02-06_Time-02-49-36.1348.log</span></span></code></pre></div>
<p>The heatmap rows are clustered using k-means clustering based on the value passed to the parameter <code>k</code>, which defaults to 25 as shown below.</p>
<div class="sourceCode" id="cb358"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb358-1"><a href="peak2genelinkage-with-archr.html#cb358-1" tabindex="-1"></a>p</span></code></pre></div>
<p><img src="ArchR_files/figure-html/unnamed-chunk-348-1.svg" width="672" /></p>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="co-accessibility-with-archr.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="identification-of-positive-tf-regulators.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": null
},
"fontsettings": null,
"edit": null,
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "none"
},
"info": false
});
});
</script>

</body>

</html>
