<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>20.1 Projecting bulk ATAC-seq data | ArchR: Robust and scaleable analysis of single-cell chromatin accessibility data.</title>
  <meta name="description" content="A guide to ArchR" />
  <meta name="generator" content="bookdown 0.41 and GitBook 2.6.7" />

  <meta property="og:title" content="20.1 Projecting bulk ATAC-seq data | ArchR: Robust and scaleable analysis of single-cell chromatin accessibility data." />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="A guide to ArchR" />
  <meta name="github-repo" content="GreenleafLab/ArchR" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="20.1 Projecting bulk ATAC-seq data | ArchR: Robust and scaleable analysis of single-cell chromatin accessibility data." />
  
  <meta name="twitter:description" content="A guide to ArchR" />
  

<meta name="author" content="Jeffrey Granja and Ryan Corces" />


<meta name="date" content="2025-01-23" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="images/favicon.ico" type="image/x-icon" />
<link rel="prev" href="integration-with-bulk-atac-seq.html"/>
<link rel="next" href="multiomic-data-analysis-in-archr.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<link href="libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="../index.html">Back to ArchR Home</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><img src="images/ArchR_Logo_Integrated.png" alt="" width="200"></a></li>
<li class="chapter" data-level="1" data-path="debugging-and-troubleshooting-in-archr.html"><a href="debugging-and-troubleshooting-in-archr.html"><i class="fa fa-check"></i><b>1</b> Debugging and Troubleshooting in ArchR</a></li>
<li class="chapter" data-level="2" data-path="manage-archrs-dependencies.html"><a href="manage-archrs-dependencies.html"><i class="fa fa-check"></i><b>2</b> Manage ArchR’s Dependencies</a>
<ul>
<li class="chapter" data-level="2.1" data-path="using-renv-to-manage-dependencies.html"><a href="using-renv-to-manage-dependencies.html"><i class="fa fa-check"></i><b>2.1</b> Using renv to manage dependencies</a></li>
<li class="chapter" data-level="2.2" data-path="other-environment-managers.html"><a href="other-environment-managers.html"><i class="fa fa-check"></i><b>2.2</b> Other environment managers</a></li>
<li class="chapter" data-level="2.3" data-path="using-docker-for-pre-compiled-environment-setup.html"><a href="using-docker-for-pre-compiled-environment-setup.html"><i class="fa fa-check"></i><b>2.3</b> Using Docker for pre-compiled environment setup</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="getting-started-with-archr.html"><a href="getting-started-with-archr.html"><i class="fa fa-check"></i><b>3</b> Getting Started with ArchR</a>
<ul>
<li class="chapter" data-level="3.1" data-path="a-brief-primer-on-atac-seq-terminology.html"><a href="a-brief-primer-on-atac-seq-terminology.html"><i class="fa fa-check"></i><b>3.1</b> A Brief Primer on ATAC-seq Terminology</a></li>
<li class="chapter" data-level="3.2" data-path="why-use-archr.html"><a href="why-use-archr.html"><i class="fa fa-check"></i><b>3.2</b> Why use ArchR?</a></li>
<li class="chapter" data-level="3.3" data-path="what-is-an-arrow-file-archrproject.html"><a href="what-is-an-arrow-file-archrproject.html"><i class="fa fa-check"></i><b>3.3</b> What is an Arrow file / <code>ArchRProject</code>?</a></li>
<li class="chapter" data-level="3.4" data-path="input-file-types-in-archr.html"><a href="input-file-types-in-archr.html"><i class="fa fa-check"></i><b>3.4</b> Input File Types in ArchR</a></li>
<li class="chapter" data-level="3.5" data-path="getting-set-up.html"><a href="getting-set-up.html"><i class="fa fa-check"></i><b>3.5</b> Getting Set Up</a></li>
<li class="chapter" data-level="3.6" data-path="creating-arrow-files.html"><a href="creating-arrow-files.html"><i class="fa fa-check"></i><b>3.6</b> Creating Arrow Files</a></li>
<li class="chapter" data-level="3.7" data-path="per-cell-quality-control.html"><a href="per-cell-quality-control.html"><i class="fa fa-check"></i><b>3.7</b> Per-cell Quality Control</a></li>
<li class="chapter" data-level="3.8" data-path="using-bam-files-for-arrow-file-creation.html"><a href="using-bam-files-for-arrow-file-creation.html"><i class="fa fa-check"></i><b>3.8</b> Using BAM files for Arrow File creation</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="doublet-inference-with-archr.html"><a href="doublet-inference-with-archr.html"><i class="fa fa-check"></i><b>4</b> Doublet Inference with ArchR</a>
<ul>
<li class="chapter" data-level="4.1" data-path="how-does-doublet-identification-work-in-archr.html"><a href="how-does-doublet-identification-work-in-archr.html"><i class="fa fa-check"></i><b>4.1</b> How does doublet identification work in ArchR?</a></li>
<li class="chapter" data-level="4.2" data-path="inferring-scatac-seq-doublets-with-archr.html"><a href="inferring-scatac-seq-doublets-with-archr.html"><i class="fa fa-check"></i><b>4.2</b> Inferring scATAC-seq Doublets with ArchR</a></li>
<li class="chapter" data-level="4.3" data-path="using-demuxlet-with-archr.html"><a href="using-demuxlet-with-archr.html"><i class="fa fa-check"></i><b>4.3</b> Using <em>demuxlet</em> with ArchR</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="creating-an-archrproject.html"><a href="creating-an-archrproject.html"><i class="fa fa-check"></i><b>5</b> Creating an ArchRProject</a>
<ul>
<li class="chapter" data-level="5.1" data-path="creating-an-archrproject-1.html"><a href="creating-an-archrproject-1.html"><i class="fa fa-check"></i><b>5.1</b> Creating An ArchRProject</a></li>
<li class="chapter" data-level="5.2" data-path="manipulating-an-archrproject.html"><a href="manipulating-an-archrproject.html"><i class="fa fa-check"></i><b>5.2</b> Manipulating An ArchRProject</a></li>
<li class="chapter" data-level="5.3" data-path="plotting-sample-statistics-from-an-archrproject.html"><a href="plotting-sample-statistics-from-an-archrproject.html"><i class="fa fa-check"></i><b>5.3</b> Plotting Sample Statistics from an ArchRProject</a></li>
<li class="chapter" data-level="5.4" data-path="plotting-sample-fragment-size-distribution-and-tss-enrichment-profiles..html"><a href="plotting-sample-fragment-size-distribution-and-tss-enrichment-profiles..html"><i class="fa fa-check"></i><b>5.4</b> Plotting Sample Fragment Size Distribution and TSS Enrichment Profiles.</a></li>
<li class="chapter" data-level="5.5" data-path="saving-and-loading-an-archrproject.html"><a href="saving-and-loading-an-archrproject.html"><i class="fa fa-check"></i><b>5.5</b> Saving and Loading an <code>ArchRProject</code></a></li>
<li class="chapter" data-level="5.6" data-path="filtering-doublets-from-an-archrproject.html"><a href="filtering-doublets-from-an-archrproject.html"><i class="fa fa-check"></i><b>5.6</b> Filtering Doublets from an ArchRProject</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="dimensionality-reduction-with-archr.html"><a href="dimensionality-reduction-with-archr.html"><i class="fa fa-check"></i><b>6</b> Dimensionality Reduction with ArchR</a>
<ul>
<li class="chapter" data-level="6.1" data-path="archrs-lsi-implementation.html"><a href="archrs-lsi-implementation.html"><i class="fa fa-check"></i><b>6.1</b> ArchR’s LSI Implementation</a></li>
<li class="chapter" data-level="6.2" data-path="iterative-latent-semantic-indexing-lsi.html"><a href="iterative-latent-semantic-indexing-lsi.html"><i class="fa fa-check"></i><b>6.2</b> Iterative Latent Semantic Indexing (LSI)</a></li>
<li class="chapter" data-level="6.3" data-path="estimated-lsi.html"><a href="estimated-lsi.html"><i class="fa fa-check"></i><b>6.3</b> Estimated LSI</a></li>
<li class="chapter" data-level="6.4" data-path="batch-effect-correction-wtih-harmony.html"><a href="batch-effect-correction-wtih-harmony.html"><i class="fa fa-check"></i><b>6.4</b> Batch Effect Correction wtih Harmony</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="clustering-with-archr.html"><a href="clustering-with-archr.html"><i class="fa fa-check"></i><b>7</b> Clustering with ArchR</a>
<ul>
<li class="chapter" data-level="7.1" data-path="clustering-using-seurats-findclusters-function.html"><a href="clustering-using-seurats-findclusters-function.html"><i class="fa fa-check"></i><b>7.1</b> Clustering using Seurat’s <code>FindClusters()</code> function</a></li>
<li class="chapter" data-level="7.2" data-path="clustering-using-scran.html"><a href="clustering-using-scran.html"><i class="fa fa-check"></i><b>7.2</b> Clustering using <code>scran</code></a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="single-cell-embeddings.html"><a href="single-cell-embeddings.html"><i class="fa fa-check"></i><b>8</b> Single-cell Embeddings</a>
<ul>
<li class="chapter" data-level="8.1" data-path="uniform-manifold-approximation-and-projection-umap.html"><a href="uniform-manifold-approximation-and-projection-umap.html"><i class="fa fa-check"></i><b>8.1</b> Uniform Manifold Approximation and Projection (UMAP)</a></li>
<li class="chapter" data-level="8.2" data-path="t-stocastic-neighbor-embedding-t-sne.html"><a href="t-stocastic-neighbor-embedding-t-sne.html"><i class="fa fa-check"></i><b>8.2</b> t-Stocastic Neighbor Embedding (t-SNE)</a></li>
<li class="chapter" data-level="8.3" data-path="dimensionality-reduction-after-harmony.html"><a href="dimensionality-reduction-after-harmony.html"><i class="fa fa-check"></i><b>8.3</b> Dimensionality Reduction After Harmony</a></li>
<li class="chapter" data-level="8.4" data-path="highlighting-specific-cells-on-an-embedding.html"><a href="highlighting-specific-cells-on-an-embedding.html"><i class="fa fa-check"></i><b>8.4</b> Highlighting specific cells on an embedding</a></li>
<li class="chapter" data-level="8.5" data-path="importing-an-embedding-from-external-software.html"><a href="importing-an-embedding-from-external-software.html"><i class="fa fa-check"></i><b>8.5</b> Importing an embedding from external software</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="gene-scores-and-marker-genes-with-archr.html"><a href="gene-scores-and-marker-genes-with-archr.html"><i class="fa fa-check"></i><b>9</b> Gene Scores and Marker Genes with ArchR</a>
<ul>
<li class="chapter" data-level="9.1" data-path="calculating-gene-scores-in-archr.html"><a href="calculating-gene-scores-in-archr.html"><i class="fa fa-check"></i><b>9.1</b> Calculating Gene Scores in ArchR</a></li>
<li class="chapter" data-level="9.2" data-path="identification-of-marker-features.html"><a href="identification-of-marker-features.html"><i class="fa fa-check"></i><b>9.2</b> Identification of Marker Features</a></li>
<li class="chapter" data-level="9.3" data-path="identifying-marker-genes.html"><a href="identifying-marker-genes.html"><i class="fa fa-check"></i><b>9.3</b> Identifying Marker Genes</a></li>
<li class="chapter" data-level="9.4" data-path="visualizing-marker-genes-on-an-embedding.html"><a href="visualizing-marker-genes-on-an-embedding.html"><i class="fa fa-check"></i><b>9.4</b> Visualizing Marker Genes on an Embedding</a></li>
<li class="chapter" data-level="9.5" data-path="marker-genes-imputation-with-magic.html"><a href="marker-genes-imputation-with-magic.html"><i class="fa fa-check"></i><b>9.5</b> Marker Genes Imputation with MAGIC</a></li>
<li class="chapter" data-level="9.6" data-path="module-scores.html"><a href="module-scores.html"><i class="fa fa-check"></i><b>9.6</b> Module Scores</a></li>
<li class="chapter" data-level="9.7" data-path="track-plotting-with-archrbrowser.html"><a href="track-plotting-with-archrbrowser.html"><i class="fa fa-check"></i><b>9.7</b> Track Plotting with ArchRBrowser</a></li>
<li class="chapter" data-level="9.8" data-path="launching-the-interactive-archrbrowser.html"><a href="launching-the-interactive-archrbrowser.html"><i class="fa fa-check"></i><b>9.8</b> Launching the Interactive ArchRBrowser</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="defining-cluster-identity-with-scrna-seq.html"><a href="defining-cluster-identity-with-scrna-seq.html"><i class="fa fa-check"></i><b>10</b> Defining Cluster Identity with scRNA-seq</a>
<ul>
<li class="chapter" data-level="10.1" data-path="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html"><i class="fa fa-check"></i><b>10.1</b> Cross-platform linkage of scATAC-seq cells with scRNA-seq cells</a></li>
<li class="chapter" data-level="10.2" data-path="adding-pseudo-scrna-seq-profiles-for-each-scatac-seq-cell.html"><a href="adding-pseudo-scrna-seq-profiles-for-each-scatac-seq-cell.html"><i class="fa fa-check"></i><b>10.2</b> Adding Pseudo-scRNA-seq profiles for each scATAC-seq cell</a></li>
<li class="chapter" data-level="10.3" data-path="labeling-scatac-seq-clusters-with-scrna-seq-information.html"><a href="labeling-scatac-seq-clusters-with-scrna-seq-information.html"><i class="fa fa-check"></i><b>10.3</b> Labeling scATAC-seq clusters with scRNA-seq information</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="pseudo-bulk-replicates-in-archr.html"><a href="pseudo-bulk-replicates-in-archr.html"><i class="fa fa-check"></i><b>11</b> Pseudo-bulk Replicates in ArchR</a>
<ul>
<li class="chapter" data-level="11.1" data-path="how-does-archr-make-pseudo-bulk-replicates.html"><a href="how-does-archr-make-pseudo-bulk-replicates.html"><i class="fa fa-check"></i><b>11.1</b> How Does ArchR Make Pseudo-bulk Replicates?</a></li>
<li class="chapter" data-level="11.2" data-path="making-pseudo-bulk-replicates.html"><a href="making-pseudo-bulk-replicates.html"><i class="fa fa-check"></i><b>11.2</b> Making Pseudo-bulk Replicates</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="calling-peaks-with-archr.html"><a href="calling-peaks-with-archr.html"><i class="fa fa-check"></i><b>12</b> Calling Peaks with ArchR</a>
<ul>
<li class="chapter" data-level="12.1" data-path="the-iterative-overlap-peak-merging-procedure.html"><a href="the-iterative-overlap-peak-merging-procedure.html"><i class="fa fa-check"></i><b>12.1</b> The Iterative Overlap Peak Merging Procedure</a></li>
<li class="chapter" data-level="12.2" data-path="calling-peaks-w-macs2.html"><a href="calling-peaks-w-macs2.html"><i class="fa fa-check"></i><b>12.2</b> Calling Peaks w/ Macs2</a></li>
<li class="chapter" data-level="12.3" data-path="calling-peaks-w-tilematrix.html"><a href="calling-peaks-w-tilematrix.html"><i class="fa fa-check"></i><b>12.3</b> Calling Peaks w/ TileMatrix</a></li>
<li class="chapter" data-level="12.4" data-path="adding-a-peak-matrix.html"><a href="adding-a-peak-matrix.html"><i class="fa fa-check"></i><b>12.4</b> Adding a Peak Matrix</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="identifying-marker-peaks.html"><a href="identifying-marker-peaks.html"><i class="fa fa-check"></i><b>13</b> Identifying Marker Peaks</a>
<ul>
<li class="chapter" data-level="13.1" data-path="identifying-marker-peaks-with-archr.html"><a href="identifying-marker-peaks-with-archr.html"><i class="fa fa-check"></i><b>13.1</b> Identifying Marker Peaks with ArchR</a></li>
<li class="chapter" data-level="13.2" data-path="plotting-marker-peaks-in-archr.html"><a href="plotting-marker-peaks-in-archr.html"><i class="fa fa-check"></i><b>13.2</b> Plotting Marker Peaks in ArchR</a></li>
<li class="chapter" data-level="13.3" data-path="pairwise-testing-between-groups.html"><a href="pairwise-testing-between-groups.html"><i class="fa fa-check"></i><b>13.3</b> Pairwise Testing Between Groups</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="motif-and-feature-enrichment-with-archr.html"><a href="motif-and-feature-enrichment-with-archr.html"><i class="fa fa-check"></i><b>14</b> Motif and Feature Enrichment with ArchR</a>
<ul>
<li class="chapter" data-level="14.1" data-path="motif-enrichment.html"><a href="motif-enrichment.html"><i class="fa fa-check"></i><b>14.1</b> Motif Enrichment</a></li>
<li class="chapter" data-level="14.2" data-path="archr-enrichment.html"><a href="archr-enrichment.html"><i class="fa fa-check"></i><b>14.2</b> ArchR Enrichment</a></li>
<li class="chapter" data-level="14.3" data-path="custom-enrichment.html"><a href="custom-enrichment.html"><i class="fa fa-check"></i><b>14.3</b> Custom Enrichment</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="chromvar-deviatons-enrichment-with-archr.html"><a href="chromvar-deviatons-enrichment-with-archr.html"><i class="fa fa-check"></i><b>15</b> ChromVAR Deviatons Enrichment with ArchR</a>
<ul>
<li class="chapter" data-level="15.1" data-path="motif-deviations.html"><a href="motif-deviations.html"><i class="fa fa-check"></i><b>15.1</b> Motif Deviations</a></li>
<li class="chapter" data-level="15.2" data-path="archr-and-custom-deviations.html"><a href="archr-and-custom-deviations.html"><i class="fa fa-check"></i><b>15.2</b> ArchR and Custom Deviations</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="footprinting-with-archr.html"><a href="footprinting-with-archr.html"><i class="fa fa-check"></i><b>16</b> Footprinting with ArchR</a>
<ul>
<li class="chapter" data-level="16.1" data-path="motif-footprinting.html"><a href="motif-footprinting.html"><i class="fa fa-check"></i><b>16.1</b> Motif Footprinting</a></li>
<li class="chapter" data-level="16.2" data-path="normalization-of-footprints-for-tn5-bias.html"><a href="normalization-of-footprints-for-tn5-bias.html"><i class="fa fa-check"></i><b>16.2</b> Normalization of Footprints for Tn5 Bias</a></li>
<li class="chapter" data-level="16.3" data-path="feature-footprinting.html"><a href="feature-footprinting.html"><i class="fa fa-check"></i><b>16.3</b> Feature Footprinting</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="integrative-analysis-with-archr.html"><a href="integrative-analysis-with-archr.html"><i class="fa fa-check"></i><b>17</b> Integrative Analysis with ArchR</a>
<ul>
<li class="chapter" data-level="17.1" data-path="creating-low-overlapping-aggregates-of-cells.html"><a href="creating-low-overlapping-aggregates-of-cells.html"><i class="fa fa-check"></i><b>17.1</b> Creating Low-Overlapping Aggregates of Cells</a></li>
<li class="chapter" data-level="17.2" data-path="co-accessibility-with-archr.html"><a href="co-accessibility-with-archr.html"><i class="fa fa-check"></i><b>17.2</b> Co-accessibility with ArchR</a></li>
<li class="chapter" data-level="17.3" data-path="peak2genelinkage-with-archr.html"><a href="peak2genelinkage-with-archr.html"><i class="fa fa-check"></i><b>17.3</b> Peak2GeneLinkage with ArchR</a></li>
<li class="chapter" data-level="17.4" data-path="identification-of-positive-tf-regulators.html"><a href="identification-of-positive-tf-regulators.html"><i class="fa fa-check"></i><b>17.4</b> Identification of Positive TF-Regulators</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="trajectory-analysis-with-archr.html"><a href="trajectory-analysis-with-archr.html"><i class="fa fa-check"></i><b>18</b> Trajectory Analysis with ArchR</a>
<ul>
<li class="chapter" data-level="18.1" data-path="myeloid-trajectory---monocyte-differentiation.html"><a href="myeloid-trajectory---monocyte-differentiation.html"><i class="fa fa-check"></i><b>18.1</b> Myeloid Trajectory - Monocyte Differentiation</a></li>
<li class="chapter" data-level="18.2" data-path="lymphoid-trajectory---b-cell-cifferentiation.html"><a href="lymphoid-trajectory---b-cell-cifferentiation.html"><i class="fa fa-check"></i><b>18.2</b> Lymphoid Trajectory - B Cell Cifferentiation</a></li>
<li class="chapter" data-level="18.3" data-path="monocle3-trajectories.html"><a href="monocle3-trajectories.html"><i class="fa fa-check"></i><b>18.3</b> Monocle3 Trajectories</a></li>
<li class="chapter" data-level="18.4" data-path="slingshot-trajectories.html"><a href="slingshot-trajectories.html"><i class="fa fa-check"></i><b>18.4</b> Slingshot Trajectories</a></li>
<li class="chapter" data-level="18.5" data-path="comparing-the-different-trajectories.html"><a href="comparing-the-different-trajectories.html"><i class="fa fa-check"></i><b>18.5</b> Comparing the different trajectories</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="handling-archr-output.html"><a href="handling-archr-output.html"><i class="fa fa-check"></i><b>19</b> Handling ArchR Output</a>
<ul>
<li class="chapter" data-level="19.1" data-path="exporting-fragment-level-data.html"><a href="exporting-fragment-level-data.html"><i class="fa fa-check"></i><b>19.1</b> Exporting fragment-level data</a></li>
<li class="chapter" data-level="19.2" data-path="exporting-matrix-level-data.html"><a href="exporting-matrix-level-data.html"><i class="fa fa-check"></i><b>19.2</b> Exporting matrix-level data</a></li>
<li class="chapter" data-level="19.3" data-path="exporting-pseudo-bulked-data-to-a-summarizedexperiment.html"><a href="exporting-pseudo-bulked-data-to-a-summarizedexperiment.html"><i class="fa fa-check"></i><b>19.3</b> Exporting pseudo-bulked data to a <code>SummarizedExperiment</code></a></li>
<li class="chapter" data-level="19.4" data-path="exporting-pseudo-bulked-data-to-a-bigwig-file.html"><a href="exporting-pseudo-bulked-data-to-a-bigwig-file.html"><i class="fa fa-check"></i><b>19.4</b> Exporting pseudo-bulked data to a bigWig file</a></li>
</ul></li>
<li class="chapter" data-level="20" data-path="integration-with-bulk-atac-seq.html"><a href="integration-with-bulk-atac-seq.html"><i class="fa fa-check"></i><b>20</b> Integration with bulk ATAC-seq</a>
<ul>
<li class="chapter" data-level="20.1" data-path="projecting-bulk-atac-seq-data.html"><a href="projecting-bulk-atac-seq-data.html"><i class="fa fa-check"></i><b>20.1</b> Projecting bulk ATAC-seq data</a></li>
</ul></li>
<li class="chapter" data-level="21" data-path="multiomic-data-analysis-in-archr.html"><a href="multiomic-data-analysis-in-archr.html"><i class="fa fa-check"></i><b>21</b> Multiomic data analysis in ArchR</a>
<ul>
<li class="chapter" data-level="21.1" data-path="importing-data-and-setting-up-a-multiome-project.html"><a href="importing-data-and-setting-up-a-multiome-project.html"><i class="fa fa-check"></i><b>21.1</b> Importing data and setting up a Multiome project</a></li>
<li class="chapter" data-level="21.2" data-path="analysis-of-multiome-data-in-archr.html"><a href="analysis-of-multiome-data-in-archr.html"><i class="fa fa-check"></i><b>21.2</b> Analysis of multiome data in ArchR</a></li>
</ul></li>
<li class="chapter" data-level="22" data-path="plot-aesthetics-in-archr.html"><a href="plot-aesthetics-in-archr.html"><i class="fa fa-check"></i><b>22</b> Plot aesthetics in ArchR</a>
<ul>
<li class="chapter" data-level="22.1" data-path="using-and-manipulating-palettes-in-archr.html"><a href="using-and-manipulating-palettes-in-archr.html"><i class="fa fa-check"></i><b>22.1</b> Using and manipulating palettes in ArchR</a></li>
</ul></li>
<li class="chapter" data-level="23" data-path="testing-your-archr-installation.html"><a href="testing-your-archr-installation.html"><i class="fa fa-check"></i><b>23</b> Testing Your ArchR Installation</a></li>
<li class="chapter" data-level="24" data-path="session-information.html"><a href="session-information.html"><i class="fa fa-check"></i><b>24</b> Session Information</a></li>
<li class="divider"></li>
<li><a href="https://bookdown.org" target="_blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">ArchR: Robust and scaleable analysis of single-cell chromatin accessibility data.</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="projecting-bulk-atac-seq-data" class="section level2 hasAnchor" number="20.1">
<h2><span class="header-section-number">20.1</span> Projecting bulk ATAC-seq data<a href="projecting-bulk-atac-seq-data.html#projecting-bulk-atac-seq-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Projection of bulk ATAC-seq data into a single-cell dimensionality reduction is handled by the <code>projectBulkATAC()</code> function. This function accepts a <code>SummarizedExperiment</code> object containing the relevant bulk ATAC-seq data, subsets this object based on overlap of the peak regions in the bulk ATAC-seq <code>SummarizedExperiment</code> and single-cell <code>ArchRProject</code>, simulates sub-sampled “pseudocells”, and projects these pseudocells into the specified dimensionality reduction and embedding. A “pseudocell” is a downsampled representation of a bulk ATAC-seq library</p>
<p>To get the <code>projectBulkATAC()</code> function to work, the <code>SummarizedExperiment</code> object must be properly formatted. In particular, there needs to be an <code>assay</code> named “counts” containing the raw counts for each peak (row) for each sample (column). The <code>rowRanges</code> can be any peak set and the <code>projectBulkATAC()</code> function will match those rows to the <code>peakSet</code> of your <code>ArchRProject</code> based on overlaps. This means that it is not required to have a counts matrix based on the exact same set of peaks as used in your <code>ArchRProject</code>, enabling projection of bulk ATAC-seq data analyzed previously, for example from publicly available sources.</p>
<p>Below, we demonstrate the core functionality of <code>projectBulkATAC()</code> using a data set of bulk ATAC-seq data from FACS-enriched hematopoietic cell types (Corces et al. <em>Nature Genetics</em> 2015). First, lets download the counts matrix from GEO and fix the column names to be more human readable.</p>
<div class="sourceCode" id="cb444"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb444-1"><a href="projecting-bulk-atac-seq-data.html#cb444-1" tabindex="-1"></a><span class="co">#Download bulk ATAC-seq data and organize raw counts matrix</span></span>
<span id="cb444-2"><a href="projecting-bulk-atac-seq-data.html#cb444-2" tabindex="-1"></a><span class="fu">download.file</span>(<span class="at">url =</span> <span class="st">&quot;https://ftp.ncbi.nlm.nih.gov/geo/series/GSE74nnn/GSE74912/suppl/GSE74912_ATACseq_All_Counts.txt.gz&quot;</span>, <span class="at">destfile =</span> <span class="st">&quot;Corces_HemeAML_BulkATAC.txt.gz&quot;</span>)</span>
<span id="cb444-3"><a href="projecting-bulk-atac-seq-data.html#cb444-3" tabindex="-1"></a><span class="fu">download.file</span>(<span class="at">url =</span> <span class="st">&quot;https://jeffgranja.s3.amazonaws.com/ArchR/TestData/Corces_HemeAML_ReHeader.txt&quot;</span>, <span class="at">destfile =</span> <span class="st">&quot;Corces_HemeAML_ReHeader.txt&quot;</span>)</span>
<span id="cb444-4"><a href="projecting-bulk-atac-seq-data.html#cb444-4" tabindex="-1"></a></span>
<span id="cb444-5"><a href="projecting-bulk-atac-seq-data.html#cb444-5" tabindex="-1"></a>df <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="at">file =</span> <span class="st">&quot;Corces_HemeAML_BulkATAC.txt.gz&quot;</span>, <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">sep =</span> <span class="st">&#39;</span><span class="sc">\t</span><span class="st">&#39;</span>, <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>)</span>
<span id="cb444-6"><a href="projecting-bulk-atac-seq-data.html#cb444-6" tabindex="-1"></a>header <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="at">file =</span> <span class="st">&quot;Corces_HemeAML_ReHeader.txt&quot;</span>, <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">sep =</span> <span class="st">&#39;</span><span class="sc">\t</span><span class="st">&#39;</span>, <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>)</span>
<span id="cb444-7"><a href="projecting-bulk-atac-seq-data.html#cb444-7" tabindex="-1"></a><span class="fu">colnames</span>(df) <span class="ot">&lt;-</span> header<span class="sc">$</span>NewHeader</span>
<span id="cb444-8"><a href="projecting-bulk-atac-seq-data.html#cb444-8" tabindex="-1"></a>df[<span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>]</span>
<span id="cb444-9"><a href="projecting-bulk-atac-seq-data.html#cb444-9" tabindex="-1"></a><span class="do">##    chr  start    end Donor4983_1A_HSC Donor4983_2A_MPP Donor4983_3A_LMPP</span></span>
<span id="cb444-10"><a href="projecting-bulk-atac-seq-data.html#cb444-10" tabindex="-1"></a><span class="do">## 1 chr1  10025  10525                1                0                 4</span></span>
<span id="cb444-11"><a href="projecting-bulk-atac-seq-data.html#cb444-11" tabindex="-1"></a><span class="do">## 2 chr1  13252  13752                0                5                 6</span></span>
<span id="cb444-12"><a href="projecting-bulk-atac-seq-data.html#cb444-12" tabindex="-1"></a><span class="do">## 3 chr1  16019  16519                4                2                 6</span></span>
<span id="cb444-13"><a href="projecting-bulk-atac-seq-data.html#cb444-13" tabindex="-1"></a><span class="do">## 4 chr1  96376  96876                0                0                 0</span></span>
<span id="cb444-14"><a href="projecting-bulk-atac-seq-data.html#cb444-14" tabindex="-1"></a><span class="do">## 5 chr1 115440 115940                0                3                 0</span></span>
<span id="cb444-15"><a href="projecting-bulk-atac-seq-data.html#cb444-15" tabindex="-1"></a><span class="do">## 6 chr1 235393 235893                0                2                 2</span></span></code></pre></div>
<p>Inspecting the first few rows and columns of this <code>data.frame</code> shows that the first three columns are actually the peak coordinates and the remaining columns represent the individual samples. Because of this, we need to strip out the peak information and create a separate <code>GenomicRanges</code> object to represent the peak regions.</p>
<div class="sourceCode" id="cb445"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb445-1"><a href="projecting-bulk-atac-seq-data.html#cb445-1" tabindex="-1"></a>bulk_gr <span class="ot">&lt;-</span> <span class="fu">GRanges</span>(<span class="at">seqnames =</span> df<span class="sc">$</span>chr, <span class="at">ranges =</span> <span class="fu">IRanges</span>(<span class="at">start =</span> df<span class="sc">$</span>start<span class="sc">+</span><span class="dv">1</span>, <span class="at">end =</span> df<span class="sc">$</span>end))</span>
<span id="cb445-2"><a href="projecting-bulk-atac-seq-data.html#cb445-2" tabindex="-1"></a>df <span class="ot">&lt;-</span> df[,<span class="fu">which</span>(<span class="fu">colnames</span>(df) <span class="sc">%ni%</span> <span class="fu">c</span>(<span class="st">&quot;chr&quot;</span>,<span class="st">&quot;start&quot;</span>,<span class="st">&quot;end&quot;</span>))]</span>
<span id="cb445-3"><a href="projecting-bulk-atac-seq-data.html#cb445-3" tabindex="-1"></a><span class="fu">rownames</span>(df) <span class="ot">&lt;-</span> <span class="fu">paste</span>(bulk_gr)</span></code></pre></div>
<p>We don’t need all of the data in this counts matrix so for simplicity we will subset the counts matrix to only include a few cell types (columns) with data that is well-represented in our tutorial dataset.</p>
<div class="sourceCode" id="cb446"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb446-1"><a href="projecting-bulk-atac-seq-data.html#cb446-1" tabindex="-1"></a>colsToKeep <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Donor4983_10A_CD8Tcell&quot;</span>,</span>
<span id="cb446-2"><a href="projecting-bulk-atac-seq-data.html#cb446-2" tabindex="-1"></a>                <span class="st">&quot;Donor4983_9A_CD4Tcell&quot;</span>,</span>
<span id="cb446-3"><a href="projecting-bulk-atac-seq-data.html#cb446-3" tabindex="-1"></a>                <span class="st">&quot;Donor5483_13A_Bcell&quot;</span>,</span>
<span id="cb446-4"><a href="projecting-bulk-atac-seq-data.html#cb446-4" tabindex="-1"></a>                <span class="st">&quot;Donor5483_11B_NKcell&quot;</span>,</span>
<span id="cb446-5"><a href="projecting-bulk-atac-seq-data.html#cb446-5" tabindex="-1"></a>                <span class="st">&quot;Donor7256_7B_Mono&quot;</span>)</span>
<span id="cb446-6"><a href="projecting-bulk-atac-seq-data.html#cb446-6" tabindex="-1"></a>df <span class="ot">&lt;-</span> df[,colsToKeep]</span></code></pre></div>
<p>Now we are ready to create our <code>SummarizedExperiment</code> object using the raw counts and peak regions.</p>
<div class="sourceCode" id="cb447"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb447-1"><a href="projecting-bulk-atac-seq-data.html#cb447-1" tabindex="-1"></a>seBulk <span class="ot">&lt;-</span> <span class="fu">SummarizedExperiment</span>(<span class="at">assays =</span> <span class="fu">SimpleList</span>(<span class="at">counts =</span> <span class="fu">as.matrix</span>(df)), <span class="at">rowRanges =</span> bulk_gr)</span>
<span id="cb447-2"><a href="projecting-bulk-atac-seq-data.html#cb447-2" tabindex="-1"></a>seBulk</span>
<span id="cb447-3"><a href="projecting-bulk-atac-seq-data.html#cb447-3" tabindex="-1"></a><span class="do">## class: RangedSummarizedExperiment </span></span>
<span id="cb447-4"><a href="projecting-bulk-atac-seq-data.html#cb447-4" tabindex="-1"></a><span class="do">## dim: 590650 5 </span></span>
<span id="cb447-5"><a href="projecting-bulk-atac-seq-data.html#cb447-5" tabindex="-1"></a><span class="do">## metadata(0):</span></span>
<span id="cb447-6"><a href="projecting-bulk-atac-seq-data.html#cb447-6" tabindex="-1"></a><span class="do">## assays(1): counts</span></span>
<span id="cb447-7"><a href="projecting-bulk-atac-seq-data.html#cb447-7" tabindex="-1"></a><span class="do">## rownames(590650): chr1:10026-10525 chr1:13253-13752 ...</span></span>
<span id="cb447-8"><a href="projecting-bulk-atac-seq-data.html#cb447-8" tabindex="-1"></a><span class="do">##   chrX:154912442-154912941 chrX:155259861-155260360</span></span>
<span id="cb447-9"><a href="projecting-bulk-atac-seq-data.html#cb447-9" tabindex="-1"></a><span class="do">## rowData names(0):</span></span>
<span id="cb447-10"><a href="projecting-bulk-atac-seq-data.html#cb447-10" tabindex="-1"></a><span class="do">## colnames(5): Donor4983_10A_CD8Tcell Donor4983_9A_CD4Tcell</span></span>
<span id="cb447-11"><a href="projecting-bulk-atac-seq-data.html#cb447-11" tabindex="-1"></a><span class="do">##   Donor5483_13A_Bcell Donor5483_11B_NKcell Donor7256_7B_Mono</span></span>
<span id="cb447-12"><a href="projecting-bulk-atac-seq-data.html#cb447-12" tabindex="-1"></a><span class="do">## colData names(0):</span></span></code></pre></div>
<p>To project these bulk samples using <code>projectBulkATAC()</code>, we specify the <code>reducedDims</code> and <code>embedding</code> we want to use as well as the number of “pseudocells” we want to project.</p>
<div class="sourceCode" id="cb448"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb448-1"><a href="projecting-bulk-atac-seq-data.html#cb448-1" tabindex="-1"></a>bulkPro <span class="ot">&lt;-</span> <span class="fu">projectBulkATAC</span>(<span class="at">ArchRProj =</span> projHeme5,</span>
<span id="cb448-2"><a href="projecting-bulk-atac-seq-data.html#cb448-2" tabindex="-1"></a>                           <span class="at">seATAC =</span> seBulk,</span>
<span id="cb448-3"><a href="projecting-bulk-atac-seq-data.html#cb448-3" tabindex="-1"></a>                           <span class="at">reducedDims =</span> <span class="st">&quot;IterativeLSI&quot;</span>,</span>
<span id="cb448-4"><a href="projecting-bulk-atac-seq-data.html#cb448-4" tabindex="-1"></a>                           <span class="at">embedding =</span> <span class="st">&quot;UMAP&quot;</span>,</span>
<span id="cb448-5"><a href="projecting-bulk-atac-seq-data.html#cb448-5" tabindex="-1"></a>                           <span class="at">n =</span> <span class="dv">250</span>)</span>
<span id="cb448-6"><a href="projecting-bulk-atac-seq-data.html#cb448-6" tabindex="-1"></a><span class="do">## ArchR logging to : ArchRLogs/ArchR-projectBulkATAC-1a1cff0c1c-Date-2025-01-23_Time-23-32-57.421948.log</span></span>
<span id="cb448-7"><a href="projecting-bulk-atac-seq-data.html#cb448-7" tabindex="-1"></a><span class="do">## If there is an issue, please report to github with logFile!</span></span>
<span id="cb448-8"><a href="projecting-bulk-atac-seq-data.html#cb448-8" tabindex="-1"></a><span class="do">## Overlap Ratio of Reduced Dims Features = 0.99084</span></span>
<span id="cb448-9"><a href="projecting-bulk-atac-seq-data.html#cb448-9" tabindex="-1"></a><span class="do">## 2025-01-23 23:32:58.020513 :</span></span>
<span id="cb448-10"><a href="projecting-bulk-atac-seq-data.html#cb448-10" tabindex="-1"></a><span class="do">## 23:33:06 Creating temp directory /tmp/RtmpLozqnR/dir1a118e17bce</span></span>
<span id="cb448-11"><a href="projecting-bulk-atac-seq-data.html#cb448-11" tabindex="-1"></a><span class="do">## untar: using cmd = &#39;/usr/bin/tar -xf &#39;/workspace/ArchR/ArchR_Website_Testing/bookdown/Save-ProjHeme1/Embeddings/Save-Uwot-UMAP-Params-IterativeLSI-1a14b0e7d95-Date-2025-01-23_Time-20-13-46.56756.tar&#39; -C &#39;/tmp/RtmpLozqnR/dir1a118e17bce&#39;&#39;</span></span>
<span id="cb448-12"><a href="projecting-bulk-atac-seq-data.html#cb448-12" tabindex="-1"></a><span class="do">## 23:33:10 Read 6250 rows and found 30 numeric columns</span></span>
<span id="cb448-13"><a href="projecting-bulk-atac-seq-data.html#cb448-13" tabindex="-1"></a><span class="do">## 23:33:10 Processing block 1 of 1</span></span>
<span id="cb448-14"><a href="projecting-bulk-atac-seq-data.html#cb448-14" tabindex="-1"></a><span class="do">## 23:33:10 Writing NN index file to temp file /tmp/RtmpLozqnR/file1a1675bf888</span></span>
<span id="cb448-15"><a href="projecting-bulk-atac-seq-data.html#cb448-15" tabindex="-1"></a><span class="do">## 23:33:10 Searching Annoy index using 8 threads, search_k = 3000</span></span>
<span id="cb448-16"><a href="projecting-bulk-atac-seq-data.html#cb448-16" tabindex="-1"></a><span class="do">## 23:33:10 Commencing smooth kNN distance calibration using 8 threads with target n_neighbors = 30</span></span>
<span id="cb448-17"><a href="projecting-bulk-atac-seq-data.html#cb448-17" tabindex="-1"></a><span class="do">## 23:33:10 Initializing by weighted average of neighbor coordinates using 8 threads</span></span>
<span id="cb448-18"><a href="projecting-bulk-atac-seq-data.html#cb448-18" tabindex="-1"></a><span class="do">## 23:33:10 Commencing optimization for 67 epochs, with 187500 positive edges</span></span>
<span id="cb448-19"><a href="projecting-bulk-atac-seq-data.html#cb448-19" tabindex="-1"></a><span class="do">## 23:33:10 Finished</span></span>
<span id="cb448-20"><a href="projecting-bulk-atac-seq-data.html#cb448-20" tabindex="-1"></a><span class="do">## ArchR logging successful to : ArchRLogs/ArchR-projectBulkATAC-1a1cff0c1c-Date-2025-01-23_Time-23-32-57.421948.log</span></span>
<span id="cb448-21"><a href="projecting-bulk-atac-seq-data.html#cb448-21" tabindex="-1"></a>bulkPro</span>
<span id="cb448-22"><a href="projecting-bulk-atac-seq-data.html#cb448-22" tabindex="-1"></a><span class="do">## List of length 3</span></span>
<span id="cb448-23"><a href="projecting-bulk-atac-seq-data.html#cb448-23" tabindex="-1"></a><span class="do">## names(3): simulatedBulkUMAP singleCellUMAP simulatedReducedDims</span></span></code></pre></div>
<p>The output of <code>projectBulkATAC()</code> is a list object containing 3 entries discussed below.</p>
<p><code>simulatedBulkUMAP</code> is a <code>DataFrame</code> containing the embedding coordinates for the simulated “pseduocells”.</p>
<div class="sourceCode" id="cb449"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb449-1"><a href="projecting-bulk-atac-seq-data.html#cb449-1" tabindex="-1"></a><span class="fu">head</span>(bulkPro<span class="sc">$</span>simulatedBulkUMAP)</span>
<span id="cb449-2"><a href="projecting-bulk-atac-seq-data.html#cb449-2" tabindex="-1"></a><span class="do">## DataFrame with 6 rows and 3 columns</span></span>
<span id="cb449-3"><a href="projecting-bulk-atac-seq-data.html#cb449-3" tabindex="-1"></a><span class="do">##                              UMAP1      UMAP2                   Type</span></span>
<span id="cb449-4"><a href="projecting-bulk-atac-seq-data.html#cb449-4" tabindex="-1"></a><span class="do">##                          &lt;numeric&gt;  &lt;numeric&gt;                  &lt;Rle&gt;</span></span>
<span id="cb449-5"><a href="projecting-bulk-atac-seq-data.html#cb449-5" tabindex="-1"></a><span class="do">## Donor4983_10A_CD8Tcell#1   9.37652 -0.2009891 Donor4983_10A_CD8Tcell</span></span>
<span id="cb449-6"><a href="projecting-bulk-atac-seq-data.html#cb449-6" tabindex="-1"></a><span class="do">## Donor4983_10A_CD8Tcell#2   9.69217  0.4535988 Donor4983_10A_CD8Tcell</span></span>
<span id="cb449-7"><a href="projecting-bulk-atac-seq-data.html#cb449-7" tabindex="-1"></a><span class="do">## Donor4983_10A_CD8Tcell#3   9.65991 -0.0697212 Donor4983_10A_CD8Tcell</span></span>
<span id="cb449-8"><a href="projecting-bulk-atac-seq-data.html#cb449-8" tabindex="-1"></a><span class="do">## Donor4983_10A_CD8Tcell#4   9.65814 -0.0221541 Donor4983_10A_CD8Tcell</span></span>
<span id="cb449-9"><a href="projecting-bulk-atac-seq-data.html#cb449-9" tabindex="-1"></a><span class="do">## Donor4983_10A_CD8Tcell#5   9.64607  0.1114746 Donor4983_10A_CD8Tcell</span></span>
<span id="cb449-10"><a href="projecting-bulk-atac-seq-data.html#cb449-10" tabindex="-1"></a><span class="do">## Donor4983_10A_CD8Tcell#6   9.46927 -0.2376311 Donor4983_10A_CD8Tcell</span></span></code></pre></div>
<p><code>singleCellUMAP</code> is a <code>DataFrame</code> containing the embedding coordinates for all of the original single-cells in your <code>ArchRProject</code>.</p>
<div class="sourceCode" id="cb450"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb450-1"><a href="projecting-bulk-atac-seq-data.html#cb450-1" tabindex="-1"></a><span class="fu">head</span>(bulkPro<span class="sc">$</span>singleCellUMAP)</span>
<span id="cb450-2"><a href="projecting-bulk-atac-seq-data.html#cb450-2" tabindex="-1"></a><span class="do">## DataFrame with 6 rows and 3 columns</span></span>
<span id="cb450-3"><a href="projecting-bulk-atac-seq-data.html#cb450-3" tabindex="-1"></a><span class="do">##                                        UMAP1     UMAP2   Type</span></span>
<span id="cb450-4"><a href="projecting-bulk-atac-seq-data.html#cb450-4" tabindex="-1"></a><span class="do">##                                    &lt;numeric&gt; &lt;numeric&gt;  &lt;Rle&gt;</span></span>
<span id="cb450-5"><a href="projecting-bulk-atac-seq-data.html#cb450-5" tabindex="-1"></a><span class="do">## scATAC_BMMC_R1#TTATGTCAGTGATTAG-1 -2.0100708 -0.463790 scATAC</span></span>
<span id="cb450-6"><a href="projecting-bulk-atac-seq-data.html#cb450-6" tabindex="-1"></a><span class="do">## scATAC_BMMC_R1#AAGATAGTCACCGCGA-1  8.3619858  1.214364 scATAC</span></span>
<span id="cb450-7"><a href="projecting-bulk-atac-seq-data.html#cb450-7" tabindex="-1"></a><span class="do">## scATAC_BMMC_R1#GCATTGAAGATTCCGT-1 -0.1737829 -9.489737 scATAC</span></span>
<span id="cb450-8"><a href="projecting-bulk-atac-seq-data.html#cb450-8" tabindex="-1"></a><span class="do">## scATAC_BMMC_R1#TATGTTCAGGGTTCCC-1 -0.1669594 -7.583627 scATAC</span></span>
<span id="cb450-9"><a href="projecting-bulk-atac-seq-data.html#cb450-9" tabindex="-1"></a><span class="do">## scATAC_BMMC_R1#TCCATCGGTCCCGTGA-1 -0.0378193 -8.868483 scATAC</span></span>
<span id="cb450-10"><a href="projecting-bulk-atac-seq-data.html#cb450-10" tabindex="-1"></a><span class="do">## scATAC_BMMC_R1#AGTTACGAGAACGTCG-1 -2.8937029  0.609718 scATAC</span></span></code></pre></div>
<p><code>simulatedReducedDims</code> is a <code>DataFrame</code> containing the coordinates in the given reduced dimensions (i.e. “IterativeLSI”) for each of the simulated “pseudocells”.</p>
<div class="sourceCode" id="cb451"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb451-1"><a href="projecting-bulk-atac-seq-data.html#cb451-1" tabindex="-1"></a><span class="fu">head</span>(bulkPro<span class="sc">$</span>simulatedReducedDims[,<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>])</span>
<span id="cb451-2"><a href="projecting-bulk-atac-seq-data.html#cb451-2" tabindex="-1"></a><span class="do">##                               LSI1     LSI2       LSI3        LSI4      LSI5</span></span>
<span id="cb451-3"><a href="projecting-bulk-atac-seq-data.html#cb451-3" tabindex="-1"></a><span class="do">## Donor4983_10A_CD8Tcell#1 -4.625283 1.686137 -0.5859489 -0.16511159 0.5173089</span></span>
<span id="cb451-4"><a href="projecting-bulk-atac-seq-data.html#cb451-4" tabindex="-1"></a><span class="do">## Donor4983_10A_CD8Tcell#2 -4.655096 1.604512 -0.3136310 -0.22394294 0.4209449</span></span>
<span id="cb451-5"><a href="projecting-bulk-atac-seq-data.html#cb451-5" tabindex="-1"></a><span class="do">## Donor4983_10A_CD8Tcell#3 -4.677772 1.683076 -0.5080028 -0.01157532 0.5635535</span></span>
<span id="cb451-6"><a href="projecting-bulk-atac-seq-data.html#cb451-6" tabindex="-1"></a><span class="do">## Donor4983_10A_CD8Tcell#4 -4.631163 1.706387 -0.4764640 -0.02528689 0.4869747</span></span>
<span id="cb451-7"><a href="projecting-bulk-atac-seq-data.html#cb451-7" tabindex="-1"></a><span class="do">## Donor4983_10A_CD8Tcell#5 -4.643288 1.659233 -0.5008208 -0.17292483 0.3627571</span></span>
<span id="cb451-8"><a href="projecting-bulk-atac-seq-data.html#cb451-8" tabindex="-1"></a><span class="do">## Donor4983_10A_CD8Tcell#6 -4.614042 1.690643 -0.3981789 -0.01228556 0.6056827</span></span></code></pre></div>
<p>Once we have the output of <code>projectBulkATAC()</code>, we often just want to plot this data to see where in our embedding the simulated “pseudocells” fall. There are many ways to do this but the easiest is to just concatenate the data from <code>projectBulkATAC()</code>, create a color palette, and use <code>ggPoint()</code> to plot.</p>
<div class="sourceCode" id="cb452"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb452-1"><a href="projecting-bulk-atac-seq-data.html#cb452-1" tabindex="-1"></a><span class="co">#concatenate single-cell and pseudocell embedding positions</span></span>
<span id="cb452-2"><a href="projecting-bulk-atac-seq-data.html#cb452-2" tabindex="-1"></a>pro_df <span class="ot">&lt;-</span> <span class="fu">rbind</span>(bulkPro<span class="sc">$</span>singleCellUMAP, bulkPro<span class="sc">$</span>simulatedBulkUMAP)</span>
<span id="cb452-3"><a href="projecting-bulk-atac-seq-data.html#cb452-3" tabindex="-1"></a></span>
<span id="cb452-4"><a href="projecting-bulk-atac-seq-data.html#cb452-4" tabindex="-1"></a><span class="co">#create a color palette and force the scATAC cells to be grey to enable visualization of the project bulk ATAC data</span></span>
<span id="cb452-5"><a href="projecting-bulk-atac-seq-data.html#cb452-5" tabindex="-1"></a>pal <span class="ot">&lt;-</span> <span class="fu">paletteDiscrete</span>(<span class="at">values =</span> <span class="fu">unique</span>(<span class="fu">as.character</span>(pro_df<span class="sc">$</span>Type)), <span class="at">set =</span> <span class="st">&quot;stallion&quot;</span>)</span>
<span id="cb452-6"><a href="projecting-bulk-atac-seq-data.html#cb452-6" tabindex="-1"></a>pal[<span class="st">&quot;scATAC&quot;</span>] <span class="ot">&lt;-</span> <span class="st">&quot;#BABABA&quot;</span></span>
<span id="cb452-7"><a href="projecting-bulk-atac-seq-data.html#cb452-7" tabindex="-1"></a></span>
<span id="cb452-8"><a href="projecting-bulk-atac-seq-data.html#cb452-8" tabindex="-1"></a><span class="co">#plot using ggPoint</span></span>
<span id="cb452-9"><a href="projecting-bulk-atac-seq-data.html#cb452-9" tabindex="-1"></a><span class="fu">ggPoint</span>(<span class="at">x =</span> pro_df<span class="sc">$</span>UMAP1,</span>
<span id="cb452-10"><a href="projecting-bulk-atac-seq-data.html#cb452-10" tabindex="-1"></a>        <span class="at">y =</span> pro_df<span class="sc">$</span>UMAP2,</span>
<span id="cb452-11"><a href="projecting-bulk-atac-seq-data.html#cb452-11" tabindex="-1"></a>        <span class="at">discrete =</span> <span class="cn">TRUE</span>,</span>
<span id="cb452-12"><a href="projecting-bulk-atac-seq-data.html#cb452-12" tabindex="-1"></a>        <span class="at">color =</span> <span class="fu">as.character</span>(pro_df<span class="sc">$</span>Type),</span>
<span id="cb452-13"><a href="projecting-bulk-atac-seq-data.html#cb452-13" tabindex="-1"></a>        <span class="at">pal =</span> pal,</span>
<span id="cb452-14"><a href="projecting-bulk-atac-seq-data.html#cb452-14" tabindex="-1"></a>        <span class="at">xlabel =</span> <span class="st">&quot;UMAP Dimension 1&quot;</span>,</span>
<span id="cb452-15"><a href="projecting-bulk-atac-seq-data.html#cb452-15" tabindex="-1"></a>        <span class="at">ylabel =</span> <span class="st">&quot;UMAP Dimension 2&quot;</span>,</span>
<span id="cb452-16"><a href="projecting-bulk-atac-seq-data.html#cb452-16" tabindex="-1"></a>        <span class="at">title =</span> <span class="st">&quot;Heme Bulk ATAC-seq Projection&quot;</span>)</span></code></pre></div>
<p><img src="ArchR_files/figure-html/unnamed-chunk-450-1.svg" width="672" /></p>
<p>In the resulting projection UMAP plot, you can see that the bulk ATAC-seq pseudocells pile up in the regions occupied by the corresponding single-cells.</p>
<p>At this point in the tutorial, we are done modifying our <code>ArchRProject</code> so lets save it one last time to make sure all of our changes are saved on disk. We will just save it on top of our previous <code>projHeme5</code> directory.</p>
<p>projHeme5 &lt;- saveArchRProject(ArchRProj = projHeme5, outputDirectory = “Save-ProjHeme5”, load = TRUE)</p>

</div>
<!-- </div> -->
            </section>

          </div>
        </div>
      </div>
<a href="integration-with-bulk-atac-seq.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="multiomic-data-analysis-in-archr.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": null
},
"fontsettings": null,
"edit": null,
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "none"
},
"info": false
});
});
</script>

</body>

</html>
