<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>9.7 Track Plotting with ArchRBrowser | ArchR: Robust and scaleable analysis of single-cell chromatin accessibility data.</title>
  <meta name="description" content="A guide to ArchR" />
  <meta name="generator" content="bookdown 0.41 and GitBook 2.6.7" />

  <meta property="og:title" content="9.7 Track Plotting with ArchRBrowser | ArchR: Robust and scaleable analysis of single-cell chromatin accessibility data." />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="A guide to ArchR" />
  <meta name="github-repo" content="GreenleafLab/ArchR" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="9.7 Track Plotting with ArchRBrowser | ArchR: Robust and scaleable analysis of single-cell chromatin accessibility data." />
  
  <meta name="twitter:description" content="A guide to ArchR" />
  

<meta name="author" content="Jeffrey Granja and Ryan Corces" />


<meta name="date" content="2025-01-23" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="images/favicon.ico" type="image/x-icon" />
<link rel="prev" href="module-scores.html"/>
<link rel="next" href="launching-the-interactive-archrbrowser.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<link href="libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="../index.html">Back to ArchR Home</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><img src="images/ArchR_Logo_Integrated.png" alt="" width="200"></a></li>
<li class="chapter" data-level="1" data-path="debugging-and-troubleshooting-in-archr.html"><a href="debugging-and-troubleshooting-in-archr.html"><i class="fa fa-check"></i><b>1</b> Debugging and Troubleshooting in ArchR</a></li>
<li class="chapter" data-level="2" data-path="manage-archrs-dependencies.html"><a href="manage-archrs-dependencies.html"><i class="fa fa-check"></i><b>2</b> Manage ArchR’s Dependencies</a>
<ul>
<li class="chapter" data-level="2.1" data-path="using-renv-to-manage-dependencies.html"><a href="using-renv-to-manage-dependencies.html"><i class="fa fa-check"></i><b>2.1</b> Using renv to manage dependencies</a></li>
<li class="chapter" data-level="2.2" data-path="other-environment-managers.html"><a href="other-environment-managers.html"><i class="fa fa-check"></i><b>2.2</b> Other environment managers</a></li>
<li class="chapter" data-level="2.3" data-path="using-docker-for-pre-compiled-environment-setup.html"><a href="using-docker-for-pre-compiled-environment-setup.html"><i class="fa fa-check"></i><b>2.3</b> Using Docker for pre-compiled environment setup</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="getting-started-with-archr.html"><a href="getting-started-with-archr.html"><i class="fa fa-check"></i><b>3</b> Getting Started with ArchR</a>
<ul>
<li class="chapter" data-level="3.1" data-path="a-brief-primer-on-atac-seq-terminology.html"><a href="a-brief-primer-on-atac-seq-terminology.html"><i class="fa fa-check"></i><b>3.1</b> A Brief Primer on ATAC-seq Terminology</a></li>
<li class="chapter" data-level="3.2" data-path="why-use-archr.html"><a href="why-use-archr.html"><i class="fa fa-check"></i><b>3.2</b> Why use ArchR?</a></li>
<li class="chapter" data-level="3.3" data-path="what-is-an-arrow-file-archrproject.html"><a href="what-is-an-arrow-file-archrproject.html"><i class="fa fa-check"></i><b>3.3</b> What is an Arrow file / <code>ArchRProject</code>?</a></li>
<li class="chapter" data-level="3.4" data-path="input-file-types-in-archr.html"><a href="input-file-types-in-archr.html"><i class="fa fa-check"></i><b>3.4</b> Input File Types in ArchR</a></li>
<li class="chapter" data-level="3.5" data-path="getting-set-up.html"><a href="getting-set-up.html"><i class="fa fa-check"></i><b>3.5</b> Getting Set Up</a></li>
<li class="chapter" data-level="3.6" data-path="creating-arrow-files.html"><a href="creating-arrow-files.html"><i class="fa fa-check"></i><b>3.6</b> Creating Arrow Files</a></li>
<li class="chapter" data-level="3.7" data-path="per-cell-quality-control.html"><a href="per-cell-quality-control.html"><i class="fa fa-check"></i><b>3.7</b> Per-cell Quality Control</a></li>
<li class="chapter" data-level="3.8" data-path="using-bam-files-for-arrow-file-creation.html"><a href="using-bam-files-for-arrow-file-creation.html"><i class="fa fa-check"></i><b>3.8</b> Using BAM files for Arrow File creation</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="doublet-inference-with-archr.html"><a href="doublet-inference-with-archr.html"><i class="fa fa-check"></i><b>4</b> Doublet Inference with ArchR</a>
<ul>
<li class="chapter" data-level="4.1" data-path="how-does-doublet-identification-work-in-archr.html"><a href="how-does-doublet-identification-work-in-archr.html"><i class="fa fa-check"></i><b>4.1</b> How does doublet identification work in ArchR?</a></li>
<li class="chapter" data-level="4.2" data-path="inferring-scatac-seq-doublets-with-archr.html"><a href="inferring-scatac-seq-doublets-with-archr.html"><i class="fa fa-check"></i><b>4.2</b> Inferring scATAC-seq Doublets with ArchR</a></li>
<li class="chapter" data-level="4.3" data-path="using-demuxlet-with-archr.html"><a href="using-demuxlet-with-archr.html"><i class="fa fa-check"></i><b>4.3</b> Using <em>demuxlet</em> with ArchR</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="creating-an-archrproject.html"><a href="creating-an-archrproject.html"><i class="fa fa-check"></i><b>5</b> Creating an ArchRProject</a>
<ul>
<li class="chapter" data-level="5.1" data-path="creating-an-archrproject-1.html"><a href="creating-an-archrproject-1.html"><i class="fa fa-check"></i><b>5.1</b> Creating An ArchRProject</a></li>
<li class="chapter" data-level="5.2" data-path="manipulating-an-archrproject.html"><a href="manipulating-an-archrproject.html"><i class="fa fa-check"></i><b>5.2</b> Manipulating An ArchRProject</a></li>
<li class="chapter" data-level="5.3" data-path="plotting-sample-statistics-from-an-archrproject.html"><a href="plotting-sample-statistics-from-an-archrproject.html"><i class="fa fa-check"></i><b>5.3</b> Plotting Sample Statistics from an ArchRProject</a></li>
<li class="chapter" data-level="5.4" data-path="plotting-sample-fragment-size-distribution-and-tss-enrichment-profiles..html"><a href="plotting-sample-fragment-size-distribution-and-tss-enrichment-profiles..html"><i class="fa fa-check"></i><b>5.4</b> Plotting Sample Fragment Size Distribution and TSS Enrichment Profiles.</a></li>
<li class="chapter" data-level="5.5" data-path="saving-and-loading-an-archrproject.html"><a href="saving-and-loading-an-archrproject.html"><i class="fa fa-check"></i><b>5.5</b> Saving and Loading an <code>ArchRProject</code></a></li>
<li class="chapter" data-level="5.6" data-path="filtering-doublets-from-an-archrproject.html"><a href="filtering-doublets-from-an-archrproject.html"><i class="fa fa-check"></i><b>5.6</b> Filtering Doublets from an ArchRProject</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="dimensionality-reduction-with-archr.html"><a href="dimensionality-reduction-with-archr.html"><i class="fa fa-check"></i><b>6</b> Dimensionality Reduction with ArchR</a>
<ul>
<li class="chapter" data-level="6.1" data-path="archrs-lsi-implementation.html"><a href="archrs-lsi-implementation.html"><i class="fa fa-check"></i><b>6.1</b> ArchR’s LSI Implementation</a></li>
<li class="chapter" data-level="6.2" data-path="iterative-latent-semantic-indexing-lsi.html"><a href="iterative-latent-semantic-indexing-lsi.html"><i class="fa fa-check"></i><b>6.2</b> Iterative Latent Semantic Indexing (LSI)</a></li>
<li class="chapter" data-level="6.3" data-path="estimated-lsi.html"><a href="estimated-lsi.html"><i class="fa fa-check"></i><b>6.3</b> Estimated LSI</a></li>
<li class="chapter" data-level="6.4" data-path="batch-effect-correction-wtih-harmony.html"><a href="batch-effect-correction-wtih-harmony.html"><i class="fa fa-check"></i><b>6.4</b> Batch Effect Correction wtih Harmony</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="clustering-with-archr.html"><a href="clustering-with-archr.html"><i class="fa fa-check"></i><b>7</b> Clustering with ArchR</a>
<ul>
<li class="chapter" data-level="7.1" data-path="clustering-using-seurats-findclusters-function.html"><a href="clustering-using-seurats-findclusters-function.html"><i class="fa fa-check"></i><b>7.1</b> Clustering using Seurat’s <code>FindClusters()</code> function</a></li>
<li class="chapter" data-level="7.2" data-path="clustering-using-scran.html"><a href="clustering-using-scran.html"><i class="fa fa-check"></i><b>7.2</b> Clustering using <code>scran</code></a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="single-cell-embeddings.html"><a href="single-cell-embeddings.html"><i class="fa fa-check"></i><b>8</b> Single-cell Embeddings</a>
<ul>
<li class="chapter" data-level="8.1" data-path="uniform-manifold-approximation-and-projection-umap.html"><a href="uniform-manifold-approximation-and-projection-umap.html"><i class="fa fa-check"></i><b>8.1</b> Uniform Manifold Approximation and Projection (UMAP)</a></li>
<li class="chapter" data-level="8.2" data-path="t-stocastic-neighbor-embedding-t-sne.html"><a href="t-stocastic-neighbor-embedding-t-sne.html"><i class="fa fa-check"></i><b>8.2</b> t-Stocastic Neighbor Embedding (t-SNE)</a></li>
<li class="chapter" data-level="8.3" data-path="dimensionality-reduction-after-harmony.html"><a href="dimensionality-reduction-after-harmony.html"><i class="fa fa-check"></i><b>8.3</b> Dimensionality Reduction After Harmony</a></li>
<li class="chapter" data-level="8.4" data-path="highlighting-specific-cells-on-an-embedding.html"><a href="highlighting-specific-cells-on-an-embedding.html"><i class="fa fa-check"></i><b>8.4</b> Highlighting specific cells on an embedding</a></li>
<li class="chapter" data-level="8.5" data-path="importing-an-embedding-from-external-software.html"><a href="importing-an-embedding-from-external-software.html"><i class="fa fa-check"></i><b>8.5</b> Importing an embedding from external software</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="gene-scores-and-marker-genes-with-archr.html"><a href="gene-scores-and-marker-genes-with-archr.html"><i class="fa fa-check"></i><b>9</b> Gene Scores and Marker Genes with ArchR</a>
<ul>
<li class="chapter" data-level="9.1" data-path="calculating-gene-scores-in-archr.html"><a href="calculating-gene-scores-in-archr.html"><i class="fa fa-check"></i><b>9.1</b> Calculating Gene Scores in ArchR</a></li>
<li class="chapter" data-level="9.2" data-path="identification-of-marker-features.html"><a href="identification-of-marker-features.html"><i class="fa fa-check"></i><b>9.2</b> Identification of Marker Features</a></li>
<li class="chapter" data-level="9.3" data-path="identifying-marker-genes.html"><a href="identifying-marker-genes.html"><i class="fa fa-check"></i><b>9.3</b> Identifying Marker Genes</a></li>
<li class="chapter" data-level="9.4" data-path="visualizing-marker-genes-on-an-embedding.html"><a href="visualizing-marker-genes-on-an-embedding.html"><i class="fa fa-check"></i><b>9.4</b> Visualizing Marker Genes on an Embedding</a></li>
<li class="chapter" data-level="9.5" data-path="marker-genes-imputation-with-magic.html"><a href="marker-genes-imputation-with-magic.html"><i class="fa fa-check"></i><b>9.5</b> Marker Genes Imputation with MAGIC</a></li>
<li class="chapter" data-level="9.6" data-path="module-scores.html"><a href="module-scores.html"><i class="fa fa-check"></i><b>9.6</b> Module Scores</a></li>
<li class="chapter" data-level="9.7" data-path="track-plotting-with-archrbrowser.html"><a href="track-plotting-with-archrbrowser.html"><i class="fa fa-check"></i><b>9.7</b> Track Plotting with ArchRBrowser</a></li>
<li class="chapter" data-level="9.8" data-path="launching-the-interactive-archrbrowser.html"><a href="launching-the-interactive-archrbrowser.html"><i class="fa fa-check"></i><b>9.8</b> Launching the Interactive ArchRBrowser</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="defining-cluster-identity-with-scrna-seq.html"><a href="defining-cluster-identity-with-scrna-seq.html"><i class="fa fa-check"></i><b>10</b> Defining Cluster Identity with scRNA-seq</a>
<ul>
<li class="chapter" data-level="10.1" data-path="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html"><i class="fa fa-check"></i><b>10.1</b> Cross-platform linkage of scATAC-seq cells with scRNA-seq cells</a></li>
<li class="chapter" data-level="10.2" data-path="adding-pseudo-scrna-seq-profiles-for-each-scatac-seq-cell.html"><a href="adding-pseudo-scrna-seq-profiles-for-each-scatac-seq-cell.html"><i class="fa fa-check"></i><b>10.2</b> Adding Pseudo-scRNA-seq profiles for each scATAC-seq cell</a></li>
<li class="chapter" data-level="10.3" data-path="labeling-scatac-seq-clusters-with-scrna-seq-information.html"><a href="labeling-scatac-seq-clusters-with-scrna-seq-information.html"><i class="fa fa-check"></i><b>10.3</b> Labeling scATAC-seq clusters with scRNA-seq information</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="pseudo-bulk-replicates-in-archr.html"><a href="pseudo-bulk-replicates-in-archr.html"><i class="fa fa-check"></i><b>11</b> Pseudo-bulk Replicates in ArchR</a>
<ul>
<li class="chapter" data-level="11.1" data-path="how-does-archr-make-pseudo-bulk-replicates.html"><a href="how-does-archr-make-pseudo-bulk-replicates.html"><i class="fa fa-check"></i><b>11.1</b> How Does ArchR Make Pseudo-bulk Replicates?</a></li>
<li class="chapter" data-level="11.2" data-path="making-pseudo-bulk-replicates.html"><a href="making-pseudo-bulk-replicates.html"><i class="fa fa-check"></i><b>11.2</b> Making Pseudo-bulk Replicates</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="calling-peaks-with-archr.html"><a href="calling-peaks-with-archr.html"><i class="fa fa-check"></i><b>12</b> Calling Peaks with ArchR</a>
<ul>
<li class="chapter" data-level="12.1" data-path="the-iterative-overlap-peak-merging-procedure.html"><a href="the-iterative-overlap-peak-merging-procedure.html"><i class="fa fa-check"></i><b>12.1</b> The Iterative Overlap Peak Merging Procedure</a></li>
<li class="chapter" data-level="12.2" data-path="calling-peaks-w-macs2.html"><a href="calling-peaks-w-macs2.html"><i class="fa fa-check"></i><b>12.2</b> Calling Peaks w/ Macs2</a></li>
<li class="chapter" data-level="12.3" data-path="calling-peaks-w-tilematrix.html"><a href="calling-peaks-w-tilematrix.html"><i class="fa fa-check"></i><b>12.3</b> Calling Peaks w/ TileMatrix</a></li>
<li class="chapter" data-level="12.4" data-path="adding-a-peak-matrix.html"><a href="adding-a-peak-matrix.html"><i class="fa fa-check"></i><b>12.4</b> Adding a Peak Matrix</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="identifying-marker-peaks.html"><a href="identifying-marker-peaks.html"><i class="fa fa-check"></i><b>13</b> Identifying Marker Peaks</a>
<ul>
<li class="chapter" data-level="13.1" data-path="identifying-marker-peaks-with-archr.html"><a href="identifying-marker-peaks-with-archr.html"><i class="fa fa-check"></i><b>13.1</b> Identifying Marker Peaks with ArchR</a></li>
<li class="chapter" data-level="13.2" data-path="plotting-marker-peaks-in-archr.html"><a href="plotting-marker-peaks-in-archr.html"><i class="fa fa-check"></i><b>13.2</b> Plotting Marker Peaks in ArchR</a></li>
<li class="chapter" data-level="13.3" data-path="pairwise-testing-between-groups.html"><a href="pairwise-testing-between-groups.html"><i class="fa fa-check"></i><b>13.3</b> Pairwise Testing Between Groups</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="motif-and-feature-enrichment-with-archr.html"><a href="motif-and-feature-enrichment-with-archr.html"><i class="fa fa-check"></i><b>14</b> Motif and Feature Enrichment with ArchR</a>
<ul>
<li class="chapter" data-level="14.1" data-path="motif-enrichment.html"><a href="motif-enrichment.html"><i class="fa fa-check"></i><b>14.1</b> Motif Enrichment</a></li>
<li class="chapter" data-level="14.2" data-path="archr-enrichment.html"><a href="archr-enrichment.html"><i class="fa fa-check"></i><b>14.2</b> ArchR Enrichment</a></li>
<li class="chapter" data-level="14.3" data-path="custom-enrichment.html"><a href="custom-enrichment.html"><i class="fa fa-check"></i><b>14.3</b> Custom Enrichment</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="chromvar-deviatons-enrichment-with-archr.html"><a href="chromvar-deviatons-enrichment-with-archr.html"><i class="fa fa-check"></i><b>15</b> ChromVAR Deviatons Enrichment with ArchR</a>
<ul>
<li class="chapter" data-level="15.1" data-path="motif-deviations.html"><a href="motif-deviations.html"><i class="fa fa-check"></i><b>15.1</b> Motif Deviations</a></li>
<li class="chapter" data-level="15.2" data-path="archr-and-custom-deviations.html"><a href="archr-and-custom-deviations.html"><i class="fa fa-check"></i><b>15.2</b> ArchR and Custom Deviations</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="footprinting-with-archr.html"><a href="footprinting-with-archr.html"><i class="fa fa-check"></i><b>16</b> Footprinting with ArchR</a>
<ul>
<li class="chapter" data-level="16.1" data-path="motif-footprinting.html"><a href="motif-footprinting.html"><i class="fa fa-check"></i><b>16.1</b> Motif Footprinting</a></li>
<li class="chapter" data-level="16.2" data-path="normalization-of-footprints-for-tn5-bias.html"><a href="normalization-of-footprints-for-tn5-bias.html"><i class="fa fa-check"></i><b>16.2</b> Normalization of Footprints for Tn5 Bias</a></li>
<li class="chapter" data-level="16.3" data-path="feature-footprinting.html"><a href="feature-footprinting.html"><i class="fa fa-check"></i><b>16.3</b> Feature Footprinting</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="integrative-analysis-with-archr.html"><a href="integrative-analysis-with-archr.html"><i class="fa fa-check"></i><b>17</b> Integrative Analysis with ArchR</a>
<ul>
<li class="chapter" data-level="17.1" data-path="creating-low-overlapping-aggregates-of-cells.html"><a href="creating-low-overlapping-aggregates-of-cells.html"><i class="fa fa-check"></i><b>17.1</b> Creating Low-Overlapping Aggregates of Cells</a></li>
<li class="chapter" data-level="17.2" data-path="co-accessibility-with-archr.html"><a href="co-accessibility-with-archr.html"><i class="fa fa-check"></i><b>17.2</b> Co-accessibility with ArchR</a></li>
<li class="chapter" data-level="17.3" data-path="peak2genelinkage-with-archr.html"><a href="peak2genelinkage-with-archr.html"><i class="fa fa-check"></i><b>17.3</b> Peak2GeneLinkage with ArchR</a></li>
<li class="chapter" data-level="17.4" data-path="identification-of-positive-tf-regulators.html"><a href="identification-of-positive-tf-regulators.html"><i class="fa fa-check"></i><b>17.4</b> Identification of Positive TF-Regulators</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="trajectory-analysis-with-archr.html"><a href="trajectory-analysis-with-archr.html"><i class="fa fa-check"></i><b>18</b> Trajectory Analysis with ArchR</a>
<ul>
<li class="chapter" data-level="18.1" data-path="myeloid-trajectory---monocyte-differentiation.html"><a href="myeloid-trajectory---monocyte-differentiation.html"><i class="fa fa-check"></i><b>18.1</b> Myeloid Trajectory - Monocyte Differentiation</a></li>
<li class="chapter" data-level="18.2" data-path="lymphoid-trajectory---b-cell-cifferentiation.html"><a href="lymphoid-trajectory---b-cell-cifferentiation.html"><i class="fa fa-check"></i><b>18.2</b> Lymphoid Trajectory - B Cell Cifferentiation</a></li>
<li class="chapter" data-level="18.3" data-path="monocle3-trajectories.html"><a href="monocle3-trajectories.html"><i class="fa fa-check"></i><b>18.3</b> Monocle3 Trajectories</a></li>
<li class="chapter" data-level="18.4" data-path="slingshot-trajectories.html"><a href="slingshot-trajectories.html"><i class="fa fa-check"></i><b>18.4</b> Slingshot Trajectories</a></li>
<li class="chapter" data-level="18.5" data-path="comparing-the-different-trajectories.html"><a href="comparing-the-different-trajectories.html"><i class="fa fa-check"></i><b>18.5</b> Comparing the different trajectories</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="handling-archr-output.html"><a href="handling-archr-output.html"><i class="fa fa-check"></i><b>19</b> Handling ArchR Output</a>
<ul>
<li class="chapter" data-level="19.1" data-path="exporting-fragment-level-data.html"><a href="exporting-fragment-level-data.html"><i class="fa fa-check"></i><b>19.1</b> Exporting fragment-level data</a></li>
<li class="chapter" data-level="19.2" data-path="exporting-matrix-level-data.html"><a href="exporting-matrix-level-data.html"><i class="fa fa-check"></i><b>19.2</b> Exporting matrix-level data</a></li>
<li class="chapter" data-level="19.3" data-path="exporting-pseudo-bulked-data-to-a-summarizedexperiment.html"><a href="exporting-pseudo-bulked-data-to-a-summarizedexperiment.html"><i class="fa fa-check"></i><b>19.3</b> Exporting pseudo-bulked data to a <code>SummarizedExperiment</code></a></li>
<li class="chapter" data-level="19.4" data-path="exporting-pseudo-bulked-data-to-a-bigwig-file.html"><a href="exporting-pseudo-bulked-data-to-a-bigwig-file.html"><i class="fa fa-check"></i><b>19.4</b> Exporting pseudo-bulked data to a bigWig file</a></li>
</ul></li>
<li class="chapter" data-level="20" data-path="integration-with-bulk-atac-seq.html"><a href="integration-with-bulk-atac-seq.html"><i class="fa fa-check"></i><b>20</b> Integration with bulk ATAC-seq</a>
<ul>
<li class="chapter" data-level="20.1" data-path="projecting-bulk-atac-seq-data.html"><a href="projecting-bulk-atac-seq-data.html"><i class="fa fa-check"></i><b>20.1</b> Projecting bulk ATAC-seq data</a></li>
</ul></li>
<li class="chapter" data-level="21" data-path="multiomic-data-analysis-in-archr.html"><a href="multiomic-data-analysis-in-archr.html"><i class="fa fa-check"></i><b>21</b> Multiomic data analysis in ArchR</a>
<ul>
<li class="chapter" data-level="21.1" data-path="importing-data-and-setting-up-a-multiome-project.html"><a href="importing-data-and-setting-up-a-multiome-project.html"><i class="fa fa-check"></i><b>21.1</b> Importing data and setting up a Multiome project</a></li>
<li class="chapter" data-level="21.2" data-path="analysis-of-multiome-data-in-archr.html"><a href="analysis-of-multiome-data-in-archr.html"><i class="fa fa-check"></i><b>21.2</b> Analysis of multiome data in ArchR</a></li>
</ul></li>
<li class="chapter" data-level="22" data-path="plot-aesthetics-in-archr.html"><a href="plot-aesthetics-in-archr.html"><i class="fa fa-check"></i><b>22</b> Plot aesthetics in ArchR</a>
<ul>
<li class="chapter" data-level="22.1" data-path="using-and-manipulating-palettes-in-archr.html"><a href="using-and-manipulating-palettes-in-archr.html"><i class="fa fa-check"></i><b>22.1</b> Using and manipulating palettes in ArchR</a></li>
</ul></li>
<li class="chapter" data-level="23" data-path="testing-your-archr-installation.html"><a href="testing-your-archr-installation.html"><i class="fa fa-check"></i><b>23</b> Testing Your ArchR Installation</a></li>
<li class="chapter" data-level="24" data-path="session-information.html"><a href="session-information.html"><i class="fa fa-check"></i><b>24</b> Session Information</a></li>
<li class="divider"></li>
<li><a href="https://bookdown.org" target="_blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">ArchR: Robust and scaleable analysis of single-cell chromatin accessibility data.</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="track-plotting-with-archrbrowser" class="section level2 hasAnchor" number="9.7">
<h2><span class="header-section-number">9.7</span> Track Plotting with ArchRBrowser<a href="track-plotting-with-archrbrowser.html#track-plotting-with-archrbrowser" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>In addition to plotting gene scores per cell as a UMAP overlay, we can browse the local chromatin accessibility at these marker genes on a per cluster basis with genome browser tracks. To do this, we use the <code>plotBrowserTrack()</code> function which will create a list of plots, one for each of the genes specified by <code>markerGenes</code>. This function will plot a single track for each group in the <code>groupBy</code> parameter.</p>
<p>It is worth noting that the <code>plotBrowserTrack()</code> function plots “pseudobulk” level data from the single-cell ATAC-seq data by grouping many individual cells together. The cells selected are random and the number of cells selected and the cell groupings that can be displayed are based on the <code>maxCells</code> and <code>minCells</code> parameters respectively. As such, to ensure reproducibility of the functionality of the <code>plotBrowserTrack()</code> function, you should make sure to set a consistent seed for random number generation prior to plotting.</p>
<div class="sourceCode" id="cb140"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb140-1"><a href="track-plotting-with-archrbrowser.html#cb140-1" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">1</span>)</span>
<span id="cb140-2"><a href="track-plotting-with-archrbrowser.html#cb140-2" tabindex="-1"></a></span>
<span id="cb140-3"><a href="track-plotting-with-archrbrowser.html#cb140-3" tabindex="-1"></a>markerGenes  <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb140-4"><a href="track-plotting-with-archrbrowser.html#cb140-4" tabindex="-1"></a>    <span class="st">&quot;CD34&quot;</span>, <span class="co">#Early Progenitor</span></span>
<span id="cb140-5"><a href="track-plotting-with-archrbrowser.html#cb140-5" tabindex="-1"></a>    <span class="st">&quot;GATA1&quot;</span>, <span class="co">#Erythroid</span></span>
<span id="cb140-6"><a href="track-plotting-with-archrbrowser.html#cb140-6" tabindex="-1"></a>    <span class="st">&quot;PAX5&quot;</span>, <span class="st">&quot;MS4A1&quot;</span>, <span class="co">#B-Cell Trajectory</span></span>
<span id="cb140-7"><a href="track-plotting-with-archrbrowser.html#cb140-7" tabindex="-1"></a>    <span class="st">&quot;CD14&quot;</span>, <span class="co">#Monocytes</span></span>
<span id="cb140-8"><a href="track-plotting-with-archrbrowser.html#cb140-8" tabindex="-1"></a>    <span class="st">&quot;CD3D&quot;</span>, <span class="st">&quot;CD8A&quot;</span>, <span class="st">&quot;TBX21&quot;</span>, <span class="st">&quot;IL7R&quot;</span> <span class="co">#TCells</span></span>
<span id="cb140-9"><a href="track-plotting-with-archrbrowser.html#cb140-9" tabindex="-1"></a>  )</span>
<span id="cb140-10"><a href="track-plotting-with-archrbrowser.html#cb140-10" tabindex="-1"></a></span>
<span id="cb140-11"><a href="track-plotting-with-archrbrowser.html#cb140-11" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">plotBrowserTrack</span>(</span>
<span id="cb140-12"><a href="track-plotting-with-archrbrowser.html#cb140-12" tabindex="-1"></a>    <span class="at">ArchRProj =</span> projHeme2, </span>
<span id="cb140-13"><a href="track-plotting-with-archrbrowser.html#cb140-13" tabindex="-1"></a>    <span class="at">groupBy =</span> <span class="st">&quot;Clusters&quot;</span>, </span>
<span id="cb140-14"><a href="track-plotting-with-archrbrowser.html#cb140-14" tabindex="-1"></a>    <span class="at">geneSymbol =</span> markerGenes, </span>
<span id="cb140-15"><a href="track-plotting-with-archrbrowser.html#cb140-15" tabindex="-1"></a>    <span class="at">upstream =</span> <span class="dv">50000</span>,</span>
<span id="cb140-16"><a href="track-plotting-with-archrbrowser.html#cb140-16" tabindex="-1"></a>    <span class="at">downstream =</span> <span class="dv">50000</span></span>
<span id="cb140-17"><a href="track-plotting-with-archrbrowser.html#cb140-17" tabindex="-1"></a>)</span>
<span id="cb140-18"><a href="track-plotting-with-archrbrowser.html#cb140-18" tabindex="-1"></a><span class="do">## ArchR logging to : ArchRLogs/ArchR-plotBrowserTrack-1a16462d18d-Date-2025-01-23_Time-20-19-16.837045.log</span></span>
<span id="cb140-19"><a href="track-plotting-with-archrbrowser.html#cb140-19" tabindex="-1"></a><span class="do">## If there is an issue, please report to github with logFile!</span></span>
<span id="cb140-20"><a href="track-plotting-with-archrbrowser.html#cb140-20" tabindex="-1"></a><span class="do">## 2025-01-23 20:19:17.501278 : Validating Region, 0.011 mins elapsed.</span></span>
<span id="cb140-21"><a href="track-plotting-with-archrbrowser.html#cb140-21" tabindex="-1"></a><span class="do">## GRanges object with 9 ranges and 2 metadata columns:</span></span>
<span id="cb140-22"><a href="track-plotting-with-archrbrowser.html#cb140-22" tabindex="-1"></a><span class="do">##       seqnames              ranges strand |     gene_id      symbol</span></span>
<span id="cb140-23"><a href="track-plotting-with-archrbrowser.html#cb140-23" tabindex="-1"></a><span class="do">##          &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt;</span></span>
<span id="cb140-24"><a href="track-plotting-with-archrbrowser.html#cb140-24" tabindex="-1"></a><span class="do">##   [1]     chr1 208059883-208084683      - |         947        CD34</span></span>
<span id="cb140-25"><a href="track-plotting-with-archrbrowser.html#cb140-25" tabindex="-1"></a><span class="do">##   [2]     chrX   48644982-48652717      + |        2623       GATA1</span></span>
<span id="cb140-26"><a href="track-plotting-with-archrbrowser.html#cb140-26" tabindex="-1"></a><span class="do">##   [3]     chr9   36838531-37034476      - |        5079        PAX5</span></span>
<span id="cb140-27"><a href="track-plotting-with-archrbrowser.html#cb140-27" tabindex="-1"></a><span class="do">##   [4]    chr11   60223282-60238225      + |         931       MS4A1</span></span>
<span id="cb140-28"><a href="track-plotting-with-archrbrowser.html#cb140-28" tabindex="-1"></a><span class="do">##   [5]     chr5 140011313-140013286      - |         929        CD14</span></span>
<span id="cb140-29"><a href="track-plotting-with-archrbrowser.html#cb140-29" tabindex="-1"></a><span class="do">##   [6]    chr11 118209789-118213459      - |         915        CD3D</span></span>
<span id="cb140-30"><a href="track-plotting-with-archrbrowser.html#cb140-30" tabindex="-1"></a><span class="do">##   [7]     chr2   87011728-87035519      - |         925        CD8A</span></span>
<span id="cb140-31"><a href="track-plotting-with-archrbrowser.html#cb140-31" tabindex="-1"></a><span class="do">##   [8]    chr17   45810610-45823485      + |       30009       TBX21</span></span>
<span id="cb140-32"><a href="track-plotting-with-archrbrowser.html#cb140-32" tabindex="-1"></a><span class="do">##   [9]     chr5   35856977-35879705      + |        3575        IL7R</span></span>
<span id="cb140-33"><a href="track-plotting-with-archrbrowser.html#cb140-33" tabindex="-1"></a><span class="do">##   -------</span></span>
<span id="cb140-34"><a href="track-plotting-with-archrbrowser.html#cb140-34" tabindex="-1"></a><span class="do">##   seqinfo: 24 sequences from hg19 genome</span></span>
<span id="cb140-35"><a href="track-plotting-with-archrbrowser.html#cb140-35" tabindex="-1"></a><span class="do">## 2025-01-23 20:19:17.584247 : Adding Bulk Tracks (1 of 9), 0.012 mins elapsed.</span></span>
<span id="cb140-36"><a href="track-plotting-with-archrbrowser.html#cb140-36" tabindex="-1"></a><span class="do">## 2025-01-23 20:19:18.921915 : Adding Gene Tracks (1 of 9), 0.035 mins elapsed.</span></span>
<span id="cb140-37"><a href="track-plotting-with-archrbrowser.html#cb140-37" tabindex="-1"></a><span class="do">## 2025-01-23 20:19:19.195664 : Plotting, 0.039 mins elapsed.</span></span>
<span id="cb140-38"><a href="track-plotting-with-archrbrowser.html#cb140-38" tabindex="-1"></a><span class="do">## 2025-01-23 20:19:21.503763 : Adding Bulk Tracks (2 of 9), 0.078 mins elapsed.</span></span>
<span id="cb140-39"><a href="track-plotting-with-archrbrowser.html#cb140-39" tabindex="-1"></a><span class="do">## 2025-01-23 20:19:19.067302 : Adding Gene Tracks (2 of 9), 0.037 mins elapsed.</span></span>
<span id="cb140-40"><a href="track-plotting-with-archrbrowser.html#cb140-40" tabindex="-1"></a><span class="do">## 2025-01-23 20:19:19.402905 : Plotting, 0.043 mins elapsed.</span></span>
<span id="cb140-41"><a href="track-plotting-with-archrbrowser.html#cb140-41" tabindex="-1"></a><span class="do">## 2025-01-23 20:19:20.157611 : Adding Bulk Tracks (3 of 9), 0.055 mins elapsed.</span></span>
<span id="cb140-42"><a href="track-plotting-with-archrbrowser.html#cb140-42" tabindex="-1"></a><span class="do">## 2025-01-23 20:19:21.392212 : Adding Gene Tracks (3 of 9), 0.076 mins elapsed.</span></span>
<span id="cb140-43"><a href="track-plotting-with-archrbrowser.html#cb140-43" tabindex="-1"></a><span class="do">## 2025-01-23 20:19:21.603658 : Plotting, 0.079 mins elapsed.</span></span>
<span id="cb140-44"><a href="track-plotting-with-archrbrowser.html#cb140-44" tabindex="-1"></a><span class="do">## 2025-01-23 20:19:22.483531 : Adding Bulk Tracks (4 of 9), 0.094 mins elapsed.</span></span>
<span id="cb140-45"><a href="track-plotting-with-archrbrowser.html#cb140-45" tabindex="-1"></a><span class="do">## 2025-01-23 20:19:23.732543 : Adding Gene Tracks (4 of 9), 0.115 mins elapsed.</span></span>
<span id="cb140-46"><a href="track-plotting-with-archrbrowser.html#cb140-46" tabindex="-1"></a><span class="do">## 2025-01-23 20:19:23.948389 : Plotting, 0.119 mins elapsed.</span></span>
<span id="cb140-47"><a href="track-plotting-with-archrbrowser.html#cb140-47" tabindex="-1"></a><span class="do">## 2025-01-23 20:19:24.627934 : Adding Bulk Tracks (5 of 9), 0.13 mins elapsed.</span></span>
<span id="cb140-48"><a href="track-plotting-with-archrbrowser.html#cb140-48" tabindex="-1"></a><span class="do">## 2025-01-23 20:19:25.889378 : Adding Gene Tracks (5 of 9), 0.151 mins elapsed.</span></span>
<span id="cb140-49"><a href="track-plotting-with-archrbrowser.html#cb140-49" tabindex="-1"></a><span class="do">## 2025-01-23 20:19:26.101466 : Plotting, 0.154 mins elapsed.</span></span>
<span id="cb140-50"><a href="track-plotting-with-archrbrowser.html#cb140-50" tabindex="-1"></a><span class="do">## 2025-01-23 20:19:26.947893 : Adding Bulk Tracks (6 of 9), 0.169 mins elapsed.</span></span>
<span id="cb140-51"><a href="track-plotting-with-archrbrowser.html#cb140-51" tabindex="-1"></a><span class="do">## 2025-01-23 20:19:28.157929 : Adding Gene Tracks (6 of 9), 0.189 mins elapsed.</span></span>
<span id="cb140-52"><a href="track-plotting-with-archrbrowser.html#cb140-52" tabindex="-1"></a><span class="do">## 2025-01-23 20:19:28.371197 : Plotting, 0.192 mins elapsed.</span></span>
<span id="cb140-53"><a href="track-plotting-with-archrbrowser.html#cb140-53" tabindex="-1"></a><span class="do">## 2025-01-23 20:19:29.11274 : Adding Bulk Tracks (7 of 9), 0.205 mins elapsed.</span></span>
<span id="cb140-54"><a href="track-plotting-with-archrbrowser.html#cb140-54" tabindex="-1"></a><span class="do">## 2025-01-23 20:19:30.466452 : Adding Gene Tracks (7 of 9), 0.227 mins elapsed.</span></span>
<span id="cb140-55"><a href="track-plotting-with-archrbrowser.html#cb140-55" tabindex="-1"></a><span class="do">## 2025-01-23 20:19:30.678786 : Plotting, 0.231 mins elapsed.</span></span>
<span id="cb140-56"><a href="track-plotting-with-archrbrowser.html#cb140-56" tabindex="-1"></a><span class="do">## 2025-01-23 20:19:31.453751 : Adding Bulk Tracks (8 of 9), 0.244 mins elapsed.</span></span>
<span id="cb140-57"><a href="track-plotting-with-archrbrowser.html#cb140-57" tabindex="-1"></a><span class="do">## 2025-01-23 20:19:32.735101 : Adding Gene Tracks (8 of 9), 0.265 mins elapsed.</span></span>
<span id="cb140-58"><a href="track-plotting-with-archrbrowser.html#cb140-58" tabindex="-1"></a><span class="do">## 2025-01-23 20:19:32.950608 : Plotting, 0.269 mins elapsed.</span></span>
<span id="cb140-59"><a href="track-plotting-with-archrbrowser.html#cb140-59" tabindex="-1"></a><span class="do">## 2025-01-23 20:19:33.843326 : Adding Bulk Tracks (9 of 9), 0.283 mins elapsed.</span></span>
<span id="cb140-60"><a href="track-plotting-with-archrbrowser.html#cb140-60" tabindex="-1"></a><span class="do">## 2025-01-23 20:19:35.041549 : Adding Gene Tracks (9 of 9), 0.303 mins elapsed.</span></span>
<span id="cb140-61"><a href="track-plotting-with-archrbrowser.html#cb140-61" tabindex="-1"></a><span class="do">## 2025-01-23 20:19:35.241047 : Plotting, 0.307 mins elapsed.</span></span>
<span id="cb140-62"><a href="track-plotting-with-archrbrowser.html#cb140-62" tabindex="-1"></a><span class="do">## ArchR logging successful to : ArchRLogs/ArchR-plotBrowserTrack-1a16462d18d-Date-2025-01-23_Time-20-19-16.837045.log</span></span></code></pre></div>
<p>Here, the output of <code>plotBrowserTrack()</code> is a list of plots. To plot a track of a specific gene, we can select one from the list.</p>
<div class="sourceCode" id="cb141"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb141-1"><a href="track-plotting-with-archrbrowser.html#cb141-1" tabindex="-1"></a>grid<span class="sc">::</span><span class="fu">grid.newpage</span>()</span>
<span id="cb141-2"><a href="track-plotting-with-archrbrowser.html#cb141-2" tabindex="-1"></a>grid<span class="sc">::</span><span class="fu">grid.draw</span>(p<span class="sc">$</span>CD14)</span></code></pre></div>
<p><img src="ArchR_files/figure-html/unnamed-chunk-138-1.svg" width="672" /></p>
<p>We can save a multi-page PDF with a single page for each gene locus in our plot list using the <code>plotPDF()</code> function.</p>
<div class="sourceCode" id="cb142"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb142-1"><a href="track-plotting-with-archrbrowser.html#cb142-1" tabindex="-1"></a><span class="fu">plotPDF</span>(<span class="at">plotList =</span> p, </span>
<span id="cb142-2"><a href="track-plotting-with-archrbrowser.html#cb142-2" tabindex="-1"></a>    <span class="at">name =</span> <span class="st">&quot;Plot-Tracks-Marker-Genes.pdf&quot;</span>, </span>
<span id="cb142-3"><a href="track-plotting-with-archrbrowser.html#cb142-3" tabindex="-1"></a>    <span class="at">ArchRProj =</span> projHeme2, </span>
<span id="cb142-4"><a href="track-plotting-with-archrbrowser.html#cb142-4" tabindex="-1"></a>    <span class="at">addDOC =</span> <span class="cn">FALSE</span>, <span class="at">width =</span> <span class="dv">5</span>, <span class="at">height =</span> <span class="dv">5</span>)</span>
<span id="cb142-5"><a href="track-plotting-with-archrbrowser.html#cb142-5" tabindex="-1"></a><span class="do">## Plotting Gtable!</span></span>
<span id="cb142-6"><a href="track-plotting-with-archrbrowser.html#cb142-6" tabindex="-1"></a><span class="do">## NULL</span></span>
<span id="cb142-7"><a href="track-plotting-with-archrbrowser.html#cb142-7" tabindex="-1"></a><span class="do">## Plotting Gtable!</span></span>
<span id="cb142-8"><a href="track-plotting-with-archrbrowser.html#cb142-8" tabindex="-1"></a><span class="do">## NULL</span></span>
<span id="cb142-9"><a href="track-plotting-with-archrbrowser.html#cb142-9" tabindex="-1"></a><span class="do">## Plotting Gtable!</span></span>
<span id="cb142-10"><a href="track-plotting-with-archrbrowser.html#cb142-10" tabindex="-1"></a><span class="do">## NULL</span></span>
<span id="cb142-11"><a href="track-plotting-with-archrbrowser.html#cb142-11" tabindex="-1"></a><span class="do">## Plotting Gtable!</span></span>
<span id="cb142-12"><a href="track-plotting-with-archrbrowser.html#cb142-12" tabindex="-1"></a><span class="do">## NULL</span></span>
<span id="cb142-13"><a href="track-plotting-with-archrbrowser.html#cb142-13" tabindex="-1"></a><span class="do">## Plotting Gtable!</span></span>
<span id="cb142-14"><a href="track-plotting-with-archrbrowser.html#cb142-14" tabindex="-1"></a><span class="do">## NULL</span></span>
<span id="cb142-15"><a href="track-plotting-with-archrbrowser.html#cb142-15" tabindex="-1"></a><span class="do">## Plotting Gtable!</span></span>
<span id="cb142-16"><a href="track-plotting-with-archrbrowser.html#cb142-16" tabindex="-1"></a><span class="do">## NULL</span></span>
<span id="cb142-17"><a href="track-plotting-with-archrbrowser.html#cb142-17" tabindex="-1"></a><span class="do">## Plotting Gtable!</span></span>
<span id="cb142-18"><a href="track-plotting-with-archrbrowser.html#cb142-18" tabindex="-1"></a><span class="do">## NULL</span></span>
<span id="cb142-19"><a href="track-plotting-with-archrbrowser.html#cb142-19" tabindex="-1"></a><span class="do">## Plotting Gtable!</span></span>
<span id="cb142-20"><a href="track-plotting-with-archrbrowser.html#cb142-20" tabindex="-1"></a><span class="do">## NULL</span></span>
<span id="cb142-21"><a href="track-plotting-with-archrbrowser.html#cb142-21" tabindex="-1"></a><span class="do">## Plotting Gtable!</span></span>
<span id="cb142-22"><a href="track-plotting-with-archrbrowser.html#cb142-22" tabindex="-1"></a><span class="do">## NULL</span></span></code></pre></div>
<p>The <code>plotBrowserTrack()</code> function is highly versatile and therefore has a lot of different parameter options so you should explore all of those capabilities in the <a href="https://www.archrproject.com/reference/plotBrowserTrack.html">function documentation</a>. There are multiple different types of plot elements that can be plotted and these are controlled using the <code>plotSummary</code> parameter. Those options include:</p>
<ol style="list-style-type: decimal">
<li><code>bulkTrack</code> - A pseudobulk representation of the ATAC-seq signal in the given region</li>
<li><code>scTrack</code> - The per-cell integration signal scATAC-seq signal. Each row represents an individual cell. The number of cells plotted per cell grouping is controlled by the <code>scCellsMax</code> parameter</li>
<li><code>featureTrack</code> - Regions to be shown as blocks below the tracks (i.e peak regions, or SNPs, or some other feature</li>
<li><code>geneTrack</code> - Line diagrams of genes with introns and exons shown. We’ve adopted the convention of plotting “plus-stranded” genes (those that point left-to-right with the TSS on the left) in red and “minus-stranded” genes (those that point right-to-left with the TSS on the right) in blue.</li>
<li><code>loopTrack</code> - Arcs showing links between a peak and a gene, for example from co-accessibility or peak-to-gene linkage analysis. The actual loops plotted is controlled by the <code>loops</code> parameter.</li>
</ol>
<p>For example, instead of providing a set of genes to plot via <code>geneSymbol</code>, you could provide a set of regions to plot as a <code>GRanges</code> object to the <code>region</code> parameter.</p>
<p>You can also change which regions are shown as “features” by manipulating the <code>features</code> paramter. The default is to show the <code>peakSet</code> of your <code>ArchRProject</code> but you can provide a <code>GRanges</code> (for a single feature track) or a <strong>named</strong> <code>GRangesList</code> (for multiple feature tracks). If you provide a <code>GRangesList</code>, its best to provide a named list because the names will be used to label the individual feature tracks on the plot.</p>
<p>By default, the pseudobulk ATAC-seq tracks will be normalized based on <code>ReadsInTSS</code> (from <code>cellColData</code>) but this can be changed via the <code>normMethod</code> parameter if desired. Lastly, you can highlight specific regions in the plots using the <code>hightlight</code> and <code>highlightFill</code> paramters. There are tons of ways to use the <code>plotBrowserTrack()</code> function, far more than can be adequately reviewed in this manual so take the time to explore.</p>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="module-scores.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="launching-the-interactive-archrbrowser.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": null
},
"fontsettings": null,
"edit": null,
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "none"
},
"info": false
});
});
</script>

</body>

</html>
