<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>3.8 Using BAM files for Arrow File creation | ArchR: Robust and scaleable analysis of single-cell chromatin accessibility data.</title>
  <meta name="description" content="A guide to ArchR" />
  <meta name="generator" content="bookdown 0.31 and GitBook 2.6.7" />

  <meta property="og:title" content="3.8 Using BAM files for Arrow File creation | ArchR: Robust and scaleable analysis of single-cell chromatin accessibility data." />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="A guide to ArchR" />
  <meta name="github-repo" content="GreenleafLab/ArchR" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="3.8 Using BAM files for Arrow File creation | ArchR: Robust and scaleable analysis of single-cell chromatin accessibility data." />
  
  <meta name="twitter:description" content="A guide to ArchR" />
  

<meta name="author" content="Jeffrey Granja and Ryan Corces" />


<meta name="date" content="2022-12-23" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="images/favicon.ico" type="image/x-icon" />
<link rel="prev" href="per-cell-quality-control.html"/>
<link rel="next" href="doublet-inference-with-archr.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<link href="libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="../index.html">Back to ArchR Home</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><img src="images/ArchR_Logo_Integrated.png" alt="" width="200"></a></li>
<li class="chapter" data-level="1" data-path="debugging-and-troubleshooting-in-archr.html"><a href="debugging-and-troubleshooting-in-archr.html"><i class="fa fa-check"></i><b>1</b> Debugging and Troubleshooting in ArchR</a></li>
<li class="chapter" data-level="2" data-path="manage-archrs-dependencies.html"><a href="manage-archrs-dependencies.html"><i class="fa fa-check"></i><b>2</b> Manage ArchR’s Dependencies</a>
<ul>
<li class="chapter" data-level="2.1" data-path="using-renv-to-manage-dependencies.html"><a href="using-renv-to-manage-dependencies.html"><i class="fa fa-check"></i><b>2.1</b> Using renv to manage dependencies</a></li>
<li class="chapter" data-level="2.2" data-path="other-environment-managers.html"><a href="other-environment-managers.html"><i class="fa fa-check"></i><b>2.2</b> Other environment managers</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="getting-started-with-archr.html"><a href="getting-started-with-archr.html"><i class="fa fa-check"></i><b>3</b> Getting Started with ArchR</a>
<ul>
<li class="chapter" data-level="3.1" data-path="a-brief-primer-on-atac-seq-terminology.html"><a href="a-brief-primer-on-atac-seq-terminology.html"><i class="fa fa-check"></i><b>3.1</b> A Brief Primer on ATAC-seq Terminology</a></li>
<li class="chapter" data-level="3.2" data-path="why-use-archr.html"><a href="why-use-archr.html"><i class="fa fa-check"></i><b>3.2</b> Why use ArchR?</a></li>
<li class="chapter" data-level="3.3" data-path="what-is-an-arrow-file-archrproject.html"><a href="what-is-an-arrow-file-archrproject.html"><i class="fa fa-check"></i><b>3.3</b> What is an Arrow file / <code>ArchRProject</code>?</a></li>
<li class="chapter" data-level="3.4" data-path="input-file-types-in-archr.html"><a href="input-file-types-in-archr.html"><i class="fa fa-check"></i><b>3.4</b> Input File Types in ArchR</a></li>
<li class="chapter" data-level="3.5" data-path="getting-set-up.html"><a href="getting-set-up.html"><i class="fa fa-check"></i><b>3.5</b> Getting Set Up</a></li>
<li class="chapter" data-level="3.6" data-path="creating-arrow-files.html"><a href="creating-arrow-files.html"><i class="fa fa-check"></i><b>3.6</b> Creating Arrow Files</a></li>
<li class="chapter" data-level="3.7" data-path="per-cell-quality-control.html"><a href="per-cell-quality-control.html"><i class="fa fa-check"></i><b>3.7</b> Per-cell Quality Control</a></li>
<li class="chapter" data-level="3.8" data-path="using-bam-files-for-arrow-file-creation.html"><a href="using-bam-files-for-arrow-file-creation.html"><i class="fa fa-check"></i><b>3.8</b> Using BAM files for Arrow File creation</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="doublet-inference-with-archr.html"><a href="doublet-inference-with-archr.html"><i class="fa fa-check"></i><b>4</b> Doublet Inference with ArchR</a>
<ul>
<li class="chapter" data-level="4.1" data-path="how-does-doublet-identification-work-in-archr.html"><a href="how-does-doublet-identification-work-in-archr.html"><i class="fa fa-check"></i><b>4.1</b> How does doublet identification work in ArchR?</a></li>
<li class="chapter" data-level="4.2" data-path="inferring-scatac-seq-doublets-with-archr.html"><a href="inferring-scatac-seq-doublets-with-archr.html"><i class="fa fa-check"></i><b>4.2</b> Inferring scATAC-seq Doublets with ArchR</a></li>
<li class="chapter" data-level="4.3" data-path="using-demuxlet-with-archr.html"><a href="using-demuxlet-with-archr.html"><i class="fa fa-check"></i><b>4.3</b> Using <em>demuxlet</em> with ArchR</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="creating-an-archrproject.html"><a href="creating-an-archrproject.html"><i class="fa fa-check"></i><b>5</b> Creating an ArchRProject</a>
<ul>
<li class="chapter" data-level="5.1" data-path="creating-an-archrproject-1.html"><a href="creating-an-archrproject-1.html"><i class="fa fa-check"></i><b>5.1</b> Creating An ArchRProject</a></li>
<li class="chapter" data-level="5.2" data-path="manipulating-an-archrproject.html"><a href="manipulating-an-archrproject.html"><i class="fa fa-check"></i><b>5.2</b> Manipulating An ArchRProject</a></li>
<li class="chapter" data-level="5.3" data-path="plotting-sample-statistics-from-an-archrproject.html"><a href="plotting-sample-statistics-from-an-archrproject.html"><i class="fa fa-check"></i><b>5.3</b> Plotting Sample Statistics from an ArchRProject</a></li>
<li class="chapter" data-level="5.4" data-path="plotting-sample-fragment-size-distribution-and-tss-enrichment-profiles..html"><a href="plotting-sample-fragment-size-distribution-and-tss-enrichment-profiles..html"><i class="fa fa-check"></i><b>5.4</b> Plotting Sample Fragment Size Distribution and TSS Enrichment Profiles.</a></li>
<li class="chapter" data-level="5.5" data-path="saving-and-loading-an-archrproject.html"><a href="saving-and-loading-an-archrproject.html"><i class="fa fa-check"></i><b>5.5</b> Saving and Loading an <code>ArchRProject</code></a></li>
<li class="chapter" data-level="5.6" data-path="filtering-doublets-from-an-archrproject.html"><a href="filtering-doublets-from-an-archrproject.html"><i class="fa fa-check"></i><b>5.6</b> Filtering Doublets from an ArchRProject</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="dimensionality-reduction-with-archr.html"><a href="dimensionality-reduction-with-archr.html"><i class="fa fa-check"></i><b>6</b> Dimensionality Reduction with ArchR</a>
<ul>
<li class="chapter" data-level="6.1" data-path="archrs-lsi-implementation.html"><a href="archrs-lsi-implementation.html"><i class="fa fa-check"></i><b>6.1</b> ArchR’s LSI Implementation</a></li>
<li class="chapter" data-level="6.2" data-path="iterative-latent-semantic-indexing-lsi.html"><a href="iterative-latent-semantic-indexing-lsi.html"><i class="fa fa-check"></i><b>6.2</b> Iterative Latent Semantic Indexing (LSI)</a></li>
<li class="chapter" data-level="6.3" data-path="estimated-lsi.html"><a href="estimated-lsi.html"><i class="fa fa-check"></i><b>6.3</b> Estimated LSI</a></li>
<li class="chapter" data-level="6.4" data-path="batch-effect-correction-wtih-harmony.html"><a href="batch-effect-correction-wtih-harmony.html"><i class="fa fa-check"></i><b>6.4</b> Batch Effect Correction wtih Harmony</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="clustering-with-archr.html"><a href="clustering-with-archr.html"><i class="fa fa-check"></i><b>7</b> Clustering with ArchR</a>
<ul>
<li class="chapter" data-level="7.1" data-path="clustering-using-seurats-findclusters-function.html"><a href="clustering-using-seurats-findclusters-function.html"><i class="fa fa-check"></i><b>7.1</b> Clustering using Seurat’s <code>FindClusters()</code> function</a></li>
<li class="chapter" data-level="7.2" data-path="clustering-using-scran.html"><a href="clustering-using-scran.html"><i class="fa fa-check"></i><b>7.2</b> Clustering using <code>scran</code></a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="single-cell-embeddings.html"><a href="single-cell-embeddings.html"><i class="fa fa-check"></i><b>8</b> Single-cell Embeddings</a>
<ul>
<li class="chapter" data-level="8.1" data-path="uniform-manifold-approximation-and-projection-umap.html"><a href="uniform-manifold-approximation-and-projection-umap.html"><i class="fa fa-check"></i><b>8.1</b> Uniform Manifold Approximation and Projection (UMAP)</a></li>
<li class="chapter" data-level="8.2" data-path="t-stocastic-neighbor-embedding-t-sne.html"><a href="t-stocastic-neighbor-embedding-t-sne.html"><i class="fa fa-check"></i><b>8.2</b> t-Stocastic Neighbor Embedding (t-SNE)</a></li>
<li class="chapter" data-level="8.3" data-path="dimensionality-reduction-after-harmony.html"><a href="dimensionality-reduction-after-harmony.html"><i class="fa fa-check"></i><b>8.3</b> Dimensionality Reduction After Harmony</a></li>
<li class="chapter" data-level="8.4" data-path="highlighting-specific-cells-on-an-embedding.html"><a href="highlighting-specific-cells-on-an-embedding.html"><i class="fa fa-check"></i><b>8.4</b> Highlighting specific cells on an embedding</a></li>
<li class="chapter" data-level="8.5" data-path="importing-an-embedding-from-external-software.html"><a href="importing-an-embedding-from-external-software.html"><i class="fa fa-check"></i><b>8.5</b> Importing an embedding from external software</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="gene-scores-and-marker-genes-with-archr.html"><a href="gene-scores-and-marker-genes-with-archr.html"><i class="fa fa-check"></i><b>9</b> Gene Scores and Marker Genes with ArchR</a>
<ul>
<li class="chapter" data-level="9.1" data-path="calculating-gene-scores-in-archr.html"><a href="calculating-gene-scores-in-archr.html"><i class="fa fa-check"></i><b>9.1</b> Calculating Gene Scores in ArchR</a></li>
<li class="chapter" data-level="9.2" data-path="identification-of-marker-features.html"><a href="identification-of-marker-features.html"><i class="fa fa-check"></i><b>9.2</b> Identification of Marker Features</a></li>
<li class="chapter" data-level="9.3" data-path="identifying-marker-genes.html"><a href="identifying-marker-genes.html"><i class="fa fa-check"></i><b>9.3</b> Identifying Marker Genes</a></li>
<li class="chapter" data-level="9.4" data-path="visualizing-marker-genes-on-an-embedding.html"><a href="visualizing-marker-genes-on-an-embedding.html"><i class="fa fa-check"></i><b>9.4</b> Visualizing Marker Genes on an Embedding</a></li>
<li class="chapter" data-level="9.5" data-path="marker-genes-imputation-with-magic.html"><a href="marker-genes-imputation-with-magic.html"><i class="fa fa-check"></i><b>9.5</b> Marker Genes Imputation with MAGIC</a></li>
<li class="chapter" data-level="9.6" data-path="module-scores.html"><a href="module-scores.html"><i class="fa fa-check"></i><b>9.6</b> Module Scores</a></li>
<li class="chapter" data-level="9.7" data-path="track-plotting-with-archrbrowser.html"><a href="track-plotting-with-archrbrowser.html"><i class="fa fa-check"></i><b>9.7</b> Track Plotting with ArchRBrowser</a></li>
<li class="chapter" data-level="9.8" data-path="launching-the-interactive-archrbrowser.html"><a href="launching-the-interactive-archrbrowser.html"><i class="fa fa-check"></i><b>9.8</b> Launching the Interactive ArchRBrowser</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="defining-cluster-identity-with-scrna-seq.html"><a href="defining-cluster-identity-with-scrna-seq.html"><i class="fa fa-check"></i><b>10</b> Defining Cluster Identity with scRNA-seq</a>
<ul>
<li class="chapter" data-level="10.1" data-path="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html"><a href="cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html"><i class="fa fa-check"></i><b>10.1</b> Cross-platform linkage of scATAC-seq cells with scRNA-seq cells</a></li>
<li class="chapter" data-level="10.2" data-path="adding-pseudo-scrna-seq-profiles-for-each-scatac-seq-cell.html"><a href="adding-pseudo-scrna-seq-profiles-for-each-scatac-seq-cell.html"><i class="fa fa-check"></i><b>10.2</b> Adding Pseudo-scRNA-seq profiles for each scATAC-seq cell</a></li>
<li class="chapter" data-level="10.3" data-path="labeling-scatac-seq-clusters-with-scrna-seq-information.html"><a href="labeling-scatac-seq-clusters-with-scrna-seq-information.html"><i class="fa fa-check"></i><b>10.3</b> Labeling scATAC-seq clusters with scRNA-seq information</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="pseudo-bulk-replicates-in-archr.html"><a href="pseudo-bulk-replicates-in-archr.html"><i class="fa fa-check"></i><b>11</b> Pseudo-bulk Replicates in ArchR</a>
<ul>
<li class="chapter" data-level="11.1" data-path="how-does-archr-make-pseudo-bulk-replicates.html"><a href="how-does-archr-make-pseudo-bulk-replicates.html"><i class="fa fa-check"></i><b>11.1</b> How Does ArchR Make Pseudo-bulk Replicates?</a></li>
<li class="chapter" data-level="11.2" data-path="making-pseudo-bulk-replicates.html"><a href="making-pseudo-bulk-replicates.html"><i class="fa fa-check"></i><b>11.2</b> Making Pseudo-bulk Replicates</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="calling-peaks-with-archr.html"><a href="calling-peaks-with-archr.html"><i class="fa fa-check"></i><b>12</b> Calling Peaks with ArchR</a>
<ul>
<li class="chapter" data-level="12.1" data-path="the-iterative-overlap-peak-merging-procedure.html"><a href="the-iterative-overlap-peak-merging-procedure.html"><i class="fa fa-check"></i><b>12.1</b> The Iterative Overlap Peak Merging Procedure</a></li>
<li class="chapter" data-level="12.2" data-path="calling-peaks-w-macs2.html"><a href="calling-peaks-w-macs2.html"><i class="fa fa-check"></i><b>12.2</b> Calling Peaks w/ Macs2</a></li>
<li class="chapter" data-level="12.3" data-path="calling-peaks-w-tilematrix.html"><a href="calling-peaks-w-tilematrix.html"><i class="fa fa-check"></i><b>12.3</b> Calling Peaks w/ TileMatrix</a></li>
<li class="chapter" data-level="12.4" data-path="adding-a-peak-matrix.html"><a href="adding-a-peak-matrix.html"><i class="fa fa-check"></i><b>12.4</b> Adding a Peak Matrix</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="identifying-marker-peaks.html"><a href="identifying-marker-peaks.html"><i class="fa fa-check"></i><b>13</b> Identifying Marker Peaks</a>
<ul>
<li class="chapter" data-level="13.1" data-path="identifying-marker-peaks-with-archr.html"><a href="identifying-marker-peaks-with-archr.html"><i class="fa fa-check"></i><b>13.1</b> Identifying Marker Peaks with ArchR</a></li>
<li class="chapter" data-level="13.2" data-path="plotting-marker-peaks-in-archr.html"><a href="plotting-marker-peaks-in-archr.html"><i class="fa fa-check"></i><b>13.2</b> Plotting Marker Peaks in ArchR</a></li>
<li class="chapter" data-level="13.3" data-path="pairwise-testing-between-groups.html"><a href="pairwise-testing-between-groups.html"><i class="fa fa-check"></i><b>13.3</b> Pairwise Testing Between Groups</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="motif-and-feature-enrichment-with-archr.html"><a href="motif-and-feature-enrichment-with-archr.html"><i class="fa fa-check"></i><b>14</b> Motif and Feature Enrichment with ArchR</a>
<ul>
<li class="chapter" data-level="14.1" data-path="motif-enrichment.html"><a href="motif-enrichment.html"><i class="fa fa-check"></i><b>14.1</b> Motif Enrichment</a></li>
<li class="chapter" data-level="14.2" data-path="archr-enrichment.html"><a href="archr-enrichment.html"><i class="fa fa-check"></i><b>14.2</b> ArchR Enrichment</a></li>
<li class="chapter" data-level="14.3" data-path="custom-enrichment.html"><a href="custom-enrichment.html"><i class="fa fa-check"></i><b>14.3</b> Custom Enrichment</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="chromvar-deviatons-enrichment-with-archr.html"><a href="chromvar-deviatons-enrichment-with-archr.html"><i class="fa fa-check"></i><b>15</b> ChromVAR Deviatons Enrichment with ArchR</a>
<ul>
<li class="chapter" data-level="15.1" data-path="motif-deviations.html"><a href="motif-deviations.html"><i class="fa fa-check"></i><b>15.1</b> Motif Deviations</a></li>
<li class="chapter" data-level="15.2" data-path="archr-and-custom-deviations.html"><a href="archr-and-custom-deviations.html"><i class="fa fa-check"></i><b>15.2</b> ArchR and Custom Deviations</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="footprinting-with-archr.html"><a href="footprinting-with-archr.html"><i class="fa fa-check"></i><b>16</b> Footprinting with ArchR</a>
<ul>
<li class="chapter" data-level="16.1" data-path="motif-footprinting.html"><a href="motif-footprinting.html"><i class="fa fa-check"></i><b>16.1</b> Motif Footprinting</a></li>
<li class="chapter" data-level="16.2" data-path="normalization-of-footprints-for-tn5-bias.html"><a href="normalization-of-footprints-for-tn5-bias.html"><i class="fa fa-check"></i><b>16.2</b> Normalization of Footprints for Tn5 Bias</a></li>
<li class="chapter" data-level="16.3" data-path="feature-footprinting.html"><a href="feature-footprinting.html"><i class="fa fa-check"></i><b>16.3</b> Feature Footprinting</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="integrative-analysis-with-archr.html"><a href="integrative-analysis-with-archr.html"><i class="fa fa-check"></i><b>17</b> Integrative Analysis with ArchR</a>
<ul>
<li class="chapter" data-level="17.1" data-path="creating-low-overlapping-aggregates-of-cells.html"><a href="creating-low-overlapping-aggregates-of-cells.html"><i class="fa fa-check"></i><b>17.1</b> Creating Low-Overlapping Aggregates of Cells</a></li>
<li class="chapter" data-level="17.2" data-path="co-accessibility-with-archr.html"><a href="co-accessibility-with-archr.html"><i class="fa fa-check"></i><b>17.2</b> Co-accessibility with ArchR</a></li>
<li class="chapter" data-level="17.3" data-path="peak2genelinkage-with-archr.html"><a href="peak2genelinkage-with-archr.html"><i class="fa fa-check"></i><b>17.3</b> Peak2GeneLinkage with ArchR</a></li>
<li class="chapter" data-level="17.4" data-path="identification-of-positive-tf-regulators.html"><a href="identification-of-positive-tf-regulators.html"><i class="fa fa-check"></i><b>17.4</b> Identification of Positive TF-Regulators</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="trajectory-analysis-with-archr.html"><a href="trajectory-analysis-with-archr.html"><i class="fa fa-check"></i><b>18</b> Trajectory Analysis with ArchR</a>
<ul>
<li class="chapter" data-level="18.1" data-path="myeloid-trajectory---monocyte-differentiation.html"><a href="myeloid-trajectory---monocyte-differentiation.html"><i class="fa fa-check"></i><b>18.1</b> Myeloid Trajectory - Monocyte Differentiation</a></li>
<li class="chapter" data-level="18.2" data-path="lymphoid-trajectory---b-cell-cifferentiation.html"><a href="lymphoid-trajectory---b-cell-cifferentiation.html"><i class="fa fa-check"></i><b>18.2</b> Lymphoid Trajectory - B Cell Cifferentiation</a></li>
<li class="chapter" data-level="18.3" data-path="monocle3-trajectories.html"><a href="monocle3-trajectories.html"><i class="fa fa-check"></i><b>18.3</b> Monocle3 Trajectories</a></li>
<li class="chapter" data-level="18.4" data-path="slingshot-trajectories.html"><a href="slingshot-trajectories.html"><i class="fa fa-check"></i><b>18.4</b> Slingshot Trajectories</a></li>
<li class="chapter" data-level="18.5" data-path="comparing-the-different-trajectories.html"><a href="comparing-the-different-trajectories.html"><i class="fa fa-check"></i><b>18.5</b> Comparing the different trajectories</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="handling-archr-output.html"><a href="handling-archr-output.html"><i class="fa fa-check"></i><b>19</b> Handling ArchR Output</a>
<ul>
<li class="chapter" data-level="19.1" data-path="exporting-fragment-level-data.html"><a href="exporting-fragment-level-data.html"><i class="fa fa-check"></i><b>19.1</b> Exporting fragment-level data</a></li>
<li class="chapter" data-level="19.2" data-path="exporting-matrix-level-data.html"><a href="exporting-matrix-level-data.html"><i class="fa fa-check"></i><b>19.2</b> Exporting matrix-level data</a></li>
<li class="chapter" data-level="19.3" data-path="exporting-pseudo-bulked-data-to-a-summarizedexperiment.html"><a href="exporting-pseudo-bulked-data-to-a-summarizedexperiment.html"><i class="fa fa-check"></i><b>19.3</b> Exporting pseudo-bulked data to a <code>SummarizedExperiment</code></a></li>
<li class="chapter" data-level="19.4" data-path="exporting-pseudo-bulked-data-to-a-bigwig-file.html"><a href="exporting-pseudo-bulked-data-to-a-bigwig-file.html"><i class="fa fa-check"></i><b>19.4</b> Exporting pseudo-bulked data to a bigWig file</a></li>
</ul></li>
<li class="chapter" data-level="20" data-path="integration-with-bulk-atac-seq.html"><a href="integration-with-bulk-atac-seq.html"><i class="fa fa-check"></i><b>20</b> Integration with bulk ATAC-seq</a>
<ul>
<li class="chapter" data-level="20.1" data-path="projecting-bulk-atac-seq-data.html"><a href="projecting-bulk-atac-seq-data.html"><i class="fa fa-check"></i><b>20.1</b> Projecting bulk ATAC-seq data</a></li>
</ul></li>
<li class="chapter" data-level="21" data-path="multiomic-data-analysis-in-archr.html"><a href="multiomic-data-analysis-in-archr.html"><i class="fa fa-check"></i><b>21</b> Multiomic data analysis in ArchR</a>
<ul>
<li class="chapter" data-level="21.1" data-path="importing-data-and-setting-up-a-multiome-project.html"><a href="importing-data-and-setting-up-a-multiome-project.html"><i class="fa fa-check"></i><b>21.1</b> Importing data and setting up a Multiome project</a></li>
<li class="chapter" data-level="21.2" data-path="analysis-of-multiome-data-in-archr.html"><a href="analysis-of-multiome-data-in-archr.html"><i class="fa fa-check"></i><b>21.2</b> Analysis of multiome data in ArchR</a></li>
</ul></li>
<li class="chapter" data-level="22" data-path="plot-aesthetics-in-archr.html"><a href="plot-aesthetics-in-archr.html"><i class="fa fa-check"></i><b>22</b> Plot aesthetics in ArchR</a>
<ul>
<li class="chapter" data-level="22.1" data-path="using-and-manipulating-palettes-in-archr.html"><a href="using-and-manipulating-palettes-in-archr.html"><i class="fa fa-check"></i><b>22.1</b> Using and manipulating palettes in ArchR</a></li>
</ul></li>
<li class="chapter" data-level="23" data-path="testing-your-archr-installation.html"><a href="testing-your-archr-installation.html"><i class="fa fa-check"></i><b>23</b> Testing Your ArchR Installation</a></li>
<li class="chapter" data-level="24" data-path="session-information.html"><a href="session-information.html"><i class="fa fa-check"></i><b>24</b> Session Information</a></li>
<li class="divider"></li>
<li><a href="https://bookdown.org" target="_blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">ArchR: Robust and scaleable analysis of single-cell chromatin accessibility data.</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="using-bam-files-for-arrow-file-creation" class="section level2 hasAnchor" number="3.8">
<h2><span class="header-section-number">3.8</span> Using BAM files for Arrow File creation<a href="using-bam-files-for-arrow-file-creation.html#using-bam-files-for-arrow-file-creation" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Though fragment files are strongly preferred because of their very standardized format, BAM files can also be used as input for Arrow File creation in ArchR. The key to getting this to work is understanding your BAM file structure, in particular which field of the BAM file contains the cell-specific barcode information. In this example, we will download some data from 10x Genomics to illustrate how to use BAM files as input to <code>createArrowFiles()</code>. Using <code>samtools view</code></p>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="using-bam-files-for-arrow-file-creation.html#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="co">#set timeout to prevent interrupting large file download</span></span>
<span id="cb31-2"><a href="using-bam-files-for-arrow-file-creation.html#cb31-2" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">timeout=</span><span class="dv">1000</span>)</span>
<span id="cb31-3"><a href="using-bam-files-for-arrow-file-creation.html#cb31-3" aria-hidden="true" tabindex="-1"></a><span class="fu">dir.create</span>(<span class="at">path =</span> <span class="st">&quot;./PBMC_BAM&quot;</span>)</span>
<span id="cb31-4"><a href="using-bam-files-for-arrow-file-creation.html#cb31-4" aria-hidden="true" tabindex="-1"></a><span class="do">## Warning in dir.create(path = &quot;./PBMC_BAM&quot;): &#39;./PBMC_BAM&#39; already exists</span></span>
<span id="cb31-5"><a href="using-bam-files-for-arrow-file-creation.html#cb31-5" aria-hidden="true" tabindex="-1"></a>down_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb31-6"><a href="using-bam-files-for-arrow-file-creation.html#cb31-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">fileUrl =</span> <span class="fu">c</span>(<span class="st">&quot;https://cf.10xgenomics.com/samples/cell-atac/2.0.0/atac_pbmc_500_nextgem/atac_pbmc_500_nextgem_possorted_bam.bam&quot;</span>,</span>
<span id="cb31-7"><a href="using-bam-files-for-arrow-file-creation.html#cb31-7" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;https://cf.10xgenomics.com/samples/cell-atac/2.0.0/atac_pbmc_500_nextgem/atac_pbmc_500_nextgem_possorted_bam.bam.bai&quot;</span>),</span>
<span id="cb31-8"><a href="using-bam-files-for-arrow-file-creation.html#cb31-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">md5sum =</span> <span class="fu">c</span>(<span class="st">&quot;8140a2218ecdfd276aca5c4bb999c989&quot;</span>,<span class="st">&quot;d3c5f5a00fec76378f2a947749ff2cf5&quot;</span>)</span>
<span id="cb31-9"><a href="using-bam-files-for-arrow-file-creation.html#cb31-9" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb31-10"><a href="using-bam-files-for-arrow-file-creation.html#cb31-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-11"><a href="using-bam-files-for-arrow-file-creation.html#cb31-11" aria-hidden="true" tabindex="-1"></a><span class="co">#we will use a hidden ArchR function to do the download. This automatically checks the md5sum.</span></span>
<span id="cb31-12"><a href="using-bam-files-for-arrow-file-creation.html#cb31-12" aria-hidden="true" tabindex="-1"></a>ArchR<span class="sc">:::</span><span class="fu">.downloadFiles</span>(<span class="at">filesUrl =</span> down_df, <span class="at">pathDownload =</span> <span class="st">&quot;./PBMC_BAM&quot;</span>, <span class="at">threads =</span> <span class="dv">1</span>)</span>
<span id="cb31-13"><a href="using-bam-files-for-arrow-file-creation.html#cb31-13" aria-hidden="true" tabindex="-1"></a><span class="do">## Downloading files to ./PBMC_BAM...</span></span>
<span id="cb31-14"><a href="using-bam-files-for-arrow-file-creation.html#cb31-14" aria-hidden="true" tabindex="-1"></a><span class="do">## File exists! Skipping file atac_pbmc_500_nextgem_possorted_bam.bam...</span></span>
<span id="cb31-15"><a href="using-bam-files-for-arrow-file-creation.html#cb31-15" aria-hidden="true" tabindex="-1"></a><span class="do">## File exists! Skipping file atac_pbmc_500_nextgem_possorted_bam.bam.bai...</span></span>
<span id="cb31-16"><a href="using-bam-files-for-arrow-file-creation.html#cb31-16" aria-hidden="true" tabindex="-1"></a><span class="do">## [[1]]</span></span>
<span id="cb31-17"><a href="using-bam-files-for-arrow-file-creation.html#cb31-17" aria-hidden="true" tabindex="-1"></a><span class="do">## NULL</span></span>
<span id="cb31-18"><a href="using-bam-files-for-arrow-file-creation.html#cb31-18" aria-hidden="true" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb31-19"><a href="using-bam-files-for-arrow-file-creation.html#cb31-19" aria-hidden="true" tabindex="-1"></a><span class="do">## [[2]]</span></span>
<span id="cb31-20"><a href="using-bam-files-for-arrow-file-creation.html#cb31-20" aria-hidden="true" tabindex="-1"></a><span class="do">## NULL</span></span></code></pre></div>
<p>If we were to use <code>samtools view</code> from the command line to look at this BAM file, the first line would look like this:</p>
<pre><code>A00519:269:H7FM2DRXX:2:1201:1985:18834  83      chr1    9997    0       50M     =       10022   -25     CAGATAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAAC  ,,,:FF,,F,FF,FFFFFFF:FFFFFFFFFFFFFFF:FF:FFF,:FFFFF      NM:i:1  MD:Z:1C48       AS:i:48 XS:i:47 CR:Z:GCGGGTTAGAACGTCG       CY:Z:FFFFFFFF:FFFFFFF   CB:Z:GCGGGTTAGAACGTCG-1 BC:Z:ATCGTACT   QT:Z:FFFFFFFF   RG:Z:atac_pbmc_500_nextgem:MissingLibrary:1:H7FM2DRXX:2</code></pre>
<p>Here, the <code>CB</code> tag is used to store the cell-specific barcode, in this case <code>CB:Z:GCGGGTTAGAACGTCG-1</code>. This means that we will pass <code>bcTag = "CB"</code> to <code>createArrowFiles()</code> to tell it to look for the <code>CB</code> tag. If, for example, your BAM files come from a scATAC-seq technology other than 10x Genomics, the relevant tag will likely be different. On top of that, the format of the cell-specific barcode may also be different. In these cases, it may be necessary to use the <code>gsubExpression</code> parameter to clean up the cell-specific barcode string. The other important input parameter is <code>bamFlag</code> which will determine which reads or fragments are viewed as valid. The value passed to <code>bamFlag</code> should be in the format of a <code>scanBamFlag</code> used by <code>ScanBam</code> in <code>Rsamtools</code>.</p>
<p>In addition to understanding the structure of you BAM file, you may need to pre-process your BAM file by:</p>
<ol style="list-style-type: decimal">
<li>Removing any fragments less than 20 bp in length</li>
<li>Marking/removing PCR duplicates</li>
<li>Removing barcode reads that are <code>NA</code></li>
</ol>
<p>In the case of our example data that we just downloaded, we can proceed to Arrow file creation.</p>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="using-bam-files-for-arrow-file-creation.html#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="fu">addArchRGenome</span>(<span class="st">&quot;hg38&quot;</span>)</span>
<span id="cb33-2"><a href="using-bam-files-for-arrow-file-creation.html#cb33-2" aria-hidden="true" tabindex="-1"></a><span class="do">## Setting default genome to Hg38.</span></span>
<span id="cb33-3"><a href="using-bam-files-for-arrow-file-creation.html#cb33-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-4"><a href="using-bam-files-for-arrow-file-creation.html#cb33-4" aria-hidden="true" tabindex="-1"></a>ArrowBam <span class="ot">&lt;-</span> <span class="fu">createArrowFiles</span>(</span>
<span id="cb33-5"><a href="using-bam-files-for-arrow-file-creation.html#cb33-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">inputFiles =</span> <span class="st">&quot;./PBMC_BAM/atac_pbmc_500_nextgem_possorted_bam.bam&quot;</span>,</span>
<span id="cb33-6"><a href="using-bam-files-for-arrow-file-creation.html#cb33-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">sampleNames =</span> <span class="st">&quot;bam10x&quot;</span>,</span>
<span id="cb33-7"><a href="using-bam-files-for-arrow-file-creation.html#cb33-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">minTSS =</span> <span class="dv">4</span>, <span class="co">#Dont set this too high because you can always increase later</span></span>
<span id="cb33-8"><a href="using-bam-files-for-arrow-file-creation.html#cb33-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">minFrags =</span> <span class="dv">1000</span>, </span>
<span id="cb33-9"><a href="using-bam-files-for-arrow-file-creation.html#cb33-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">addTileMat =</span> <span class="cn">FALSE</span>,</span>
<span id="cb33-10"><a href="using-bam-files-for-arrow-file-creation.html#cb33-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">addGeneScoreMat =</span> <span class="cn">FALSE</span>,</span>
<span id="cb33-11"><a href="using-bam-files-for-arrow-file-creation.html#cb33-11" aria-hidden="true" tabindex="-1"></a>  <span class="at">bcTag =</span> <span class="st">&quot;CB&quot;</span>,</span>
<span id="cb33-12"><a href="using-bam-files-for-arrow-file-creation.html#cb33-12" aria-hidden="true" tabindex="-1"></a>  <span class="at">bamFlag =</span> <span class="fu">list</span>(<span class="at">isMinusStrand =</span> <span class="cn">FALSE</span>, <span class="at">isProperPair =</span> <span class="cn">TRUE</span>, <span class="at">isDuplicate =</span> <span class="cn">FALSE</span>)</span>
<span id="cb33-13"><a href="using-bam-files-for-arrow-file-creation.html#cb33-13" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb33-14"><a href="using-bam-files-for-arrow-file-creation.html#cb33-14" aria-hidden="true" tabindex="-1"></a><span class="do">## Using GeneAnnotation set by addArchRGenome(Hg38)!</span></span>
<span id="cb33-15"><a href="using-bam-files-for-arrow-file-creation.html#cb33-15" aria-hidden="true" tabindex="-1"></a><span class="do">## Using GeneAnnotation set by addArchRGenome(Hg38)!</span></span>
<span id="cb33-16"><a href="using-bam-files-for-arrow-file-creation.html#cb33-16" aria-hidden="true" tabindex="-1"></a><span class="do">## ArchR logging to : ArchRLogs/ArchR-createArrows-3d01d231080e4b-Date-2022-12-20_Time-13-32-52.log</span></span>
<span id="cb33-17"><a href="using-bam-files-for-arrow-file-creation.html#cb33-17" aria-hidden="true" tabindex="-1"></a><span class="do">## If there is an issue, please report to github with logFile!</span></span>
<span id="cb33-18"><a href="using-bam-files-for-arrow-file-creation.html#cb33-18" aria-hidden="true" tabindex="-1"></a><span class="do">## Cleaning Temporary Files</span></span>
<span id="cb33-19"><a href="using-bam-files-for-arrow-file-creation.html#cb33-19" aria-hidden="true" tabindex="-1"></a><span class="do">## subThreading Disabled since ArchRLocking is TRUE see `addArchRLocking`</span></span>
<span id="cb33-20"><a href="using-bam-files-for-arrow-file-creation.html#cb33-20" aria-hidden="true" tabindex="-1"></a><span class="do">## 2022-12-20 13:32:52 : Batch Execution w/ safelapply!, 0 mins elapsed.</span></span>
<span id="cb33-21"><a href="using-bam-files-for-arrow-file-creation.html#cb33-21" aria-hidden="true" tabindex="-1"></a><span class="do">## (bam10x : 1 of 1) Determining Arrow Method to use!</span></span>
<span id="cb33-22"><a href="using-bam-files-for-arrow-file-creation.html#cb33-22" aria-hidden="true" tabindex="-1"></a><span class="do">## 2022-12-20 13:32:52 : (bam10x : 1 of 1) Reading In Fragments from inputFiles (readMethod = bam), 0.001 mins elapsed.</span></span>
<span id="cb33-23"><a href="using-bam-files-for-arrow-file-creation.html#cb33-23" aria-hidden="true" tabindex="-1"></a><span class="do">## 2022-12-20 13:32:52 : (bam10x : 1 of 1) Tabix Bam To Temporary File, 0.001 mins elapsed.</span></span>
<span id="cb33-24"><a href="using-bam-files-for-arrow-file-creation.html#cb33-24" aria-hidden="true" tabindex="-1"></a><span class="do">## Warning in sprintf(&quot;%s Reading BamFile %s Percent&quot;, prefix, round(100 * : one</span></span>
<span id="cb33-25"><a href="using-bam-files-for-arrow-file-creation.html#cb33-25" aria-hidden="true" tabindex="-1"></a><span class="do">## argument not used by format &#39;%s Reading BamFile %s Percent&#39;</span></span>
<span id="cb33-26"><a href="using-bam-files-for-arrow-file-creation.html#cb33-26" aria-hidden="true" tabindex="-1"></a><span class="do">## 2022-12-20 13:33:09 : (bam10x : 1 of 1) Reading BamFile 8 Percent, 0.291 mins elapsed.</span></span>
<span id="cb33-27"><a href="using-bam-files-for-arrow-file-creation.html#cb33-27" aria-hidden="true" tabindex="-1"></a><span class="do">## Warning in sprintf(&quot;%s Reading BamFile %s Percent&quot;, prefix, round(100 * : one</span></span>
<span id="cb33-28"><a href="using-bam-files-for-arrow-file-creation.html#cb33-28" aria-hidden="true" tabindex="-1"></a><span class="do">## argument not used by format &#39;%s Reading BamFile %s Percent&#39;</span></span>
<span id="cb33-29"><a href="using-bam-files-for-arrow-file-creation.html#cb33-29" aria-hidden="true" tabindex="-1"></a><span class="do">## 2022-12-20 13:33:24 : (bam10x : 1 of 1) Reading BamFile 17 Percent, 0.529 mins elapsed.</span></span>
<span id="cb33-30"><a href="using-bam-files-for-arrow-file-creation.html#cb33-30" aria-hidden="true" tabindex="-1"></a><span class="do">## Warning in sprintf(&quot;%s Reading BamFile %s Percent&quot;, prefix, round(100 * : one</span></span>
<span id="cb33-31"><a href="using-bam-files-for-arrow-file-creation.html#cb33-31" aria-hidden="true" tabindex="-1"></a><span class="do">## argument not used by format &#39;%s Reading BamFile %s Percent&#39;</span></span>
<span id="cb33-32"><a href="using-bam-files-for-arrow-file-creation.html#cb33-32" aria-hidden="true" tabindex="-1"></a><span class="do">## 2022-12-20 13:33:37 : (bam10x : 1 of 1) Reading BamFile 25 Percent, 0.759 mins elapsed.</span></span>
<span id="cb33-33"><a href="using-bam-files-for-arrow-file-creation.html#cb33-33" aria-hidden="true" tabindex="-1"></a><span class="do">## Warning in sprintf(&quot;%s Reading BamFile %s Percent&quot;, prefix, round(100 * : one</span></span>
<span id="cb33-34"><a href="using-bam-files-for-arrow-file-creation.html#cb33-34" aria-hidden="true" tabindex="-1"></a><span class="do">## argument not used by format &#39;%s Reading BamFile %s Percent&#39;</span></span>
<span id="cb33-35"><a href="using-bam-files-for-arrow-file-creation.html#cb33-35" aria-hidden="true" tabindex="-1"></a><span class="do">## 2022-12-20 13:33:51 : (bam10x : 1 of 1) Reading BamFile 33 Percent, 0.978 mins elapsed.</span></span>
<span id="cb33-36"><a href="using-bam-files-for-arrow-file-creation.html#cb33-36" aria-hidden="true" tabindex="-1"></a><span class="do">## Warning in sprintf(&quot;%s Reading BamFile %s Percent&quot;, prefix, round(100 * : one</span></span>
<span id="cb33-37"><a href="using-bam-files-for-arrow-file-creation.html#cb33-37" aria-hidden="true" tabindex="-1"></a><span class="do">## argument not used by format &#39;%s Reading BamFile %s Percent&#39;</span></span>
<span id="cb33-38"><a href="using-bam-files-for-arrow-file-creation.html#cb33-38" aria-hidden="true" tabindex="-1"></a><span class="do">## 2022-12-20 13:34:03 : (bam10x : 1 of 1) Reading BamFile 42 Percent, 1.194 mins elapsed.</span></span>
<span id="cb33-39"><a href="using-bam-files-for-arrow-file-creation.html#cb33-39" aria-hidden="true" tabindex="-1"></a><span class="do">## Warning in sprintf(&quot;%s Reading BamFile %s Percent&quot;, prefix, round(100 * : one</span></span>
<span id="cb33-40"><a href="using-bam-files-for-arrow-file-creation.html#cb33-40" aria-hidden="true" tabindex="-1"></a><span class="do">## argument not used by format &#39;%s Reading BamFile %s Percent&#39;</span></span>
<span id="cb33-41"><a href="using-bam-files-for-arrow-file-creation.html#cb33-41" aria-hidden="true" tabindex="-1"></a><span class="do">## 2022-12-20 13:34:17 : (bam10x : 1 of 1) Reading BamFile 50 Percent, 1.422 mins elapsed.</span></span>
<span id="cb33-42"><a href="using-bam-files-for-arrow-file-creation.html#cb33-42" aria-hidden="true" tabindex="-1"></a><span class="do">## Warning in sprintf(&quot;%s Reading BamFile %s Percent&quot;, prefix, round(100 * : one</span></span>
<span id="cb33-43"><a href="using-bam-files-for-arrow-file-creation.html#cb33-43" aria-hidden="true" tabindex="-1"></a><span class="do">## argument not used by format &#39;%s Reading BamFile %s Percent&#39;</span></span>
<span id="cb33-44"><a href="using-bam-files-for-arrow-file-creation.html#cb33-44" aria-hidden="true" tabindex="-1"></a><span class="do">## 2022-12-20 13:34:28 : (bam10x : 1 of 1) Reading BamFile 58 Percent, 1.608 mins elapsed.</span></span>
<span id="cb33-45"><a href="using-bam-files-for-arrow-file-creation.html#cb33-45" aria-hidden="true" tabindex="-1"></a><span class="do">## Warning in sprintf(&quot;%s Reading BamFile %s Percent&quot;, prefix, round(100 * : one</span></span>
<span id="cb33-46"><a href="using-bam-files-for-arrow-file-creation.html#cb33-46" aria-hidden="true" tabindex="-1"></a><span class="do">## argument not used by format &#39;%s Reading BamFile %s Percent&#39;</span></span>
<span id="cb33-47"><a href="using-bam-files-for-arrow-file-creation.html#cb33-47" aria-hidden="true" tabindex="-1"></a><span class="do">## 2022-12-20 13:34:41 : (bam10x : 1 of 1) Reading BamFile 67 Percent, 1.812 mins elapsed.</span></span>
<span id="cb33-48"><a href="using-bam-files-for-arrow-file-creation.html#cb33-48" aria-hidden="true" tabindex="-1"></a><span class="do">## Warning in sprintf(&quot;%s Reading BamFile %s Percent&quot;, prefix, round(100 * : one</span></span>
<span id="cb33-49"><a href="using-bam-files-for-arrow-file-creation.html#cb33-49" aria-hidden="true" tabindex="-1"></a><span class="do">## argument not used by format &#39;%s Reading BamFile %s Percent&#39;</span></span>
<span id="cb33-50"><a href="using-bam-files-for-arrow-file-creation.html#cb33-50" aria-hidden="true" tabindex="-1"></a><span class="do">## 2022-12-20 13:34:53 : (bam10x : 1 of 1) Reading BamFile 75 Percent, 2.021 mins elapsed.</span></span>
<span id="cb33-51"><a href="using-bam-files-for-arrow-file-creation.html#cb33-51" aria-hidden="true" tabindex="-1"></a><span class="do">## Warning in sprintf(&quot;%s Reading BamFile %s Percent&quot;, prefix, round(100 * : one</span></span>
<span id="cb33-52"><a href="using-bam-files-for-arrow-file-creation.html#cb33-52" aria-hidden="true" tabindex="-1"></a><span class="do">## argument not used by format &#39;%s Reading BamFile %s Percent&#39;</span></span>
<span id="cb33-53"><a href="using-bam-files-for-arrow-file-creation.html#cb33-53" aria-hidden="true" tabindex="-1"></a><span class="do">## 2022-12-20 13:35:06 : (bam10x : 1 of 1) Reading BamFile 83 Percent, 2.238 mins elapsed.</span></span>
<span id="cb33-54"><a href="using-bam-files-for-arrow-file-creation.html#cb33-54" aria-hidden="true" tabindex="-1"></a><span class="do">## Warning in sprintf(&quot;%s Reading BamFile %s Percent&quot;, prefix, round(100 * : one</span></span>
<span id="cb33-55"><a href="using-bam-files-for-arrow-file-creation.html#cb33-55" aria-hidden="true" tabindex="-1"></a><span class="do">## argument not used by format &#39;%s Reading BamFile %s Percent&#39;</span></span>
<span id="cb33-56"><a href="using-bam-files-for-arrow-file-creation.html#cb33-56" aria-hidden="true" tabindex="-1"></a><span class="do">## 2022-12-20 13:35:15 : (bam10x : 1 of 1) Reading BamFile 92 Percent, 2.394 mins elapsed.</span></span>
<span id="cb33-57"><a href="using-bam-files-for-arrow-file-creation.html#cb33-57" aria-hidden="true" tabindex="-1"></a><span class="do">## Warning in sprintf(&quot;%s Reading BamFile %s Percent&quot;, prefix, round(100 * : one</span></span>
<span id="cb33-58"><a href="using-bam-files-for-arrow-file-creation.html#cb33-58" aria-hidden="true" tabindex="-1"></a><span class="do">## argument not used by format &#39;%s Reading BamFile %s Percent&#39;</span></span>
<span id="cb33-59"><a href="using-bam-files-for-arrow-file-creation.html#cb33-59" aria-hidden="true" tabindex="-1"></a><span class="do">## 2022-12-20 13:35:24 : (bam10x : 1 of 1) Reading BamFile 100 Percent, 2.539 mins elapsed.</span></span>
<span id="cb33-60"><a href="using-bam-files-for-arrow-file-creation.html#cb33-60" aria-hidden="true" tabindex="-1"></a><span class="do">## 2022-12-20 13:35:25 : (bam10x : 1 of 1) Successful creation of Temporary File, 2.552 mins elapsed.</span></span>
<span id="cb33-61"><a href="using-bam-files-for-arrow-file-creation.html#cb33-61" aria-hidden="true" tabindex="-1"></a><span class="do">## 2022-12-20 13:35:25 : (bam10x : 1 of 1) Creating ArrowFile From Temporary File, 2.552 mins elapsed.</span></span>
<span id="cb33-62"><a href="using-bam-files-for-arrow-file-creation.html#cb33-62" aria-hidden="true" tabindex="-1"></a><span class="do">## 2022-12-20 13:36:02 : (bam10x : 1 of 1) Successful creation of Arrow File, 3.168 mins elapsed.</span></span>
<span id="cb33-63"><a href="using-bam-files-for-arrow-file-creation.html#cb33-63" aria-hidden="true" tabindex="-1"></a><span class="do">## 2022-12-20 13:36:59 : (bam10x : 1 of 1) CellStats : Number of Cells Pass Filter = 568 , 4.119 mins elapsed.</span></span>
<span id="cb33-64"><a href="using-bam-files-for-arrow-file-creation.html#cb33-64" aria-hidden="true" tabindex="-1"></a><span class="do">## 2022-12-20 13:36:59 : (bam10x : 1 of 1) CellStats : Median Frags = 16887 , 4.119 mins elapsed.</span></span>
<span id="cb33-65"><a href="using-bam-files-for-arrow-file-creation.html#cb33-65" aria-hidden="true" tabindex="-1"></a><span class="do">## 2022-12-20 13:36:59 : (bam10x : 1 of 1) CellStats : Median TSS Enrichment = 19.811 , 4.119 mins elapsed.</span></span>
<span id="cb33-66"><a href="using-bam-files-for-arrow-file-creation.html#cb33-66" aria-hidden="true" tabindex="-1"></a><span class="do">## 2022-12-20 13:37:00 : (bam10x : 1 of 1) Adding Additional Feature Counts!, 4.143 mins elapsed.</span></span>
<span id="cb33-67"><a href="using-bam-files-for-arrow-file-creation.html#cb33-67" aria-hidden="true" tabindex="-1"></a><span class="do">## 2022-12-20 13:37:52 : (bam10x : 1 of 1) Removing Fragments from Filtered Cells, 5.002 mins elapsed.</span></span>
<span id="cb33-68"><a href="using-bam-files-for-arrow-file-creation.html#cb33-68" aria-hidden="true" tabindex="-1"></a><span class="do">## 2022-12-20 13:37:52 : (bam10x : 1 of 1) Creating Filtered Arrow File, 5.002 mins elapsed.</span></span>
<span id="cb33-69"><a href="using-bam-files-for-arrow-file-creation.html#cb33-69" aria-hidden="true" tabindex="-1"></a><span class="do">## 2022-12-20 13:38:11 : (bam10x : 1 of 1) Finished Constructing Filtered Arrow File!, 5.322 mins elapsed.</span></span>
<span id="cb33-70"><a href="using-bam-files-for-arrow-file-creation.html#cb33-70" aria-hidden="true" tabindex="-1"></a><span class="do">## 2022-12-20 13:38:11 : (bam10x : 1 of 1) Finished Creating Arrow File, 5.323 mins elapsed.</span></span>
<span id="cb33-71"><a href="using-bam-files-for-arrow-file-creation.html#cb33-71" aria-hidden="true" tabindex="-1"></a><span class="do">## ArchR logging successful to : ArchRLogs/ArchR-createArrows-3d01d231080e4b-Date-2022-12-20_Time-13-32-52.log</span></span></code></pre></div>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="using-bam-files-for-arrow-file-creation.html#cb34-1" aria-hidden="true" tabindex="-1"></a>ArrowBam</span>
<span id="cb34-2"><a href="using-bam-files-for-arrow-file-creation.html#cb34-2" aria-hidden="true" tabindex="-1"></a><span class="do">## [1] &quot;bam10x.arrow&quot;</span></span></code></pre></div>

</div>
<!-- </div> -->
            </section>

          </div>
        </div>
      </div>
<a href="per-cell-quality-control.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="doublet-inference-with-archr.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": null
},
"fontsettings": null,
"edit": null,
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "none"
},
"info": false
});
});
</script>

</body>

</html>
