---
output:
  html_document:
    theme: yeti  # many options for theme, this one is my favorite.
params:
  threads: 20
  rdata: ""
  token: ""
  ref: ""
  out: ""
---

```{r, include=FALSE, eval=FALSE}
params2 <- params
unlockBinding("params", env = .GlobalEnv)
#load("Chapter-13-Footprints.Rdata")
load(params$rdata)
params <- params2
rm(params2)
devtools::install_github("GreenleafLab/ArchR", 
  auth_token = params$token, 
  ref = params$ref,
  repos = BiocManager::repositories(),
  dependencies = FALSE
)
library(ArchR)
fn <- unclass(lsf.str(envir = asNamespace("ArchR"), all = TRUE))
fn <- fn[!grepl("\\.", fn)]
fn <- fn[fn!="ArchRProj"]
for (i in seq_along(fn)){
    tryCatch({
        eval(parse(text = paste0(fn[i], "<-ArchR::", fn[i])))
    }, error = function(x) {
    })
}
addArchRThreads(threads = params$threads)
addArchRGenome("hg19")
# fn <- unclass(lsf.str(envir = asNamespace("ArchR"), all = TRUE))
# fn <- fn[fn!="ArchRProj"]
# for (i in seq_along(fn)) {
#     tryCatch({
#         eval(parse(text = paste0(fn[i], "<-ArchR:::", fn[i])))
#     }, error = function(x) {
#     })
# }
set.seed(1)
```

# Calling Peaks with ArchR

Calling peaks is one of the most fundamental processes in ATAC-seq data analysis. Because per-cell scATAC-seq data is essentially binary (accessible or not accessible), we cannot call peaks on an individual cell basis. For this reason, we defined groups of cells, typically clusters, in a previous chapter. Moreover, we created pseudo-bulk replicates to allow us to assess the reproducibility of our peak calls.

## The Iterative Overlap Peak Merging Procedure

We first introduced [a strategy for iterative overlap peak merging in 2018](https://science.sciencemag.org/content/362/6413/eaav1898). Other peak merging strategies suffer from a few key issues that we outline below.

### Fixed-width vs Variable-width Peaks
We use 501-bp fixed-width peaks because they make downstream computation easier as peak length does not need to be normalized. Moreover, the vast majority of peaks in ATAC-seq are less than 501-bp wide. Using variable-width peaks also makes it difficult to merge peak calls from multiple samples. In general, we do not feel that the potential benefit derived from using variable-width peaks outweighs the costs. More broadly, most analyses are stable with respect to the peak set or peak style used.

Below, we use the same toy example of a few cell types with a few different peaks to illustrate the differences between these often used peak merging methods.

### Raw Peak Overlap Using _bedtools merge_

Raw peak overlap involves taking any peaks that overlap each other and merging these into a single larger peak. In this scheme, daisy-chaining becomes a large problem because peaks that dont directly overlap each other get included in the same larger peak because they are bridged by a shared internal peak. Another problem with this type of approach is that, if you want to keep track of peak summits, you are forced to either pick a single new summit for each new merged peak or keep track of all of the summits that apply to each new merged peak. Typically, this type of peak merging approach is implemented using the `bedtools merge` command.

![](images/peakCalling_RawOverlap.png){width=350px}

### Clustered Overlap Using _bedtools cluster_

Clustered overlap takes peaks that cluster together and picks a single winner. This is often done by using the `bedtools cluster` command and then keeping the most significant peak in each cluster. In our experience, this ends up under-calling peaks and misses smaller peaks located nearby.

![](images/peakCalling_ClusteredOverlap.png){width=350px}

### Iterative Overlap In ArchR

Iterative overlap removal avoids the issues mentioned above. Peaks are first ranked by their significance. The most significant peak is retained and any peak that directly overlaps with the most significant peak is removed from further analysis. Then, of the remaining peaks, this process is repeated until no more peaks exist. This avoids daisy-chaining and still allows for use of fixed-width peaks.

![](images/peakCalling_IterativeOverlap.png){width=350px}

### Comparison of Peak Calling Methods

Comparing the peak calls resulting from all of these methods directly shows clear differences in the final peak sets. It is our opinion that the iterative overlap peak merging process yields the best peak set with the fewest caveats.

![](images/peakCalling_Comparison.png){width=350px}

### So how does this all work in ArchR?

The iterative overlap peak merging procedure is performed in a tiered fashion to optimally preserve cell type-specific peaks.

Imagine a situation where you had 3 cell types, A, B, and C, and each cell type had 3 pseudo-bulk replicates. ArchR uses a function called `addReproduciblePeakSet()` to perform this iterative overlap peak merging procedure. First, ArchR would call peaks for each pseudo-bulk replicate individually. Then, ArchR would analyze all of the pseudo-bulk replicates from a single cell type together, performing the first iteration of iterative overlap removal. It is important to note that ArchR uses a normalized metric of significance for peaks to compare the significance of peaks called across different samples. This is because the reported MACS2 significance is proportional to the sequencing depth so peak significance is not immediately comparable across samples. After the first iteration of iterative overlap removal, ArchR checks to see the reproducibility of each peak across pseudo-bulk replicates and only keeps peaks that pass a threshold indicated by the `reproducibility` parameter. At the end of this process, we would have a single merged peak set for each of the 3 cell types, A, B, and C. 

Then, we would repeat this procedure to merge the A, B, and C peak sets. To do this, we re-normalize the peak significance across the different cell types, and perform the iterative overlap removal. The final result of this is a single merged peak set of fixed-width peaks.

### What if I don't like this iterative overlap peak merging process?

The iterative overlap peak merging process is implemented by ArchR via `addReproduciblePeakSet()` but you can always use your own peak set via `ArchRProj <- addPeakSet()`.


## Calling Peaks w/ Macs2

As mentioned above, we generate a reproducible peak set in ArchR using the `addReproduciblePeakSet()` function. By default ArchR attempts to call peaks using MACS2; however, ArchR also implements its own native peak caller which could be used when MACS2 cannot be installed (for example, we have not successfully installed MACS2 on Windows) - this alternative peak calling method is described in the next section.

To call peaks using MACS2, ArchR must be able to find the MACS2 executable. First, ArchR looks in your `PATH` environment variable. If this is unsuccessful, ArchR attempts to determine if you have installed MACS2 with either `pip` or `pip3`. If neither of these is successful, ArchR gives up and provides an error message. If you have MACS2 installed and ArchR cannot find it, you should provide the path to the `addReproduciblePeakSet()` function via the `pathToMacs2` parameter.

```{r eval=FALSE}
pathToMacs2 <- findMacs2()
```
> \## Searching For MACS2..  
## Found with $path!  

With the path to MACS2 identified, we can then create a reproducible merged peak set w/ MACS2 (~5-10 minutes). To avoid bias from pseudo-bulk replicates that have very few cells, we can provide a cutoff for the upper limit of the number of peaks called per cell via the `peaksPerCell` parameter. This prevents clusters with very few cells from contributing lots of low quality peaks to the merged peak set. There are many other parameters that can be tweaked in `addReproduciblePeakSet()` - try `?addReproduciblePeakSet` for more information.

Each `ArchRProject` object can only contain a single peak set. As such, we assign the output of `addReproduciblePeakSet()` to our desired `ArchRProject`. If you would like to experiment with different peak sets, you must save a copy of your `ArchRProject` and thus also copy the Arrow files. While this does use more on-disk storage space, this is unavoidable given the structure of Arrow files and the storage of peak matrix information in the Arrow files.
```{r eval=FALSE}
projHeme4 <- addReproduciblePeakSet(
	ArchRProj = projHeme4, 
	groupBy = "Clusters2", 
	pathToMacs2 = pathToMacs2
)
```
> \## ArchR logging to : ArchRLogs/ArchR-addReproduciblePeakSet-faba719c230c-Date-2020-04-15_Time-10-25-16.log  
## If there is an issue, please report to github with logFile!  
## Calling Peaks with Macs2  
## 2020-04-15 10:25:17 : Peak Calling Parameters!, 0.014 mins elapsed.  
##                 Group nCells nCellsUsed nReplicates nMin nMax maxPeaks  
## B                   B    439        433           2  182  251   150000  
## CD4.M           CD4.M    678        617           2  117  500   150000  
## CD4.N           CD4.N   1271        552           2   52  500   150000  
## CLP               CLP    387        387           2   82  305   150000  
## Erythroid   Erythroid    879        694           2  194  500   150000  
## GMP               GMP    793        751           2  251  500   150000  
## Mono             Mono   2632       1000           2  500  500   150000  
## NK                 NK    851        801           2  301  500   150000  
## pDC               pDC    320        311           2  150  161   150000  
## PreB             PreB    351        351           2   40  311   150000  
## Progenitor Progenitor   1650        672           2  172  500   150000  
## 2020-04-15 10:25:17 : Batching Peak Calls!, 0.014 mins elapsed.  
## 2020-04-15 10:25:17 : Batch Execution w/ safelapply!, 0 mins elapsed.  
## 2020-04-15 10:27:01 : Identifying Reproducible Peaks!, 1.74 mins elapsed.  
## 2020-04-15 10:27:09 : Creating Union Peak Set!, 1.886 mins elapsed.  
## Converged after 7 iterations!  
## [1] "plotting ggplot!"  
## 2020-04-15 10:27:15 : Finished Creating Union Peak Set (144009)!, 1.98 mins elapsed.  

To retrieve this peak set as a `GRanges` object, we use the `getPeakSet()` function. This peak set contains an annotation for the group from which each peak originated. However, these annotations do not inherently mean that the given peak was only called in that group, rather that the annotated group had the highest normalized significance for that peak call.
```{r eval=FALSE}
getPeakSet(projHeme4)
```
> \## GRanges object with 144009 ranges and 12 metadata columns:  
##             seqnames              ranges strand |     score  
##                <Rle>           <IRanges>  <Rle> | <numeric>  
##        Mono     chr1       752499-752999      * |  24.54003  
##          NK     chr1       762651-763151      * | 141.22064  
##           B     chr1       801006-801506      * |  14.18461  
##           B     chr1       805039-805539      * |  37.30365  
##         CLP     chr1       845325-845825      * |   2.81281  
##         ...      ...                 ...    ... .       ...  
##   Erythroid     chrX 154664540-154665040      * |   7.09786  
##          NK     chrX 154807324-154807824      * |   9.38477  
##        PreB     chrX 154840785-154841285      * |   3.29501  
##        PreB     chrX 154842404-154842904      * |   7.68692  
##          NK     chrX 154862017-154862517      * |     12.44  
##             replicateScoreQuantile groupScoreQuantile Reproducibility  
##                          <numeric>          <numeric>       <numeric>  
##        Mono                  0.812              0.655               2  
##          NK                  0.887              0.797               2  
##           B                    0.7              0.418               2  
##           B                  0.959                0.9               2  
##         CLP                  0.706              0.311               2  
##         ...                    ...                ...             ...  
##   Erythroid                   0.69              0.279               2  
##          NK                  0.424              0.166               2  
##        PreB                  0.669              0.274               2  
##        PreB                  0.932              0.772               2  
##          NK                  0.492              0.202               2  
##                         GroupReplicate distToGeneStart nearestGene    peakType  
##                            <character>       <integer> <character> <character>  
##        Mono      Mono._.scATAC_PBMC_R1           10152   LINC00115      Distal  
##          NK        NK._.scATAC_PBMC_R1               0   LINC00115    Promoter   
##           B         B._.scATAC_PBMC_R1           10925      FAM41C      Distal  
##           B         B._.scATAC_BMMC_R1            6892      FAM41C    Intronic  
##         CLP       CLP._.scATAC_BMMC_R1            9241   LINC02593      Distal  
##         ...                        ...             ...         ...         ...  
##   Erythroid Erythroid._.scATAC_BMMC_R1          100803       CLIC2      Distal  
##          NK        NK._.scATAC_PBMC_R1           35047       TMLHE    Intronic  
##        PreB                PreB._.Rep2            1586       TMLHE    Intronic  
##        PreB                PreB._.Rep2              31       TMLHE    Promoter  
##          NK        NK._.scATAC_PBMC_R1           19644       TMLHE      Distal  
##             distToTSS  nearestTSS        GC       idx  
##             <integer> <character> <numeric> <integer>  
##        Mono     10152  uc001aau.3     0.483         1  
##          NK         0  uc001aau.3    0.6906         2  
##           B     10925  uc021oei.1    0.4371         3  
##           B      6892  uc021oei.1    0.7285         4  
##         CLP      1239  uc010nxu.2    0.7904         5  
##         ...       ...         ...       ...       ...  
##   Erythroid     22387  uc004cim.2     0.515      3462  
##          NK     35047  uc004cin.3     0.525      3463  
##        PreB      1586  uc004cin.3     0.485      3464  
##        PreB        31  uc004cin.3    0.5888      3465  
##          NK     19644  uc004cin.3    0.4212      3466  
##   -------  
##   seqinfo: 23 sequences from an unspecified genome; no seqlengths  

## Calling Peaks w/ TileMatrix

As mentioned previously, ArchR also implements its own native peak caller. While we have benchmarked this peak caller against MACS2 and note very similar performances, we do not recommend using this native peak caller unless absolutely necessary.

The ArchR native peak caller calls peaks on the 500-bp `TileMatrix` and we indicate to `addReproduciblePeakSet()` that we want to use this peak caller via the `peakMethod` parameter. Note that we are not storing the output into the `projHeme4` object because we do not intend to keep this peak set and this analysis is only for illustrative purposes. Storage into the `ArchRProject` object would overwrite the previous reproducible peak set already stored in `projHeme4`.

```{r eval=FALSE}
projHemeTmp <- addReproduciblePeakSet(
	ArchRProj = projHeme4, 
	groupBy = "Clusters2",
	peakMethod = "Tiles",
	method = "p"
)
```
> \## ArchR logging to : ArchRLogs/ArchR-addReproduciblePeakSet-faba49485623-Date-2020-04-15_Time-10-27-15.log  
## If there is an issue, please report to github with logFile!  
## Calling Peaks with TileMatrix   
## Group Coverages Already Computed Returning Groups, Set force = TRUE to Recompute!  
## List of length 11  
## names(11): B CD4.M CD4.N CLP Erythroid GMP Mono NK pDC PreB Progenitor  
## 2020-04-15 10:27:16 : Peak Calling Parameters!, 0.012 mins elapsed.  
##                 Group nCells nCellsUsed nReplicates nMin nMax maxPeaks  
## B                   B    439        433           2  182  251   150000  
## CD4.M           CD4.M    678        617           2  117  500   150000  
## CD4.N           CD4.N   1271        552           2   52  500   150000  
## CLP               CLP    387        387           2   82  305   150000  
## Erythroid   Erythroid    879        694           2  194  500   150000  
## GMP               GMP    793        751           2  251  500   150000  
## Mono             Mono   2632       1000           2  500  500   150000  
## NK                 NK    851        801           2  301  500   150000  
## pDC               pDC    320        311           2  150  161   150000  
## PreB             PreB    351        351           2   40  311   150000  
## Progenitor Progenitor   1650        672           2  172  500   150000  
## 2020-04-15 10:27:24 : Computing Total Accessibility Across All Features, 0.138 mins elapsed.  
## 2020-04-15 10:27:27 : Computing Pseudo-Grouped Tile Matrix, 0.19 mins elapsed.  
## 2020-04-15 10:28:07 : Created Pseudo-Grouped Tile Matrix (0.474 GB), 0.866 mins elapsed.  
## Expectation = 0.137769200114613Expectation = 0.170233276575844Expectation = 0.502430087836881Expectation = 0.0743484690298421Expectation = 0.231170400914268Expectation = 0.0293481232153502Expectation = 0.188425424281447Expectation = 0.0412872862125409Expectation = 0.369108556109225Expectation = 0.120760726012825Expectation = 0.33554874172927Expectation = 0.330837760307742Expectation = 0.41068352704434Expectation = 0.343789007051322Expectation = 0.571660173038985Expectation = 0.180700422552374Expectation = 0.126306931769154Expectation = 0.107652051338631Expectation = 0.200607316117261Expectation = 0.0248573103536859Expectation = 0.38762741617292Expectation = 0.0990605372969163  
## 2020-04-15 10:28:51 : Creating Group Peak Sets with Annotations!, 1.59 mins elapsed.  
## 2020-04-15 10:29:03 : Creating Union Peak Set with Annotations!, 1.789 mins elapsed.  
## Annotating Peaks : Nearest Gene  
## Annotating Peaks : Gene  
## Annotating Peaks : TSS  
## Annotating Peaks : GC  
## [1] "plotting ggplot!"  
## 2020-04-15 10:29:22 : Finished Creating Union Peak Set (271917)!, 2.104 mins elapsed.  

```{r include=FALSE, eval=FALSE,echo=FALSE}
# JJJ the structure of the GRanges objects from the two peak calling methods is different. the TileMatrix GRange has something called "Group" whereas the MACS2 GRange does not.
```

We can similarly examine this merged peak set.
```{r eval=FALSE}
getPeakSet(projHemeTmp)
```
> \## GRanges object with 271917 ranges and 9 metadata columns:  
##            seqnames              ranges strand |   mlog10p      Group  
##               <Rle>           <IRanges>  <Rle> | <numeric>      <Rle>  
##        [1]     chr1       752000-752499      * |     0.742       Mono  
##        [2]     chr1       752500-752999      * |     6.561       Mono  
##        [3]     chr1       758500-758999      * |      1.11         NK  
##        [4]     chr1       762000-762499      * |     2.461         NK  
##        [5]     chr1       762500-762999      * |    22.536         NK  
##        ...      ...                 ...    ... .       ...        ...  
##   [271913]     chrX 154862000-154862499      * |     1.966         NK  
##   [271914]     chrX 154862500-154862999      * |     1.334         NK  
##   [271915]     chrX 154912500-154912999      * |      2.01  Erythroid  
##   [271916]     chrX 154997000-154997499      * |     1.349 Progenitor  
##   [271917]     chrX 154998000-154998499      * |     1.227       Mono  
##            distToGeneStart nearestGene    peakType distToTSS  nearestTSS  
##                  <integer> <character> <character> <integer> <character>  
##        [1]           10652   LINC00115      Distal     10652  uc001aau.3  
##        [2]           10152   LINC00115      Distal     10152  uc001aau.3  
##        [3]            4152   LINC00115      Distal      4152  uc001aau.3  
##        [4]             652   LINC00115    Promoter       652  uc001aau.3  
##        [5]             152   LINC00115    Promoter       152  uc001aau.3  
##        ...             ...         ...         ...       ...         ...  
##   [271913]           19626       TMLHE      Distal     19626  uc004cin.3  
##   [271914]           20126       TMLHE      Distal     20126  uc004cin.3  
##   [271915]           70126       TMLHE      Distal     70126  uc004cin.3  
##   [271916]          154626       TMLHE      Distal       201  uc004cin.3  
##   [271917]          155626       TMLHE      Distal       797  uc004cin.3  
##                   GC       idx  
##            <numeric> <integer>  
##        [1]     0.376         1  
##        [2]     0.484         2  
##        [3]      0.56         3  
##        [4]     0.574         4  
##        [5]     0.684         5  
##        ...       ...       ...  
##   [271913]      0.43      6212  
##   [271914]      0.42      6213  
##   [271915]     0.368      6214  
##   [271916]     0.542      6215  
##   [271917]     0.476      6216  
##   -------   
##   seqinfo: 23 sequences from an unspecified genome; no seqlengths  


### Comparing the two peak calling methods

To compare the merged peak set generated using MACS2 vs the merged peak set generated using the ArchR native `TileMatrix` peak caller, we can check the perfecent of overlapping peaks etc.

First, we check the percent of MACS2-called peaks that are overlapped by the `TileMatrix`-called peaks
```{r eval=FALSE}
length(subsetByOverlaps(getPeakSet(projHeme4), getPeakSet(projHemeTmp))) / length(getPeakSet(projHeme4))
```
> \## [1] 0.9627246

Then, we check the converse - the percent of `TileMatrix`-called peaks that are overlapped by MACS2-called peaks. You can see that this overlap is not as strong.
```{r eval=FALSE}
length(subsetByOverlaps(getPeakSet(projHemeTmp), getPeakSet(projHeme4))) / length(getPeakSet(projHemeTmp))
```
> \## [1] 0.7533365

If we increase the margins of the peaks to be wider (1000-bp peaks instead of 500-bp peaks), the percent of MACS2-called peaks that are overlapped does not change much.
```{r eval=FALSE}
length(subsetByOverlaps(resize(getPeakSet(projHeme4), 1000, "center"), getPeakSet(projHemeTmp))) / length(getPeakSet(projHeme4))
```
> \## [1] 0.9676687

But the percent of `TileMatrix`-called peaks overlapped by MACS2 does increase.
```{r eval=FALSE}
length(subsetByOverlaps(getPeakSet(projHemeTmp), resize(getPeakSet(projHeme4), 1000, "center"))) / length(getPeakSet(projHemeTmp))
```
> \## [1] 0.8287639

## Add Peak Matrix

We can now save our original `projHeme4` using the `saveArchRProject()` function. This `ArchRProject` contains the MACS2-derived merged peak set.

```{r eval=FALSE}
saveArchRProject(ArchRProj = projHeme4, outputDirectory = "Save-ProjHeme4", load = FALSE)
```
> \## Copying ArchRProject to new outputDirectory : /oak/stanford/groups/howchang/users/jgranja/ArchRTutorial/ArchRBook/BookOutput4/Save-ProjHeme4  
## Copying Arrow Files...  
## Copying Arrow Files (1 of 3)  
## Copying Arrow Files (2 of 3)  
## Copying Arrow Files (3 of 3)  
## Getting ImputeWeights  
## Dropping ImputeWeights...  
## Copying Other Files...  
## Copying Other Files (1 of 6): Embeddings  
## Copying Other Files (2 of 6): GroupCoverages  
## Copying Other Files (3 of 6): IterativeLSI    
## Copying Other Files (4 of 6): IterativeLSI2  
## Copying Other Files (5 of 6): PeakCalls  
## Copying Other Files (6 of 6): Plots  
## Saving ArchRProject...  

To prepare for downstream analyses, we can create a new `ArchRProject` called `projHeme5` and add a new matrix to it containing insertion counts within our new merged peak set. 
```{r eval=FALSE}
projHeme5 <- addPeakMatrix(projHeme4)
```
> \## ArchR logging to : ArchRLogs/ArchR-addPeakMatrix-faba377aa2f2-Date-2020-04-15_Time-10-29-40.log  
## If there is an issue, please report to github with logFile!  
## 2020-04-15 10:29:41 : Batch Execution w/ safelapply!, 0 mins elapsed.  
## ArchR logging successful to : ArchRLogs/ArchR-addPeakMatrix-faba377aa2f2-Date-2020-04-15_Time-10-29-40.log  

We can now see that a new matrix has been added to `projHeme5` called "PeakMatrix". This is another reserved-name matrix similar to `GeneScoreMatrix` and `TileMatrix`. As mentioned previously, each `ArchRProject` object can only have one peak set and one `PeakMatrix`. You can, of course, create an unlimited number of custom feature matrices of different names but `PeakMatrix` is reserved for the matrix of insertion counts derived from the peak set stored in the `ArchRProject`.
```{r eval=FALSE}
getAvailableMatrices(projHeme5)
```
> \## [1] "GeneIntegrationMatrix" "GeneScoreMatrix"       "PeakMatrix"             
## [4] "TileMatrix"

```{r, include=FALSE, eval=FALSE}
save.image(params$out, compress = FALSE)
```
