---
output:
  html_document:
    theme: yeti  # many options for theme, this one is my favorite.
params:
  threads: 20
  rdata: ""
  token: ""
  ref: ""
  out: ""
---

```{r, include=FALSE, eval=FALSE}
params2 <- params
unlockBinding("params", env = .GlobalEnv)
#load("Chapter-13-Footprints.Rdata")
load(params$rdata)
params <- params2
rm(params2)
devtools::install_github("GreenleafLab/ArchR", 
  auth_token = params$token, 
  ref = params$ref,
  repos = BiocManager::repositories(),
  dependencies = FALSE
)
library(ArchR)
fn <- unclass(lsf.str(envir = asNamespace("ArchR"), all = TRUE))
fn <- fn[!grepl("\\.", fn)]
fn <- fn[fn!="ArchRProj"]
for (i in seq_along(fn)){
    tryCatch({
        eval(parse(text = paste0(fn[i], "<-ArchR::", fn[i])))
    }, error = function(x) {
    })
}
addArchRThreads(threads = params$threads)
addArchRGenome("hg19")
# fn <- unclass(lsf.str(envir = asNamespace("ArchR"), all = TRUE))
# fn <- fn[fn!="ArchRProj"]
# for (i in seq_along(fn)) {
#     tryCatch({
#         eval(parse(text = paste0(fn[i], "<-ArchR:::", fn[i])))
#     }, error = function(x) {
#     })
# }
set.seed(1)
```

# Trajectory Analysis with ArchR

To order cells in pseudo-time, ArchR creates cellular trajectories that order cells across a lower N-dimensional subspace within an ArchRProject. Previously, we have performed this ordering in the 2-dimensional UMAP subspace but ArchR has improved upon this methodology to enable alignment within an N-dimensional subspace (i.e. LSI). First, ArchR requires a user-defined trajectory backbone that provides a rough ordering of cell groups/clusters. For example, given user-determined cluster identities, one might provide the cluster IDs for a stem cell cluster, then a progenitor cell cluster, and then a differentiated cell cluster that correspond to a known or presumed biologically relevant cellular trajectory (i.e. providing the cluster IDs for HSC, to GMP, to Monocyte). Next, for each cluster, ArchR calculates the mean coordinates for each cell group/cluster in N-dimensions and retains cells whose Euclidean distance to those mean coordinates is in the top 5% of all cells. Next, ArchR computes the distance for each cell from cluster i to the mean coordinates of cluster i+1 along the trajectory and computes a pseudo-time vector based on these distances for each iteration of i. This allows ArchR to determine an N-dimensional coordinate and a pseudo-time value for each of the cells retained as part of the trajectory based on the Euclidean distance to the cell group/cluster mean coordinates. Next, ArchR fits a continuous trajectory to each N-dimensional coordinate based on the pseudo-time value using the `smooth.spline` function. Then, ArchR aligns all cells to the trajectory based on their Euclidean distance to the nearest point along the manifold. ArchR then scales this alignment to 100 and stores this pseudo-time in the ArchRProject for downstream analyses.

ArchR can create matrices that convey pseudo-time trends across features stored within the Arrow files. For example, ArchR can analyze changes in TF deviations, gene scores, or integrated gene expression across pseudo-time to identify regulators or regulatory elements that are dynamic throughout the cellular trajectory. First, ArchR groups cells in small user-defined quantile increments (default = 1/100) across the cellular trajectory. ArchR then smooths this matrix per feature using a user-defined smoothing window (default = 9/100) using the `data.table::frollmean` function. ArchR then returns this smoothed pseudo-time x feature matrix as a `SummarizedExperiment` for downstream analyses. ArchR additionally can correlate two of these smoothed pseudo-time x feature matrices using name matching (i.e. positive regulators with chromVAR TF deviations and gene score/integration profiles) or by genomic position overlap methods (i.e. peak-to-gene linkages) using low-overlapping cellular aggregates as described in previous sections. Thus, ArchR facilitates integrative analyses across cellular trajectories, revealing correlated regulatory dynamics across multi-modal data.


## Myeloid Trajectory - Monocyte Differentiation

In this section, we will create a cellular trajectory that approximates the differentiation of HSCs into fully differentiated monocytes. To start, lets review the clusters and cell types that we defined previously, stored in `cellColData` in columns named "Clusters" and "Clusters2". Overlaying these cell groupings on our UMAP embedding shows the different cell types that we are interested in.

```{r eval=FALSE}
p1 <- plotEmbedding(ArchRProj = projHeme5, colorBy = "cellColData", name = "Clusters", embedding = "UMAP")
```
> \## ArchR logging to : ArchRLogs/ArchR-plotEmbedding-15e1326ab29a8-Date-2020-04-15_Time-14-59-47.log  
## If there is an issue, please report to github with logFile!  
## Getting UMAP Embedding  
## ColorBy = cellColData  
## Plotting Embedding  
## 1   
## ArchR logging successful to : ArchRLogs/ArchR-plotEmbedding-15e1326ab29a8-Date-2020-04-15_Time-14-59-47.log  

```{r eval=FALSE}
p2 <- plotEmbedding(ArchRProj = projHeme5, colorBy = "cellColData", name = "Clusters2", embedding = "UMAP")
```
> \## ArchR logging to : ArchRLogs/ArchR-plotEmbedding-15e1339bfbfcf-Date-2020-04-15_Time-14-59-49.log  
## If there is an issue, please report to github with logFile!  
## Getting UMAP Embedding  
## ColorBy = cellColData  
## Plotting Embedding  
## 1   
## ArchR logging successful to : ArchRLogs/ArchR-plotEmbedding-15e1339bfbfcf-Date-2020-04-15_Time-14-59-49.log  

```{r eval=FALSE}
ggAlignPlots(p1, p2, type = "h")
```

```{r, include=FALSE, eval=FALSE}
plotPDF(
    ggAlignPlots(p1, p2, type = "h", draw=FALSE), 
    name = "Plot-UMAP-Clusters12-Combined.pdf", 
    ArchRProj = projHeme5,
    addDOC = FALSE,
    width = 10, 
    height = 5
)
ArchR:::.convertToPNG(ArchRProj = projHeme5)
system("cp Figures/*.png images/HemeWalkthrough/PNG/")
system("cp Figures/*.pdf images/HemeWalkthrough/PDF/")
```

![](images/HemeWalkthrough/PNG/Plot-UMAP-Clusters12-Combined_1.png){width=800 height=400}

### Pseudo-time UMAPs and individual feature plots

We will use the cell type definitions that are stored in "Clusters2". As mentioned above, we are creating a trajectory from stem cells ("Progenitor"), through the commited myeloid progenitor ("GMP"), to monocytes ("Mono"). The first step in creating a trajectory is to create a trajectory backbone in the form of an ordered vector of cell group labels.
```{r eval=FALSE}
trajectory <- c("Progenitor", "GMP", "Mono")
trajectory
```
> \## [1] "Progenitor" "GMP"        "Mono"

We use the `addTrajectory()` function to create a trajectory and we add this to our `ArchRProject`. We will call this trajectory "MyeloidU". What this does is it creates a new column in `cellColData` called "MyeloidU" that stores the pseudo-time value for each cell in the trajectory. Cells that are not part of the trajectory are labeled with `NA`.
```{r eval=FALSE}
projHeme5 <- addTrajectory(
    ArchRProj = projHeme5, 
    name = "MyeloidU", 
    groupBy = "Clusters2",
    trajectory = trajectory, 
    embedding = "UMAP", 
    force = TRUE
)
```
> \## ArchR logging to : ArchRLogs/ArchR-addTrajectory-15e132f14b12b-Date-2020-04-15_Time-15-00-01.log  
## If there is an issue, please report to github with logFile!  
## Filtering outliers  
## Initial Alignment Before Spline Fit  
## Spline Fit  
## KNN to Spline  
## Aligning cells not in trajectory  
## ArchR logging successful to : ArchRLogs/ArchR-addTrajectory-15e132f14b12b-Date-2020-04-15_Time-15-00-01.log  

We can look at this information and see that each cell has a unique pseudotime value between 0 and 100. We exclude cells with `NA` values because these are not part of the trajectory.
```{r eval=FALSE}
head(projHeme5$MyeloidU[!is.na(projHeme5$MyeloidU)])
```
> \## [1] 46.07479 53.75027 44.82834 43.18828 47.49617 43.21015

To plot this trajectory, we use the `plotTrajectory()` function which overlays the pseudo-time values on our UMAP embedding and displays an arrow approximating the trajectory path from the spline-fit. Cells that are not part of the trajectory are colored gray. In this example, we use `colorBy = "cellColData"`  to tell ArchR to look within `cellColData` for the column specified by `name` - in this case, the "MyeloidU" pseudo-time trajectory. While it may seem conterintuitive to list "MyeloidU" for both `trajectory` and `name`, this is done because `trajectory` tells ArchR which subset of cells we are interested in and `name` tells ArchR how to color that subset of cells.
```{r eval=FALSE}
p <- plotTrajectory(projHeme5, trajectory = "MyeloidU", colorBy = "cellColData", name = "MyeloidU")
```
> \## ArchR logging to : ArchRLogs/ArchR-plotTrajectory-15e136ad2f619-Date-2020-04-15_Time-15-00-03.log  
## If there is an issue, please report to github with logFile!  
## Plotting  
## Warning: Removed 5678 rows containing non-finite values (stat_summary_hex).  
## Plotting Trajectory  
## Adding Inferred Arrow Trajectory to Plot  
## ArchR logging successful to : ArchRLogs/ArchR-plotTrajectory-15e136ad2f619-Date-2020-04-15_Time-15-00-03.log  

```{r eval=FALSE}
p[[1]]
```
> \## Warning: Removed 5678 rows containing non-finite values (stat_summary_hex).

```{r, include=FALSE, eval=FALSE}
plotPDF(p, name = "Plot-MyeloidU-Traj-UMAP.pdf", ArchRProj = projHeme5, addDOC = FALSE, width = 5, height = 5)
ArchR:::.convertToPNG(ArchRProj = projHeme5)
system("cp Figures/*.png images/HemeWalkthrough/PNG/")
system("cp Figures/*.pdf images/HemeWalkthrough/PDF/")
```

![](images/HemeWalkthrough/PNG/Plot-MyeloidU-Traj-UMAP_1.png){width=500 height=500}

To save an editable vectorized version of this plot, we use `plotPDF()`.

```{r eval=FALSE}
plotPDF(p, name = "Plot-MyeloidU-Traj-UMAP.pdf", ArchRProj = projHeme5, addDOC = FALSE, width = 5, height = 5)
```
> \## [1] "plotting ggplot!"  
## Warning: Removed 5678 rows containing non-finite values (stat_summary_hex).  
## Warning: Removed 5678 rows containing non-finite values (stat_summary_hex).  
## [1] "plotting ggplot!"  
## `geom_smooth()` using method = 'gam' and formula 'y ~ s(x, bs = "cs")'  
## `geom_smooth()` using method = 'gam' and formula 'y ~ s(x, bs = "cs")'  
## [1] 0  

We can also overlay other features on the trajectory within our UMAP embedding. This allows us to display specific features only within the cells that are relevant to our trajectory.

If you have not already added impute weights to your `projHeme5` project, lets do that now.
```{r eval=FALSE}
projHeme5 <- addImputeWeights(projHeme5)
```
> \## 2020-04-15 15:00:28 : Computing Impute Weights Using Magic (Cell 2018), 0 mins elapsed.  
## 2020-04-15 15:00:40 : Completed Getting Magic Weights!, 0.206 mins elapsed.

Then, we can plot the "MyeloidU" trajectory but color the cells by the gene score value of the CEBPB gene, a known regulator of monocyte function that becomes active during the differentiation process. We indicate the matrix to use via the `colorBy` parameter and the feature to use via the `name` parameter. 
```{r eval=FALSE}
p1 <- plotTrajectory(projHeme5, trajectory = "MyeloidU", colorBy = "GeneScoreMatrix", name = "CEBPB", continuousSet = "horizonExtra")
```
> \## Getting ImputeWeights  
## ArchR logging to : ArchRLogs/ArchR-plotTrajectory-15e13179524ae-Date-2020-04-15_Time-15-00-40.log  
## If there is an issue, please report to github with logFile!  
## Getting Matrix Values...  
## Getting Matrix Values...  
##   
## Imputing Matrix  
## Using weights on disk  
## Using weights on disk  
## Plotting  
## Warning: Removed 5678 rows containing non-finite values (stat_summary_hex).  
## Plotting Trajectory  
## Adding Inferred Arrow Trajectory to Plot  
## ArchR logging successful to : ArchRLogs/ArchR-plotTrajectory-15e13179524ae-Date-2020-04-15_Time-15-00-40.log  

We can repeat this process but color the cells by their linked gene expression via the `GeneIntegrationMatrix`.
```{r eval=FALSE}
p2 <- plotTrajectory(projHeme5, trajectory = "MyeloidU", colorBy = "GeneIntegrationMatrix", name = "CEBPB", continuousSet = "blueYellow")
```
> \## Getting ImputeWeights  
## ArchR logging to : ArchRLogs/ArchR-plotTrajectory-15e133644196a-Date-2020-04-15_Time-15-00-47.log  
## If there is an issue, please report to github with logFile!  
## Getting Matrix Values...  
## Getting Matrix Values...  
##   
## Imputing Matrix  
## Using weights on disk  
## Using weights on disk  
## Plotting  
## Warning: Removed 5678 rows containing non-finite values (stat_summary_hex).  
## Plotting Trajectory   
## Adding Inferred Arrow Trajectory to Plot  
## ArchR logging successful to : ArchRLogs/ArchR-plotTrajectory-15e133644196a-Date-2020-04-15_Time-15-00-47.log  

The `plotTrajectory()` function actually returns a list of relevant plots. The first plot in the list is a UMAP embedding, colorized as specified in the function call.

Comparing these UMAP plots side-by-side for the gene score and gene expression, we see that the activity of the CEBPB gene is highly specific to monocyte cells in the later part of the pseudo-time trajectory.
```{r eval=FALSE}
ggAlignPlots(p1[[1]], p2[[1]], type = "h")
```
> \## Warning: Removed 5678 rows containing non-finite values (stat_summary_hex).  
## Warning: Removed 5678 rows containing non-finite values (stat_summary_hex).

```{r, include=FALSE, eval=FALSE}
plotPDF(
    ggAlignPlots(p1[[1]], p2[[1]], type = "h", draw = FALSE), 
    name = "Plot-UMAP-CEBPB-Combined.pdf", 
    ArchRProj = projHeme5,
    addDOC = FALSE,
    width = 10, 
    height = 5
)

plotPDF(
    ggAlignPlots(p1[[2]], p2[[2]], type = "h", draw = FALSE), 
    name = "Plot-UMAP-CEBPB-Combined2.pdf", 
    ArchRProj = projHeme5,
    addDOC = FALSE,
    width = 10, 
    height = 5
)
ArchR:::.convertToPNG(ArchRProj = projHeme2)
system("cp Figures/*.png images/HemeWalkthrough/PNG/")
system("cp Figures/*.pdf images/HemeWalkthrough/PDF/")
```

![](images/HemeWalkthrough/PNG/Plot-UMAP-CEBPB-Combined_1.png){width=800 height=400}

The second plot that is returned by `plotTrajectory()` is a dot plot of pseudo-time versus the value of the relevant feature, in this case, the gene score or gene expression of CEBPB. In this case, the cells are colored by their pseudo-time.
```{r eval=FALSE}
ggAlignPlots(p1[[2]], p2[[2]], type = "h")
```
> \## `geom_smooth()` using method = 'gam' and formula 'y ~ s(x, bs = "cs")'  
## `geom_smooth()` using method = 'gam' and formula 'y ~ s(x, bs = "cs")'

![](images/HemeWalkthrough/PNG/Plot-UMAP-CEBPB-Combined2_1.png){width=800 height=400}

### Pseudo-time heatmaps

We can visualize changes in many features across pseudo-time using heatmaps. To do this, we first retrieve the trajectory of interest from the `ArchRProject` using the `getTrajectory()` function which returns the trajectory as a `SummarizedExperiment` object. We will create these pseudo-time heatmaps for motifs, gene scores, gene expression, and peak accessibility by passing the corresponding matrix to the `useMatrix` parameter. 

```{r eval=FALSE}
trajMM  <- getTrajectory(ArchRProj = projHeme5, name = "MyeloidU", useMatrix = "MotifMatrix", log2Norm = FALSE)
```
> \## Creating Trajectory Group Matrix..  
## Some values are below 0, this could be a DeviationsMatrix in which scaleTo should be set = NULL.  
## Continuing without depth normalization!  
## Smoothing...

We then pass this `SummarizedExperiment` to the `plotTrajectoryHeatmap()` function.
```{r eval=FALSE}
p1 <- plotTrajectoryHeatmap(trajMM, pal = paletteContinuous(set = "solarExtra"))
```
> \## ArchR logging to : ArchRLogs/ArchR-plotTrajectoryHeatmap-15e131c8ea2de-Date-2020-04-15_Time-15-01-31.log  
## If there is an issue, please report to github with logFile!  
## useSeqnames is NULL or greater than 1 with a Sparse.Assays.Matrix trajectory input.  
## 2020-04-15 15:01:31 :  
## force=FALSE thus continuing with subsetting useSeqnames = z  
## 2020-04-15 15:01:31 :  
## Removing rows with NA values...  
## 2020-04-15 15:01:31 :  
## Preparing Main Heatmap..  
## ArchR logging successful to : ArchRLogs/ArchR-plotTrajectoryHeatmap-15e131c8ea2de-Date-2020-04-15_Time-15-01-31.log  

```{r, include=FALSE, eval=FALSE}
plotPDF(p1, name = "Plot-MyeloidU-Traj-Heatmaps.pdf", ArchRProj = projHeme5, addDOC = FALSE, width = 6, height = 8)
ArchR:::.convertToPNG(ArchRProj = projHeme2)
system("cp Figures/*.png images/HemeWalkthrough/PNG/")
system("cp Figures/*.pdf images/HemeWalkthrough/PDF/")
```

![](images/HemeWalkthrough/PNG/Plot-MyeloidU-Traj-Heatmaps_1.png){width=600 height=800}

We can perform the same steps to obtain a pseudo-time heatmap of gene scores by setting `useMatrix = "GeneScoreMatrix"`.
```{r eval=FALSE}
trajGSM <- getTrajectory(ArchRProj = projHeme5, name = "MyeloidU", useMatrix = "GeneScoreMatrix", log2Norm = TRUE)
```
> \## Creating Trajectory Group Matrix..  
## Smoothing...

```{r eval=FALSE}
p2 <- trajectoryHeatmap(trajGSM,  pal = paletteContinuous(set = "horizonExtra"))
```
> \## ArchR logging to : ArchRLogs/ArchR-trajectoryHeatmap-15e132f1f4173-Date-2020-04-15_Time-15-01-51.log  
## If there is an issue, please report to github with logFile!  
## Preparing Main Heatmap..  
## ArchR logging successful to : ArchRLogs/ArchR-trajectoryHeatmap-15e132f1f4173-Date-2020-04-15_Time-15-01-51.log  

```{r, include=FALSE, eval=FALSE}
plotPDF(p1, p2, name = "Plot-MyeloidU-Traj-Heatmaps.pdf", ArchRProj = projHeme5, addDOC = FALSE, width = 6, height = 8)
ArchR:::.convertToPNG(ArchRProj = projHeme2)
system("cp Figures/*.png images/HemeWalkthrough/PNG/")
system("cp Figures/*.pdf images/HemeWalkthrough/PDF/")
```

![](images/HemeWalkthrough/PNG/Plot-MyeloidU-Traj-Heatmaps_2.png){width=600 height=800}

Similarly, we can obtain a pseudo-time heatmap of gene expression by setting `useMatrix = "GeneIntegrationMatrix"`.
```{r eval=FALSE}
trajGIM <- getTrajectory(ArchRProj = projHeme5, name = "MyeloidU", useMatrix = "GeneIntegrationMatrix", log2Norm = FALSE)
```
> \## Creating Trajectory Group Matrix..  
## Smoothing...

```{r eval=FALSE}
p3 <- plotTrajectoryHeatmap(trajGIM,  pal = paletteContinuous(set = "blueYellow"))
```
> \## ArchR logging to : ArchRLogs/ArchR-plotTrajectoryHeatmap-15e1333158a4-Date-2020-04-15_Time-15-02-18.log  
## If there is an issue, please report to github with logFile!  
## Preparing Main Heatmap..  
## ArchR logging successful to : ArchRLogs/ArchR-plotTrajectoryHeatmap-15e1333158a4-Date-2020-04-15_Time-15-02-18.log  

```{r, include=FALSE, eval=FALSE}
plotPDF(p1, p2, p3, name = "Plot-MyeloidU-Traj-Heatmaps.pdf", ArchRProj = projHeme5, addDOC = FALSE, width = 6, height = 8)
ArchR:::.convertToPNG(ArchRProj = projHeme2)
system("cp Figures/*.png images/HemeWalkthrough/PNG/")
system("cp Figures/*.pdf images/HemeWalkthrough/PDF/")
```

![](images/HemeWalkthrough/PNG/Plot-MyeloidU-Traj-Heatmaps_3.png){width=600 height=800}

Lastly, we can obtain a pseudo-time heatmap of peak accessibility by setting `useMatrix = "PeakMatrix"`.
```{r eval=FALSE}
trajPM  <- getTrajectory(ArchRProj = projHeme5, name = "MyeloidU", useMatrix = "PeakMatrix", log2Norm = TRUE)
```
> \## Creating Trajectory Group Matrix..  
## Smoothing...

```{r eval=FALSE}
p4 <- plotTrajectoryHeatmap(trajPM, pal = paletteContinuous(set = "solarExtra"))
```
> \## ArchR logging to : ArchRLogs/ArchR-plotTrajectoryHeatmap-15e131ecfb872-Date-2020-04-15_Time-15-03-01.log  
## If there is an issue, please report to github with logFile!  
## Preparing Main Heatmap..  
## ArchR logging successful to : ArchRLogs/ArchR-plotTrajectoryHeatmap-15e131ecfb872-Date-2020-04-15_Time-15-03-01.log  

```{r, include=FALSE, eval=FALSE}
plotPDF(p1, p2, p3, p4, name = "Plot-MyeloidU-Traj-Heatmaps.pdf", ArchRProj = projHeme5, addDOC = FALSE, width = 6, height = 8)
ArchR:::.convertToPNG(ArchRProj = projHeme2)
system("cp Figures/*.png images/HemeWalkthrough/PNG/")
system("cp Figures/*.pdf images/HemeWalkthrough/PDF/")
```

![](images/HemeWalkthrough/PNG/Plot-MyeloidU-Traj-Heatmaps_4.png){width=600 height=800}

To save an editable vectorized version of this plot, we use `plotPDF()`.

```{r eval=FALSE}
plotPDF(p1, p2, p3, p4, name = "Plot-MyeloidU-Traj-Heatmaps.pdf", ArchRProj = projHeme5, addDOC = FALSE, width = 6, height = 8)
```
> \## [1] 0

### Integrative pseudo-time analyses

We can also perform integrative analyses, such as identification of positive TF regulators by integration of gene scores / gene expression with motif accessibility across pseudo-time. This can be very powerful, for example in identifying drivers of differentiation. To do this, we use the `correlateTrajectories()` function which takes two `SummarizedExperiment` objects retrived from the `getTrajectories()` function.

First, lets find motifs whose accessibility across pseudo-time is correlated with the gene score of the TF gene.
```{r eval=FALSE}
corGSM_MM <- correlateTrajectories(trajGSM, trajMM)
```
> \## ArchR logging to : ArchRLogs/ArchR-correlateTrajectories-15e13b9ca02a-Date-2020-04-15_Time-15-03-37.log  
## If there is an issue, please report to github with logFile!  
## Found 36 Correlated Pairings!  
## 2020-04-15 15:03:56 :  

The primary output of `correlateTrajectories()` is a list object containing a `DataFrame` object as the first entry in the list. This `DataFrame` has columns named `idx1`, `matchname1`, `name1`, and `VarAssay1` which correspond to the index, match name, unaltered name, and the variance quantile of the features from the first trajectory (gene scores) passed to the `correlateTrajectories()` function. A "variance quantile" is a normalized measure of the given feature which allows us to derive a correlation across disparate assays. This `DataFrame` contains all of the features that met the cutoffs specified in the `correlateTrajectories()` function.

```{r eval=FALSE}
corGSM_MM[[1]]
```
> \## DataFrame with 36 rows and 12 columns  
##          idx1      idx2 matchname1 matchname2       name1        name2  
##     <integer> <integer>    <array>    <array> <character>  <character>  
## 1          82      1081     PRDM16     PRDM16 chr1:PRDM16 z:PRDM16_211  
## 2         731       932       TAL1       TAL1   chr1:TAL1    z:TAL1_62  
## 3         818      1013        JUN        JUN    chr1:JUN    z:JUN_143  
## 4        2034      1002       ATF3       ATF3   chr1:ATF3   z:ATF3_132  
## 5        2369      1254      GATA3      GATA3 chr10:GATA3  z:GATA3_384  
## ...       ...       ...        ...        ...         ...          ...  
## 32      20775      1022      CEBPD      CEBPD  chr8:CEBPD  z:CEBPD_152  
## 33      20780      1031      SNAI2      SNAI2  chr8:SNAI2  z:SNAI2_161  
## 34      21658      1003      NFIL3      NFIL3  chr9:NFIL3  z:NFIL3_133  
## 35      21793      1078       KLF4       KLF4   chr9:KLF4   z:KLF4_208  
## 36      22097      1565       RXRA       RXRA   chr9:RXRA   z:RXRA_695  
##           Correlation         VarAssay1         VarAssay2            TStat  
##             <numeric>         <numeric>         <numeric>        <numeric>  
## 1   0.859588836579798 0.999783802481948 0.860344827586207 16.6530781620713  
## 2   0.687938524545333 0.866735849872443 0.987931034482759 9.38348743943718  
## 3   0.530335586469274 0.993989708998141 0.963218390804598 6.19265152142139  
## 4   0.776280918431044 0.985168850261599 0.936781609195402 12.1905867620732  
## 5   0.714559116657296 0.801573917931422 0.989655172413793  10.111530171907  
## ...               ...               ...               ...              ...  
## 32   0.73809434207349 0.993816750983699 0.998275862068966 10.8296844080467  
## 33  0.643337129188769  0.87252994335625 0.983333333333333 8.31877485229719  
## 34  0.833740261914099 0.992303368357331 0.893103448275862 14.9479403447508  
## 35   0.82265894955927 0.998702814891685 0.905747126436782 14.3243696597426  
## 36   0.59509316489084  0.88640982401522 0.890229885057471 7.33039570155334  
##                     Pval                  FDR  
##                <numeric>            <numeric>  
## 1   2.47466579579935e-30 4.04855324192774e-27  
## 2   2.63756432350363e-15 1.00350121703533e-13  
## 3   1.38781951660218e-08 1.55511830764464e-07  
## 4   2.36537648993666e-21 3.22479661461365e-19  
## 5    6.9322610846337e-17 3.78039304482025e-15  
## ...                  ...                  ...  
## 32  1.93304822797782e-18 1.31769454207155e-16  
## 33  5.27319420246801e-13  1.3914428572964e-11  
## 34  5.06319064599113e-27 2.70186610322912e-24  
## 35   8.9984663184347e-26 2.45358181615986e-23  
## 36  6.60626612679289e-11 1.25672690505037e-09  

We can then subset our corresponding trajectory `SummarizedExperiment` objects to only contain the elements that passed significance above.
```{r eval=FALSE}
trajGSM2 <- trajGSM[corGSM_MM[[1]]$name1, ]
trajMM2 <- trajMM[corGSM_MM[[1]]$name2, ]
```
To best order these features, we can create a new trajectory where the values of these two trajectories are multiplied. This will allow us to create side-by-side heatmaps that are identically ordered by row.
```{r eval=FALSE}
trajCombined <- trajGSM2
assay(trajCombined) <- t(apply(assay(trajGSM2), 1, scale)) + t(apply(assay(trajMM2), 1, scale))
```

```{r include=FALSE, eval=FALSE,echo=FALSE}
#JJJ This whole row order thing went a bit over my head. Could use some extra clarification on how this works.
```

We can extract the optimal row order from the return of the `plotTrajectoryHeatmap()` function.
```{r eval=FALSE}
combinedMat <- plotTrajectoryHeatmap(trajCombined, returnMat = TRUE, varCutOff = 0)
```
> \## ArchR logging to : ArchRLogs/ArchR-plotTrajectoryHeatmap-15e136ff55add-Date-2020-04-15_Time-15-03-56.log  
## If there is an issue, please report to github with logFile!  
## Preparing Main Heatmap..  
## ArchR logging successful to : ArchRLogs/ArchR-plotTrajectoryHeatmap-15e136ff55add-Date-2020-04-15_Time-15-03-56.log  

```{r eval=FALSE}
rowOrder <- match(rownames(combinedMat), rownames(trajGSM2))
```

With this, we are now ready to create our paired heatmaps. First, we will create the heatmap for the gene score trajectory. We specify the desired row order via the `rowOrder` parameter.
```{r eval=FALSE}
ht1 <- plotTrajectoryHeatmap(trajGSM2,  pal = paletteContinuous(set = "horizonExtra"),  varCutOff = 0, rowOrder = rowOrder)
```
> \## ArchR logging to : ArchRLogs/ArchR-plotTrajectoryHeatmap-15e13aac5f3b-Date-2020-04-15_Time-15-03-57.log  
## If there is an issue, please report to github with logFile!  
## Preparing Main Heatmap..  
## ArchR logging successful to : ArchRLogs/ArchR-plotTrajectoryHeatmap-15e13aac5f3b-Date-2020-04-15_Time-15-03-57.log  

Then, we will create the heatmap for the motif trajectory, again specifying the row order via the `rowOrder` parameter
```{r eval=FALSE}
ht2 <- plotTrajectoryHeatmap(trajMM2, pal = paletteContinuous(set = "solarExtra"), varCutOff = 0, rowOrder = rowOrder)
```
> \## ArchR logging to : ArchRLogs/ArchR-plotTrajectoryHeatmap-15e1323caf784-Date-2020-04-15_Time-15-03-57.log  
## If there is an issue, please report to github with logFile!  
## useSeqnames is NULL or greater than 1 with a Sparse.Assays.Matrix trajectory input.  
## 2020-04-15 15:03:58 :  
## force=FALSE thus continuing with subsetting useSeqnames = z  
## 2020-04-15 15:03:58 :  
## Preparing Main Heatmap..  
## ArchR logging successful to : ArchRLogs/ArchR-plotTrajectoryHeatmap-15e1323caf784-Date-2020-04-15_Time-15-03-57.log  

Plotting these two heatmaps side-by-side, we see that the rows are matched across the two heatmaps. You may notice that the analysis captures both GATA3 and GATA3-AS1 (an anti-sense transcript of GATA3). This is due to how feature matching is performed and the anti-sense transcript entry could be removed manually in post-processing or programatically if desired.
```{r eval=FALSE}
ht1 + ht2
```

```{r, include=FALSE, eval=FALSE}
plotPDF(ht1 + ht2, name = "Plot-MyeloidU-GS-TF.pdf", ArchRProj = projHeme5, addDOC = FALSE, width = 8, height = 6)
ArchR:::.convertToPNG(ArchRProj = projHeme2)
system("cp Figures/*.png images/HemeWalkthrough/PNG/")
system("cp Figures/*.pdf images/HemeWalkthrough/PDF/")
```

![](images/HemeWalkthrough/PNG/Plot-MyeloidU-GS-TF_1.png){width=800 height=600}

We can repeat this same exact process but use gene expression from the `GeneIntegrationMatrix` instead of gene scores. Because this is the same analytical workflow, we do not repeat our explanations for each step.
```{r eval=FALSE}
corGIM_MM <- correlateTrajectories(trajGIM, trajMM)
```
> \## ArchR logging to : ArchRLogs/ArchR-correlateTrajectories-15e13f7dfd04-Date-2020-04-15_Time-15-04-14.log  
## If there is an issue, please report to github with logFile!  
## Found 54 Correlated Pairings!  
## 2020-04-15 15:04:30 :  

```{r eval=FALSE}
corGIM_MM[[1]]
```
> \## DataFrame with 54 rows and 12 columns  
##          idx1      idx2 matchname1 matchname2       name1       name2  
##     <integer> <integer>    <array>    <array> <character> <character>  
## 1         295      1601      RUNX3      RUNX3  chr1:RUNX3 z:RUNX3_731  
## 2         680      1612       NFIA       NFIA   chr1:NFIA  z:NFIA_742  
## 3        1227      1512      MEF2D      MEF2D  chr1:MEF2D z:MEF2D_642  
## 4        1317       930       USF1       USF1   chr1:USF1   z:USF1_60  
## 5        1664      1002       ATF3       ATF3   chr1:ATF3  z:ATF3_132  
## ...       ...       ...        ...        ...         ...         ...  
## 50      16981      1696      KLF10      KLF10  chr8:KLF10 z:KLF10_826  
## 51      17463      1003      NFIL3      NFIL3  chr9:NFIL3 z:NFIL3_133  
## 52      17560      1078       KLF4       KLF4   chr9:KLF4  z:KLF4_208  
## 53      17803      1565       RXRA       RXRA   chr9:RXRA  z:RXRA_695  
## 54      18090       888       TFE3       TFE3   chrX:TFE3   z:TFE3_18  
##           Correlation         VarAssay1         VarAssay2            TStat  
##             <numeric>         <numeric>         <numeric>        <numeric>  
## 1   0.506721180509532 0.871189720982743  0.98448275862069 5.81861222342308  
## 2   0.899884055575892 0.878017310897264 0.919540229885057 20.4260514493749  
## 3   0.787682536829653 0.815493790656416 0.841379310344828  12.656866457322  
## 4   0.653303634000368 0.804204075049728 0.875287356321839 8.54234597517229  
## 5   0.847460093565636 0.813020805333047 0.936781609195402 15.8034906427774  
## ...               ...               ...               ...              ...  
## 50   0.57964654735637 0.953335842159024 0.864367816091954 7.04188649257934  
## 51  0.849563937747037 0.933928283425622 0.893103448275862 15.9440428922802  
## 52  0.820699669277503  0.98876404494382 0.905747126436782 14.2196070232556  
## 53  0.899871735441299 0.960485995376593 0.890229885057471  20.424581355724  
## 54  0.832322259515336 0.801247244771786 0.886206896551724 14.8650341763864  
##                     Pval                  FDR  
##                <numeric>            <numeric>  
## 1   7.47838268446286e-08 2.00835886256963e-07  
## 2   4.28849849649256e-37 2.40519926139511e-35  
## 3   2.45911764921563e-22  2.0096926995314e-21  
## 4    1.7427906036477e-13 7.90509804908143e-13  
## 5   1.05531067509617e-28  1.5631789374862e-27  
## ...                  ...                  ...  
## 50  2.63272438071192e-10 8.66605108651006e-10  
## 51  5.63648466586524e-29 9.05658892074619e-28  
## 52  1.46590261078753e-25  1.5687313439583e-24  
## 53  4.31312103836676e-37 2.40519926139511e-35  
## 54   7.4030081939147e-27 8.88360983269764e-26  

```{r eval=FALSE}
trajGIM2 <- trajGIM[corGIM_MM[[1]]$name1, ]
trajMM2 <- trajMM[corGIM_MM[[1]]$name2, ]

trajCombined <- trajGIM2
assay(trajCombined) <- t(apply(assay(trajGIM2), 1, scale)) + t(apply(assay(trajMM2), 1, scale))

combinedMat <- plotTrajectoryHeatmap(trajCombined, returnMat = TRUE, varCutOff = 0)
```
> \## ArchR logging to : ArchRLogs/ArchR-plotTrajectoryHeatmap-15e137bbcc972-Date-2020-04-15_Time-15-04-30.log  
## If there is an issue, please report to github with logFile!  
## Preparing Main Heatmap..  
## ArchR logging successful to : ArchRLogs/ArchR-plotTrajectoryHeatmap-15e137bbcc972-Date-2020-04-15_Time-15-04-30.log  

```{r eval=FALSE}
rowOrder <- match(rownames(combinedMat), rownames(trajGIM2))
```
```{r eval=FALSE}
ht1 <- plotTrajectoryHeatmap(trajGIM2,  pal = paletteContinuous(set = "blueYellow"),  varCutOff = 0, rowOrder = rowOrder)
```
> \## ArchR logging to : ArchRLogs/ArchR-plotTrajectoryHeatmap-15e1331e06f29-Date-2020-04-15_Time-15-04-31.log  
## If there is an issue, please report to github with logFile!  
## Preparing Main Heatmap..  
## ArchR logging successful to : ArchRLogs/ArchR-plotTrajectoryHeatmap-15e1331e06f29-Date-2020-04-15_Time-15-04-31.log  

```{r eval=FALSE}
ht2 <- plotTrajectoryHeatmap(trajMM2, pal = paletteContinuous(set = "solarExtra"), varCutOff = 0, rowOrder = rowOrder)
```
> \## ArchR logging to : ArchRLogs/ArchR-plotTrajectoryHeatmap-15e1345e2da25-Date-2020-04-15_Time-15-04-32.log  
## If there is an issue, please report to github with logFile!   
## useSeqnames is NULL or greater than 1 with a Sparse.Assays.Matrix trajectory input.  
## 2020-04-15 15:04:32 :  
## force=FALSE thus continuing with subsetting useSeqnames = z  
## 2020-04-15 15:04:32 :    
## Preparing Main Heatmap..  
## ArchR logging successful to : ArchRLogs/ArchR-plotTrajectoryHeatmap-15e1345e2da25-Date-2020-04-15_Time-15-04-32.log  

```{r eval=FALSE}
ht1 + ht2
```

```{r, include=FALSE, eval=FALSE}
plotPDF(ht1 + ht2, name = "Plot-MyeloidU-GEx-TF.pdf", ArchRProj = projHeme5, addDOC = FALSE, width = 8, height = 6)
ArchR:::.convertToPNG(ArchRProj = projHeme2)
system("cp Figures/*.png images/HemeWalkthrough/PNG/")
system("cp Figures/*.pdf images/HemeWalkthrough/PDF/")
```

![](images/HemeWalkthrough/PNG/Plot-MyeloidU-GEx-TF_1.png){width=800 height=600}


## Lymphoid Trajectory - B Cell Cifferentiation

As a second example of a trajectory, we will create a B cell trajectory from progenitor cells, through the common lymphoid progenitor and pre-B cell all the way to fully differentiated B cells. Because this analysis is essentially repeated from the monocyte trajectory in the previous section, we do not provide explanations for the code snippets. If you are trying to learn how to perform trajectory analysis, check out the monocyte trajectory section in this chapter instead.

```{r eval=FALSE}
p1 <- plotEmbedding(ArchRProj = projHeme5, colorBy = "cellColData", name = "Clusters", embedding = "UMAP")
```
> \## ArchR logging to : ArchRLogs/ArchR-plotEmbedding-15e1340853c58-Date-2020-04-15_Time-15-04-41.log  
## If there is an issue, please report to github with logFile!  
## Getting UMAP Embedding  
## ColorBy = cellColData  
## Plotting Embedding  
## 1   
## ArchR logging successful to : ArchRLogs/ArchR-plotEmbedding-15e1340853c58-Date-2020-04-15_Time-15-04-41.log  

```{r eval=FALSE}
p2 <- plotEmbedding(ArchRProj = projHeme5, colorBy = "cellColData", name = "Clusters2", embedding = "UMAP")
```
> \## ArchR logging to : ArchRLogs/ArchR-plotEmbedding-15e1326401e05-Date-2020-04-15_Time-15-04-42.log  
## If there is an issue, please report to github with logFile!  
## Getting UMAP Embedding  
## ColorBy = cellColData  
## Plotting Embedding  
## 1   
## ArchR logging successful to : ArchRLogs/ArchR-plotEmbedding-15e1326401e05-Date-2020-04-15_Time-15-04-42.log  

```{r eval=FALSE}
ggAlignPlots(p1, p2, type = "h")
```

```{r, include=FALSE, eval=FALSE}
plotPDF(
    ggAlignPlots(p1, p2, type = "h", draw=FALSE), 
    name = "Plot-UMAP-Clusters12-Combined.pdf", 
    ArchRProj = projHeme5,
    addDOC = FALSE,
    width = 10, 
    height = 5
)
ArchR:::.convertToPNG(ArchRProj = projHeme2)
system("cp Figures/*.png images/HemeWalkthrough/PNG/")
system("cp Figures/*.pdf images/HemeWalkthrough/PDF/")
```

![](images/HemeWalkthrough/PNG/Plot-UMAP-Clusters12-Combined_1.png){width=800 height=400}

### Pseudo-time UMAPs and individual feature plots

```{r eval=FALSE}
trajectory <- c("Progenitor", "CLP", "PreB", "B")
trajectory
```
> \## [1] "Progenitor" "CLP"        "PreB"       "B"

```{r eval=FALSE}
projHeme5 <- addTrajectory(
    ArchRProj = projHeme5, 
    name = "LymphoidU", 
    groupBy = "Clusters2",
    trajectory = trajectory, 
    embedding = "UMAP", 
    force = TRUE
)
```
> \## ArchR logging to : ArchRLogs/ArchR-addTrajectory-15e1351d16b68-Date-2020-04-15_Time-15-04-55.log  
## If there is an issue, please report to github with logFile!  
## Filtering outliers  
## Initial Alignment Before Spline Fit  
## Spline Fit   
## KNN to Spline  
## Aligning cells not in trajectory  
## ArchR logging successful to : ArchRLogs/ArchR-addTrajectory-15e1351d16b68-Date-2020-04-15_Time-15-04-55.log  

```{r eval=FALSE}
head(projHeme5$LymphoidU[!is.na(projHeme5$LymphoidU)]) #NA means not in trajectory
```
> \## [1] 87.630792 81.913303 84.342302 81.763827 83.632287  4.073244

```{r eval=FALSE}
p <- plotTrajectory(projHeme5, trajectory = "LymphoidU", colorBy = "cellColData", name = "LymphoidU")
```
> \## ArchR logging to : ArchRLogs/ArchR-plotTrajectory-15e1377b52455-Date-2020-04-15_Time-15-04-56.log  
## If there is an issue, please report to github with logFile!  
## Plotting  
## Warning: Removed 7575 rows containing non-finite values (stat_summary_hex).  
## Plotting Trajectory  
## Adding Inferred Arrow Trajectory to Plot  
## ArchR logging successful to : ArchRLogs/ArchR-plotTrajectory-15e1377b52455-Date-2020-04-15_Time-15-04-56.log  

```{r eval=FALSE}
p[[1]]
```
> \## Warning: Removed 7575 rows containing non-finite values (stat_summary_hex).

```{r, include=FALSE, eval=FALSE}
plotPDF(p, name = "Plot-LymphoidU-Traj-UMAP.pdf", ArchRProj = projHeme5, addDOC = FALSE, width = 5, height = 5)
ArchR:::.convertToPNG(ArchRProj = projHeme2)
system("cp Figures/*.png images/HemeWalkthrough/PNG/")
system("cp Figures/*.pdf images/HemeWalkthrough/PDF/")
```

![](images/HemeWalkthrough/PNG/Plot-LymphoidU-Traj-UMAP_1.png){width=500 height=500}

To save an editable vectorized version of this plot, we use `plotPDF()`.

```{r eval=FALSE}
plotPDF(p, name = "Plot-LymphoidU-Traj-UMAP.pdf", ArchRProj = projHeme5, addDOC = FALSE, width = 5, height = 5)
```
> \## [1] "plotting ggplot!"  
## Warning: Removed 7575 rows containing non-finite values (stat_summary_hex).  
## Warning: Removed 7575 rows containing non-finite values (stat_summary_hex).  
## [1] "plotting ggplot!"  
## `geom_smooth()` using method = 'gam' and formula 'y ~ s(x, bs = "cs")'  
## `geom_smooth()` using method = 'gam' and formula 'y ~ s(x, bs = "cs")'  
## [1] 0  

```{r eval=FALSE}
p1 <- plotTrajectory(projHeme5, trajectory = "LymphoidU", colorBy = "GeneScoreMatrix", name = "PAX5", continuousSet = "horizonExtra")
```
> \## Getting ImputeWeights  
## ArchR logging to : ArchRLogs/ArchR-plotTrajectory-15e13576dc585-Date-2020-04-15_Time-15-05-20.log  
## If there is an issue, please report to github with logFile!  
## Getting Matrix Values...  
## Getting Matrix Values...  
##   
## Imputing Matrix  
## Using weights on disk  
## Using weights on disk  
## Plotting  
## Warning: Removed 7575 rows containing non-finite values (stat_summary_hex).  
## Plotting Trajectory  
## Adding Inferred Arrow Trajectory to Plot  
## ArchR logging successful to : ArchRLogs/ArchR-plotTrajectory-15e13576dc585-Date-2020-04-15_Time-15-05-20.log  

```{r eval=FALSE}
p2 <- plotTrajectory(projHeme5, trajectory = "LymphoidU", colorBy = "GeneIntegrationMatrix", name = "PAX5", continuousSet = "blueYellow")
```
> \## Getting ImputeWeights  
## ArchR logging to : ArchRLogs/ArchR-plotTrajectory-15e13ed9c20a-Date-2020-04-15_Time-15-05-27.log  
## If there is an issue, please report to github with logFile!  
## Getting Matrix Values...  
## Getting Matrix Values...  
##   
## Imputing Matrix  
## Using weights on disk  
## Using weights on disk  
## Plotting  
## Warning: Removed 7575 rows containing non-finite values (stat_summary_hex).  
## Plotting Trajectory  
## Adding Inferred Arrow Trajectory to Plot   
## ArchR logging successful to : ArchRLogs/ArchR-plotTrajectory-15e13ed9c20a-Date-2020-04-15_Time-15-05-27.log  

```{r eval=FALSE}
ggAlignPlots(p1[[1]], p2[[1]], type = "h")
```
> \## Warning: Removed 7575 rows containing non-finite values (stat_summary_hex).  
## Warning: Removed 7575 rows containing non-finite values (stat_summary_hex).

```{r, include=FALSE, eval=FALSE}
plotPDF(
    ggAlignPlots(p1[[1]], p2[[1]], type = "h", draw = FALSE), 
    name = "Plot-UMAP-PAX5-Combined.pdf", 
    ArchRProj = projHeme5,
    addDOC = FALSE,
    width = 10, 
    height = 5
)

plotPDF(
    ggAlignPlots(p1[[2]], p2[[2]], type = "h", draw = FALSE), 
    name = "Plot-UMAP-PAX5-Combined2.pdf", 
    ArchRProj = projHeme5,
    addDOC = FALSE,
    width = 10, 
    height = 5
)
ArchR:::.convertToPNG(ArchRProj = projHeme2)
system("cp Figures/*.png images/HemeWalkthrough/PNG/")
system("cp Figures/*.pdf images/HemeWalkthrough/PDF/")
```

![](images/HemeWalkthrough/PNG/Plot-UMAP-PAX5-Combined_1.png){width=800 height=400}

```{r eval=FALSE}
ggAlignPlots(p1[[2]], p2[[2]], type = "h")
```
> \## `geom_smooth()` using method = 'gam' and formula 'y ~ s(x, bs = "cs")'  
## `geom_smooth()` using method = 'gam' and formula 'y ~ s(x, bs = "cs")'

![](images/HemeWalkthrough/PNG/Plot-UMAP-PAX5-Combined2_1.png){width=800 height=400}

### Pseudo-time heatmaps

```{r eval=FALSE}
trajMM  <- getTrajectory(ArchRProj = projHeme5, name = "LymphoidU", useMatrix = "MotifMatrix", log2Norm = FALSE)
```
> \## Creating Trajectory Group Matrix..  
## Some values are below 0, this could be a DeviationsMatrix in which scaleTo should be set = NULL.  
## Continuing without depth normalization!  
## Smoothing...  

```{r eval=FALSE}
p1 <- plotTrajectoryHeatmap(trajMM, pal = paletteContinuous(set = "solarExtra"))
```
> \## ArchR logging to : ArchRLogs/ArchR-plotTrajectoryHeatmap-15e133830d097-Date-2020-04-15_Time-15-06-14.log  
## If there is an issue, please report to github with logFile!  
## useSeqnames is NULL or greater than 1 with a Sparse.Assays.Matrix trajectory input.  
## 2020-04-15 15:06:15 :  
## force=FALSE thus continuing with subsetting useSeqnames = z  
## 2020-04-15 15:06:15 :  
## Removing rows with NA values...  
## 2020-04-15 15:06:15 :  
## Preparing Main Heatmap..  
## ArchR logging successful to : ArchRLogs/ArchR-plotTrajectoryHeatmap-15e133830d097-Date-2020-04-15_Time-15-06-14.log  

```{r, include=FALSE, eval=FALSE}
plotPDF(p1, name = "Plot-LymphoidU-Traj-Heatmaps.pdf", ArchRProj = projHeme5, addDOC = FALSE, width = 6, height = 8)
ArchR:::.convertToPNG(ArchRProj = projHeme2)
system("cp Figures/*.png images/HemeWalkthrough/PNG/")
system("cp Figures/*.pdf images/HemeWalkthrough/PDF/")
```

![](images/HemeWalkthrough/PNG/Plot-LymphoidU-Traj-Heatmaps_1.png){width=600 height=800}

```{r eval=FALSE}
trajGSM <- getTrajectory(ArchRProj = projHeme5, name = "LymphoidU", useMatrix = "GeneScoreMatrix", log2Norm = TRUE)
```
> \## Creating Trajectory Group Matrix..  
## Smoothing...

```{r eval=FALSE}
p2 <- plotTrajectoryHeatmap(trajGSM,  pal = paletteContinuous(set = "horizonExtra"))
```
> \## ArchR logging to : ArchRLogs/ArchR-plotTrajectoryHeatmap-15e13357a3282-Date-2020-04-15_Time-15-06-36.log  
## If there is an issue, please report to github with logFile!  
## Preparing Main Heatmap..  
## ArchR logging successful to : ArchRLogs/ArchR-plotTrajectoryHeatmap-15e13357a3282-Date-2020-04-15_Time-15-06-36.log  

```{r, include=FALSE, eval=FALSE}
plotPDF(p1, p2, name = "Plot-LymphoidU-Traj-Heatmaps.pdf", ArchRProj = projHeme5, addDOC = FALSE, width = 6, height = 8)
ArchR:::.convertToPNG(ArchRProj = projHeme2)
system("cp Figures/*.png images/HemeWalkthrough/PNG/")
system("cp Figures/*.pdf images/HemeWalkthrough/PDF/")
```

![](images/HemeWalkthrough/PNG/Plot-LymphoidU-Traj-Heatmaps_2.png){width=600 height=800}


```{r eval=FALSE}
trajGIM <- getTrajectory(ArchRProj = projHeme5, name = "LymphoidU", useMatrix = "GeneIntegrationMatrix", log2Norm = FALSE)
```
> \## Creating Trajectory Group Matrix..  
## Smoothing...

```{r eval=FALSE}
p3 <- plotTrajectoryHeatmap(trajGIM,  pal = paletteContinuous(set = "blueYellow"))
```
> \## ArchR logging to : ArchRLogs/ArchR-plotTrajectoryHeatmap-15e131325dad9-Date-2020-04-15_Time-15-07-06.log  
## If there is an issue, please report to github with logFile!  
## Preparing Main Heatmap..  
## ArchR logging successful to : ArchRLogs/ArchR-plotTrajectoryHeatmap-15e131325dad9-Date-2020-04-15_Time-15-07-06.log  

```{r, include=FALSE, eval=FALSE}
plotPDF(p1, p2, p3, name = "Plot-LymphoidU-Traj-Heatmaps.pdf", ArchRProj = projHeme5, addDOC = FALSE, width = 6, height = 8)
ArchR:::.convertToPNG(ArchRProj = projHeme2)
system("cp Figures/*.png images/HemeWalkthrough/PNG/")
system("cp Figures/*.pdf images/HemeWalkthrough/PDF/")
```

![](images/HemeWalkthrough/PNG/Plot-LymphoidU-Traj-Heatmaps_3.png){width=600 height=800}

```{r eval=FALSE}
trajPM  <- getTrajectory(ArchRProj = projHeme5, name = "LymphoidU", useMatrix = "PeakMatrix", log2Norm = TRUE)
```
> \## Creating Trajectory Group Matrix..  
## Smoothing...

```{r eval=FALSE}
p4 <- plotTrajectoryHeatmap(trajPM, pal = paletteContinuous(set = "solarExtra"))
```
> \## ArchR logging to : ArchRLogs/ArchR-plotTrajectoryHeatmap-15e1351d7cd4d-Date-2020-04-15_Time-15-07-48.log  
## If there is an issue, please report to github with logFile!  
## Preparing Main Heatmap..  
## ArchR logging successful to : ArchRLogs/ArchR-plotTrajectoryHeatmap-15e1351d7cd4d-Date-2020-04-15_Time-15-07-48.log  

```{r, include=FALSE, eval=FALSE}
plotPDF(p1, p2, p3, p4, name = "Plot-LymphoidU-Traj-Heatmaps.pdf", ArchRProj = projHeme5, addDOC = FALSE, width = 6, height = 8)
ArchR:::.convertToPNG(ArchRProj = projHeme2)
system("cp Figures/*.png images/HemeWalkthrough/PNG/")
system("cp Figures/*.pdf images/HemeWalkthrough/PDF/")
```

![](images/HemeWalkthrough/PNG/Plot-LymphoidU-Traj-Heatmaps_4.png){width=600 height=800}

To save an editable vectorized version of this plot, we use `plotPDF()`.

```{r eval=FALSE}
plotPDF(p1, p2, p3, p4, name = "Plot-LymphoidU-Traj-Heatmaps.pdf", ArchRProj = projHeme5, addDOC = FALSE, width = 6, height = 8)
```
> \## [1] 0

### Integrative pseudo-time analyses

```{r eval=FALSE}
corGSM_MM <- correlateTrajectories(trajGSM, trajMM)
```
> \## ArchR logging to : ArchRLogs/ArchR-correlateTrajectories-15e133e72f66-Date-2020-04-15_Time-15-08-26.log  
## If there is an issue, please report to github with logFile!  
## Found 16 Correlated Pairings!  
## 2020-04-15 15:08:47 :  

```{r eval=FALSE}
corGSM_MM[[1]]$matchname1
```
> \##  [1] "CREM"   "NFE2"   "FOXO1"  "JDP2"   "MAFG"   "MAFG"   "RFX2"   "KLF2"  
##  [9] "CEBPA"  "CEBPA"  "FOSL2"  "BCL11A" "GATA2"  "IRF2"   "IRF4"   "PAX5"

```{r eval=FALSE}
corGSM_MM[[1]]
```
> \## DataFrame with 16 rows and 12 columns  
##          idx1      idx2 matchname1 matchname2       name1        name2  
##     <integer> <integer>    <array>    <array> <character>  <character>  
## 1        2503       978       CREM       CREM  chr10:CREM   z:CREM_108  
## 2        5181       989       NFE2       NFE2  chr12:NFE2   z:NFE2_119  
## 3        5958      1231      FOXO1      FOXO1 chr13:FOXO1  z:FOXO1_361  
## 4        6721       995       JDP2       JDP2  chr14:JDP2   z:JDP2_125  
## 5       10077      1018       MAFG       MAFG  chr17:MAFG   z:MAFG_148  
## ...       ...       ...        ...        ...         ...          ...  
## 12      12398      1064     BCL11A     BCL11A chr2:BCL11A z:BCL11A_194  
## 13      15802      1258      GATA2      GATA2  chr3:GATA2  z:GATA2_388  
## 14      17102      1504       IRF2       IRF2   chr4:IRF2   z:IRF2_634  
## 15      18161      1502       IRF4       IRF4   chr6:IRF4   z:IRF4_632  
## 16      21499      1579       PAX5       PAX5   chr9:PAX5   z:PAX5_709  
##           Correlation         VarAssay1         VarAssay2            TStat  
##             <numeric>         <numeric>         <numeric>        <numeric>  
## 1   0.592000045415225 0.812340554330436 0.875287356321839 7.27165360687288  
## 2   0.762361661991419 0.993427595451204 0.989655172413793 11.6618731859923  
## 3   0.658467426103544 0.855580057940935 0.879310344827586 8.66119456905554  
## 4   0.649844641802372 0.819518311929779 0.991954022988506 8.46390278863183  
## 5   0.502090536187102 0.886798979547715 0.935632183908046 5.74740534991767  
## ...               ...               ...               ...              ...  
## 12  0.609993999834133 0.867557400441043 0.968965517241379 7.62062915462215  
## 13  0.549172289236984 0.945561464954382 0.986781609195402  6.5052874492361   
## 14  0.643963832332729 0.828685086695205 0.970114942528736 8.33261518735298  
## 15  0.536406221175027   0.8282526916591 0.952873563218391  6.2919464214338  
## 16  0.803092200275283 0.978942361741687 0.962068965517241 13.3425248033796  
##                     Pval                  FDR  
##                <numeric>            <numeric>  
## 1   8.76476821331047e-11  4.6198810517901e-09  
## 2   3.15171893793416e-20 1.02998174891688e-17  
## 3   9.66043618744996e-14 1.43501388457211e-11  
## 4   2.57124018372512e-13 3.50117205017237e-11  
## 5   1.02530728526766e-07 2.83957983750399e-06  
## ...                  ...                  ...  
## 12  1.62079037573397e-11 1.10348811414554e-09  
## 13  3.29134663353072e-09 1.25071172074167e-07  
## 14  4.92446003514203e-13 6.18966745955545e-11  
## 15  8.81247545774464e-09 2.93869079550097e-07  
## 16  9.15917087670499e-24 3.74152130313399e-21  

```{r eval=FALSE}
trajGSM2 <- trajGSM[corGSM_MM[[1]]$name1, ]
trajMM2 <- trajMM[corGSM_MM[[1]]$name2, ]

trajCombined <- trajGSM2
assay(trajCombined) <- t(apply(assay(trajGSM2), 1, scale)) + t(apply(assay(trajMM2), 1, scale))

combinedMat <- plotTrajectoryHeatmap(trajCombined, returnMat = TRUE, varCutOff = 0)
```
> \## ArchR logging to : ArchRLogs/ArchR-plotTrajectoryHeatmap-15e136c64f4ee-Date-2020-04-15_Time-15-08-47.log  
## If there is an issue, please report to github with logFile!  
## Preparing Main Heatmap..  
## ArchR logging successful to : ArchRLogs/ArchR-plotTrajectoryHeatmap-15e136c64f4ee-Date-2020-04-15_Time-15-08-47.log  

```{r eval=FALSE}
rowOrder <- match(rownames(combinedMat), rownames(trajGSM2))

ht1 <- plotTrajectoryHeatmap(trajGSM2,  pal = paletteContinuous(set = "horizonExtra"),  varCutOff = 0, rowOrder = rowOrder)
```
> \## ArchR logging to : ArchRLogs/ArchR-plotTrajectoryHeatmap-15e1344a2d751-Date-2020-04-15_Time-15-08-47.log  
## If there is an issue, please report to github with logFile!  
## Preparing Main Heatmap..  
## ArchR logging successful to : ArchRLogs/ArchR-plotTrajectoryHeatmap-15e1344a2d751-Date-2020-04-15_Time-15-08-47.log  

```{r eval=FALSE}
ht2 <- plotTrajectoryHeatmap(trajMM2,  pal = paletteContinuous(set = "solarExtra"), varCutOff = 0, rowOrder = rowOrder)
```
> \## ArchR logging to : ArchRLogs/ArchR-plotTrajectoryHeatmap-15e1365304b53-Date-2020-04-15_Time-15-08-48.log  
## If there is an issue, please report to github with logFile!  
## useSeqnames is NULL or greater than 1 with a Sparse.Assays.Matrix trajectory input.  
## 2020-04-15 15:08:49 :  
## force=FALSE thus continuing with subsetting useSeqnames = z  
## 2020-04-15 15:08:49 :  
## Preparing Main Heatmap..  
## ArchR logging successful to : ArchRLogs/ArchR-plotTrajectoryHeatmap-15e1365304b53-Date-2020-04-15_Time-15-08-48.log  

```{r eval=FALSE}
ht1 + ht2
```

```{r, include=FALSE, eval=FALSE}
plotPDF(ht1 + ht2, name = "Plot-LymphoidU-GS-TF.pdf", ArchRProj = projHeme5, addDOC = FALSE, width = 8, height = 6)
ArchR:::.convertToPNG(ArchRProj = projHeme2)
system("cp Figures/*.png images/HemeWalkthrough/PNG/")
system("cp Figures/*.pdf images/HemeWalkthrough/PDF/")
```

![](images/HemeWalkthrough/PNG/Plot-LymphoidU-GS-TF_1.png){width=800 height=600}

```{r eval=FALSE}
corGIM_MM <- correlateTrajectories(trajGIM, trajMM)
```
> \## ArchR logging to : ArchRLogs/ArchR-correlateTrajectories-15e134c10b516-Date-2020-04-15_Time-15-09-05.log  
## If there is an issue, please report to github with logFile!  
## Found 37 Correlated Pairings!  
## 2020-04-15 15:09:20 :  

```{r eval=FALSE}
corGIM_MM[[1]]$matchname1
```
> \##  [1] "NFIA"    "NFKB2"   "IRF7"    "ETS1"    "NFE2"    "STAT2"   "FOXO1"  
##  [8] "IRF9"    "FOS"     "MEF2A"   "IRF8"    "MAFG"    "TCF4"    "TCF3"   
## [15] "NFIC"    "KLF2"    "CEBPA"   "POU2F2"  "FOSB"    "FOSL2"   "REL"    
## [22] "NFE2L2"  "CREB1"   "RUNX1"   "ATF4"    "SMARCC1" "FOXP1"   "GATA2"  
## [29] "HLTF"    "REST"    "LEF1"    "IRF2"    "MEF2C"   "IRF1"    "EBF1"   
## [36] "TFEB"    "PAX5"

```{r eval=FALSE}
corGIM_MM[[1]]
```
> \## DataFrame with 37 rows and 12 columns  
##          idx1      idx2 matchname1 matchname2       name1       name2  
##     <integer> <integer>    <array>    <array> <character> <character>  
## 1         680      1612       NFIA       NFIA   chr1:NFIA  z:NFIA_742  
## 2        2428      1584      NFKB2      NFKB2 chr10:NFKB2 z:NFKB2_714  
## 3        2625      1505       IRF7       IRF7  chr11:IRF7  z:IRF7_635  
## 4        3790      1202       ETS1       ETS1  chr11:ETS1  z:ETS1_332  
## 5        4287       989       NFE2       NFE2  chr12:NFE2  z:NFE2_119  
## ...       ...       ...        ...        ...         ...         ...  
## 33      14165      1510      MEF2C      MEF2C  chr5:MEF2C z:MEF2C_640  
## 34      14287      1499       IRF1       IRF1   chr5:IRF1  z:IRF1_629  
## 35      14558       937       EBF1       EBF1   chr5:EBF1   z:EBF1_67  
## 36      15164       902       TFEB       TFEB   chr6:TFEB   z:TFEB_32  
## 37      17354      1579       PAX5       PAX5   chr9:PAX5  z:PAX5_709  
##           Correlation         VarAssay1         VarAssay2            TStat  
##             <numeric>         <numeric>         <numeric>        <numeric>  
## 1   0.776554032919386 0.895435729261868 0.955747126436782 12.2013882485613  
## 2   0.842642341279197 0.893070265039514  0.88735632183908 15.4914421620131  
## 3   0.829368082571345 0.919144132035912  0.91551724137931 14.6953411117392  
## 4    0.85457028998599 0.947529702704156 0.901724137931034 16.2896821299438  
## 5   0.710070464598055 0.928874791677867 0.989655172413793 9.98300419241652  
## ...               ...               ...               ...              ...  
## 33   0.65886205886934 0.990860706413634 0.904597701149425 8.67036514488553  
## 34  0.852573236159675 0.977151766034084 0.981609195402299 16.1498641717228  
## 35  0.947958169110169 0.986344820170959 0.992528735632184 29.4737987999312  
## 36   0.71455800087286 0.934842212784259 0.842528735632184 10.1114979100535  
## 37  0.835610609654717 0.968066232998226 0.962068965517241 15.0587807907066  
##                     Pval                  FDR  
##                <numeric>            <numeric>  
## 1   2.24403341264388e-21 3.40224420626652e-20  
## 2   4.28486739529724e-28 1.38888805226876e-26  
## 3   1.61521163596779e-26 4.05409887519786e-25  
## 4   1.21844806414286e-29 5.72670590147145e-28  
## 5   1.31753444224449e-16 1.51034436062174e-15  
## ...                  ...                  ...  
## 33  9.23012667696911e-14 7.74671346102765e-13  
## 34  2.25997962853195e-29 9.23643848182623e-28  
## 35  1.65431752552523e-50 3.88764618498429e-48  
## 36  6.93337834458196e-17 8.04614277025561e-16  
## 37  3.05084830514743e-27 8.43469825540759e-26  

```{r eval=FALSE}
trajGIM2 <- trajGIM[corGIM_MM[[1]]$name1, ]
trajMM2 <- trajMM[corGIM_MM[[1]]$name2, ]

trajCombined <- trajGIM2
assay(trajCombined) <- t(apply(assay(trajGIM2), 1, scale)) + t(apply(assay(trajMM2), 1, scale))

combinedMat <- plotTrajectoryHeatmap(trajCombined, returnMat = TRUE, varCutOff = 0)
```
> \## ArchR logging to : ArchRLogs/ArchR-plotTrajectoryHeatmap-15e131cd6b30f-Date-2020-04-15_Time-15-09-21.log  
## If there is an issue, please report to github with logFile!  
## Preparing Main Heatmap..  
## ArchR logging successful to : ArchRLogs/ArchR-plotTrajectoryHeatmap-15e131cd6b30f-Date-2020-04-15_Time-15-09-21.log  

```{r eval=FALSE}
rowOrder <- match(rownames(combinedMat), rownames(trajGIM2))

ht1 <- plotTrajectoryHeatmap(trajGIM2,  pal = paletteContinuous(set = "blueYellow"),  varCutOff = 0, rowOrder = rowOrder)
```
> \## ArchR logging to : ArchRLogs/ArchR-plotTrajectoryHeatmap-15e137c439868-Date-2020-04-15_Time-15-09-21.log  
## If there is an issue, please report to github with logFile!  
## Preparing Main Heatmap..  
## ArchR logging successful to : ArchRLogs/ArchR-plotTrajectoryHeatmap-15e137c439868-Date-2020-04-15_Time-15-09-21.log  

```{r eval=FALSE}
ht2 <- plotTrajectoryHeatmap(trajMM2, pal = paletteContinuous(set = "solarExtra"), varCutOff = 0, rowOrder = rowOrder)
```
> \## ArchR logging to : ArchRLogs/ArchR-plotTrajectoryHeatmap-15e13755647e8-Date-2020-04-15_Time-15-09-22.log  
## If there is an issue, please report to github with logFile!  
## useSeqnames is NULL or greater than 1 with a Sparse.Assays.Matrix trajectory input.  
## 2020-04-15 15:09:23 :  
## force=FALSE thus continuing with subsetting useSeqnames = z  
## 2020-04-15 15:09:23 :  
## Preparing Main Heatmap..  
## ArchR logging successful to : ArchRLogs/ArchR-plotTrajectoryHeatmap-15e13755647e8-Date-2020-04-15_Time-15-09-22.log  

```{r eval=FALSE}
ht1 + ht2
```

```{r, include=FALSE, eval=FALSE}
plotPDF(ht1 + ht2, name = "Plot-LymphoidU-GEx-TF.pdf", ArchRProj = projHeme5, addDOC = FALSE, width = 8, height = 6)
ArchR:::.convertToPNG(ArchRProj = projHeme2)
system("cp Figures/*.png images/HemeWalkthrough/PNG/")
system("cp Figures/*.pdf images/HemeWalkthrough/PDF/")
```

![](images/HemeWalkthrough/PNG/Plot-LymphoidU-GEx-TF_1.png){width=800 height=600}

```{r, include=FALSE, eval=FALSE}
save.image(params$out, compress = FALSE)
```

