---
output:
  html_document:
    theme: yeti  # many options for theme, this one is my favorite.
params:
  threads: 20
  rdata: ""
  token: ""
  ref: ""
  out: ""
---

```{r, include=FALSE, eval=FALSE}
params2 <- params
unlockBinding("params", env = .GlobalEnv)
#load("Chapter-13-Footprints.Rdata")
load(params$rdata)
params <- params2
rm(params2)
devtools::install_github("GreenleafLab/ArchR", 
  auth_token = params$token, 
  ref = params$ref,
  repos = BiocManager::repositories(),
  dependencies = FALSE
)
library(ArchR)
fn <- unclass(lsf.str(envir = asNamespace("ArchR"), all = TRUE))
fn <- fn[!grepl("\\.", fn)]
fn <- fn[fn!="ArchRProj"]
for (i in seq_along(fn)){
    tryCatch({
        eval(parse(text = paste0(fn[i], "<-ArchR::", fn[i])))
    }, error = function(x) {
    })
}
addArchRThreads(threads = params$threads)
addArchRGenome("hg19")
# fn <- unclass(lsf.str(envir = asNamespace("ArchR"), all = TRUE))
# fn <- fn[fn!="ArchRProj"]
# for (i in seq_along(fn)) {
#     tryCatch({
#         eval(parse(text = paste0(fn[i], "<-ArchR:::", fn[i])))
#     }, error = function(x) {
#     })
# }
set.seed(1)
```

# Gene Scores and Marker Genes with ArchR

While ArchR is able to robustly call clusters, it is not possible to know _a priori_ which cell type is represented by each cluster. This task is often left to manual annotation because every application is different.

To do this cell type annotation, we use prior knowledge of cell type-specific marker genes and we estimate gene expression for these genes from our chromatin accessibility data by uing _gene scores_. A gene score is essentially a prediction of how highly expressed a gene will be based on the accessibility of regulatory elements in the vicinity of the gene. To create these gene scores, ArchR allows for the use of complex user-supplied custom distance-weighted accessibility models.

## Calculating Gene Scores in ArchR

In our publication, we tested over 50 different gene score models and identified a class of models that consistently outperformed the rest across a variety of test conditions. This class of model, which is implemented as the default in ArchR, has three major components:

1. Accessibility within the entire gene body contributes to the gene score.
2. An exponential weighting function that accounts for the activity of putative distal regulatory elements in a distance-dependent fashion.
3. Imposed gene boundaries that minimizes the contribution of unrelated regulatory elements to the gene score.

So how does ArchR calculate gene scores? For each chromosome, ArchR creates a tile matrix using a user-defined tile size that is not pre-computed (default is 500 bp), overlaps these tiles with the user-defined gene window (default is 100 kb on either side of the gene), and then computes the distance from each tile (start or end) to the gene body (with optional extensions upstream or downstream) or gene start. We have found that the best predictor of gene expression is the local accessibility of the gene region which includes the promoter and gene body. As mentioned above, to properly account for distal accessibility for a given gene, ArchR identifies the subset of tiles that are within the gene window and do not cross another gene region. This filtering allows for inclusion of distal regulatory elements that could improve the accuracy of predicting gene expression values but excludes regulatory elements more likely to be associated with another gene (for ex. the promoter of a nearby gene). The distance from each tile to the gene is then converted to a distance weight using a user-defined accessibility model (default is e(-abs(distance)/5000) + e-1). When the gene body is included in the gene region (where the distance-based weight is the maximum weight possible), we found that extremely large genes can bias the overall gene scores. In these cases, the total gene scores can vary substantially due to the inclusion of insertions in both introns and exons. To help adjust for these large differences in gene size, ArchR applies a separate weight for the inverse of the gene size (1 / gene size) and scales this inverse weight linearly from 1 to a user-defined hard maximum (default of 5). Smaller genes thus receive larger relative weights, partially normalizing this length effect. The corresponding distance and gene size weights are then multiplied by the number of Tn5 insertions within each tile and summed across all tiles within the gene window, while still accounting for nearby gene regions as described above. This summed accessibility is a “gene score” and is depth normalized across all genes to a user-defined constant (default of 10,000). Computed gene scores are then stored in the corresponding Arrow file for downstream analyses.

To illustrate what the default ArchR gene score model looks like, we provide this toy example showing the weighting applied across the entire gene region:

<center>
![](images/GeneActivityScore_Schematic.png){width=500px}
</center>

Gene scores are calculated for each Arrow file at the time of creation if the parameter `addGeneScoreMat` is set to `TRUE` - this is the default behavior. If you wish to change any of the default parameters for gene score calculation during Arrow file creation, you can use the `GeneScoreMatParams` parameter. By default, gene scores are depth normalized based on the `scaleTo` parameter. If you do not wish to add gene scores to your Arrow file at the time of creation, gene scores can be added to Arrow files at any time by using the `addGeneScoreMatrix()` function. Gene scores are stored in a matrix within each Arrow file called "GeneScoreMatrix". Once caluclated, gene scores can be used for many different analyses. For example, individual cells of an embedding can be colored by their gene scores to help with the identification of various cell types. We illustrate the application of gene scores throughout the rest of this chapter.

It is important to note that not all genes behave well with gene scores. In particular, genes that reside in very gene-dense areas can be problematic. Thus, it is always best to sanity check all gene score analyses by looking at sequencing tracks which is described in a later chapter.

## Identification of Marker Features

In addition to using prior knowledge of relevant marker genes for annotation of clusters, ArchR enables unbiased identification of marker features for any given cell groupings (for example, clusters). These features can be anything - peaks, genes (based on gene scores), or transcription factor motifs (based on chromVAR deviations). ArchR does this using the `getMarkerFeatures()` function which can take as input any matrix via the `useMatrix` parameter and it identifies features unique to the groups indicated by the `groupBy` parameter. If the `useMatrix` parameter is set to "GeneScoreMatrix", then the function will identify the genes that appear to be uniquely active in each cell type. This provides an unbiased way of seeing which genes are predicted to be active in each cluster and can aid in cluster annotation.

As mentioned above, the same `getMarkerFeatures()` function can be used with any matrix stored in the Arrow files to identify features that are specific to certain cell groups. This is accomplished via the `useMatrix` parameter. For example, `useMatrix = "TileMatrix"` would identify genomic regions that are highly specific to a certain cell group and `useMatrix = "PeakMatrix"` would identify peaks that are highly specific to a certain cell group. Examples of how to use the `getMarkerFeatures()` function on other feature types are provided in later chapters.

### How does marker feature identification happen?

This process of marker feature identification hinges on the selection of a group of bias-matched background cells for each cell group. Across all features, each cell group is compared to its own background group of cells to determine if the given cell group has significantly higher accessibility.

<center>
![](images/markerFeature_schematic.png){width=500px}
</center>

The selection of these background cell groups is critical to the success of this process and is performed across the multi-dimensional space provided by the user via the `bias` argument to `getMarkerFeatures()`. For each cell in the cell group, ArchR finds the nearest neighbor cell across the provided multi-dimensional space that is not a member of the given cell group and adds it to the background group of cells. In this way, ArchR creates a group of bias-matched cells that is as similar as possible to the given cell group, thus enabling a more robust determination of significance even if the group of cells is small.

The way ArchR does this is by taking all of the dimensions provided via the `bias` parameter and quantile normalizing their values to distribute the variance of each dimension across the same relative scale. Taking a toy example, if the parameters `TSS` and `log10(Num Fragments)` were provided to `bias`, the pre-quantile normalized values might look like this:

<center>
![](images/background_preNorm.png){width=500px}
</center>

Here, the relative variance across the y-axis is very small compared to the variance across the x-axis. If we normalize these axes so that their values range from 0 to 1 instead, we make the relative variance much more equal. Importantly, we also change the nearest neighbors dramatically as indicated to the right of this plot.

<center>
![](images/background_postNorm.png){width=500px}
</center>

ArchR normalizes all of the dimensions and uses euclidean distance in this normalized multidimensional space to find the nearest neighbors.

## Identifying Marker Genes

To identify marker genes based on gene scores, we call the `getMarkerFeatures()` function with `useMatrix = "GeneScoreMatrix"`. We specify that we want to know the cluster-specific features with `groupBy = "Clusters"` which tells ArchR to use the "Clusters" column in `cellColData` to stratify cell groups.
```{r eval=FALSE}
markersGS <- getMarkerFeatures(
    ArchRProj = projHeme2, 
    useMatrix = "GeneScoreMatrix", 
    groupBy = "Clusters",
    bias = c("TSSEnrichment", "log10(nFrags)"),
    testMethod = "wilcoxon"
)
```
> \## ArchR logging to : ArchRLogs/ArchR-getMarkerFeatures-f4e4232fa80b-Date-2020-04-15_Time-10-04-48.log  
## If there is an issue, please report to github with logFile!  
## MatrixClass = Sparse.Double.Matrix  
## 2020-04-15 10:04:49 : Matching Known Biases, 0.009 mins elapsed. 
##   
## ###########  
## 2020-04-15 10:05:23 : Completed Pairwise Tests, 0.585 mins elapsed.  
## ###########  
## ArchR logging successful to : ArchRLogs/ArchR-getMarkerFeatures-f4e4232fa80b-Date-2020-04-15_Time-10-04-48.log  

This function returns a `SummarizedExperiment` object containing relevant information on the marker features identified. This type of return value is common in ArchR and is one of the key ways that ArchR enables downstream data analysis. `SummarizedExperiment` objects are similar to a matrix where rows represent features of interest (i.e. genes) and columns represent samples. A `SummarizedExperiment` object contains one or more assays, each represented by a matrix-like object of numeric data, and metadata that applies to the rows or columns of the assays matrices. It is beyond the scope of this tutorial to dive into `SummarizedExperiment` objects but check out the [bioconductor page](https://bioconductor.org/packages/release/bioc/html/SummarizedExperiment.html) if you need more information.

We can get a list of `DataFrame` objects, one for each of our clusters, containing the relevant marker features using the `getMarkers()` function:
```{r eval=FALSE}
markerList <- getMarkers(markersGS, cutOff = "FDR <= 0.01 & Log2FC >= 1.25")
markerList$C6
```
> \## DataFrame with 291 rows and 9 columns  
##       seqnames     start       end  strand         name     idx  
##          <Rle>   <array>   <array> <array>      <array> <array>  
## 8362     chr16  49891830  49524515       2       ZNF423     539  
## 7025     chr14 105714879 105717430       1        BTBD6     764  
## 8726     chr16  89043504  88941263       2      CBFA2T3     903  
## 7024     chr14 105781914 105675623       2         BRF1     763  
## 17158     chr5    997455    988425       2 LOC100506688      18  
## ...        ...       ...       ...     ...          ...     ...  
## 19084     chr6 119812467 119773712       2    LOC285762     925  
## 18913     chr6  83072923  83077133       1         TPBG     754  
## 12814     chr2 124782864 125672863       1      CNTNAP5     755  
## 20390     chr7 150147962 150176483       1       GIMAP8    1028  
## 7384     chr15  52569397  52569314       2      MIR1266     347  
##                 Log2FC                  FDR          MeanDiff  
##              <numeric>            <numeric>         <numeric>  
## 8362  1.94888141545084 2.95473053609015e-21  2.08129078745695  
## 7025  1.88748443360838 3.25360925877231e-15  2.29925343722754  
## 8726   1.5858058377813 3.25360925877231e-15  2.11727644857422  
## 7024  1.41061475820385 2.00400144783449e-14   1.9923490244715  
## 17158 3.16686924437513 2.00400144783449e-14 0.693424444489265  
## ...                ...                  ...               ...  
## 19084 1.36625245611966  0.00853516274073803 0.211727000065234  
## 18913 2.30499450627443  0.00854168013784213 0.544454051635953  
## 12814 2.28305835231306  0.00893526022954534  0.82240450922043  
## 20390 1.52933035478039  0.00910586862792073 0.516763460205267  
## 7384  2.16292224987129  0.00986664274558125  0.22839476831448  

To visualize all of the marker features simultaneously, we can create a heatmap using the `markerHeatmap()` function, optionally supplying some marker genes to label on the heatmap via the `labelMarkers` parameter.
```{r eval=FALSE}
markerGenes  <- c(
    "CD34", #Early Progenitor
    "GATA1", #Erythroid
    "PAX5", "MS4A1", "EBF1", "MME", #B-Cell Trajectory
    "CD14", "CEBPB", "MPO", #Monocytes
    "IRF8", 
    "CD3D", "CD8A", "TBX21", "IL7R" #TCells
  )

heatmapGS <- markerHeatmap(
  seMarker = markersGS, 
  cutOff = "FDR <= 0.01 & Log2FC >= 1.25", 
  labelMarkers = markerGenes,
  transpose = TRUE
)
```
> \## Printing Top Marker Genes:  
## C1:  
##  PADI4, PADI6, CDA, LINC01226, TMEM61, RPE65, PIN1P1, GNAT2, HSD3BP4, SRGAP2D, SNX27, THEM5, S100A8, FCAMR, UCN3, MIR604, MIR938, MIR4679-2, OR56B4, OR52B2  
## C2:  
##  HES4, FHAD1, PLA2G2C, CELA3A, RHD, FGR, MYCL, BTBD19, TCTEX1D4, SLC5A9, L1TD1, PALMD, FRRS1, GPR88, MYBPHL, CASQ2, NHLH2, SLC22A15, CTSS, S100A11  
## C3:  
##  PLEKHM2, MIR4422, CLCA3P, LOC729970, ALG14, TMEM56, SLC16A4, FCRL3, AIM2, SLAMF6, C1orf220, MIR4424, RALGPS2, RHEX, FCMR, BPNT1, KMO, LINC00700, ARID5B, MIR548AV  
## C4:  
##  LINC01342, LINC01346, SERINC2, RIMS3, SLFNL1-AS1, EDN2, ROR1, C1orf141, CACNA1S, KLHDC8A, SLC16A9, MIR346, GLYATL2, FAM111B, OR5AN1, RAB30-AS1, BTG4, TEAD4, CD9, PIANP  
## C5:  
##  LRRC38, FAM131C, PLA2G2D, LOC653160, TSPAN1, TTC39A, GBP1P1, LOC100129046, HAO2, RORC, S100A5, S100A4, ILDR2, FMO1, ANGPTL1, LAD1, TNNI1, VASH2, GJC2, KCNK1  
## C6:  
##  TTLL10, ATP13A2, PAX7, CSMD2, MIR3659, TRABD2B, ERICH3, ABCA4, NTNG1, LOR, FAM189B, UCK2, PHLDA3, LEMD1, SERTAD4-AS1, ESRRG, TRIM67, LOC149373, C1orf131, DEPP1  
## C7:  
##  TNFRSF4, RUNX3, LEXM, IL12RB2, VCAM1, CHIA, BCL2L15, CD160, PDZK1, SH2D2A, NTRK1, PYHIN1, SLAMF7, FCGR3A, CD247, LOC100505918, FASLG, GAS5-AS1, ATP6V1G3, GREM2  
## C8:  
##  RCAN3, HPCAL4, ANKRD13C, HHLA3, ZRANB2, PLPPR5, PRMT6, AMIGO1, CHI3L2, CD2, PTGFRN, ACP6, NIT1, NUF2, MIR557, IL19, MIR1265, ANKRD26, ZEB1, ZEB1-AS1  
## C9:  
##  ESPN, MIR4252, RCAN3AS, HOOK1, ZNF37BP, ZNF239, SLC10A1, TMEM202, RRN3P1, TP53TG3D, MT1F, ZNF286B, FOXO3B, TRIM16L, PEX12, LOC729966, PDCD2L, ZNF283, ZNF610, TLX2  
## C10:  
##  MIR4689, HES2, NPPA-AS1, MFN2, FBLIM1, KLHDC7A, RPS14P3, FAM43B, MIR378F, GRHL3, TRIM63, GJB3, GJA4, EPHA10, RHBDL2, PPIE, RIMKLA, FAM183A, LINC00853, PDZK1IP1  
## C11:  
##  NPPA, SFTA1P, WT1-AS, MIR4488, LRRC10B, CACNA1C-AS4, LOC728739, CCNA1, LPO, MPO, TTR, PRTN3, ELANE, OR10H2, NCAN, OXER1, INSM1, BPIFA1, EMILIN3, COL9A3  
## C12:  
##  LINC02593, VWA1, CHD5, TMEM51-AS1, MFAP2, PADI3, HSPG2, C1QB, MIR4253, KIAA1522, C1orf94, LINC01343, KNCN, DMRTB1, CDCP2, ACOT11, ASB17, CYR61, SLC6A17, LOC643441  
## Identified 2662 markers!  
## Adding Annotations..  
## Preparing Main Heatmap..  

To plot this heatmap, we can use the `ComplexHeatmap::draw()` function because the `heatmapGS` object is actually a list of heatmaps:
```{r eval=FALSE}
ComplexHeatmap::draw(heatmapGS, heatmap_legend_side = "bot", annotation_legend_side = "bot")
```

```{r, include=FALSE, eval=FALSE}
plotPDF(heatmapGS, name = "GeneScores-Marker-Heatmap", width = 8, height = 6, ArchRProj = projHeme2, addDOC = FALSE)
ArchR:::.convertToPNG(ArchRProj = projHeme2)
system("cp Figures/*.png images/HemeWalkthrough/PNG/")
system("cp Figures/*.pdf images/HemeWalkthrough/PDF/")
```

![](images/HemeWalkthrough/PNG/GeneScores-Marker-Heatmap_1.png){width=800 height=600}

To save an editable vectorized version of this plot, we use `plotPDF()`.

```{r eval=FALSE}
plotPDF(heatmapGS, name = "GeneScores-Marker-Heatmap", width = 8, height = 6, ArchRProj = projHeme2, addDOC = FALSE)
```
> \## [1] 0

[Download PDF : GeneScores-Marker-Heatmap.pdf](images/HemeWalkthrough/PDF/GeneScores-Marker-Heatmap.pdf)

## Visualizing Marker Genes on an Embedding

As mentioned previously, we can overlay per-cell gene scores on our UMAP embedding. This is accomplished using the `colorBy` and `name` parameters in the `plotEmbedding()` function.
```{r eval=FALSE}
markerGenes  <- c(
    "CD34",  #Early Progenitor
    "GATA1", #Erythroid
    "PAX5", "MS4A1", "MME", #B-Cell Trajectory
    "CD14", "MPO", #Monocytes
    "CD3D", "CD8A"#TCells
  )

p <- plotEmbedding(
    ArchRProj = projHeme2, 
    colorBy = "GeneScoreMatrix", 
    name = markerGenes, 
    embedding = "UMAP",
    quantCut = c(0.01, 0.95),
    imputeWeights = NULL
)
```
> \## ArchR logging to : ArchRLogs/ArchR-plotEmbedding-f4e44c020f3e-Date-2020-04-15_Time-10-05-43.log  
## If there is an issue, please report to github with logFile!  
## Getting UMAP Embedding  
## ColorBy = GeneScoreMatrix  
## Getting Matrix Values...  
## Getting Matrix Values...  
##   
## Plotting Embedding  
## 1 2 3 4 5 6 7 8 9   
## ArchR logging successful to : ArchRLogs/ArchR-plotEmbedding-f4e44c020f3e-Date-2020-04-15_Time-10-05-43.log  

To plot a specific gene, we can subset this plot list:
```{r, eval=FALSE}
p$CD14
```
![](images/HemeWalkthrough/PNG/Plot-UMAP-Marker-Genes-WO-Imputation_6.png){width=800 height=800}

To plot all genes we can use `cowplot` to arrange the various marker genes into a single plot.

```{r, eval=FALSE}
p2 <- lapply(p, function(x){
    x + guides(color = FALSE, fill = FALSE) + 
    theme_ArchR(baseSize = 6.5) +
    theme(plot.margin = unit(c(0, 0, 0, 0), "cm")) +
    theme(
        axis.text.x=element_blank(), 
        axis.ticks.x=element_blank(), 
        axis.text.y=element_blank(), 
        axis.ticks.y=element_blank()
    )
})
do.call(cowplot::plot_grid, c(list(ncol = 3),p2))
```

```{r, include=FALSE, eval=FALSE}
plotPDF(
    do.call(cowplot::plot_grid, c(list(ncol = 3),p2)), 
    name = "Plot-UMAP-Markers-WO-Imputation.pdf", 
    ArchRProj = projHeme2,
    addDOC = FALSE,
    width = 10, 
    height = 10
)
ArchR:::.convertToPNG(ArchRProj = projHeme2)
system("cp Figures/*.png images/HemeWalkthrough/PNG/")
system("cp Figures/*.pdf images/HemeWalkthrough/PDF/")
```

![](images/HemeWalkthrough/PNG/Plot-UMAP-Markers-WO-Imputation_1.png){width=800 height=800}

To save an editable vectorized version of this plot, we use the `plotPDF()` function.

```{r eval=FALSE}
plotPDF(plotList = p, 
    name = "Plot-UMAP-Marker-Genes-WO-Imputation.pdf", 
    ArchRProj = projHeme2, 
    addDOC = FALSE, width = 5, height = 5)
```
> \## [1] "plotting ggplot!"  
## [1] "plotting ggplot!"  
## [1] "plotting ggplot!"  
## [1] "plotting ggplot!"  
## [1] "plotting ggplot!"  
## [1] "plotting ggplot!"  
## [1] "plotting ggplot!"  
## [1] "plotting ggplot!"  
## [1] "plotting ggplot!"  
## [1] 0  

[Download PDF : Plot-UMAP-Marker-Genes-WO-Imputation.pdf](images/HemeWalkthrough/PDF/Plot-UMAP-Marker-Genes-WO-Imputation.pdf)

## Marker Genes Imputation with MAGIC

In the previous section, you may have noticed that some of the gene score plots appear quite variable. This is because of the sparsity of scATAC-seq data. We can use [MAGIC](https://github.com/KrishnaswamyLab/MAGIC) to impute gene scores by smoothing signal across nearby cells. In our hands, this greatly improves the visual interpretation of gene scores. To do this, we first add impute weights to our `ArchRProject`.

```{r eval=FALSE}
projHeme2 <- addImputeWeights(projHeme2)
```
> \## 2020-04-15 10:06:25 : Computing Impute Weights Using Magic (Cell 2018), 0 mins elapsed.  
## 2020-04-15 10:06:36 : Completed Getting Magic Weights!, 0.19 mins elapsed.  

These impute weights can then be passed to `plotEmbedding()` when plotting gene scores overlayed on the UMAP embedding.
```{r eval=FALSE}
markerGenes  <- c(
    "CD34",  #Early Progenitor
    "GATA1", #Erythroid
    "PAX5", "MS4A1", "MME", #B-Cell Trajectory
    "CD14", "MPO", #Monocytes
    "CD3D", "CD8A"#TCells
  )

p <- plotEmbedding(
    ArchRProj = projHeme2, 
    colorBy = "GeneScoreMatrix", 
    name = markerGenes, 
    embedding = "UMAP",
    imputeWeights = getImputeWeights(projHeme2)
)
```
> \## Getting ImputeWeights  
## ArchR logging to : ArchRLogs/ArchR-plotEmbedding-f4e41cd9b447-Date-2020-04-15_Time-10-06-36.log  
## If there is an issue, please report to github with logFile!  
## Getting UMAP Embedding  
## ColorBy = GeneScoreMatrix  
## Getting Matrix Values...  
## Getting Matrix Values...  
##   
## Imputing Matrix  
## Using weights on disk  
## Using weights on disk  
## Plotting Embedding  
## 1 2 3 4 5 6 7 8 9   
## ArchR logging successful to : ArchRLogs/ArchR-plotEmbedding-f4e41cd9b447-Date-2020-04-15_Time-10-06-36.log  

As before, we can subset this plot list to select a specific gene.
```{r, eval=FALSE}
p$CD14
```

![](images/HemeWalkthrough/PNG/Plot-UMAP-Marker-Genes-W-Imputation_6.png){width=800 height=800}

Or we can plot all the marker genes at once using `cowplot`.

```{r, eval=FALSE}
#Rearrange for grid plotting
p2 <- lapply(p, function(x){
    x + guides(color = FALSE, fill = FALSE) + 
    theme_ArchR(baseSize = 6.5) +
    theme(plot.margin = unit(c(0, 0, 0, 0), "cm")) +
    theme(
        axis.text.x=element_blank(), 
        axis.ticks.x=element_blank(), 
        axis.text.y=element_blank(), 
        axis.ticks.y=element_blank()
    )
})
do.call(cowplot::plot_grid, c(list(ncol = 3),p2))
```

```{r, include=FALSE, eval=FALSE}
plotPDF(
    do.call(cowplot::plot_grid, c(list(ncol = 3),p2)), 
    name = "Plot-UMAP-Markers-W-Imputation.pdf", 
    ArchRProj = projHeme2,
    addDOC = FALSE,
    width = 10, 
    height = 10
)
ArchR:::.convertToPNG(ArchRProj = projHeme2)
system("cp Figures/*.png images/HemeWalkthrough/PNG/")
system("cp Figures/*.pdf images/HemeWalkthrough/PDF/")
```

![](images/HemeWalkthrough/PNG/Plot-UMAP-Markers-W-Imputation_1.png){width=800 height=800}

To save an editable vectorized version of this plot, we use the `plotPDF()` function.

```{r eval=FALSE}
plotPDF(plotList = p, 
    name = "Plot-UMAP-Marker-Genes-W-Imputation.pdf", 
    ArchRProj = projHeme2, 
    addDOC = FALSE, width = 5, height = 5)
```
> \## [1] "plotting ggplot!"  
## [1] "plotting ggplot!"  
## [1] "plotting ggplot!"  
## [1] "plotting ggplot!"  
## [1] "plotting ggplot!"  
## [1] "plotting ggplot!"  
## [1] "plotting ggplot!"  
## [1] "plotting ggplot!"  
## [1] "plotting ggplot!"  
## [1] 0  

[Download PDF : Plot-UMAP-Marker-Genes-W-Imputation.pdf](images/HemeWalkthrough/PDF/Plot-UMAP-Marker-Genes-W-Imputation.pdf)

## Track Plotting with ArchRBrowser

In addition to plotting gene scores per cell as a UMAP overlay, we can browse the local chromatin accessibility at these marker genes on a per cluster basis with genome browser tracks. To do this, we use the `plotBrowserTrack()` function which will create a list of plots, one for each of the genes specified by `markerGenes`. This function will plot a single track for each group in the `groupBy` parameter.

```{r eval=FALSE}
markerGenes  <- c(
    "CD34", #Early Progenitor
    "GATA1", #Erythroid
    "PAX5", "MS4A1", #B-Cell Trajectory
    "CD14", #Monocytes
    "CD3D", "CD8A", "TBX21", "IL7R" #TCells
  )

p <- plotBrowserTrack(
    ArchRProj = projHeme2, 
    groupBy = "Clusters", 
    geneSymbol = markerGenes, 
    upstream = 50000,
    downstream = 50000
)
```
> \## ArchR logging to : ArchRLogs/ArchR-plotBrowserTrack-f4e46f9da45e-Date-2020-04-15_Time-10-07-24.log  
## If there is an issue, please report to github with logFile!  
## 2020-04-15 10:07:24 : Validating Region, 0.006 mins elapsed.  
## GRanges object with 9 ranges and 2 metadata columns:   
##       seqnames              ranges strand |     gene_id      symbol  
##          <Rle>           <IRanges>  <Rle> | <character> <character>  
##   [1]     chr1 208059883-208084683      - |         947        CD34  
##   [2]     chrX   48644982-48652717      + |        2623       GATA1  
##   [3]     chr9   36838531-37034476      - |        5079        PAX5  
##   [4]    chr11   60223282-60238225      + |         931       MS4A1  
##   [5]     chr5 140011313-140013286      - |         929        CD14  
##   [6]    chr11 118209789-118213459      - |         915        CD3D  
##   [7]     chr2   87011728-87035519      - |         925        CD8A  
##   [8]    chr17   45810610-45823485      + |       30009       TBX21  
##   [9]     chr5   35856977-35879705      + |        3575        IL7R  
##   -------  
##   seqinfo: 24 sequences from hg19 genome  
## 2020-04-15 10:07:25 : Adding Bulk Tracks (1 of 9), 0.009 mins elapsed.  
## 2020-04-15 10:07:27 : Adding Gene Tracks (1 of 9), 0.044 mins elapsed.  
## 2020-04-15 10:07:27 : Plotting, 0.054 mins elapsed.  
## 2020-04-15 10:07:29 : Adding Bulk Tracks (2 of 9), 0.09 mins elapsed.  
## 2020-04-15 10:07:31 : Adding Gene Tracks (2 of 9), 0.108 mins elapsed.  
## 2020-04-15 10:07:31 : Plotting, 0.117 mins elapsed.  
## 2020-04-15 10:07:34 : Adding Bulk Tracks (3 of 9), 0.157 mins elapsed.  
## 2020-04-15 10:07:35 : Adding Gene Tracks (3 of 9), 0.175 mins elapsed.  
## 2020-04-15 10:07:35 : Plotting, 0.183 mins elapsed.  
## 2020-04-15 10:07:37 : Adding Bulk Tracks (4 of 9), 0.221 mins elapsed.  
## 2020-04-15 10:07:38 : Adding Gene Tracks (4 of 9), 0.239 mins elapsed.  
## 2020-04-15 10:07:39 : Plotting, 0.249 mins elapsed.  
## 2020-04-15 10:07:41 : Adding Bulk Tracks (5 of 9), 0.279 mins elapsed.  
## 2020-04-15 10:07:42 : Adding Gene Tracks (5 of 9), 0.295 mins elapsed.  
## 2020-04-15 10:07:42 : Plotting, 0.301 mins elapsed.  
## 2020-04-15 10:07:44 : Adding Bulk Tracks (6 of 9), 0.339 mins elapsed.  
## 2020-04-15 10:07:46 : Adding Gene Tracks (6 of 9), 0.358 mins elapsed.  
## 2020-04-15 10:07:46 : Plotting, 0.367 mins elapsed.  
## 2020-04-15 10:07:49 : Adding Bulk Tracks (7 of 9), 0.409 mins elapsed.  
## 2020-04-15 10:07:50 : Adding Gene Tracks (7 of 9), 0.429 mins elapsed.  
## 2020-04-15 10:07:50 : Plotting, 0.44 mins elapsed.  
## 2020-04-15 10:07:54 : Adding Bulk Tracks (8 of 9), 0.502 mins elapsed.  
## 2020-04-15 10:07:55 : Adding Gene Tracks (8 of 9), 0.52 mins elapsed.  
## 2020-04-15 10:07:56 : Plotting, 0.528 mins elapsed.  
## 2020-04-15 10:07:58 : Adding Bulk Tracks (9 of 9), 0.567 mins elapsed.  
## 2020-04-15 10:07:59 : Adding Gene Tracks (9 of 9), 0.585 mins elapsed.  
## 2020-04-15 10:08:00 : Plotting, 0.593 mins elapsed.  
## ArchR logging successful to : ArchRLogs/ArchR-plotBrowserTrack-f4e46f9da45e-Date-2020-04-15_Time-10-07-24.log  

To plot a track of a specific gene, we can select one from the list.

```{r eval=FALSE}
grid::grid.newpage()
grid::grid.draw(p$CD14)
```

```{r, include=FALSE, eval=FALSE}
plotPDF(plotList = p, 
    name = "Plot-Tracks-Marker-Genes.pdf", 
    ArchRProj = projHeme2, 
    addDOC = FALSE, 
    width = 5, 
    height = 5
)
ArchR:::.convertToPNG(ArchRProj = projHeme2)
system("cp Figures/*.png images/HemeWalkthrough/PNG/")
system("cp Figures/*.pdf images/HemeWalkthrough/PDF/")
```

![](images/HemeWalkthrough/PNG/Plot-Tracks-Marker-Genes_5.png){width=500 height=500}

We can save a multi-page PDF with a single page for each gene locus in our plot list using the `plotPDF()` function.

```{r eval=FALSE}
plotPDF(plotList = p, 
    name = "Plot-Tracks-Marker-Genes.pdf", 
    ArchRProj = projHeme2, 
    addDOC = FALSE, width = 5, height = 5)
```
> \## NULL  
## NULL  
## NULL  
## NULL  
## NULL  
## NULL  
## NULL  
## NULL  
## NULL  
## [1] 0  

[Download PDF : Plot-Tracks-Marker-Genes.pdf](images/HemeWalkthrough/PDF/Plot-Tracks-Marker-Genes.pdf)

## Launching the ArchRBrowser

One challenge inherent to scATAC-seq data analysis is genome-track level visualizations of chromatin accessibility observed within groups of. Traditionally, track visualization requires grouping the scATAC-seq fragments, creating a genome coverage bigwig, and normalizing this track for quantitative visualization. Typically, end-users use a genome browser such as the [WashU Epigenome Browser](https://epigenomegateway.wustl.edu/), the [UCSC Genome Browser](http://genome.ucsc.edu/), or the [IGV browser](http://software.broadinstitute.org/software/igv/) to visualize these sequencing tracks. This process involves using multiple software and any change to the cellular groups or addition of more samples requires re-generation of bigwig files etc., which can become time consuming.

For this reason, ArchR has a Shiny-based interactive genome browser that can be launched with a single line of code `ArchRBrowser(ArchRProj)`. The data storage strategy implemented in Arrow files allows this interactive browser to dynamically change the cell groupings, resolution, and normalization, enabling real-time track-level visualizations. The ArchR Genome Browser also creates high-quality vectorized images in PDF format for publication or distribution. Additionally, the browser accepts user-supplied input files such as a `GenomicRanges` object to display features, via the `features` parameter, or genomic interaction files that define co-accessibility, peak-to-gene linkages, or loops from chromatin conformation data via the `loops` parameter. For `loops` the expected format is a `GRanges` object whose start position represents the center position of one loop anchor and whose end position represents the center position of the other loop anchor. 

To launch our local interactive genome browser, we use the `ArchRBrowser()` function.
```{r eval=FALSE}
#ArchRBrowser(projHeme2)
```
When we start, we see a screen that looks like this:
![](images/HemeWalkthrough/PNG/ArchR_Browser_1.png){width=800 height=500}

By selecting a gene via the "Gene Symbol" box, we can begin browsing. You may need to click the "Plot Track" button to force your browser session to update.
![](images/HemeWalkthrough/PNG/ArchR_Browser_2.png){width=800 height=500}

Once we have plotted a gene locus, we see a single track representing each of the clusters in our data.
![](images/HemeWalkthrough/PNG/ArchR_Browser_3.png){width=800 height=500}

If we click the "Additional Parameters" tab, we can specify which clusters to display and which to hide.
![](images/HemeWalkthrough/PNG/ArchR_Browser_4.png){width=800 height=500}

By unclicking the check boes next to Clusters 1, 2, and 3, we will remove them from the plot.
![](images/HemeWalkthrough/PNG/ArchR_Browser_5.png){width=800 height=500}

When we return to the "Plot" tab, we should see an updated plot with Clusters 1, 2, and 3 removed. Again, you may need to click the "Plot Track" button to force your browser session to update.
![](images/HemeWalkthrough/PNG/ArchR_Browser_6.png){width=800 height=500}

At any point, we can export a vectorized PDF of the current track view using the "Donwload the Track" button.

```{r, include=FALSE, eval=FALSE}
save.image(params$out, compress = FALSE)
```















