---
output:
  html_document:
params:
  threads: 20
  rdata: ""
  token: ""
  ref: ""
  out: ""
---

```{r, include=FALSE, eval=FALSE}
params2 <- params
unlockBinding("params", env = .GlobalEnv)
#load("Chapter-13-Footprints.Rdata")
load(params$rdata)
params <- params2
rm(params2)
devtools::install_github("GreenleafLab/ArchR", 
  auth_token = params$token, 
  ref = params$ref,
  repos = BiocManager::repositories(),
  dependencies = FALSE
)
library(ArchR)
fn <- unclass(lsf.str(envir = asNamespace("ArchR"), all = TRUE))
fn <- fn[!grepl("\\.", fn)]
fn <- fn[fn!="ArchRProj"]
for (i in seq_along(fn)){
    tryCatch({
        eval(parse(text = paste0(fn[i], "<-ArchR::", fn[i])))
    }, error = function(x) {
    })
}
addArchRThreads(threads = params$threads)
addArchRGenome("hg19")
# fn <- unclass(lsf.str(envir = asNamespace("ArchR"), all = TRUE))
# fn <- fn[fn!="ArchRProj"]
# for (i in seq_along(fn)) {
#     tryCatch({
#         eval(parse(text = paste0(fn[i], "<-ArchR:::", fn[i])))
#     }, error = function(x) {
#     })
# }
set.seed(1)
```

# Calling Peaks with ArchR


## Calling Peaks w/ Macs2

We can now save our original projHeme3 using `saveArchRProject` from ArchR.

```{r eval=FALSE}
saveArchRProject(ArchRProj = projHeme3, outputDirectory = "Save-ProjHeme3", load = FALSE)
```
> \## Copying ArchRProject to new outputDirectory : /oak/stanford/groups/howchang/users/jgranja/ArchRTutorial/ArchRBook/BookOutput4/Save-ProjHeme3  
## Copying Arrow Files...  
## Copying Arrow Files (1 of 3)  
## Copying Arrow Files (2 of 3)  
## Copying Arrow Files (3 of 3)  
## Getting ImputeWeights  
## Dropping ImputeWeights...  
## Copying Other Files...  
## Copying Other Files (1 of 4): Embeddings  
## Copying Other Files (2 of 4): IterativeLSI  
## Copying Other Files (3 of 4): IterativeLSI2  
## Copying Other Files (4 of 4): Plots  
## Saving ArchRProject...  

```{r eval=FALSE}
#Create Group Coverage Files that can be used for downstream analysis (~5-10 minutes)
projHeme4 <- addGroupCoverages(ArchRProj = projHeme3, groupBy = "Clusters2")
```
> \## ArchR logging to : ArchRLogs/ArchR-addGroupCoverages-faba40cd6ba1-Date-2020-04-15_Time-10-23-14.log  
## If there is an issue, please report to github with logFile!  
## B CellGroups N = 2CD4.M CellGroups N = 2CD4.N CellGroups N = 2CLP CellGroups N = 2Erythroid CellGroups N = 2GMP CellGroups N = 2Mono CellGroups N = 2NK CellGroups N = 2pDC CellGroups N = 2PreB CellGroups N = 2Progenitor CellGroups N = 2  
## B CellGroups N = 2CD4.M CellGroups N = 2CD4.N CellGroups N = 2CLP CellGroups N = 2Erythroid CellGroups N = 2GMP CellGroups N = 2Mono CellGroups N = 2NK CellGroups N = 2pDC CellGroups N = 2PreB CellGroups N = 2Progenitor CellGroups N = 2  
## B CellGroups N = 2CD4.M CellGroups N = 2CD4.N CellGroups N = 2CLP CellGroups N = 2Erythroid CellGroups N = 2GMP CellGroups N = 2Mono CellGroups N = 2NK CellGroups N = 2pDC CellGroups N = 2PreB CellGroups N = 2Progenitor CellGroups N = 2  
## B CellGroups N = 2CD4.M CellGroups N = 2CD4.N CellGroups N = 2CLP CellGroups N = 2Erythroid CellGroups N = 2GMP CellGroups N = 2Mono CellGroups N = 2NK CellGroups N = 2pDC CellGroups N = 2PreB CellGroups N = 2Progenitor CellGroups N = 2  
## B CellGroups N = 2CD4.M CellGroups N = 2CD4.N CellGroups N = 2CLP CellGroups N = 2Erythroid CellGroups N = 2GMP CellGroups N = 2Mono CellGroups N = 2NK CellGroups N = 2pDC CellGroups N = 2PreB CellGroups N = 2Progenitor CellGroups N = 2  
## B CellGroups N = 2CD4.M CellGroups N = 2CD4.N CellGroups N = 2CLP CellGroups N = 2Erythroid CellGroups N = 2GMP CellGroups N = 2Mono CellGroups N = 2NK CellGroups N = 2pDC CellGroups N = 2PreB CellGroups N = 2Progenitor CellGroups N = 2  
## B CellGroups N = 2CD4.M CellGroups N = 2CD4.N CellGroups N = 2CLP CellGroups N = 2Erythroid CellGroups N = 2GMP CellGroups N = 2Mono CellGroups N = 2NK CellGroups N = 2pDC CellGroups N = 2PreB CellGroups N = 2Progenitor CellGroups N = 2  
## B CellGroups N = 2CD4.M CellGroups N = 2CD4.N CellGroups N = 2CLP CellGroups N = 2Erythroid CellGroups N = 2GMP CellGroups N = 2Mono CellGroups N = 2NK CellGroups N = 2pDC CellGroups N = 2PreB CellGroups N = 2Progenitor CellGroups N = 2  
## B CellGroups N = 2CD4.M CellGroups N = 2CD4.N CellGroups N = 2CLP CellGroups N = 2Erythroid CellGroups N = 2GMP CellGroups N = 2Mono CellGroups N = 2NK CellGroups N = 2pDC CellGroups N = 2PreB CellGroups N = 2Progenitor CellGroups N = 2  
## B CellGroups N = 2CD4.M CellGroups N = 2CD4.N CellGroups N = 2CLP CellGroups N = 2Erythroid CellGroups N = 2GMP CellGroups N = 2Mono CellGroups N = 2NK CellGroups N = 2pDC CellGroups N = 2PreB CellGroups N = 2Progenitor CellGroups N = 2  
## B CellGroups N = 2CD4.M CellGroups N = 2CD4.N CellGroups N = 2CLP CellGroups N = 2Erythroid CellGroups N = 2GMP CellGroups N = 2Mono CellGroups N = 2NK CellGroups N = 2pDC CellGroups N = 2PreB CellGroups N = 2Progenitor CellGroups N = 2  
## 2020-04-15 10:23:17 : Creating Coverage Files!, 0.046 mins elapsed.  
## 2020-04-15 10:23:17 : Batch Execution w/ safelapply!, 0.046 mins elapsed.  
## 2020-04-15 10:24:14 : Adding Kmer Bias to Coverage Files!, 0.996 mins elapsed.  
##  
## Completed Kmer Bias Calculation  
## Adding Kmer Bias (1 of 22)  
## Adding Kmer Bias (2 of 22)  
## Adding Kmer Bias (3 of 22)  
## Adding Kmer Bias (4 of 22)  
## Adding Kmer Bias (5 of 22)  
## Adding Kmer Bias (6 of 22)  
## Adding Kmer Bias (7 of 22)  
## Adding Kmer Bias (8 of 22)  
## Adding Kmer Bias (9 of 22)  
## Adding Kmer Bias (10 of 22)  
## Adding Kmer Bias (11 of 22)  
## Adding Kmer Bias (12 of 22)  
## Adding Kmer Bias (13 of 22)  
## Adding Kmer Bias (14 of 22)  
## Adding Kmer Bias (15 of 22)  
## Adding Kmer Bias (16 of 22)  
## Adding Kmer Bias (17 of 22)  
## Adding Kmer Bias (18 of 22)  
## Adding Kmer Bias (19 of 22)  
## Adding Kmer Bias (20 of 22)  
## Adding Kmer Bias (21 of 22)  
## Adding Kmer Bias (22 of 22)  
## 2020-04-15 10:25:16 : Finished Creation of Coverage Files!, 2.022 mins elapsed.   
## ArchR logging successful to : ArchRLogs/ArchR-addGroupCoverages-faba40cd6ba1-Date-2020-04-15_Time-10-23-14.log  

```{r eval=FALSE}
#Find Path to Macs2 binary. This function sometimes struggles if you are using conda environment
#for installation of python packages. If this function fails try providing a direct path manually
pathToMacs2 <- findMacs2()
```
> \## Searching For MACS2..  
## Found with $path!  

```{r eval=FALSE}
#Call Reproducible Peaks w/ Macs2 (~5-10 minutes)
projHeme4 <- addReproduciblePeakSet(
	ArchRProj = projHeme4, 
	groupBy = "Clusters2", 
	pathToMacs2 = pathToMacs2
)
```
> \## ArchR logging to : ArchRLogs/ArchR-addReproduciblePeakSet-faba719c230c-Date-2020-04-15_Time-10-25-16.log  
## If there is an issue, please report to github with logFile!  
## Calling Peaks with Macs2  
## 2020-04-15 10:25:17 : Peak Calling Parameters!, 0.014 mins elapsed.  
##                 Group nCells nCellsUsed nReplicates nMin nMax maxPeaks  
## B                   B    439        433           2  182  251   150000  
## CD4.M           CD4.M    678        617           2  117  500   150000  
## CD4.N           CD4.N   1271        552           2   52  500   150000  
## CLP               CLP    387        387           2   82  305   150000  
## Erythroid   Erythroid    879        694           2  194  500   150000  
## GMP               GMP    793        751           2  251  500   150000  
## Mono             Mono   2632       1000           2  500  500   150000  
## NK                 NK    851        801           2  301  500   150000  
## pDC               pDC    320        311           2  150  161   150000  
## PreB             PreB    351        351           2   40  311   150000  
## Progenitor Progenitor   1650        672           2  172  500   150000  
## 2020-04-15 10:25:17 : Batching Peak Calls!, 0.014 mins elapsed.  
## 2020-04-15 10:25:17 : Batch Execution w/ safelapply!, 0 mins elapsed.  
## 2020-04-15 10:27:01 : Identifying Reproducible Peaks!, 1.74 mins elapsed.  
## 2020-04-15 10:27:09 : Creating Union Peak Set!, 1.886 mins elapsed.  
## Converged after 7 iterations!  
## [1] "plotting ggplot!"  
## 2020-04-15 10:27:15 : Finished Creating Union Peak Set (144009)!, 1.98 mins elapsed.  

```{r eval=FALSE}
getPeakSet(projHeme4)
```
> \## GRanges object with 144009 ranges and 12 metadata columns:  
##             seqnames              ranges strand |     score  
##                <Rle>           <IRanges>  <Rle> | <numeric>  
##        Mono     chr1       752499-752999      * |  24.54003  
##          NK     chr1       762651-763151      * | 141.22064  
##           B     chr1       801006-801506      * |  14.18461  
##           B     chr1       805039-805539      * |  37.30365  
##         CLP     chr1       845325-845825      * |   2.81281  
##         ...      ...                 ...    ... .       ...  
##   Erythroid     chrX 154664540-154665040      * |   7.09786  
##          NK     chrX 154807324-154807824      * |   9.38477  
##        PreB     chrX 154840785-154841285      * |   3.29501  
##        PreB     chrX 154842404-154842904      * |   7.68692  
##          NK     chrX 154862017-154862517      * |     12.44  
##             replicateScoreQuantile groupScoreQuantile Reproducibility  
##                          <numeric>          <numeric>       <numeric>  
##        Mono                  0.812              0.655               2  
##          NK                  0.887              0.797               2  
##           B                    0.7              0.418               2  
##           B                  0.959                0.9               2  
##         CLP                  0.706              0.311               2  
##         ...                    ...                ...             ...  
##   Erythroid                   0.69              0.279               2  
##          NK                  0.424              0.166               2  
##        PreB                  0.669              0.274               2  
##        PreB                  0.932              0.772               2  
##          NK                  0.492              0.202               2  
##                         GroupReplicate distToGeneStart nearestGene    peakType  
##                            <character>       <integer> <character> <character>  
##        Mono      Mono._.scATAC_PBMC_R1           10152   LINC00115      Distal  
##          NK        NK._.scATAC_PBMC_R1               0   LINC00115    Promoter   
##           B         B._.scATAC_PBMC_R1           10925      FAM41C      Distal  
##           B         B._.scATAC_BMMC_R1            6892      FAM41C    Intronic  
##         CLP       CLP._.scATAC_BMMC_R1            9241   LINC02593      Distal  
##         ...                        ...             ...         ...         ...  
##   Erythroid Erythroid._.scATAC_BMMC_R1          100803       CLIC2      Distal  
##          NK        NK._.scATAC_PBMC_R1           35047       TMLHE    Intronic  
##        PreB                PreB._.Rep2            1586       TMLHE    Intronic  
##        PreB                PreB._.Rep2              31       TMLHE    Promoter  
##          NK        NK._.scATAC_PBMC_R1           19644       TMLHE      Distal  
##             distToTSS  nearestTSS        GC       idx  
##             <integer> <character> <numeric> <integer>  
##        Mono     10152  uc001aau.3     0.483         1  
##          NK         0  uc001aau.3    0.6906         2  
##           B     10925  uc021oei.1    0.4371         3  
##           B      6892  uc021oei.1    0.7285         4  
##         CLP      1239  uc010nxu.2    0.7904         5  
##         ...       ...         ...       ...       ...  
##   Erythroid     22387  uc004cim.2     0.515      3462  
##          NK     35047  uc004cin.3     0.525      3463  
##        PreB      1586  uc004cin.3     0.485      3464  
##        PreB        31  uc004cin.3    0.5888      3465  
##          NK     19644  uc004cin.3    0.4212      3466  
##   -------  
##   seqinfo: 23 sequences from an unspecified genome; no seqlengths  

## Calling Peaks w/ TileMatrix

```{r eval=FALSE}
#Call Reproducible Peaks w/ TileMatrix (~2 minutes)
projHemeTmp <- addReproduciblePeakSet(
	ArchRProj = projHeme4, 
	groupBy = "Clusters2",
	peakMethod = "Tiles",
	method = "p"
)
```
> \## ArchR logging to : ArchRLogs/ArchR-addReproduciblePeakSet-faba49485623-Date-2020-04-15_Time-10-27-15.log  
## If there is an issue, please report to github with logFile!  
## Calling Peaks with TileMatrix   
## Group Coverages Already Computed Returning Groups, Set force = TRUE to Recompute!  
## List of length 11  
## names(11): B CD4.M CD4.N CLP Erythroid GMP Mono NK pDC PreB Progenitor  
## 2020-04-15 10:27:16 : Peak Calling Parameters!, 0.012 mins elapsed.  
##                 Group nCells nCellsUsed nReplicates nMin nMax maxPeaks  
## B                   B    439        433           2  182  251   150000  
## CD4.M           CD4.M    678        617           2  117  500   150000  
## CD4.N           CD4.N   1271        552           2   52  500   150000  
## CLP               CLP    387        387           2   82  305   150000  
## Erythroid   Erythroid    879        694           2  194  500   150000  
## GMP               GMP    793        751           2  251  500   150000  
## Mono             Mono   2632       1000           2  500  500   150000  
## NK                 NK    851        801           2  301  500   150000  
## pDC               pDC    320        311           2  150  161   150000  
## PreB             PreB    351        351           2   40  311   150000  
## Progenitor Progenitor   1650        672           2  172  500   150000  
## 2020-04-15 10:27:24 : Computing Total Accessibility Across All Features, 0.138 mins elapsed.  
## 2020-04-15 10:27:27 : Computing Pseudo-Grouped Tile Matrix, 0.19 mins elapsed.  
## 2020-04-15 10:28:07 : Created Pseudo-Grouped Tile Matrix (0.474 GB), 0.866 mins elapsed.  
## Expectation = 0.137769200114613Expectation = 0.170233276575844Expectation = 0.502430087836881Expectation = 0.0743484690298421Expectation = 0.231170400914268Expectation = 0.0293481232153502Expectation = 0.188425424281447Expectation = 0.0412872862125409Expectation = 0.369108556109225Expectation = 0.120760726012825Expectation = 0.33554874172927Expectation = 0.330837760307742Expectation = 0.41068352704434Expectation = 0.343789007051322Expectation = 0.571660173038985Expectation = 0.180700422552374Expectation = 0.126306931769154Expectation = 0.107652051338631Expectation = 0.200607316117261Expectation = 0.0248573103536859Expectation = 0.38762741617292Expectation = 0.0990605372969163  
## 2020-04-15 10:28:51 : Creating Group Peak Sets with Annotations!, 1.59 mins elapsed.  
## 2020-04-15 10:29:03 : Creating Union Peak Set with Annotations!, 1.789 mins elapsed.  
## Annotating Peaks : Nearest Gene  
## Annotating Peaks : Gene  
## Annotating Peaks : TSS  
## Annotating Peaks : GC  
## [1] "plotting ggplot!"  
## 2020-04-15 10:29:22 : Finished Creating Union Peak Set (271917)!, 2.104 mins elapsed.  

```{r eval=FALSE}
getPeakSet(projHemeTmp)
```
> \## GRanges object with 271917 ranges and 9 metadata columns:  
##            seqnames              ranges strand |   mlog10p      Group  
##               <Rle>           <IRanges>  <Rle> | <numeric>      <Rle>  
##        [1]     chr1       752000-752499      * |     0.742       Mono  
##        [2]     chr1       752500-752999      * |     6.561       Mono  
##        [3]     chr1       758500-758999      * |      1.11         NK  
##        [4]     chr1       762000-762499      * |     2.461         NK  
##        [5]     chr1       762500-762999      * |    22.536         NK  
##        ...      ...                 ...    ... .       ...        ...  
##   [271913]     chrX 154862000-154862499      * |     1.966         NK  
##   [271914]     chrX 154862500-154862999      * |     1.334         NK  
##   [271915]     chrX 154912500-154912999      * |      2.01  Erythroid  
##   [271916]     chrX 154997000-154997499      * |     1.349 Progenitor  
##   [271917]     chrX 154998000-154998499      * |     1.227       Mono  
##            distToGeneStart nearestGene    peakType distToTSS  nearestTSS  
##                  <integer> <character> <character> <integer> <character>  
##        [1]           10652   LINC00115      Distal     10652  uc001aau.3  
##        [2]           10152   LINC00115      Distal     10152  uc001aau.3  
##        [3]            4152   LINC00115      Distal      4152  uc001aau.3  
##        [4]             652   LINC00115    Promoter       652  uc001aau.3  
##        [5]             152   LINC00115    Promoter       152  uc001aau.3  
##        ...             ...         ...         ...       ...         ...  
##   [271913]           19626       TMLHE      Distal     19626  uc004cin.3  
##   [271914]           20126       TMLHE      Distal     20126  uc004cin.3  
##   [271915]           70126       TMLHE      Distal     70126  uc004cin.3  
##   [271916]          154626       TMLHE      Distal       201  uc004cin.3  
##   [271917]          155626       TMLHE      Distal       797  uc004cin.3  
##                   GC       idx  
##            <numeric> <integer>  
##        [1]     0.376         1  
##        [2]     0.484         2  
##        [3]      0.56         3  
##        [4]     0.574         4  
##        [5]     0.684         5  
##        ...       ...       ...  
##   [271913]      0.43      6212  
##   [271914]      0.42      6213  
##   [271915]     0.368      6214  
##   [271916]     0.542      6215  
##   [271917]     0.476      6216  
##   -------   
##   seqinfo: 23 sequences from an unspecified genome; no seqlengths  

Lets see how well this compares to Macs2

```{r eval=FALSE}
#Peak Overlap
length(subsetByOverlaps(getPeakSet(projHeme4), getPeakSet(projHemeTmp))) / length(getPeakSet(projHeme4))
```
> \## [1] 0.9627246

```{r eval=FALSE}
length(subsetByOverlaps(getPeakSet(projHemeTmp), getPeakSet(projHeme4))) / length(getPeakSet(projHemeTmp))
```
> \## [1] 0.7533365

```{r eval=FALSE}
#Extend to 1kb
length(subsetByOverlaps(resize(getPeakSet(projHeme4), 1000, "center"), getPeakSet(projHemeTmp))) / length(getPeakSet(projHeme4))
```
> \## [1] 0.9676687

```{r eval=FALSE}
length(subsetByOverlaps(getPeakSet(projHemeTmp), resize(getPeakSet(projHeme4), 1000, "center"))) / length(getPeakSet(projHemeTmp))
```
> \## [1] 0.8287639

## Add Peak Matrix

We can now save our original projHeme4 using `saveArchRProject` from ArchR.

```{r eval=FALSE}
saveArchRProject(ArchRProj = projHeme4, outputDirectory = "Save-ProjHeme4", load = FALSE)
```
> \## Copying ArchRProject to new outputDirectory : /oak/stanford/groups/howchang/users/jgranja/ArchRTutorial/ArchRBook/BookOutput4/Save-ProjHeme4  
## Copying Arrow Files...  
## Copying Arrow Files (1 of 3)  
## Copying Arrow Files (2 of 3)  
## Copying Arrow Files (3 of 3)  
## Getting ImputeWeights  
## Dropping ImputeWeights...  
## Copying Other Files...  
## Copying Other Files (1 of 6): Embeddings  
## Copying Other Files (2 of 6): GroupCoverages  
## Copying Other Files (3 of 6): IterativeLSI    
## Copying Other Files (4 of 6): IterativeLSI2  
## Copying Other Files (5 of 6): PeakCalls  
## Copying Other Files (6 of 6): Plots  
## Saving ArchRProject...  

```{r eval=FALSE}
getPeakSet(projHeme4)
```
> \## GRanges object with 144009 ranges and 12 metadata columns:  
##             seqnames              ranges strand |     score  
##                <Rle>           <IRanges>  <Rle> | <numeric>  
##        Mono     chr1       752499-752999      * |  24.54003  
##          NK     chr1       762651-763151      * | 141.22064  
##           B     chr1       801006-801506      * |  14.18461  
##           B     chr1       805039-805539      * |  37.30365  
##         CLP     chr1       845325-845825      * |   2.81281  
##         ...      ...                 ...    ... .       ...   
##   Erythroid     chrX 154664540-154665040      * |   7.09786  
##          NK     chrX 154807324-154807824      * |   9.38477  
##        PreB     chrX 154840785-154841285      * |   3.29501  
##        PreB     chrX 154842404-154842904      * |   7.68692  
##          NK     chrX 154862017-154862517      * |     12.44  
##             replicateScoreQuantile groupScoreQuantile Reproducibility  
##                          <numeric>          <numeric>       <numeric>  
##        Mono                  0.812              0.655               2  
##          NK                  0.887              0.797               2  
##           B                    0.7              0.418               2  
##           B                  0.959                0.9               2  
##         CLP                  0.706              0.311               2  
##         ...                    ...                ...             ...  
##   Erythroid                   0.69              0.279               2  
##          NK                  0.424              0.166               2  
##        PreB                  0.669              0.274               2  
##        PreB                  0.932              0.772               2  
##          NK                  0.492              0.202               2  
##                         GroupReplicate distToGeneStart nearestGene    peakType  
##                            <character>       <integer> <character> <character>  
##        Mono      Mono._.scATAC_PBMC_R1           10152   LINC00115      Distal  
##          NK        NK._.scATAC_PBMC_R1               0   LINC00115    Promoter  
##           B         B._.scATAC_PBMC_R1           10925      FAM41C      Distal  
##           B         B._.scATAC_BMMC_R1            6892      FAM41C    Intronic  
##         CLP       CLP._.scATAC_BMMC_R1            9241   LINC02593      Distal  
##         ...                        ...             ...         ...         ...  
##   Erythroid Erythroid._.scATAC_BMMC_R1          100803       CLIC2      Distal  
##          NK        NK._.scATAC_PBMC_R1           35047       TMLHE    Intronic  
##        PreB                PreB._.Rep2            1586       TMLHE    Intronic  
##        PreB                PreB._.Rep2              31       TMLHE    Promoter  
##          NK        NK._.scATAC_PBMC_R1           19644       TMLHE      Distal  
##             distToTSS  nearestTSS        GC       idx  
##             <integer> <character> <numeric> <integer>  
##        Mono     10152  uc001aau.3     0.483         1  
##          NK         0  uc001aau.3    0.6906         2  
##           B     10925  uc021oei.1    0.4371         3  
##           B      6892  uc021oei.1    0.7285         4  
##         CLP      1239  uc010nxu.2    0.7904         5  
##         ...       ...         ...       ...       ...  
##   Erythroid     22387  uc004cim.2     0.515      3462  
##          NK     35047  uc004cin.3     0.525      3463  
##        PreB      1586  uc004cin.3     0.485      3464  
##        PreB        31  uc004cin.3    0.5888      3465  
##          NK     19644  uc004cin.3    0.4212      3466  
##   -------  
##   seqinfo: 23 sequences from an unspecified genome; no seqlengths  

```{r eval=FALSE}
#~1-2 Minutes
projHeme5 <- addPeakMatrix(projHeme4)
```
> \## ArchR logging to : ArchRLogs/ArchR-addPeakMatrix-faba377aa2f2-Date-2020-04-15_Time-10-29-40.log  
## If there is an issue, please report to github with logFile!  
## 2020-04-15 10:29:41 : Batch Execution w/ safelapply!, 0 mins elapsed.  
## ArchR logging successful to : ArchRLogs/ArchR-addPeakMatrix-faba377aa2f2-Date-2020-04-15_Time-10-29-40.log  

```{r eval=FALSE}
getAvailableMatrices(projHeme5)
```
> \## [1] "GeneIntegrationMatrix" "GeneScoreMatrix"       "PeakMatrix"             
## [4] "TileMatrix"

## Session Information

```{r eval=FALSE}
Sys.Date()
```
> \## [1] "2020-04-15"

```{r eval=FALSE}
sessionInfo()
```

```{r, include=FALSE, eval=FALSE}
save.image(params$out, compress = FALSE)
```
