---
output:
  html_document:
    theme: yeti  # many options for theme, this one is my favorite.
params:
  threads: 20
  rdata: ""
  token: ""
  ref: ""
  out: ""
---

```{r, include=FALSE, eval=FALSE}
params2 <- params
unlockBinding("params", env = .GlobalEnv)
#load("Chapter-13-Footprints.Rdata")
load(params$rdata)
params <- params2
rm(params2)
devtools::install_github("GreenleafLab/ArchR", 
  auth_token = params$token, 
  ref = params$ref,
  repos = BiocManager::repositories(),
  dependencies = FALSE
)
library(ArchR)
fn <- unclass(lsf.str(envir = asNamespace("ArchR"), all = TRUE))
fn <- fn[!grepl("\\.", fn)]
fn <- fn[fn!="ArchRProj"]
for (i in seq_along(fn)){
    tryCatch({
        eval(parse(text = paste0(fn[i], "<-ArchR::", fn[i])))
    }, error = function(x) {
    })
}
addArchRThreads(threads = params$threads)
addArchRGenome("hg19")
# fn <- unclass(lsf.str(envir = asNamespace("ArchR"), all = TRUE))
# fn <- fn[fn!="ArchRProj"]
# for (i in seq_along(fn)) {
#     tryCatch({
#         eval(parse(text = paste0(fn[i], "<-ArchR:::", fn[i])))
#     }, error = function(x) {
#     })
# }
set.seed(1)
```

# Identifying Marker Peaks with ArchR

We have introduced the identification of marker features previously in the chapter discussing gene scores. The same function (`getMarkerFeatures()`) can be used to identify marker features from any matrix stored in an `ArchRProject`. Marker features are features that are unique to a specific cell grouping. These can be very useful in understanding cluster- or cell type-specific biology. In this chapter, we will discuss how to use this function to identify marker peaks.

## Identifying Marker Peaks with ArchR

Often times, we are interested to know which peaks are unique to an individual cluster or a small group of clusters. We can do this in an unsupervised fashion in ArchR using the `addMarkerFeatures()` function in combination with `useMatrix = "PeakMatrix"`.

First, lets remind ourselves of the cell types that we are working with in `projHeme5` and their relative proportions.
```{r eval=FALSE}
#Our scRNA labels
table(projHeme5$Clusters2)
```
> \##          B      CD4.M      CD4.N        CLP  Erythroid        GMP       Mono   
##        439        678       1271        387        879        793       2632   
##         NK        pDC       PreB Progenitor   
##        851        320        351       1650  

Now, we are ready to identify marker peaks by calling the `addMarkerFeatures()` function with `useMatrix = "PeakMatrix"`. Additionally, we tell ArchR to account for differences in data quality amongst the cell groups by setting the `bias` parameter to account for TSS enrichment and the number of unique fragments per cell.
```{r eval=FALSE}
markersPeaks <- getMarkerFeatures(
	ArchRProj = projHeme5, 
	useMatrix = "PeakMatrix", 
	groupBy = "Clusters2",
  bias = c("TSSEnrichment", "log10(nFrags)"),
  testMethod = "wilcoxon"
)
```
> \## ArchR logging to : ArchRLogs/ArchR-getMarkerFeatures-ff9f78e54080-Date-2020-04-15_Time-10-31-43.log  
## If there is an issue, please report to github with logFile!  
## MatrixClass = Sparse.Integer.Matrix  
## 2020-04-15 10:31:44 : Matching Known Biases, 0.011 mins elapsed.  
##  
## ###########  
## 2020-04-15 10:33:02 : Completed Pairwise Tests, 1.301 mins elapsed.  
## ###########  
## ArchR logging successful to : ArchRLogs/ArchR-getMarkerFeatures-ff9f78e54080-Date-2020-04-15_Time-10-31-43.log  

The object returned by the `getMarkerFeatures()` function is a `SummarizedExperiment` that contains a few different `assays`.
```{r eval=FALSE}
markersPeaks
```
> \## class: SummarizedExperiment   
## dim: 144009 11   
## metadata(2): MatchInfo Params   
## assays(6): Log2FC Mean ... AUC MeanBGD  
## rownames(144009): 1 2 ... 144008 144009  
## rowData names(4): seqnames idx start end  
## colnames(11): B CD4.M ... PreB Progenitor  
## colData names(0):  

We can use the `getMarkers()` function to retrieve particular slices of this `SummarizedExperiment` that we are interested in. The default behavior of this function is to return a list of `DataFrame` objects, one for each cell group.
```{r eval=FALSE}
markerList <- getMarkers(markersPeaks, cutOff = "FDR <= 0.01 & Log2FC >= 1")
markerList
```
> \## List of length 11  
## names(11): B CD4.M CD4.N CLP Erythroid GMP Mono NK pDC PreB Progenitor

If we are interested in the marker peaks for a specific cell group, we can access this from the list via the `$` accessor.
```{r eval=FALSE}
markerList$Erythroid
```
> \## DataFrame with 2267 rows and 7 columns  
##        seqnames     idx     start       end           Log2FC  
##           <Rle> <array>   <array>   <array>        <numeric>  
## 6915       chr1    6915 110407005 110407505 6.13156087199941  
## 89187     chr22    1265  30129827  30130327 4.31793618061602  
## 2715       chr1    2715  27869127  27869627 8.81830883930008  
## 9273       chr1    9273 164681433 164681933 4.09761463117515  
## 46278     chr15    2875  74902688  74903188 5.94807465189627  
## ...         ...     ...       ...       ...              ...  
## 124072     chr7    2640  47608032  47608532 1.57044726023651  
## 13738      chr1   13738 248018421 248018921 2.18899218987721  
## 90192     chr22    2270  39632393  39632893 2.09633755722991  
## 30353     chr12    2227  47600919  47601419 1.98865307113263  
## 51935     chr16    3926  70729551  70730051 2.33056750082649  
##                         FDR          MeanDiff  
##                   <numeric>         <numeric>  
## 6915    8.5785377923846e-15 0.908051252811916  
## 89187  1.00297518036045e-13   1.0458579648286  
## 2715   3.96321274318867e-12 0.908093680025248  
## 9273   3.96321274318867e-12 0.743482307090926  
## 46278  3.96321274318867e-12 0.739990672777532  
## ...                     ...               ...  
## 124072  0.00939699723029238 0.345440715489147  
## 13738   0.00940220652828868 0.214851028991864  
## 90192   0.00960407719383025 0.270261700407864  
## 30353   0.00978687988910092 0.348162927084217  
## 51935   0.00982181746221214 0.327156785147329  

Instead of a list of `DataFrame` objects, we can use `getMarkers()` to return a `GRangesList` object by setting `returnGR = TRUE`.
```{r eval=FALSE}
markerList <- getMarkers(markersPeaks, cutOff = "FDR <= 0.01 & Log2FC >= 1", returnGR = TRUE)
markerList
```
> \## GRangesList object of length 11:  
## $B  
## GRanges object with 594 ranges and 3 metadata columns:  
##         seqnames              ranges strand |           Log2FC  
##            <Rle>           <IRanges>  <Rle> |        <numeric>  
##     [1]     chr2 232537191-232537691      * | 4.60006288742652  
##     [2]    chr12   92566305-92566805      * |  4.3389825818619  
##     [3]     chr3   13152070-13152570      * | 4.44200650760392  
##     [4]     chr9   37409171-37409671      * | 3.32811859032542  
##     [5]     chr1 160759469-160759969      * | 5.66666023794885  
##     ...      ...                 ...    ... .              ...  
##   [590]     chr8 128222178-128222678      * |  6.1639699924098  
##   [591]     chr9   93643862-93644362      * | 6.57677988412609  
##   [592]     chrX     6656761-6657261      * |  6.4417508983755  
##   [593]    chr14   81425876-81426376      * | 4.62521198459712  
##   [594]     chr7   63765291-63765791      * | 4.10893943379989  
##                          FDR          MeanDiff  
##                    <numeric>         <numeric>  
##     [1] 1.61183300601365e-12  1.09207677080711  
##     [2] 9.91817474781581e-10 0.993976831608143  
##     [3] 4.25560916639498e-09  1.20028559994278  
##     [4] 1.45797719387195e-08  1.03421998926309  
##     [5] 1.58046323198917e-08  0.80285226332275  
##     ...                  ...               ...  
##   [590]   0.0098441599959607  0.20375624865997  
##   [591]   0.0098441599959607  0.27221052581318  
##   [592]   0.0098441599959607 0.247631445395118  
##   [593]  0.00994427516588697 0.432810696716752  
##   [594]  0.00998116335641915 0.326535661912967  
##   -------  
##   seqinfo: 23 sequences from an unspecified genome; no seqlengths  
##   
## ...  
## <10 more elements>  

This `GRangesList` object can similarly be subset to a `GRanges` object for a particular cell group using the `$` accessor.
```{r eval=FALSE}
markerList$Erythroid
```
> \## GRanges object with 2267 ranges and 3 metadata columns:  
##          seqnames              ranges strand |           Log2FC  
##             <Rle>           <IRanges>  <Rle> |        <numeric>  
##      [1]     chr1 110407005-110407505      * | 6.13156087199941  
##      [2]    chr22   30129827-30130327      * | 4.31793618061602  
##      [3]     chr1   27869127-27869627      * | 8.81830883930008  
##      [4]     chr1 164681433-164681933      * | 4.09761463117515  
##      [5]    chr15   74902688-74903188      * | 5.94807465189627  
##      ...      ...                 ...    ... .              ...  
##   [2263]     chr7   47608032-47608532      * | 1.57044726023651  
##   [2264]     chr1 248018421-248018921      * | 2.18899218987721  
##   [2265]    chr22   39632393-39632893      * | 2.09633755722991  
##   [2266]    chr12   47600919-47601419      * | 1.98865307113263  
##   [2267]    chr16   70729551-70730051      * | 2.33056750082649  
##                           FDR          MeanDiff  
##                     <numeric>         <numeric>  
##      [1]  8.5785377923846e-15 0.908051252811916  
##      [2] 1.00297518036045e-13   1.0458579648286  
##      [3] 3.96321274318867e-12 0.908093680025248  
##      [4] 3.96321274318867e-12 0.743482307090926  
##      [5] 3.96321274318867e-12 0.739990672777532  
##      ...                  ...               ...  
##   [2263]  0.00939699723029238 0.345440715489147  
##   [2264]  0.00940220652828868 0.214851028991864  
##   [2265]  0.00960407719383025 0.270261700407864  
##   [2266]  0.00978687988910092 0.348162927084217  
##   [2267]  0.00982181746221214 0.327156785147329  
##   -------  
##   seqinfo: 23 sequences from an unspecified genome; no seqlengths  

## Plotting Marker Peaks in ArchR

ArchR provides multiple plotting functions to interact with the `SummarizedExperiment` objects returned by `getMarkerFeatures()`.

### Marker Peak Heatmaps

We can visualize these marker peaks (or any features output by `getMarkerFeatures()`) as a heatmap using the `markerHeatmap()` function.

```{r eval=FALSE}
heatmapPeaks <- markerHeatmap(
  seMarker = markersPeaks, 
  cutOff = "FDR <= 0.1 & Log2FC >= 0.5",
  transpose = TRUE
)
```
> \## Identified 40034 markers!  
## Preparing Main Heatmap..

We can plot this heatmap using `draw()`.
```{r eval=FALSE}
draw(heatmapPeaks, heatmap_legend_side = "bot", annotation_legend_side = "bot")
```

```{r, include=FALSE, eval=FALSE}
plotPDF(heatmapPeaks, name = "Peak-Marker-Heatmap", width = 8, height = 6, ArchRProj = projHeme5, addDOC = FALSE)
ArchR:::.convertToPNG(ArchRProj = projHeme5)
system("cp Figures/*.png images/HemeWalkthrough/PNG/")
system("cp Figures/*.pdf images/HemeWalkthrough/PDF/")
```

![](images/HemeWalkthrough/PNG/Peak-Marker-Heatmap_1.png){width=800 height=600}

To save an editable vectorized version of this plot, we use the `plotPDF()` function.
```{r eval=FALSE}
plotPDF(heatmapPeaks, name = "Peak-Marker-Heatmap", width = 8, height = 6, ArchRProj = projHeme5, addDOC = FALSE)
```
> \## [1] 0

### Marker Peak MA and Volcano Plots

Instead of plotting a heatmap, we can also plot an MA or Volcano plot for any individual cell group. To do this, we use the `markerPlot()` function. For an MA plot we specify `plotAs = "MA"`. Here we specify the "Erythroid" cell group via the `name` parameter.
```{r eval=FALSE}
pma <- markerPlot(seMarker = markersPeaks, name = "Erythroid", cutOff = "FDR <= 0.1 & Log2FC >= 1", plotAs = "MA")
pma
```
> \## Warning: Removed 55 rows containing missing values (geom_point_rast).

```{r, include=FALSE, eval=FALSE}
plotPDF(pma, name = "Erythroid-Markers-MA-Volcano", width = 5, height = 5, ArchRProj = projHeme5, addDOC = FALSE)
ArchR:::.convertToPNG(ArchRProj = projHeme5)
system("cp Figures/*.png images/HemeWalkthrough/PNG/")
system("cp Figures/*.pdf images/HemeWalkthrough/PDF/")
```

![](images/HemeWalkthrough/PNG/Erythroid-Markers-MA-Volcano_1.png){width=600 height=600}

Similarly, for a Volcano plot, we specify `plotAs = "Volcano"`.
```{r eval=FALSE}
pv <- markerPlot(seMarker = markersPeaks, name = "Erythroid", cutOff = "FDR <= 0.1 & Log2FC >= 1", plotAs = "Volcano")
pv
```
> \## Warning: Removed 55 rows containing missing values (geom_point_rast).

```{r, include=FALSE, eval=FALSE}
plotPDF(pma, pv, name = "Erythroid-Markers-MA-Volcano", width = 5, height = 5, ArchRProj = projHeme5, addDOC = FALSE)
ArchR:::.convertToPNG(ArchRProj = projHeme5)
system("cp Figures/*.png images/HemeWalkthrough/PNG/")
system("cp Figures/*.pdf images/HemeWalkthrough/PDF/")
```

![](images/HemeWalkthrough/PNG/Erythroid-Markers-MA-Volcano_2.png){width=600 height=600}

To save an editable vectorized version of these plots, we use the `plotPDF()` function.

```{r eval=FALSE}
plotPDF(pma, pv, name = "Erythroid-Markers-MA-Volcano", width = 5, height = 5, ArchRProj = projHeme5, addDOC = FALSE)
```
> \## [1] "plotting ggplot!"  
## Warning: Removed 55 rows containing missing values (geom_point_rast).  
## Warning: Removed 55 rows containing missing values (geom_point_rast).  
## [1] "plotting ggplot!"  
## Warning: Removed 55 rows containing missing values (geom_point_rast).  
## Warning: Removed 55 rows containing missing values (geom_point_rast).  
## [1] 0  

### Marker Peaks in Browser Tracks

Additionally we can see these peak regions overlayed on our browser tracks by passing the relevant peak regions to the `features` parameterin the `plotBrowserTrack()` function. This will add an additional BED-style track of marker peak regions to the bottom of our ArchR track plot. Here we specify plotting the GATA1 gene via the `geneSymbol` parameter.

```{r eval=FALSE}
p <- plotBrowserTrack(
    ArchRProj = projHeme5, 
    groupBy = "Clusters2", 
    geneSymbol = c("GATA1"),
    features =  getMarkers(markersPeaks, cutOff = "FDR <= 0.1 & Log2FC >= 1", returnGR = TRUE)["Erythroid"],
    upstream = 50000,
    downstream = 50000
)
```
> \## ArchR logging to : ArchRLogs/ArchR-plotBrowserTrack-ff9f580fdbcb-Date-2020-04-15_Time-10-35-20.log  
## If there is an issue, please report to github with logFile!  
## Warning: 'GenomicRangesList' is deprecated.  
## Use 'GRangesList(..., compress=FALSE)' instead.  
## See help("Deprecated")  
## 2020-04-15 10:35:23 : Validating Region, 0.045 mins elapsed.  
## GRanges object with 1 range and 2 metadata columns:  
##       seqnames            ranges strand |     gene_id      symbol  
##          <Rle>         <IRanges>  <Rle> | <character> <character>  
##   [1]     chrX 48644982-48652717      + |        2623       GATA1  
##   -------  
##   seqinfo: 24 sequences from hg19 genome  
## 2020-04-15 10:35:23 : Adding Bulk Tracks (1 of 1), 0.047 mins elapsed.  
## 2020-04-15 10:35:26 : Adding Feature Tracks (1 of 1), 0.095 mins elapsed.  
## 2020-04-15 10:35:26 : Adding Gene Tracks (1 of 1), 0.1 mins elapsed.  
## 2020-04-15 10:35:27 : Plotting, 0.109 mins elapsed.  
## ArchR logging successful to : ArchRLogs/ArchR-plotBrowserTrack-ff9f580fdbcb-Date-2020-04-15_Time-10-35-20.log  

We can plot this using `grid::grid.draw()`.
```{r eval=FALSE}
grid::grid.newpage()
grid::grid.draw(p$GATA1)
```

```{r, include=FALSE, eval=FALSE}
plotPDF(p, name = "Plot-Tracks-With-Features", width = 5, height = 5, ArchRProj = projHeme5, addDOC = FALSE)
ArchR:::.convertToPNG(ArchRProj = projHeme5)
system("cp Figures/*.png images/HemeWalkthrough/PNG/")
system("cp Figures/*.pdf images/HemeWalkthrough/PDF/")
```

![](images/HemeWalkthrough/PNG/Plot-Tracks-With-Features_1.png){width=600 height=600}

To save an editable vectorized version of this plot, we use the `plotPDF()` function.

```{r eval=FALSE}
plotPDF(p, name = "Plot-Tracks-With-Features", width = 5, height = 5, ArchRProj = projHeme5, addDOC = FALSE)
```
> \## NULL  
## [1] 0

## Pairwise Testing Between Groups

Marker feature identification is a very specific type of differential test. However, ArchR also enables standard differential testing using the same `getMarkerFeatures()` function. The trick is to set `useGroups` to one of the two cell groups and `bgdGroups` to the other cell group. This performs a differential test between the two provided groups. To be explicitly clear, the cell groups here are user-defined. They can be anything that fits your application. All you need to do is add a new column to `cellColData` that defines the groups that you would like to compare. In all differential tests, the peaks that are higher in the group passed to `useGroups` will have positive fold change values while the peaks that are higher in the group passed to `bgdGroups` will have negative fold change values. 

Here we perform a pairwise test between the "Erythroid" cell group and the "Progenitor" cell group.

```{r eval=FALSE}
markerTest <- getMarkerFeatures(
  ArchRProj = projHeme5, 
  useMatrix = "PeakMatrix",
  groupBy = "Clusters2",
  testMethod = "wilcoxon",
  bias = c("TSSEnrichment", "log10(nFrags)"),
  useGroups = "Erythroid",
  bgdGroups = "Progenitor"
)
```
> \## ArchR logging to : ArchRLogs/ArchR-getMarkerFeatures-ff9f77b592d7-Date-2020-04-15_Time-10-35-37.log  
## If there is an issue, please report to github with logFile!  
## MatrixClass = Sparse.Integer.Matrix  
## 2020-04-15 10:35:41 : Matching Known Biases, 0.057 mins elapsed.  
##   
## 2020-04-15 10:35:44 : Computing Pairwise Tests (1 of 1), 0.107 mins elapsed.  
## ###########  
## 2020-04-15 10:36:13 : Completed Pairwise Tests, 0.584 mins elapsed.   
## ###########  
## ArchR logging successful to : ArchRLogs/ArchR-getMarkerFeatures-ff9f77b592d7-Date-2020-04-15_Time-10-35-37.log  

We can then plot an MA or Volcano plot using the `markerPlot()` function. We indicate an MA plot using `plotAs = "MA"`.

```{r eval=FALSE}
pma <- markerPlot(seMarker = markerTest, name = "Erythroid", cutOff = "FDR <= 0.1 & abs(Log2FC) >= 1", plotAs = "MA")
pma
```
> \## Warning: Removed 57 rows containing missing values (geom_point_rast).

```{r, include=FALSE, eval=FALSE}
plotPDF(pma, name = "Erythroid-vs-Progenitor-Markers-MA-Volcano", width = 5, height = 5, ArchRProj = projHeme5, addDOC = FALSE)
ArchR:::.convertToPNG(ArchRProj = projHeme5)
system("cp Figures/*.png images/HemeWalkthrough/PNG/")
system("cp Figures/*.pdf images/HemeWalkthrough/PDF/")
```

![](images/HemeWalkthrough/PNG/Erythroid-vs-Progenitor-Markers-MA-Volcano_1.png){width=600 height=600}

Similarly, we plot a volvano plot using `plotAs = "Volcano"`.
```{r eval=FALSE}
pv <- markerPlot(seMarker = markerTest, name = "Erythroid", cutOff = "FDR <= 0.1 & abs(Log2FC) >= 1", plotAs = "Volcano")
pv
```
> \## Warning: Removed 57 rows containing missing values (geom_point_rast).

```{r, include=FALSE, eval=FALSE}
plotPDF(pma, pv, name = "Erythroid-vs-Progenitor-Markers-MA-Volcano", width = 5, height = 5, ArchRProj = projHeme5, addDOC = FALSE)
ArchR:::.convertToPNG(ArchRProj = projHeme5)
system("cp Figures/*.png images/HemeWalkthrough/PNG/")
system("cp Figures/*.pdf images/HemeWalkthrough/PDF/")
```

![](images/HemeWalkthrough/PNG/Erythroid-vs-Progenitor-Markers-MA-Volcano_2.png){width=600 height=600}

To save an editable vectorized version of these plots, we use the `plotPDF()` function.

```{r eval=FALSE}
plotPDF(pma, pv, name = "Erythroid-vs-Progenitor-Markers-MA-Volcano", width = 5, height = 5, ArchRProj = projHeme5, addDOC = FALSE)
```
> \## [1] "plotting ggplot!"  
## Warning: Removed 57 rows containing missing values (geom_point_rast).  
## Warning: Removed 57 rows containing missing values (geom_point_rast).  
## [1] "plotting ggplot!"  
## Warning: Removed 57 rows containing missing values (geom_point_rast).  
## Warning: Removed 57 rows containing missing values (geom_point_rast).  
## [1] 0  

We will continue this differential analysis in the next chapter by looking for motif enrichments in our differentially accessible peaks.


```{r, include=FALSE, eval=FALSE}
save.image(params$out, compress = FALSE)
```

```{r, include=FALSE, eval=FALSE}
ArchR:::.convertToPNG(ArchRProj = projHeme2)
system("cp Figures/*.png images/HemeWalkthrough/PNG/")
system("cp Figures/*.pdf images/HemeWalkthrough/PDF/")
```
