---
output:
  html_document:
    theme: yeti  # many options for theme, this one is my favorite.
params:
  threads: 20
  rdata: ""
  token: ""
  ref: ""
  out: ""
---

```{r, include=FALSE, eval=FALSE}
params2 <- params
unlockBinding("params", env = .GlobalEnv)
#load("Chapter-13-Footprints.Rdata")
load(params$rdata)
params <- params2
rm(params2)
devtools::install_github("GreenleafLab/ArchR", 
  auth_token = params$token, 
  ref = params$ref,
  repos = BiocManager::repositories(),
  dependencies = FALSE
)
library(ArchR)
fn <- unclass(lsf.str(envir = asNamespace("ArchR"), all = TRUE))
fn <- fn[!grepl("\\.", fn)]
fn <- fn[fn!="ArchRProj"]
for (i in seq_along(fn)){
    tryCatch({
        eval(parse(text = paste0(fn[i], "<-ArchR::", fn[i])))
    }, error = function(x) {
    })
}
addArchRThreads(threads = params$threads)
addArchRGenome("hg19")
# fn <- unclass(lsf.str(envir = asNamespace("ArchR"), all = TRUE))
# fn <- fn[fn!="ArchRProj"]
# for (i in seq_along(fn)) {
#     tryCatch({
#         eval(parse(text = paste0(fn[i], "<-ArchR:::", fn[i])))
#     }, error = function(x) {
#     })
# }
set.seed(1)
```

# Peak Annotation Enrichment with ArchR

## Motif Enrichment

```{r eval=FALSE}
#Motif Search in Peak Set and add to Peak Annotations
if("Motif" %ni% names(projHeme5@peakAnnotation)){
  projHeme5 <- addMotifAnnotations(ArchRProj = projHeme5, motifSet = "cisbp", name = "Motif")
}

#Identify Motif Enrichments
enrichMotifs <- peakAnnoEnrichment(
    seMarker = markersPeaks,
    ArchRProj = projHeme5,
    peakAnnotation = "Motif",
    cutOff = "FDR <= 0.1 & Log2FC >= 0.5"
  )
```
> \## 2020-04-15 10:43:01 : Computing Enrichments 1 of 11, 0.046 mins elapsed.  
## 2020-04-15 10:43:01 : Computing Enrichments 2 of 11, 0.058 mins elapsed.  
## 2020-04-15 10:43:02 : Computing Enrichments 3 of 11, 0.072 mins elapsed.  
## 2020-04-15 10:43:03 : Computing Enrichments 4 of 11, 0.094 mins elapsed.  
## 2020-04-15 10:43:04 : Computing Enrichments 5 of 11, 0.103 mins elapsed.  
## 2020-04-15 10:43:05 : Computing Enrichments 6 of 11, 0.113 mins elapsed.  
## 2020-04-15 10:43:05 : Computing Enrichments 7 of 11, 0.121 mins elapsed.  
## 2020-04-15 10:43:06 : Computing Enrichments 8 of 11, 0.13 mins elapsed.  
## 2020-04-15 10:43:06 : Computing Enrichments 9 of 11, 0.139 mins elapsed.  
## 2020-04-15 10:43:07 : Computing Enrichments 10 of 11, 0.148 mins elapsed.  
## 2020-04-15 10:43:08 : Computing Enrichments 11 of 11, 0.17 mins elapsed.  

```{r eval=FALSE}
enrichMotifs
```
> \## class: SummarizedExperiment   
## dim: 870 11   
## metadata(0):  
## assays(10): mlog10Padj mlog10p ... CompareFrequency feature  
## rownames(870): TFAP2B_1 TFAP2D_2 ... TBX18_869 TBX22_870  
## rowData names(0):  
## colnames(11): B CD4.M ... PreB Progenitor  
## colData names(0):  

```{r eval=FALSE}
#Heatmap
heatmapEM <- enrichHeatmap(enrichMotifs, n = 7, transpose = TRUE)
```
> \## Adding Annotations..  
## Preparing Main Heatmap..  

```{r eval=FALSE}
draw(heatmapEM, heatmap_legend_side = "bot", annotation_legend_side = "bot")
```

```{r, include=FALSE, eval=FALSE}
plotPDF(heatmapEM, name = "Motifs-Enriched-Marker-Heatmap", width = 8, height = 6, ArchRProj = projHeme5, addDOC = FALSE)
ArchR:::.convertToPNG(ArchRProj = projHeme5)
system("cp Figures/*.png images/HemeWalkthrough/PNG/")
system("cp Figures/*.pdf images/HemeWalkthrough/PDF/")
```

![](images/HemeWalkthrough/PNG/Motifs-Enriched-Marker-Heatmap_1.png){width=800 height=600}

We can then plot this 

```{r eval=FALSE}
plotPDF(heatmapEM, name = "Motifs-Enriched-Marker-Heatmap", width = 8, height = 6, ArchRProj = projHeme5, addDOC = FALSE)
```
> \## [1] 0

## ArchR Enrichment

### Encode TFBS

```{r eval=FALSE}
projHeme5 <- addArchRAnnotations(ArchRProj = projHeme5, collection = "EncodeTFBS")
```
> \## Annotation ArchR-Hg19-v1.Anno does not exist! Downloading..  
## Annotating Chr:  1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 X  

```{r eval=FALSE}
#Identify EncodeTFBS Enrichments
enrichEncode <- peakAnnoEnrichment(
    seMarker = markersPeaks,
    ArchRProj = projHeme5,
    peakAnnotation = "EncodeTFBS",
    cutOff = "FDR <= 0.1 & Log2FC >= 0.5"
  )
```
> \## 2020-04-15 10:43:34 : Computing Enrichments 1 of 11, 0.036 mins elapsed.  
## 2020-04-15 10:43:34 : Computing Enrichments 2 of 11, 0.041 mins elapsed.  
## 2020-04-15 10:43:34 : Computing Enrichments 3 of 11, 0.047 mins elapsed.  
## 2020-04-15 10:43:35 : Computing Enrichments 4 of 11, 0.054 mins elapsed.  
## 2020-04-15 10:43:35 : Computing Enrichments 5 of 11, 0.063 mins elapsed.  
## 2020-04-15 10:43:36 : Computing Enrichments 6 of 11, 0.066 mins elapsed.  
## 2020-04-15 10:43:36 : Computing Enrichments 7 of 11, 0.071 mins elapsed.   
## 2020-04-15 10:43:36 : Computing Enrichments 8 of 11, 0.075 mins elapsed.  
## 2020-04-15 10:43:36 : Computing Enrichments 9 of 11, 0.079 mins elapsed.  
## 2020-04-15 10:43:37 : Computing Enrichments 10 of 11, 0.083 mins elapsed.  
## 2020-04-15 10:43:37 : Computing Enrichments 11 of 11, 0.087 mins elapsed.  

```{r eval=FALSE}
enrichEncode
```
> \## class: SummarizedExperiment   
## dim: 689 11   
## metadata(0):  
## assays(10): mlog10Padj mlog10p ... CompareFrequency feature  
## rownames(689): 1.CTCF-Dnd41... 2.EZH2_39-Dnd41... ...  
##   688.CTCF-WERI_Rb_1... 689.CTCF-WI_38...  
## rowData names(0):   
## colnames(11): B CD4.M ... PreB Progenitor  
## colData names(0):  

```{r eval=FALSE}
#Heatmap
heatmapEncode <- enrichHeatmap(enrichEncode, n = 7, transpose = TRUE)
```
> \## Adding Annotations..  
## Preparing Main Heatmap..  

```{r eval=FALSE}
draw(heatmapEncode, heatmap_legend_side = "bot", annotation_legend_side = "bot")
```

```{r, include=FALSE, eval=FALSE}
plotPDF(heatmapEncode, name = "EncodeTFBS-Enriched-Marker-Heatmap", width = 8, height = 6, ArchRProj = projHeme5, addDOC = FALSE)
ArchR:::.convertToPNG(ArchRProj = projHeme5)
system("cp Figures/*.png images/HemeWalkthrough/PNG/")
system("cp Figures/*.pdf images/HemeWalkthrough/PDF/")
```

![](images/HemeWalkthrough/PNG/EncodeTFBS-Enriched-Marker-Heatmap_1.png){width=800 height=600}

We can then plot this 

```{r eval=FALSE}
plotPDF(heatmapEncode, name = "EncodeTFBS-Enriched-Marker-Heatmap", width = 8, height = 6, ArchRProj = projHeme5, addDOC = FALSE)
```
> \## [1] 0

### Bulk ATAC-seq

```{r eval=FALSE}
projHeme5 <- addArchRAnnotations(ArchRProj = projHeme5, collection = "ATAC")
```
> \## Annotating Chr:  1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 X

```{r eval=FALSE}
#Identify ATAC Enrichments
enrichATAC <- peakAnnoEnrichment(
    seMarker = markersPeaks,
    ArchRProj = projHeme5,
    peakAnnotation = "ATAC",
    cutOff = "FDR <= 0.1 & Log2FC >= 0.5"
  )
```
> \## 2020-04-15 10:43:53 : Computing Enrichments 1 of 11, 0.027 mins elapsed.  
## 2020-04-15 10:43:54 : Computing Enrichments 2 of 11, 0.028 mins elapsed.  
## 2020-04-15 10:43:54 : Computing Enrichments 3 of 11, 0.03 mins elapsed.  
## 2020-04-15 10:43:54 : Computing Enrichments 4 of 11, 0.034 mins elapsed.  
## 2020-04-15 10:43:54 : Computing Enrichments 5 of 11, 0.036 mins elapsed.  
## 2020-04-15 10:43:54 : Computing Enrichments 6 of 11, 0.038 mins elapsed.  
## 2020-04-15 10:43:54 : Computing Enrichments 7 of 11, 0.04 mins elapsed.  
## 2020-04-15 10:43:54 : Computing Enrichments 8 of 11, 0.043 mins elapsed.  
## 2020-04-15 10:43:55 : Computing Enrichments 9 of 11, 0.045 mins elapsed.  
## 2020-04-15 10:43:55 : Computing Enrichments 10 of 11, 0.047 mins elapsed.  
## 2020-04-15 10:43:55 : Computing Enrichments 11 of 11, 0.049 mins elapsed.  

```{r eval=FALSE}
enrichATAC
```
> \## class: SummarizedExperiment   
## dim: 96 11   
## metadata(0):  
## assays(10): mlog10Padj mlog10p ... CompareFrequency feature  
## rownames(96): Brain_Astrocytes Brain_Excitatory_neurons ... Heme_MPP  
##   Heme_NK  
## rowData names(0):  
## colnames(11): B CD4.M ... PreB Progenitor  
## colData names(0):  

```{r eval=FALSE}
#Heatmap
heatmapATAC <- enrichHeatmap(enrichATAC, n = 7, transpose = TRUE)
```
> \## Adding Annotations..  
## Preparing Main Heatmap..  

```{r eval=FALSE}
draw(heatmapATAC, heatmap_legend_side = "bot", annotation_legend_side = "bot")
```

```{r, include=FALSE, eval=FALSE}
plotPDF(heatmapATAC, name = "ATAC-Enriched-Marker-Heatmap", width = 8, height = 6, ArchRProj = projHeme5, addDOC = FALSE)
ArchR:::.convertToPNG(ArchRProj = projHeme5)
system("cp Figures/*.png images/HemeWalkthrough/PNG/")
system("cp Figures/*.pdf images/HemeWalkthrough/PDF/")
```

![](images/HemeWalkthrough/PNG/ATAC-Enriched-Marker-Heatmap_1.png){width=800 height=600}

We can then plot this 

```{r eval=FALSE}
plotPDF(heatmapATAC, name = "ATAC-Enriched-Marker-Heatmap", width = 8, height = 6, ArchRProj = projHeme5, addDOC = FALSE)
```
> \## [1] 0

### Codex TFBS

```{r eval=FALSE}
projHeme5 <- addArchRAnnotations(ArchRProj = projHeme5, collection = "Codex")
```
> \## Annotating Chr:  1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 X

```{r eval=FALSE}
#Identify Codex Enrichments
enrichCodex <- peakAnnoEnrichment(
    seMarker = markersPeaks,
    ArchRProj = projHeme5,
    peakAnnotation = "Codex",
    cutOff = "FDR <= 0.1 & Log2FC >= 0.5"
  )
```
> \## 2020-04-15 10:44:11 : Computing Enrichments 1 of 11, 0.025 mins elapsed.  
## 2020-04-15 10:44:11 : Computing Enrichments 2 of 11, 0.027 mins elapsed.  
## 2020-04-15 10:44:11 : Computing Enrichments 3 of 11, 0.029 mins elapsed.  
## 2020-04-15 10:44:11 : Computing Enrichments 4 of 11, 0.032 mins elapsed.  
## 2020-04-15 10:44:11 : Computing Enrichments 5 of 11, 0.034 mins elapsed.  
## 2020-04-15 10:44:11 : Computing Enrichments 6 of 11, 0.035 mins elapsed.  
## 2020-04-15 10:44:11 : Computing Enrichments 7 of 11, 0.036 mins elapsed.  
## 2020-04-15 10:44:11 : Computing Enrichments 8 of 11, 0.037 mins elapsed.  
## 2020-04-15 10:44:11 : Computing Enrichments 9 of 11, 0.038 mins elapsed.  
## 2020-04-15 10:44:11 : Computing Enrichments 10 of 11, 0.039 mins elapsed.  
## 2020-04-15 10:44:12 : Computing Enrichments 11 of 11, 0.041 mins elapsed.  

```{r eval=FALSE}
enrichCodex
```
> \## class: SummarizedExperiment   
## dim: 189 11   
## metadata(0):  
## assays(10): mlog10Padj mlog10p ... CompareFrequency feature  
## rownames(189): 1.STAT5-No_drug_(DMSO)... 2.RUNX3-GM12878_cell_fr... ...  
##   188.TP53-codex_Embryonic... 189.TP53-codex_Embryonic...  
## rowData names(0):  
## colnames(11): B CD4.M ... PreB Progenitor  
## colData names(0):  

```{r eval=FALSE}
#Heatmap
heatmapCodex <- enrichHeatmap(enrichCodex, n = 7, transpose = TRUE)
```
> \## Adding Annotations..  
## Preparing Main Heatmap..  

```{r eval=FALSE}
draw(heatmapCodex, heatmap_legend_side = "bot", annotation_legend_side = "bot")
```

```{r, include=FALSE, eval=FALSE}
plotPDF(heatmapCodex, name = "Codex-Enriched-Marker-Heatmap", width = 8, height = 6, ArchRProj = projHeme5, addDOC = FALSE)
ArchR:::.convertToPNG(ArchRProj = projHeme5)
system("cp Figures/*.png images/HemeWalkthrough/PNG/")
system("cp Figures/*.pdf images/HemeWalkthrough/PDF/")
```

![](images/HemeWalkthrough/PNG/Codex-Enriched-Marker-Heatmap_1.png){width=800 height=600}

We can then plot this 

```{r eval=FALSE}
plotPDF(heatmapCodex, name = "Codex-Enriched-Marker-Heatmap", width = 8, height = 6, ArchRProj = projHeme5, addDOC = FALSE)
```
> \## [1] 0

## Custom Enrichment

```{r eval=FALSE}
#Custom Peak Annotations
EncodePeaks <- c(
  Encode_K562_GATA1 = "https://www.encodeproject.org/files/ENCFF632NQI/@@download/ENCFF632NQI.bed.gz",
  Encode_GM12878_CEBPB = "https://www.encodeproject.org/files/ENCFF761MGJ/@@download/ENCFF761MGJ.bed.gz",
  Encode_K562_Ebf1 = "https://www.encodeproject.org/files/ENCFF868VSY/@@download/ENCFF868VSY.bed.gz",
  Encode_K562_Pax5 = "https://www.encodeproject.org/files/ENCFF339KUO/@@download/ENCFF339KUO.bed.gz"
)
projHeme5 <- addPeakAnnotations(ArchRProj = projHeme5, regions = EncodePeaks, name = "ChIP")
```
> \## 2020-04-15 10:44:23 : Creating Peak Overlap Matrix, 0.036 mins elapsed.

```{r eval=FALSE}
#Identify ChIP Enrichments
enrichRegions <- peakAnnoEnrichment(
    seMarker = markersPeaks,
    ArchRProj = projHeme5,
    peakAnnotation = "ChIP",
    cutOff = "FDR <= 0.1 & Log2FC >= 0.5"
  )
```
> \## 2020-04-15 10:44:25 : Computing Enrichments 1 of 11, 0.027 mins elapsed.  
## 2020-04-15 10:44:25 : Computing Enrichments 2 of 11, 0.027 mins elapsed.  
## 2020-04-15 10:44:25 : Computing Enrichments 3 of 11, 0.027 mins elapsed.  
## 2020-04-15 10:44:25 : Computing Enrichments 4 of 11, 0.028 mins elapsed.  
## 2020-04-15 10:44:25 : Computing Enrichments 5 of 11, 0.028 mins elapsed.  
## 2020-04-15 10:44:25 : Computing Enrichments 6 of 11, 0.029 mins elapsed.  
## 2020-04-15 10:44:25 : Computing Enrichments 7 of 11, 0.029 mins elapsed.  
## 2020-04-15 10:44:25 : Computing Enrichments 8 of 11, 0.03 mins elapsed.  
## 2020-04-15 10:44:25 : Computing Enrichments 9 of 11, 0.03 mins elapsed.  
## 2020-04-15 10:44:25 : Computing Enrichments 10 of 11, 0.03 mins elapsed.  
## 2020-04-15 10:44:26 : Computing Enrichments 11 of 11, 0.033 mins elapsed.  

```{r eval=FALSE}
enrichRegions
```
> \## class: SummarizedExperiment 
## dim: 4 11 
## metadata(0):
## assays(10): mlog10Padj mlog10p ... CompareFrequency feature
## rownames(4): Encode_K562_GATA1 Encode_GM12878_CEBPB Encode_K562_Ebf1
##   Encode_K562_Pax5
## rowData names(0):
## colnames(11): B CD4.M ... PreB Progenitor
## colData names(0):

```{r eval=FALSE}
#Heatmap
heatmapRegions <- enrichHeatmap(enrichRegions, n = 7, transpose = TRUE)
```
> \## Adding Annotations..  
## Preparing Main Heatmap..  

```{r eval=FALSE}
draw(heatmapRegions, heatmap_legend_side = "bot", annotation_legend_side = "bot")
```

```{r, include=FALSE, eval=FALSE}
plotPDF(heatmapRegions, name = "Regions-Enriched-Marker-Heatmap", width = 8, height = 6, ArchRProj = projHeme5, addDOC = FALSE)
ArchR:::.convertToPNG(ArchRProj = projHeme5)
system("cp Figures/*.png images/HemeWalkthrough/PNG/")
system("cp Figures/*.pdf images/HemeWalkthrough/PDF/")
```

![](images/HemeWalkthrough/PNG/Regions-Enriched-Marker-Heatmap_1.png){width=800 height=600}

We can then plot this 

```{r eval=FALSE}
plotPDF(heatmapRegions, name = "Regions-Enriched-Marker-Heatmap", width = 8, height = 6, ArchRProj = projHeme5, addDOC = FALSE)
```
> \## [1] 0


```{r, include=FALSE, eval=FALSE}
save.image(params$out, compress = FALSE)
```


