---
output:
  html_document:
    theme: yeti  # many options for theme, this one is my favorite.
params:
  threads: 20
  rdata: ""
  token: ""
  ref: ""
  out: ""
---

```{r, include=FALSE, eval=FALSE}
params2 <- params
unlockBinding("params", env = .GlobalEnv)
#load("Chapter-13-Footprints.Rdata")
load(params$rdata)
params <- params2
rm(params2)
devtools::install_github("GreenleafLab/ArchR", 
  auth_token = params$token, 
  ref = params$ref,
  repos = BiocManager::repositories(),
  dependencies = FALSE
)
library(ArchR)
fn <- unclass(lsf.str(envir = asNamespace("ArchR"), all = TRUE))
fn <- fn[!grepl("\\.", fn)]
fn <- fn[fn!="ArchRProj"]
for (i in seq_along(fn)){
    tryCatch({
        eval(parse(text = paste0(fn[i], "<-ArchR::", fn[i])))
    }, error = function(x) {
    })
}
addArchRThreads(threads = params$threads)
addArchRGenome("hg19")
# fn <- unclass(lsf.str(envir = asNamespace("ArchR"), all = TRUE))
# fn <- fn[fn!="ArchRProj"]
# for (i in seq_along(fn)) {
#     tryCatch({
#         eval(parse(text = paste0(fn[i], "<-ArchR:::", fn[i])))
#     }, error = function(x) {
#     })
# }
set.seed(1)
```

# Footprinting with ArchR

Transcription factor (TF) footprinting allows for the prediction of the precise binding location of a TF at a particular locus. This is because the DNA bases that are bound by the TF are actually protected from transposition while the DNA bases immediately adjacent to TF binding are accessible.

![](/images/footprintingSchematic.png){width=250px}

Ideally, TF footprinting is performed at a single site to determine the precise binding location of the TF. However, in practice, this requires very high sequencing depth, often much higher depth than what most users would obtain from either bulk or single-cell ATAC-seq. To get around this problem, we can combine Tn5 insertion locations across many instances of predicted TF binding. For example, we can take all peaks that harbor a CTCF motif and make an aggregate TF footprint for CTCF across the whole genome.

The accuracy of this footprint relies on generating a reliable curated list of predicted binding sites for the TF of interest. ArchR does this in a naive way via the `addMotifAnnotations()` functions by searching the peak region for any DNA sequence matching a motif. Depending on the degeneracy of the motif of interest, this may or may not be sufficient. These motif annotations get added to the __QQQArchRProject???__ as a binary representation on a per peak basis (0 = no motif, 1 = motif present). Once you have these motif annotations, ArchR performs footprinting using the `plotFootprints()` function which takes as input an `ArchRProject` object and a `GenomicRanges` object that contains the positions of the motifs. These positions can be accessed from the `ArchRProject` via the `getPositions()` function.

Perhaps most importantly, the footprinting analyses in ArchR account for known Tn5 insertion sequence bias.To do this, ArchR uses a matrix of hexamer position frequencies and a matrix of k-mer frequencies at Tn5 insertion sites:

![](/images/footprintingMethods.png){width=1000px}

All put together, the `plotFootprints()` function generates footprint plots that take into account Tn5 insertion bias.


## Motif Footprinting

Transcription factor footprinting can also be done in ArchR with a single command. We note that the footprints generated by the tutorial data are not as clean as would be desired but this is because of the small size of the tutorial dataset.

```{r eval=FALSE}
motifPositions <- getPositions(projHeme5)
motifPositions
```
> \## GRangesList object of length 870:  
## $TFAP2B_1  
## GRanges object with 16773 ranges and 1 metadata column:  
##           seqnames              ranges strand |            score  
##              <Rle>           <IRanges>  <Rle> |        <numeric>  
##       [1]     chr1       852468-852479      + | 8.17731199746359  
##       [2]     chr1       873916-873927      + | 8.32673820065588  
##       [3]     chr1       873916-873927      - | 8.32673820065588  
##       [4]     chr1       896671-896682      + | 9.96223327271814  
##       [5]     chr1       896671-896682      - | 8.92408377606486  
##       ...      ...                 ...    ... .              ...  
##   [16769]     chrX 153991101-153991112      + | 8.39549159740639  
##   [16770]     chrX 154299568-154299579      + | 8.90119825654299  
##   [16771]     chrX 154664929-154664940      - | 8.16690864294221  
##   [16772]     chrX 154807684-154807695      + | 9.57636587154549  
##   [16773]     chrX 154807684-154807695      - | 10.6117355833828  
##   -------  
##   seqinfo: 23 sequences from an unspecified genome; no seqlengths  
##   
## ...  
## <869 more elements>

Find TF-Motifs

```{r eval=FALSE}
motifs <- c("GATA1", "CEBPA", "EBF1", "IRF4", "TBX21", "PAX5")
markerMotifs <- unlist(lapply(motifs, function(x) grep(x, names(motifPositions), value = TRUE)))
markerMotifs <- markerMotifs[markerMotifs %ni% "SREBF1_22"]
markerMotifs
```
> \## [1] "GATA1_383" "CEBPA_155" "EBF1_67"   "IRF4_632"  "TBX21_780" "PAX5_709"

```{r eval=FALSE}
#Create Group Coverage Files if you have not done so already
projHeme5 <- addGroupCoverages(ArchRProj = projHeme5, groupBy = "Clusters2")
```

Now we can compute footprints

```{r eval=FALSE}
#Plot Motif Footprints from positions list
#Recommend doing a few motifs not entire motif set
seFoot <- getFootprints(
  ArchRProj = projHeme5, 
  positions = motifPositions[markerMotifs], 
  groupBy = "Clusters2"
)
```
> \## ArchR logging to : ArchRLogs/ArchR-getFootprints-10b7a1d11042e-Date-2020-04-15_Time-11-15-50.log  
## If there is an issue, please report to github with logFile!  
## 2020-04-15 11:15:51 : Computing Kmer Bias Table, 0.024 mins elapsed.  
## 2020-04-15 11:16:13 : Finished Computing Kmer Tables, 0.361 mins elapsed.  
## 2020-04-15 11:16:13 : Computing Footprints, 0.385 mins elapsed.  
## 2020-04-15 11:16:22 : Computing Footprints Bias, 0.539 mins elapsed.   
## 2020-04-15 11:16:28 : Summarizing Footprints, 0.642 mins elapsed.  

### No Normalization

```{r eval=FALSE}
#Plot Motif Footprints from positions list
#Recommend doing a few motifs not entire motif set
plotFootprints(
  seFoot = seFoot,
  ArchRProj = projHeme5, 
  normMethod = "None",
  plotName = "Footprints-No-Normalization",
  addDOC = FALSE,
  smoothWindow = 5
)
```
> \## ArchR logging to : ArchRLogs/ArchR-plotFootprints-10b7a22669572-Date-2020-04-15_Time-11-16-30.log  
## If there is an issue, please report to github with logFile!  
## 2020-04-15 11:16:31 : Plotting Footprint : GATA1_383 (1 of 6), 0.009 mins elapsed.  
## Applying smoothing window to footprint  
## Normalizing by flanking regions    
## NormMethod = None  
## 2020-04-15 11:16:35 : Plotting Footprint : CEBPA_155 (2 of 6), 0.077 mins elapsed.  
## Applying smoothing window to footprint  
## Normalizing by flanking regions  
## NormMethod = None  
## 2020-04-15 11:16:38 : Plotting Footprint : EBF1_67 (3 of 6), 0.125 mins elapsed.  
## Applying smoothing window to footprint  
## Normalizing by flanking regions  
## NormMethod = None  
## 2020-04-15 11:16:41 : Plotting Footprint : IRF4_632 (4 of 6), 0.173 mins elapsed.    
## Applying smoothing window to footprint  
## Normalizing by flanking regions  
## NormMethod = None  
## 2020-04-15 11:16:44 : Plotting Footprint : TBX21_780 (5 of 6), 0.221 mins elapsed.  
## Applying smoothing window to footprint  
## Normalizing by flanking regions  
## NormMethod = None  
## 2020-04-15 11:16:46 : Plotting Footprint : PAX5_709 (6 of 6), 0.27 mins elapsed.  
## Applying smoothing window to footprint  
## Normalizing by flanking regions  
## NormMethod = None  
## ArchR logging successful to : ArchRLogs/ArchR-plotFootprints-10b7a22669572-Date-2020-04-15_Time-11-16-30.log  

```{r, include=FALSE, eval=FALSE}
ArchR:::.convertToPNG(ArchRProj = projHeme2)
system("cp Figures/*.png images/HemeWalkthrough/PNG/")
system("cp Figures/*.pdf images/HemeWalkthrough/PDF/")
```

By default the plots will be in the outputDirectory of the ArchRProject. The challenge is that if you requested to plot all motifs and returned this as a ggplot object it could be extremely large. An example of motif footprints ran from this function are seen below.

![](images/HemeWalkthrough/PNG/Footprints-No-Normalization_1.png){width=250 height=400}
![](images/HemeWalkthrough/PNG/Footprints-No-Normalization_2.png){width=250 height=400}
![](images/HemeWalkthrough/PNG/Footprints-No-Normalization_3.png){width=250 height=400}

### Subtract Tn5 Bias 

```{r eval=FALSE}
#Re-Plot with different normalization method
plotFootprints(
  seFoot = seFoot,
  ArchRProj = projHeme5, 
  normMethod = "Subtract",
  plotName = "Footprints-Subtract-Bias",
  addDOC = FALSE,
  smoothWindow = 5
)
```
> \## ArchR logging to : ArchRLogs/ArchR-plotFootprints-10b7a2038428-Date-2020-04-15_Time-11-16-59.log  
## If there is an issue, please report to github with logFile!  
## 2020-04-15 11:16:59 : Plotting Footprint : GATA1_383 (1 of 6), 0.007 mins elapsed.  
## Applying smoothing window to footprint  
## Normalizing by flanking regions   
## NormMethod = Subtract  
## 2020-04-15 11:17:02 : Plotting Footprint : CEBPA_155 (2 of 6), 0.065 mins elapsed.  
## Applying smoothing window to footprint  
## Normalizing by flanking regions  
## NormMethod = Subtract  
## 2020-04-15 11:17:05 : Plotting Footprint : EBF1_67 (3 of 6), 0.109 mins elapsed.  
## Applying smoothing window to footprint  
## Normalizing by flanking regions  
## NormMethod = Subtract  
## 2020-04-15 11:17:08 : Plotting Footprint : IRF4_632 (4 of 6), 0.155 mins elapsed.  
## Applying smoothing window to footprint  
## Normalizing by flanking regions  
## NormMethod = Subtract  
## 2020-04-15 11:17:11 : Plotting Footprint : TBX21_780 (5 of 6), 0.199 mins elapsed.  
## Applying smoothing window to footprint  
## Normalizing by flanking regions  
## NormMethod = Subtract  
## 2020-04-15 11:17:13 : Plotting Footprint : PAX5_709 (6 of 6), 0.245 mins elapsed.  
## Applying smoothing window to footprint  
## Normalizing by flanking regions  
## NormMethod = Subtract  
## ArchR logging successful to : ArchRLogs/ArchR-plotFootprints-10b7a2038428-Date-2020-04-15_Time-11-16-59.log  

```{r, include=FALSE, eval=FALSE}
ArchR:::.convertToPNG(ArchRProj = projHeme2)
system("cp Figures/*.png images/HemeWalkthrough/PNG/")
system("cp Figures/*.pdf images/HemeWalkthrough/PDF/")
```

By default the plots will be in the outputDirectory of the ArchRProject. The challenge is that if you requested to plot all motifs and returned this as a ggplot object it could be extremely large. An example of motif footprints ran from this function are seen below.

![](images/HemeWalkthrough/PNG/Footprints-Subtract-Bias_1.png){width=250 height=400}
![](images/HemeWalkthrough/PNG/Footprints-Subtract-Bias_2.png){width=250 height=400}
![](images/HemeWalkthrough/PNG/Footprints-Subtract-Bias_3.png){width=250 height=400}

### Divide Tn5 Bias 

```{r eval=FALSE}
#Re-Plot with different normalization method
plotFootprints(
  seFoot = seFoot,
  ArchRProj = projHeme5, 
  normMethod = "Divide",
  plotName = "Footprints-Divide-Bias",
  addDOC = FALSE,
  smoothWindow = 5
)
```
> \## ArchR logging to : ArchRLogs/ArchR-plotFootprints-10b7a703225b0-Date-2020-04-15_Time-11-17-23.log  
## If there is an issue, please report to github with logFile!  
## 2020-04-15 11:17:23 : Plotting Footprint : GATA1_383 (1 of 6), 0.008 mins elapsed.  
## Applying smoothing window to footprint  
## Normalizing by flanking regions  
## NormMethod = Divide  
## 2020-04-15 11:17:27 : Plotting Footprint : CEBPA_155 (2 of 6), 0.073 mins elapsed.  
## Applying smoothing window to footprint  
## Normalizing by flanking regions  
## NormMethod = Divide  
## 2020-04-15 11:17:30 : Plotting Footprint : EBF1_67 (3 of 6), 0.13 mins elapsed.  
## Applying smoothing window to footprint  
## Normalizing by flanking regions  
## NormMethod = Divide  
## 2020-04-15 11:17:34 : Plotting Footprint : IRF4_632 (4 of 6), 0.189 mins elapsed.  
## Applying smoothing window to footprint  
## Normalizing by flanking regions  
## NormMethod = Divide  
## 2020-04-15 11:17:38 : Plotting Footprint : TBX21_780 (5 of 6), 0.248 mins elapsed.  
## Applying smoothing window to footprint  
## Normalizing by flanking regions    
## NormMethod = Divide  
## 2020-04-15 11:17:41 : Plotting Footprint : PAX5_709 (6 of 6), 0.307 mins elapsed.  
## Applying smoothing window to footprint  
## Normalizing by flanking regions  
## NormMethod = Divide  
## ArchR logging successful to : ArchRLogs/ArchR-plotFootprints-10b7a703225b0-Date-2020-04-15_Time-11-17-23.log  

```{r, include=FALSE, eval=FALSE}
ArchR:::.convertToPNG(ArchRProj = projHeme2)
system("cp Figures/*.png images/HemeWalkthrough/PNG/")
system("cp Figures/*.pdf images/HemeWalkthrough/PDF/")
```

By default the plots will be in the outputDirectory of the ArchRProject. The challenge is that if you requested to plot all motifs and returned this as a ggplot object it could be extremely large. An example of motif footprints ran from this function are seen below.

![](images/HemeWalkthrough/PNG/Footprints-Divide-Bias_1.png){width=250 height=400}
![](images/HemeWalkthrough/PNG/Footprints-Divide-Bias_2.png){width=250 height=400}
![](images/HemeWalkthrough/PNG/Footprints-Divide-Bias_3.png){width=250 height=400}

## Feature Footprinting

```{r eval=FALSE}
#Create Group Coverage Files if you have not done so already
projHeme5 <- addGroupCoverages(ArchRProj = projHeme5, groupBy = "Clusters2")
```

### No Normalization

```{r eval=FALSE}
#Plot Motif Footprints from positions list
#Recommend doing a few motifs not entire motif set
seTSS <- getFootprints(
  ArchRProj = projHeme5, 
  positions = GRangesList(TSS = getTSS(projHeme5)), 
  groupBy = "Clusters2",
  flank = 2000
)
```
> \## ArchR logging to : ArchRLogs/ArchR-getFootprints-10b7a47a82723-Date-2020-04-15_Time-11-17-52.log  
## If there is an issue, please report to github with logFile!   
## 2020-04-15 11:17:52 : Computing Kmer Bias Table, 0.012 mins elapsed.  
## 2020-04-15 11:18:24 : Finished Computing Kmer Tables, 0.535 mins elapsed.  
## 2020-04-15 11:18:24 : Computing Footprints, 0.547 mins elapsed.  
## 2020-04-15 11:18:36 : Computing Footprints Bias, 0.734 mins elapsed.  
## 2020-04-15 11:18:43 : Summarizing Footprints, 0.852 mins elapsed.  

```{r eval=FALSE}
plotFootprints(
  seFoot = seTSS,
  ArchRProj = projHeme5, 
  normMethod = "None",
  plotName = "TSS-No-Normalization",
  addDOC = FALSE,
  flank = 2000,
  flankNorm = 100
)
```
> \## ArchR logging to : ArchRLogs/ArchR-plotFootprints-10b7a1b7faa1-Date-2020-04-15_Time-11-18-46.log  
## If there is an issue, please report to github with logFile!  
## 2020-04-15 11:18:46 : Plotting Footprint : TSS (1 of 1), 0.007 mins elapsed.  
## Normalizing by flanking regions  
## NormMethod = None  
## ArchR logging successful to : ArchRLogs/ArchR-plotFootprints-10b7a1b7faa1-Date-2020-04-15_Time-11-18-46.log  

By default the plots will be in the outputDirectory of the ArchRProject. The challenge is that if you requested to plot all motifs and returned this as a ggplot object it could be extremely large. An example of motif footprints ran from this function are seen below.

```{r, include=FALSE, eval=FALSE}
ArchR:::.convertToPNG(ArchRProj = projHeme2)
system("cp Figures/*.png images/HemeWalkthrough/PNG/")
system("cp Figures/*.pdf images/HemeWalkthrough/PDF/")
```

<center>
![](images/HemeWalkthrough/PNG/TSS-No-Normalization_1.png){width=250 height=400}
</center>

```{r, include=FALSE, eval=FALSE}
save.image(params$out, compress = FALSE)
```

