---
title: "Chapter 14 - Integrative Analysis with ArchR"
output:
  html_document:
    theme: yeti  # many options for theme, this one is my favorite.
params:
  threads: 20
  rdata: ""
  token: ""
  ref: ""
  out: ""
---

```{r, include=FALSE, eval=FALSE}
params2 <- params
unlockBinding("params", env = .GlobalEnv)
#load("Chapter-13-Footprints.Rdata")
load(params$rdata)
params <- params2
rm(params2)
devtools::install_github("GreenleafLab/ArchR", 
  auth_token = params$token, 
  ref = params$ref,
  repos = BiocManager::repositories(),
  dependencies = FALSE
)
library(ArchR)
fn <- unclass(lsf.str(envir = asNamespace("ArchR"), all = TRUE))
fn <- fn[!grepl("\\.", fn)]
fn <- fn[fn!="ArchRProj"]
for (i in seq_along(fn)){
    tryCatch({
        eval(parse(text = paste0(fn[i], "<-ArchR::", fn[i])))
    }, error = function(x) {
    })
}
addArchRThreads(threads = params$threads)
addArchRGenome("hg19")
# fn <- unclass(lsf.str(envir = asNamespace("ArchR"), all = TRUE))
# fn <- fn[fn!="ArchRProj"]
# for (i in seq_along(fn)) {
#     tryCatch({
#         eval(parse(text = paste0(fn[i], "<-ArchR:::", fn[i])))
#     }, error = function(x) {
#     })
# }
set.seed(1)
```

# Integrative Analysis with ArchR

## CoAccessibility with ArchR


```{r eval=FALSE}
projHeme5 <- addCoAccessibility(
	ArchRProj = projHeme5,
	reducedDims = "IterativeLSI"
)
```

```{r eval=FALSE}
cA <- getCoAccessibility(
	ArchRProj = projHeme5,
	corCutOff = 0.5,
	resolution = 1,
	returnLoops = FALSE
)

cA
# DataFrame with 65736 rows and 4 columns
#       queryHits subjectHits seqnames       correlation
#       <integer>   <integer>    <Rle>         <numeric>
# 1             8          13     chr1  0.69736362363962
# 2            13           8     chr1  0.69736362363962
# 3            52          64     chr1 0.502479053390956
# 4            54          55     chr1 0.573405135422231
# 5            55          54     chr1 0.573405135422231
# ...         ...         ...      ...               ...
# 65732    158297      158296     chrX   0.5541398895206
# 65733    158332      158333     chrX 0.504941172383233
# 65734    158333      158332     chrX 0.504941172383233
# 65735    158339      158340     chrX 0.506143160385986
# 65736    158340      158339     chrX 0.506143160385985

metadata(cA)[[1]]
# GRanges object with 158378 ranges and 0 metadata columns:
#             seqnames              ranges strand
#                <Rle>           <IRanges>  <Rle>
#   Cluster10     chr1       752327-752827      *
#    Cluster9     chr1       760866-761366      *
#    Cluster2     chr1       762700-763200      *
#    Cluster2     chr1       779639-780139      *
#    Cluster8     chr1       793519-794019      *
#         ...      ...                 ...    ...
#    Cluster5     chrX 154841853-154842353      *
#   Cluster12     chrX 154842354-154842854      *
#    Cluster4     chrX 154862032-154862532      *
#    Cluster5     chrX 154912371-154912871      *
#    Cluster6     chrX 154996898-154997398      *


cA <- getCoAccessibility(
	ArchRProj = projHeme5,
	corCutOff = 0.5,
	resolution = 1,
	returnLoops = TRUE
)

# GenomicRangesList of length 1
# names(1): CoAccessibility

cA[[1]]
# GRanges object with 32868 ranges and 1 metadata column:
#           seqnames              ranges strand |             value
#              <Rle>           <IRanges>  <Rle> |         <numeric>
#       [1]     chr1       845640-856561      * |  0.69736362363962
#       [2]     chr1      968601-1004143      * | 0.502479053390956
#       [3]     chr1       974287-975100      * | 0.573405135422231
#       [4]     chr1     1034332-1034887      * | 0.807870076075987
#       [5]     chr1     1072846-1079621      * | 0.591975205782561
#       ...      ...                 ...    ... .               ...
#   [32864]     chrX 153597267-153637554      * | 0.501648373320333
#   [32865]     chrX 153637554-153659408      * | 0.534554389595235
#   [32866]     chrX 153659408-153659914      * |   0.5541398895206
#   [32867]     chrX 153959662-153960350      * | 0.504941172383233
#   [32868]     chrX 153980214-153990355      * | 0.506143160385986
```


```{r eval=FALSE}
cA <- getCoAccessibility(
	ArchRProj = projHeme5,
	corCutOff = 0.5,
	resolution = 1000,
	returnLoops = TRUE
)

cA[[1]]
# GRanges object with 30315 ranges and 1 metadata column:
#           seqnames              ranges strand |             value
#              <Rle>           <IRanges>  <Rle> |         <numeric>
#       [1]     chr1       845500-856500      * |  0.69736362363962
#       [2]     chr1      968500-1004500      * | 0.502479053390956
#       [3]     chr1       974500-975500      * | 0.573405135422231
#       [4]     chr1     1072500-1079500      * | 0.591975205782561
#       [5]     chr1     1072500-1109500      * | 0.509755959269061
#       ...      ...                 ...    ... .               ...
#   [30311]     chrX 153596500-153597500      * | 0.526200963980073
#   [30312]     chrX 153597500-153637500      * | 0.501648373320333
#   [30313]     chrX 153637500-153659500      * | 0.534554389595235
#   [30314]     chrX 153959500-153960500      * | 0.504941172383233
#   [30315]     chrX 153980500-153990500      * | 0.506143160385986
#   -------
#   seqinfo: 23 sequences from an unspecified genome; no seqlengths
```

```{r eval=FALSE}
cA <- getCoAccessibility(
	ArchRProj = projHeme5,
	corCutOff = 0.5,
	resolution = 10000,
	returnLoops = TRUE
)

cA[[1]]
# GRanges object with 15990 ranges and 1 metadata column:
#           seqnames              ranges strand |             value
#              <Rle>           <IRanges>  <Rle> |         <numeric>
#       [1]     chr1       845000-855000      * |  0.69736362363962
#       [2]     chr1      965000-1005000      * | 0.502479053390956
#       [3]     chr1     1075000-1085000      * | 0.542909007799088
#       [4]     chr1     1075000-1105000      * | 0.522706329518404
#       [5]     chr1     1105000-1115000      * | 0.607428937524039
#       ...      ...                 ...    ... .               ...
#   [15986]     chrX 153305000-153345000      * |  0.63313940661629
#   [15987]     chrX 153595000-153635000      * | 0.501648373320333
#   [15988]     chrX 153635000-153655000      * | 0.534554389595235
#   [15989]     chrX 153955000-153965000      * | 0.504941172383233
#   [15990]     chrX 153985000-153995000      * | 0.506143160385986
#   -------
#   seqinfo: 23 sequences from an unspecified genome; no seqlengths
```

Plotting browser tracks with CoAccessibility

```{r eval=FALSE}
markerGenes  <- c(
    "CD34", #Early Progenitor
    "GATA1", #Erythroid
    "PAX5", "MS4A1", #B-Cell Trajectory
    "CD14", #Monocytes
    "CD3D", "CD8A", "TBX21", "IL7R" #TCells
  )

p <- ArchRBrowserTrack(
    ArchRProj = projHeme5, 
    groupBy = "Clusters2", 
    geneSymbol = markerGenes, 
    upstream = 50000,
    downstream = 50000,
    loops = getCoAccessibility(projHeme5)
)
```

To plot a track we can simply print one from the list

```{r eval=FALSE}
grid::grid.newpage()
grid::grid.draw(p$CD14)
```

```{r, include=FALSE, eval=FALSE}
plotPDF(plotList = p, 
    name = "Plot-Tracks-Marker-Genes-with-CoAccessibility.pdf", 
    ArchRProj = projHeme5, 
    addDOC = FALSE, width = 5, height = 5)
ArchR:::.convertToPNG(ArchRProj = projHeme5)
system("cp Figures/*.png images/HemeWalkthrough/PNG/")
system("cp Figures/*.pdf images/HemeWalkthrough/PDF/")
```

![](images/HemeWalkthrough/PNG/Plot-Tracks-Marker-Genes-with-CoAccessibility_5.png){width=500 height=500}

```{r eval=FALSE}
plotPDF(plotList = p, 
    name = "Plot-Tracks-Marker-Genes-with-CoAccessibility.pdf", 
    ArchRProj = projHeme5, 
    addDOC = FALSE, width = 5, height = 5)
```

[Download PDF : Plot-Tracks-Marker-Genes.pdf](images/HemeWalkthrough/PDF/Plot-Tracks-Marker-Genes-with-CoAccessibility.pdf)


## Peak2GeneLinkage with ArchR


```{r eval=FALSE}
projHeme5 <- addPeak2GeneLinks(
	ArchRProj = projHeme5,
	reducedDims = "IterativeLSI"
)
```

```{r eval=FALSE}
p2g <- getPeak2GeneLinks(
	ArchRProj = projHeme5,
	corCutOff = 0.45,
	resolution = 1,
	returnLoops = FALSE
)

p2g
# DataFrame with 65736 rows and 4 columns
#       queryHits subjectHits seqnames       correlation
#       <integer>   <integer>    <Rle>         <numeric>
# 1             8          13     chr1  0.69736362363962
# 2            13           8     chr1  0.69736362363962
# 3            52          64     chr1 0.502479053390956
# 4            54          55     chr1 0.573405135422231
# 5            55          54     chr1 0.573405135422231
# ...         ...         ...      ...               ...
# 65732    158297      158296     chrX   0.5541398895206
# 65733    158332      158333     chrX 0.504941172383233
# 65734    158333      158332     chrX 0.504941172383233
# 65735    158339      158340     chrX 0.506143160385986
# 65736    158340      158339     chrX 0.506143160385985

metadata(p2g)[[1]]
# GRanges object with 158378 ranges and 0 metadata columns:
#             seqnames              ranges strand
#                <Rle>           <IRanges>  <Rle>
#   Cluster10     chr1       752327-752827      *
#    Cluster9     chr1       760866-761366      *
#    Cluster2     chr1       762700-763200      *
#    Cluster2     chr1       779639-780139      *
#    Cluster8     chr1       793519-794019      *
#         ...      ...                 ...    ...
#    Cluster5     chrX 154841853-154842353      *
#   Cluster12     chrX 154842354-154842854      *
#    Cluster4     chrX 154862032-154862532      *
#    Cluster5     chrX 154912371-154912871      *
#    Cluster6     chrX 154996898-154997398      *


p2g <- getPeak2GeneLinks(
	ArchRProj = projHeme5,
	corCutOff = 0.45,
	resolution = 1,
	returnLoops = TRUE
)

# GenomicRangesList of length 1
# names(1): CoAccessibility

p2g[[1]]
# GRanges object with 32868 ranges and 1 metadata column:
#           seqnames              ranges strand |             value
#              <Rle>           <IRanges>  <Rle> |         <numeric>
#       [1]     chr1       845640-856561      * |  0.69736362363962
#       [2]     chr1      968601-1004143      * | 0.502479053390956
#       [3]     chr1       974287-975100      * | 0.573405135422231
#       [4]     chr1     1034332-1034887      * | 0.807870076075987
#       [5]     chr1     1072846-1079621      * | 0.591975205782561
#       ...      ...                 ...    ... .               ...
#   [32864]     chrX 153597267-153637554      * | 0.501648373320333
#   [32865]     chrX 153637554-153659408      * | 0.534554389595235
#   [32866]     chrX 153659408-153659914      * |   0.5541398895206
#   [32867]     chrX 153959662-153960350      * | 0.504941172383233
#   [32868]     chrX 153980214-153990355      * | 0.506143160385986
```


```{r eval=FALSE}
p2g <- getPeak2GeneLinks(
	ArchRProj = projHeme5,
	corCutOff = 0.45,
	resolution = 1000,
	returnLoops = TRUE
)

p2g[[1]]
# GRanges object with 30315 ranges and 1 metadata column:
#           seqnames              ranges strand |             value
#              <Rle>           <IRanges>  <Rle> |         <numeric>
#       [1]     chr1       845500-856500      * |  0.69736362363962
#       [2]     chr1      968500-1004500      * | 0.502479053390956
#       [3]     chr1       974500-975500      * | 0.573405135422231
#       [4]     chr1     1072500-1079500      * | 0.591975205782561
#       [5]     chr1     1072500-1109500      * | 0.509755959269061
#       ...      ...                 ...    ... .               ...
#   [30311]     chrX 153596500-153597500      * | 0.526200963980073
#   [30312]     chrX 153597500-153637500      * | 0.501648373320333
#   [30313]     chrX 153637500-153659500      * | 0.534554389595235
#   [30314]     chrX 153959500-153960500      * | 0.504941172383233
#   [30315]     chrX 153980500-153990500      * | 0.506143160385986
#   -------
#   seqinfo: 23 sequences from an unspecified genome; no seqlengths
```

```{r eval=FALSE}
p2g <- getPeak2GeneLinks(
	ArchRProj = projHeme5,
	corCutOff = 0.45,
	resolution = 10000,
	returnLoops = TRUE
)

p2g[[1]]
# GRanges object with 15990 ranges and 1 metadata column:
#           seqnames              ranges strand |             value
#              <Rle>           <IRanges>  <Rle> |         <numeric>
#       [1]     chr1       845000-855000      * |  0.69736362363962
#       [2]     chr1      965000-1005000      * | 0.502479053390956
#       [3]     chr1     1075000-1085000      * | 0.542909007799088
#       [4]     chr1     1075000-1105000      * | 0.522706329518404
#       [5]     chr1     1105000-1115000      * | 0.607428937524039
#       ...      ...                 ...    ... .               ...
#   [15986]     chrX 153305000-153345000      * |  0.63313940661629
#   [15987]     chrX 153595000-153635000      * | 0.501648373320333
#   [15988]     chrX 153635000-153655000      * | 0.534554389595235
#   [15989]     chrX 153955000-153965000      * | 0.504941172383233
#   [15990]     chrX 153985000-153995000      * | 0.506143160385986
#   -------
#   seqinfo: 23 sequences from an unspecified genome; no seqlengths
```

Plotting browser tracks with Peak2Gene Links

```{r eval=FALSE}
markerGenes  <- c(
    "CD34", #Early Progenitor
    "GATA1", #Erythroid
    "PAX5", "MS4A1", #B-Cell Trajectory
    "CD14", #Monocytes
    "CD3D", "CD8A", "TBX21", "IL7R" #TCells
  )

p <- ArchRBrowserTrack(
    ArchRProj = projHeme5, 
    groupBy = "Clusters2", 
    geneSymbol = markerGenes, 
    upstream = 50000,
    downstream = 50000,
    loops = getPeak2GeneLinks(projHeme5)
)
```

To plot a track we can simply print one from the list

```{r eval=FALSE}
grid::grid.newpage()
grid::grid.draw(p$CD14)
```

```{r, include=FALSE, eval=FALSE}
plotPDF(plotList = p, 
    name = "Plot-Tracks-Marker-Genes-with-Peak2GeneLinks.pdf", 
    ArchRProj = projHeme5, 
    addDOC = FALSE, width = 5, height = 5)
ArchR:::.convertToPNG(ArchRProj = projHeme5)
system("cp Figures/*.png images/HemeWalkthrough/PNG/")
system("cp Figures/*.pdf images/HemeWalkthrough/PDF/")
```

![](images/HemeWalkthrough/PNG/Plot-Tracks-Marker-Genes-with-Peak2GeneLinks_5.png){width=500 height=500}

```{r eval=FALSE}
plotPDF(plotList = p, 
    name = "Plot-Tracks-Marker-Genes-with-Peak2GeneLinks.pdf", 
    ArchRProj = projHeme5, 
    addDOC = FALSE, width = 5, height = 5)
```

[Download PDF : Plot-Tracks-Marker-Genes.pdf](images/HemeWalkthrough/PDF/Plot-Tracks-Marker-Genes-with-Peak2GeneLinks.pdf)

Lastly we can make a peak2gene heatmap

```{r eval=FALSE}
p <- peak2GeneHeatmap(ArchRProj = projHeme5, groupBy = "Clusters2")
p
```

```{r, include=FALSE, eval=FALSE}
plotPDF(p, name = "Peak2GeneHeatmap", width = 6, height = 8, ArchRProj = projHeme5, addDOC = FALSE)
ArchR:::.convertToPNG(ArchRProj = projHeme5)
system("cp Figures/*.png images/HemeWalkthrough/PNG/")
system("cp Figures/*.pdf images/HemeWalkthrough/PDF/")
```

![](images/HemeWalkthrough/PNG/Peak2GeneHeatmap_1.png){width=600 height=800}


## Identification of Positive TF-Regulators

Step 1. Identify Deviant TF Motifs

```{r eval=FALSE}
seGroupMotif <- exportGroupSE(ArchRProj = projHeme5, useMatrix = "MotifMatrix", groupBy = "Clusters2")
seGroupMotif

#Get Deviation Z-Scores
seZ <- seGroupMotif[rowData(seGroupMotif)$seqnames=="z",]

#Max Delta
rowData(seZ)$maxDelta <- lapply(seq_len(ncol(seZ)), function(x){
  rowMaxs(assay(seZ) - assay(seZ)[,x])
}) %>% Reduce("cbind", .) %>% rowMaxs

```

Step 2. Identify Correlated TF Motifs and TF Gene Score/Expression

```{r eval=FALSE}
corGSM_MM <- correlateMatrices(
	ArchRProj = projHeme5,
	useMatrix1 = "GeneScoreMatrix",
	useMatrix2 = "MotifMatrix",
	reducedDims = "IterativeLSI"
)
corGSM_MM

corGIM_MM <- correlateMatrices(
	ArchRProj = projHeme5,
	useMatrix1 = "GeneIntegrationMatrix",
	useMatrix2 = "MotifMatrix",
	reducedDims = "IterativeLSI"
)
corGIM_MM
```

Step 3. Add Maximum Delta Deviation to the Correlation Data Frame

```{r eval=FALSE}
corGSM_MM$maxDelta <- rowData(seZ)[match(corGSM_MM$MotifMatrix_name, rowData(seZ)$name), "maxDelta"]
corGIM_MM$maxDelta <- rowData(seZ)[match(corGIM_MM$MotifMatrix_name, rowData(seZ)$name), "maxDelta"]
```

Step 4. Identify Positive TF Regulators </br> </br>

For Gene Scores 

```{r eval=FALSE}
corGSM_MM <- corGSM_MM[order(abs(corGSM_MM$cor), decreasing = TRUE), ]
corGSM_MM <- corGSM_MM[which(!duplicated(gsub("\\-.*","",corGSM_MM[,"MotifMatrix_name"]))), ]
corGSM_MM$TFRegulator <- "NO"
corGSM_MM$TFRegulator[which(corGSM_MM$cor > 0.5 & corGSM_MM$padj < 0.01 & corGSM_MM$maxDelta > quantile(corGSM_MM$maxDelta, 0.75))] <- "YES"
sort(corGSM_MM[corGSM_MM$TFRegulator=="YES",1])

p <- ggplot(data.frame(corGSM_MM), aes(cor, maxDelta, color = TFRegulator)) +
  geom_point() + 
  theme_ArchR() +
  geom_vline(xintercept = 0, lty = "dashed") + 
  scale_color_manual(values = c("NO"="darkgrey", "YES"="firebrick3")) +
  xlab("Correlation To Gene Expression") +
  ylab("Max TF Motif Delta") +
  scale_y_continuous(
  	expand = c(0,0), 
  	limits = c(0, max(corGSM_MM$maxDelta)*1.05)
  )

p
```

```{r, include=FALSE, eval=FALSE}
plotPDF(p, name = "TF-Regulators-GeneScores", width = 5, height = 5, ArchRProj = projHeme5, addDOC = FALSE)
ArchR:::.convertToPNG(ArchRProj = projHeme5)
system("cp Figures/*.png images/HemeWalkthrough/PNG/")
system("cp Figures/*.pdf images/HemeWalkthrough/PDF/")
```

![](images/HemeWalkthrough/PNG/TF-Regulators-GeneScores_1.png){width=500 height=500}

For Gene Expression 

```{r eval=FALSE}
corGIM_MM <- corGIM_MM[order(abs(corGIM_MM$cor), decreasing = TRUE), ]
corGIM_MM <- corGIM_MM[which(!duplicated(gsub("\\-.*","",corGIM_MM[,"MotifMatrix_name"]))), ]
corGIM_MM$TFRegulator <- "NO"
corGIM_MM$TFRegulator[which(corGIM_MM$cor > 0.5 & corGIM_MM$padj < 0.01 & corGIM_MM$maxDelta > quantile(corGIM_MM$maxDelta, 0.75))] <- "YES"
sort(corGIM_MM[corGIM_MM$TFRegulator=="YES",1])

p <- ggplot(data.frame(corGIM_MM), aes(cor, maxDelta, color = TFRegulator)) +
  geom_point() + 
  theme_ArchR() +
  geom_vline(xintercept = 0, lty = "dashed") + 
  scale_color_manual(values = c("NO"="darkgrey", "YES"="firebrick3")) +
  xlab("Correlation To Gene Expression") +
  ylab("Max TF Motif Delta") +
  scale_y_continuous(
  	expand = c(0,0), 
  	limits = c(0, max(corGIM_MM$maxDelta)*1.05)
  )

p
```


```{r, include=FALSE, eval=FALSE}
plotPDF(p, name = "TF-Regulators-GeneExpression", width = 5, height = 5, ArchRProj = projHeme5, addDOC = FALSE)
ArchR:::.convertToPNG(ArchRProj = projHeme5)
system("cp Figures/*.png images/HemeWalkthrough/PNG/")
system("cp Figures/*.pdf images/HemeWalkthrough/PDF/")
```

![](images/HemeWalkthrough/PNG/TF-Regulators-GeneExpression_1.png){width=500 height=500}

## Session Information

```{r eval=FALSE}
Sys.Date()


sessionInfo()
```

```{r, include=FALSE, eval=FALSE}
save.image(params$out, compress = FALSE)
```

```{r, include=FALSE, eval=FALSE}
ArchR:::.convertToPNG(ArchRProj = projHeme5)
system("cp Figures/*.png images/HemeWalkthrough/PNG/")
system("cp Figures/*.pdf images/HemeWalkthrough/PDF/")
```












