<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Getting Started With ArchR • ArchR</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.3.7/yeti/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../../pkgdown.css" rel="stylesheet">
<script src="../../pkgdown.js"></script><meta property="og:title" content="Getting Started With ArchR">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../../index.html">ArchR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.7</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../../index.html">
    <span class="fa fa-home"></span>
     
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fa fa-graduation-cap"></span>
     
    Tutorial
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../../articles/Articles/tutorial_0_intro.html">ArchR Basics</a>
    </li>
    <li>
      <a href="../../articles/Articles/tutorial_1_ArchRProject.html">1 - Setting up an ArchRProject</a>
    </li>
    <li>
      <a href="../../articles/Articles/tutorial_2_dimensionalityReduction.html">2 - Dimenstionality Reduction</a>
    </li>
    <li>
      <a href="../../articles/Articles/tutorial_3_identifyingCellTypes.html">3 - Identifying Cell Types</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fa fa-file-text-o"></span>
     
    Vignettes
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../../articles/Articles/inputFiles.html">Input File Types</a>
    </li>
    <li>
      <a href="../../articles/Articles/doubletRemoval.html">Doublet Removal</a>
    </li>
    <li>
      <a href="../../articles/Articles/qcFilters.html">QC Filters</a>
    </li>
    <li>
      <a href="../../articles/Articles/peakCalling.html">Peak Calling</a>
    </li>
    <li>
      <a href="../../articles/Articles/dimReduction.html">Dimensionality Reduction</a>
    </li>
    <li>
      <a href="../../articles/Articles/geneScores.html">Gene Activity Scores</a>
    </li>
    <li>
      <a href="../../articles/Articles/markerFeatures.html">Marker Feature Identification</a>
    </li>
    <li>
      <a href="../../articles/Articles/signalTracks.html">Signal Track Ploting</a>
    </li>
    <li>
      <a href="../../articles/Articles/footprinting.html">TF Footprinting</a>
    </li>
    <li>
      <a href="../../articles/Articles/pseudobulkGeneration.html">Pseudobulk Replicate Generation</a>
    </li>
    <li>
      <a href="../../articles/Articles/annotations.html">Gene and Genome Annotation</a>
    </li>
  </ul>
</li>
<li>
  <a href="../../articles/Articles/faq.html">
    <span class="fa fa-question-circle-o"></span>
     
    FAQ
  </a>
</li>
<li>
  <a href="../../reference/index.html">
    <span class="fa fa-code"></span>
     
    Functions
  </a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/GreenleafLab/ArchR">
    <span class="fa fa-github"></span>
     
    ArchR on GitHub
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Getting Started With ArchR</h1>
            
      
      
      <div class="hidden name"><code>tutorial.Rmd</code></div>

    </div>

    
    
<p>The following tutorial shows the basics of setting up and interacting with an <code>ArchRProject</code> using a gold-standard downsampled dataset of hematopoietic cells <a href="https://www.ncbi.nlm.nih.gov/pubmed/31792411">Granja* et al. Nature Biotechnology 2019</a>. This tutorial and all of the accompanying vignettes assume that you are running ArchR <strong>locally</strong>. <strong>QQQ</strong> Once all of these steps work for you, feel free to <a href="articles/Articles/clusterComputing.html">set up ArchR to work in a cluster environment</a>. This tutorial does not explain every detail of every step but rather shows you how to perform the most common analyses. Please see the <a href="articles/index.html">Vignettes section</a> for more details on each major analytical step and all of the major features of ArchR.</p>
<div id="what-is-an-arrowfile-archrproject" class="section level1">
<h1 class="hasAnchor">
<a href="#what-is-an-arrowfile-archrproject" class="anchor"></a>1. What is an <code>ArrowFile</code> / <code>ArchRProject</code>?</h1>
<p>The base unit of an analytical project in ArchR is called an <code>ArrowFile</code>. Each <code>ArrowFile</code> stores all of the data associated with an individual sample (i.e. metadata, accessible fragments, and data matrices). Here, an “individual sample” would be the most detailed unit of analysis desired (for ex. a single replicate of a particular condition). During creation and as additional analyses are performed, ArchR updates and edits each <code>ArrowFile</code> to contain additional layers of information. It is worth noting that, to ArchR, an <code>ArrowFile</code> is actually just a path to an external file stored on disk. More explicitly, an <code>ArrowFile</code> is not an R-language object that is stored in memory. Because of this, we use an <code>ArchRProject</code> object to associate these <code>ArrowFiles</code> together into a single analytical framework that can be rapidly accessed in R. This <code>ArchRProject</code> object is small in size and is stored in memory.</p>
<center>
<img src="../../images/ArchRProject_Schematic.png" width="700">
</center>
<p>Certain actions can be taken directly on <code>ArrowFiles</code> while other actions are taken on an <code>ArchRProject</code> which in turn updates each associated <code>ArrowFile</code>. Because <code>ArrowFiles</code> are stored as large HDF5-format files, “get-er” functions in ArchR retrieve data by interacting with the <code>ArchRProject</code> while “add-er” functions either (i) add data directly to <code>ArrowFiles</code>, (ii) add data directly to an <code>ArchRProject</code>, or (iii) add data to <code>ArrowFiles</code> by interacting with an <code>ArchRProject</code>.</p>
<center>
<img src="../../images/ArchR_FunctionSchematic.png" width="400">
</center>
</div>
<div id="getting-set-up" class="section level1">
<h1 class="hasAnchor">
<a href="#getting-set-up" class="anchor"></a>2. Getting Set Up</h1>
<p>The first thing we do is set up our working directory, load our gene and genome annotations, and set the number of threads we would like to use. Depending on the configuration of your local environment, you may need to modify the number of <code>threads</code> used below in <code><a href="../../reference/addArchRThreads.html">addArchRThreads()</a></code>. By default ArchR uses half of the total number of <code>threads</code> available but you can adjust this manually as you see fit. If you are using windows, the usable <code>threads</code> will automatically be set to 1 because the parallel processing in ArchR is build for Unix-based operating systems.</p>
<p>For the purposes of this tutorial, we provide the gene and genome annotations but you can create your own using the <code>createGeneAnnotation()</code> and <code><a href="../../reference/createGenomeAnnotation.html">createGenomeAnnotation()</a></code> functions. See the <a href="articles/Articles/annotations.html">Gene and Genome Annotations vignette</a> for more information.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#Load R Libraries</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(ArchR)

<span class="co">#Create a new folder and set this as the working directory for tutorial analyses</span>
wd &lt;-<span class="st"> "ArchR_tutorial"</span>
<span class="kw"><a href="https://rdrr.io/r/base/files2.html">dir.create</a></span>(wd, <span class="dt">showWarnings =</span> <span class="ot">FALSE</span>, <span class="dt">recursive =</span> <span class="ot">TRUE</span>)
<span class="kw"><a href="https://rdrr.io/r/base/getwd.html">setwd</a></span>(wd)

<span class="co">#Load genome annotations. Available annotations are for "Hg19", "Hg38", "Mm9", or "Mm10". QQQ THIS DOESNT MAKE SENSE. THE DATA IS HUMAN, NOT MOUSE, AND IS ONLY AVAILABLE FOR ONE GENOME FOR THE TUTORIAL? WOULD REMOVE.</span>
<span class="kw"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="st">"geneAnnoHg19"</span>)
<span class="kw"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="st">"genomeAnnoHg19"</span>)
geneAnno &lt;-<span class="st"> </span>geneAnnoHg19
genomeAnno &lt;-<span class="st"> </span>genomeAnnoHg19

<span class="co">#Set Default Threads for ArchR Functions</span>
<span class="kw"><a href="../../reference/addArchRThreads.html">addArchRThreads</a></span>(<span class="dt">threads =</span> <span class="kw"><a href="https://rdrr.io/r/base/Round.html">floor</a></span>(parallel::<span class="kw">detectCores</span>()/<span class="dv">2</span>)) <span class="co">#QQQ I favor including default values to make it clear what happens. Open to suggestions though.</span></code></pre></div>
</div>
<div id="creating-arrow-files" class="section level1 tabset tabset-fade tabset-pills">
<h1 class="hasAnchor">
<a href="#creating-arrow-files" class="anchor"></a>3. Creating Arrow Files</h1>
<p>For this tutorial, we will download a collection of fragment files. Fragment files are one of the base file types of the 10x Genomics analytical platform (and other platforms) and can be easily created from any BAM file. See <a href="articles/Articles/inputFiles.html">the ArchR input file types vignette</a> for information on making your own fragment files for input to ArchR. Once we have our fragment files, we provide their paths as a character vector to <code><a href="../../reference/createArrowFiles.html">createArrowFiles()</a></code>. During creation, some basic metadata and matrices are added to each <code>ArrowFile</code> including a “TileMatrix” containing insertion counts across genome-wide 500-bp bins (see <code><a href="../../reference/addTileMatrix.html">addTileMatrix()</a></code>) and a “GeneScoreMatrix” that is determined based on weighting insertion counts in tiles nearby a gene promoter (see <code><a href="../../reference/addGeneScoreMatrix.html">addGeneScoreMatrix()</a></code>). These gene activity scores are described in more depth in the <a href="articles/Articles/geneScores.html">Gene Activity Score vignette</a>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#Get Tutorial Data ~2.2GB To Download (if downloaded already ArchR will bypass downloading).</span>
inputFiles &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/getTutorialData.html">getTutorialData</a></span>(<span class="st">"Hematopoiesis"</span>)

<span class="co">#Create Arrow Files (~10-15 minutes) w/ helpful messages displaying progress.</span>
<span class="co">#For each sample, this step will:</span>
<span class="co"># 1. Read accessible fragments.</span>
<span class="co"># 2. Calculate QC Information for each cell (TSS Enrichment, Nucleosome info).</span>
<span class="co"># 3. Filter cells based on QC parameters.</span>
<span class="co"># 4. Create a genome-wide TileMatrix using 500-bp bins.</span>
<span class="co"># 5. Create a GeneScoreMatrix using the provided geneAnnotation.</span>
ArrowFiles &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/createArrowFiles.html">createArrowFiles</a></span>(
  <span class="dt">inputFiles =</span> inputFiles,
  <span class="dt">sampleNames =</span> <span class="kw"><a href="https://rdrr.io/r/base/names.html">names</a></span>(inputFiles),
  <span class="dt">geneAnnotation =</span> geneAnno,
  <span class="dt">genomeAnnotation =</span> genomeAnno,
  <span class="dt">filterTSS =</span> <span class="dv">4</span>,
  <span class="dt">filterFrags =</span> <span class="dv">1000</span>,
  <span class="dt">addTileMat =</span> <span class="ot">TRUE</span>,
  <span class="dt">addGeneScoreMat =</span> <span class="ot">TRUE</span>
  )</code></pre></div>
<hr>
<div id="qc-tss-scores-by-unique-fragments" class="section level2">
<h2 class="hasAnchor">
<a href="#qc-tss-scores-by-unique-fragments" class="anchor"></a>QC TSS Scores by Unique Fragments</h2>
<p>Since this was plotted prior to creation of an ArchRProject we go to QualityControl/scATAC_BMMC_R2/scATAC_BMMC_R2-TSS_by_Unique_Frags.pdf for the plot below.</p>
<div class="figure">
<img src="../../images/Tutorial2/scATAC_BMMC_R2-TSS_by_Unique_Frags_1.png" width="600" height="600">
</div>
</div>
<div id="qc-fragment-size-distribution" class="section level2">
<h2 class="hasAnchor">
<a href="#qc-fragment-size-distribution" class="anchor"></a>QC Fragment Size Distribution</h2>
<p>Since this was plotted prior to creation of an ArchRProject we go to QualityControl/scATAC_BMMC_R2/scATAC_BMMC_R2-Fragment_Size_Distribution.pdf for the plot below.</p>
<div class="figure">
<img src="../../images/Tutorial2/scATAC_BMMC_R2-Fragment_Size_Distribution_1.png" width="600" height="400">
</div>
</div>
</div>
<div id="tidying-up-our-data-and-creating-an-archrproject" class="section level1">
<h1 class="hasAnchor">
<a href="#tidying-up-our-data-and-creating-an-archrproject" class="anchor"></a>4. Tidying up our data and creating an <code class="tabset tabset-fade tabset-pills">ArchRProject</code>
</h1>
<p>One major source of trouble in single-cell data is the contribution of “doublets” to the analysis. A doublet refers to a single droplet that received a single barcoded bead and more than one nucleus. This causes the reads from more than one cell to appear as a single cell. We remove these computationally and describe this doublet removal process in more depth in the <a href="articles/Articles/doubletRemoval.html">doublet removal vignette</a>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#Add Infered Doublet Scores to each Arrow File (~2-5 min per sample)</span>
doubScores &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/addDoubletScores.html">addDoubletScores</a></span>(ArrowFiles)

<span class="co">#Create ArchRProject</span>
<span class="co">#The outputDirectory here describes where all downstream analyses and plots go.</span>
proj &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/ArchRProject.html">ArchRProject</a></span>(
  <span class="dt">ArrowFiles =</span> ArrowFiles, 
  <span class="dt">geneAnnotation =</span> geneAnno,
  <span class="dt">genomeAnnotation =</span> genomeAnno,
  <span class="dt">outputDirectory =</span> <span class="st">"Heme_Tutorial2"</span>
)

<span class="co">#To see information about ArchR Project created simply input</span>
proj

<span class="co">#We can now visualize numeric metadata per grouping with a violin plot now that we have created an ArchR Project.</span>
<span class="co">#For example we will plot TSS / Doublet Enrichment Scores per sample.</span>
plotList &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/list.html">list</a></span>()
plotList[[<span class="dv">1</span>]] &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/plotGroups.html">plotGroups</a></span>(<span class="dt">ArchRProj =</span> proj, 
  <span class="dt">groupBy =</span> <span class="st">"Sample"</span>, 
  <span class="dt">colorBy =</span> <span class="st">"colData"</span>, 
  <span class="dt">name =</span> <span class="st">"TSSEnrichment"</span>,
)
plotList[[<span class="dv">2</span>]] &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/plotGroups.html">plotGroups</a></span>(<span class="dt">ArchRProj =</span> proj, 
  <span class="dt">groupBy =</span> <span class="st">"Sample"</span>, 
  <span class="dt">colorBy =</span> <span class="st">"colData"</span>, 
  <span class="dt">name =</span> <span class="st">"DoubletEnrichment"</span>,
)
<span class="kw"><a href="../../reference/plotPDF.html">plotPDF</a></span>(<span class="dt">plotList =</span> plotList, <span class="dt">name =</span> <span class="st">"TSS-Doublet-Enrichment"</span>, <span class="dt">width =</span> <span class="dv">4</span>, <span class="dt">height =</span> <span class="dv">4</span>,  <span class="dt">ArchRProj =</span> proj, <span class="dt">addDOC =</span> <span class="ot">FALSE</span>)

<span class="co">#Filter Doublets</span>
<span class="co">#The automatic filtering rate will be based on how many cells are in the sample, if there</span>
<span class="co">#are 5,000 cells ArchR will remove up to 250 (~5%) of the cells. If you believe more cells</span>
<span class="co">#should be excluded change the filterRatio argument apropriately.</span>
proj &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/filterDoublets.html">filterDoublets</a></span>(proj)</code></pre></div>
<hr>
<div id="example-doublet-results" class="section level2 tabset tabset-fade tabset-pills">
<h2 class="hasAnchor">
<a href="#example-doublet-results" class="anchor"></a>Example Doublet Results</h2>
<p>Since this was plotted prior to creation of an ArchRProject we go to QualityControl/scATAC_BMMC_R2/scATAC_BMMC_R2-Doublet-Summary.pdf for the plot below.</p>
<div id="section" class="section level3">
<h3 class="hasAnchor">
<a href="#section" class="anchor"></a>1</h3>
<div class="figure">
<img src="../../images/Tutorial2/scATAC_BMMC_R2-Doublet-Summary_1.png" width="600" height="600">
</div>
</div>
<div id="section-1" class="section level3">
<h3 class="hasAnchor">
<a href="#section-1" class="anchor"></a>2</h3>
<div class="figure">
<img src="../../images/Tutorial2/scATAC_BMMC_R2-Doublet-Summary_2.png" width="600" height="600">
</div>
</div>
<div id="section-2" class="section level3">
<h3 class="hasAnchor">
<a href="#section-2" class="anchor"></a>3</h3>
<div class="figure">
<img src="../../images/Tutorial2/scATAC_BMMC_R2-Doublet-Summary_3.png" width="600" height="600">
</div>
</div>
</div>
<div id="tss-doublet-enrichment" class="section level2 tabset tabset-fade tabset-pills">
<h2 class="hasAnchor">
<a href="#tss-doublet-enrichment" class="anchor"></a>TSS / Doublet Enrichment</h2>
<div id="section-3" class="section level3">
<h3 class="hasAnchor">
<a href="#section-3" class="anchor"></a>1</h3>
<div class="figure">
<img src="../../images/Tutorial2/TSS-Doublet-Enrichment_1.png" width="600" height="600">
</div>
</div>
<div id="section-4" class="section level3">
<h3 class="hasAnchor">
<a href="#section-4" class="anchor"></a>2</h3>
<div class="figure">
<img src="../../images/Tutorial2/TSS-Doublet-Enrichment_2.png" width="600" height="600">
</div>
</div>
</div>
</div>
<div id="dimensionality-reduction" class="section level1 tabset tabset-fade tabset-pills">
<h1 class="hasAnchor">
<a href="#dimensionality-reduction" class="anchor"></a>5. Dimensionality Reduction</h1>
<p>At this point, we have an ArchR project that is ready to be used in downstream visualizations and analyses. The first thing we will do is use an iterative latent semantic indexing (LSI) approach to define clusters in our data. Once we have identified clusters in our data, we can plot a UMAP embedding. For more details, see the <a href="articles/Articles/dimReduction.html">dimensionality reduction vignette</a>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#Reduce Dimensions with Iterative LSI (~5-10 minutes)</span>
proj &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/addIterativeLSI.html">addIterativeLSI</a></span>(
  <span class="dt">ArchRProj =</span> proj, 
  <span class="dt">useMatrix =</span> <span class="st">"TileMatrix"</span>, 
  <span class="dt">reducedDimsOut =</span> <span class="st">"IterativeLSI"</span>
)

<span class="co">#Identify Clusters from Iterative LSI</span>
<span class="co">#By default ArchR uses Seurat's FindClusters function because it is fast/robust producing reasonable clustering.</span>
<span class="co">#The larger the resolution the more clusters will be called. The lower the resolution hte less clusters will be called.</span>
<span class="co">#It is recommended to compare the results from your clusters and your embeddings and find params that best agree across</span>
<span class="co">#both analyses for clarity.</span>
proj &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/addClusters.html">addClusters</a></span>(<span class="dt">input =</span> proj, <span class="dt">reducedDims =</span> <span class="st">"IterativeLSI"</span>, <span class="dt">resolution =</span> <span class="fl">0.6</span>)

<span class="co">#Compute a UMAP embedding to visualize our tiled accessibility matrix in a 2-d setting.</span>
proj &lt;-<span class="st"> </span><span class="kw">addEmbedding</span>(
  <span class="dt">ArchRProj =</span> proj, 
  <span class="dt">reducedDims =</span> <span class="st">"IterativeLSI"</span>, 
  <span class="dt">embedding =</span> <span class="st">"UMAP"</span>, 
  <span class="dt">embeddingParams =</span> <span class="kw"><a href="https://rdrr.io/r/base/list.html">list</a></span>(<span class="dt">min_dist =</span> <span class="fl">0.4</span>) <span class="co">#see uwot::umap for alternative params</span>
)

<span class="co">#Plot the UMAP Embedding with Metadata Overlayed such as Experimental Sample and Clusters.</span>
<span class="co">#To change plotting aesthetics see ?plotEmbedding parameters.</span>
plotList &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/list.html">list</a></span>()
plotList[[<span class="dv">1</span>]] &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/plotEmbedding.html">plotEmbedding</a></span>(<span class="dt">ArchRProj =</span> proj, <span class="dt">colorBy =</span> <span class="st">"colData"</span>, <span class="dt">name =</span> <span class="st">"Sample"</span>)
plotList[[<span class="dv">2</span>]] &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/plotEmbedding.html">plotEmbedding</a></span>(<span class="dt">ArchRProj =</span> proj, <span class="dt">colorBy =</span> <span class="st">"colData"</span>, <span class="dt">name =</span> <span class="st">"Clusters"</span>, <span class="dt">plotParams =</span> <span class="kw"><a href="https://rdrr.io/r/base/list.html">list</a></span>(<span class="dt">labelMeans=</span><span class="ot">TRUE</span>))
<span class="co">#Different ArchRPalette "circus"</span>
plotList[[<span class="dv">3</span>]] &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/plotEmbedding.html">plotEmbedding</a></span>(<span class="dt">ArchRProj =</span> proj, <span class="dt">colorBy =</span> <span class="st">"colData"</span>, <span class="dt">name =</span> <span class="st">"Clusters"</span>, <span class="dt">discreteSet =</span> <span class="st">"circus"</span>, <span class="dt">plotParams =</span> <span class="kw"><a href="https://rdrr.io/r/base/list.html">list</a></span>(<span class="dt">labelMeans=</span><span class="ot">TRUE</span>))
<span class="kw"><a href="../../reference/plotPDF.html">plotPDF</a></span>(<span class="dt">plotList =</span> plotList, <span class="dt">name =</span> <span class="st">"UMAP-Samples-Clusters"</span>, <span class="dt">width =</span> <span class="dv">6</span>, <span class="dt">height =</span> <span class="dv">6</span>, <span class="dt">ArchRProj =</span> proj, <span class="dt">addDOC =</span> <span class="ot">FALSE</span>)</code></pre></div>
<hr>
<div id="umap-w-clusters" class="section level2 tabset tabset-fade tabset-pills">
<h2 class="hasAnchor">
<a href="#umap-w-clusters" class="anchor"></a>UMAP w/ Clusters</h2>
<p>This <a href="../../images/Tutorial/UMAP-Samples-Clusters.pdf">plot</a> shows gene experimental samples and clusters described above overlayed onto the UMAP embedding. (Note if you see a blank space below try firefox or safari)</p>
<div id="section-5" class="section level3">
<h3 class="hasAnchor">
<a href="#section-5" class="anchor"></a>1</h3>
<div class="figure">
<img src="../../images/Tutorial2/UMAP-Samples-Clusters_1.png" width="600" height="600">
</div>
</div>
<div id="section-6" class="section level3">
<h3 class="hasAnchor">
<a href="#section-6" class="anchor"></a>2</h3>
<div class="figure">
<img src="../../images/Tutorial2/UMAP-Samples-Clusters_2.png" width="600" height="600">
</div>
</div>
<div id="section-7" class="section level3">
<h3 class="hasAnchor">
<a href="#section-7" class="anchor"></a>3</h3>
<div class="figure">
<img src="../../images/Tutorial2/UMAP-Samples-Clusters_3.png" width="600" height="600">
</div>
</div>
</div>
<div id="tsne-w-clusters" class="section level2 tabset tabset-fade tabset-pills">
<h2 class="hasAnchor">
<a href="#tsne-w-clusters" class="anchor"></a>TSNE w/ Clusters</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#Compute a TSNE embedding to visualize our tiled accessibility matrix in a 2-d setting.</span>
proj &lt;-<span class="st"> </span><span class="kw">addEmbedding</span>(
  <span class="dt">ArchRProj =</span> proj, 
  <span class="dt">reducedDims =</span> <span class="st">"IterativeLSI"</span>, 
  <span class="dt">embedding =</span> <span class="st">"Rtsne"</span>, 
  <span class="dt">embeddingParams =</span> <span class="kw"><a href="https://rdrr.io/r/base/list.html">list</a></span>(<span class="dt">perplexity =</span> <span class="dv">50</span>)
  )

<span class="co">#Plot the TSNE Embedding with Metadata Overlayed such as Experimental Sample and Clusters.</span>
<span class="co">#To change plotting aesthetics see ?plotEmbedding parameters.</span>
plotList &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/list.html">list</a></span>()
plotList[[<span class="dv">1</span>]] &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/plotEmbedding.html">plotEmbedding</a></span>(<span class="dt">ArchRProj =</span> proj, <span class="dt">colorBy =</span> <span class="st">"colData"</span>, <span class="dt">embedding =</span> <span class="st">"Rtsne"</span>, <span class="dt">name =</span> <span class="st">"Sample"</span>)
plotList[[<span class="dv">2</span>]] &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/plotEmbedding.html">plotEmbedding</a></span>(<span class="dt">ArchRProj =</span> proj, <span class="dt">colorBy =</span> <span class="st">"colData"</span>, <span class="dt">embedding =</span> <span class="st">"Rtsne"</span>, <span class="dt">name =</span> <span class="st">"Clusters"</span>, <span class="dt">plotParams =</span> <span class="kw"><a href="https://rdrr.io/r/base/list.html">list</a></span>(<span class="dt">labelMeans=</span><span class="ot">TRUE</span>))
<span class="co">#Different ArchRPalette "circus"</span>
plotList[[<span class="dv">3</span>]] &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/plotEmbedding.html">plotEmbedding</a></span>(<span class="dt">ArchRProj =</span> proj, <span class="dt">colorBy =</span> <span class="st">"colData"</span>, <span class="dt">embedding =</span> <span class="st">"Rtsne"</span>, <span class="dt">name =</span> <span class="st">"Clusters"</span>, <span class="dt">discreteSet =</span> <span class="st">"circus"</span>, <span class="dt">plotParams =</span> <span class="kw"><a href="https://rdrr.io/r/base/list.html">list</a></span>(<span class="dt">labelMeans=</span><span class="ot">TRUE</span>))
<span class="kw"><a href="../../reference/plotPDF.html">plotPDF</a></span>(<span class="dt">plotList =</span> plotList, <span class="dt">name =</span> <span class="st">"TSNE-Samples-Clusters"</span>, <span class="dt">width =</span> <span class="dv">6</span>, <span class="dt">height =</span> <span class="dv">6</span>, <span class="dt">ArchRProj =</span> proj, <span class="dt">addDOC =</span> <span class="ot">FALSE</span>)</code></pre></div>
<div id="section-8" class="section level3">
<h3 class="hasAnchor">
<a href="#section-8" class="anchor"></a>1</h3>
<div class="figure">
<img src="../../images/Tutorial2/TSNE-Samples-Clusters_1.png" width="600" height="600">
</div>
</div>
<div id="section-9" class="section level3">
<h3 class="hasAnchor">
<a href="#section-9" class="anchor"></a>2</h3>
<div class="figure">
<img src="../../images/Tutorial2/TSNE-Samples-Clusters_2.png" width="600" height="600">
</div>
</div>
<div id="section-10" class="section level3">
<h3 class="hasAnchor">
<a href="#section-10" class="anchor"></a>3</h3>
<div class="figure">
<img src="../../images/Tutorial2/TSNE-Samples-Clusters_3.png" width="600" height="600">
</div>
</div>
</div>
<div id="umap-w-custom-coldata" class="section level2 tabset tabset-fade tabset-pills">
<h2 class="hasAnchor">
<a href="#umap-w-custom-coldata" class="anchor"></a>UMAP w/ Custom ColData</h2>
<p>To add your own information for plotting ontop of UMAP embedding we will show an example here.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#Get Cell Names</span>
cellNames &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/getCellNames.html">getCellNames</a></span>(<span class="dt">ArchRProj =</span> proj)

<span class="co">#Get Sample names</span>
sampleNames &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/getCellColData.html">getCellColData</a></span>(<span class="dt">ArchRProj =</span> proj, <span class="dt">select =</span> <span class="st">"Sample"</span>, <span class="dt">drop =</span> <span class="ot">TRUE</span>)

<span class="co">#Remove Replicate Name</span>
sampleNames &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span>(<span class="st">"_R1"</span>, <span class="st">""</span>, <span class="kw"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span>(<span class="st">"_R2"</span>, <span class="st">""</span>, sampleNames))

<span class="co">#Add Info to cellColData</span>
proj &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/addCellColData.html">addCellColData</a></span>(<span class="dt">ArchRProj =</span> proj, <span class="dt">data =</span> sampleNames, <span class="dt">cells =</span> cellNames, <span class="dt">name =</span> <span class="st">"Type"</span>)

<span class="co">#Plot the UMAP Embedding with Metadata Overlayed such as Experimental Sample and Clusters.</span>
<span class="co">#To change plotting aesthetics see ?plotEmbedding parameters.</span>
plotList &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/list.html">list</a></span>()
plotList[[<span class="dv">1</span>]] &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/plotEmbedding.html">plotEmbedding</a></span>(<span class="dt">ArchRProj =</span> proj, <span class="dt">colorBy =</span> <span class="st">"colData"</span>, <span class="dt">name =</span> <span class="st">"Type"</span>, <span class="dt">plotParams =</span> <span class="kw"><a href="https://rdrr.io/r/base/list.html">list</a></span>(<span class="dt">labelMeans=</span><span class="ot">TRUE</span>))
plotList[[<span class="dv">2</span>]] &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/plotEmbedding.html">plotEmbedding</a></span>(<span class="dt">ArchRProj =</span> proj, <span class="dt">colorBy =</span> <span class="st">"colData"</span>, <span class="dt">name =</span> <span class="st">"Type"</span>, <span class="dt">discreteSet =</span> <span class="st">"circus"</span>, <span class="dt">plotParams =</span> <span class="kw"><a href="https://rdrr.io/r/base/list.html">list</a></span>(<span class="dt">labelMeans=</span><span class="ot">TRUE</span>))
<span class="kw"><a href="../../reference/plotPDF.html">plotPDF</a></span>(<span class="dt">plotList =</span> plotList, <span class="dt">name =</span> <span class="st">"UMAP-Samples-Type"</span>, <span class="dt">width =</span> <span class="dv">6</span>, <span class="dt">height =</span> <span class="dv">6</span>, <span class="dt">ArchRProj =</span> proj, <span class="dt">addDOC =</span> <span class="ot">FALSE</span>)</code></pre></div>
<div id="section-11" class="section level3">
<h3 class="hasAnchor">
<a href="#section-11" class="anchor"></a>1</h3>
<div class="figure">
<img src="../../images/Tutorial2/UMAP-Samples-Type_1.png" width="600" height="600">
</div>
</div>
<div id="section-12" class="section level3">
<h3 class="hasAnchor">
<a href="#section-12" class="anchor"></a>2</h3>
<div class="figure">
<img src="../../images/Tutorial2/UMAP-Samples-Type_2.png" width="600" height="600">
</div>
</div>
</div>
</div>
<div id="identifying-cluster-cell-types-using-marker-genes" class="section level1 tabset tabset-fade tabset-pills">
<h1 class="hasAnchor">
<a href="#identifying-cluster-cell-types-using-marker-genes" class="anchor"></a>6. Identifying Cluster Cell Types Using Marker Genes</h1>
<p>In order to understand which clusters correspond to which cell types, we use a supervised approach based on prior knowledge of the genes that are active in specific cell types. We determine <em>gene activity scores</em> for each putative marker gene based on chromatin accessibility signal in the region surrounding the gene’s promoter. We can then overlay these <em>gene activity scores</em> on our UMAP embedding to visualize the relationship between gene activity and cluster. For more details, see the <a href="articles/Articles/geneScores.html">marker genes vignette</a>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#Add Imputation Weights for imputing numerical values based on Magic (see van Dijk et. al. 2018).</span>
proj &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/addImputeWeights.html">addImputeWeights</a></span>(<span class="dt">ArchRProj =</span> proj)

markerGenes  &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(
    <span class="st">"CD34"</span>, <span class="co">#Early Progenitor</span>
    <span class="st">"GATA1"</span>, <span class="co">#Erythroid</span>
    <span class="st">"PAX5"</span>, <span class="st">"MS4A1"</span>, <span class="co">#B-Cell Trajectory</span>
    <span class="st">"CD14"</span>, <span class="co">#Monocytes</span>
    <span class="st">"CD3D"</span>, <span class="st">"CD8A"</span>, <span class="st">"TBX21"</span>, <span class="st">"IL7R"</span> <span class="co">#TCells</span>
  )

<span class="co">#Plot the UMAP Embedding with Marker Genes Overlayed w/ Imputation</span>
plotList &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/list.html">list</a></span>()
plotList[[<span class="dv">1</span>]] &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/plotEmbedding.html">plotEmbedding</a></span>(<span class="dt">ArchRProj =</span> proj, <span class="dt">colorBy =</span> <span class="st">"GeneScoreMatrix"</span>, <span class="dt">name =</span> markerGenes, <span class="dt">imputeWeights =</span> <span class="kw"><a href="../../reference/getImputeWeights.html">getImputeWeights</a></span>(proj))
<span class="kw"><a href="../../reference/plotPDF.html">plotPDF</a></span>(<span class="dt">plotList =</span> plotList, <span class="dt">name =</span> <span class="st">"UMAP-Marker-Gene-Scores-w-Imputation"</span>, <span class="dt">width =</span> <span class="dv">6</span>, <span class="dt">height =</span> <span class="dv">6</span>, <span class="dt">ArchRProj =</span> proj, <span class="dt">addDOC =</span> <span class="ot">FALSE</span>)

<span class="co">#Plot the UMAP Embedding with Marker Genes Overlayed w/o Imputation</span>
plotList &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/list.html">list</a></span>()
plotList[[<span class="dv">1</span>]] &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/plotEmbedding.html">plotEmbedding</a></span>(<span class="dt">ArchRProj =</span> proj, <span class="dt">colorBy =</span> <span class="st">"GeneScoreMatrix"</span>, <span class="dt">name =</span> markerGenes)
<span class="kw"><a href="../../reference/plotPDF.html">plotPDF</a></span>(<span class="dt">plotList =</span> plotList, <span class="dt">name =</span> <span class="st">"UMAP-Marker-Gene-Scores-wo-Imputation"</span>, <span class="dt">width =</span> <span class="dv">6</span>, <span class="dt">height =</span> <span class="dv">6</span>, <span class="dt">ArchRProj =</span> proj, <span class="dt">addDOC =</span> <span class="ot">FALSE</span>)

<span class="co">#Plot Tracks at Marker Genes</span>
plotTracks &lt;-<span class="st"> </span><span class="kw">ArchRRegionTrack</span>(<span class="dt">ArchRProj =</span> proj, <span class="dt">geneSymbol =</span> markerGenes)
<span class="kw"><a href="../../reference/plotPDF.html">plotPDF</a></span>(<span class="dt">plotList =</span> plotTracks, <span class="dt">name =</span> <span class="st">"Plot-Tracks-Marker-Genes"</span>, <span class="dt">width =</span> <span class="dv">6</span>, <span class="dt">height =</span> <span class="dv">8</span>, <span class="dt">ArchRProj =</span> proj, <span class="dt">addDOC =</span> <span class="ot">FALSE</span>)

<span class="co">#Identify Marker Gene through Pairwise Test vs Bias-Matched Background </span>
markersGS &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/markerFeatures.html">markerFeatures</a></span>(<span class="dt">ArchRProj =</span> proj, <span class="dt">useMatrix =</span> <span class="st">"GeneScoreMatrix"</span>, <span class="dt">groupBy =</span> <span class="st">"Clusters"</span>)
heatmapGS &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/markerHeatmap.html">markerHeatmap</a></span>(
  <span class="dt">seMarker =</span> markersGS, 
  <span class="dt">cutOff =</span> <span class="st">"FDR &lt;= 0.01 &amp; Log2FC &gt;= 1"</span>, 
  <span class="dt">labelMarkers =</span> markerGenes
)
<span class="kw"><a href="../../reference/plotPDF.html">plotPDF</a></span>(heatmapGS, <span class="dt">name =</span> <span class="st">"GeneScores-Marker-Heatmap"</span>, <span class="dt">width =</span> <span class="dv">8</span>, <span class="dt">height =</span> <span class="dv">12</span>, <span class="dt">ArchRProj =</span> proj, <span class="dt">addDOC =</span> <span class="ot">FALSE</span>)</code></pre></div>
<hr>
<div id="umap-genescores-w-imputation" class="section level2 tabset tabset-fade tabset-pills">
<h2 class="hasAnchor">
<a href="#umap-genescores-w-imputation" class="anchor"></a>UMAP GeneScores w/ Imputation</h2>
<p>This <a href="../../images/Tutorial/UMAP-Marker-Gene-Scores-w-Imputation.pdf">plot</a> shows gene activity scores of the marker genes described above overlayed onto the UMAP embedding. The gene scores are imputed using Magic when running addImupteWeights. (Note if you see a blank space below try firefox or safari)</p>
<div id="section-13" class="section level3">
<h3 class="hasAnchor">
<a href="#section-13" class="anchor"></a>1</h3>
<div class="figure">
<img src="../../images/Tutorial2/UMAP-Marker-Gene-Scores-w-Imputation_1.png" width="600" height="600">
</div>
</div>
<div id="section-14" class="section level3">
<h3 class="hasAnchor">
<a href="#section-14" class="anchor"></a>2</h3>
<div class="figure">
<img src="../../images/Tutorial2/UMAP-Marker-Gene-Scores-w-Imputation_2.png" width="600" height="600">
</div>
</div>
<div id="section-15" class="section level3">
<h3 class="hasAnchor">
<a href="#section-15" class="anchor"></a>3</h3>
<div class="figure">
<img src="../../images/Tutorial2/UMAP-Marker-Gene-Scores-w-Imputation_3.png" width="600" height="600">
</div>
</div>
<div id="section-16" class="section level3">
<h3 class="hasAnchor">
<a href="#section-16" class="anchor"></a>4</h3>
<div class="figure">
<img src="../../images/Tutorial2/UMAP-Marker-Gene-Scores-w-Imputation_4.png" width="600" height="600">
</div>
</div>
<div id="section-17" class="section level3">
<h3 class="hasAnchor">
<a href="#section-17" class="anchor"></a>5</h3>
<div class="figure">
<img src="../../images/Tutorial2/UMAP-Marker-Gene-Scores-w-Imputation_5.png" width="600" height="600">
</div>
</div>
<div id="section-18" class="section level3">
<h3 class="hasAnchor">
<a href="#section-18" class="anchor"></a>6</h3>
<div class="figure">
<img src="../../images/Tutorial2/UMAP-Marker-Gene-Scores-w-Imputation_6.png" width="600" height="600">
</div>
</div>
<div id="section-19" class="section level3">
<h3 class="hasAnchor">
<a href="#section-19" class="anchor"></a>7</h3>
<div class="figure">
<img src="../../images/Tutorial2/UMAP-Marker-Gene-Scores-w-Imputation_7.png" width="600" height="600">
</div>
</div>
<div id="section-20" class="section level3">
<h3 class="hasAnchor">
<a href="#section-20" class="anchor"></a>8</h3>
<div class="figure">
<img src="../../images/Tutorial2/UMAP-Marker-Gene-Scores-w-Imputation_8.png" width="600" height="600">
</div>
</div>
<div id="section-21" class="section level3">
<h3 class="hasAnchor">
<a href="#section-21" class="anchor"></a>9</h3>
<div class="figure">
<img src="../../images/Tutorial2/UMAP-Marker-Gene-Scores-w-Imputation_9.png" width="600" height="600">
</div>
</div>
</div>
<div id="umap-genescores-wo-imputation" class="section level2 tabset tabset-fade tabset-pills">
<h2 class="hasAnchor">
<a href="#umap-genescores-wo-imputation" class="anchor"></a>UMAP GeneScores w/o Imputation</h2>
<p>This <a href="../../images/Tutorial/UMAP-Marker-Gene-Scores-wo-Imputation.pdf">plot</a> shows gene activity scores of the marker genes described above overlayed onto the UMAP embedding. (Note if you see a blank space below try firefox or safari)</p>
<div id="section-22" class="section level3">
<h3 class="hasAnchor">
<a href="#section-22" class="anchor"></a>1</h3>
<div class="figure">
<img src="../../images/Tutorial2/UMAP-Marker-Gene-Scores-wo-Imputation_1.png" width="600" height="600">
</div>
</div>
<div id="section-23" class="section level3">
<h3 class="hasAnchor">
<a href="#section-23" class="anchor"></a>2</h3>
<div class="figure">
<img src="../../images/Tutorial2/UMAP-Marker-Gene-Scores-wo-Imputation_2.png" width="600" height="600">
</div>
</div>
<div id="section-24" class="section level3">
<h3 class="hasAnchor">
<a href="#section-24" class="anchor"></a>3</h3>
<div class="figure">
<img src="../../images/Tutorial2/UMAP-Marker-Gene-Scores-wo-Imputation_3.png" width="600" height="600">
</div>
</div>
<div id="section-25" class="section level3">
<h3 class="hasAnchor">
<a href="#section-25" class="anchor"></a>4</h3>
<div class="figure">
<img src="../../images/Tutorial2/UMAP-Marker-Gene-Scores-wo-Imputation_4.png" width="600" height="600">
</div>
</div>
<div id="section-26" class="section level3">
<h3 class="hasAnchor">
<a href="#section-26" class="anchor"></a>5</h3>
<div class="figure">
<img src="../../images/Tutorial2/UMAP-Marker-Gene-Scores-wo-Imputation_5.png" width="600" height="600">
</div>
</div>
<div id="section-27" class="section level3">
<h3 class="hasAnchor">
<a href="#section-27" class="anchor"></a>6</h3>
<div class="figure">
<img src="../../images/Tutorial2/UMAP-Marker-Gene-Scores-wo-Imputation_6.png" width="600" height="600">
</div>
</div>
<div id="section-28" class="section level3">
<h3 class="hasAnchor">
<a href="#section-28" class="anchor"></a>7</h3>
<div class="figure">
<img src="../../images/Tutorial2/UMAP-Marker-Gene-Scores-wo-Imputation_7.png" width="600" height="600">
</div>
</div>
<div id="section-29" class="section level3">
<h3 class="hasAnchor">
<a href="#section-29" class="anchor"></a>8</h3>
<div class="figure">
<img src="../../images/Tutorial2/UMAP-Marker-Gene-Scores-wo-Imputation_8.png" width="600" height="600">
</div>
</div>
<div id="section-30" class="section level3">
<h3 class="hasAnchor">
<a href="#section-30" class="anchor"></a>9</h3>
<div class="figure">
<img src="../../images/Tutorial2/UMAP-Marker-Gene-Scores-wo-Imputation_9.png" width="600" height="600">
</div>
</div>
</div>
<div id="track-plots" class="section level2 tabset tabset-fade tabset-pills">
<h2 class="hasAnchor">
<a href="#track-plots" class="anchor"></a>Track-Plots</h2>
<div id="section-31" class="section level3">
<h3 class="hasAnchor">
<a href="#section-31" class="anchor"></a>1</h3>
<div class="figure">
<img src="../../images/Tutorial2/Plot-Tracks-Marker-Genes_1.png" width="600" height="900">
</div>
</div>
<div id="section-32" class="section level3">
<h3 class="hasAnchor">
<a href="#section-32" class="anchor"></a>2</h3>
<div class="figure">
<img src="../../images/Tutorial2/Plot-Tracks-Marker-Genes_2.png" width="600" height="900">
</div>
</div>
<div id="section-33" class="section level3">
<h3 class="hasAnchor">
<a href="#section-33" class="anchor"></a>3</h3>
<div class="figure">
<img src="../../images/Tutorial2/Plot-Tracks-Marker-Genes_3.png" width="600" height="900">
</div>
</div>
<div id="section-34" class="section level3">
<h3 class="hasAnchor">
<a href="#section-34" class="anchor"></a>4</h3>
<div class="figure">
<img src="../../images/Tutorial2/Plot-Tracks-Marker-Genes_4.png" width="600" height="900">
</div>
</div>
<div id="section-35" class="section level3">
<h3 class="hasAnchor">
<a href="#section-35" class="anchor"></a>5</h3>
<div class="figure">
<img src="../../images/Tutorial2/Plot-Tracks-Marker-Genes_5.png" width="600" height="900">
</div>
</div>
<div id="section-36" class="section level3">
<h3 class="hasAnchor">
<a href="#section-36" class="anchor"></a>6</h3>
<div class="figure">
<img src="../../images/Tutorial2/Plot-Tracks-Marker-Genes_6.png" width="600" height="900">
</div>
</div>
<div id="section-37" class="section level3">
<h3 class="hasAnchor">
<a href="#section-37" class="anchor"></a>7</h3>
<div class="figure">
<img src="../../images/Tutorial2/Plot-Tracks-Marker-Genes_7.png" width="600" height="900">
</div>
</div>
<div id="section-38" class="section level3">
<h3 class="hasAnchor">
<a href="#section-38" class="anchor"></a>8</h3>
<div class="figure">
<img src="../../images/Tutorial2/Plot-Tracks-Marker-Genes_8.png" width="600" height="900">
</div>
</div>
<div id="section-39" class="section level3">
<h3 class="hasAnchor">
<a href="#section-39" class="anchor"></a>9</h3>
<div class="figure">
<img src="../../images/Tutorial2/Plot-Tracks-Marker-Genes_9.png" width="600" height="900">
</div>
</div>
</div>
<div id="marker-genescores-heatmap" class="section level2 tabset tabset-fade tabset-pills">
<h2 class="hasAnchor">
<a href="#marker-genescores-heatmap" class="anchor"></a>Marker GeneScores Heatmap</h2>
<div class="figure">
<img src="../../images/Tutorial2/GeneScores-Marker-Heatmap_1.png" width="600" height="900">
</div>
</div>
</div>
<div id="creating-a-reproducible-peak-set" class="section level1 tabset tabset-fade tabset-pills">
<h1 class="hasAnchor">
<a href="#creating-a-reproducible-peak-set" class="anchor"></a>7. Creating a Reproducible Peak Set</h1>
<p>One of the most complicated aspects about ATAC-seq and scATAC-seq analysis is the generation of a reproducible and robust peak set. In ArchR, we use an iterative overlap removal process that we first described in <a href="https://www.ncbi.nlm.nih.gov/pubmed/30361341">Corces* &amp; Granja* et al. Science 2018</a>. This process is described in detail in the <a href="articles/Articles/peakCalling.html">peak calling vignette</a>.</p>
<p>To robustly call peaks, we first merge the sparse single-cell data into pseudo-bulk replicates by aggregating the insertions from many individual cells into a single group. We make multiple pseudo-bulk replicates for each cluster to enable an assessment of peak reproducibility. This process of pseudo-bulk generation is described in detail in the <a href="articles/Articles/pseudobulkGeneration.html">pseudo-bulk generation vignette</a>. We than call peaks using <a href="https://github.com/taoliu/MACS">MACS2</a> and perform our iterative overlap removal. Once we obtain a finalized non-overlapping peak set, we can then collect insertion counts in each peak for each single cell and associate this with the corresponding <code>ArrowFile</code> via the <code>ArchRProject</code>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#Create Group Coverage Files that can be used for downstream analysis (~5-10 minutes)</span>
proj &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/addGroupCoverages.html">addGroupCoverages</a></span>(<span class="dt">ArchRProj =</span> proj, <span class="dt">groupBy =</span> <span class="st">"Clusters"</span>)

<span class="co">#Call Reproducible Peaks w/ Macs2 (~5-10 minutes)</span>
proj &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/addReproduciblePeakSet.html">addReproduciblePeakSet</a></span>(<span class="dt">ArchRProj =</span> proj, <span class="dt">groupBy =</span> <span class="st">"Clusters"</span>)

<span class="co">#Add Peak Matrix</span>
proj &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/addPeakMatrix.html">addPeakMatrix</a></span>(<span class="dt">ArchRProj =</span> proj)

<span class="co">#We can now plot the fraction of reads/insertions for each cell per sample and see how that compares to TSS Enrichment scores.</span>
plotList &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/list.html">list</a></span>()
plotList[[<span class="dv">1</span>]] &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/plotGroups.html">plotGroups</a></span>(<span class="dt">ArchRProj =</span> proj, 
  <span class="dt">groupBy =</span> <span class="st">"Sample"</span>, 
  <span class="dt">colorBy =</span> <span class="st">"colData"</span>, 
  <span class="dt">name =</span> <span class="st">"FRIP"</span>,
)
plotList[[<span class="dv">2</span>]] &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/plotGroups.html">plotGroups</a></span>(<span class="dt">ArchRProj =</span> proj, 
  <span class="dt">groupBy =</span> <span class="st">"Sample"</span>, 
  <span class="dt">colorBy =</span> <span class="st">"colData"</span>, 
  <span class="dt">name =</span> <span class="st">"TSSEnrichment"</span>,
)
<span class="kw"><a href="../../reference/plotPDF.html">plotPDF</a></span>(<span class="dt">plotList =</span> plotList, <span class="dt">name =</span> <span class="st">"FRIP-TSS-Enrichment"</span>, <span class="dt">width =</span> <span class="dv">4</span>, <span class="dt">height =</span> <span class="dv">4</span>,  <span class="dt">ArchRProj =</span> proj, <span class="dt">addDOC =</span> <span class="ot">FALSE</span>)</code></pre></div>
<div id="reproducible-peaks-per-cluster" class="section level2">
<h2 class="hasAnchor">
<a href="#reproducible-peaks-per-cluster" class="anchor"></a>Reproducible Peaks per Cluster</h2>
<div class="figure">
<img src="../../images/Tutorial2/Peak-Call-Summary_1.png" width="600" height="400">
</div>
</div>
<div id="frip-tss-enrichment-per-sample" class="section level2 tabset tabset-fade tabset-pills">
<h2 class="hasAnchor">
<a href="#frip-tss-enrichment-per-sample" class="anchor"></a>FRIP / TSS Enrichment Per Sample</h2>
<div id="section-40" class="section level3">
<h3 class="hasAnchor">
<a href="#section-40" class="anchor"></a>1</h3>
<div class="figure">
<img src="../../images/Tutorial2/FRIP-TSS-Enrichment_1.png" width="600" height="600">
</div>
</div>
<div id="section-41" class="section level3">
<h3 class="hasAnchor">
<a href="#section-41" class="anchor"></a>2</h3>
<div class="figure">
<img src="../../images/Tutorial2/FRIP-TSS-Enrichment_2.png" width="600" height="600">
</div>
</div>
</div>
</div>
<div id="identifying-marker-peaks" class="section level1 tabset tabset-fade tabset-pills">
<h1 class="hasAnchor">
<a href="#identifying-marker-peaks" class="anchor"></a>8. Identifying Marker Peaks</h1>
<p>Often times, we are interested to know which peaks are unique to an individual cluster or a small group of clusters. We can do this in an unsupervised fashion in ArchR:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#Identify Marker Peaks</span>
markersPeaks &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/markerFeatures.html">markerFeatures</a></span>(<span class="dt">ArchRProj =</span> proj, <span class="dt">useMatrix =</span> <span class="st">"PeakMatrix"</span>, <span class="dt">groupBy =</span> <span class="st">"Clusters"</span>)

<span class="co">#Visualize Markers as a heatmap</span>
heatmapPeaks &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/markerHeatmap.html">markerHeatmap</a></span>(
  <span class="dt">seMarker =</span> markersPeaks, 
  <span class="dt">cutOff =</span> <span class="st">"FDR &lt;= 0.1 &amp; Log2FC &gt;= 1"</span>
)
<span class="kw"><a href="../../reference/plotPDF.html">plotPDF</a></span>(heatmapPeaks, <span class="dt">name =</span> <span class="st">"Peak-Marker-Heatmap"</span>, <span class="dt">width =</span> <span class="dv">8</span>, <span class="dt">height =</span> <span class="dv">12</span>, <span class="dt">ArchRProj =</span> proj, <span class="dt">addDOC =</span> <span class="ot">FALSE</span>)</code></pre></div>
<div id="markers-heatmap" class="section level2">
<h2 class="hasAnchor">
<a href="#markers-heatmap" class="anchor"></a>Markers Heatmap</h2>
<div class="figure">
<img src="../../images/Tutorial2/Peak-Marker-Heatmap_1.png" width="600" height="900">
</div>
</div>
<div id="markers-mavolcano-plot" class="section level2 tabset tabset-fade tabset-pills">
<h2 class="hasAnchor">
<a href="#markers-mavolcano-plot" class="anchor"></a>Markers MA/Volcano-Plot</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#Marker Peaks for Cluster 6 (Erythroid)</span>
plotList &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/list.html">list</a></span>()
plotList[[<span class="dv">1</span>]] &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/markerPlot.html">markerPlot</a></span>(<span class="dt">seMarker =</span> markersPeaks, <span class="dt">name =</span> <span class="st">"Cluster6"</span>, <span class="dt">cutOff =</span> <span class="st">"FDR &lt;= 0.1 &amp; abs(Log2FC) &gt;= 1"</span>, <span class="dt">plotAs =</span> <span class="st">"MA"</span>)
plotList[[<span class="dv">2</span>]] &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/markerPlot.html">markerPlot</a></span>(<span class="dt">seMarker =</span> markersPeaks, <span class="dt">name =</span> <span class="st">"Cluster6"</span>, <span class="dt">cutOff =</span> <span class="st">"FDR &lt;= 0.1 &amp; abs(Log2FC) &gt;= 1"</span>, <span class="dt">plotAs =</span> <span class="st">"Volcano"</span>)
<span class="kw"><a href="../../reference/plotPDF.html">plotPDF</a></span>(<span class="dt">plotList =</span> plotList, <span class="dt">name =</span> <span class="st">"Cluster6-Marker-Peaks"</span>, <span class="dt">width =</span> <span class="dv">6</span>, <span class="dt">height =</span> <span class="dv">6</span>, <span class="dt">ArchRProj =</span> proj, <span class="dt">addDOC =</span> <span class="ot">FALSE</span>)</code></pre></div>
<div id="section-42" class="section level3">
<h3 class="hasAnchor">
<a href="#section-42" class="anchor"></a>1</h3>
<div class="figure">
<img src="../../images/Tutorial2/Cluster6-Marker-Peaks_1.png" width="600" height="600">
</div>
</div>
<div id="section-43" class="section level3">
<h3 class="hasAnchor">
<a href="#section-43" class="anchor"></a>2</h3>
<div class="figure">
<img src="../../images/Tutorial2/Cluster6-Marker-Peaks_1.png" width="600" height="600">
</div>
</div>
</div>
<div id="track-of-peak-markers" class="section level2 tabset tabset-fade tabset-pills">
<h2 class="hasAnchor">
<a href="#track-of-peak-markers" class="anchor"></a>Track of Peak Markers</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#Get Marker Regions</span>
markerRegions &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/markerGR.html">markerGR</a></span>(<span class="dt">seMarker =</span> markersPeaks, <span class="dt">cutOff =</span> <span class="st">"FDR &lt;= 0.1 &amp; Log2FC &gt; 1"</span>)

<span class="co">#Plot Tracks at Marker Genes</span>
plotTracks &lt;-<span class="st"> </span><span class="kw">ArchRRegionTrack</span>(
  <span class="dt">ArchRProj =</span> proj, 
  <span class="dt">geneSymbol =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"GATA1"</span>, <span class="st">"PAX5"</span>, <span class="st">"CD8A"</span>), 
  <span class="dt">features =</span> markerRegions,
  <span class="dt">plotSummary =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"bulkTrack"</span>, <span class="st">"featureTrack"</span>, <span class="st">"geneTrack"</span>),
  <span class="dt">sizes =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="dv">10</span>, <span class="dv">4</span>, <span class="dv">4</span>)
  )
<span class="kw"><a href="../../reference/plotPDF.html">plotPDF</a></span>(<span class="dt">plotList =</span> plotTracks, <span class="dt">name =</span> <span class="st">"Plot-Tracks-w-Marker-Peaks-Genes"</span>, <span class="dt">width =</span> <span class="dv">6</span>, <span class="dt">height =</span> <span class="dv">8</span>, <span class="dt">ArchRProj =</span> proj, <span class="dt">addDOC =</span> <span class="ot">FALSE</span>)</code></pre></div>
<div id="section-44" class="section level3">
<h3 class="hasAnchor">
<a href="#section-44" class="anchor"></a>1</h3>
<div class="figure">
<img src="../../images/Tutorial2/Plot-Tracks-w-Marker-Peaks-Genes_1.png" width="600" height="800">
</div>
</div>
<div id="section-45" class="section level3">
<h3 class="hasAnchor">
<a href="#section-45" class="anchor"></a>2</h3>
<div class="figure">
<img src="../../images/Tutorial2/Plot-Tracks-w-Marker-Peaks-Genes_2.png" width="600" height="800">
</div>
</div>
<div id="section-46" class="section level3">
<h3 class="hasAnchor">
<a href="#section-46" class="anchor"></a>3</h3>
<div class="figure">
<img src="../../images/Tutorial2/Plot-Tracks-w-Marker-Peaks-Genes_3.png" width="600" height="800">
</div>
</div>
</div>
<div id="pairwise-test-between-groups" class="section level2 tabset tabset-fade tabset-pills">
<h2 class="hasAnchor">
<a href="#pairwise-test-between-groups" class="anchor"></a>Pairwise-Test Between Groups</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#Pairwise Test between Cluster 6 (Erythroid) and Cluster 10 (B-Cells)</span>
testClust6_10 &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/markerFeatures.html">markerFeatures</a></span>(
  <span class="dt">ArchRProj =</span> proj, 
  <span class="dt">useMatrix =</span> <span class="st">"PeakMatrix"</span>,
  <span class="dt">groupBy =</span> <span class="st">"Clusters"</span>,
  <span class="dt">testMethod =</span> <span class="st">"wilcoxon"</span>,
  <span class="dt">bias =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"TSSEnrichment"</span>, <span class="st">"log10(nFrags)"</span>),
  <span class="dt">useGroups =</span> <span class="st">"Cluster6"</span>,
  <span class="dt">bgdGroups =</span> <span class="st">"Cluster10"</span>
)

<span class="co">#Plot Differential Plot</span>
plotList &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/list.html">list</a></span>()
plotList[[<span class="dv">1</span>]] &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/markerPlot.html">markerPlot</a></span>(<span class="dt">seMarker =</span> testClust6_10, <span class="dt">name =</span> <span class="st">"Cluster6"</span>, <span class="dt">cutOff =</span> <span class="st">"FDR &lt;= 0.1 &amp; abs(Log2FC) &gt;= 1"</span>, <span class="dt">plotAs =</span> <span class="st">"MA"</span>)
plotList[[<span class="dv">2</span>]] &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/markerPlot.html">markerPlot</a></span>(<span class="dt">seMarker =</span> testClust6_10, <span class="dt">name =</span> <span class="st">"Cluster6"</span>, <span class="dt">cutOff =</span> <span class="st">"FDR &lt;= 0.1 &amp; abs(Log2FC) &gt;= 1"</span>, <span class="dt">plotAs =</span> <span class="st">"Volcano"</span>)
<span class="kw"><a href="../../reference/plotPDF.html">plotPDF</a></span>(<span class="dt">plotList =</span> plotList, <span class="dt">name =</span> <span class="st">"Cluster6-Cluster10-Marker-Peaks"</span>, <span class="dt">width =</span> <span class="dv">6</span>, <span class="dt">height =</span> <span class="dv">6</span>, <span class="dt">ArchRProj =</span> proj, <span class="dt">addDOC =</span> <span class="ot">FALSE</span>)</code></pre></div>
<div id="section-47" class="section level3">
<h3 class="hasAnchor">
<a href="#section-47" class="anchor"></a>1</h3>
<div class="figure">
<img src="../../images/Tutorial2/Cluster6-Cluster10-Marker-Peaks_1.png" width="600" height="600">
</div>
</div>
<div id="section-48" class="section level3">
<h3 class="hasAnchor">
<a href="#section-48" class="anchor"></a>2</h3>
<div class="figure">
<img src="../../images/Tutorial2/Cluster6-Cluster10-Marker-Peaks_2.png" width="600" height="600">
</div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#Add Motif Peak Annotations if not already added!</span>
proj &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/addMotifAnnotations.html">addMotifAnnotations</a></span>(<span class="dt">ArchRProj =</span> proj, <span class="dt">motifSet =</span> <span class="st">"cisbp"</span>, <span class="dt">name =</span> <span class="st">"Motif"</span>)

<span class="co">#Identify Motif Enrichments</span>
motifsUp &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/peakAnnoEnrichment.html">peakAnnoEnrichment</a></span>(
    <span class="dt">seMarker =</span> testClust6_10,
    <span class="dt">ArchRProj =</span> proj,
    <span class="dt">peakAnnotation =</span> <span class="st">"Motif"</span>,
    <span class="dt">cutOff =</span> <span class="st">"FDR &lt;= 0.1 &amp; Log2FC &gt;= 0.5"</span>
  )

<span class="co">#Create data.frame</span>
df &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span>(<span class="dt">TF =</span> <span class="kw"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(motifsUp), <span class="dt">mlog10FDR =</span> <span class="kw">assay</span>(motifsUp)[,<span class="dv">1</span>])
df &lt;-<span class="st"> </span>df[<span class="kw"><a href="https://rdrr.io/r/base/order.html">order</a></span>(df$mlog10FDR, <span class="dt">decreasing =</span> <span class="ot">TRUE</span>),]
df$rank &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/seq.html">seq_len</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span>(df))

<span class="co">#Ggplot</span>
ggUp &lt;-<span class="st"> </span><span class="kw">ggplot</span>(df, <span class="kw">aes</span>(rank, mlog10FDR)) +<span class="st"> </span>
<span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">size =</span> <span class="dv">1</span>) +
<span class="st">  </span>ggrepel::<span class="kw">geom_label_repel</span>(
        <span class="dt">data =</span> df[<span class="kw"><a href="https://rdrr.io/r/base/rev.html">rev</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/seq.html">seq_len</a></span>(<span class="dv">30</span>)), ], <span class="kw">aes</span>(<span class="dt">x =</span> rank, <span class="dt">y =</span> mlog10FDR, <span class="dt">label =</span> TF), 
        <span class="dt">size =</span> <span class="fl">1.5</span>,
        <span class="dt">nudge_x =</span> <span class="dv">2</span>
  ) +<span class="st"> </span><span class="kw"><a href="../../reference/theme_ArchR.html">theme_ArchR</a></span>() +<span class="st"> </span><span class="kw">ylab</span>(<span class="st">"-log10(FDR) Motif Enrichment"</span>) +<span class="st"> </span><span class="kw">xlab</span>(<span class="st">"Rank Sorted TFs Enriched"</span>)
<span class="kw"><a href="../../reference/plotPDF.html">plotPDF</a></span>(ggUp, <span class="dt">name =</span> <span class="st">"Cluster6-Cluster10-Motifs-Up"</span>, <span class="dt">width =</span> <span class="dv">6</span>, <span class="dt">height =</span> <span class="dv">6</span>, <span class="dt">ArchRProj =</span> proj, <span class="dt">addDOC =</span> <span class="ot">FALSE</span>)</code></pre></div>
<div class="figure">
<img src="../../images/Tutorial2/Cluster6-Cluster10-Motifs-Up_1.png" width="600" height="600">
</div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">motifsDo &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/peakAnnoEnrichment.html">peakAnnoEnrichment</a></span>(
    <span class="dt">seMarker =</span> testClust6_10,
    <span class="dt">ArchRProj =</span> proj,
    <span class="dt">peakAnnotation =</span> <span class="st">"Motif"</span>,
    <span class="dt">cutOff =</span> <span class="st">"FDR &lt;= 0.1 &amp; Log2FC &lt;= -0.5"</span>
  )

<span class="co">#Create data.frame</span>
df &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span>(<span class="dt">TF =</span> <span class="kw"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(motifsDo), <span class="dt">mlog10FDR =</span> <span class="kw">assay</span>(motifsDo)[,<span class="dv">1</span>])
df &lt;-<span class="st"> </span>df[<span class="kw"><a href="https://rdrr.io/r/base/order.html">order</a></span>(df$mlog10FDR, <span class="dt">decreasing =</span> <span class="ot">TRUE</span>),]
df$rank &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/seq.html">seq_len</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span>(df))

<span class="co">#Ggplot</span>
ggDo &lt;-<span class="st"> </span><span class="kw">ggplot</span>(df, <span class="kw">aes</span>(rank, mlog10FDR)) +<span class="st"> </span>
<span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">size =</span> <span class="dv">1</span>) +
<span class="st">  </span>ggrepel::<span class="kw">geom_label_repel</span>(
        <span class="dt">data =</span> df[<span class="kw"><a href="https://rdrr.io/r/base/rev.html">rev</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/seq.html">seq_len</a></span>(<span class="dv">30</span>)), ], <span class="kw">aes</span>(<span class="dt">x =</span> rank, <span class="dt">y =</span> mlog10FDR, <span class="dt">label =</span> TF), 
        <span class="dt">size =</span> <span class="fl">1.5</span>,
        <span class="dt">nudge_x =</span> <span class="dv">2</span>
  ) +<span class="st"> </span><span class="kw"><a href="../../reference/theme_ArchR.html">theme_ArchR</a></span>() +<span class="st"> </span><span class="kw">ylab</span>(<span class="st">"-log10(FDR) Motif Enrichment"</span>) +<span class="st"> </span><span class="kw">xlab</span>(<span class="st">"Rank Sorted TFs Enriched"</span>)
<span class="kw"><a href="../../reference/plotPDF.html">plotPDF</a></span>(ggDo, <span class="dt">name =</span> <span class="st">"Cluster6-Cluster10-Motifs-Down"</span>, <span class="dt">width =</span> <span class="dv">6</span>, <span class="dt">height =</span> <span class="dv">6</span>, <span class="dt">ArchRProj =</span> proj, <span class="dt">addDOC =</span> <span class="ot">FALSE</span>)</code></pre></div>
<div class="figure">
<img src="../../images/Tutorial2/Cluster6-Cluster10-Motifs-Down_1.png" width="600" height="600">
</div>
</div>
</div>
</div>
<div id="performing-peak-annotation-enrichments" class="section level1 tabset tabset-fade tabset-pills">
<h1 class="hasAnchor">
<a href="#performing-peak-annotation-enrichments" class="anchor"></a>9. Performing Peak Annotation Enrichments</h1>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#Motif Search in Peak Set and add to Peak Annotations</span>
proj &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/addMotifAnnotations.html">addMotifAnnotations</a></span>(<span class="dt">ArchRProj =</span> proj, <span class="dt">motifSet =</span> <span class="st">"cisbp"</span>, <span class="dt">name =</span> <span class="st">"Motif"</span>)

<span class="co">#Identify Motif Enrichments</span>
enrichMotifs &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/peakAnnoEnrichment.html">peakAnnoEnrichment</a></span>(
    <span class="dt">seMarker =</span> markersPeaks,
    <span class="dt">ArchRProj =</span> proj,
    <span class="dt">peakAnnotation =</span> <span class="st">"Motif"</span>,
    <span class="dt">cutOff =</span> <span class="st">"FDR &lt;= 0.1 &amp; Log2FC &gt;= 0.5"</span>
  )
heatmapEM &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/enrichHeatmap.html">enrichHeatmap</a></span>(enrichMotifs, <span class="dt">n =</span> <span class="dv">10</span>)
<span class="kw"><a href="../../reference/plotPDF.html">plotPDF</a></span>(heatmapEM, <span class="dt">name =</span> <span class="st">"Motifs-Enrich-Heatmap"</span>, <span class="dt">width =</span> <span class="dv">8</span>, <span class="dt">height =</span> <span class="dv">12</span>, <span class="dt">ArchRProj =</span> proj, <span class="dt">addDOC =</span> <span class="ot">FALSE</span>)

<span class="co">#Custom Peak Annotations</span>
EncodePeaks &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(
  <span class="dt">Encode_K562_GATA1 =</span> <span class="st">"https://www.encodeproject.org/files/ENCFF632NQI/@@download/ENCFF632NQI.bed.gz"</span>,
  <span class="dt">Encode_GM12878_CEBPB =</span> <span class="st">"https://www.encodeproject.org/files/ENCFF761MGJ/@@download/ENCFF761MGJ.bed.gz"</span>,
  <span class="dt">Encode_K562_Ebf1 =</span> <span class="st">"https://www.encodeproject.org/files/ENCFF868VSY/@@download/ENCFF868VSY.bed.gz"</span>,
  <span class="dt">Encode_K562_Pax5 =</span> <span class="st">"https://www.encodeproject.org/files/ENCFF339KUO/@@download/ENCFF339KUO.bed.gz"</span>
)
proj &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/addPeakAnnotations.html">addPeakAnnotations</a></span>(<span class="dt">ArchRProj =</span> proj, <span class="dt">regions =</span> EncodePeaks, <span class="dt">name =</span> <span class="st">"ChIP"</span>)

<span class="co">#Identify ChIP Enrichments</span>
enrichRegions &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/peakAnnoEnrichment.html">peakAnnoEnrichment</a></span>(
    <span class="dt">seMarker =</span> markersPeaks,
    <span class="dt">ArchRProj =</span> proj,
    <span class="dt">peakAnnotation =</span> <span class="st">"ChIP"</span>,
    <span class="dt">cutOff =</span> <span class="st">"FDR &lt;= 0.1 &amp; Log2FC &gt;= 0.5"</span>
  )
heatmapER &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/enrichHeatmap.html">enrichHeatmap</a></span>(enrichRegions, <span class="dt">n =</span> <span class="dv">10</span>)
<span class="kw"><a href="../../reference/plotPDF.html">plotPDF</a></span>(heatmapER, <span class="dt">name =</span> <span class="st">"Regions-Enrich-Heatmap"</span>, <span class="dt">width =</span> <span class="dv">8</span>, <span class="dt">height =</span> <span class="dv">12</span>, <span class="dt">ArchRProj =</span> proj, <span class="dt">addDOC =</span> <span class="ot">FALSE</span>)</code></pre></div>
<div id="motif-enrichment" class="section level2">
<h2 class="hasAnchor">
<a href="#motif-enrichment" class="anchor"></a>Motif Enrichment</h2>
<div class="figure">
<img src="../../images/Tutorial2/Motifs-Enrich-Heatmap_1.png" width="600" height="900">
</div>
</div>
<div id="region-enrichment" class="section level2">
<h2 class="hasAnchor">
<a href="#region-enrichment" class="anchor"></a>Region Enrichment</h2>
<div class="figure">
<img src="../../images/Tutorial2/Regions-Enrich-Heatmap_1.png" width="600" height="900">
</div>
</div>
</div>
<div id="computing-chromvar-deviations" class="section level1 tabset tabset-fade tabset-pills">
<h1 class="hasAnchor">
<a href="#computing-chromvar-deviations" class="anchor"></a>10. Computing chromVAR Deviations</h1>
<p>Using the reproducible peak set that we defined above, we can use ArchR to calculate TF/Annotation deviations on a single-cell basis for transcription factors/annotations in the peaks identified in each cluster. We can then overlay these deviations on on UMAP embedding. This effectively infers differences in TF activity across all single cells and is very useful in identifying regulatory factors governing cell fate.</p>
<div id="motif-deviations" class="section level2 tabset tabset-fade tabset-pills">
<h2 class="hasAnchor">
<a href="#motif-deviations" class="anchor"></a>Motif Deviations</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#Add Background Peaks to ArchR Project</span>
proj &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/addBgdPeaks.html">addBgdPeaks</a></span>(proj)

<span class="co">#Add chromVAR Deviations (~20-25 min if using CisBP Motif Set)</span>
proj &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/addDeviationsMatrix.html">addDeviationsMatrix</a></span>(<span class="dt">ArchRProj =</span> proj, <span class="dt">peakAnnotation =</span> <span class="st">"Motif"</span>)

<span class="co">#Plot Variable TFs</span>
plotVarDev &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/getVarDeviations.html">getVarDeviations</a></span>(proj, <span class="dt">plot =</span> <span class="ot">TRUE</span>)
<span class="kw"><a href="../../reference/plotPDF.html">plotPDF</a></span>(plotVarDev, <span class="dt">name =</span> <span class="st">"Plot-Var-Deviations"</span>, <span class="dt">width =</span> <span class="dv">6</span>, <span class="dt">height =</span> <span class="dv">6</span>, <span class="dt">ArchRProj =</span> proj, <span class="dt">addDOC =</span> <span class="ot">FALSE</span>)</code></pre></div>
<div class="figure">
<img src="../../images/Tutorial2/Plot-Var-Deviations_1.png" width="600" height="600">
</div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#To access motif need to specify deviations,z : motif_name</span>
<span class="co">#Try getFeatures with MotifMatrix to see available names</span>
<span class="kw"><a href="../../reference/getFeatures.html">getFeatures</a></span>(proj, <span class="dt">select =</span> <span class="st">"PAX5"</span>, <span class="dt">useMatrix =</span> <span class="st">"MotifMatrix"</span>)

<span class="co">#Define the list of motifs to plot</span>
markerMotifs &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"GATA1_383"</span>, <span class="st">"CEBPA_155"</span>, <span class="st">"EBF1_67"</span>, <span class="st">"IRF4_632"</span>, <span class="st">"TBX21_780"</span>, <span class="st">"PAX5_709"</span>)

<span class="co">#Plot Violins of motif deviations</span>
plotList &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/list.html">list</a></span>()
plotList[[<span class="dv">1</span>]] &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/plotGroups.html">plotGroups</a></span>(<span class="dt">ArchRProj =</span> proj, 
  <span class="dt">groupBy =</span> <span class="st">"Clusters"</span>, 
  <span class="dt">colorBy =</span> <span class="st">"MotifMatrix"</span>, 
  <span class="dt">name =</span> <span class="kw"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span>(<span class="st">"z:"</span>,markerMotifs[<span class="dv">1</span>]),
  <span class="dt">imputeWeights =</span> <span class="kw"><a href="../../reference/getImputeWeights.html">getImputeWeights</a></span>(proj)
)
<span class="kw"><a href="../../reference/plotPDF.html">plotPDF</a></span>(<span class="dt">plotList =</span> plotList, <span class="dt">name =</span> <span class="st">"Groups-Marker-MotifDeviations-w-Imputation"</span>, <span class="dt">width =</span> <span class="dv">4</span>, <span class="dt">height =</span> <span class="dv">3</span>,  <span class="dt">ArchRProj =</span> proj, <span class="dt">addDOC =</span> <span class="ot">FALSE</span>)</code></pre></div>
<div class="figure">
<img src="../../images/Tutorial2/Groups-Marker-MotifDeviations-w-Imputation_1.png" width="600" height="450">
</div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#Plot the UMAP Embedding with chromVAR Deviations Overlayed</span>
plotList &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/list.html">list</a></span>()
plotList[[<span class="dv">1</span>]] &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/plotEmbedding.html">plotEmbedding</a></span>(<span class="dt">ArchRProj =</span> proj, <span class="dt">colorBy =</span> <span class="st">"colData"</span>, <span class="dt">name =</span> <span class="st">"Clusters"</span>)
plotList[[<span class="dv">2</span>]] &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/plotEmbedding.html">plotEmbedding</a></span>(<span class="dt">ArchRProj =</span> proj, <span class="dt">colorBy =</span> <span class="st">"MotifMatrix"</span>, <span class="dt">name =</span> <span class="kw"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span>(<span class="st">"z:"</span>,markerMotifs), <span class="dt">imputeWeights =</span> <span class="kw"><a href="../../reference/getImputeWeights.html">getImputeWeights</a></span>(proj))
<span class="kw"><a href="../../reference/plotPDF.html">plotPDF</a></span>(<span class="dt">plotList =</span> plotList, <span class="dt">name =</span> <span class="st">"UMAP-Marker-MotifDeviations-w-Imputation"</span>, <span class="dt">width =</span> <span class="dv">6</span>, <span class="dt">height =</span> <span class="dv">6</span>, <span class="dt">ArchRProj =</span> proj, <span class="dt">addDOC =</span> <span class="ot">FALSE</span>)</code></pre></div>
<div id="section-49" class="section level3">
<h3 class="hasAnchor">
<a href="#section-49" class="anchor"></a>1</h3>
<div class="figure">
<img src="../../images/Tutorial2/UMAP-Marker-MotifDeviations-w-Imputation_1.png" width="600" height="600">
</div>
</div>
<div id="section-50" class="section level3">
<h3 class="hasAnchor">
<a href="#section-50" class="anchor"></a>2</h3>
<div class="figure">
<img src="../../images/Tutorial2/UMAP-Marker-MotifDeviations-w-Imputation_2.png" width="600" height="600">
</div>
</div>
<div id="section-51" class="section level3">
<h3 class="hasAnchor">
<a href="#section-51" class="anchor"></a>3</h3>
<div class="figure">
<img src="../../images/Tutorial2/UMAP-Marker-MotifDeviations-w-Imputation_3.png" width="600" height="600">
</div>
</div>
<div id="section-52" class="section level3">
<h3 class="hasAnchor">
<a href="#section-52" class="anchor"></a>4</h3>
<div class="figure">
<img src="../../images/Tutorial2/UMAP-Marker-MotifDeviations-w-Imputation_4.png" width="600" height="600">
</div>
</div>
<div id="section-53" class="section level3">
<h3 class="hasAnchor">
<a href="#section-53" class="anchor"></a>5</h3>
<div class="figure">
<img src="../../images/Tutorial2/UMAP-Marker-MotifDeviations-w-Imputation_5.png" width="600" height="600">
</div>
</div>
<div id="section-54" class="section level3">
<h3 class="hasAnchor">
<a href="#section-54" class="anchor"></a>6</h3>
<div class="figure">
<img src="../../images/Tutorial2/UMAP-Marker-MotifDeviations-w-Imputation_6.png" width="600" height="600">
</div>
</div>
</div>
<div id="region-deviations" class="section level2 tabset tabset-fade tabset-pills">
<h2 class="hasAnchor">
<a href="#region-deviations" class="anchor"></a>Region Deviations</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#Add Background Peaks to ArchR Project</span>
proj &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/addBgdPeaks.html">addBgdPeaks</a></span>(proj)

<span class="co">#Add chromVAR Deviations</span>
proj &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/addDeviationsMatrix.html">addDeviationsMatrix</a></span>(<span class="dt">ArchRProj =</span> proj, <span class="dt">peakAnnotation =</span> <span class="st">"ChIP"</span>)

<span class="co">#To access motif need to specify deviations,z : motif_name</span>
<span class="co">#Try getFeatures with MotifMatrix to see available names</span>
<span class="kw"><a href="../../reference/getFeatures.html">getFeatures</a></span>(proj, <span class="dt">select =</span> <span class="st">"PAX5"</span>, <span class="dt">useMatrix =</span> <span class="st">"ChIPMatrix"</span>)

<span class="co">#Define the list of regions to plot</span>
markerRegions &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"Encode_K562_GATA1"</span>, <span class="st">"Encode_GM12878_CEBPB"</span>, <span class="st">"Encode_K562_Ebf1"</span>, <span class="st">"Encode_K562_Pax5"</span>)

<span class="co">#Plot Violins of ChIP-seq deviations</span>
plotList &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/list.html">list</a></span>()
plotList[[<span class="dv">1</span>]] &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/plotGroups.html">plotGroups</a></span>(<span class="dt">ArchRProj =</span> proj, 
  <span class="dt">groupBy =</span> <span class="st">"Clusters"</span>, 
  <span class="dt">colorBy =</span> <span class="st">"ChIPMatrix"</span>, 
  <span class="dt">name =</span> <span class="kw"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span>(<span class="st">"z:"</span>,markerRegions[<span class="dv">1</span>]),
  <span class="dt">imputeWeights =</span> <span class="kw"><a href="../../reference/getImputeWeights.html">getImputeWeights</a></span>(proj)
)
<span class="kw"><a href="../../reference/plotPDF.html">plotPDF</a></span>(<span class="dt">plotList =</span> plotList, <span class="dt">name =</span> <span class="st">"Groups-Marker-ChIPDeviations-w-Imputation"</span>, <span class="dt">width =</span> <span class="dv">4</span>, <span class="dt">height =</span> <span class="dv">3</span>,  <span class="dt">ArchRProj =</span> proj, <span class="dt">addDOC =</span> <span class="ot">FALSE</span>)</code></pre></div>
<div class="figure">
<img src="../../images/Tutorial2/Groups-Marker-ChIPDeviations-w-Imputation_1.png" width="600" height="450">
</div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#Plot the UMAP Embedding with chromVAR Deviations Overlayed</span>
plotList &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/list.html">list</a></span>()
plotList[[<span class="dv">1</span>]] &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/plotEmbedding.html">plotEmbedding</a></span>(<span class="dt">ArchRProj =</span> proj, <span class="dt">colorBy =</span> <span class="st">"colData"</span>, <span class="dt">name =</span> <span class="st">"Clusters"</span>)
plotList[[<span class="dv">2</span>]] &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/plotEmbedding.html">plotEmbedding</a></span>(<span class="dt">ArchRProj =</span> proj, <span class="dt">colorBy =</span> <span class="st">"ChIPMatrix"</span>, <span class="dt">name =</span> <span class="kw"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span>(<span class="st">"z:"</span>,markerRegions), <span class="dt">imputeWeights =</span> <span class="kw"><a href="../../reference/getImputeWeights.html">getImputeWeights</a></span>(proj))
<span class="kw"><a href="../../reference/plotPDF.html">plotPDF</a></span>(<span class="dt">plotList =</span> plotList, <span class="dt">name =</span> <span class="st">"UMAP-Marker-ChIPDeviations-w-Imputation"</span>, <span class="dt">width =</span> <span class="dv">6</span>, <span class="dt">height =</span> <span class="dv">6</span>, <span class="dt">ArchRProj =</span> proj, <span class="dt">addDOC =</span> <span class="ot">FALSE</span>)</code></pre></div>
<div id="section-55" class="section level3">
<h3 class="hasAnchor">
<a href="#section-55" class="anchor"></a>1</h3>
<div class="figure">
<img src="../../images/Tutorial2/UMAP-Marker-ChIPDeviations-w-Imputation_1.png" width="600" height="600">
</div>
</div>
<div id="section-56" class="section level3">
<h3 class="hasAnchor">
<a href="#section-56" class="anchor"></a>2</h3>
<div class="figure">
<img src="../../images/Tutorial2/UMAP-Marker-ChIPDeviations-w-Imputation_2.png" width="600" height="600">
</div>
</div>
<div id="section-57" class="section level3">
<h3 class="hasAnchor">
<a href="#section-57" class="anchor"></a>3</h3>
<div class="figure">
<img src="../../images/Tutorial2/UMAP-Marker-ChIPDeviations-w-Imputation_3.png" width="600" height="600">
</div>
</div>
<div id="section-58" class="section level3">
<h3 class="hasAnchor">
<a href="#section-58" class="anchor"></a>4</h3>
<div class="figure">
<img src="../../images/Tutorial2/UMAP-Marker-ChIPDeviations-w-Imputation_4.png" width="600" height="600">
</div>
</div>
</div>
</div>
<div id="performing-tf-footprinting" class="section level1 tabset tabset-fade tabset-pills">
<h1 class="hasAnchor">
<a href="#performing-tf-footprinting" class="anchor"></a>11. Performing TF Footprinting</h1>
<p>Transcription factor footprinting can also be done in ArchR with a single command. We note that the footprints generated by the tutorial data are not as clean as would be desired but this is because of the small size of the tutorial dataset.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#Plot Motif Footprints from positions list</span>
<span class="co">#Recommend doing a few motifs not entire motif set</span>
seFoot_None &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/plotFootprints.html">plotFootprints</a></span>(
  <span class="dt">ArchRProj =</span> proj, 
  <span class="dt">positions =</span> <span class="kw"><a href="../../reference/getPositions.html">getPositions</a></span>(proj)[markerMotifs], 
  <span class="dt">groupBy =</span> <span class="st">"Clusters"</span>,
  <span class="dt">normMethod =</span> <span class="st">"none"</span>,
  <span class="dt">plotName =</span> <span class="st">"Footprints-No-Normalization"</span>
)</code></pre></div>
<div id="no-normalization" class="section level2 tabset tabset-fade tabset-pills">
<h2 class="hasAnchor">
<a href="#no-normalization" class="anchor"></a>No Normalization</h2>
<div id="section-59" class="section level3">
<h3 class="hasAnchor">
<a href="#section-59" class="anchor"></a>1</h3>
<div class="figure">
<img src="../../images/Tutorial2/Footprints-No-Normalization_1.png" width="600" height="900">
</div>
</div>
<div id="section-60" class="section level3">
<h3 class="hasAnchor">
<a href="#section-60" class="anchor"></a>2</h3>
<div class="figure">
<img src="../../images/Tutorial2/Footprints-No-Normalization_2.png" width="600" height="900">
</div>
</div>
<div id="section-61" class="section level3">
<h3 class="hasAnchor">
<a href="#section-61" class="anchor"></a>3</h3>
<div class="figure">
<img src="../../images/Tutorial2/Footprints-No-Normalization_3.png" width="600" height="900">
</div>
</div>
<div id="section-62" class="section level3">
<h3 class="hasAnchor">
<a href="#section-62" class="anchor"></a>4</h3>
<div class="figure">
<img src="../../images/Tutorial2/Footprints-No-Normalization_4.png" width="600" height="900">
</div>
</div>
<div id="section-63" class="section level3">
<h3 class="hasAnchor">
<a href="#section-63" class="anchor"></a>5</h3>
<div class="figure">
<img src="../../images/Tutorial2/Footprints-No-Normalization_5.png" width="600" height="900">
</div>
</div>
<div id="section-64" class="section level3">
<h3 class="hasAnchor">
<a href="#section-64" class="anchor"></a>6</h3>
<div class="figure">
<img src="../../images/Tutorial2/Footprints-No-Normalization_6.png" width="600" height="900">
</div>
</div>
</div>
<div id="divide-bias" class="section level2 tabset tabset-fade tabset-pills">
<h2 class="hasAnchor">
<a href="#divide-bias" class="anchor"></a>Divide Bias</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#Re-Plot with different normalization method</span>
seFoot_Divide &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/plotFootprints.html">plotFootprints</a></span>(
  <span class="dt">ArchRProj =</span> proj, 
  <span class="dt">inputSE =</span> seFoot_None, 
  <span class="dt">positions =</span> <span class="kw"><a href="../../reference/getPositions.html">getPositions</a></span>(proj)[markerMotifs], 
  <span class="dt">groupBy =</span> <span class="st">"Clusters"</span>,
  <span class="dt">normMethod =</span> <span class="st">"divide"</span>,
  <span class="dt">plotName =</span> <span class="st">"Footprints-Divide-Bias"</span>
)</code></pre></div>
<div id="section-65" class="section level3">
<h3 class="hasAnchor">
<a href="#section-65" class="anchor"></a>1</h3>
<div class="figure">
<img src="../../images/Tutorial2/Footprints-Divide-Bias_1.png" width="600" height="900">
</div>
</div>
<div id="section-66" class="section level3">
<h3 class="hasAnchor">
<a href="#section-66" class="anchor"></a>2</h3>
<div class="figure">
<img src="../../images/Tutorial2/Footprints-Divide-Bias_2.png" width="600" height="900">
</div>
</div>
<div id="section-67" class="section level3">
<h3 class="hasAnchor">
<a href="#section-67" class="anchor"></a>3</h3>
<div class="figure">
<img src="../../images/Tutorial2/Footprints-Divide-Bias_3.png" width="600" height="900">
</div>
</div>
<div id="section-68" class="section level3">
<h3 class="hasAnchor">
<a href="#section-68" class="anchor"></a>4</h3>
<div class="figure">
<img src="../../images/Tutorial2/Footprints-Divide-Bias_4.png" width="600" height="900">
</div>
</div>
<div id="section-69" class="section level3">
<h3 class="hasAnchor">
<a href="#section-69" class="anchor"></a>5</h3>
<div class="figure">
<img src="../../images/Tutorial2/Footprints-Divide-Bias_5.png" width="600" height="900">
</div>
</div>
<div id="section-70" class="section level3">
<h3 class="hasAnchor">
<a href="#section-70" class="anchor"></a>6</h3>
<div class="figure">
<img src="../../images/Tutorial2/Footprints-Divide-Bias_6.png" width="600" height="900">
</div>
</div>
</div>
<div id="tss-enrichment" class="section level2 tabset tabset-fade tabset-pills">
<h2 class="hasAnchor">
<a href="#tss-enrichment" class="anchor"></a>TSS Enrichment</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#Add Coverages for Samples prior to footprint</span>
proj &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/addGroupCoverages.html">addGroupCoverages</a></span>(<span class="dt">ArchRProj =</span> proj, <span class="dt">groupBy =</span> <span class="st">"Sample"</span>)

<span class="co">#TSS by sample</span>
TSS_Samples &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/plotFootprints.html">plotFootprints</a></span>(
  <span class="dt">ArchRProj =</span> proj, 
  <span class="dt">positions =</span> <span class="kw">GRangesList</span>(<span class="dt">TSS =</span> <span class="kw"><a href="../../reference/getTSS.html">getTSS</a></span>(proj)), 
  <span class="dt">groupBy =</span> <span class="st">"Sample"</span>,
  <span class="dt">normMethod =</span> <span class="st">"none"</span>,
  <span class="dt">plotName =</span> <span class="st">"TSS_Samples"</span>,
  <span class="dt">flank =</span> <span class="dv">2000</span>,
  <span class="dt">flankNorm =</span> <span class="dv">100</span>
)</code></pre></div>
<div class="figure">
<img src="../../images/Tutorial2/TSS_Samples_1.png" width="600" height="900">
</div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#TSS by clusters</span>
TSS_Clusters &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/plotFootprints.html">plotFootprints</a></span>(
  <span class="dt">ArchRProj =</span> proj, 
  <span class="dt">positions =</span> <span class="kw">GRangesList</span>(<span class="dt">TSS =</span> <span class="kw"><a href="../../reference/getTSS.html">getTSS</a></span>(proj)), 
  <span class="dt">groupBy =</span> <span class="st">"Clusters"</span>,
  <span class="dt">normMethod =</span> <span class="st">"none"</span>,
  <span class="dt">plotName =</span> <span class="st">"TSS_Clusters"</span>,
  <span class="dt">flank =</span> <span class="dv">2000</span>,
  <span class="dt">flankNorm =</span> <span class="dv">100</span>
)</code></pre></div>
<div class="figure">
<img src="../../images/Tutorial2/TSS_Clusters_1.png" width="600" height="900">
</div>
</div>
</div>
<div id="analyze-trajectories" class="section level1 tabset tabset-fade tabset-pills">
<h1 class="hasAnchor">
<a href="#analyze-trajectories" class="anchor"></a>12. Analyze Trajectories</h1>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#Myeloid Trajectory</span>
<span class="co">#First we need to create a Trajectory and add it to ArchRProj cellColData</span>
proj &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/addTrajectory.html">addTrajectory</a></span>(<span class="dt">ArchRProj =</span> proj, <span class="dt">name =</span> <span class="st">"Myeloid"</span>, <span class="dt">trajectory =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span>(<span class="st">"Cluster"</span>, <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="dv">7</span>,<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">4</span>,<span class="dv">5</span>))))

<span class="co">#Additionally we can plot these features on top of the UMAP embedding</span>
plotList &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/list.html">list</a></span>()
plotList[[<span class="dv">1</span>]] &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/plotEmbedding.html">plotEmbedding</a></span>(<span class="dt">ArchRProj =</span> proj, <span class="dt">colorBy =</span> <span class="st">"colData"</span>, <span class="dt">name =</span> <span class="st">"Clusters"</span>, <span class="dt">plotParams =</span> <span class="kw"><a href="https://rdrr.io/r/base/list.html">list</a></span>(<span class="dt">labelMeans=</span><span class="ot">TRUE</span>))
plotList[[<span class="dv">2</span>]] &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/plotTrajectory.html">plotTrajectory</a></span>(proj, <span class="dt">trajectory =</span> <span class="st">"Myeloid"</span>, <span class="dt">colorBy =</span> <span class="st">"colData"</span>, <span class="dt">name =</span> <span class="st">"Myeloid"</span>)
plotList[[<span class="dv">3</span>]] &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/plotTrajectory.html">plotTrajectory</a></span>(proj, <span class="dt">trajectory =</span> <span class="st">"Myeloid"</span>, <span class="dt">colorBy =</span> <span class="st">"GeneScoreMatrix"</span>, <span class="dt">name =</span> <span class="st">"CEBPB"</span>)
plotList[[<span class="dv">4</span>]] &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/plotTrajectory.html">plotTrajectory</a></span>(proj, <span class="dt">trajectory =</span> <span class="st">"Myeloid"</span>, <span class="dt">colorBy =</span> <span class="st">"MotifMatrix"</span>, <span class="dt">name =</span> <span class="st">"z:CEBPB_140"</span>)
<span class="kw"><a href="../../reference/plotPDF.html">plotPDF</a></span>(<span class="dt">plotList =</span> plotList, <span class="dt">name =</span> <span class="st">"Plot-UMAP-Myeloid-Trajectory"</span>, <span class="dt">width =</span> <span class="dv">6</span>, <span class="dt">height =</span> <span class="dv">6</span>, <span class="dt">ArchRProj =</span> proj, <span class="dt">addDOC =</span> <span class="ot">FALSE</span>)

<span class="co">#Next we can access variable features across the trajectory and summarize</span>
TrajMyeloidMM &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/getTrajectory.html">getTrajectory</a></span>(<span class="dt">ArchRProj =</span> proj, <span class="dt">name =</span> <span class="st">"Myeloid"</span>, <span class="dt">useMatrix =</span> <span class="st">"MotifMatrix"</span>, <span class="dt">log2Norm =</span> <span class="ot">FALSE</span>)
TrajMyeloidGS &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/getTrajectory.html">getTrajectory</a></span>(<span class="dt">ArchRProj =</span> proj, <span class="dt">name =</span> <span class="st">"Myeloid"</span>, <span class="dt">useMatrix =</span> <span class="st">"GeneScoreMatrix"</span>)

<span class="co">#We can then plot a heatmap for this trajectory to see variable features</span>
plotList &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/list.html">list</a></span>()
plotList[[<span class="dv">1</span>]] &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/trajectoryHeatmap.html">trajectoryHeatmap</a></span>(TrajMyeloidMM)
plotList[[<span class="dv">2</span>]] &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/trajectoryHeatmap.html">trajectoryHeatmap</a></span>(TrajMyeloidGS)
<span class="kw"><a href="../../reference/plotPDF.html">plotPDF</a></span>(<span class="dt">plotList =</span> plotList, <span class="dt">name =</span> <span class="st">"Myeloid-Trajectory-Heatmap"</span>, <span class="dt">width =</span> <span class="dv">8</span>, <span class="dt">height =</span> <span class="dv">12</span>, <span class="dt">ArchRProj =</span> proj, <span class="dt">addDOC =</span> <span class="ot">FALSE</span>)</code></pre></div>
<div id="trajectory-umaps" class="section level2 tabset tabset-fade tabset-pills">
<h2 class="hasAnchor">
<a href="#trajectory-umaps" class="anchor"></a>Trajectory UMAPs</h2>
<div id="section-71" class="section level3">
<h3 class="hasAnchor">
<a href="#section-71" class="anchor"></a>1</h3>
<div class="figure">
<img src="../../images/Tutorial2/Plot-UMAP-Myeloid-Trajectory_1.png" width="600" height="600">
</div>
</div>
<div id="section-72" class="section level3">
<h3 class="hasAnchor">
<a href="#section-72" class="anchor"></a>2</h3>
<div class="figure">
<img src="../../images/Tutorial2/Plot-UMAP-Myeloid-Trajectory_2.png" width="600" height="600">
</div>
</div>
<div id="section-73" class="section level3">
<h3 class="hasAnchor">
<a href="#section-73" class="anchor"></a>3</h3>
<div class="figure">
<img src="../../images/Tutorial2/Plot-UMAP-Myeloid-Trajectory_3.png" width="600" height="600">
</div>
</div>
<div id="section-74" class="section level3">
<h3 class="hasAnchor">
<a href="#section-74" class="anchor"></a>4</h3>
<div class="figure">
<img src="../../images/Tutorial2/Plot-UMAP-Myeloid-Trajectory_4.png" width="600" height="600">
</div>
</div>
<div id="section-75" class="section level3">
<h3 class="hasAnchor">
<a href="#section-75" class="anchor"></a>5</h3>
<div class="figure">
<img src="../../images/Tutorial2/Plot-UMAP-Myeloid-Trajectory_5.png" width="600" height="600">
</div>
</div>
<div id="section-76" class="section level3">
<h3 class="hasAnchor">
<a href="#section-76" class="anchor"></a>6</h3>
<div class="figure">
<img src="../../images/Tutorial2/Plot-UMAP-Myeloid-Trajectory_6.png" width="600" height="600">
</div>
</div>
<div id="section-77" class="section level3">
<h3 class="hasAnchor">
<a href="#section-77" class="anchor"></a>7</h3>
<div class="figure">
<img src="../../images/Tutorial2/Plot-UMAP-Myeloid-Trajectory_7.png" width="600" height="600">
</div>
</div>
</div>
<div id="trajectory-heatmaps" class="section level2 tabset tabset-fade tabset-pills">
<h2 class="hasAnchor">
<a href="#trajectory-heatmaps" class="anchor"></a>Trajectory Heatmaps</h2>
<div id="section-78" class="section level3">
<h3 class="hasAnchor">
<a href="#section-78" class="anchor"></a>1</h3>
<div class="figure">
<img src="../../images/Tutorial2/Myeloid-Trajectory-Heatmap_1.png" width="600" height="900">
</div>
</div>
<div id="section-79" class="section level3">
<h3 class="hasAnchor">
<a href="#section-79" class="anchor"></a>2</h3>
<div class="figure">
<img src="../../images/Tutorial2/Myeloid-Trajectory-Heatmap_2.png" width="600" height="900">
</div>
</div>
</div>
</div>
<div id="references" class="section level1">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>13. References</h1>
<ol style="list-style-type: decimal">
<li>
<li>
<li>
<li>
<li>
</ol>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#what-is-an-arrowfile-archrproject">1. What is an <code>ArrowFile</code> / <code>ArchRProject</code>?</a></li>
      <li><a href="#getting-set-up">2. Getting Set Up</a></li>
      <li>
<a href="#creating-arrow-files">3. Creating Arrow Files</a><ul class="nav nav-pills nav-stacked">
<li><a href="#qc-tss-scores-by-unique-fragments">QC TSS Scores by Unique Fragments</a></li>
      <li><a href="#qc-fragment-size-distribution">QC Fragment Size Distribution</a></li>
      </ul>
</li>
      <li>
<a href="#tidying-up-our-data-and-creating-an-archrproject">4. Tidying up our data and creating an <code class="tabset tabset-fade tabset-pills">ArchRProject</code></a><ul class="nav nav-pills nav-stacked">
<li><a href="#example-doublet-results">Example Doublet Results</a></li>
      <li><a href="#tss-doublet-enrichment">TSS / Doublet Enrichment</a></li>
      </ul>
</li>
      <li>
<a href="#dimensionality-reduction">5. Dimensionality Reduction</a><ul class="nav nav-pills nav-stacked">
<li><a href="#umap-w-clusters">UMAP w/ Clusters</a></li>
      <li><a href="#tsne-w-clusters">TSNE w/ Clusters</a></li>
      <li><a href="#umap-w-custom-coldata">UMAP w/ Custom ColData</a></li>
      </ul>
</li>
      <li>
<a href="#identifying-cluster-cell-types-using-marker-genes">6. Identifying Cluster Cell Types Using Marker Genes</a><ul class="nav nav-pills nav-stacked">
<li><a href="#umap-genescores-w-imputation">UMAP GeneScores w/ Imputation</a></li>
      <li><a href="#umap-genescores-wo-imputation">UMAP GeneScores w/o Imputation</a></li>
      <li><a href="#track-plots">Track-Plots</a></li>
      <li><a href="#marker-genescores-heatmap">Marker GeneScores Heatmap</a></li>
      </ul>
</li>
      <li>
<a href="#creating-a-reproducible-peak-set">7. Creating a Reproducible Peak Set</a><ul class="nav nav-pills nav-stacked">
<li><a href="#reproducible-peaks-per-cluster">Reproducible Peaks per Cluster</a></li>
      <li><a href="#frip-tss-enrichment-per-sample">FRIP / TSS Enrichment Per Sample</a></li>
      </ul>
</li>
      <li>
<a href="#identifying-marker-peaks">8. Identifying Marker Peaks</a><ul class="nav nav-pills nav-stacked">
<li><a href="#markers-heatmap">Markers Heatmap</a></li>
      <li><a href="#markers-mavolcano-plot">Markers MA/Volcano-Plot</a></li>
      <li><a href="#track-of-peak-markers">Track of Peak Markers</a></li>
      <li><a href="#pairwise-test-between-groups">Pairwise-Test Between Groups</a></li>
      </ul>
</li>
      <li>
<a href="#performing-peak-annotation-enrichments">9. Performing Peak Annotation Enrichments</a><ul class="nav nav-pills nav-stacked">
<li><a href="#motif-enrichment">Motif Enrichment</a></li>
      <li><a href="#region-enrichment">Region Enrichment</a></li>
      </ul>
</li>
      <li>
<a href="#computing-chromvar-deviations">10. Computing chromVAR Deviations</a><ul class="nav nav-pills nav-stacked">
<li><a href="#motif-deviations">Motif Deviations</a></li>
      <li><a href="#region-deviations">Region Deviations</a></li>
      </ul>
</li>
      <li>
<a href="#performing-tf-footprinting">11. Performing TF Footprinting</a><ul class="nav nav-pills nav-stacked">
<li><a href="#no-normalization">No Normalization</a></li>
      <li><a href="#divide-bias">Divide Bias</a></li>
      <li><a href="#tss-enrichment">TSS Enrichment</a></li>
      </ul>
</li>
      <li>
<a href="#analyze-trajectories">12. Analyze Trajectories</a><ul class="nav nav-pills nav-stacked">
<li><a href="#trajectory-umaps">Trajectory UMAPs</a></li>
      <li><a href="#trajectory-heatmaps">Trajectory Heatmaps</a></li>
      </ul>
</li>
      <li><a href="#references">13. References</a></li>
      </ul>
</div>
      </div>

</div>



      <footer><div class="copyright">
  <p>Developed by <a href="https://twitter.com/JeffreyGranja">Jeffrey Granja</a>, <a href="https://twitter.com/doctorcorces">Ryan Corces</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.4.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
