<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>A Brief Tutorial of ArchR • ArchR</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.3.7/yeti/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../../pkgdown.css" rel="stylesheet">
<script src="../../pkgdown.js"></script><!-- docsearch --><script src="../../docsearch.js"></script><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.1/docsearch.min.css" integrity="sha256-QOSRU/ra9ActyXkIBbiIB144aDBdtvXBcNc3OTNuX/Q=" crossorigin="anonymous">
<link href="../../docsearch.css" rel="stylesheet">
<script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/jquery.mark.min.js" integrity="sha256-4HLtjeVgH0eIB3aZ9mLYF6E8oU5chNdjU6p6rrXpl9U=" crossorigin="anonymous"></script><link href="../../extra.css" rel="stylesheet">
<meta property="og:title" content="A Brief Tutorial of ArchR">
<meta property="og:description" content="">
<meta property="og:image" content="https://greenleaflab.github.io/ArchR_Website/logo.png">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-158769145-1"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-158769145-1');
</script>
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../../index.html">ArchR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.9.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../../index.html">
    <span class="fa fa-home"></span>
     
  </a>
</li>
<li>
  <a href="../../articles/Articles/tutorial.html">
    <span class="fa fa-graduation-cap"></span>
     
    Quick Tutorial
  </a>
</li>
<li>
  <a href="../../bookdown/index.html">
    <span class="fa fa-file-text-o"></span>
     
    Full Manual
  </a>
</li>
<li>
  <a href="../../articles/Articles/faq.html">
    <span class="fa fa-question-circle-o"></span>
     
    FAQ
  </a>
</li>
<li>
  <a href="../../reference/index.html">
    <span class="fa fa-code"></span>
     
    Functions
  </a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/GreenleafLab/ArchR">
    <span class="fa fa-github"></span>
     
    ArchR on GitHub
  </a>
</li>
      </ul>
<form class="navbar-form navbar-right hidden-xs hidden-sm" role="search">
        <div class="form-group">
          <input type="search" class="form-control" name="search-input" id="search-input" placeholder="Search..." aria-label="Search for..." autocomplete="off">
</div>
      </form>
      
    </div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>A Brief Tutorial of ArchR</h1>
            
      
      
      <div class="hidden name"><code>tutorial.Rmd</code></div>

    </div>

    
    
<p>This is a <strong><em>brief</em></strong> ArchR tutorial. It does not showcase all of ArchR’s functionality nor explain each of the steps in detail. Please see <a href="bookdown/index.html">the searchable full user’s manual</a> for an in-depth tutorial using real-world data and a full description of all aspects of ArchR analysis. <strong>Importantly, this tutorial and the full user manual assume that you are running ArchR locally</strong>.</p>
<hr>
<div id="getting-set-up" class="section level2">
<h2 class="hasAnchor">
<a href="#getting-set-up" class="anchor"></a>Getting Set Up</h2>
<p>First, we load the ArchR library. If this fails, you have not properly installed ArchR and should <a href="https://www.archrproject.com/index.html">revisit the installation instructions</a>. We also recommend setting and remembering a known seed to facilitate replication of operations requiring randomization.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1"></a><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(ArchR)</span>
<span id="cb1-2"><a href="#cb1-2"></a><span class="kw"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span>(<span class="dv">1</span>)</span></code></pre></div>
<p>Next, we set the default number of threads for parallelized operations in ArchR functions. You should change the value passed the <code>threads</code> to match the specifications of your local machine.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1"></a><span class="kw"><a href="../../reference/addArchRThreads.html">addArchRThreads</a></span>(<span class="dt">threads =</span> <span class="dv">16</span>) </span></code></pre></div>
<blockquote>
<p>## Setting default number of Parallel threads to 16.</p>
</blockquote>
<p>The Hematopoeisis tutorial data can be downloaded using the <code><a href="../../reference/getTutorialData.html">getTutorialData()</a></code> function. The tutorial data is approximately 0.5 GB in size. If you have already downloaded the tutorial in the current working directory, ArchR will bypass downloading.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1"></a>inputFiles &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/getTutorialData.html">getTutorialData</a></span>(<span class="st">"Hematopoiesis"</span>)</span>
<span id="cb3-2"><a href="#cb3-2"></a>inputFiles</span></code></pre></div>
<blockquote>
<p>## scATAC_BMMC_R1<br>
## “HemeFragments/scATAC_BMMC_R1.fragments.tsv.gz”<br>
## scATAC_CD34_BMMC_R1<br>
## “HemeFragments/scATAC_CD34_BMMC_R1.fragments.tsv.gz”<br>
## scATAC_PBMC_R1<br>
## “HemeFragments/scATAC_PBMC_R1.fragments.tsv.gz”</p>
</blockquote>
<p>Before we begin, we need add a reference genome annotation for ArchR to have access to chromosome and gene information. ArchR natively supports hg19, hg38, mm9, and mm10.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1"></a><span class="kw"><a href="../../reference/addArchRGenome.html">addArchRGenome</a></span>(<span class="st">"hg19"</span>)</span></code></pre></div>
<blockquote>
<p>## Setting default genome to Hg19.</p>
</blockquote>
</div>
<div id="creating-arrow-files" class="section level2">
<h2 class="hasAnchor">
<a href="#creating-arrow-files" class="anchor"></a>Creating Arrow Files</h2>
<p>Now we will create our Arrow files which will take 10-15 minutes. For each sample, this step will:</p>
<ol style="list-style-type: decimal">
<li>Read accessible fragments from the provided input files.</li>
<li>Calculate quality control information for each cell (i.e. TSS enrichment scores and nucleosome info).</li>
<li>Filter cells based on quality control parameters.</li>
<li>Create a genome-wide TileMatrix using 500-bp bins.</li>
<li>Create a GeneScoreMatrix using the custom <code>geneAnnotation</code> that was defined when we called <code><a href="../../reference/addArchRGenome.html">addArchRGenome()</a></code>.</li>
</ol>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1"></a>ArrowFiles &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/createArrowFiles.html">createArrowFiles</a></span>(</span>
<span id="cb5-2"><a href="#cb5-2"></a>  <span class="dt">inputFiles =</span> inputFiles,</span>
<span id="cb5-3"><a href="#cb5-3"></a>  <span class="dt">sampleNames =</span> <span class="kw"><a href="https://rdrr.io/r/base/names.html">names</a></span>(inputFiles),</span>
<span id="cb5-4"><a href="#cb5-4"></a>  <span class="dt">filterTSS =</span> <span class="dv">4</span>, <span class="co">#Dont set this too high because you can always increase later</span></span>
<span id="cb5-5"><a href="#cb5-5"></a>  <span class="dt">filterFrags =</span> <span class="dv">1000</span>, </span>
<span id="cb5-6"><a href="#cb5-6"></a>  <span class="dt">addTileMat =</span> <span class="ot">TRUE</span>,</span>
<span id="cb5-7"><a href="#cb5-7"></a>  <span class="dt">addGeneScoreMat =</span> <span class="ot">TRUE</span></span>
<span id="cb5-8"><a href="#cb5-8"></a>)</span></code></pre></div>
<blockquote>
<p>## Using GeneAnnotation set by addArchRGenome(Hg19)!<br>
## Using GeneAnnotation set by addArchRGenome(Hg19)!<br>
## ArchR logging to : ArchRLogs/ArchR-createArrows-69ef6ba2e1c7-Date-2020-04-21_Time-16-18-35.log<br>
## If there is an issue, please report to github with logFile!<br>
## Cleaning Temporary Files<br>
## 2020-04-21 16:18:35 : Batch Execution w/ safelapply!, 0 mins elapsed.<br>
## ArchR logging successful to : ArchRLogs/ArchR-createArrows-69ef6ba2e1c7-Date-2020-04-21_Time-16-18-35.log</p>
</blockquote>
<p>We can inspect the <code>ArrowFiles</code> object to see that it is actually just a character vector of Arrow file paths.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1"></a>ArrowFiles</span></code></pre></div>
<blockquote>
<p>## “scATAC_BMMC_R1.arrow” “scATAC_CD34_BMMC_R1.arrow”<br>
## “scATAC_PBMC_R1.arrow”</p>
</blockquote>
<p>Strict quality control (QC) of scATAC-seq data is essential to remove the contribution of low-quality cells. In ArchR, we consider three characteristics of data:</p>
<ol style="list-style-type: decimal">
<li>The number of unique nuclear fragments (i.e. not mapping to mitochondrial DNA).</li>
<li>The signal-to-background ratio. Low signal-to-background ratio is often attributed to dead or dying cells which have de-chromatinzed DNA which allows for random transposition genome-wide.</li>
<li>The fragment size distribution. Due to nucleosomal periodicity, we expect to see depletion of fragments that are the length of DNA wrapped around a nucleosome (approximately 147 bp).</li>
</ol>
<p>For our tutorial data, an example of these QC plots is shown below for the <strong>BMMC</strong> sample:</p>
<p><img src="../../images/HemeShort/PNG/scATAC_BMMC_R1-TSS_by_Unique_Frags_1.png" width="330" height="330"><img src="../../images/HemeShort/PNG/scATAC_BMMC_R1-Fragment_Size_Distribution_1.png" width="430" height="330"></p>
</div>
<div id="inferring-doublets" class="section level2">
<h2 class="hasAnchor">
<a href="#inferring-doublets" class="anchor"></a>Inferring Doublets</h2>
<p>After Arrow file creation, we can infer potential doublets (a single droplet containing multiple cells) that can confound downstream results. This is done using the <code><a href="../../reference/addDoubletScores.html">addDoubletScores()</a></code> function.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1"></a>doubScores &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/addDoubletScores.html">addDoubletScores</a></span>(</span>
<span id="cb7-2"><a href="#cb7-2"></a>  <span class="dt">input =</span> ArrowFiles,</span>
<span id="cb7-3"><a href="#cb7-3"></a>  <span class="dt">k =</span> <span class="dv">10</span>, <span class="co">#Refers to how many cells near a "pseudo-doublet" to count.</span></span>
<span id="cb7-4"><a href="#cb7-4"></a>  <span class="dt">knnMethod =</span> <span class="st">"UMAP"</span>, <span class="co">#Refers to the embedding to use for nearest neighbor search.</span></span>
<span id="cb7-5"><a href="#cb7-5"></a>  <span class="dt">LSIMethod =</span> <span class="dv">1</span></span>
<span id="cb7-6"><a href="#cb7-6"></a>)</span></code></pre></div>
<blockquote>
<p>## ArchR logging to : ArchRLogs/ArchR-addDoubletScores-69ef56ebfd4d-Date-2020-04-21_Time-16-24-31.log<br>
## If there is an issue, please report to github with logFile!<br>
## 2020-04-21 16:24:31 : Batch Execution w/ safelapply!, 0 mins elapsed.<br>
## 2020-04-21 16:24:31 : scATAC_BMMC_R1 (1 of 3) : Computing Doublet Statistics, 0 mins elapsed.<br>
## scATAC_BMMC_R1 (1 of 3) : UMAP Projection R^2 = 0.97229<br>
## scATAC_BMMC_R1 (1 of 3) : UMAP Projection R^2 = 0.97229<br>
## 2020-04-21 16:26:31 : scATAC_CD34_BMMC_R1 (2 of 3) : Computing Doublet Statistics, 2.007 mins elapsed.<br>
## scATAC_CD34_BMMC_R1 (2 of 3) : UMAP Projection R^2 = 0.99077<br>
## scATAC_CD34_BMMC_R1 (2 of 3) : UMAP Projection R^2 = 0.99077<br>
## 2020-04-21 16:28:03 : scATAC_PBMC_R1 (3 of 3) : Computing Doublet Statistics, 3.531 mins elapsed.<br>
## scATAC_PBMC_R1 (3 of 3) : UMAP Projection R^2 = 0.97496<br>
## scATAC_PBMC_R1 (3 of 3) : UMAP Projection R^2 = 0.97496<br>
## ArchR logging successful to : ArchRLogs/ArchR-addDoubletScores-69ef56ebfd4d-Date-2020-04-21_Time-16-24-31.log</p>
</blockquote>
</div>
<div id="creating-an-archrproject" class="section level2">
<h2 class="hasAnchor">
<a href="#creating-an-archrproject" class="anchor"></a>Creating an <code>ArchRProject</code>
</h2>
<p>With our Arrow files in hand, we are now ready to create an <code>ArchRProject</code>. An <code>ArchRProject</code> is associated with a set of Arrow files and is the backbone of nearly all ArchR analyses.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1"></a>proj &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/ArchRProject.html">ArchRProject</a></span>(</span>
<span id="cb8-2"><a href="#cb8-2"></a>  <span class="dt">ArrowFiles =</span> ArrowFiles, </span>
<span id="cb8-3"><a href="#cb8-3"></a>  <span class="dt">outputDirectory =</span> <span class="st">"HemeTutorial"</span>,</span>
<span id="cb8-4"><a href="#cb8-4"></a>  <span class="dt">copyArrows =</span> <span class="ot">TRUE</span> <span class="co">#This is recommened so that you maintain an unaltered copy for later usage.</span></span>
<span id="cb8-5"><a href="#cb8-5"></a>)</span></code></pre></div>
<blockquote>
<p>## Using GeneAnnotation set by addArchRGenome(Hg19)!<br>
## Using GeneAnnotation set by addArchRGenome(Hg19)!<br>
## Validating Arrows…<br>
## Getting SampleNames…<br>
##<br>
## Copying ArrowFiles to Ouptut Directory! If you want to save disk space set copyArrows = FALSE<br>
## 1 2 3<br>
## Getting Cell Metadata…<br>
##<br>
## Merging Cell Metadata…<br>
## Initializing ArchRProject…</p>
</blockquote>
<p>We can also ask which data matrices are available within the <code>ArchRProject</code> which will be useful downstream once we start adding to this project:</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1"></a><span class="kw"><a href="../../reference/getAvailableMatrices.html">getAvailableMatrices</a></span>(proj)</span></code></pre></div>
<blockquote>
<p>## [1] “GeneScoreMatrix” “TileMatrix”</p>
</blockquote>
<p>Now we can filter putative doublets based on the previously determined doublet scores using the <code><a href="../../reference/filterDoublets.html">filterDoublets()</a></code> function. This doesn’t physically remove data from the Arrow files but rather tells the <code>ArchRProject</code> to ignore these cells for downstream analysis.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1"></a>proj &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/filterDoublets.html">filterDoublets</a></span>(<span class="dt">ArchRProj =</span> proj)</span></code></pre></div>
<blockquote>
<p>## Filtering 410 cells from ArchRProject!<br>
## scATAC_BMMC_R1 : 243 of 4932 (4.9%)<br>
## scATAC_CD34_BMMC_R1 : 107 of 3275 (3.3%)<br>
## scATAC_PBMC_R1 : 60 of 2454 (2.4%)</p>
</blockquote>
</div>
<div id="dimensionality-reduction-and-clustering" class="section level2">
<h2 class="hasAnchor">
<a href="#dimensionality-reduction-and-clustering" class="anchor"></a>Dimensionality Reduction and Clustering</h2>
<p>ArchR implements an iterative LSI dimensionality reduction via the <code><a href="../../reference/addIterativeLSI.html">addIterativeLSI()</a></code> function.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1"></a>proj &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/addIterativeLSI.html">addIterativeLSI</a></span>(<span class="dt">ArchRProj =</span> proj, <span class="dt">useMatrix =</span> <span class="st">"TileMatrix"</span>, <span class="dt">name =</span> <span class="st">"IterativeLSI"</span>)</span></code></pre></div>
<blockquote>
<p>## Checking Inputs…<br>
## ArchR logging to : ArchRLogs/ArchR-addIterativeLSI-69ef14e7f129-Date-2020-04-21_Time-16-29-45.log<br>
## If there is an issue, please report to github with logFile!<br>
## 2020-04-21 16:29:46 : Computing Total Accessibility Across All Features, 0.005 mins elapsed.<br>
## 2020-04-21 16:29:49 : Computing Top Features, 0.068 mins elapsed.<br>
## ###########<br>
## 2020-04-21 16:29:50 : Running LSI (1 of 2) on Top Features, 0.077 mins elapsed.<br>
## ###########<br>
## 2020-04-21 16:29:50 : Sampling Cells (N = 10002) for Estimated LSI, 0.078 mins elapsed.<br>
## 2020-04-21 16:29:50 : Creating Sampled Partial Matrix, 0.078 mins elapsed.<br>
## 2020-04-21 16:29:58 : Computing Estimated LSI (projectAll = FALSE), 0.208 mins elapsed.<br>
## 2020-04-21 16:30:38 : Identifying Clusters, 0.876 mins elapsed.<br>
## 2020-04-21 16:31:00 : Identified 5 Clusters, 1.241 mins elapsed.<br>
## 2020-04-21 16:31:00 : Saving LSI Iteration, 1.242 mins elapsed.<br>
## 2020-04-21 16:31:21 : Creating Cluster Matrix on the total Group Features, 1.593 mins elapsed.<br>
## 2020-04-21 16:31:31 : Computing Variable Features, 1.759 mins elapsed.<br>
## ###########<br>
## 2020-04-21 16:31:31 : Running LSI (2 of 2) on Variable Features, 1.764 mins elapsed.<br>
## ###########<br>
## 2020-04-21 16:31:31 : Creating Partial Matrix, 1.764 mins elapsed.<br>
## 2020-04-21 16:31:40 : Computing LSI, 1.911 mins elapsed.<br>
## 2020-04-21 16:32:18 : Finished Running IterativeLSI, 2.541 mins elapsed.</p>
</blockquote>
<p>To call clusters in this reduced dimension sub-space, we use the <code><a href="../../reference/addClusters.html">addClusters()</a></code> function which uses <a href="https://satijalab.org/seurat/">Seurat’s</a> graph clustering as the default clustering method.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1"></a>proj &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/addClusters.html">addClusters</a></span>(<span class="dt">input =</span> proj, <span class="dt">reducedDims =</span> <span class="st">"IterativeLSI"</span>)</span></code></pre></div>
<blockquote>
<p>## ArchR logging to : ArchRLogs/ArchR-addClusters-69ef73b1d963-Date-2020-04-21_Time-16-32-18.log<br>
## If there is an issue, please report to github with logFile!<br>
## 2020-04-21 16:32:19 : Running Seurats FindClusters (Stuart et al. Cell 2019), 0.004 mins elapsed.<br>
## Computing nearest neighbor graph<br>
## Computing SNN<br>
## Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck<br>
##<br>
## Number of nodes: 10251<br>
## Number of edges: 498555<br>
##<br>
## Running Louvain algorithm…<br>
## Maximum modularity in 10 random starts: 0.8575<br>
## Number of communities: 11<br>
## Elapsed time: 0 seconds<br>
## 2020-04-21 16:32:38 : Testing Outlier Clusters, 0.32 mins elapsed.<br>
## 2020-04-21 16:32:38 : Assigning Cluster Names to 11 Clusters, 0.32 mins elapsed.<br>
## 2020-04-21 16:32:38 : Finished addClusters, 0.322 mins elapsed.</p>
</blockquote>
</div>
<div id="visualizing-in-a-2d-umap-embedding" class="section level2">
<h2 class="hasAnchor">
<a href="#visualizing-in-a-2d-umap-embedding" class="anchor"></a>Visualizing in a 2D UMAP Embedding</h2>
<p>We can visualize our scATAC-seq data using a 2-dimensional representation such as Uniform Manifold Approximation and Projection (UMAP). To do this, we add a UMAP embedding to our <code>ArchRProject</code> object with the <code><a href="../../reference/addUMAP.html">addUMAP()</a></code> function. This function uses the <a href="https://github.com/jlmelville/uwot">uwot package</a> to perform UMAP.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1"></a>proj &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/addUMAP.html">addUMAP</a></span>(<span class="dt">ArchRProj =</span> proj, <span class="dt">reducedDims =</span> <span class="st">"IterativeLSI"</span>)</span></code></pre></div>
<blockquote>
<p>## 16:32:38 UMAP embedding parameters a = 0.7669 b = 1.223<br>
## 16:32:38 Read 10251 rows and found 30 numeric columns<br>
## 16:32:38 Using Annoy for neighbor search, n_neighbors = 40<br>
## 16:32:38 Building Annoy index with metric = cosine, n_trees = 50<br>
## 0% 10 20 30 40 50 60 70 80 90 100%<br>
## [—-|—-|—-|—-|—-|—-|—-|—-|—-|—-|<br>
## **************************************************|<br>
## 16:32:41 Writing NN index file to temp file /tmp/RtmpJ4Z9d9/file69ef223c61d2<br>
## 16:32:41 Searching Annoy index using 10 threads, search_k = 4000<br>
## 16:32:42 Annoy recall = 100%<br>
## 16:32:44 Commencing smooth kNN distance calibration using 10 threads<br>
## 16:32:45 Initializing from normalized Laplacian + noise<br>
## 16:32:46 Commencing optimization for 200 epochs, with 623598 positive edges<br>
## 16:32:58 Optimization finished</p>
</blockquote>
<p>Using this UMAP, we can visualize various attributes of our cells which are stored in a matrix called <code>cellColData</code> in our <code>ArchRProject</code>. To do this, we use the <code><a href="../../reference/plotEmbedding.html">plotEmbedding()</a></code> function and we specify the variable to use for coloration via a combination of the <code>colorBy</code> and <code>name</code> parameters.</p>
<p>For example, we can color by “Sample”:</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1"></a>p1 &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/plotEmbedding.html">plotEmbedding</a></span>(<span class="dt">ArchRProj =</span> proj, <span class="dt">colorBy =</span> <span class="st">"cellColData"</span>, <span class="dt">name =</span> <span class="st">"Sample"</span>, <span class="dt">embedding =</span> <span class="st">"UMAP"</span>)</span></code></pre></div>
<blockquote>
<p>## ArchR logging to : ArchRLogs/ArchR-plotEmbedding-69ef25160449-Date-2020-04-21_Time-16-33-00.log<br>
## If there is an issue, please report to github with logFile!<br>
## Getting UMAP Embedding<br>
## ColorBy = cellColData<br>
## Plotting Embedding<br>
## 1<br>
## ArchR logging successful to : ArchRLogs/ArchR-plotEmbedding-69ef25160449-Date-2020-04-21_Time-16-33-00.log</p>
</blockquote>
<p>Or we can color by “Clusters”:</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1"></a>p2 &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/plotEmbedding.html">plotEmbedding</a></span>(<span class="dt">ArchRProj =</span> proj, <span class="dt">colorBy =</span> <span class="st">"cellColData"</span>, <span class="dt">name =</span> <span class="st">"Clusters"</span>, <span class="dt">embedding =</span> <span class="st">"UMAP"</span>)</span></code></pre></div>
<blockquote>
<p>## ArchR logging to : ArchRLogs/ArchR-plotEmbedding-69ef420f8c0a-Date-2020-04-21_Time-16-33-01.log<br>
## If there is an issue, please report to github with logFile!<br>
## Getting UMAP Embedding<br>
## ColorBy = cellColData<br>
## Plotting Embedding<br>
## 1<br>
## ArchR logging successful to : ArchRLogs/ArchR-plotEmbedding-69ef420f8c0a-Date-2020-04-21_Time-16-33-01.log</p>
</blockquote>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1"></a><span class="kw"><a href="../../reference/ggAlignPlots.html">ggAlignPlots</a></span>(p1, p2, <span class="dt">type =</span> <span class="st">"h"</span>)</span></code></pre></div>
<p><img src="../../images/HemeShort/PNG/Plot-UMAP-Sample-Clusters-Combined_1.png" width="900"></p>
<p>To save an editable vectorized version of this plot, we use the <code><a href="../../reference/plotPDF.html">plotPDF()</a></code> function.</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1"></a><span class="kw"><a href="../../reference/plotPDF.html">plotPDF</a></span>(p1,p2, <span class="dt">name =</span> <span class="st">"Plot-UMAP-Sample-Clusters.pdf"</span>,</span>
<span id="cb17-2"><a href="#cb17-2"></a>        <span class="dt">ArchRProj =</span> proj, <span class="dt">addDOC =</span> <span class="ot">FALSE</span>, <span class="dt">width =</span> <span class="dv">5</span>, <span class="dt">height =</span> <span class="dv">5</span>)</span></code></pre></div>
<blockquote>
<p>## [1] “plotting ggplot!”<br>
## [1] “plotting ggplot!”<br>
## [1] 0</p>
</blockquote>
<p><a href="../../images/HemeShort/PDF/Plot-UMAP-Sample-Clusters.pdf">Download PDF : Plot-UMAP-Sample-Clusters.pdf</a></p>
</div>
<div id="assigning-clusters-with-gene-scores" class="section level2">
<h2 class="hasAnchor">
<a href="#assigning-clusters-with-gene-scores" class="anchor"></a>Assigning Clusters with Gene Scores</h2>
<p>We can try to assign biological labels to these clusters using marker genes of known hematopoietic regulators. First, we add imputation weights using <a href="https://github.com/KrishnaswamyLab/MAGIC">MAGIC</a> to help smooth the dropout noise in our gene scores.</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1"></a>proj &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/addImputeWeights.html">addImputeWeights</a></span>(proj)</span></code></pre></div>
<blockquote>
<p>## ArchR logging to : ArchRLogs/ArchR-addImputeWeights-69ef433c71d0-Date-2020-04-21_Time-16-33-19.log<br>
## If there is an issue, please report to github with logFile!<br>
## 2020-04-21 16:33:19 : Computing Impute Weights Using Magic (Cell 2018), 0 mins elapsed.</p>
</blockquote>
<p>Now we can overlay our marker gene scores on our 2D UMAP embedding.</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1"></a>markerGenes  &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(</span>
<span id="cb19-2"><a href="#cb19-2"></a>    <span class="st">"CD34"</span>,  <span class="co">#Early Progenitor</span></span>
<span id="cb19-3"><a href="#cb19-3"></a>    <span class="st">"GATA1"</span>, <span class="co">#Erythroid</span></span>
<span id="cb19-4"><a href="#cb19-4"></a>    <span class="st">"PAX5"</span>, <span class="st">"MS4A1"</span>, <span class="st">"MME"</span>, <span class="co">#B-Cell Trajectory</span></span>
<span id="cb19-5"><a href="#cb19-5"></a>    <span class="st">"CD14"</span>, <span class="st">"MPO"</span>, <span class="co">#Monocytes</span></span>
<span id="cb19-6"><a href="#cb19-6"></a>    <span class="st">"CD3D"</span>, <span class="st">"CD8A"</span><span class="co">#TCells</span></span>
<span id="cb19-7"><a href="#cb19-7"></a>  )</span>
<span id="cb19-8"><a href="#cb19-8"></a></span>
<span id="cb19-9"><a href="#cb19-9"></a>p &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/plotEmbedding.html">plotEmbedding</a></span>(</span>
<span id="cb19-10"><a href="#cb19-10"></a>    <span class="dt">ArchRProj =</span> proj, </span>
<span id="cb19-11"><a href="#cb19-11"></a>    <span class="dt">colorBy =</span> <span class="st">"GeneScoreMatrix"</span>, </span>
<span id="cb19-12"><a href="#cb19-12"></a>    <span class="dt">name =</span> markerGenes, </span>
<span id="cb19-13"><a href="#cb19-13"></a>    <span class="dt">embedding =</span> <span class="st">"UMAP"</span>,</span>
<span id="cb19-14"><a href="#cb19-14"></a>    <span class="dt">imputeWeights =</span> <span class="kw"><a href="../../reference/getImputeWeights.html">getImputeWeights</a></span>(proj)</span>
<span id="cb19-15"><a href="#cb19-15"></a>)</span></code></pre></div>
<blockquote>
<p>## Getting ImputeWeights<br>
## ArchR logging to : ArchRLogs/ArchR-plotEmbedding-69ef2f10a62d-Date-2020-04-21_Time-16-33-31.log<br>
## If there is an issue, please report to github with logFile!<br>
## Getting UMAP Embedding<br>
## ColorBy = GeneScoreMatrix<br>
## Getting Matrix Values…<br>
## 2020-04-21 16:33:32 :<br>
##<br>
## Imputing Matrix<br>
## Using weights on disk<br>
## Using weights on disk<br>
## Plotting Embedding<br>
## 1 2 3 4 5 6 7 8 9<br>
## ArchR logging successful to : ArchRLogs/ArchR-plotEmbedding-69ef2f10a62d-Date-2020-04-21_Time-16-33-31.log</p>
</blockquote>
<p>To plot a specific gene we can subset this plot list using the gene name.</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1"></a>p<span class="op">$</span>CD14</span></code></pre></div>
<center>
<img src="../../images/HemeShort/PNG/Plot-UMAP-Marker-Genes-W-Imputation_6.png" width="500" height="500">
</center>
<p>To plot all genes we can use <code>cowplot</code> to arrange the 9 different plots together. Each of these marker genes lights up the corresponding cell clusters. For example, we infer that the cells that have the highest gene score for CD3D, a known T cell marker, are in fact T cells.</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1"></a><span class="co">#Rearrange for grid plotting</span></span>
<span id="cb21-2"><a href="#cb21-2"></a>p2 &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span>(p, <span class="cf">function</span>(x){</span>
<span id="cb21-3"><a href="#cb21-3"></a>    x <span class="op">+</span><span class="st"> </span><span class="kw">guides</span>(<span class="dt">color =</span> <span class="ot">FALSE</span>, <span class="dt">fill =</span> <span class="ot">FALSE</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb21-4"><a href="#cb21-4"></a><span class="st">    </span><span class="kw"><a href="../../reference/theme_ArchR.html">theme_ArchR</a></span>(<span class="dt">baseSize =</span> <span class="fl">6.5</span>) <span class="op">+</span></span>
<span id="cb21-5"><a href="#cb21-5"></a><span class="st">    </span><span class="kw">theme</span>(<span class="dt">plot.margin =</span> <span class="kw">unit</span>(<span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>), <span class="st">"cm"</span>)) <span class="op">+</span></span>
<span id="cb21-6"><a href="#cb21-6"></a><span class="st">    </span><span class="kw">theme</span>(</span>
<span id="cb21-7"><a href="#cb21-7"></a>        <span class="dt">axis.text.x=</span><span class="kw">element_blank</span>(), </span>
<span id="cb21-8"><a href="#cb21-8"></a>        <span class="dt">axis.ticks.x=</span><span class="kw">element_blank</span>(), </span>
<span id="cb21-9"><a href="#cb21-9"></a>        <span class="dt">axis.text.y=</span><span class="kw">element_blank</span>(), </span>
<span id="cb21-10"><a href="#cb21-10"></a>        <span class="dt">axis.ticks.y=</span><span class="kw">element_blank</span>()</span>
<span id="cb21-11"><a href="#cb21-11"></a>    )</span>
<span id="cb21-12"><a href="#cb21-12"></a>})</span>
<span id="cb21-13"><a href="#cb21-13"></a><span class="kw"><a href="https://rdrr.io/r/base/do.call.html">do.call</a></span>(cowplot<span class="op">::</span>plot_grid, <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/list.html">list</a></span>(<span class="dt">ncol =</span> <span class="dv">3</span>),p2))</span></code></pre></div>
<center>
<img src="../../images/HemeShort/PNG/Plot-UMAP-Markers-W-Imputation_1.png" width="900" height="900">
</center>
<p>To save an editable vectorized version of this plot, we use the <code><a href="../../reference/plotPDF.html">plotPDF()</a></code> function.</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1"></a><span class="kw"><a href="../../reference/plotPDF.html">plotPDF</a></span>(<span class="dt">plotList =</span> p, </span>
<span id="cb22-2"><a href="#cb22-2"></a>    <span class="dt">name =</span> <span class="st">"Plot-UMAP-Marker-Genes-W-Imputation.pdf"</span>, </span>
<span id="cb22-3"><a href="#cb22-3"></a>    <span class="dt">ArchRProj =</span> proj, </span>
<span id="cb22-4"><a href="#cb22-4"></a>    <span class="dt">addDOC =</span> <span class="ot">FALSE</span>, <span class="dt">width =</span> <span class="dv">5</span>, <span class="dt">height =</span> <span class="dv">5</span>)</span></code></pre></div>
<blockquote>
<p>## [1] “plotting ggplot!”<br>
## [1] “plotting ggplot!”<br>
## [1] “plotting ggplot!”<br>
## [1] “plotting ggplot!”<br>
## [1] “plotting ggplot!”<br>
## [1] “plotting ggplot!”<br>
## [1] “plotting ggplot!”<br>
## [1] “plotting ggplot!”<br>
## [1] “plotting ggplot!”<br>
## [1] 0</p>
</blockquote>
<p><a href="../../images/HemeShort/PDF/Plot-UMAP-Marker-Genes-W-Imputation.pdf">Download PDF : Plot-UMAP-Marker-Genes-W-Imputation.pdf</a></p>
</div>
<div id="visualizing-genome-browser-tracks" class="section level2">
<h2 class="hasAnchor">
<a href="#visualizing-genome-browser-tracks" class="anchor"></a>Visualizing Genome Browser Tracks</h2>
<p>In addition to plotting gene scores per cell as a UMAP overlay, we can browse the local chromatin accessibility at these marker genes on a per cluster basis with genome browser tracks. To do this, we use the <code><a href="../../reference/plotBrowserTrack.html">plotBrowserTrack()</a></code> function which will create a list of plots, one for each of the genes specified by <code>markerGenes</code>.</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1"></a>p &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/plotBrowserTrack.html">plotBrowserTrack</a></span>(</span>
<span id="cb23-2"><a href="#cb23-2"></a>    <span class="dt">ArchRProj =</span> proj, </span>
<span id="cb23-3"><a href="#cb23-3"></a>    <span class="dt">groupBy =</span> <span class="st">"Clusters"</span>, </span>
<span id="cb23-4"><a href="#cb23-4"></a>    <span class="dt">geneSymbol =</span> markerGenes, </span>
<span id="cb23-5"><a href="#cb23-5"></a>    <span class="dt">upstream =</span> <span class="dv">50000</span>,</span>
<span id="cb23-6"><a href="#cb23-6"></a>    <span class="dt">downstream =</span> <span class="dv">50000</span></span>
<span id="cb23-7"><a href="#cb23-7"></a>)</span></code></pre></div>
<blockquote>
<p>## ArchR logging to : ArchRLogs/ArchR-plotBrowserTrack-69ef1ca18f89-Date-2020-04-21_Time-16-34-17.log<br>
## If there is an issue, please report to github with logFile!<br>
## 2020-04-21 16:34:17 : Validating Region, 0.006 mins elapsed.<br>
## GRanges object with 9 ranges and 2 metadata columns:<br>
## seqnames ranges strand | gene_id symbol<br>
## <rle><iranges><rle> | <character><character><br>
## [1] chr1 208059883-208084683 - | 947 CD34<br>
## [2] chrX 48644982-48652717 + | 2623 GATA1<br>
## [3] chr9 36838531-37034476 - | 5079 PAX5<br>
## [4] chr11 60223282-60238225 + | 931 MS4A1<br>
## [5] chr3 154741913-154901518 + | 4311 MME<br>
## [6] chr5 140011313-140013286 - | 929 CD14<br>
## [7] chr17 56347217-56358296 - | 4353 MPO<br>
## [8] chr11 118209789-118213459 - | 915 CD3D<br>
## [9] chr2 87011728-87035519 - | 925 CD8A<br>
## ——-<br>
## seqinfo: 24 sequences from hg19 genome<br>
## 2020-04-21 16:34:17 : Adding Bulk Tracks (1 of 9), 0.007 mins elapsed.<br>
## 2020-04-21 16:34:19 : Adding Gene Tracks (1 of 9), 0.033 mins elapsed.<br>
## 2020-04-21 16:34:19 : Plotting, 0.043 mins elapsed.<br>
## 2020-04-21 16:34:22 : Adding Bulk Tracks (2 of 9), 0.084 mins elapsed.<br>
## 2020-04-21 16:34:22 : Adding Gene Tracks (2 of 9), 0.099 mins elapsed.<br>
## 2020-04-21 16:34:23 : Plotting, 0.105 mins elapsed.<br>
## 2020-04-21 16:34:26 : Adding Bulk Tracks (3 of 9), 0.156 mins elapsed.<br>
## 2020-04-21 16:34:27 : Adding Gene Tracks (3 of 9), 0.174 mins elapsed.<br>
## 2020-04-21 16:34:27 : Plotting, 0.182 mins elapsed.<br>
## 2020-04-21 16:34:30 : Adding Bulk Tracks (4 of 9), 0.219 mins elapsed.<br>
## 2020-04-21 16:34:31 : Adding Gene Tracks (4 of 9), 0.235 mins elapsed.<br>
## 2020-04-21 16:34:31 : Plotting, 0.242 mins elapsed.<br>
## 2020-04-21 16:34:32 : Adding Bulk Tracks (5 of 9), 0.265 mins elapsed.<br>
## 2020-04-21 16:34:34 : Adding Gene Tracks (5 of 9), 0.284 mins elapsed.<br>
## 2020-04-21 16:34:34 : Plotting, 0.291 mins elapsed.<br>
## 2020-04-21 16:34:36 : Adding Bulk Tracks (6 of 9), 0.318 mins elapsed.<br>
## 2020-04-21 16:34:37 : Adding Gene Tracks (6 of 9), 0.334 mins elapsed.<br>
## 2020-04-21 16:34:37 : Plotting, 0.339 mins elapsed.<br>
## 2020-04-21 16:34:39 : Adding Bulk Tracks (7 of 9), 0.374 mins elapsed.<br>
## 2020-04-21 16:34:40 : Adding Gene Tracks (7 of 9), 0.391 mins elapsed.<br>
## 2020-04-21 16:34:41 : Plotting, 0.401 mins elapsed.<br>
## 2020-04-21 16:34:44 : Adding Bulk Tracks (8 of 9), 0.454 mins elapsed.<br>
## 2020-04-21 16:34:45 : Adding Gene Tracks (8 of 9), 0.473 mins elapsed.<br>
## 2020-04-21 16:34:45 : Plotting, 0.482 mins elapsed.<br>
## 2020-04-21 16:34:47 : Adding Bulk Tracks (9 of 9), 0.515 mins elapsed.<br>
## 2020-04-21 16:34:49 : Adding Gene Tracks (9 of 9), 0.537 mins elapsed.<br>
## 2020-04-21 16:34:49 : Plotting, 0.546 mins elapsed.<br>
## ArchR logging successful to : ArchRLogs/ArchR-plotBrowserTrack-69ef1ca18f89-Date-2020-04-21_Time-16-34-17.log</character></character></rle></iranges></rle></p>
</blockquote>
<p>To plot a track of a specific gene, we can simply select one from the list.</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1"></a>grid<span class="op">::</span><span class="kw"><a href="https://rdrr.io/r/grid/grid.newpage.html">grid.newpage</a></span>()</span>
<span id="cb24-2"><a href="#cb24-2"></a>grid<span class="op">::</span><span class="kw"><a href="https://rdrr.io/r/grid/grid.draw.html">grid.draw</a></span>(p<span class="op">$</span>CD14)</span></code></pre></div>
<center>
<img src="../../images/HemeShort/PNG/Plot-Tracks-Marker-Genes_6.png" width="700">
</center>
<p>We can save a multi-page PDF with a single page for each gene locus in our plot list using the <code><a href="../../reference/plotPDF.html">plotPDF()</a></code> function.</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1"></a><span class="kw"><a href="../../reference/plotPDF.html">plotPDF</a></span>(<span class="dt">plotList =</span> p, </span>
<span id="cb25-2"><a href="#cb25-2"></a>    <span class="dt">name =</span> <span class="st">"Plot-Tracks-Marker-Genes.pdf"</span>, </span>
<span id="cb25-3"><a href="#cb25-3"></a>    <span class="dt">ArchRProj =</span> proj, </span>
<span id="cb25-4"><a href="#cb25-4"></a>    <span class="dt">addDOC =</span> <span class="ot">FALSE</span>, <span class="dt">width =</span> <span class="dv">5</span>, <span class="dt">height =</span> <span class="dv">5</span>)</span></code></pre></div>
<blockquote>
<p>## NULL<br>
## NULL<br>
## NULL<br>
## NULL<br>
## NULL<br>
## NULL<br>
## NULL<br>
## NULL<br>
## NULL<br>
## [1] 0</p>
</blockquote>
<p><a href="../../images/HemeShort/PDF/Plot-Tracks-Marker-Genes.pdf">Download PDF : Plot-Tracks-Marker-Genes.pdf</a></p>
<p>Last but certainly not least, ArchR natively supports an interactive and dynamic genome browser that can be launched locally via a <a href="https://shiny.rstudio.com/">shiny app</a>. To do this, we use the <code><a href="../../reference/ArchRBrowser.html">ArchRBrowser()</a></code> function.</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1"></a><span class="kw"><a href="../../reference/ArchRBrowser.html">ArchRBrowser</a></span>(<span class="dt">ArchRProj =</span> proj)</span></code></pre></div>
<p>This launches a dynamic genome browser session with a whole host of features including export of vectorized tracks for publication.</p>
</div>
<div id="saving-and-loading-an-archrproject" class="section level2">
<h2 class="hasAnchor">
<a href="#saving-and-loading-an-archrproject" class="anchor"></a>Saving and Loading an <code>ArchRProject</code>
</h2>
<p>To easily save an <code>ArchRProject</code> for later use or for sharing with collaborators, we use the <code><a href="../../reference/saveArchRProject.html">saveArchRProject()</a></code> function. This copies the current <code>ArchRProject</code> object and all of the Arrow files to a specified directory. If we don’t specify an output directory (as below), <code><a href="../../reference/saveArchRProject.html">saveArchRProject()</a></code> uses the output directory that we specified upon creation of our <code>ArchRProject</code>. In this case that is the folder “HemeTutorial”.</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1"></a>proj &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/saveArchRProject.html">saveArchRProject</a></span>(<span class="dt">ArchRProj =</span> proj)</span></code></pre></div>
<blockquote>
<p>## Saving ArchRProject…<br>
## Loading ArchRProject…<br>
## Successfully loaded ArchRProject!</p>
</blockquote>
<p>When we are ready to load this saved <code>ArchRProject</code> we use the <code><a href="../../reference/loadArchRProject.html">loadArchRProject()</a></code> object and provide the path to the folder containing the saved <code>ArchRProject</code> object.</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1"></a>proj &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/loadArchRProject.html">loadArchRProject</a></span>(<span class="dt">path =</span> <span class="st">"HemeTutorial"</span>)</span></code></pre></div>
<blockquote>
<p>## Successfully loaded ArchRProject!</p>
</blockquote>
</div>
<div id="session-information" class="section level2">
<h2 class="hasAnchor">
<a href="#session-information" class="anchor"></a>Session Information</h2>
<p>This tutorial was run on the date specified below.</p>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1"></a><span class="kw"><a href="https://rdrr.io/r/base/Sys.time.html">Sys.Date</a></span>()</span></code></pre></div>
<blockquote>
<p>## [1] “2020-04-21”</p>
</blockquote>
<p>The <code><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo()</a></code> at run time was:</p>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1"></a><span class="kw"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span>()</span></code></pre></div>
<blockquote>
<p>## R version 3.6.1 (2019-07-05)<br>
## Platform: x86_64-pc-linux-gnu (64-bit)<br>
## Running under: CentOS Linux 7 (Core)<br>
##<br>
## Matrix products: default<br>
## BLAS/LAPACK: /share/software/user/open/openblas/0.2.19/lib/libopenblasp-r0.2.19.so<br>
##<br>
## locale:<br>
## [1] LC_CTYPE=en_US.UTF-8 LC_NUMERIC=C<br>
## [3] LC_TIME=en_US.UTF-8 LC_COLLATE=en_US.UTF-8<br>
## [5] LC_MONETARY=en_US.UTF-8 LC_MESSAGES=en_US.UTF-8<br>
## [7] LC_PAPER=en_US.UTF-8 LC_NAME=C<br>
## [9] LC_ADDRESS=C LC_TELEPHONE=C<br>
## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C<br>
##<br>
## attached base packages:<br>
## [1] parallel stats4 stats graphics grDevices utils datasets<br>
## [8] methods base<br>
##<br>
## other attached packages:<br>
## [1] ArchR_0.9.1 magrittr_1.5<br>
## [3] rhdf5_2.30.1 Matrix_1.2-17<br>
## [5] data.table_1.12.8 SummarizedExperiment_1.16.1<br>
## [7] DelayedArray_0.12.2 BiocParallel_1.20.1<br>
## [9] matrixStats_0.56.0 Biobase_2.46.0<br>
## [11] GenomicRanges_1.38.0 GenomeInfoDb_1.22.1<br>
## [13] IRanges_2.20.2 S4Vectors_0.24.3<br>
## [15] BiocGenerics_0.32.0 ggplot2_3.2.1<br>
##<br>
## loaded via a namespace (and not attached):<br>
## [1] Rcpp_1.0.4 pillar_1.4.3 compiler_3.6.1<br>
## [4] XVector_0.26.0 tools_3.6.1 bitops_1.0-6<br>
## [7] zlibbioc_1.32.0 lifecycle_0.1.0 tibble_2.1.3<br>
## [10] gtable_0.3.0 lattice_0.20-38 pkgconfig_2.0.3<br>
## [13] rlang_0.4.5 GenomeInfoDbData_1.2.2 withr_2.1.2<br>
## [16] dplyr_0.8.4 grid_3.6.1 tidyselect_1.0.0<br>
## [19] glue_1.4.0 R6_2.4.1 Rhdf5lib_1.8.0<br>
## [22] purrr_0.3.3 scales_1.1.0 assertthat_0.2.1<br>
## [25] colorspace_1.4-1 RCurl_1.98-1.1 lazyeval_0.2.2<br>
## [28] munsell_0.5.0 crayon_1.3.4</p>
</blockquote>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#getting-set-up">Getting Set Up</a></li>
      <li><a href="#creating-arrow-files">Creating Arrow Files</a></li>
      <li><a href="#inferring-doublets">Inferring Doublets</a></li>
      <li><a href="#creating-an-archrproject">Creating an <code>ArchRProject</code></a></li>
      <li><a href="#dimensionality-reduction-and-clustering">Dimensionality Reduction and Clustering</a></li>
      <li><a href="#visualizing-in-a-2d-umap-embedding">Visualizing in a 2D UMAP Embedding</a></li>
      <li><a href="#assigning-clusters-with-gene-scores">Assigning Clusters with Gene Scores</a></li>
      <li><a href="#visualizing-genome-browser-tracks">Visualizing Genome Browser Tracks</a></li>
      <li><a href="#saving-and-loading-an-archrproject">Saving and Loading an <code>ArchRProject</code></a></li>
      <li><a href="#session-information">Session Information</a></li>
      </ul>
</div>
      </div>

</div>



      <footer><div class="copyright">
  <p>Developed by <a href="https://twitter.com/JeffreyGranja">Jeffrey Granja</a>, <a href="https://twitter.com/doctorcorces">Ryan Corces</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.4.1.</p>
</div>

      </footer>
</div>

  
<script src="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.1/docsearch.min.js" integrity="sha256-GKvGqXDznoRYHCwKXGnuchvKSwmx9SRMrZOTh2g4Sb0=" crossorigin="anonymous"></script><script>
  docsearch({
    
    
    apiKey: '252c2c36fedc32b3ed803d0db4877b96',
    indexName: 'archr',
    inputSelector: 'input#search-input.form-control',
    transformData: function(hits) {
      return hits.map(function (hit) {
        hit.url = updateHitURL(hit);
        return hit;
      });
    }
  });
</script>
</body>
</html>
