<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Add GeneScoreMatrix to ArrowFiles or an ArchRProject — addGeneScoreMatrix • ArchR</title>

<!-- favicons -->
<link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png" />
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png" />
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png" />
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png" />

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/yeti/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous" />


<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>


<!-- docsearch -->
<script src="../docsearch.js"></script>
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.3/docsearch.min.css" integrity="sha256-QOSRU/ra9ActyXkIBbiIB144aDBdtvXBcNc3OTNuX/Q=" crossorigin="anonymous" />
<link href="../docsearch.css" rel="stylesheet">
<script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/jquery.mark.min.js" integrity="sha256-4HLtjeVgH0eIB3aZ9mLYF6E8oU5chNdjU6p6rrXpl9U=" crossorigin="anonymous"></script>


  <link href="../extra.css" rel="stylesheet">
  

<meta property="og:title" content="Add GeneScoreMatrix to ArrowFiles or an ArchRProject — addGeneScoreMatrix" />
<meta property="og:description" content="This function, for each sample, will independently compute counts for each tile
per cell and then infer gene activity scores." />
<meta property="og:image" content="https://greenleaflab.github.io/ArchR_Website/logo.svg" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


<!-- Global site tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-158769145-1"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-158769145-1');
</script>


  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">ArchR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.0.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fa fa-home"></span>
     
  </a>
</li>
<li>
  <a href="../articles/Articles/tutorial.html">
    <span class="fa fa-graduation-cap"></span>
     
    Quick Tutorial
  </a>
</li>
<li>
  <a href="../bookdown/index.html">
    <span class="fa fa-file-text-o"></span>
     
    Full Manual
  </a>
</li>
<li>
  <a href="../articles/Articles/faq.html">
    <span class="fa fa-question-circle-o"></span>
     
    FAQ
  </a>
</li>
<li>
  <a href="../reference/index.html">
    <span class="fa fa-code"></span>
     
    Functions
  </a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/GreenleafLab/ArchR">
    <span class="fa fa-github"></span>
     
    ArchR on GitHub
  </a>
</li>
      </ul>
      
      <form class="navbar-form navbar-right hidden-xs hidden-sm" role="search">
        <div class="form-group">
          <input type="search" class="form-control" name="search-input" id="search-input" placeholder="Search..." aria-label="Search for..." autocomplete="off">
        </div>
      </form>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Add GeneScoreMatrix to ArrowFiles or an ArchRProject</h1>
    
    <div class="hidden name"><code>addGeneScoreMatrix.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>This function, for each sample, will independently compute counts for each tile
per cell and then infer gene activity scores.</p>
    </div>

    <pre class="usage"><span class='fu'>addGeneScoreMatrix</span><span class='op'>(</span>
  input <span class='op'>=</span> <span class='cn'>NULL</span>,
  genes <span class='op'>=</span> <span class='fu'><a href='getGenes.html'>getGenes</a></span><span class='op'>(</span><span class='va'>input</span><span class='op'>)</span>,
  geneModel <span class='op'>=</span> <span class='st'>"exp(-abs(x)/5000) + exp(-1)"</span>,
  matrixName <span class='op'>=</span> <span class='st'>"GeneScoreMatrix"</span>,
  extendUpstream <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>1000</span>, <span class='fl'>1e+05</span><span class='op'>)</span>,
  extendDownstream <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>1000</span>, <span class='fl'>1e+05</span><span class='op'>)</span>,
  geneUpstream <span class='op'>=</span> <span class='fl'>5000</span>,
  geneDownstream <span class='op'>=</span> <span class='fl'>0</span>,
  useGeneBoundaries <span class='op'>=</span> <span class='cn'>TRUE</span>,
  useTSS <span class='op'>=</span> <span class='cn'>FALSE</span>,
  extendTSS <span class='op'>=</span> <span class='cn'>FALSE</span>,
  tileSize <span class='op'>=</span> <span class='fl'>500</span>,
  ceiling <span class='op'>=</span> <span class='fl'>4</span>,
  geneScaleFactor <span class='op'>=</span> <span class='fl'>5</span>,
  scaleTo <span class='op'>=</span> <span class='fl'>10000</span>,
  excludeChr <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"chrY"</span>, <span class='st'>"chrM"</span><span class='op'>)</span>,
  blacklist <span class='op'>=</span> <span class='fu'><a href='getBlacklist.html'>getBlacklist</a></span><span class='op'>(</span><span class='va'>input</span><span class='op'>)</span>,
  threads <span class='op'>=</span> <span class='fu'><a href='getArchRThreads.html'>getArchRThreads</a></span><span class='op'>(</span><span class='op'>)</span>,
  parallelParam <span class='op'>=</span> <span class='cn'>NULL</span>,
  subThreading <span class='op'>=</span> <span class='cn'>TRUE</span>,
  force <span class='op'>=</span> <span class='cn'>FALSE</span>,
  logFile <span class='op'>=</span> <span class='fu'><a href='createLogFile.html'>createLogFile</a></span><span class='op'>(</span><span class='st'>"addGeneScoreMatrix"</span><span class='op'>)</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>input</th>
      <td><p>An <code>ArchRProject</code> object or character vector of ArrowFiles.</p></td>
    </tr>
    <tr>
      <th>genes</th>
      <td><p>A stranded <code>GRanges</code> object containing the ranges associated with all gene start and end coordinates.</p></td>
    </tr>
    <tr>
      <th>geneModel</th>
      <td><p>A string giving a "gene model function" used for weighting peaks for gene score calculation. This string
should be a function of <code>x</code>, where <code>x</code> is the stranded distance from the transcription start site of the gene.</p></td>
    </tr>
    <tr>
      <th>matrixName</th>
      <td><p>The name to be used for storage of the gene activity score matrix in the provided <code>ArchRProject</code> or ArrowFiles.</p></td>
    </tr>
    <tr>
      <th>extendUpstream</th>
      <td><p>The minimum and maximum number of basepairs upstream of the transcription start site to consider for gene
activity score calculation.</p></td>
    </tr>
    <tr>
      <th>extendDownstream</th>
      <td><p>The minimum and maximum number of basepairs downstream of the transcription start site or transcription termination site
(based on 'useTSS') to consider for gene activity score calculation.</p></td>
    </tr>
    <tr>
      <th>geneUpstream</th>
      <td><p>An integer describing the number of bp upstream the gene to extend the gene body. This effectively makes the gene body larger as there
are proximal peaks that should be weighted equally to the gene body. This parameter is used if 'useTSS=FALSE'.</p></td>
    </tr>
    <tr>
      <th>geneDownstream</th>
      <td><p>An integer describing the number of bp downstream the gene to extend the gene body.This effectively makes the gene body larger as there
are proximal peaks that should be weighted equally to the gene body. This parameter is used if 'useTSS=FALSE'.</p></td>
    </tr>
    <tr>
      <th>useGeneBoundaries</th>
      <td><p>A boolean value indicating whether gene boundaries should be employed during gene activity score
calculation. Gene boundaries refers to the process of preventing tiles from contributing to the gene score of a given gene
if there is a second gene's transcription start site between the tile and the gene of interest.</p></td>
    </tr>
    <tr>
      <th>useTSS</th>
      <td><p>A boolean describing whether to build gene model based on gene TSS or the gene body.</p></td>
    </tr>
    <tr>
      <th>extendTSS</th>
      <td><p>A boolean describing whether to extend the gene TSS. By default useTSS uses the 1bp TSS while this parameter enables the extension of this
region with 'geneUpstream' and 'geneDownstream' respectively.</p></td>
    </tr>
    <tr>
      <th>tileSize</th>
      <td><p>The size of the tiles used for binning counts prior to gene activity score calculation.</p></td>
    </tr>
    <tr>
      <th>ceiling</th>
      <td><p>The maximum counts per tile allowed. This is used to prevent large biases in tile counts.</p></td>
    </tr>
    <tr>
      <th>geneScaleFactor</th>
      <td><p>A numeric scaling factor to weight genes based on the inverse of there length i.e. (Scale Factor)/(Gene Length). This
is scaled from 1 to the scale factor. Small genes will be the scale factor while extremely large genes will be closer to 1. This scaling helps with
the relative gene score value.</p></td>
    </tr>
    <tr>
      <th>scaleTo</th>
      <td><p>Each column in the calculated gene score matrix will be normalized to a column sum designated by <code>scaleTo</code>.</p></td>
    </tr>
    <tr>
      <th>excludeChr</th>
      <td><p>A character vector containing the <code>seqnames</code> of the chromosomes that should be excluded from this analysis.</p></td>
    </tr>
    <tr>
      <th>blacklist</th>
      <td><p>A <code>GRanges</code> object containing genomic regions to blacklist that may be extremeley over-represented and thus
biasing the geneScores for genes nearby that locus.</p></td>
    </tr>
    <tr>
      <th>threads</th>
      <td><p>The number of threads to be used for parallel computing.</p></td>
    </tr>
    <tr>
      <th>parallelParam</th>
      <td><p>A list of parameters to be passed for biocparallel/batchtools parallel computing.</p></td>
    </tr>
    <tr>
      <th>subThreading</th>
      <td><p>A boolean determining whether possible use threads within each multi-threaded subprocess if greater than the number of input samples.</p></td>
    </tr>
    <tr>
      <th>force</th>
      <td><p>A boolean value indicating whether to force the matrix indicated by <code>matrixName</code> to be overwritten if it already exist in the given <code>input</code>.</p></td>
    </tr>
    <tr>
      <th>logFile</th>
      <td><p>The path to a file to be used for logging ArchR output.</p></td>
    </tr>
    </table>


  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by <a href='https://twitter.com/JeffreyGranja'>Jeffrey Granja</a>, <a href='https://twitter.com/doctorcorces'>Ryan Corces</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  
<script src="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.1/docsearch.min.js" integrity="sha256-GKvGqXDznoRYHCwKXGnuchvKSwmx9SRMrZOTh2g4Sb0=" crossorigin="anonymous"></script>
<script>
  docsearch({
    
    
    apiKey: '252c2c36fedc32b3ed803d0db4877b96',
    indexName: 'archr',
    inputSelector: 'input#search-input.form-control',
    transformData: function(hits) {
      return hits.map(function (hit) {
        hit.url = updateHitURL(hit);
        return hit;
      });
    }
  });
</script>



  </body>
</html>


