<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Add a GeneIntegrationMatrix to ArrowFiles or an ArchRProject — addGeneIntegrationMatrix • ArchR</title>

<!-- favicons -->
<link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png" />
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png" />
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png" />
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png" />
<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.3.7/yeti/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>


<!-- docsearch -->
<script src="../docsearch.js"></script>
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.1/docsearch.min.css" integrity="sha256-QOSRU/ra9ActyXkIBbiIB144aDBdtvXBcNc3OTNuX/Q=" crossorigin="anonymous" />
<link href="../docsearch.css" rel="stylesheet">
<script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/jquery.mark.min.js" integrity="sha256-4HLtjeVgH0eIB3aZ9mLYF6E8oU5chNdjU6p6rrXpl9U=" crossorigin="anonymous"></script>


  <link href="../extra.css" rel="stylesheet">
  
<meta property="og:title" content="Add a GeneIntegrationMatrix to ArrowFiles or an ArchRProject — addGeneIntegrationMatrix" />

<meta property="og:description" content="This function, will integrate multiple subsets of scATAC cells with a scRNA experiment, compute matched scRNA profiles and
then store this in each samples ArrowFile." />

<meta property="og:image" content="https://greenleaflab.github.io/ArchR_Website/logo.png" />
<meta name="twitter:card" content="summary" />



<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


<!-- Global site tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-158769145-1"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-158769145-1');
</script>


  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">ArchR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.9.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fa fa-home"></span>
     
  </a>
</li>
<li>
  <a href="../articles/Articles/tutorial.html">
    <span class="fa fa-graduation-cap"></span>
     
    Quick Tutorial
  </a>
</li>
<li>
  <a href="../bookdown/index.html">
    <span class="fa fa-file-text-o"></span>
     
    Full Manual
  </a>
</li>
<li>
  <a href="../articles/Articles/faq.html">
    <span class="fa fa-question-circle-o"></span>
     
    FAQ
  </a>
</li>
<li>
  <a href="../reference/index.html">
    <span class="fa fa-code"></span>
     
    Functions
  </a>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/GreenleafLab/ArchR">
    <span class="fa fa-github"></span>
     
    ArchR on GitHub
  </a>
</li>
      </ul>
      
      <form class="navbar-form navbar-right hidden-xs hidden-sm" role="search">
        <div class="form-group">
          <input type="search" class="form-control" name="search-input" id="search-input" placeholder="Search..." aria-label="Search for..." autocomplete="off">
        </div>
      </form>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Add a GeneIntegrationMatrix to ArrowFiles or an ArchRProject</h1>
    
    <div class="hidden name"><code>addGeneIntegrationMatrix.Rd</code></div>
    </div>

    <div class="ref-description">
    
    <p>This function, will integrate multiple subsets of scATAC cells with a scRNA experiment, compute matched scRNA profiles and
then store this in each samples ArrowFile.</p>
    
    </div>

    <pre class="usage"><span class='fu'>addGeneIntegrationMatrix</span>(
  <span class='kw'>ArchRProj</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>useMatrix</span> <span class='kw'>=</span> <span class='st'>"GeneScoreMatrix"</span>,
  <span class='kw'>matrixName</span> <span class='kw'>=</span> <span class='st'>"GeneIntegrationMatrix"</span>,
  <span class='kw'>reducedDims</span> <span class='kw'>=</span> <span class='st'>"IterativeLSI"</span>,
  <span class='kw'>seRNA</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>groupATAC</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>groupRNA</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>groupList</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>sampleCellsATAC</span> <span class='kw'>=</span> <span class='fl'>10000</span>,
  <span class='kw'>sampleCellsRNA</span> <span class='kw'>=</span> <span class='fl'>10000</span>,
  <span class='kw'>embeddingATAC</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>embeddingRNA</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>nGenes</span> <span class='kw'>=</span> <span class='fl'>2000</span>,
  <span class='kw'>useImputation</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>,
  <span class='kw'>reduction</span> <span class='kw'>=</span> <span class='st'>"cca"</span>,
  <span class='kw'>addToArrow</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>,
  <span class='kw'>scaleTo</span> <span class='kw'>=</span> <span class='fl'>10000</span>,
  <span class='kw'>nameCell</span> <span class='kw'>=</span> <span class='st'>"predictedCell"</span>,
  <span class='kw'>nameGroup</span> <span class='kw'>=</span> <span class='st'>"predictedGroup"</span>,
  <span class='kw'>nameScore</span> <span class='kw'>=</span> <span class='st'>"predictedScore"</span>,
  <span class='kw'>transferParams</span> <span class='kw'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span>(),
  <span class='kw'>threads</span> <span class='kw'>=</span> <span class='fu'><a href='getArchRThreads.html'>getArchRThreads</a></span>(),
  <span class='kw'>verbose</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>,
  <span class='kw'>force</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>,
  <span class='kw'>logFile</span> <span class='kw'>=</span> <span class='fu'><a href='createLogFile.html'>createLogFile</a></span>(<span class='st'>"addGeneIntegrationMatrix"</span>),
  <span class='no'>...</span>
)</pre>
    
    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>ArchRProj</th>
      <td><p>An <code>ArchRProject</code> object.</p></td>
    </tr>
    <tr>
      <th>useMatrix</th>
      <td><p>The name of a matrix in the <code>ArchRProject</code> containing gene scores to be used for RNA integration.</p></td>
    </tr>
    <tr>
      <th>matrixName</th>
      <td><p>The name to use for the output matrix containing scRNA-seq integration to be stored in the <code>ArchRProject</code>.</p></td>
    </tr>
    <tr>
      <th>reducedDims</th>
      <td><p>The name of the <code>reducedDims</code> object (i.e. "IterativeLSI") to retrieve from the designated <code>ArchRProject</code>.
This <code>reducedDims</code> will be used in weighting the transfer of data to scRNA to scATAC. See <code><a href='https://rdrr.io/pkg/Seurat/man/TransferData.html'>Seurat::TransferData</a></code> for more info.</p></td>
    </tr>
    <tr>
      <th>seRNA</th>
      <td><p>A <code>SeuratObject</code> or a scRNA-seq <code>SummarizedExperiment</code> (cell x gene) to be integrated with the scATAC-seq data.</p></td>
    </tr>
    <tr>
      <th>groupATAC</th>
      <td><p>A column name in <code>cellColData</code> of the <code>ArchRProj</code> that will be used to determine the subgroupings specified in <code>groupList</code>.
This is used to constrain the integration to occur across biologically relevant groups.</p></td>
    </tr>
    <tr>
      <th>groupRNA</th>
      <td><p>A column name in either <code>colData</code> (if <code>SummarizedExperiment</code>) or <code>metadata</code> (if <code>SeuratObject</code>) of <code>seRNA</code> that
will be used to determine the subgroupings specified in <code>groupList</code>. This is used to constrain the integration to occur across biologically relevant groups.
Additionally this groupRNA is used for the <code>nameGroup</code> output of this function.</p></td>
    </tr>
    <tr>
      <th>groupList</th>
      <td><p>A list of cell groupings for both ATAC-seq and RNA-seq cells to be used for RNA-ATAC integration.
This is used to constrain the integration to occur across biologically relevant groups. The format of this should be a list of groups
with subgroups of ATAC and RNA specifying cells to integrate from both platforms.
For example <code>groupList</code> &lt;- list(groupA = list(ATAC = cellsATAC_A, RNA = cellsRNA_A), groupB = list(ATAC = cellsATAC_B, RNA = cellsRNA_B))</p></td>
    </tr>
    <tr>
      <th>sampleCellsATAC</th>
      <td><p>An integer describing the number of scATAC-seq cells to be used for integration.
This number will be evenly sampled across the total number of cells in the ArchRProject.</p></td>
    </tr>
    <tr>
      <th>sampleCellsRNA</th>
      <td><p>An integer describing the number of scRNA-seq cells to be used for integration.</p></td>
    </tr>
    <tr>
      <th>embeddingATAC</th>
      <td><p>A <code>data.frame</code> of cell embeddings such as a UMAP for scATAC-seq cells to be used for density sampling. The <code>data.frame</code> object
should have a row for each single cell described in <code>row.names</code> and 2 columns, one for each dimension of the embedding.</p></td>
    </tr>
    <tr>
      <th>embeddingRNA</th>
      <td><p>A <code>data.frame</code> of cell embeddings such as a UMAP for scRNA-seq cells to be used for density sampling. The <code>data.frame</code> object
should have a row for each single cell described in <code>row.names</code> and 2 columns, one for each dimension of the embedding.</p></td>
    </tr>
    <tr>
      <th>nGenes</th>
      <td><p>The number of variable genes determined by <code><a href='https://rdrr.io/pkg/Seurat/man/FindVariableFeatures.html'>Seurat::FindVariableGenes()</a></code> to use for integration.</p></td>
    </tr>
    <tr>
      <th>useImputation</th>
      <td><p>A boolean value indicating whether to use imputation for creating the Gene Score Matrix prior to integration.</p></td>
    </tr>
    <tr>
      <th>reduction</th>
      <td><p>The Seurat reduction method to use for integrating modalities. See <code><a href='https://rdrr.io/pkg/Seurat/man/FindTransferAnchors.html'>Seurat::FindTransferAnchors()</a></code> for possible reduction methods.</p></td>
    </tr>
    <tr>
      <th>addToArrow</th>
      <td><p>A boolean value indicating whether to add the log2-normalized transcript counts from the integrated matched RNA to the Arrow files.</p></td>
    </tr>
    <tr>
      <th>scaleTo</th>
      <td><p>Each column in the integrated RNA matrix will be normalized to a column sum designated by <code>scaleTo</code> prior to adding to Arrow files.</p></td>
    </tr>
    <tr>
      <th>nameCell</th>
      <td><p>A column name to add to <code>cellColData</code> for the predicted scRNA-seq cell in the specified <code>ArchRProject</code>. This is useful for identifying which cell was closest to the scATAC-seq cell.</p></td>
    </tr>
    <tr>
      <th>nameGroup</th>
      <td><p>A column name to add to <code>cellColData</code> for the predicted scRNA-seq group in the specified <code>ArchRProject</code>. See <code>groupRNA</code> for more details.</p></td>
    </tr>
    <tr>
      <th>nameScore</th>
      <td><p>A column name to add to <code>cellColData</code> for the predicted scRNA-seq score in the specified <code>ArchRProject</code>. These scores represent
the assignment accuracy of the group in the RNA cells. Lower scores represent ambiguous predictions and higher scores represent precise predictions.</p></td>
    </tr>
    <tr>
      <th>transferParams</th>
      <td><p>Additional params to be passed to <code><a href='https://rdrr.io/pkg/Seurat/man/TransferData.html'>Seurat::TransferData</a></code>.</p></td>
    </tr>
    <tr>
      <th>threads</th>
      <td><p>The number of threads to be used for parallel computing.</p></td>
    </tr>
    <tr>
      <th>verbose</th>
      <td><p>A boolean value that determines whether standard output includes verbose sections.</p></td>
    </tr>
    <tr>
      <th>force</th>
      <td><p>A boolean value indicating whether to force the matrix indicated by <code>matrixName</code> to be overwritten if it already exists in the given <code>input</code>.</p></td>
    </tr>
    <tr>
      <th>logFile</th>
      <td><p>The path to a file to be used for logging ArchR output.</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>Additional params to be added to <code><a href='https://rdrr.io/pkg/Seurat/man/FindTransferAnchors.html'>Seurat::FindTransferAnchors</a></code></p></td>
    </tr>
    </table>
    

  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
                </ul>

  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by <a href='https://twitter.com/JeffreyGranja'>Jeffrey Granja</a>, <a href='https://twitter.com/doctorcorces'>Ryan Corces</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.4.1.</p>
</div>

      </footer>
   </div>

  
<script src="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.1/docsearch.min.js" integrity="sha256-GKvGqXDznoRYHCwKXGnuchvKSwmx9SRMrZOTh2g4Sb0=" crossorigin="anonymous"></script>
<script>
  docsearch({
    
    
    apiKey: '252c2c36fedc32b3ed803d0db4877b96',
    indexName: 'archr',
    inputSelector: 'input#search-input.form-control',
    transformData: function(hits) {
      return hits.map(function (hit) {
        hit.url = updateHitURL(hit);
        return hit;
      });
    }
  });
</script>



  </body>
</html>


