<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Add Peak2GeneLinks to an ArchRProject — addPeak2GeneLinks • ArchR</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/yeti/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><!-- docsearch --><script src="../docsearch.js"></script><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.3/docsearch.min.css" integrity="sha256-QOSRU/ra9ActyXkIBbiIB144aDBdtvXBcNc3OTNuX/Q=" crossorigin="anonymous"><link href="../docsearch.css" rel="stylesheet"><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/jquery.mark.min.js" integrity="sha256-4HLtjeVgH0eIB3aZ9mLYF6E8oU5chNdjU6p6rrXpl9U=" crossorigin="anonymous"></script><link href="../extra.css" rel="stylesheet"><meta property="og:title" content="Add Peak2GeneLinks to an ArchRProject — addPeak2GeneLinks"><meta property="og:description" content="This function will add peak-to-gene links to a given ArchRProject"><meta property="og:image" content="https://greenleaflab.github.io/ArchR_Website/logo.svg"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-158769145-1"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-158769145-1');
</script></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">ArchR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.0.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../index.html">
    <span class="fa fa-home"></span>
     
  </a>
</li>
<li>
  <a href="../articles/Articles/tutorial.html">
    <span class="fa fa-graduation-cap"></span>
     
    Quick Tutorial
  </a>
</li>
<li>
  <a href="../bookdown/index.html">
    <span class="fa fa-file-text-o"></span>
     
    Full Manual
  </a>
</li>
<li>
  <a href="../reference/index.html">
    <span class="fa fa-code"></span>
     
    Functions
  </a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/GreenleafLab/ArchR" class="external-link">
    <span class="fa fa-github"></span>
     
    ArchR on GitHub
  </a>
</li>
      </ul><form class="navbar-form navbar-right hidden-xs hidden-sm" role="search">
        <div class="form-group">
          <input type="search" class="form-control" name="search-input" id="search-input" placeholder="Search..." aria-label="Search for..." autocomplete="off"></div>
      </form>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Add Peak2GeneLinks to an ArchRProject</h1>
    
    <div class="hidden name"><code>addPeak2GeneLinks.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>This function will add peak-to-gene links to a given ArchRProject</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">addPeak2GeneLinks</span><span class="op">(</span></span>
<span>  ArchRProj <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  reducedDims <span class="op">=</span> <span class="st">"IterativeLSI"</span>,</span>
<span>  useMatrix <span class="op">=</span> <span class="st">"GeneIntegrationMatrix"</span>,</span>
<span>  dimsToUse <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">30</span>,</span>
<span>  scaleDims <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  corCutOff <span class="op">=</span> <span class="fl">0.75</span>,</span>
<span>  cellsToUse <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  excludeChr <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  k <span class="op">=</span> <span class="fl">100</span>,</span>
<span>  knnIteration <span class="op">=</span> <span class="fl">500</span>,</span>
<span>  overlapCutoff <span class="op">=</span> <span class="fl">0.8</span>,</span>
<span>  maxDist <span class="op">=</span> <span class="fl">250000</span>,</span>
<span>  scaleTo <span class="op">=</span> <span class="fl">10</span><span class="op">^</span><span class="fl">4</span>,</span>
<span>  log2Norm <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  predictionCutoff <span class="op">=</span> <span class="fl">0.4</span>,</span>
<span>  addEmpiricalPval <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  addPermutedPval <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  nperm <span class="op">=</span> <span class="fl">100</span>,</span>
<span>  seed <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  threads <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">max</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">floor</a></span><span class="op">(</span><span class="fu"><a href="getArchRThreads.html">getArchRThreads</a></span><span class="op">(</span><span class="op">)</span><span class="op">/</span><span class="fl">2</span><span class="op">)</span>, <span class="fl">1</span><span class="op">)</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  logFile <span class="op">=</span> <span class="fu"><a href="createLogFile.html">createLogFile</a></span><span class="op">(</span><span class="st">"addPeak2GeneLinks"</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>ArchRProj</dt>
<dd><p>An <code>ArchRProject</code> object.</p></dd>


<dt>reducedDims</dt>
<dd><p>The name of the <code>reducedDims</code> object (i.e. "IterativeLSI") to retrieve from the designated <code>ArchRProject</code>.</p></dd>


<dt>useMatrix</dt>
<dd><p>The name of the matrix containing gene expression information to be used for determining peak-to-gene links. See <code>getAvailableMatrices(ArchRProj)</code></p></dd>


<dt>dimsToUse</dt>
<dd><p>A vector containing the dimensions from the <code>reducedDims</code> object to use in clustering.</p></dd>


<dt>scaleDims</dt>
<dd><p>A boolean value that indicates whether to z-score the reduced dimensions for each cell. This is useful for minimizing
the contribution of strong biases (dominating early PCs) and lowly abundant populations. However, this may lead to stronger sample-specific
biases since it is over-weighting latent PCs. If set to <code>NULL</code> this will scale the dimensions based on the value of <code>scaleDims</code> when the
<code>reducedDims</code> were originally created during dimensionality reduction. This idea was introduced by Timothy Stuart.</p></dd>


<dt>corCutOff</dt>
<dd><p>A numeric cutoff for the correlation of each dimension to the sequencing depth. If the dimension has a
correlation to sequencing depth that is greater than the <code>corCutOff</code>, it will be excluded from analysis.</p></dd>


<dt>cellsToUse</dt>
<dd><p>A character vector of cellNames to compute peak-to-gene links on if desired to run on a subset of the total cells.</p></dd>


<dt>excludeChr</dt>
<dd><p>A character vector containing the <code>seqnames</code> of the chromosomes that should be excluded from this analysis.</p></dd>


<dt>k</dt>
<dd><p>The number of k-nearest neighbors to use for creating single-cell groups for correlation analyses.</p></dd>


<dt>knnIteration</dt>
<dd><p>The number of k-nearest neighbor groupings to test for passing the supplied <code>overlapCutoff</code>.</p></dd>


<dt>overlapCutoff</dt>
<dd><p>The maximum allowable overlap between the current group and all previous groups to permit the current
group be added to the group list during k-nearest neighbor calculations.</p></dd>


<dt>maxDist</dt>
<dd><p>The maximum allowable distance in basepairs between two peaks to consider for co-accessibility.</p></dd>


<dt>scaleTo</dt>
<dd><p>The total insertion counts from the designated group of single cells is summed across all relevant peak regions
from the <code>peakSet</code> of the <code>ArchRProject</code> and normalized to the total depth provided by <code>scaleTo</code>.</p></dd>


<dt>log2Norm</dt>
<dd><p>A boolean value indicating whether to log2 transform the single-cell groups prior to computing co-accessibility correlations.</p></dd>


<dt>predictionCutoff</dt>
<dd><p>A numeric describing the cutoff for RNA integration to use when picking cells for groupings.</p></dd>


<dt>addEmpiricalPval</dt>
<dd><p>Add empirical p-values based on randomly correlating peaks and genes not on the same seqname.</p></dd>


<dt>addPermutedPval</dt>
<dd><p>Add permuted p-values based on shuffle sample correlating peaks and genes. This approach was adapted from
Regner et al 2021 "A multi-omic single-cell landscape of human gynecologic malignancies".</p></dd>


<dt>nperm</dt>
<dd><p>An integer representing the number of permutations to run for Regner et al 2021 approach.</p></dd>


<dt>seed</dt>
<dd><p>A number to be used as the seed for random number generation required in knn determination. It is recommended
to keep track of the seed used so that you can reproduce results downstream.</p></dd>


<dt>threads</dt>
<dd><p>The number of threads to be used for parallel computing.</p></dd>


<dt>verbose</dt>
<dd><p>A boolean value that determines whether standard output should be printed.</p></dd>


<dt>logFile</dt>
<dd><p>The path to a file to be used for logging ArchR output.</p></dd>

</dl></div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span></span>
<span><span class="co"># Get Test ArchR Project</span></span>
<span><span class="va">proj</span> <span class="op">&lt;-</span> <span class="fu"><a href="getTestProject.html">getTestProject</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Add P2G Links</span></span>
<span><span class="va">proj</span> <span class="op">&lt;-</span> <span class="fu">addPeak2GeneLinks</span><span class="op">(</span><span class="va">proj</span>, k <span class="op">=</span> <span class="fl">20</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Get P2G Links</span></span>
<span><span class="va">p2g</span> <span class="op">&lt;-</span> <span class="fu"><a href="getPeak2GeneLinks.html">getPeak2GeneLinks</a></span><span class="op">(</span><span class="va">proj</span><span class="op">)</span></span>
<span></span></code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by <a href="https://twitter.com/JeffreyGranja" class="external-link">Jeffrey Granja</a>, <a href="https://twitter.com/doctorcorces" class="external-link">Ryan Corces</a>.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  
<script src="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.1/docsearch.min.js" integrity="sha256-GKvGqXDznoRYHCwKXGnuchvKSwmx9SRMrZOTh2g4Sb0=" crossorigin="anonymous"></script><script>
  docsearch({
    
    
    apiKey: '252c2c36fedc32b3ed803d0db4877b96',
    indexName: 'archr',
    inputSelector: 'input#search-input.form-control',
    transformData: function(hits) {
      return hits.map(function (hit) {
        hit.url = updateHitURL(hit);
        return hit;
      });
    }
  });
</script></body></html>

