<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Add a Reproducible Peak Set to an ArchRProject — addReproduciblePeakSet • ArchR</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/yeti/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><!-- docsearch --><script src="../docsearch.js"></script><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.3/docsearch.min.css" integrity="sha256-QOSRU/ra9ActyXkIBbiIB144aDBdtvXBcNc3OTNuX/Q=" crossorigin="anonymous"><link href="../docsearch.css" rel="stylesheet"><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/jquery.mark.min.js" integrity="sha256-4HLtjeVgH0eIB3aZ9mLYF6E8oU5chNdjU6p6rrXpl9U=" crossorigin="anonymous"></script><link href="../extra.css" rel="stylesheet"><meta property="og:title" content="Add a Reproducible Peak Set to an ArchRProject — addReproduciblePeakSet"><meta property="og:description" content='This function will get insertions from coverage files, call peaks,
and merge peaks to get a "Union Reproducible Peak Set".'><meta property="og:image" content="https://greenleaflab.github.io/ArchR_Website/logo.svg"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-158769145-1"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-158769145-1');
</script></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">ArchR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.0.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../index.html">
    <span class="fa fa-home"></span>
     
  </a>
</li>
<li>
  <a href="../articles/Articles/tutorial.html">
    <span class="fa fa-graduation-cap"></span>
     
    Quick Tutorial
  </a>
</li>
<li>
  <a href="../bookdown/index.html">
    <span class="fa fa-file-text-o"></span>
     
    Full Manual
  </a>
</li>
<li>
  <a href="../reference/index.html">
    <span class="fa fa-code"></span>
     
    Functions
  </a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/GreenleafLab/ArchR" class="external-link">
    <span class="fa fa-github"></span>
     
    ArchR on GitHub
  </a>
</li>
      </ul><form class="navbar-form navbar-right hidden-xs hidden-sm" role="search">
        <div class="form-group">
          <input type="search" class="form-control" name="search-input" id="search-input" placeholder="Search..." aria-label="Search for..." autocomplete="off"></div>
      </form>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Add a Reproducible Peak Set to an ArchRProject</h1>
    
    <div class="hidden name"><code>addReproduciblePeakSet.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>This function will get insertions from coverage files, call peaks,
and merge peaks to get a "Union Reproducible Peak Set".</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">addReproduciblePeakSet</span><span class="op">(</span></span>
<span>  ArchRProj <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  groupBy <span class="op">=</span> <span class="st">"Clusters"</span>,</span>
<span>  peakMethod <span class="op">=</span> <span class="st">"Macs2"</span>,</span>
<span>  reproducibility <span class="op">=</span> <span class="st">"2"</span>,</span>
<span>  peaksPerCell <span class="op">=</span> <span class="fl">500</span>,</span>
<span>  maxPeaks <span class="op">=</span> <span class="fl">150000</span>,</span>
<span>  minCells <span class="op">=</span> <span class="fl">25</span>,</span>
<span>  excludeChr <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"chrM"</span>, <span class="st">"chrY"</span><span class="op">)</span>,</span>
<span>  pathToMacs2 <span class="op">=</span> <span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/chartr.html" class="external-link">tolower</a></span><span class="op">(</span><span class="va">peakMethod</span><span class="op">)</span> <span class="op">==</span> <span class="st">"macs2"</span><span class="op">)</span> <span class="fu"><a href="findMacs2.html">findMacs2</a></span><span class="op">(</span><span class="op">)</span> <span class="kw">else</span> <span class="cn">NULL</span>,</span>
<span>  genomeSize <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  shift <span class="op">=</span> <span class="op">-</span><span class="fl">75</span>,</span>
<span>  extsize <span class="op">=</span> <span class="fl">150</span>,</span>
<span>  method <span class="op">=</span> <span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/chartr.html" class="external-link">tolower</a></span><span class="op">(</span><span class="va">peakMethod</span><span class="op">)</span> <span class="op">==</span> <span class="st">"macs2"</span><span class="op">)</span> <span class="st">"q"</span> <span class="kw">else</span> <span class="st">"p"</span>,</span>
<span>  cutOff <span class="op">=</span> <span class="fl">0.1</span>,</span>
<span>  additionalParams <span class="op">=</span> <span class="st">"--nomodel --nolambda"</span>,</span>
<span>  extendSummits <span class="op">=</span> <span class="fl">250</span>,</span>
<span>  promoterRegion <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">2000</span>, <span class="fl">100</span><span class="op">)</span>,</span>
<span>  genomeAnnotation <span class="op">=</span> <span class="fu"><a href="getGenomeAnnotation.html">getGenomeAnnotation</a></span><span class="op">(</span><span class="va">ArchRProj</span><span class="op">)</span>,</span>
<span>  geneAnnotation <span class="op">=</span> <span class="fu"><a href="getGeneAnnotation.html">getGeneAnnotation</a></span><span class="op">(</span><span class="va">ArchRProj</span><span class="op">)</span>,</span>
<span>  plot <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  threads <span class="op">=</span> <span class="fu"><a href="getArchRThreads.html">getArchRThreads</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>  parallelParam <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  force <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  logFile <span class="op">=</span> <span class="fu"><a href="createLogFile.html">createLogFile</a></span><span class="op">(</span><span class="st">"addReproduciblePeakSet"</span><span class="op">)</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>ArchRProj</dt>
<dd><p>An <code>ArchRProject</code> object.</p></dd>


<dt>groupBy</dt>
<dd><p>The name of the column in <code>cellColData</code> to use for grouping cells together for peak calling.</p></dd>


<dt>peakMethod</dt>
<dd><p>The name of peak calling method to be used. Options include "Macs2" for using macs2 callpeak or "Tiles" for using a TileMatrix.</p></dd>


<dt>reproducibility</dt>
<dd><p>A string that indicates how peak reproducibility should be handled. This string is dynamic and can be a
function of <code>n</code> where <code>n</code> is the number of samples being assessed. For example, <code>reproducibility = "2"</code> means at least 2 samples
must have a peak call at this locus and <code>reproducibility = "(n+1)/2"</code> means that the majority of samples must have a peak call at this locus.</p></dd>


<dt>peaksPerCell</dt>
<dd><p>The upper limit of the number of peaks that can be identified per cell-grouping in <code>groupBy</code>. This is useful
for controlling how many peaks can be called from cell groups with low cell numbers.</p></dd>


<dt>maxPeaks</dt>
<dd><p>A numeric threshold for the maximum peaks to retain per group from <code>groupBy</code> in the union reproducible peak set.</p></dd>


<dt>minCells</dt>
<dd><p>The minimum allowable number of unique cells that was used to create the coverage files on which peaks are called.
This is important to allow for exclusion of pseudo-bulk replicates derived from very low cell numbers.</p></dd>


<dt>excludeChr</dt>
<dd><p>A character vector containing the <code>seqnames</code> of the chromosomes that should be excluded from peak calling.</p></dd>


<dt>pathToMacs2</dt>
<dd><p>The full path to the MACS2 executable.</p></dd>


<dt>genomeSize</dt>
<dd><p>The genome size to be used for MACS2 peak calling (see MACS2 documentation). This is required if genome is not hg19, hg38, mm9, or mm10.</p></dd>


<dt>shift</dt>
<dd><p>The number of basepairs to shift each Tn5 insertion. When combined with <code>extsize</code> this allows you to create proper fragments,
centered at the Tn5 insertion site, for use with MACS2 (see MACS2 documentation).</p></dd>


<dt>extsize</dt>
<dd><p>The number of basepairs to extend the MACS2 fragment after <code>shift</code> has been applied. When combined with <code>extsize</code> this
allows you to create proper fragments, centered at the Tn5 insertion site, for use with MACS2 (see MACS2 documentation).</p></dd>


<dt>method</dt>
<dd><p>The method to use for significance testing in MACS2. Options are "p" for p-value and "q" for q-value. When combined with
<code>cutOff</code> this gives the method and significance threshold for peak calling (see MACS2 documentation).</p></dd>


<dt>cutOff</dt>
<dd><p>The numeric significance cutOff for the testing method indicated by <code>method</code> (see MACS2 documentation).</p></dd>


<dt>additionalParams</dt>
<dd><p>A string of additional parameters to pass to MACS2 (see MACS2 documentation).</p></dd>


<dt>extendSummits</dt>
<dd><p>The number of basepairs to extend peak summits (in both directions) to obtain final fixed-width peaks. For example,
<code>extendSummits = 250</code> will create 501-bp fixed-width peaks from the 1-bp summits.</p></dd>


<dt>promoterRegion</dt>
<dd><p>A vector of two integers specifying the distance in basepairs upstream and downstream of a TSS to be included as a promoter region.
Peaks called within one of these regions will be annotated as a "promoter" peak. For example, <code>promoterRegion = c(2000, 100)</code> will annotate any peak within the region
2000 bp upstream and 100 bp downstream of a TSS as a "promoter" peak.</p></dd>


<dt>genomeAnnotation</dt>
<dd><p>The genomeAnnotation (see <code><a href="createGenomeAnnotation.html">createGenomeAnnotation()</a></code>) to be used for generating peak metadata such as nucleotide
information (GC content) or chromosome sizes.</p></dd>


<dt>geneAnnotation</dt>
<dd><p>The geneAnnotation (see <code><a href="createGeneAnnotation.html">createGeneAnnotation()</a></code>) to be used for labeling peaks as "promoter", "exonic", etc.</p></dd>


<dt>plot</dt>
<dd><p>A boolean describing whether to plot peak annotation results.</p></dd>


<dt>threads</dt>
<dd><p>The number of threads to be used for parallel computing.</p></dd>


<dt>parallelParam</dt>
<dd><p>A list of parameters to be passed for biocparallel/batchtools parallel computing.</p></dd>


<dt>force</dt>
<dd><p>A boolean value indicating whether to force the reproducible peak set to be overwritten if it already exist in the given <code>ArchRProject</code> peakSet.</p></dd>


<dt>verbose</dt>
<dd><p>A boolean value that determines whether standard output includes verbose sections.</p></dd>


<dt>logFile</dt>
<dd><p>The path to a file to be used for logging ArchR output.</p></dd>


<dt>...</dt>
<dd><p>Additional parameters to be pass to <code><a href="addGroupCoverages.html">addGroupCoverages()</a></code> to get sample-guided pseudobulk cell-groupings. Only used for TileMatrix-based
peak calling (not for MACS2). See <code><a href="addGroupCoverages.html">addGroupCoverages()</a></code> for more info.</p></dd>

</dl></div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span></span>
<span><span class="co"># Get Test ArchR Project</span></span>
<span><span class="va">proj</span> <span class="op">&lt;-</span> <span class="fu"><a href="getTestProject.html">getTestProject</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Add Peak Matrix Tiles</span></span>
<span><span class="va">proj</span> <span class="op">&lt;-</span> <span class="fu">addReproduciblePeakSet</span><span class="op">(</span><span class="va">proj</span>, peakMethod <span class="op">=</span> <span class="st">"tiles"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Add Peak Matrix Macs2 (Preferred)</span></span>
<span><span class="va">proj</span> <span class="op">&lt;-</span> <span class="fu">addReproduciblePeakSet</span><span class="op">(</span><span class="va">proj</span>, peakMethod <span class="op">=</span> <span class="st">"macs2"</span><span class="op">)</span></span>
<span></span></code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by <a href="https://twitter.com/JeffreyGranja" class="external-link">Jeffrey Granja</a>, <a href="https://twitter.com/doctorcorces" class="external-link">Ryan Corces</a>.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  
<script src="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.1/docsearch.min.js" integrity="sha256-GKvGqXDznoRYHCwKXGnuchvKSwmx9SRMrZOTh2g4Sb0=" crossorigin="anonymous"></script><script>
  docsearch({
    
    
    apiKey: '252c2c36fedc32b3ed803d0db4877b96',
    indexName: 'archr',
    inputSelector: 'input#search-input.form-control',
    transformData: function(hits) {
      return hits.map(function (hit) {
        hit.url = updateHitURL(hit);
        return hit;
      });
    }
  });
</script></body></html>

