<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Plot an ArchR Region Track — plotBrowserTrack • ArchR</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/yeti/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><!-- docsearch --><script src="../docsearch.js"></script><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.3/docsearch.min.css" integrity="sha256-QOSRU/ra9ActyXkIBbiIB144aDBdtvXBcNc3OTNuX/Q=" crossorigin="anonymous"><link href="../docsearch.css" rel="stylesheet"><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/jquery.mark.min.js" integrity="sha256-4HLtjeVgH0eIB3aZ9mLYF6E8oU5chNdjU6p6rrXpl9U=" crossorigin="anonymous"></script><link href="../extra.css" rel="stylesheet"><meta property="og:title" content="Plot an ArchR Region Track — plotBrowserTrack"><meta property="og:description" content="This function will plot the coverage at an input region in the style of a browser track. It allows for normalization of the signal
which enables direct comparison across samples."><meta property="og:image" content="https://greenleaflab.github.io/ArchR_Website/logo.svg"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-158769145-1"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-158769145-1');
</script></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">ArchR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.0.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../index.html">
    <span class="fa fa-home"></span>
     
  </a>
</li>
<li>
  <a href="../articles/Articles/tutorial.html">
    <span class="fa fa-graduation-cap"></span>
     
    Quick Tutorial
  </a>
</li>
<li>
  <a href="../bookdown/index.html">
    <span class="fa fa-file-text-o"></span>
     
    Full Manual
  </a>
</li>
<li>
  <a href="../reference/index.html">
    <span class="fa fa-code"></span>
     
    Functions
  </a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/GreenleafLab/ArchR" class="external-link">
    <span class="fa fa-github"></span>
     
    ArchR on GitHub
  </a>
</li>
      </ul><form class="navbar-form navbar-right hidden-xs hidden-sm" role="search">
        <div class="form-group">
          <input type="search" class="form-control" name="search-input" id="search-input" placeholder="Search..." aria-label="Search for..." autocomplete="off"></div>
      </form>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Plot an ArchR Region Track</h1>
    
    <div class="hidden name"><code>plotBrowserTrack.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>This function will plot the coverage at an input region in the style of a browser track. It allows for normalization of the signal
which enables direct comparison across samples.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="fu">plotBrowserTrack</span><span class="op">(</span>
  ArchRProj <span class="op">=</span> <span class="cn">NULL</span>,
  region <span class="op">=</span> <span class="cn">NULL</span>,
  groupBy <span class="op">=</span> <span class="st">"Clusters"</span>,
  useGroups <span class="op">=</span> <span class="cn">NULL</span>,
  plotSummary <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"bulkTrack"</span>, <span class="st">"featureTrack"</span>, <span class="st">"loopTrack"</span>, <span class="st">"geneTrack"</span><span class="op">)</span>,
  sizes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">10</span>, <span class="fl">1.5</span>, <span class="fl">3</span>, <span class="fl">4</span><span class="op">)</span>,
  features <span class="op">=</span> <span class="fu"><a href="getPeakSet.html">getPeakSet</a></span><span class="op">(</span><span class="va">ArchRProj</span><span class="op">)</span>,
  loops <span class="op">=</span> <span class="fu"><a href="getCoAccessibility.html">getCoAccessibility</a></span><span class="op">(</span><span class="va">ArchRProj</span><span class="op">)</span>,
  geneSymbol <span class="op">=</span> <span class="cn">NULL</span>,
  useMatrix <span class="op">=</span> <span class="cn">NULL</span>,
  log2Norm <span class="op">=</span> <span class="cn">TRUE</span>,
  upstream <span class="op">=</span> <span class="fl">50000</span>,
  downstream <span class="op">=</span> <span class="fl">50000</span>,
  tileSize <span class="op">=</span> <span class="fl">250</span>,
  minCells <span class="op">=</span> <span class="fl">25</span>,
  normMethod <span class="op">=</span> <span class="st">"ReadsInTSS"</span>,
  threads <span class="op">=</span> <span class="fu"><a href="getArchRThreads.html">getArchRThreads</a></span><span class="op">(</span><span class="op">)</span>,
  ylim <span class="op">=</span> <span class="cn">NULL</span>,
  pal <span class="op">=</span> <span class="cn">NULL</span>,
  baseSize <span class="op">=</span> <span class="fl">7</span>,
  scTileSize <span class="op">=</span> <span class="fl">0.5</span>,
  scCellsMax <span class="op">=</span> <span class="fl">100</span>,
  borderWidth <span class="op">=</span> <span class="fl">0.4</span>,
  tickWidth <span class="op">=</span> <span class="fl">0.4</span>,
  facetbaseSize <span class="op">=</span> <span class="fl">7</span>,
  geneAnnotation <span class="op">=</span> <span class="fu"><a href="getGeneAnnotation.html">getGeneAnnotation</a></span><span class="op">(</span><span class="va">ArchRProj</span><span class="op">)</span>,
  title <span class="op">=</span> <span class="st">""</span>,
  verbose <span class="op">=</span> <span class="cn">TRUE</span>,
  logFile <span class="op">=</span> <span class="fu"><a href="createLogFile.html">createLogFile</a></span><span class="op">(</span><span class="st">"plotBrowserTrack"</span><span class="op">)</span>
<span class="op">)</span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>ArchRProj</dt>
<dd><p>An <code>ArchRProject</code> object.</p></dd>
<dt>region</dt>
<dd><p>A <code>GRanges</code> region that indicates the region to be plotted. If more than one region exists in the <code>GRanges</code> object,
all will be plotted. If no region is supplied, then the <code>geneSymbol</code> argument can be used to center the plot window at the
transcription start site of the supplied gene.</p></dd>
<dt>groupBy</dt>
<dd><p>A string that indicates how cells should be grouped. This string corresponds to one of the standard or
user-supplied <code>cellColData</code> metadata columns (for example, "Clusters"). Cells with the same value annotated in this metadata
column will be grouped together and the average signal will be plotted.</p></dd>
<dt>useGroups</dt>
<dd><p>A character vector that is used to select a subset of groups by name from the designated <code>groupBy</code> column in
<code>cellColData</code>. This limits the groups to be plotted.</p></dd>
<dt>plotSummary</dt>
<dd><p>A character vector containing the features to be potted. Possible values include "bulkTrack" (the ATAC-seq signal),
"scTrack" (scATAC-seq signal), "featureTrack" (i.e. the peak regions), "geneTrack" (line diagrams of genes with introns and exons shown.
Blue-colored genes are on the minus strand and red-colored genes are on the plus strand), and "loopTrack" (links between a peak and a gene).</p></dd>
<dt>sizes</dt>
<dd><p>A numeric vector containing up to 3 values that indicate the sizes of the individual components passed to <code>plotSummary</code>.
The order must be the same as <code>plotSummary</code>.</p></dd>
<dt>features</dt>
<dd><p>A <code>GRanges</code> object containing the "features" to be plotted via the "featureTrack". This should be thought of as a
bed track. i.e. the set of peaks obtained using <code>getPeakSet(ArchRProj))</code>.</p></dd>
<dt>loops</dt>
<dd><p>A <code>GRanges</code> object containing the "loops" to be plotted via the "loopTrack".
This <code>GRanges</code> object start represents the center position of one loop anchor and the end represents the center position of another loop anchor.
A "loopTrack" draws an arc between two genomic regions that show some type of interaction. This type of track can be used
to display chromosome conformation capture data or co-accessibility links obtained using <code><a href="getCoAccessibility.html">getCoAccessibility()</a></code>.</p></dd>
<dt>geneSymbol</dt>
<dd><p>If <code>region</code> is not supplied, plotting can be centered at the transcription start site corresponding to the gene symbol(s) passed here.</p></dd>
<dt>useMatrix</dt>
<dd><p>If supplied geneSymbol, one can plot the corresponding GeneScores/GeneExpression within this matrix. I.E. "GeneScoreMatrix"</p></dd>
<dt>log2Norm</dt>
<dd><p>If supplied geneSymbol, Log2 normalize the corresponding GeneScores/GeneExpression matrix before plotting.</p></dd>
<dt>upstream</dt>
<dd><p>The number of basepairs upstream of the transcription start site of <code>geneSymbol</code> to extend the plotting window.
If <code>region</code> is supplied, this argument is ignored.</p></dd>
<dt>downstream</dt>
<dd><p>The number of basepairs downstream of the transcription start site of <code>geneSymbol</code> to extend the plotting window.
If <code>region</code> is supplied, this argument is ignored.</p></dd>
<dt>tileSize</dt>
<dd><p>The numeric width of the tile/bin in basepairs for plotting ATAC-seq signal tracks. All insertions in a single bin will be summed.</p></dd>
<dt>minCells</dt>
<dd><p>The minimum number of cells contained within a cell group to allow for this cell group to be plotted. This argument can be
used to exclude pseudo-bulk replicates generated from low numbers of cells.</p></dd>
<dt>normMethod</dt>
<dd><p>The name of the column in <code>cellColData</code> by which normalization should be performed. The recommended and default value
is "ReadsInTSS" which simultaneously normalizes tracks based on sequencing depth and sample data quality.</p></dd>
<dt>threads</dt>
<dd><p>The number of threads to use for parallel execution.</p></dd>
<dt>ylim</dt>
<dd><p>The numeric quantile y-axis limit to be used for for "bulkTrack" plotting. If not provided, the y-axis limit will be c(0, 0.999).</p></dd>
<dt>pal</dt>
<dd><p>A custom palette (see <code>paletteDiscrete</code> or <code>ArchRPalettes</code>) used to override coloring for groups.</p></dd>
<dt>baseSize</dt>
<dd><p>The numeric font size to be used in the plot. This applies to all plot labels.</p></dd>
<dt>scTileSize</dt>
<dd><p>The width of the tiles in scTracks. Larger numbers may make cells overlap more. Default is 0.5 for about 100 cells.</p></dd>
<dt>scCellsMax</dt>
<dd><p>The maximum number of cells for scTracks.</p></dd>
<dt>borderWidth</dt>
<dd><p>The numeric line width to be used for plot borders.</p></dd>
<dt>tickWidth</dt>
<dd><p>The numeric line width to be used for axis tick marks.</p></dd>
<dt>facetbaseSize</dt>
<dd><p>The numeric font size to be used in the facets (gray boxes used to provide track labels) of the plot.</p></dd>
<dt>geneAnnotation</dt>
<dd><p>The <code>geneAnnotation</code> object to be used for plotting the "geneTrack" object. See <code><a href="createGeneAnnotation.html">createGeneAnnotation()</a></code> for more info.</p></dd>
<dt>title</dt>
<dd><p>The title to add at the top of the plot next to the plot's genomic coordinates.</p></dd>
<dt>verbose</dt>
<dd><p>A boolean value that determines whether standard output should be printed.</p></dd>
<dt>logFile</dt>
<dd><p>The path to a file to be used for logging ArchR output.</p></dd>
</dl></div>

  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by <a href="https://twitter.com/JeffreyGranja" class="external-link">Jeffrey Granja</a>, <a href="https://twitter.com/doctorcorces" class="external-link">Ryan Corces</a>.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.1.</p>
</div>

      </footer></div>

  
<script src="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.1/docsearch.min.js" integrity="sha256-GKvGqXDznoRYHCwKXGnuchvKSwmx9SRMrZOTh2g4Sb0=" crossorigin="anonymous"></script><script>
  docsearch({
    
    
    apiKey: '252c2c36fedc32b3ed803d0db4877b96',
    indexName: 'archr',
    inputSelector: 'input#search-input.form-control',
    transformData: function(hits) {
      return hits.map(function (hit) {
        hit.url = updateHitURL(hit);
        return hit;
      });
    }
  });
</script></body></html>

